{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 1. Compute Results Overview Tables"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The success of an assembly is evaluated by the computation of metrics in two defined ways: globally through statistics inherent to the complete set of sequences that were assembled, and relative to the replicons present in the sample. \n",
    "\n",
    "The computation of the global metrics is done through custom python code from the complete set of sequences assembled. \n",
    "\n",
    "The following metrics are computed for the complete and filtered set of assembled sequences, restricted to contigs of length above a specified minimum size: \n",
    "\n",
    "- **Contig sizes**\n",
    "    - **Contigs:** The total number of contigs in the assembly;\n",
    "    - **Basepairs:** The total number of bases in the assembly;\n",
    "    - **Maximum sequence length:** The length of the largest contig in the assembly.\n",
    "    - **Number of ‘N’s:** Number of uncalled bases (N's) \n",
    "- **Contiguity**\n",
    "    - **Nx (where 0  < x  ⩽ 100):** Length for which the collection of all assembled sequences of that length or longer in an assembly covers at least a given percentage of the total length of the assembly\n",
    "- **Misassembly**\n",
    "    - **Misassemblies** - Number of aligned contigs that contain a misassembly event\n",
    "\n",
    "After filtering, the sequences are mapped with the reference replicons and the metrics are computed through custom python code. The metrics are calculated, filtered for a minimum length, for each replicon in the file provided input references. \n",
    "\n",
    "- **Contig sizes**\n",
    "    - **Contigs:** The total number of contigs in the assembly;\n",
    "    - **Basepairs:** The total number of bases in the assembly;\n",
    "    - **Number of ‘N’s:** Number of uncalled bases (N's) \n",
    "- **COMPASS**\n",
    "    - **(Breadth of) Coverage:** Ratio of covered sequence on the reference by aligned contigs;\n",
    "    - **Multiplicity:** Ratio of the length of alignable assembled sequence to covered sequence on the reference;\n",
    "    - **Validity:** Ratio of the length of the alignable assembled sequence to total basepairs in the aligned contigs;\n",
    "    - **Parsimony:** Cost of the assembly (multiplicity over validity);\n",
    "- **Contiguity**\n",
    "    - **Contiguity:** longest single alignment between the assembly and the reference, relative to the reference length;\n",
    "    - **NAx (where 0  < x  ⩽ 100):** Length for which the collection of aligned assembled sequences of that length or longer in an assembly covers at least a given percentage of the total length of the reference replicon;\n",
    "    - **NGx (where 0  < x  ⩽ 100):** Length for which the collection of aligned contigs of that length or longer covers at least a given percentage of the sequence of the reference.\n",
    "    - **Lx (where 0  < x  ⩽ 100):** Minimal number of contigs that cover x % of the sequence of the reference;\n",
    "- **Identity**\n",
    "    - **Identity:** Ratio of identical basepairs in all aligned contigs to the reference;\n",
    "    - **Lowest identity:** Identity of the lowest scoring contig to the reference.\n",
    "    - **PLS**: Phred-like score per contig, per assembler.\n",
    "- **Misassembly**\n",
    "    - **Misassemblies** - Number of aligned contigs that contain a misassembly event\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "from plotly.offline import plot\n",
    "import glob\n",
    "import fnmatch\n",
    "import plotly.graph_objects as go\n",
    "from plotly.subplots import make_subplots\n",
    "import json\n",
    "import pandas as pd\n",
    "from itertools import groupby\n",
    "import csv\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Global variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "ASSEMBLER_PROCESS_LIST = [\"ABYSS\", \"BCALM2\", \"GATBMINIAPIPELINE\", \"METAHIPMER2\", \"MINIA\", \"MEGAHIT\", \"METASPADES\", \"UNICYCLER\", \"SPADES\",\n",
    "                          \"SKESA\", \"VELVETOPTIMIZER\", \"IDBA\"]\n",
    "PROCESS_TO_NAME = {\"ABYSS\": \"AbYSS\",\n",
    "                   \"BCALM2\": \"BCALM2\", \n",
    "                   \"GATBMINIAPIPELINE\": \"GATBMiniaPipeline\",\n",
    "                   \"METAHIPMER2\": \"MetaHipMer2\",\n",
    "                   \"MINIA\": \"MINIA\",\n",
    "                   \"MEGAHIT\": \"MEGAHIT\", \n",
    "                   \"METASPADES\": \"metaSPAdes\", \n",
    "                   \"UNICYCLER\": \"Unicycler\", \n",
    "                   \"SPADES\": \"SPAdes\",\n",
    "                    \"SKESA\": \"SKESA\",\n",
    "                   \"VELVETOPTIMIZER\": \"VelvetOptimizer\",\n",
    "                   \"IDBA\": \"IDBA-UD\"}\n",
    "\n",
    "log_distributed = ['LHS', 'LNN', 'ERR2935805']\n",
    "even_distribution = ['EMS', 'LNN', 'ERR2984773']\n",
    "\n",
    "genomic_assemblers = ['AbYSS', 'BCALM2', 'MINIA', 'SKESA', 'SPAdes', 'Unicycler', 'VelvetOptimizer']\n",
    "metagenomic_assemblers = ['MetaHipMer2','GATBMiniaPipeline', 'IDBA-UD', 'MEGAHIT', 'metaSPAdes']\n",
    "single_kmer = ['BCALM2', 'MINIA', 'AbYSS']\n",
    "multiple_kmer = ['SKESA', 'SPAdes', 'Unicycler', 'VelvetOptimizer', 'GATBMiniaPipeline', \n",
    "                 'IDBA-UD', 'MEGAHIT', 'metaSPAdes', 'MetaHipMer2']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Global metrics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pipeline_report_tables.json data from run2...\n",
      "Processing pipeline_report_tables.json data from run1...\n",
      "Processing pipeline_report_tables.json data from run3...\n",
      "Processing pipeline_report_tables.json data from run2...\n"
     ]
    }
   ],
   "source": [
    "report_glob = glob.glob('../Results/*/*/report/pipeline_report_tables.json')\n",
    "global_pipeline_metrics_df = pd.DataFrame()\n",
    "\n",
    "for pipeline_report_file in report_glob:\n",
    "    report_file_name = pipeline_report_file.split('/')[-1]\n",
    "    stats_run = pipeline_report_file.split('/')[-3]\n",
    "    print('Processing {0} data from {1}...'.format(report_file_name, stats_run))\n",
    "    \n",
    "    with open(pipeline_report_file) as _fh:\n",
    "        json_report = json.load(_fh)\n",
    "        for sample in json_report.keys():\n",
    "            for line in json_report[sample]['GlobalTable']:\n",
    "                assembler = line['assembler']\n",
    "                global_pipeline_metrics_df = global_pipeline_metrics_df.append({'run': stats_run,\n",
    "                                                                                'sample': sample,\n",
    "                                                                                'assembler': line['assembler'],\n",
    "                                                                                'contigs': int(line['original']['contigs']),\n",
    "                                                                                'basepairs': int(line['original']['basepairs']),\n",
    "                                                                                'max_contig': int(line['original']['max_contig_size']),\n",
    "                                                                                'n50': int(line['original']['N50']),\n",
    "                                                                                'mapped_reads': line['original']['mapped_reads'],\n",
    "                                                                                'Ns': int(line['original']['Ns']),\n",
    "                                                                                'misassembled contigs': line['filtered']['misassembled_contigs'],\n",
    "                                                                                'misassembly events': line['filtered']['misassembly_events'],\n",
    "                                                                                'filtered_contigs': line['filtered']['contigs'],\n",
    "                                                                                'filtered_basepairs': line['filtered']['basepairs'],\n",
    "                                                                                'filtered_n50': line['filtered']['N50'],\n",
    "                                                                                'filtered_Ns': line['filtered']['Ns'],\n",
    "                                                                                'filtered_mapped_reads': line['filtered']['mapped_reads'],\n",
    "                                                                               },\n",
    "                                                                               ignore_index=True)\n",
    "global_pipeline_metrics_df['distribution'] = np.where(global_pipeline_metrics_df['sample'].isin(log_distributed), 'Log', 'Even')\n",
    "global_pipeline_metrics_df['type'] = np.where(global_pipeline_metrics_df['assembler'].isin(genomic_assemblers), 'Genomic', 'Metagenomic')\n",
    "global_pipeline_metrics_df['algorythm'] = np.where(global_pipeline_metrics_df['assembler'].isin(single_kmer), 'Single k-mer De Bruijn graph', 'Multiple k-mer De Bruijn graph')\n",
    "global_pipeline_metrics_df[['contigs','basepairs','max_contig', 'Ns','n50', 'filtered_n50','misassembled contigs', 'misassembly events']] = global_pipeline_metrics_df[['contigs','basepairs','max_contig', 'Ns','n50','filtered_n50','misassembled contigs', 'misassembly events']].apply(pd.to_numeric)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Preview data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Ns                       float64\n",
       "assembler                 object\n",
       "basepairs                float64\n",
       "contigs                  float64\n",
       "filtered_Ns              float64\n",
       "filtered_basepairs       float64\n",
       "filtered_contigs         float64\n",
       "filtered_mapped_reads    float64\n",
       "filtered_n50             float64\n",
       "mapped_reads             float64\n",
       "max_contig               float64\n",
       "misassembled contigs     float64\n",
       "misassembly events       float64\n",
       "n50                      float64\n",
       "run                       object\n",
       "sample                    object\n",
       "distribution              object\n",
       "type                      object\n",
       "algorythm                 object\n",
       "dtype: object"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "global_pipeline_metrics_df.dtypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ns</th>\n",
       "      <th>assembler</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>contigs</th>\n",
       "      <th>filtered_Ns</th>\n",
       "      <th>filtered_basepairs</th>\n",
       "      <th>filtered_contigs</th>\n",
       "      <th>filtered_mapped_reads</th>\n",
       "      <th>filtered_n50</th>\n",
       "      <th>mapped_reads</th>\n",
       "      <th>max_contig</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>n50</th>\n",
       "      <th>run</th>\n",
       "      <th>sample</th>\n",
       "      <th>distribution</th>\n",
       "      <th>type</th>\n",
       "      <th>algorythm</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9.0</td>\n",
       "      <td>ABySS</td>\n",
       "      <td>78728514.0</td>\n",
       "      <td>677859.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3150980.0</td>\n",
       "      <td>356.0</td>\n",
       "      <td>79.208904</td>\n",
       "      <td>18568.0</td>\n",
       "      <td>297.914963</td>\n",
       "      <td>88665.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>101.0</td>\n",
       "      <td>run2</td>\n",
       "      <td>ERR2935805</td>\n",
       "      <td>Log</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>BCALM2</td>\n",
       "      <td>315567452.0</td>\n",
       "      <td>8035706.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>657074.0</td>\n",
       "      <td>504.0</td>\n",
       "      <td>0.104039</td>\n",
       "      <td>1262.0</td>\n",
       "      <td>1.435470</td>\n",
       "      <td>3093.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>run2</td>\n",
       "      <td>ERR2935805</td>\n",
       "      <td>Log</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.0</td>\n",
       "      <td>GATBMiniaPipeline</td>\n",
       "      <td>17126542.0</td>\n",
       "      <td>6777.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>13891396.0</td>\n",
       "      <td>603.0</td>\n",
       "      <td>89.111413</td>\n",
       "      <td>90095.0</td>\n",
       "      <td>89.676706</td>\n",
       "      <td>542997.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>55075.0</td>\n",
       "      <td>run2</td>\n",
       "      <td>ERR2935805</td>\n",
       "      <td>Log</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.0</td>\n",
       "      <td>IDBA-UD</td>\n",
       "      <td>23587129.0</td>\n",
       "      <td>14155.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>17534580.0</td>\n",
       "      <td>2725.0</td>\n",
       "      <td>89.451482</td>\n",
       "      <td>113853.0</td>\n",
       "      <td>90.207665</td>\n",
       "      <td>632213.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>51161.0</td>\n",
       "      <td>run2</td>\n",
       "      <td>ERR2935805</td>\n",
       "      <td>Log</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.0</td>\n",
       "      <td>MEGAHIT</td>\n",
       "      <td>24498566.0</td>\n",
       "      <td>14172.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18552996.0</td>\n",
       "      <td>3044.0</td>\n",
       "      <td>93.405416</td>\n",
       "      <td>210993.0</td>\n",
       "      <td>94.140386</td>\n",
       "      <td>768410.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>82249.0</td>\n",
       "      <td>run2</td>\n",
       "      <td>ERR2935805</td>\n",
       "      <td>Log</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>168</th>\n",
       "      <td>0.0</td>\n",
       "      <td>MINIA</td>\n",
       "      <td>30648944.0</td>\n",
       "      <td>16748.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28218335.0</td>\n",
       "      <td>3574.0</td>\n",
       "      <td>92.195377</td>\n",
       "      <td>20782.0</td>\n",
       "      <td>95.964771</td>\n",
       "      <td>161812.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>17967.0</td>\n",
       "      <td>run2</td>\n",
       "      <td>EMS</td>\n",
       "      <td>Even</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169</th>\n",
       "      <td>0.0</td>\n",
       "      <td>SKESA</td>\n",
       "      <td>30249132.0</td>\n",
       "      <td>1269.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30064811.0</td>\n",
       "      <td>891.0</td>\n",
       "      <td>97.442508</td>\n",
       "      <td>74138.0</td>\n",
       "      <td>98.249020</td>\n",
       "      <td>632013.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>73839.0</td>\n",
       "      <td>run2</td>\n",
       "      <td>EMS</td>\n",
       "      <td>Even</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170</th>\n",
       "      <td>0.0</td>\n",
       "      <td>SPAdes</td>\n",
       "      <td>30664883.0</td>\n",
       "      <td>2628.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30354786.0</td>\n",
       "      <td>412.0</td>\n",
       "      <td>98.665524</td>\n",
       "      <td>208991.0</td>\n",
       "      <td>99.428625</td>\n",
       "      <td>850176.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>206515.0</td>\n",
       "      <td>run2</td>\n",
       "      <td>EMS</td>\n",
       "      <td>Even</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171</th>\n",
       "      <td>0.0</td>\n",
       "      <td>Unicycler</td>\n",
       "      <td>30301867.0</td>\n",
       "      <td>656.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30188943.0</td>\n",
       "      <td>334.0</td>\n",
       "      <td>98.048427</td>\n",
       "      <td>245050.0</td>\n",
       "      <td>99.071193</td>\n",
       "      <td>1303875.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>245050.0</td>\n",
       "      <td>run2</td>\n",
       "      <td>EMS</td>\n",
       "      <td>Even</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>172</th>\n",
       "      <td>4383.0</td>\n",
       "      <td>VelvetOptimiser</td>\n",
       "      <td>30526291.0</td>\n",
       "      <td>11069.0</td>\n",
       "      <td>4383.0</td>\n",
       "      <td>28467054.0</td>\n",
       "      <td>3035.0</td>\n",
       "      <td>92.274221</td>\n",
       "      <td>33291.0</td>\n",
       "      <td>95.893941</td>\n",
       "      <td>847777.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>29176.0</td>\n",
       "      <td>run2</td>\n",
       "      <td>EMS</td>\n",
       "      <td>Even</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>173 rows × 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Ns          assembler    basepairs    contigs  filtered_Ns  \\\n",
       "0       9.0              ABySS   78728514.0   677859.0          0.0   \n",
       "1       0.0             BCALM2  315567452.0  8035706.0          0.0   \n",
       "2       0.0  GATBMiniaPipeline   17126542.0     6777.0          0.0   \n",
       "3       0.0            IDBA-UD   23587129.0    14155.0          0.0   \n",
       "4       0.0            MEGAHIT   24498566.0    14172.0          0.0   \n",
       "..      ...                ...          ...        ...          ...   \n",
       "168     0.0              MINIA   30648944.0    16748.0          0.0   \n",
       "169     0.0              SKESA   30249132.0     1269.0          0.0   \n",
       "170     0.0             SPAdes   30664883.0     2628.0          0.0   \n",
       "171     0.0          Unicycler   30301867.0      656.0          0.0   \n",
       "172  4383.0    VelvetOptimiser   30526291.0    11069.0       4383.0   \n",
       "\n",
       "     filtered_basepairs  filtered_contigs  filtered_mapped_reads  \\\n",
       "0             3150980.0             356.0              79.208904   \n",
       "1              657074.0             504.0               0.104039   \n",
       "2            13891396.0             603.0              89.111413   \n",
       "3            17534580.0            2725.0              89.451482   \n",
       "4            18552996.0            3044.0              93.405416   \n",
       "..                  ...               ...                    ...   \n",
       "168          28218335.0            3574.0              92.195377   \n",
       "169          30064811.0             891.0              97.442508   \n",
       "170          30354786.0             412.0              98.665524   \n",
       "171          30188943.0             334.0              98.048427   \n",
       "172          28467054.0            3035.0              92.274221   \n",
       "\n",
       "     filtered_n50  mapped_reads  max_contig  misassembled contigs  \\\n",
       "0         18568.0    297.914963     88665.0                   0.0   \n",
       "1          1262.0      1.435470      3093.0                   0.0   \n",
       "2         90095.0     89.676706    542997.0                   0.0   \n",
       "3        113853.0     90.207665    632213.0                   3.0   \n",
       "4        210993.0     94.140386    768410.0                   1.0   \n",
       "..            ...           ...         ...                   ...   \n",
       "168       20782.0     95.964771    161812.0                   0.0   \n",
       "169       74138.0     98.249020    632013.0                   9.0   \n",
       "170      208991.0     99.428625    850176.0                   0.0   \n",
       "171      245050.0     99.071193   1303875.0                   0.0   \n",
       "172       33291.0     95.893941    847777.0                   6.0   \n",
       "\n",
       "     misassembly events       n50   run      sample distribution         type  \\\n",
       "0                   0.0     101.0  run2  ERR2935805          Log  Metagenomic   \n",
       "1                   0.0      35.0  run2  ERR2935805          Log      Genomic   \n",
       "2                   0.0   55075.0  run2  ERR2935805          Log  Metagenomic   \n",
       "3                   5.0   51161.0  run2  ERR2935805          Log  Metagenomic   \n",
       "4                   2.0   82249.0  run2  ERR2935805          Log  Metagenomic   \n",
       "..                  ...       ...   ...         ...          ...          ...   \n",
       "168                 0.0   17967.0  run2         EMS         Even      Genomic   \n",
       "169                18.0   73839.0  run2         EMS         Even      Genomic   \n",
       "170                 0.0  206515.0  run2         EMS         Even      Genomic   \n",
       "171                 0.0  245050.0  run2         EMS         Even      Genomic   \n",
       "172                11.0   29176.0  run2         EMS         Even  Metagenomic   \n",
       "\n",
       "                          algorythm  \n",
       "0    Multiple k-mer De Bruijn graph  \n",
       "1      Single k-mer De Bruijn graph  \n",
       "2    Multiple k-mer De Bruijn graph  \n",
       "3    Multiple k-mer De Bruijn graph  \n",
       "4    Multiple k-mer De Bruijn graph  \n",
       "..                              ...  \n",
       "168    Single k-mer De Bruijn graph  \n",
       "169  Multiple k-mer De Bruijn graph  \n",
       "170  Multiple k-mer De Bruijn graph  \n",
       "171  Multiple k-mer De Bruijn graph  \n",
       "172  Multiple k-mer De Bruijn graph  \n",
       "\n",
       "[173 rows x 19 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "global_pipeline_metrics_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Save data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "global_pipeline_metrics_df.to_csv(\"Tables/Results/Global metrics.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Global statistics per assembler type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "stats_per_assembler_type = pd.DataFrame()\n",
    "for assembly_type in global_pipeline_metrics_df.type.unique():\n",
    "    row={'Assembler Type': assembly_type}\n",
    "    lala=global_pipeline_metrics_df[(global_pipeline_metrics_df.type == assembly_type) & (global_pipeline_metrics_df.algorythm == \"Multiple k-mer De Bruijn graph\")].describe()\n",
    "    for column in lala.columns:\n",
    "        mean = lala.loc['mean',column]\n",
    "        minimum = lala.loc['min',column]\n",
    "        maximum = lala.loc['max',column]\n",
    "        row[column] = \"{} [{};{}]\".format(round(mean, 2), int(minimum), int(maximum))\n",
    "    stats_per_assembler_type=stats_per_assembler_type.append(row, ignore_index=True)\n",
    "stats_per_assembler_type = stats_per_assembler_type.set_index('Assembler Type')\n",
    "stats_per_assembler_type.to_csv(\"Tables/Results/Global metrics per assembler type multiple kmer.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ns</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>contigs</th>\n",
       "      <th>filtered_Ns</th>\n",
       "      <th>filtered_basepairs</th>\n",
       "      <th>filtered_contigs</th>\n",
       "      <th>filtered_mapped_reads</th>\n",
       "      <th>filtered_n50</th>\n",
       "      <th>mapped_reads</th>\n",
       "      <th>max_contig</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>n50</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler Type</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Metagenomic</th>\n",
       "      <td>1908.27 [0;124182]</td>\n",
       "      <td>30432237.02 [13174826;91720949]</td>\n",
       "      <td>65345.54 [640;677860]</td>\n",
       "      <td>611.43 [0;12292]</td>\n",
       "      <td>21994215.34 [13883;32021766]</td>\n",
       "      <td>1257.99 [13;5044]</td>\n",
       "      <td>84.91 [0;107]</td>\n",
       "      <td>104054.6 [1039;239185]</td>\n",
       "      <td>91.92 [1;297]</td>\n",
       "      <td>601949.54 [1157;1205839]</td>\n",
       "      <td>3.57 [0;18]</td>\n",
       "      <td>6.93 [0;36]</td>\n",
       "      <td>86314.7 [85;238611]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic</th>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>24466471.62 [2957060;46535940]</td>\n",
       "      <td>7561.62 [36;39819]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>22453968.89 [2950393;32500826]</td>\n",
       "      <td>1019.96 [21;4999]</td>\n",
       "      <td>90.33 [62;99]</td>\n",
       "      <td>148648.44 [2148;351027]</td>\n",
       "      <td>94.46 [78;107]</td>\n",
       "      <td>701726.47 [29155;1303875]</td>\n",
       "      <td>28.42 [0;572]</td>\n",
       "      <td>56.91 [0;1148]</td>\n",
       "      <td>140639.09 [1155;351027]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                Ns                        basepairs  \\\n",
       "Assembler Type                                                        \n",
       "Metagenomic     1908.27 [0;124182]  30432237.02 [13174826;91720949]   \n",
       "Genomic                  0.0 [0;0]   24466471.62 [2957060;46535940]   \n",
       "\n",
       "                              contigs       filtered_Ns  \\\n",
       "Assembler Type                                            \n",
       "Metagenomic     65345.54 [640;677860]  611.43 [0;12292]   \n",
       "Genomic            7561.62 [36;39819]         0.0 [0;0]   \n",
       "\n",
       "                            filtered_basepairs   filtered_contigs  \\\n",
       "Assembler Type                                                      \n",
       "Metagenomic       21994215.34 [13883;32021766]  1257.99 [13;5044]   \n",
       "Genomic         22453968.89 [2950393;32500826]  1019.96 [21;4999]   \n",
       "\n",
       "               filtered_mapped_reads             filtered_n50    mapped_reads  \\\n",
       "Assembler Type                                                                  \n",
       "Metagenomic            84.91 [0;107]   104054.6 [1039;239185]   91.92 [1;297]   \n",
       "Genomic                90.33 [62;99]  148648.44 [2148;351027]  94.46 [78;107]   \n",
       "\n",
       "                               max_contig misassembled contigs  \\\n",
       "Assembler Type                                                   \n",
       "Metagenomic      601949.54 [1157;1205839]          3.57 [0;18]   \n",
       "Genomic         701726.47 [29155;1303875]        28.42 [0;572]   \n",
       "\n",
       "               misassembly events                      n50  \n",
       "Assembler Type                                              \n",
       "Metagenomic           6.93 [0;36]      86314.7 [85;238611]  \n",
       "Genomic            56.91 [0;1148]  140639.09 [1155;351027]  "
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "stats_per_assembler_type"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Global stats per assembly algorithm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "stats_per_assembler_algo = pd.DataFrame()\n",
    "for assembly_algo in global_pipeline_metrics_df.algorythm.unique():\n",
    "    row={'Assembler Algorithm': assembly_algo}\n",
    "    lala=global_pipeline_metrics_df[global_pipeline_metrics_df.algorythm == assembly_algo].describe()\n",
    "    for column in lala.columns:\n",
    "        mean = lala.loc['mean',column]\n",
    "        minimum = lala.loc['min',column]\n",
    "        maximum = lala.loc['max',column]\n",
    "        row[column] = \"{} [{};{}]\".format(round(mean, 2), int(minimum), int(maximum))\n",
    "    stats_per_assembler_algo=stats_per_assembler_algo.append(row, ignore_index=True)\n",
    "stats_per_assembler_algo = stats_per_assembler_algo.set_index('Assembler Algorithm')\n",
    "stats_per_assembler_algo.to_csv(\"Tables/Results/Global metrics per assembler algorithm.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ns</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>contigs</th>\n",
       "      <th>filtered_Ns</th>\n",
       "      <th>filtered_basepairs</th>\n",
       "      <th>filtered_contigs</th>\n",
       "      <th>filtered_mapped_reads</th>\n",
       "      <th>filtered_n50</th>\n",
       "      <th>mapped_reads</th>\n",
       "      <th>max_contig</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>n50</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler Algorithm</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Multiple k-mer De Bruijn graph</th>\n",
       "      <td>1307.76 [0;124182]</td>\n",
       "      <td>28554898.26 [2957060;91720949]</td>\n",
       "      <td>47161.79 [36;677860]</td>\n",
       "      <td>419.02 [0;12292]</td>\n",
       "      <td>22138893.03 [13883;32500826]</td>\n",
       "      <td>1183.08 [13;5044]</td>\n",
       "      <td>86.62 [0;107]</td>\n",
       "      <td>118087.63 [1039;351027]</td>\n",
       "      <td>92.72 [1;297]</td>\n",
       "      <td>633347.87 [1157;1303875]</td>\n",
       "      <td>11.39 [0;572]</td>\n",
       "      <td>22.66 [0;1148]</td>\n",
       "      <td>103409.79 [85;351027]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Single k-mer De Bruijn graph</th>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>84513797.23 [15334712;364761201]</td>\n",
       "      <td>1477579.7 [14997;8175890]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>16524236.0 [0;28279162]</td>\n",
       "      <td>2476.57 [0;6250]</td>\n",
       "      <td>62.0 [0;97]</td>\n",
       "      <td>14508.9 [0;33550]</td>\n",
       "      <td>70.14 [1;99]</td>\n",
       "      <td>109037.73 [994;177038]</td>\n",
       "      <td>0.27 [0;2]</td>\n",
       "      <td>0.47 [0;4]</td>\n",
       "      <td>10128.13 [35;26474]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                Ns  \\\n",
       "Assembler Algorithm                                  \n",
       "Multiple k-mer De Bruijn graph  1307.76 [0;124182]   \n",
       "Single k-mer De Bruijn graph             0.0 [0;0]   \n",
       "\n",
       "                                                       basepairs  \\\n",
       "Assembler Algorithm                                                \n",
       "Multiple k-mer De Bruijn graph    28554898.26 [2957060;91720949]   \n",
       "Single k-mer De Bruijn graph    84513797.23 [15334712;364761201]   \n",
       "\n",
       "                                                  contigs       filtered_Ns  \\\n",
       "Assembler Algorithm                                                           \n",
       "Multiple k-mer De Bruijn graph       47161.79 [36;677860]  419.02 [0;12292]   \n",
       "Single k-mer De Bruijn graph    1477579.7 [14997;8175890]         0.0 [0;0]   \n",
       "\n",
       "                                          filtered_basepairs  \\\n",
       "Assembler Algorithm                                            \n",
       "Multiple k-mer De Bruijn graph  22138893.03 [13883;32500826]   \n",
       "Single k-mer De Bruijn graph         16524236.0 [0;28279162]   \n",
       "\n",
       "                                 filtered_contigs filtered_mapped_reads  \\\n",
       "Assembler Algorithm                                                       \n",
       "Multiple k-mer De Bruijn graph  1183.08 [13;5044]         86.62 [0;107]   \n",
       "Single k-mer De Bruijn graph     2476.57 [0;6250]           62.0 [0;97]   \n",
       "\n",
       "                                           filtered_n50   mapped_reads  \\\n",
       "Assembler Algorithm                                                      \n",
       "Multiple k-mer De Bruijn graph  118087.63 [1039;351027]  92.72 [1;297]   \n",
       "Single k-mer De Bruijn graph          14508.9 [0;33550]   70.14 [1;99]   \n",
       "\n",
       "                                              max_contig misassembled contigs  \\\n",
       "Assembler Algorithm                                                             \n",
       "Multiple k-mer De Bruijn graph  633347.87 [1157;1303875]        11.39 [0;572]   \n",
       "Single k-mer De Bruijn graph      109037.73 [994;177038]           0.27 [0;2]   \n",
       "\n",
       "                               misassembly events                    n50  \n",
       "Assembler Algorithm                                                       \n",
       "Multiple k-mer De Bruijn graph     22.66 [0;1148]  103409.79 [85;351027]  \n",
       "Single k-mer De Bruijn graph           0.47 [0;4]    10128.13 [35;26474]  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "stats_per_assembler_algo"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Global stats per assembler"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "stats_per_assembler = pd.DataFrame()\n",
    "for assembler in global_pipeline_metrics_df.assembler.unique():\n",
    "    row={'Assembler': assembler, \n",
    "         'Type':global_pipeline_metrics_df[global_pipeline_metrics_df.assembler == assembler].type.unique()[0],\n",
    "         'Algorithm': global_pipeline_metrics_df[global_pipeline_metrics_df.assembler == assembler].algorythm.unique()[0]}\n",
    "    lala=global_pipeline_metrics_df[global_pipeline_metrics_df.assembler == assembler].describe()\n",
    "    for column in lala.columns:\n",
    "        mean = lala.loc['mean',column]\n",
    "        minimum = lala.loc['min',column]\n",
    "        maximum = lala.loc['max',column]\n",
    "        row[column] = \"{} [{};{}]\".format(round(mean, 2), int(minimum), int(maximum))\n",
    "    stats_per_assembler=stats_per_assembler.append(row, ignore_index=True)\n",
    "stats_per_assembler = stats_per_assembler.set_index('Assembler')\n",
    "stats_per_assembler.to_csv(\"Tables/Results/Global metrics per assembler.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>contigs</th>\n",
       "      <th>filtered_Ns</th>\n",
       "      <th>filtered_basepairs</th>\n",
       "      <th>filtered_contigs</th>\n",
       "      <th>filtered_mapped_reads</th>\n",
       "      <th>filtered_n50</th>\n",
       "      <th>mapped_reads</th>\n",
       "      <th>max_contig</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>n50</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>1441.58 [9;3884]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>48280303.67 [20132631;91720949]</td>\n",
       "      <td>212679.33 [4769;677860]</td>\n",
       "      <td>802.5 [0;2328]</td>\n",
       "      <td>23327521.25 [3150980;30988200]</td>\n",
       "      <td>1041.75 [356;1825]</td>\n",
       "      <td>96.67 [79;107]</td>\n",
       "      <td>100939.92 [18568;192306]</td>\n",
       "      <td>135.4 [92;297]</td>\n",
       "      <td>654757.0 [88665;952618]</td>\n",
       "      <td>2.67 [0;8]</td>\n",
       "      <td>4.75 [0;14]</td>\n",
       "      <td>54339.58 [101;124246]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>141532851.73 [19635512;364761201]</td>\n",
       "      <td>2924637.8 [25377;8175890]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>10791989.4 [0;28174748]</td>\n",
       "      <td>2138.2 [0;6250]</td>\n",
       "      <td>36.91 [0;97]</td>\n",
       "      <td>9252.53 [0;31436]</td>\n",
       "      <td>48.46 [1;98]</td>\n",
       "      <td>62545.13 [994;163543]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>6119.93 [35;18682]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>24708836.73 [13638625;32810830]</td>\n",
       "      <td>2616.07 [693;6777]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>23710529.13 [13507672;30564396]</td>\n",
       "      <td>657.53 [219;814]</td>\n",
       "      <td>94.17 [82;98]</td>\n",
       "      <td>108382.33 [90095;140561]</td>\n",
       "      <td>95.14 [84;100]</td>\n",
       "      <td>674842.67 [542997;768585]</td>\n",
       "      <td>1.4 [0;3]</td>\n",
       "      <td>3.0 [0;6]</td>\n",
       "      <td>100702.93 [55075;136741]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>27009100.33 [13791669;39779384]</td>\n",
       "      <td>7055.6 [1025;20194]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>24338209.0 [13601126;31562690]</td>\n",
       "      <td>1511.27 [261;2725]</td>\n",
       "      <td>93.58 [81;99]</td>\n",
       "      <td>75249.73 [29134;113853]</td>\n",
       "      <td>95.09 [84;100]</td>\n",
       "      <td>518123.87 [303329;632213]</td>\n",
       "      <td>7.8 [2;18]</td>\n",
       "      <td>14.33 [2;36]</td>\n",
       "      <td>63523.93 [20993;105808]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>27494813.73 [13843596;41166407]</td>\n",
       "      <td>6718.87 [767;20614]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>24696076.6 [13671964;32021766]</td>\n",
       "      <td>970.4 [172;3044]</td>\n",
       "      <td>95.75 [84;100]</td>\n",
       "      <td>179387.0 [149841;210993]</td>\n",
       "      <td>96.78 [86;101]</td>\n",
       "      <td>852107.0 [632250;1205839]</td>\n",
       "      <td>7.07 [0;15]</td>\n",
       "      <td>14.67 [0;32]</td>\n",
       "      <td>148599.93 [82249;203548]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>22898207.18 [13174826;30381792]</td>\n",
       "      <td>7937.27 [882;19361]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>18810677.91 [1599587;30264953]</td>\n",
       "      <td>1663.73 [712;5044]</td>\n",
       "      <td>68.04 [3;98]</td>\n",
       "      <td>57630.27 [1175;106637]</td>\n",
       "      <td>76.41 [25;99]</td>\n",
       "      <td>258964.91 [2964;468375]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>56738.27 [658;106383]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>29645956.93 [14157309;48964620]</td>\n",
       "      <td>17286.47 [640;61644]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>24705104.67 [13672885;31967753]</td>\n",
       "      <td>977.93 [139;3392]</td>\n",
       "      <td>94.83 [83;100]</td>\n",
       "      <td>178396.73 [87983;239185]</td>\n",
       "      <td>95.69 [85;100]</td>\n",
       "      <td>804246.87 [473277;1036942]</td>\n",
       "      <td>1.67 [1;3]</td>\n",
       "      <td>4.13 [3;7]</td>\n",
       "      <td>153654.27 [3674;238611]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>27494742.73 [15334712;41187744]</td>\n",
       "      <td>30521.6 [14997;68012]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>22256482.6 [12920975;28279162]</td>\n",
       "      <td>2814.93 [762;3983]</td>\n",
       "      <td>87.09 [74;97]</td>\n",
       "      <td>19765.27 [10582;33550]</td>\n",
       "      <td>91.83 [80;99]</td>\n",
       "      <td>155530.33 [117945;177038]</td>\n",
       "      <td>0.53 [0;2]</td>\n",
       "      <td>0.93 [0;4]</td>\n",
       "      <td>14136.33 [7060;26474]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>20821825.8 [2957060;30430077]</td>\n",
       "      <td>3438.2 [36;9514]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>19785255.8 [2950393;30211740]</td>\n",
       "      <td>1271.67 [21;2889]</td>\n",
       "      <td>88.02 [62;97]</td>\n",
       "      <td>87109.87 [2148;351027]</td>\n",
       "      <td>91.14 [78;98]</td>\n",
       "      <td>437770.67 [29155;632013]</td>\n",
       "      <td>5.4 [0;12]</td>\n",
       "      <td>10.47 [0;23]</td>\n",
       "      <td>85724.0 [1155;351027]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>29707136.6 [15606257;46535940]</td>\n",
       "      <td>18662.07 [1140;39819]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>24802891.8 [13087777;32500826]</td>\n",
       "      <td>1481.27 [227;4999]</td>\n",
       "      <td>89.58 [75;99]</td>\n",
       "      <td>137422.2 [55465;208991]</td>\n",
       "      <td>97.96 [85;107]</td>\n",
       "      <td>721665.73 [263645;850176]</td>\n",
       "      <td>78.8 [0;572]</td>\n",
       "      <td>158.13 [0;1148]</td>\n",
       "      <td>115678.0 [1972;206515]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>22870452.47 [9747587;30592960]</td>\n",
       "      <td>584.6 [185;936]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>22773759.07 [9725301;30432114]</td>\n",
       "      <td>306.93 [112;443]</td>\n",
       "      <td>93.4 [81;98]</td>\n",
       "      <td>221413.27 [191424;256884]</td>\n",
       "      <td>94.27 [82;99]</td>\n",
       "      <td>945743.0 [847534;1303875]</td>\n",
       "      <td>1.07 [0;5]</td>\n",
       "      <td>2.13 [0;10]</td>\n",
       "      <td>220515.27 [191424;256884]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>11314.07 [0;124182]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>34548979.27 [15546803;57130377]</td>\n",
       "      <td>217283.07 [10924;652127]</td>\n",
       "      <td>3352.67 [0;12292]</td>\n",
       "      <td>13789106.67 [13883;28485306]</td>\n",
       "      <td>2048.27 [13;3237]</td>\n",
       "      <td>49.18 [0;95]</td>\n",
       "      <td>15393.47 [1039;34664]</td>\n",
       "      <td>53.51 [1;97]</td>\n",
       "      <td>369703.4 [1157;847809]</td>\n",
       "      <td>3.27 [0;8]</td>\n",
       "      <td>5.33 [0;14]</td>\n",
       "      <td>12361.93 [85;30453]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm                   Ns  \\\n",
       "Assembler                                                                \n",
       "ABySS              Multiple k-mer De Bruijn graph     1441.58 [9;3884]   \n",
       "BCALM2               Single k-mer De Bruijn graph            0.0 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph            0.0 [0;0]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph            0.0 [0;0]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph            0.0 [0;0]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph            0.0 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph            0.0 [0;0]   \n",
       "MINIA                Single k-mer De Bruijn graph            0.0 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph            0.0 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph            0.0 [0;0]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph            0.0 [0;0]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph  11314.07 [0;124182]   \n",
       "\n",
       "                          Type                          basepairs  \\\n",
       "Assembler                                                           \n",
       "ABySS              Metagenomic    48280303.67 [20132631;91720949]   \n",
       "BCALM2                 Genomic  141532851.73 [19635512;364761201]   \n",
       "GATBMiniaPipeline  Metagenomic    24708836.73 [13638625;32810830]   \n",
       "IDBA-UD            Metagenomic    27009100.33 [13791669;39779384]   \n",
       "MEGAHIT            Metagenomic    27494813.73 [13843596;41166407]   \n",
       "MetaHipMer2        Metagenomic    22898207.18 [13174826;30381792]   \n",
       "metaSPAdes         Metagenomic    29645956.93 [14157309;48964620]   \n",
       "MINIA                  Genomic    27494742.73 [15334712;41187744]   \n",
       "SKESA                  Genomic      20821825.8 [2957060;30430077]   \n",
       "SPAdes                 Genomic     29707136.6 [15606257;46535940]   \n",
       "Unicycler              Genomic     22870452.47 [9747587;30592960]   \n",
       "VelvetOptimiser    Metagenomic    34548979.27 [15546803;57130377]   \n",
       "\n",
       "                                     contigs        filtered_Ns  \\\n",
       "Assembler                                                         \n",
       "ABySS                212679.33 [4769;677860]     802.5 [0;2328]   \n",
       "BCALM2             2924637.8 [25377;8175890]          0.0 [0;0]   \n",
       "GATBMiniaPipeline         2616.07 [693;6777]          0.0 [0;0]   \n",
       "IDBA-UD                  7055.6 [1025;20194]          0.0 [0;0]   \n",
       "MEGAHIT                  6718.87 [767;20614]          0.0 [0;0]   \n",
       "MetaHipMer2              7937.27 [882;19361]          0.0 [0;0]   \n",
       "metaSPAdes              17286.47 [640;61644]          0.0 [0;0]   \n",
       "MINIA                  30521.6 [14997;68012]          0.0 [0;0]   \n",
       "SKESA                       3438.2 [36;9514]          0.0 [0;0]   \n",
       "SPAdes                 18662.07 [1140;39819]          0.0 [0;0]   \n",
       "Unicycler                    584.6 [185;936]          0.0 [0;0]   \n",
       "VelvetOptimiser     217283.07 [10924;652127]  3352.67 [0;12292]   \n",
       "\n",
       "                                filtered_basepairs    filtered_contigs  \\\n",
       "Assembler                                                                \n",
       "ABySS               23327521.25 [3150980;30988200]  1041.75 [356;1825]   \n",
       "BCALM2                     10791989.4 [0;28174748]     2138.2 [0;6250]   \n",
       "GATBMiniaPipeline  23710529.13 [13507672;30564396]    657.53 [219;814]   \n",
       "IDBA-UD             24338209.0 [13601126;31562690]  1511.27 [261;2725]   \n",
       "MEGAHIT             24696076.6 [13671964;32021766]    970.4 [172;3044]   \n",
       "MetaHipMer2         18810677.91 [1599587;30264953]  1663.73 [712;5044]   \n",
       "metaSPAdes         24705104.67 [13672885;31967753]   977.93 [139;3392]   \n",
       "MINIA               22256482.6 [12920975;28279162]  2814.93 [762;3983]   \n",
       "SKESA                19785255.8 [2950393;30211740]   1271.67 [21;2889]   \n",
       "SPAdes              24802891.8 [13087777;32500826]  1481.27 [227;4999]   \n",
       "Unicycler           22773759.07 [9725301;30432114]    306.93 [112;443]   \n",
       "VelvetOptimiser       13789106.67 [13883;28485306]   2048.27 [13;3237]   \n",
       "\n",
       "                  filtered_mapped_reads               filtered_n50  \\\n",
       "Assembler                                                            \n",
       "ABySS                    96.67 [79;107]   100939.92 [18568;192306]   \n",
       "BCALM2                     36.91 [0;97]          9252.53 [0;31436]   \n",
       "GATBMiniaPipeline         94.17 [82;98]   108382.33 [90095;140561]   \n",
       "IDBA-UD                   93.58 [81;99]    75249.73 [29134;113853]   \n",
       "MEGAHIT                  95.75 [84;100]   179387.0 [149841;210993]   \n",
       "MetaHipMer2                68.04 [3;98]     57630.27 [1175;106637]   \n",
       "metaSPAdes               94.83 [83;100]   178396.73 [87983;239185]   \n",
       "MINIA                     87.09 [74;97]     19765.27 [10582;33550]   \n",
       "SKESA                     88.02 [62;97]     87109.87 [2148;351027]   \n",
       "SPAdes                    89.58 [75;99]    137422.2 [55465;208991]   \n",
       "Unicycler                  93.4 [81;98]  221413.27 [191424;256884]   \n",
       "VelvetOptimiser            49.18 [0;95]      15393.47 [1039;34664]   \n",
       "\n",
       "                     mapped_reads                  max_contig  \\\n",
       "Assembler                                                       \n",
       "ABySS              135.4 [92;297]     654757.0 [88665;952618]   \n",
       "BCALM2               48.46 [1;98]       62545.13 [994;163543]   \n",
       "GATBMiniaPipeline  95.14 [84;100]   674842.67 [542997;768585]   \n",
       "IDBA-UD            95.09 [84;100]   518123.87 [303329;632213]   \n",
       "MEGAHIT            96.78 [86;101]   852107.0 [632250;1205839]   \n",
       "MetaHipMer2         76.41 [25;99]     258964.91 [2964;468375]   \n",
       "metaSPAdes         95.69 [85;100]  804246.87 [473277;1036942]   \n",
       "MINIA               91.83 [80;99]   155530.33 [117945;177038]   \n",
       "SKESA               91.14 [78;98]    437770.67 [29155;632013]   \n",
       "SPAdes             97.96 [85;107]   721665.73 [263645;850176]   \n",
       "Unicycler           94.27 [82;99]   945743.0 [847534;1303875]   \n",
       "VelvetOptimiser      53.51 [1;97]      369703.4 [1157;847809]   \n",
       "\n",
       "                  misassembled contigs misassembly events  \\\n",
       "Assembler                                                   \n",
       "ABySS                       2.67 [0;8]        4.75 [0;14]   \n",
       "BCALM2                       0.0 [0;0]          0.0 [0;0]   \n",
       "GATBMiniaPipeline            1.4 [0;3]          3.0 [0;6]   \n",
       "IDBA-UD                     7.8 [2;18]       14.33 [2;36]   \n",
       "MEGAHIT                    7.07 [0;15]       14.67 [0;32]   \n",
       "MetaHipMer2                  0.0 [0;0]          0.0 [0;0]   \n",
       "metaSPAdes                  1.67 [1;3]         4.13 [3;7]   \n",
       "MINIA                       0.53 [0;2]         0.93 [0;4]   \n",
       "SKESA                       5.4 [0;12]       10.47 [0;23]   \n",
       "SPAdes                    78.8 [0;572]    158.13 [0;1148]   \n",
       "Unicycler                   1.07 [0;5]        2.13 [0;10]   \n",
       "VelvetOptimiser             3.27 [0;8]        5.33 [0;14]   \n",
       "\n",
       "                                         n50  \n",
       "Assembler                                     \n",
       "ABySS                  54339.58 [101;124246]  \n",
       "BCALM2                    6119.93 [35;18682]  \n",
       "GATBMiniaPipeline   100702.93 [55075;136741]  \n",
       "IDBA-UD              63523.93 [20993;105808]  \n",
       "MEGAHIT             148599.93 [82249;203548]  \n",
       "MetaHipMer2            56738.27 [658;106383]  \n",
       "metaSPAdes           153654.27 [3674;238611]  \n",
       "MINIA                  14136.33 [7060;26474]  \n",
       "SKESA                  85724.0 [1155;351027]  \n",
       "SPAdes                115678.0 [1972;206515]  \n",
       "Unicycler          220515.27 [191424;256884]  \n",
       "VelvetOptimiser          12361.93 [85;30453]  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "stats_per_assembler"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Global stats per distribution"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>contigs</th>\n",
       "      <th>filtered_Ns</th>\n",
       "      <th>filtered_basepairs</th>\n",
       "      <th>filtered_contigs</th>\n",
       "      <th>filtered_mapped_reads</th>\n",
       "      <th>filtered_n50</th>\n",
       "      <th>mapped_reads</th>\n",
       "      <th>max_contig</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>n50</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>678.5 [9;1349]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>49430721.75 [20132631;78728532]</td>\n",
       "      <td>361241.5 [44621;677860]</td>\n",
       "      <td>200.0 [0;400]</td>\n",
       "      <td>8362124.25 [3150980;13573734]</td>\n",
       "      <td>784.0 [356;1212]</td>\n",
       "      <td>89.57 [79;99]</td>\n",
       "      <td>105437.0 [18568;192306]</td>\n",
       "      <td>199.54 [101;297]</td>\n",
       "      <td>469989.5 [88665;851314]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>4835.0 [101;9569]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>233321388.33 [19635512;364761201]</td>\n",
       "      <td>5418807.0 [44825;8175890]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>4688952.0 [11967;13397815]</td>\n",
       "      <td>433.0 [11;784]</td>\n",
       "      <td>32.44 [0;97]</td>\n",
       "      <td>11253.33 [1062;31436]</td>\n",
       "      <td>36.14 [1;98]</td>\n",
       "      <td>55949.33 [1212;163543]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>6255.33 [35;18682]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>14892063.33 [13638625;17126542]</td>\n",
       "      <td>2819.5 [693;6777]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>13685813.83 [13507672;13891396]</td>\n",
       "      <td>527.83 [219;762]</td>\n",
       "      <td>95.04 [89;98]</td>\n",
       "      <td>107608.33 [90095;140561]</td>\n",
       "      <td>95.72 [89;100]</td>\n",
       "      <td>602475.67 [542997;632215]</td>\n",
       "      <td>1.0 [0;2]</td>\n",
       "      <td>3.0 [0;6]</td>\n",
       "      <td>94548.33 [55075;136741]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>17234505.33 [13791669;23587129]</td>\n",
       "      <td>5722.33 [1025;14155]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>14926006.67 [13601126;17534580]</td>\n",
       "      <td>1184.67 [261;2725]</td>\n",
       "      <td>95.23 [89;99]</td>\n",
       "      <td>109414.33 [106347;113853]</td>\n",
       "      <td>95.91 [90;99]</td>\n",
       "      <td>616678.67 [585610;632213]</td>\n",
       "      <td>2.33 [2;3]</td>\n",
       "      <td>4.0 [2;5]</td>\n",
       "      <td>85565.0 [51161;105808]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>17581572.33 [13843596;24498566]</td>\n",
       "      <td>5593.67 [767;14172]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>15309833.0 [13671964;18552996]</td>\n",
       "      <td>1181.0 [172;3044]</td>\n",
       "      <td>97.05 [93;100]</td>\n",
       "      <td>197842.0 [171540;210993]</td>\n",
       "      <td>97.63 [94;100]</td>\n",
       "      <td>826891.0 [632250;1080013]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>1.33 [0;2]</td>\n",
       "      <td>152385.33 [82249;203548]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>15030376.0 [15029970;15030782]</td>\n",
       "      <td>11970.5 [11966;11975]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>10265689.0 [10264085;10267293]</td>\n",
       "      <td>5041.0 [5038;5044]</td>\n",
       "      <td>74.37 [74;74]</td>\n",
       "      <td>2173.5 [2170;2177]</td>\n",
       "      <td>85.27 [85;85]</td>\n",
       "      <td>14760.0 [14760;14760]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1513.0 [1511;1515]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>19164672.33 [14157309;27822010]</td>\n",
       "      <td>11693.67 [2099;26282]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>15386033.67 [13672885;18792639]</td>\n",
       "      <td>1238.33 [139;3392]</td>\n",
       "      <td>95.81 [89;99]</td>\n",
       "      <td>174651.33 [87983;239185]</td>\n",
       "      <td>96.17 [90;100]</td>\n",
       "      <td>796644.67 [622107;917728]</td>\n",
       "      <td>1.67 [1;3]</td>\n",
       "      <td>4.33 [3;7]</td>\n",
       "      <td>139179.67 [3674;220609]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>17498121.83 [15334712;20758864]</td>\n",
       "      <td>25649.33 [14997;37300]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>13290066.17 [12920975;13501499]</td>\n",
       "      <td>1465.67 [762;2172]</td>\n",
       "      <td>87.93 [81;97]</td>\n",
       "      <td>20765.0 [10582;33550]</td>\n",
       "      <td>92.64 [87;99]</td>\n",
       "      <td>168041.33 [163543;177038]</td>\n",
       "      <td>1.33 [0;2]</td>\n",
       "      <td>2.33 [0;4]</td>\n",
       "      <td>13902.0 [7485;26474]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>6659581.0 [2957060;8818742]</td>\n",
       "      <td>6124.0 [36;9514]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>4407308.0 [2950393;5801439]</td>\n",
       "      <td>1447.67 [21;2889]</td>\n",
       "      <td>82.19 [62;92]</td>\n",
       "      <td>120186.67 [2148;351027]</td>\n",
       "      <td>88.7 [78;95]</td>\n",
       "      <td>240146.67 [29155;632013]</td>\n",
       "      <td>4.0 [0;12]</td>\n",
       "      <td>7.67 [0;23]</td>\n",
       "      <td>117871.0 [1155;351027]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>20446071.0 [15606257;29517365]</td>\n",
       "      <td>24861.67 [7331;36214]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>15441310.0 [13087777;19595813]</td>\n",
       "      <td>2250.67 [227;4999]</td>\n",
       "      <td>84.02 [75;98]</td>\n",
       "      <td>87170.0 [55465;136684]</td>\n",
       "      <td>102.37 [99;107]</td>\n",
       "      <td>581322.33 [263645;847683]</td>\n",
       "      <td>191.0 [0;572]</td>\n",
       "      <td>383.33 [0;1148]</td>\n",
       "      <td>57051.0 [1972;122703]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>11596988.67 [9747587;13324149]</td>\n",
       "      <td>218.0 [185;264]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>11577328.67 [9725301;13315624]</td>\n",
       "      <td>166.33 [112;243]</td>\n",
       "      <td>94.33 [87;98]</td>\n",
       "      <td>230819.67 [192888;256884]</td>\n",
       "      <td>94.94 [88;99]</td>\n",
       "      <td>847562.0 [847534;847580]</td>\n",
       "      <td>1.67 [0;5]</td>\n",
       "      <td>3.33 [0;10]</td>\n",
       "      <td>230819.67 [192888;256884]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>21519.0 [0;124182]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>28904674.5 [15546803;47958628]</td>\n",
       "      <td>239235.33 [15386;525998]</td>\n",
       "      <td>2048.67 [0;12292]</td>\n",
       "      <td>5982171.17 [409222;12274643]</td>\n",
       "      <td>2093.33 [340;3237]</td>\n",
       "      <td>30.23 [0;95]</td>\n",
       "      <td>3961.5 [1169;13110]</td>\n",
       "      <td>35.99 [1;97]</td>\n",
       "      <td>75773.5 [2044;395655]</td>\n",
       "      <td>1.17 [0;7]</td>\n",
       "      <td>2.33 [0;14]</td>\n",
       "      <td>1046.0 [88;5219]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm                  Ns  \\\n",
       "Assembler                                                               \n",
       "ABySS              Multiple k-mer De Bruijn graph      678.5 [9;1349]   \n",
       "BCALM2               Single k-mer De Bruijn graph           0.0 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "MINIA                Single k-mer De Bruijn graph           0.0 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph  21519.0 [0;124182]   \n",
       "\n",
       "                          Type                          basepairs  \\\n",
       "Assembler                                                           \n",
       "ABySS              Metagenomic    49430721.75 [20132631;78728532]   \n",
       "BCALM2                 Genomic  233321388.33 [19635512;364761201]   \n",
       "GATBMiniaPipeline  Metagenomic    14892063.33 [13638625;17126542]   \n",
       "IDBA-UD            Metagenomic    17234505.33 [13791669;23587129]   \n",
       "MEGAHIT            Metagenomic    17581572.33 [13843596;24498566]   \n",
       "MetaHipMer2        Metagenomic     15030376.0 [15029970;15030782]   \n",
       "metaSPAdes         Metagenomic    19164672.33 [14157309;27822010]   \n",
       "MINIA                  Genomic    17498121.83 [15334712;20758864]   \n",
       "SKESA                  Genomic        6659581.0 [2957060;8818742]   \n",
       "SPAdes                 Genomic     20446071.0 [15606257;29517365]   \n",
       "Unicycler              Genomic     11596988.67 [9747587;13324149]   \n",
       "VelvetOptimiser    Metagenomic     28904674.5 [15546803;47958628]   \n",
       "\n",
       "                                     contigs        filtered_Ns  \\\n",
       "Assembler                                                         \n",
       "ABySS                361241.5 [44621;677860]      200.0 [0;400]   \n",
       "BCALM2             5418807.0 [44825;8175890]          0.0 [0;0]   \n",
       "GATBMiniaPipeline          2819.5 [693;6777]          0.0 [0;0]   \n",
       "IDBA-UD                 5722.33 [1025;14155]          0.0 [0;0]   \n",
       "MEGAHIT                  5593.67 [767;14172]          0.0 [0;0]   \n",
       "MetaHipMer2            11970.5 [11966;11975]          0.0 [0;0]   \n",
       "metaSPAdes             11693.67 [2099;26282]          0.0 [0;0]   \n",
       "MINIA                 25649.33 [14997;37300]          0.0 [0;0]   \n",
       "SKESA                       6124.0 [36;9514]          0.0 [0;0]   \n",
       "SPAdes                 24861.67 [7331;36214]          0.0 [0;0]   \n",
       "Unicycler                    218.0 [185;264]          0.0 [0;0]   \n",
       "VelvetOptimiser     239235.33 [15386;525998]  2048.67 [0;12292]   \n",
       "\n",
       "                                filtered_basepairs    filtered_contigs  \\\n",
       "Assembler                                                                \n",
       "ABySS                8362124.25 [3150980;13573734]    784.0 [356;1212]   \n",
       "BCALM2                  4688952.0 [11967;13397815]      433.0 [11;784]   \n",
       "GATBMiniaPipeline  13685813.83 [13507672;13891396]    527.83 [219;762]   \n",
       "IDBA-UD            14926006.67 [13601126;17534580]  1184.67 [261;2725]   \n",
       "MEGAHIT             15309833.0 [13671964;18552996]   1181.0 [172;3044]   \n",
       "MetaHipMer2         10265689.0 [10264085;10267293]  5041.0 [5038;5044]   \n",
       "metaSPAdes         15386033.67 [13672885;18792639]  1238.33 [139;3392]   \n",
       "MINIA              13290066.17 [12920975;13501499]  1465.67 [762;2172]   \n",
       "SKESA                  4407308.0 [2950393;5801439]   1447.67 [21;2889]   \n",
       "SPAdes              15441310.0 [13087777;19595813]  2250.67 [227;4999]   \n",
       "Unicycler           11577328.67 [9725301;13315624]    166.33 [112;243]   \n",
       "VelvetOptimiser       5982171.17 [409222;12274643]  2093.33 [340;3237]   \n",
       "\n",
       "                  filtered_mapped_reads               filtered_n50  \\\n",
       "Assembler                                                            \n",
       "ABySS                     89.57 [79;99]    105437.0 [18568;192306]   \n",
       "BCALM2                     32.44 [0;97]      11253.33 [1062;31436]   \n",
       "GATBMiniaPipeline         95.04 [89;98]   107608.33 [90095;140561]   \n",
       "IDBA-UD                   95.23 [89;99]  109414.33 [106347;113853]   \n",
       "MEGAHIT                  97.05 [93;100]   197842.0 [171540;210993]   \n",
       "MetaHipMer2               74.37 [74;74]         2173.5 [2170;2177]   \n",
       "metaSPAdes                95.81 [89;99]   174651.33 [87983;239185]   \n",
       "MINIA                     87.93 [81;97]      20765.0 [10582;33550]   \n",
       "SKESA                     82.19 [62;92]    120186.67 [2148;351027]   \n",
       "SPAdes                    84.02 [75;98]     87170.0 [55465;136684]   \n",
       "Unicycler                 94.33 [87;98]  230819.67 [192888;256884]   \n",
       "VelvetOptimiser            30.23 [0;95]        3961.5 [1169;13110]   \n",
       "\n",
       "                       mapped_reads                 max_contig  \\\n",
       "Assembler                                                        \n",
       "ABySS              199.54 [101;297]    469989.5 [88665;851314]   \n",
       "BCALM2                 36.14 [1;98]     55949.33 [1212;163543]   \n",
       "GATBMiniaPipeline    95.72 [89;100]  602475.67 [542997;632215]   \n",
       "IDBA-UD               95.91 [90;99]  616678.67 [585610;632213]   \n",
       "MEGAHIT              97.63 [94;100]  826891.0 [632250;1080013]   \n",
       "MetaHipMer2           85.27 [85;85]      14760.0 [14760;14760]   \n",
       "metaSPAdes           96.17 [90;100]  796644.67 [622107;917728]   \n",
       "MINIA                 92.64 [87;99]  168041.33 [163543;177038]   \n",
       "SKESA                  88.7 [78;95]   240146.67 [29155;632013]   \n",
       "SPAdes              102.37 [99;107]  581322.33 [263645;847683]   \n",
       "Unicycler             94.94 [88;99]   847562.0 [847534;847580]   \n",
       "VelvetOptimiser        35.99 [1;97]      75773.5 [2044;395655]   \n",
       "\n",
       "                  misassembled contigs misassembly events  \\\n",
       "Assembler                                                   \n",
       "ABySS                        0.0 [0;0]          0.0 [0;0]   \n",
       "BCALM2                       0.0 [0;0]          0.0 [0;0]   \n",
       "GATBMiniaPipeline            1.0 [0;2]          3.0 [0;6]   \n",
       "IDBA-UD                     2.33 [2;3]          4.0 [2;5]   \n",
       "MEGAHIT                     0.67 [0;1]         1.33 [0;2]   \n",
       "MetaHipMer2                  0.0 [0;0]          0.0 [0;0]   \n",
       "metaSPAdes                  1.67 [1;3]         4.33 [3;7]   \n",
       "MINIA                       1.33 [0;2]         2.33 [0;4]   \n",
       "SKESA                       4.0 [0;12]        7.67 [0;23]   \n",
       "SPAdes                   191.0 [0;572]    383.33 [0;1148]   \n",
       "Unicycler                   1.67 [0;5]        3.33 [0;10]   \n",
       "VelvetOptimiser             1.17 [0;7]        2.33 [0;14]   \n",
       "\n",
       "                                         n50  \n",
       "Assembler                                     \n",
       "ABySS                      4835.0 [101;9569]  \n",
       "BCALM2                    6255.33 [35;18682]  \n",
       "GATBMiniaPipeline    94548.33 [55075;136741]  \n",
       "IDBA-UD               85565.0 [51161;105808]  \n",
       "MEGAHIT             152385.33 [82249;203548]  \n",
       "MetaHipMer2               1513.0 [1511;1515]  \n",
       "metaSPAdes           139179.67 [3674;220609]  \n",
       "MINIA                   13902.0 [7485;26474]  \n",
       "SKESA                 117871.0 [1155;351027]  \n",
       "SPAdes                 57051.0 [1972;122703]  \n",
       "Unicycler          230819.67 [192888;256884]  \n",
       "VelvetOptimiser             1046.0 [88;5219]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Even\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>contigs</th>\n",
       "      <th>filtered_Ns</th>\n",
       "      <th>filtered_basepairs</th>\n",
       "      <th>filtered_contigs</th>\n",
       "      <th>filtered_mapped_reads</th>\n",
       "      <th>filtered_n50</th>\n",
       "      <th>mapped_reads</th>\n",
       "      <th>max_contig</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>n50</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>1823.12 [147;3884]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>47705094.62 [31503127;91720949]</td>\n",
       "      <td>138398.25 [4769;497797]</td>\n",
       "      <td>1103.75 [5;2328]</td>\n",
       "      <td>30810219.75 [30632246;30988200]</td>\n",
       "      <td>1170.62 [497;1825]</td>\n",
       "      <td>100.22 [81;107]</td>\n",
       "      <td>98691.38 [52698;138706]</td>\n",
       "      <td>103.33 [92;108]</td>\n",
       "      <td>747140.75 [297881;952618]</td>\n",
       "      <td>4.0 [2;8]</td>\n",
       "      <td>7.12 [3;14]</td>\n",
       "      <td>79091.88 [139;124246]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>80340494.0 [31047581;170402348]</td>\n",
       "      <td>1261858.33 [25377;3508016]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>14860681.0 [0;28174748]</td>\n",
       "      <td>3275.0 [0;6250]</td>\n",
       "      <td>39.88 [0;92]</td>\n",
       "      <td>7918.67 [0;20700]</td>\n",
       "      <td>56.66 [8;96]</td>\n",
       "      <td>66942.33 [994;161812]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>6029.67 [49;17395]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>31253352.33 [30440762;32810830]</td>\n",
       "      <td>2480.44 [943;5400]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>30393672.67 [30273320;30564396]</td>\n",
       "      <td>744.0 [657;814]</td>\n",
       "      <td>93.59 [82;98]</td>\n",
       "      <td>108898.33 [90988;122545]</td>\n",
       "      <td>94.75 [84;100]</td>\n",
       "      <td>723087.33 [632215;768585]</td>\n",
       "      <td>1.67 [1;3]</td>\n",
       "      <td>3.0 [1;6]</td>\n",
       "      <td>104806.0 [90346;121957]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>33525497.0 [30390566;39779384]</td>\n",
       "      <td>7944.44 [1822;20194]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>30613010.56 [30130413;31562690]</td>\n",
       "      <td>1729.0 [1254;2654]</td>\n",
       "      <td>92.47 [81;98]</td>\n",
       "      <td>52473.33 [29134;65108]</td>\n",
       "      <td>94.54 [84;100]</td>\n",
       "      <td>452420.67 [303329;585610]</td>\n",
       "      <td>11.44 [4;18]</td>\n",
       "      <td>21.22 [7;36]</td>\n",
       "      <td>48829.89 [20993;63725]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>34103641.33 [30545419;41166407]</td>\n",
       "      <td>7469.0 [814;20614]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>30953572.33 [30418624;32021766]</td>\n",
       "      <td>830.0 [518;1447]</td>\n",
       "      <td>94.89 [84;100]</td>\n",
       "      <td>167083.67 [149841;179870]</td>\n",
       "      <td>96.22 [86;101]</td>\n",
       "      <td>868917.67 [632274;1205839]</td>\n",
       "      <td>11.33 [9;15]</td>\n",
       "      <td>23.56 [18;32]</td>\n",
       "      <td>146076.33 [88308;178381]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>24646614.11 [13174826;30381792]</td>\n",
       "      <td>7041.0 [882;19361]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>20709564.33 [1599587;30264953]</td>\n",
       "      <td>913.22 [712;1310]</td>\n",
       "      <td>66.63 [3;98]</td>\n",
       "      <td>69954.0 [1175;106637]</td>\n",
       "      <td>74.44 [25;99]</td>\n",
       "      <td>313232.67 [2964;468375]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>69010.56 [658;106383]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>36633480.0 [30453778;48964620]</td>\n",
       "      <td>21015.0 [640;61644]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>30917818.67 [30383186;31967753]</td>\n",
       "      <td>804.33 [359;1689]</td>\n",
       "      <td>94.17 [83;100]</td>\n",
       "      <td>180893.67 [95101;238611]</td>\n",
       "      <td>95.36 [85;100]</td>\n",
       "      <td>809315.0 [473277;1036942]</td>\n",
       "      <td>1.67 [1;2]</td>\n",
       "      <td>4.0 [3;6]</td>\n",
       "      <td>163304.0 [42332;238611]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>34159156.67 [30641333;41187744]</td>\n",
       "      <td>33769.78 [16570;68012]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>28234093.56 [28205937;28279162]</td>\n",
       "      <td>3714.44 [3574;3983]</td>\n",
       "      <td>86.52 [74;92]</td>\n",
       "      <td>19098.78 [15817;20782]</td>\n",
       "      <td>91.29 [80;97]</td>\n",
       "      <td>147189.67 [117945;161812]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>14292.56 [7060;17967]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>30263322.33 [30110758;30430077]</td>\n",
       "      <td>1647.67 [1269;2322]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>30037221.0 [29835112;30211740]</td>\n",
       "      <td>1154.33 [846;1726]</td>\n",
       "      <td>91.9 [80;97]</td>\n",
       "      <td>65058.67 [43168;77870]</td>\n",
       "      <td>92.76 [81;98]</td>\n",
       "      <td>569520.0 [446163;632013]</td>\n",
       "      <td>6.33 [0;10]</td>\n",
       "      <td>12.33 [0;19]</td>\n",
       "      <td>64292.67 [42115;76924]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>35881180.33 [30442718;46535940]</td>\n",
       "      <td>14529.0 [1140;39819]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>31043946.33 [30276227;32500826]</td>\n",
       "      <td>968.33 [412;2049]</td>\n",
       "      <td>93.29 [82;99]</td>\n",
       "      <td>170923.67 [99299;208991]</td>\n",
       "      <td>95.02 [85;100]</td>\n",
       "      <td>815228.0 [745370;850176]</td>\n",
       "      <td>4.0 [0;9]</td>\n",
       "      <td>8.0 [0;18]</td>\n",
       "      <td>154762.67 [53292;206515]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>30386095.0 [30263458;30592960]</td>\n",
       "      <td>829.0 [656;936]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>30238046.0 [30093081;30432114]</td>\n",
       "      <td>400.67 [334;443]</td>\n",
       "      <td>92.78 [81;98]</td>\n",
       "      <td>215142.33 [191424;245050]</td>\n",
       "      <td>93.83 [82;99]</td>\n",
       "      <td>1011197.0 [847534;1303875]</td>\n",
       "      <td>0.67 [0;2]</td>\n",
       "      <td>1.33 [0;4]</td>\n",
       "      <td>213645.67 [191424;245050]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>4510.78 [105;8788]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>38311849.11 [30526291;57130377]</td>\n",
       "      <td>202648.22 [10924;652127]</td>\n",
       "      <td>4222.0 [0;8283]</td>\n",
       "      <td>18993730.33 [13883;28485306]</td>\n",
       "      <td>2018.22 [13;3035]</td>\n",
       "      <td>61.81 [0;93]</td>\n",
       "      <td>23014.78 [1039;34664]</td>\n",
       "      <td>65.2 [1;97]</td>\n",
       "      <td>565656.67 [1157;847809]</td>\n",
       "      <td>4.67 [0;8]</td>\n",
       "      <td>7.33 [0;11]</td>\n",
       "      <td>19905.89 [85;30453]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm                  Ns  \\\n",
       "Assembler                                                               \n",
       "ABySS              Multiple k-mer De Bruijn graph  1823.12 [147;3884]   \n",
       "BCALM2               Single k-mer De Bruijn graph           0.0 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "MINIA                Single k-mer De Bruijn graph           0.0 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph  4510.78 [105;8788]   \n",
       "\n",
       "                          Type                        basepairs  \\\n",
       "Assembler                                                         \n",
       "ABySS              Metagenomic  47705094.62 [31503127;91720949]   \n",
       "BCALM2                 Genomic  80340494.0 [31047581;170402348]   \n",
       "GATBMiniaPipeline  Metagenomic  31253352.33 [30440762;32810830]   \n",
       "IDBA-UD            Metagenomic   33525497.0 [30390566;39779384]   \n",
       "MEGAHIT            Metagenomic  34103641.33 [30545419;41166407]   \n",
       "MetaHipMer2        Metagenomic  24646614.11 [13174826;30381792]   \n",
       "metaSPAdes         Metagenomic   36633480.0 [30453778;48964620]   \n",
       "MINIA                  Genomic  34159156.67 [30641333;41187744]   \n",
       "SKESA                  Genomic  30263322.33 [30110758;30430077]   \n",
       "SPAdes                 Genomic  35881180.33 [30442718;46535940]   \n",
       "Unicycler              Genomic   30386095.0 [30263458;30592960]   \n",
       "VelvetOptimiser    Metagenomic  38311849.11 [30526291;57130377]   \n",
       "\n",
       "                                      contigs       filtered_Ns  \\\n",
       "Assembler                                                         \n",
       "ABySS                 138398.25 [4769;497797]  1103.75 [5;2328]   \n",
       "BCALM2             1261858.33 [25377;3508016]         0.0 [0;0]   \n",
       "GATBMiniaPipeline          2480.44 [943;5400]         0.0 [0;0]   \n",
       "IDBA-UD                  7944.44 [1822;20194]         0.0 [0;0]   \n",
       "MEGAHIT                    7469.0 [814;20614]         0.0 [0;0]   \n",
       "MetaHipMer2                7041.0 [882;19361]         0.0 [0;0]   \n",
       "metaSPAdes                21015.0 [640;61644]         0.0 [0;0]   \n",
       "MINIA                  33769.78 [16570;68012]         0.0 [0;0]   \n",
       "SKESA                     1647.67 [1269;2322]         0.0 [0;0]   \n",
       "SPAdes                   14529.0 [1140;39819]         0.0 [0;0]   \n",
       "Unicycler                     829.0 [656;936]         0.0 [0;0]   \n",
       "VelvetOptimiser      202648.22 [10924;652127]   4222.0 [0;8283]   \n",
       "\n",
       "                                filtered_basepairs     filtered_contigs  \\\n",
       "Assembler                                                                 \n",
       "ABySS              30810219.75 [30632246;30988200]   1170.62 [497;1825]   \n",
       "BCALM2                     14860681.0 [0;28174748]      3275.0 [0;6250]   \n",
       "GATBMiniaPipeline  30393672.67 [30273320;30564396]      744.0 [657;814]   \n",
       "IDBA-UD            30613010.56 [30130413;31562690]   1729.0 [1254;2654]   \n",
       "MEGAHIT            30953572.33 [30418624;32021766]     830.0 [518;1447]   \n",
       "MetaHipMer2         20709564.33 [1599587;30264953]    913.22 [712;1310]   \n",
       "metaSPAdes         30917818.67 [30383186;31967753]    804.33 [359;1689]   \n",
       "MINIA              28234093.56 [28205937;28279162]  3714.44 [3574;3983]   \n",
       "SKESA               30037221.0 [29835112;30211740]   1154.33 [846;1726]   \n",
       "SPAdes             31043946.33 [30276227;32500826]    968.33 [412;2049]   \n",
       "Unicycler           30238046.0 [30093081;30432114]     400.67 [334;443]   \n",
       "VelvetOptimiser       18993730.33 [13883;28485306]    2018.22 [13;3035]   \n",
       "\n",
       "                  filtered_mapped_reads               filtered_n50  \\\n",
       "Assembler                                                            \n",
       "ABySS                   100.22 [81;107]    98691.38 [52698;138706]   \n",
       "BCALM2                     39.88 [0;92]          7918.67 [0;20700]   \n",
       "GATBMiniaPipeline         93.59 [82;98]   108898.33 [90988;122545]   \n",
       "IDBA-UD                   92.47 [81;98]     52473.33 [29134;65108]   \n",
       "MEGAHIT                  94.89 [84;100]  167083.67 [149841;179870]   \n",
       "MetaHipMer2                66.63 [3;98]      69954.0 [1175;106637]   \n",
       "metaSPAdes               94.17 [83;100]   180893.67 [95101;238611]   \n",
       "MINIA                     86.52 [74;92]     19098.78 [15817;20782]   \n",
       "SKESA                      91.9 [80;97]     65058.67 [43168;77870]   \n",
       "SPAdes                    93.29 [82;99]   170923.67 [99299;208991]   \n",
       "Unicycler                 92.78 [81;98]  215142.33 [191424;245050]   \n",
       "VelvetOptimiser            61.81 [0;93]      23014.78 [1039;34664]   \n",
       "\n",
       "                      mapped_reads                  max_contig  \\\n",
       "Assembler                                                        \n",
       "ABySS              103.33 [92;108]   747140.75 [297881;952618]   \n",
       "BCALM2                56.66 [8;96]       66942.33 [994;161812]   \n",
       "GATBMiniaPipeline   94.75 [84;100]   723087.33 [632215;768585]   \n",
       "IDBA-UD             94.54 [84;100]   452420.67 [303329;585610]   \n",
       "MEGAHIT             96.22 [86;101]  868917.67 [632274;1205839]   \n",
       "MetaHipMer2          74.44 [25;99]     313232.67 [2964;468375]   \n",
       "metaSPAdes          95.36 [85;100]   809315.0 [473277;1036942]   \n",
       "MINIA                91.29 [80;97]   147189.67 [117945;161812]   \n",
       "SKESA                92.76 [81;98]    569520.0 [446163;632013]   \n",
       "SPAdes              95.02 [85;100]    815228.0 [745370;850176]   \n",
       "Unicycler            93.83 [82;99]  1011197.0 [847534;1303875]   \n",
       "VelvetOptimiser        65.2 [1;97]     565656.67 [1157;847809]   \n",
       "\n",
       "                  misassembled contigs misassembly events  \\\n",
       "Assembler                                                   \n",
       "ABySS                        4.0 [2;8]        7.12 [3;14]   \n",
       "BCALM2                       0.0 [0;0]          0.0 [0;0]   \n",
       "GATBMiniaPipeline           1.67 [1;3]          3.0 [1;6]   \n",
       "IDBA-UD                   11.44 [4;18]       21.22 [7;36]   \n",
       "MEGAHIT                   11.33 [9;15]      23.56 [18;32]   \n",
       "MetaHipMer2                  0.0 [0;0]          0.0 [0;0]   \n",
       "metaSPAdes                  1.67 [1;2]          4.0 [3;6]   \n",
       "MINIA                        0.0 [0;0]          0.0 [0;0]   \n",
       "SKESA                      6.33 [0;10]       12.33 [0;19]   \n",
       "SPAdes                       4.0 [0;9]         8.0 [0;18]   \n",
       "Unicycler                   0.67 [0;2]         1.33 [0;4]   \n",
       "VelvetOptimiser             4.67 [0;8]        7.33 [0;11]   \n",
       "\n",
       "                                         n50  \n",
       "Assembler                                     \n",
       "ABySS                  79091.88 [139;124246]  \n",
       "BCALM2                    6029.67 [49;17395]  \n",
       "GATBMiniaPipeline    104806.0 [90346;121957]  \n",
       "IDBA-UD               48829.89 [20993;63725]  \n",
       "MEGAHIT             146076.33 [88308;178381]  \n",
       "MetaHipMer2            69010.56 [658;106383]  \n",
       "metaSPAdes           163304.0 [42332;238611]  \n",
       "MINIA                  14292.56 [7060;17967]  \n",
       "SKESA                 64292.67 [42115;76924]  \n",
       "SPAdes              154762.67 [53292;206515]  \n",
       "Unicycler          213645.67 [191424;245050]  \n",
       "VelvetOptimiser          19905.89 [85;30453]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "for distribution in global_pipeline_metrics_df.distribution.unique():\n",
    "    stats_per_distribution = pd.DataFrame()\n",
    "    for assembler in global_pipeline_metrics_df.assembler.unique():\n",
    "        row={'Assembler': assembler, \n",
    "             'Type':global_pipeline_metrics_df[(global_pipeline_metrics_df.assembler == assembler) & (global_pipeline_metrics_df.distribution == distribution)].type.unique()[0],\n",
    "             'Algorithm': global_pipeline_metrics_df[global_pipeline_metrics_df.assembler == assembler].algorythm.unique()[0]}\n",
    "        lala=global_pipeline_metrics_df[(global_pipeline_metrics_df.assembler == assembler) & (global_pipeline_metrics_df.distribution == distribution)].describe()\n",
    "        for column in lala.columns:\n",
    "            mean = lala.loc['mean',column]\n",
    "            minimum = lala.loc['min',column]\n",
    "            maximum = lala.loc['max',column]\n",
    "            row[column] = \"{} [{};{}]\".format(round(mean, 2),int(minimum), int(maximum))\n",
    "        stats_per_distribution=stats_per_distribution.append(row, ignore_index=True)\n",
    "    stats_per_distribution = stats_per_distribution.set_index('Assembler')\n",
    "    stats_per_distribution.to_csv(\"Tables/Results/{} - Global metrics per assembler.csv\".format(distribution))\n",
    "    print(distribution)\n",
    "    display(stats_per_distribution)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Reference-based metrics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pipeline_report_tables.json data from run2...\n",
      "Processing pipeline_report_tables.json data from run1...\n",
      "Processing pipeline_report_tables.json data from run3...\n",
      "Processing pipeline_report_tables.json data from run2...\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>assembler</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>...</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>reference</th>\n",
       "      <th>run</th>\n",
       "      <th>sample</th>\n",
       "      <th>validity</th>\n",
       "      <th>distribution</th>\n",
       "      <th>type</th>\n",
       "      <th>algorythm</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ABySS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Bacillus_subtilis</td>\n",
       "      <td>run2</td>\n",
       "      <td>ERR2935805</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Log</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000765</td>\n",
       "      <td>1271.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>BCALM2</td>\n",
       "      <td>647119.0</td>\n",
       "      <td>0.159953</td>\n",
       "      <td>496.0</td>\n",
       "      <td>0.999942</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.000771</td>\n",
       "      <td>1.000825</td>\n",
       "      <td>Bacillus_subtilis</td>\n",
       "      <td>run2</td>\n",
       "      <td>ERR2935805</td>\n",
       "      <td>0.999946</td>\n",
       "      <td>Log</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>132.0</td>\n",
       "      <td>0.028610</td>\n",
       "      <td>31048.0</td>\n",
       "      <td>30028.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GATBMiniaPipeline</td>\n",
       "      <td>3960148.0</td>\n",
       "      <td>0.978859</td>\n",
       "      <td>205.0</td>\n",
       "      <td>0.999846</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.001038</td>\n",
       "      <td>1.001106</td>\n",
       "      <td>Bacillus_subtilis</td>\n",
       "      <td>run2</td>\n",
       "      <td>ERR2935805</td>\n",
       "      <td>0.999932</td>\n",
       "      <td>Log</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>41.0</td>\n",
       "      <td>0.057949</td>\n",
       "      <td>101732.0</td>\n",
       "      <td>101732.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IDBA-UD</td>\n",
       "      <td>3973333.0</td>\n",
       "      <td>0.982118</td>\n",
       "      <td>74.0</td>\n",
       "      <td>0.999048</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.000234</td>\n",
       "      <td>1.000349</td>\n",
       "      <td>Bacillus_subtilis</td>\n",
       "      <td>run2</td>\n",
       "      <td>ERR2935805</td>\n",
       "      <td>0.999884</td>\n",
       "      <td>Log</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>13.0</td>\n",
       "      <td>0.189992</td>\n",
       "      <td>263503.0</td>\n",
       "      <td>263503.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>MEGAHIT</td>\n",
       "      <td>3982262.0</td>\n",
       "      <td>0.984325</td>\n",
       "      <td>27.0</td>\n",
       "      <td>0.998894</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.999808</td>\n",
       "      <td>1.000243</td>\n",
       "      <td>Bacillus_subtilis</td>\n",
       "      <td>run2</td>\n",
       "      <td>ERR2935805</td>\n",
       "      <td>0.999566</td>\n",
       "      <td>Log</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2071</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>MINIA</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Staphylococcus_aureus_plasmid3</td>\n",
       "      <td>run2</td>\n",
       "      <td>EMS</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Even</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2072</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.908119</td>\n",
       "      <td>2718.0</td>\n",
       "      <td>2718.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>SKESA</td>\n",
       "      <td>2718.0</td>\n",
       "      <td>0.908119</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>Staphylococcus_aureus_plasmid3</td>\n",
       "      <td>run2</td>\n",
       "      <td>EMS</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>Even</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2073</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.999666</td>\n",
       "      <td>2992.0</td>\n",
       "      <td>2992.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>SPAdes</td>\n",
       "      <td>2992.0</td>\n",
       "      <td>0.999666</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>Staphylococcus_aureus_plasmid3</td>\n",
       "      <td>run2</td>\n",
       "      <td>EMS</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>Even</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2074</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.964250</td>\n",
       "      <td>2886.0</td>\n",
       "      <td>2886.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Unicycler</td>\n",
       "      <td>2886.0</td>\n",
       "      <td>0.964250</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>Staphylococcus_aureus_plasmid3</td>\n",
       "      <td>run2</td>\n",
       "      <td>EMS</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>Even</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2075</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>VelvetOptimiser</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Staphylococcus_aureus_plasmid3</td>\n",
       "      <td>run2</td>\n",
       "      <td>EMS</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Even</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2076 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        L90       LSA      NA50      NG50   Ns          assembler  basepairs  \\\n",
       "0       0.0  0.000000       0.0       0.0  0.0              ABySS        0.0   \n",
       "1       0.0  0.000765    1271.0       0.0  0.0             BCALM2   647119.0   \n",
       "2     132.0  0.028610   31048.0   30028.0  0.0  GATBMiniaPipeline  3960148.0   \n",
       "3      41.0  0.057949  101732.0  101732.0  0.0            IDBA-UD  3973333.0   \n",
       "4      13.0  0.189992  263503.0  263503.0  0.0            MEGAHIT  3982262.0   \n",
       "...     ...       ...       ...       ...  ...                ...        ...   \n",
       "2071    0.0  0.000000       0.0       0.0  0.0              MINIA        0.0   \n",
       "2072    0.0  0.908119    2718.0    2718.0  0.0              SKESA     2718.0   \n",
       "2073    0.0  0.999666    2992.0    2992.0  0.0             SPAdes     2992.0   \n",
       "2074    0.0  0.964250    2886.0    2886.0  0.0          Unicycler     2886.0   \n",
       "2075    0.0  0.000000       0.0       0.0  0.0    VelvetOptimiser        0.0   \n",
       "\n",
       "      breadth_of_coverage  contigs  identity  ...  misassembly events  \\\n",
       "0                0.000000      0.0  0.000000  ...                 0.0   \n",
       "1                0.159953    496.0  0.999942  ...                 0.0   \n",
       "2                0.978859    205.0  0.999846  ...                 0.0   \n",
       "3                0.982118     74.0  0.999048  ...                 2.0   \n",
       "4                0.984325     27.0  0.998894  ...                 0.0   \n",
       "...                   ...      ...       ...  ...                 ...   \n",
       "2071             0.000000      0.0  0.000000  ...                 0.0   \n",
       "2072             0.908119      1.0  1.000000  ...                 0.0   \n",
       "2073             0.999666      1.0  1.000000  ...                 0.0   \n",
       "2074             0.964250      1.0  1.000000  ...                 0.0   \n",
       "2075             0.000000      0.0  0.000000  ...                 0.0   \n",
       "\n",
       "      multiplicity  parsimony                       reference   run  \\\n",
       "0         0.000000   0.000000               Bacillus_subtilis  run2   \n",
       "1         1.000771   1.000825               Bacillus_subtilis  run2   \n",
       "2         1.001038   1.001106               Bacillus_subtilis  run2   \n",
       "3         1.000234   1.000349               Bacillus_subtilis  run2   \n",
       "4         0.999808   1.000243               Bacillus_subtilis  run2   \n",
       "...            ...        ...                             ...   ...   \n",
       "2071      0.000000   0.000000  Staphylococcus_aureus_plasmid3  run2   \n",
       "2072      1.000000   1.000000  Staphylococcus_aureus_plasmid3  run2   \n",
       "2073      1.000000   1.000000  Staphylococcus_aureus_plasmid3  run2   \n",
       "2074      1.000000   1.000000  Staphylococcus_aureus_plasmid3  run2   \n",
       "2075      0.000000   0.000000  Staphylococcus_aureus_plasmid3  run2   \n",
       "\n",
       "          sample  validity distribution         type  \\\n",
       "0     ERR2935805  0.000000          Log  Metagenomic   \n",
       "1     ERR2935805  0.999946          Log      Genomic   \n",
       "2     ERR2935805  0.999932          Log  Metagenomic   \n",
       "3     ERR2935805  0.999884          Log  Metagenomic   \n",
       "4     ERR2935805  0.999566          Log  Metagenomic   \n",
       "...          ...       ...          ...          ...   \n",
       "2071         EMS  0.000000         Even      Genomic   \n",
       "2072         EMS  1.000000         Even      Genomic   \n",
       "2073         EMS  1.000000         Even      Genomic   \n",
       "2074         EMS  1.000000         Even      Genomic   \n",
       "2075         EMS  0.000000         Even  Metagenomic   \n",
       "\n",
       "                           algorythm  \n",
       "0     Multiple k-mer De Bruijn graph  \n",
       "1       Single k-mer De Bruijn graph  \n",
       "2     Multiple k-mer De Bruijn graph  \n",
       "3     Multiple k-mer De Bruijn graph  \n",
       "4     Multiple k-mer De Bruijn graph  \n",
       "...                              ...  \n",
       "2071    Single k-mer De Bruijn graph  \n",
       "2072  Multiple k-mer De Bruijn graph  \n",
       "2073  Multiple k-mer De Bruijn graph  \n",
       "2074  Multiple k-mer De Bruijn graph  \n",
       "2075  Multiple k-mer De Bruijn graph  \n",
       "\n",
       "[2076 rows x 22 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "report_glob = glob.glob('../Results/*/*/report/pipeline_report_tables.json')\n",
    "reference_pipeline_metrics_df = pd.DataFrame()\n",
    "\n",
    "for pipeline_report_file in report_glob:\n",
    "    report_file_name = pipeline_report_file.split('/')[-1]\n",
    "    stats_run = pipeline_report_file.split('/')[-3]\n",
    "    print('Processing {0} data from {1}...'.format(report_file_name, stats_run))\n",
    "    \n",
    "    with open(pipeline_report_file) as _fh:\n",
    "        json_report = json.load(_fh)\n",
    "        for sample in json_report.keys():\n",
    "            for reference, data in json_report[sample]['ReferenceTables'].items():\n",
    "                for row in data:\n",
    "                    for item in row:\n",
    "                        reference_pipeline_metrics_df = reference_pipeline_metrics_df.append({'run': stats_run,\n",
    "                                                                                        'sample': sample,\n",
    "                                                                                        'assembler': item['assembler'],\n",
    "                                                                                        'reference': reference,\n",
    "                                                                                        'LSA': item['contiguity'],\n",
    "                                                                                        'breadth_of_coverage': item['breadth_of_coverage'],\n",
    "                                                                                        'multiplicity': item['multiplicity'],\n",
    "                                                                                        'validity': item['validity'],\n",
    "                                                                                        'parsimony': item['parsimony'],\n",
    "                                                                                        'identity': item['identity'],\n",
    "                                                                                        'lowest_identity': item['lowest_identity'],\n",
    "                                                                                        'L90': item['L90'],\n",
    "                                                                                        'contigs': item['aligned_contigs'],\n",
    "                                                                                        'NA50': item['NA50'],\n",
    "                                                                                        'NG50': item['NG50'],\n",
    "                                                                                        'basepairs': item['aligned_basepairs'],\n",
    "                                                                                        'Ns': item['Ns'],\n",
    "                                                                                        'misassembled contigs': item['misassembled_contigs'],\n",
    "                                                                                        'misassembly events': item['misassembly_events']},\n",
    "                                                                                       ignore_index=True)\n",
    "\n",
    "reference_pipeline_metrics_df['distribution'] = np.where(reference_pipeline_metrics_df['sample'].isin(log_distributed), 'Log', 'Even')\n",
    "reference_pipeline_metrics_df['type'] = np.where(reference_pipeline_metrics_df['assembler'].isin(genomic_assemblers), 'Genomic', 'Metagenomic')\n",
    "reference_pipeline_metrics_df['algorythm'] = np.where(reference_pipeline_metrics_df['assembler'].isin(single_kmer), 'Single k-mer De Bruijn graph', 'Multiple k-mer De Bruijn graph')\n",
    "reference_pipeline_metrics_df[['contigs','basepairs','L90','Ns','NA50','NG50','misassembled contigs', 'misassembly events', 'multiplicity','validity','parsimony','identity','lowest_identity']] = reference_pipeline_metrics_df[['contigs','basepairs','L90','Ns','NA50','NG50','misassembled contigs', 'misassembly events','multiplicity','validity','parsimony','identity','lowest_identity']].apply(pd.to_numeric)\n",
    "display(reference_pipeline_metrics_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "L90                     float64\n",
       "LSA                     float64\n",
       "NA50                    float64\n",
       "NG50                    float64\n",
       "Ns                      float64\n",
       "assembler                object\n",
       "basepairs               float64\n",
       "breadth_of_coverage     float64\n",
       "contigs                 float64\n",
       "identity                float64\n",
       "lowest_identity         float64\n",
       "misassembled contigs    float64\n",
       "misassembly events      float64\n",
       "multiplicity            float64\n",
       "parsimony               float64\n",
       "reference                object\n",
       "run                      object\n",
       "sample                   object\n",
       "validity                float64\n",
       "distribution             object\n",
       "type                     object\n",
       "algorythm                object\n",
       "dtype: object"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reference_pipeline_metrics_df.dtypes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Save data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "reference_pipeline_metrics_df.to_csv(\"Tables/Results/Reference metrics.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reference statistics per assembler type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_stats_per_assembler_type = pd.DataFrame()\n",
    "for assembly_type in reference_pipeline_metrics_df.type.unique():\n",
    "    row={'Assembler Type': assembly_type}\n",
    "    lala=reference_pipeline_metrics_df[reference_pipeline_metrics_df.type == assembly_type].describe()\n",
    "    for column in lala.columns:\n",
    "        mean = lala.loc['mean',column]\n",
    "        minimum = lala.loc['min',column]\n",
    "        maximum = lala.loc['max',column]\n",
    "        row[column] = \"{} [{};{}]\".format(round(mean, 2),int(minimum), int(maximum))\n",
    "    ref_stats_per_assembler_type=ref_stats_per_assembler_type.append(row, ignore_index=True)\n",
    "ref_stats_per_assembler_type = ref_stats_per_assembler_type.set_index('Assembler Type')\n",
    "ref_stats_per_assembler_type.to_csv(\"Tables/Results/Reference metrics per assembler type.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler Type</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Metagenomic</th>\n",
       "      <td>28.13 [0;1033]</td>\n",
       "      <td>0.18 [0;1]</td>\n",
       "      <td>70927.54 [0;593167]</td>\n",
       "      <td>70320.34 [0;593167]</td>\n",
       "      <td>70.54 [0;12098]</td>\n",
       "      <td>1790676.74 [0;6781605]</td>\n",
       "      <td>0.62 [0;1]</td>\n",
       "      <td>81.73 [0;2404]</td>\n",
       "      <td>0.72 [0;1]</td>\n",
       "      <td>0.69 [0;1]</td>\n",
       "      <td>0.32 [0;9]</td>\n",
       "      <td>0.6 [0;15]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.74 [0;5]</td>\n",
       "      <td>0.72 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic</th>\n",
       "      <td>37.39 [0;921]</td>\n",
       "      <td>0.13 [0;1]</td>\n",
       "      <td>54309.38 [0;475179]</td>\n",
       "      <td>53317.35 [0;475179]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1647148.01 [0;6728888]</td>\n",
       "      <td>0.53 [0;1]</td>\n",
       "      <td>121.73 [0;1877]</td>\n",
       "      <td>0.63 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>1.47 [0;565]</td>\n",
       "      <td>2.9 [0;1134]</td>\n",
       "      <td>0.63 [0;1]</td>\n",
       "      <td>0.64 [0;1]</td>\n",
       "      <td>0.63 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           L90         LSA                 NA50  \\\n",
       "Assembler Type                                                    \n",
       "Metagenomic     28.13 [0;1033]  0.18 [0;1]  70927.54 [0;593167]   \n",
       "Genomic          37.39 [0;921]  0.13 [0;1]  54309.38 [0;475179]   \n",
       "\n",
       "                               NG50               Ns               basepairs  \\\n",
       "Assembler Type                                                                 \n",
       "Metagenomic     70320.34 [0;593167]  70.54 [0;12098]  1790676.74 [0;6781605]   \n",
       "Genomic         53317.35 [0;475179]        0.0 [0;0]  1647148.01 [0;6728888]   \n",
       "\n",
       "               breadth_of_coverage          contigs    identity  \\\n",
       "Assembler Type                                                    \n",
       "Metagenomic             0.62 [0;1]   81.73 [0;2404]  0.72 [0;1]   \n",
       "Genomic                 0.53 [0;1]  121.73 [0;1877]  0.63 [0;1]   \n",
       "\n",
       "               lowest_identity misassembled contigs misassembly events  \\\n",
       "Assembler Type                                                           \n",
       "Metagenomic         0.69 [0;1]           0.32 [0;9]         0.6 [0;15]   \n",
       "Genomic              0.6 [0;1]         1.47 [0;565]       2.9 [0;1134]   \n",
       "\n",
       "               multiplicity   parsimony    validity  \n",
       "Assembler Type                                       \n",
       "Metagenomic      0.73 [0;1]  0.74 [0;5]  0.72 [0;1]  \n",
       "Genomic          0.63 [0;1]  0.64 [0;1]  0.63 [0;1]  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ref_stats_per_assembler_type"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reference stats per assembly algorithm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_stats_per_assembler_algo = pd.DataFrame()\n",
    "for assembly_algo in reference_pipeline_metrics_df.algorythm.unique():\n",
    "    row={'Assembler Algorithm': assembly_algo}\n",
    "    lala=reference_pipeline_metrics_df[reference_pipeline_metrics_df.algorythm == assembly_algo].describe()\n",
    "    for column in lala.columns:\n",
    "        mean = lala.loc['mean',column]\n",
    "        minimum = lala.loc['min',column]\n",
    "        maximum = lala.loc['max',column]\n",
    "        row[column] = \"{} [{};{}]\".format(round(mean, 2),int(minimum), int(maximum))\n",
    "    ref_stats_per_assembler_algo=ref_stats_per_assembler_algo.append(row, ignore_index=True)\n",
    "ref_stats_per_assembler_algo = ref_stats_per_assembler_algo.set_index('Assembler Algorithm')\n",
    "ref_stats_per_assembler_algo.to_csv(\"Tables/Results/Reference metrics per assembler algorithm.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler Algorithm</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Multiple k-mer De Bruijn graph</th>\n",
       "      <td>27.11 [0;1033]</td>\n",
       "      <td>0.19 [0;1]</td>\n",
       "      <td>75155.49 [0;593167]</td>\n",
       "      <td>74317.83 [0;593167]</td>\n",
       "      <td>48.34 [0;12098]</td>\n",
       "      <td>1803152.21 [0;6781605]</td>\n",
       "      <td>0.62 [0;1]</td>\n",
       "      <td>76.78 [0;2404]</td>\n",
       "      <td>0.72 [0;1]</td>\n",
       "      <td>0.68 [0;1]</td>\n",
       "      <td>0.98 [0;565]</td>\n",
       "      <td>1.92 [0;1134]</td>\n",
       "      <td>0.72 [0;1]</td>\n",
       "      <td>0.73 [0;5]</td>\n",
       "      <td>0.72 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Single k-mer De Bruijn graph</th>\n",
       "      <td>56.12 [0;921]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>9228.89 [0;65138]</td>\n",
       "      <td>8758.16 [0;64960]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1372388.53 [0;6524118]</td>\n",
       "      <td>0.4 [0;0]</td>\n",
       "      <td>205.32 [0;1495]</td>\n",
       "      <td>0.51 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.03 [0;2]</td>\n",
       "      <td>0.04 [0;4]</td>\n",
       "      <td>0.51 [0;1]</td>\n",
       "      <td>0.51 [0;1]</td>\n",
       "      <td>0.51 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           L90         LSA  \\\n",
       "Assembler Algorithm                                          \n",
       "Multiple k-mer De Bruijn graph  27.11 [0;1033]  0.19 [0;1]   \n",
       "Single k-mer De Bruijn graph     56.12 [0;921]  0.04 [0;0]   \n",
       "\n",
       "                                               NA50                 NG50  \\\n",
       "Assembler Algorithm                                                        \n",
       "Multiple k-mer De Bruijn graph  75155.49 [0;593167]  74317.83 [0;593167]   \n",
       "Single k-mer De Bruijn graph      9228.89 [0;65138]    8758.16 [0;64960]   \n",
       "\n",
       "                                             Ns               basepairs  \\\n",
       "Assembler Algorithm                                                       \n",
       "Multiple k-mer De Bruijn graph  48.34 [0;12098]  1803152.21 [0;6781605]   \n",
       "Single k-mer De Bruijn graph          0.0 [0;0]  1372388.53 [0;6524118]   \n",
       "\n",
       "                               breadth_of_coverage          contigs  \\\n",
       "Assembler Algorithm                                                   \n",
       "Multiple k-mer De Bruijn graph          0.62 [0;1]   76.78 [0;2404]   \n",
       "Single k-mer De Bruijn graph             0.4 [0;0]  205.32 [0;1495]   \n",
       "\n",
       "                                  identity lowest_identity  \\\n",
       "Assembler Algorithm                                          \n",
       "Multiple k-mer De Bruijn graph  0.72 [0;1]      0.68 [0;1]   \n",
       "Single k-mer De Bruijn graph    0.51 [0;1]       0.5 [0;1]   \n",
       "\n",
       "                               misassembled contigs misassembly events  \\\n",
       "Assembler Algorithm                                                      \n",
       "Multiple k-mer De Bruijn graph         0.98 [0;565]      1.92 [0;1134]   \n",
       "Single k-mer De Bruijn graph             0.03 [0;2]         0.04 [0;4]   \n",
       "\n",
       "                               multiplicity   parsimony    validity  \n",
       "Assembler Algorithm                                                  \n",
       "Multiple k-mer De Bruijn graph   0.72 [0;1]  0.73 [0;5]  0.72 [0;1]  \n",
       "Single k-mer De Bruijn graph     0.51 [0;1]  0.51 [0;1]  0.51 [0;1]  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ref_stats_per_assembler_algo"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reference stats per assembler"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_stats_per_assembler = pd.DataFrame()\n",
    "for assembler in reference_pipeline_metrics_df.assembler.unique():\n",
    "    row={'Assembler': assembler, \n",
    "         'Type':reference_pipeline_metrics_df[reference_pipeline_metrics_df.assembler == assembler].type.unique()[0],\n",
    "         'Algorithm': reference_pipeline_metrics_df[reference_pipeline_metrics_df.assembler == assembler].algorythm.unique()[0]}\n",
    "    lala=reference_pipeline_metrics_df[reference_pipeline_metrics_df.assembler == assembler].describe()\n",
    "    for column in lala.columns:\n",
    "        mean = lala.loc['mean',column]\n",
    "        minimum = lala.loc['min',column]\n",
    "        maximum = lala.loc['max',column]\n",
    "        row[column] = \"{} [{};{}]\".format(round(mean, 2),int(minimum), int(maximum))\n",
    "    ref_stats_per_assembler=ref_stats_per_assembler.append(row, ignore_index=True)\n",
    "ref_stats_per_assembler = ref_stats_per_assembler.set_index('Assembler')\n",
    "ref_stats_per_assembler.to_csv(\"Tables/Results/Reference metrics per assembler.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>57.11 [0;1033]</td>\n",
       "      <td>0.21 [0;0]</td>\n",
       "      <td>76534.51 [0;351511]</td>\n",
       "      <td>76400.56 [0;351511]</td>\n",
       "      <td>66.98 [0;2170]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1920076.87 [0;6781605]</td>\n",
       "      <td>0.66 [0;0]</td>\n",
       "      <td>86.45 [0;1404]</td>\n",
       "      <td>0.72 [0;1]</td>\n",
       "      <td>0.7 [0;1]</td>\n",
       "      <td>0.22 [0;8]</td>\n",
       "      <td>0.4 [0;14]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.72 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>36.16 [0;921]</td>\n",
       "      <td>0.03 [0;0]</td>\n",
       "      <td>5541.46 [0;65138]</td>\n",
       "      <td>5229.32 [0;64934]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>897715.86 [0;6515102]</td>\n",
       "      <td>0.26 [0;0]</td>\n",
       "      <td>178.13 [0;1495]</td>\n",
       "      <td>0.37 [0;1]</td>\n",
       "      <td>0.37 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.37 [0;1]</td>\n",
       "      <td>0.37 [0;1]</td>\n",
       "      <td>0.37 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>28.24 [0;459]</td>\n",
       "      <td>0.22 [0;1]</td>\n",
       "      <td>80158.39 [0;452364]</td>\n",
       "      <td>74423.64 [0;351186]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1966705.14 [0;6716522]</td>\n",
       "      <td>0.68 [0;1]</td>\n",
       "      <td>50.97 [0;604]</td>\n",
       "      <td>0.74 [0;1]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.12 [0;3]</td>\n",
       "      <td>0.25 [0;6]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.74 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>47.83 [0;292]</td>\n",
       "      <td>0.17 [0;1]</td>\n",
       "      <td>52707.09 [0;475377]</td>\n",
       "      <td>51305.18 [0;475377]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1958660.56 [0;6710005]</td>\n",
       "      <td>0.66 [0;1]</td>\n",
       "      <td>84.49 [0;449]</td>\n",
       "      <td>0.77 [0;1]</td>\n",
       "      <td>0.68 [0;1]</td>\n",
       "      <td>0.77 [0;9]</td>\n",
       "      <td>1.32 [0;15]</td>\n",
       "      <td>0.77 [0;1]</td>\n",
       "      <td>0.83 [0;5]</td>\n",
       "      <td>0.76 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>14.35 [0;147]</td>\n",
       "      <td>0.24 [0;1]</td>\n",
       "      <td>110053.78 [0;543128]</td>\n",
       "      <td>111658.51 [0;543128]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1977154.33 [0;6738936]</td>\n",
       "      <td>0.69 [0;1]</td>\n",
       "      <td>33.88 [0;223]</td>\n",
       "      <td>0.78 [0;1]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.61 [0;4]</td>\n",
       "      <td>1.24 [0;7]</td>\n",
       "      <td>0.79 [0;1]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>0.79 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>17.76 [0;110]</td>\n",
       "      <td>0.19 [0;0]</td>\n",
       "      <td>47999.15 [0;257424]</td>\n",
       "      <td>47344.33 [0;257424]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1559257.44 [0;6717767]</td>\n",
       "      <td>0.59 [0;0]</td>\n",
       "      <td>134.85 [0;2404]</td>\n",
       "      <td>0.85 [0;1]</td>\n",
       "      <td>0.84 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.85 [0;1]</td>\n",
       "      <td>0.85 [0;1]</td>\n",
       "      <td>0.85 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>14.96 [0;90]</td>\n",
       "      <td>0.22 [0;1]</td>\n",
       "      <td>104423.09 [0;593167]</td>\n",
       "      <td>103538.27 [0;593167]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1974311.24 [0;6733500]</td>\n",
       "      <td>0.69 [0;1]</td>\n",
       "      <td>29.81 [0;167]</td>\n",
       "      <td>0.79 [0;1]</td>\n",
       "      <td>0.76 [0;1]</td>\n",
       "      <td>0.15 [0;2]</td>\n",
       "      <td>0.36 [0;6]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>0.79 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>76.08 [0;649]</td>\n",
       "      <td>0.05 [0;0]</td>\n",
       "      <td>12916.32 [0;65138]</td>\n",
       "      <td>12287.0 [0;64960]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1847061.2 [0;6524118]</td>\n",
       "      <td>0.54 [0;0]</td>\n",
       "      <td>232.52 [0;1147]</td>\n",
       "      <td>0.65 [0;1]</td>\n",
       "      <td>0.63 [0;1]</td>\n",
       "      <td>0.06 [0;2]</td>\n",
       "      <td>0.09 [0;4]</td>\n",
       "      <td>0.65 [0;1]</td>\n",
       "      <td>0.65 [0;1]</td>\n",
       "      <td>0.65 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>37.44 [0;670]</td>\n",
       "      <td>0.16 [0;0]</td>\n",
       "      <td>49801.04 [0;351027]</td>\n",
       "      <td>49165.21 [0;351027]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1644801.71 [0;6673601]</td>\n",
       "      <td>0.58 [0;0]</td>\n",
       "      <td>106.01 [0;1877]</td>\n",
       "      <td>0.68 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.53 [0;12]</td>\n",
       "      <td>0.96 [0;23]</td>\n",
       "      <td>0.68 [0;1]</td>\n",
       "      <td>0.68 [0;1]</td>\n",
       "      <td>0.68 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>24.97 [0;835]</td>\n",
       "      <td>0.18 [0;1]</td>\n",
       "      <td>85323.02 [0;351104]</td>\n",
       "      <td>85314.51 [0;351104]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1951808.53 [0;6728888]</td>\n",
       "      <td>0.64 [0;1]</td>\n",
       "      <td>66.78 [0;1839]</td>\n",
       "      <td>0.79 [0;1]</td>\n",
       "      <td>0.74 [0;1]</td>\n",
       "      <td>6.65 [0;565]</td>\n",
       "      <td>13.26 [0;1134]</td>\n",
       "      <td>0.81 [0;1]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "      <td>0.79 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>12.32 [0;143]</td>\n",
       "      <td>0.24 [0;1]</td>\n",
       "      <td>117965.06 [0;475179]</td>\n",
       "      <td>114590.7 [0;475179]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1894352.78 [0;6723626]</td>\n",
       "      <td>0.63 [0;1]</td>\n",
       "      <td>25.21 [0;179]</td>\n",
       "      <td>0.66 [0;1]</td>\n",
       "      <td>0.65 [0;1]</td>\n",
       "      <td>0.09 [0;5]</td>\n",
       "      <td>0.18 [0;10]</td>\n",
       "      <td>0.66 [0;1]</td>\n",
       "      <td>0.66 [0;1]</td>\n",
       "      <td>0.66 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>19.67 [0;207]</td>\n",
       "      <td>0.05 [0;0]</td>\n",
       "      <td>19623.92 [0;338923]</td>\n",
       "      <td>22660.96 [0;464977]</td>\n",
       "      <td>407.3 [0;12098]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1142739.82 [0;6708956]</td>\n",
       "      <td>0.35 [0;0]</td>\n",
       "      <td>166.79 [0;2156]</td>\n",
       "      <td>0.45 [0;1]</td>\n",
       "      <td>0.44 [0;1]</td>\n",
       "      <td>0.28 [0;8]</td>\n",
       "      <td>0.46 [0;11]</td>\n",
       "      <td>0.45 [0;1]</td>\n",
       "      <td>0.45 [0;1]</td>\n",
       "      <td>0.45 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm             L90         LSA  \\\n",
       "Assembler                                                                       \n",
       "ABySS              Multiple k-mer De Bruijn graph  57.11 [0;1033]  0.21 [0;0]   \n",
       "BCALM2               Single k-mer De Bruijn graph   36.16 [0;921]  0.03 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph   28.24 [0;459]  0.22 [0;1]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph   47.83 [0;292]  0.17 [0;1]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph   14.35 [0;147]  0.24 [0;1]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph   17.76 [0;110]  0.19 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph    14.96 [0;90]  0.22 [0;1]   \n",
       "MINIA                Single k-mer De Bruijn graph   76.08 [0;649]  0.05 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph   37.44 [0;670]  0.16 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph   24.97 [0;835]  0.18 [0;1]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph   12.32 [0;143]  0.24 [0;1]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph   19.67 [0;207]  0.05 [0;0]   \n",
       "\n",
       "                                   NA50                  NG50  \\\n",
       "Assembler                                                       \n",
       "ABySS               76534.51 [0;351511]   76400.56 [0;351511]   \n",
       "BCALM2                5541.46 [0;65138]     5229.32 [0;64934]   \n",
       "GATBMiniaPipeline   80158.39 [0;452364]   74423.64 [0;351186]   \n",
       "IDBA-UD             52707.09 [0;475377]   51305.18 [0;475377]   \n",
       "MEGAHIT            110053.78 [0;543128]  111658.51 [0;543128]   \n",
       "MetaHipMer2         47999.15 [0;257424]   47344.33 [0;257424]   \n",
       "metaSPAdes         104423.09 [0;593167]  103538.27 [0;593167]   \n",
       "MINIA                12916.32 [0;65138]     12287.0 [0;64960]   \n",
       "SKESA               49801.04 [0;351027]   49165.21 [0;351027]   \n",
       "SPAdes              85323.02 [0;351104]   85314.51 [0;351104]   \n",
       "Unicycler          117965.06 [0;475179]   114590.7 [0;475179]   \n",
       "VelvetOptimiser     19623.92 [0;338923]   22660.96 [0;464977]   \n",
       "\n",
       "                                Ns         Type               basepairs  \\\n",
       "Assembler                                                                 \n",
       "ABySS               66.98 [0;2170]  Metagenomic  1920076.87 [0;6781605]   \n",
       "BCALM2                   0.0 [0;0]      Genomic   897715.86 [0;6515102]   \n",
       "GATBMiniaPipeline        0.0 [0;0]  Metagenomic  1966705.14 [0;6716522]   \n",
       "IDBA-UD                  0.0 [0;0]  Metagenomic  1958660.56 [0;6710005]   \n",
       "MEGAHIT                  0.0 [0;0]  Metagenomic  1977154.33 [0;6738936]   \n",
       "MetaHipMer2              0.0 [0;0]  Metagenomic  1559257.44 [0;6717767]   \n",
       "metaSPAdes               0.0 [0;0]  Metagenomic  1974311.24 [0;6733500]   \n",
       "MINIA                    0.0 [0;0]      Genomic   1847061.2 [0;6524118]   \n",
       "SKESA                    0.0 [0;0]      Genomic  1644801.71 [0;6673601]   \n",
       "SPAdes                   0.0 [0;0]      Genomic  1951808.53 [0;6728888]   \n",
       "Unicycler                0.0 [0;0]      Genomic  1894352.78 [0;6723626]   \n",
       "VelvetOptimiser    407.3 [0;12098]  Metagenomic  1142739.82 [0;6708956]   \n",
       "\n",
       "                  breadth_of_coverage          contigs    identity  \\\n",
       "Assembler                                                            \n",
       "ABySS                      0.66 [0;0]   86.45 [0;1404]  0.72 [0;1]   \n",
       "BCALM2                     0.26 [0;0]  178.13 [0;1495]  0.37 [0;1]   \n",
       "GATBMiniaPipeline          0.68 [0;1]    50.97 [0;604]  0.74 [0;1]   \n",
       "IDBA-UD                    0.66 [0;1]    84.49 [0;449]  0.77 [0;1]   \n",
       "MEGAHIT                    0.69 [0;1]    33.88 [0;223]  0.78 [0;1]   \n",
       "MetaHipMer2                0.59 [0;0]  134.85 [0;2404]  0.85 [0;1]   \n",
       "metaSPAdes                 0.69 [0;1]    29.81 [0;167]  0.79 [0;1]   \n",
       "MINIA                      0.54 [0;0]  232.52 [0;1147]  0.65 [0;1]   \n",
       "SKESA                      0.58 [0;0]  106.01 [0;1877]  0.68 [0;1]   \n",
       "SPAdes                     0.64 [0;1]   66.78 [0;1839]  0.79 [0;1]   \n",
       "Unicycler                  0.63 [0;1]    25.21 [0;179]  0.66 [0;1]   \n",
       "VelvetOptimiser            0.35 [0;0]  166.79 [0;2156]  0.45 [0;1]   \n",
       "\n",
       "                  lowest_identity misassembled contigs misassembly events  \\\n",
       "Assembler                                                                   \n",
       "ABySS                   0.7 [0;1]           0.22 [0;8]         0.4 [0;14]   \n",
       "BCALM2                 0.37 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "GATBMiniaPipeline      0.73 [0;1]           0.12 [0;3]         0.25 [0;6]   \n",
       "IDBA-UD                0.68 [0;1]           0.77 [0;9]        1.32 [0;15]   \n",
       "MEGAHIT                0.73 [0;1]           0.61 [0;4]         1.24 [0;7]   \n",
       "MetaHipMer2            0.84 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "metaSPAdes             0.76 [0;1]           0.15 [0;2]         0.36 [0;6]   \n",
       "MINIA                  0.63 [0;1]           0.06 [0;2]         0.09 [0;4]   \n",
       "SKESA                   0.6 [0;1]          0.53 [0;12]        0.96 [0;23]   \n",
       "SPAdes                 0.74 [0;1]         6.65 [0;565]     13.26 [0;1134]   \n",
       "Unicycler              0.65 [0;1]           0.09 [0;5]        0.18 [0;10]   \n",
       "VelvetOptimiser        0.44 [0;1]           0.28 [0;8]        0.46 [0;11]   \n",
       "\n",
       "                  multiplicity   parsimony    validity  \n",
       "Assembler                                               \n",
       "ABySS               0.73 [0;1]  0.73 [0;1]  0.72 [0;1]  \n",
       "BCALM2              0.37 [0;1]  0.37 [0;1]  0.37 [0;1]  \n",
       "GATBMiniaPipeline   0.75 [0;1]  0.75 [0;1]  0.74 [0;1]  \n",
       "IDBA-UD             0.77 [0;1]  0.83 [0;5]  0.76 [0;1]  \n",
       "MEGAHIT             0.79 [0;1]   0.8 [0;1]  0.79 [0;1]  \n",
       "MetaHipMer2         0.85 [0;1]  0.85 [0;1]  0.85 [0;1]  \n",
       "metaSPAdes           0.8 [0;1]   0.8 [0;1]  0.79 [0;1]  \n",
       "MINIA               0.65 [0;1]  0.65 [0;1]  0.65 [0;1]  \n",
       "SKESA               0.68 [0;1]  0.68 [0;1]  0.68 [0;1]  \n",
       "SPAdes              0.81 [0;1]  0.83 [0;1]  0.79 [0;1]  \n",
       "Unicycler           0.66 [0;1]  0.66 [0;1]  0.66 [0;1]  \n",
       "VelvetOptimiser     0.45 [0;1]  0.45 [0;1]  0.45 [0;1]  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ref_stats_per_assembler"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reference stats per reference replicon"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Per assembler for each reference replicon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Bacillus_subtilis', 'Enterococcus_faecalis',\n",
       "       'Escherichia_coli_plasmid', 'Escherichia_coli',\n",
       "       'Lactobacillus_fermentum', 'Listeria_monocytogenes',\n",
       "       'Pseudomonas_aeruginosa', 'Salmonella_enterica',\n",
       "       'Staphylococcus_aureus', 'Staphylococcus_aureus_plasmid1',\n",
       "       'Staphylococcus_aureus_plasmid2', 'Staphylococcus_aureus_plasmid3'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reference_pipeline_metrics_df.reference.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Bacillus_subtilis\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>185.83 [0;1033]</td>\n",
       "      <td>0.07 [0;0]</td>\n",
       "      <td>155739.83 [0;295233]</td>\n",
       "      <td>155677.17 [0;295233]</td>\n",
       "      <td>78.42 [0;400]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3274516.83 [0;3991152]</td>\n",
       "      <td>0.81 [0;0]</td>\n",
       "      <td>213.17 [0;1141]</td>\n",
       "      <td>0.83 [0;0]</td>\n",
       "      <td>0.83 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.84 [0;1]</td>\n",
       "      <td>0.84 [0;1]</td>\n",
       "      <td>0.83 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>30.4 [0;129]</td>\n",
       "      <td>0.01 [0;0]</td>\n",
       "      <td>18084.13 [0;65138]</td>\n",
       "      <td>17635.0 [0;64934]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1973728.93 [0;3951974]</td>\n",
       "      <td>0.49 [0;0]</td>\n",
       "      <td>411.53 [0;1467]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>95.13 [13;459]</td>\n",
       "      <td>0.11 [0;0]</td>\n",
       "      <td>172788.47 [8915;269695]</td>\n",
       "      <td>170206.53 [8732;263445]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3964562.93 [3882084;3984103]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>137.33 [27;604]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>0.4 [0;2]</td>\n",
       "      <td>1.2 [0;6]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>84.73 [18;292]</td>\n",
       "      <td>0.05 [0;0]</td>\n",
       "      <td>122499.47 [14006;252653]</td>\n",
       "      <td>114309.73 [13214;212969]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3968879.13 [3925882;3977845]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>130.4 [34;418]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.89 [0;0]</td>\n",
       "      <td>0.53 [0;2]</td>\n",
       "      <td>1.07 [0;5]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>33.67 [13;147]</td>\n",
       "      <td>0.12 [0;0]</td>\n",
       "      <td>219465.33 [26511;294549]</td>\n",
       "      <td>219633.2 [26209;294549]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3984618.47 [3969756;3989518]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>57.93 [27;223]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.86 [0;0]</td>\n",
       "      <td>0.73 [0;2]</td>\n",
       "      <td>1.67 [0;4]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>8.45 [0;16]</td>\n",
       "      <td>0.05 [0;0]</td>\n",
       "      <td>141225.27 [1145;257424]</td>\n",
       "      <td>140859.73 [0;257424]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2865084.18 [205541;3976387]</td>\n",
       "      <td>0.71 [0;0]</td>\n",
       "      <td>319.82 [27;1415]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>29.2 [9;66]</td>\n",
       "      <td>0.13 [0;0]</td>\n",
       "      <td>195863.93 [61490;343843]</td>\n",
       "      <td>195489.27 [62046;343843]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3978593.47 [3971865;3987932]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>54.0 [18;110]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>0.53 [0;2]</td>\n",
       "      <td>1.6 [0;6]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>170.4 [66;649]</td>\n",
       "      <td>0.03 [0;0]</td>\n",
       "      <td>43919.73 [6662;65138]</td>\n",
       "      <td>43207.8 [6313;64960]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>3932778.0 [3821823;3953200]</td>\n",
       "      <td>0.97 [0;0]</td>\n",
       "      <td>249.8 [116;783]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.53 [0;2]</td>\n",
       "      <td>0.93 [0;4]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>10.2 [0;20]</td>\n",
       "      <td>0.06 [0;0]</td>\n",
       "      <td>145456.6 [0;269493]</td>\n",
       "      <td>144190.8 [0;263164]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2381901.8 [0;3971432]</td>\n",
       "      <td>0.59 [0;0]</td>\n",
       "      <td>18.0 [0;34]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>23.87 [11;72]</td>\n",
       "      <td>0.14 [0;0]</td>\n",
       "      <td>235171.87 [56470;315090]</td>\n",
       "      <td>241773.47 [54127;315090]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>3975949.67 [3964395;3982668]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>44.47 [22;116]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.27 [0;2]</td>\n",
       "      <td>0.67 [0;5]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>25.67 [0;143]</td>\n",
       "      <td>0.13 [0;0]</td>\n",
       "      <td>209593.0 [0;324887]</td>\n",
       "      <td>192161.4 [0;324887]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>3164659.67 [0;3975910]</td>\n",
       "      <td>0.78 [0;0]</td>\n",
       "      <td>38.93 [0;179]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>25.4 [0;63]</td>\n",
       "      <td>0.03 [0;0]</td>\n",
       "      <td>46571.53 [0;269136]</td>\n",
       "      <td>47034.87 [0;294404]</td>\n",
       "      <td>373.93 [0;5609]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2521388.13 [0;3965142]</td>\n",
       "      <td>0.62 [0;0]</td>\n",
       "      <td>283.8 [0;1254]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>0.2 [0;3]</td>\n",
       "      <td>0.33 [0;5]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm              L90  \\\n",
       "Assembler                                                            \n",
       "ABySS              Multiple k-mer De Bruijn graph  185.83 [0;1033]   \n",
       "BCALM2               Single k-mer De Bruijn graph     30.4 [0;129]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph   95.13 [13;459]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph   84.73 [18;292]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph   33.67 [13;147]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph      8.45 [0;16]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph      29.2 [9;66]   \n",
       "MINIA                Single k-mer De Bruijn graph   170.4 [66;649]   \n",
       "SKESA              Multiple k-mer De Bruijn graph      10.2 [0;20]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph    23.87 [11;72]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph    25.67 [0;143]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph      25.4 [0;63]   \n",
       "\n",
       "                          LSA                      NA50  \\\n",
       "Assembler                                                 \n",
       "ABySS              0.07 [0;0]      155739.83 [0;295233]   \n",
       "BCALM2             0.01 [0;0]        18084.13 [0;65138]   \n",
       "GATBMiniaPipeline  0.11 [0;0]   172788.47 [8915;269695]   \n",
       "IDBA-UD            0.05 [0;0]  122499.47 [14006;252653]   \n",
       "MEGAHIT            0.12 [0;0]  219465.33 [26511;294549]   \n",
       "MetaHipMer2        0.05 [0;0]   141225.27 [1145;257424]   \n",
       "metaSPAdes         0.13 [0;0]  195863.93 [61490;343843]   \n",
       "MINIA              0.03 [0;0]     43919.73 [6662;65138]   \n",
       "SKESA              0.06 [0;0]       145456.6 [0;269493]   \n",
       "SPAdes             0.14 [0;0]  235171.87 [56470;315090]   \n",
       "Unicycler          0.13 [0;0]       209593.0 [0;324887]   \n",
       "VelvetOptimiser    0.03 [0;0]       46571.53 [0;269136]   \n",
       "\n",
       "                                       NG50               Ns         Type  \\\n",
       "Assembler                                                                   \n",
       "ABySS                  155677.17 [0;295233]    78.42 [0;400]  Metagenomic   \n",
       "BCALM2                    17635.0 [0;64934]        0.0 [0;0]      Genomic   \n",
       "GATBMiniaPipeline   170206.53 [8732;263445]        0.0 [0;0]  Metagenomic   \n",
       "IDBA-UD            114309.73 [13214;212969]        0.0 [0;0]  Metagenomic   \n",
       "MEGAHIT             219633.2 [26209;294549]        0.0 [0;0]  Metagenomic   \n",
       "MetaHipMer2            140859.73 [0;257424]        0.0 [0;0]  Metagenomic   \n",
       "metaSPAdes         195489.27 [62046;343843]        0.0 [0;0]  Metagenomic   \n",
       "MINIA                  43207.8 [6313;64960]        0.0 [0;0]      Genomic   \n",
       "SKESA                   144190.8 [0;263164]        0.0 [0;0]      Genomic   \n",
       "SPAdes             241773.47 [54127;315090]        0.0 [0;0]      Genomic   \n",
       "Unicycler               192161.4 [0;324887]        0.0 [0;0]      Genomic   \n",
       "VelvetOptimiser         47034.87 [0;294404]  373.93 [0;5609]  Metagenomic   \n",
       "\n",
       "                                      basepairs breadth_of_coverage  \\\n",
       "Assembler                                                             \n",
       "ABySS                    3274516.83 [0;3991152]          0.81 [0;0]   \n",
       "BCALM2                   1973728.93 [0;3951974]          0.49 [0;0]   \n",
       "GATBMiniaPipeline  3964562.93 [3882084;3984103]          0.98 [0;0]   \n",
       "IDBA-UD            3968879.13 [3925882;3977845]          0.98 [0;0]   \n",
       "MEGAHIT            3984618.47 [3969756;3989518]          0.98 [0;0]   \n",
       "MetaHipMer2         2865084.18 [205541;3976387]          0.71 [0;0]   \n",
       "metaSPAdes         3978593.47 [3971865;3987932]          0.98 [0;0]   \n",
       "MINIA               3932778.0 [3821823;3953200]          0.97 [0;0]   \n",
       "SKESA                     2381901.8 [0;3971432]          0.59 [0;0]   \n",
       "SPAdes             3975949.67 [3964395;3982668]          0.98 [0;0]   \n",
       "Unicycler                3164659.67 [0;3975910]          0.78 [0;0]   \n",
       "VelvetOptimiser          2521388.13 [0;3965142]          0.62 [0;0]   \n",
       "\n",
       "                            contigs    identity lowest_identity  \\\n",
       "Assembler                                                         \n",
       "ABySS               213.17 [0;1141]  0.83 [0;0]      0.83 [0;0]   \n",
       "BCALM2              411.53 [0;1467]   0.8 [0;1]       0.8 [0;1]   \n",
       "GATBMiniaPipeline   137.33 [27;604]   1.0 [0;1]      0.99 [0;1]   \n",
       "IDBA-UD              130.4 [34;418]   1.0 [0;0]      0.89 [0;0]   \n",
       "MEGAHIT              57.93 [27;223]  0.99 [0;0]      0.86 [0;0]   \n",
       "MetaHipMer2        319.82 [27;1415]   1.0 [0;1]      0.98 [0;1]   \n",
       "metaSPAdes            54.0 [18;110]   1.0 [0;0]      0.98 [0;0]   \n",
       "MINIA               249.8 [116;783]   1.0 [0;1]       1.0 [0;1]   \n",
       "SKESA                   18.0 [0;34]   0.6 [0;1]       0.6 [0;1]   \n",
       "SPAdes               44.47 [22;116]   1.0 [0;0]      0.99 [0;0]   \n",
       "Unicycler             38.93 [0;179]  0.87 [0;1]      0.87 [0;1]   \n",
       "VelvetOptimiser      283.8 [0;1254]   0.8 [0;1]       0.8 [0;1]   \n",
       "\n",
       "                  misassembled contigs misassembly events multiplicity  \\\n",
       "Assembler                                                                \n",
       "ABySS                        0.0 [0;0]          0.0 [0;0]   0.84 [0;1]   \n",
       "BCALM2                       0.0 [0;0]          0.0 [0;0]    0.8 [0;1]   \n",
       "GATBMiniaPipeline            0.4 [0;2]          1.2 [0;6]    1.0 [1;1]   \n",
       "IDBA-UD                     0.53 [0;2]         1.07 [0;5]    1.0 [1;1]   \n",
       "MEGAHIT                     0.73 [0;2]         1.67 [0;4]    1.0 [0;1]   \n",
       "MetaHipMer2                  0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "metaSPAdes                  0.53 [0;2]          1.6 [0;6]    1.0 [0;1]   \n",
       "MINIA                       0.53 [0;2]         0.93 [0;4]    1.0 [1;1]   \n",
       "SKESA                        0.0 [0;0]          0.0 [0;0]    0.6 [0;1]   \n",
       "SPAdes                      0.27 [0;2]         0.67 [0;5]    1.0 [0;1]   \n",
       "Unicycler                    0.0 [0;0]          0.0 [0;0]   0.87 [0;1]   \n",
       "VelvetOptimiser              0.2 [0;3]         0.33 [0;5]    0.8 [0;1]   \n",
       "\n",
       "                    parsimony    validity  \n",
       "Assembler                                  \n",
       "ABySS              0.84 [0;1]  0.83 [0;0]  \n",
       "BCALM2              0.8 [0;1]   0.8 [0;1]  \n",
       "GATBMiniaPipeline   1.0 [1;1]   1.0 [0;1]  \n",
       "IDBA-UD             1.0 [1;1]   1.0 [0;0]  \n",
       "MEGAHIT             1.0 [0;1]   1.0 [0;0]  \n",
       "MetaHipMer2         1.0 [1;1]   1.0 [0;1]  \n",
       "metaSPAdes          1.0 [0;1]   1.0 [0;1]  \n",
       "MINIA               1.0 [1;1]   1.0 [0;1]  \n",
       "SKESA               0.6 [0;1]   0.6 [0;1]  \n",
       "SPAdes              1.0 [0;1]   1.0 [0;1]  \n",
       "Unicycler          0.87 [0;1]  0.87 [0;1]  \n",
       "VelvetOptimiser     0.8 [0;1]   0.8 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Enterococcus_faecalis\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>15.42 [0;64]</td>\n",
       "      <td>0.1 [0;0]</td>\n",
       "      <td>127665.17 [0;241090]</td>\n",
       "      <td>127665.17 [0;241090]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1880638.58 [0;2828460]</td>\n",
       "      <td>0.66 [0;0]</td>\n",
       "      <td>28.08 [0;108]</td>\n",
       "      <td>0.67 [0;0]</td>\n",
       "      <td>0.66 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.68 [0;1]</td>\n",
       "      <td>0.68 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>20.87 [0;107]</td>\n",
       "      <td>0.01 [0;0]</td>\n",
       "      <td>6332.2 [0;30453]</td>\n",
       "      <td>6084.0 [0;30452]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>596263.2 [0;2768590]</td>\n",
       "      <td>0.21 [0;0]</td>\n",
       "      <td>69.13 [0;180]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>11.0 [0;24]</td>\n",
       "      <td>0.1 [0;0]</td>\n",
       "      <td>108568.4 [0;235254]</td>\n",
       "      <td>108568.4 [0;235254]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1679183.33 [0;2820351]</td>\n",
       "      <td>0.59 [0;0]</td>\n",
       "      <td>22.47 [0;46]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.59 [0;1]</td>\n",
       "      <td>0.2 [0;1]</td>\n",
       "      <td>0.4 [0;2]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>39.4 [0;147]</td>\n",
       "      <td>0.06 [0;0]</td>\n",
       "      <td>48945.4 [0;113206]</td>\n",
       "      <td>50217.6 [0;118897]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1679559.0 [0;2811602]</td>\n",
       "      <td>0.59 [0;0]</td>\n",
       "      <td>67.8 [0;231]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>0.56 [0;0]</td>\n",
       "      <td>0.6 [0;3]</td>\n",
       "      <td>1.2 [0;6]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>8.6 [0;17]</td>\n",
       "      <td>0.14 [0;0]</td>\n",
       "      <td>141202.6 [0;235381]</td>\n",
       "      <td>141848.6 [0;238546]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1681246.13 [0;2822833]</td>\n",
       "      <td>0.59 [0;0]</td>\n",
       "      <td>19.47 [0;35]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>0.58 [0;0]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.8 [0;2]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>14.73 [0;27]</td>\n",
       "      <td>0.07 [0;0]</td>\n",
       "      <td>65553.45 [1166;118895]</td>\n",
       "      <td>63736.91 [0;118895]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1583777.36 [9974;2811369]</td>\n",
       "      <td>0.56 [0;0]</td>\n",
       "      <td>73.82 [5;179]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>9.0 [0;25]</td>\n",
       "      <td>0.1 [0;0]</td>\n",
       "      <td>124052.6 [0;238611]</td>\n",
       "      <td>124052.6 [0;238611]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1683765.2 [0;2818677]</td>\n",
       "      <td>0.59 [0;0]</td>\n",
       "      <td>19.0 [0;51]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>0.59 [0;0]</td>\n",
       "      <td>0.2 [0;1]</td>\n",
       "      <td>0.4 [0;2]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>71.07 [0;149]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>18014.0 [0;30453]</td>\n",
       "      <td>16138.2 [0;30453]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1658143.47 [0;2770364]</td>\n",
       "      <td>0.58 [0;0]</td>\n",
       "      <td>111.6 [0;222]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "      <td>0.86 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>18.2 [0;42]</td>\n",
       "      <td>0.07 [0;0]</td>\n",
       "      <td>64106.4 [0;152709]</td>\n",
       "      <td>63471.4 [0;152709]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1683198.4 [0;2810284]</td>\n",
       "      <td>0.59 [0;0]</td>\n",
       "      <td>33.87 [0;72]</td>\n",
       "      <td>0.86 [0;1]</td>\n",
       "      <td>0.7 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "      <td>0.86 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>9.8 [0;25]</td>\n",
       "      <td>0.12 [0;0]</td>\n",
       "      <td>116484.6 [0;239100]</td>\n",
       "      <td>114500.6 [0;239100]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1680508.33 [0;2812518]</td>\n",
       "      <td>0.59 [0;0]</td>\n",
       "      <td>24.4 [0;63]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>1.4 [0;6]</td>\n",
       "      <td>2.27 [0;8]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "      <td>0.97 [0;1]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>6.2 [0;14]</td>\n",
       "      <td>0.16 [0;0]</td>\n",
       "      <td>170115.8 [0;377104]</td>\n",
       "      <td>160310.4 [0;377104]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1681967.2 [0;2813747]</td>\n",
       "      <td>0.59 [0;0]</td>\n",
       "      <td>14.0 [0;30]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>41.4 [0;106]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>12560.13 [0;30453]</td>\n",
       "      <td>11924.0 [0;30453]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1095235.53 [0;2736286]</td>\n",
       "      <td>0.38 [0;0]</td>\n",
       "      <td>64.53 [0;163]</td>\n",
       "      <td>0.53 [0;1]</td>\n",
       "      <td>0.53 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.53 [0;1]</td>\n",
       "      <td>0.53 [0;1]</td>\n",
       "      <td>0.53 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm            L90         LSA  \\\n",
       "Assembler                                                                      \n",
       "ABySS              Multiple k-mer De Bruijn graph   15.42 [0;64]   0.1 [0;0]   \n",
       "BCALM2               Single k-mer De Bruijn graph  20.87 [0;107]  0.01 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph    11.0 [0;24]   0.1 [0;0]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph   39.4 [0;147]  0.06 [0;0]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph     8.6 [0;17]  0.14 [0;0]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph   14.73 [0;27]  0.07 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph     9.0 [0;25]   0.1 [0;0]   \n",
       "MINIA                Single k-mer De Bruijn graph  71.07 [0;149]  0.02 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph    18.2 [0;42]  0.07 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph     9.8 [0;25]  0.12 [0;0]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph     6.2 [0;14]  0.16 [0;0]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph   41.4 [0;106]  0.02 [0;0]   \n",
       "\n",
       "                                     NA50                  NG50         Ns  \\\n",
       "Assembler                                                                    \n",
       "ABySS                127665.17 [0;241090]  127665.17 [0;241090]  0.0 [0;0]   \n",
       "BCALM2                   6332.2 [0;30453]      6084.0 [0;30452]  0.0 [0;0]   \n",
       "GATBMiniaPipeline     108568.4 [0;235254]   108568.4 [0;235254]  0.0 [0;0]   \n",
       "IDBA-UD                48945.4 [0;113206]    50217.6 [0;118897]  0.0 [0;0]   \n",
       "MEGAHIT               141202.6 [0;235381]   141848.6 [0;238546]  0.0 [0;0]   \n",
       "MetaHipMer2        65553.45 [1166;118895]   63736.91 [0;118895]  0.0 [0;0]   \n",
       "metaSPAdes            124052.6 [0;238611]   124052.6 [0;238611]  0.0 [0;0]   \n",
       "MINIA                   18014.0 [0;30453]     16138.2 [0;30453]  0.0 [0;0]   \n",
       "SKESA                  64106.4 [0;152709]    63471.4 [0;152709]  0.0 [0;0]   \n",
       "SPAdes                116484.6 [0;239100]   114500.6 [0;239100]  0.0 [0;0]   \n",
       "Unicycler             170115.8 [0;377104]   160310.4 [0;377104]  0.0 [0;0]   \n",
       "VelvetOptimiser        12560.13 [0;30453]     11924.0 [0;30453]  0.0 [0;0]   \n",
       "\n",
       "                          Type                  basepairs breadth_of_coverage  \\\n",
       "Assembler                                                                       \n",
       "ABySS              Metagenomic     1880638.58 [0;2828460]          0.66 [0;0]   \n",
       "BCALM2                 Genomic       596263.2 [0;2768590]          0.21 [0;0]   \n",
       "GATBMiniaPipeline  Metagenomic     1679183.33 [0;2820351]          0.59 [0;0]   \n",
       "IDBA-UD            Metagenomic      1679559.0 [0;2811602]          0.59 [0;0]   \n",
       "MEGAHIT            Metagenomic     1681246.13 [0;2822833]          0.59 [0;0]   \n",
       "MetaHipMer2        Metagenomic  1583777.36 [9974;2811369]          0.56 [0;0]   \n",
       "metaSPAdes         Metagenomic      1683765.2 [0;2818677]          0.59 [0;0]   \n",
       "MINIA                  Genomic     1658143.47 [0;2770364]          0.58 [0;0]   \n",
       "SKESA                  Genomic      1683198.4 [0;2810284]          0.59 [0;0]   \n",
       "SPAdes                 Genomic     1680508.33 [0;2812518]          0.59 [0;0]   \n",
       "Unicycler              Genomic      1681967.2 [0;2813747]          0.59 [0;0]   \n",
       "VelvetOptimiser    Metagenomic     1095235.53 [0;2736286]          0.38 [0;0]   \n",
       "\n",
       "                         contigs    identity lowest_identity  \\\n",
       "Assembler                                                      \n",
       "ABySS              28.08 [0;108]  0.67 [0;0]      0.66 [0;0]   \n",
       "BCALM2             69.13 [0;180]   0.4 [0;1]       0.4 [0;1]   \n",
       "GATBMiniaPipeline   22.47 [0;46]   0.6 [0;1]      0.59 [0;1]   \n",
       "IDBA-UD             67.8 [0;231]   0.6 [0;0]      0.56 [0;0]   \n",
       "MEGAHIT             19.47 [0;35]   0.6 [0;0]      0.58 [0;0]   \n",
       "MetaHipMer2        73.82 [5;179]   1.0 [0;1]       1.0 [0;1]   \n",
       "metaSPAdes           19.0 [0;51]   0.6 [0;0]      0.59 [0;0]   \n",
       "MINIA              111.6 [0;222]  0.87 [0;1]      0.86 [0;1]   \n",
       "SKESA               33.87 [0;72]  0.86 [0;1]       0.7 [0;1]   \n",
       "SPAdes               24.4 [0;63]  0.83 [0;1]       0.8 [0;1]   \n",
       "Unicycler            14.0 [0;30]   0.6 [0;0]       0.6 [0;0]   \n",
       "VelvetOptimiser    64.53 [0;163]  0.53 [0;1]      0.53 [0;1]   \n",
       "\n",
       "                  misassembled contigs misassembly events multiplicity  \\\n",
       "Assembler                                                                \n",
       "ABySS                        0.0 [0;0]          0.0 [0;0]   0.68 [0;1]   \n",
       "BCALM2                       0.0 [0;0]          0.0 [0;0]    0.4 [0;1]   \n",
       "GATBMiniaPipeline            0.2 [0;1]          0.4 [0;2]    0.6 [0;1]   \n",
       "IDBA-UD                      0.6 [0;3]          1.2 [0;6]    0.6 [0;1]   \n",
       "MEGAHIT                      0.4 [0;1]          0.8 [0;2]    0.6 [0;1]   \n",
       "MetaHipMer2                  0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "metaSPAdes                   0.2 [0;1]          0.4 [0;2]    0.6 [0;1]   \n",
       "MINIA                        0.0 [0;0]          0.0 [0;0]   0.87 [0;1]   \n",
       "SKESA                        0.4 [0;1]          0.4 [0;1]   0.87 [0;1]   \n",
       "SPAdes                       1.4 [0;6]         2.27 [0;8]    0.9 [0;1]   \n",
       "Unicycler                    0.0 [0;0]          0.0 [0;0]    0.6 [0;0]   \n",
       "VelvetOptimiser              0.0 [0;0]          0.0 [0;0]   0.53 [0;1]   \n",
       "\n",
       "                    parsimony    validity  \n",
       "Assembler                                  \n",
       "ABySS              0.68 [0;1]  0.67 [0;1]  \n",
       "BCALM2              0.4 [0;1]   0.4 [0;1]  \n",
       "GATBMiniaPipeline   0.6 [0;1]   0.6 [0;1]  \n",
       "IDBA-UD             0.6 [0;1]   0.6 [0;0]  \n",
       "MEGAHIT             0.6 [0;1]   0.6 [0;0]  \n",
       "MetaHipMer2         1.0 [1;1]   1.0 [1;1]  \n",
       "metaSPAdes          0.6 [0;1]   0.6 [0;1]  \n",
       "MINIA              0.87 [0;1]  0.87 [0;1]  \n",
       "SKESA              0.87 [0;1]  0.86 [0;1]  \n",
       "SPAdes             0.97 [0;1]  0.83 [0;1]  \n",
       "Unicycler           0.6 [0;0]   0.6 [0;1]  \n",
       "VelvetOptimiser    0.53 [0;1]  0.53 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Escherichia_coli_plasmid\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.17 [0;1]</td>\n",
       "      <td>0.11 [0;0]</td>\n",
       "      <td>10283.17 [0;55639]</td>\n",
       "      <td>9599.17 [0;55639]</td>\n",
       "      <td>78.0 [0;204]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>35982.58 [0;109424]</td>\n",
       "      <td>0.33 [0;0]</td>\n",
       "      <td>8.5 [0;26]</td>\n",
       "      <td>0.66 [0;0]</td>\n",
       "      <td>0.65 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>3.6 [0;18]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>2061.4 [0;7117]</td>\n",
       "      <td>1949.4 [0;6638]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>38728.2 [0;102586]</td>\n",
       "      <td>0.35 [0;0]</td>\n",
       "      <td>10.8 [0;33]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>1.8 [0;7]</td>\n",
       "      <td>0.16 [0;0]</td>\n",
       "      <td>10523.6 [0;28587]</td>\n",
       "      <td>10400.0 [0;28587]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>62504.0 [0;109450]</td>\n",
       "      <td>0.57 [0;0]</td>\n",
       "      <td>7.4 [0;24]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>3.0 [0;8]</td>\n",
       "      <td>0.16 [0;0]</td>\n",
       "      <td>13735.8 [0;28584]</td>\n",
       "      <td>13588.2 [0;28584]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>64103.93 [0;108159]</td>\n",
       "      <td>0.58 [0;0]</td>\n",
       "      <td>4.73 [0;11]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>2.0 [0;6]</td>\n",
       "      <td>0.21 [0;0]</td>\n",
       "      <td>20031.0 [0;55277]</td>\n",
       "      <td>19260.0 [0;55277]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>65170.07 [0;109516]</td>\n",
       "      <td>0.59 [0;0]</td>\n",
       "      <td>3.33 [0;9]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>2.73 [0;7]</td>\n",
       "      <td>0.15 [0;0]</td>\n",
       "      <td>11549.45 [0;28267]</td>\n",
       "      <td>11139.55 [0;28267]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>58789.45 [0;107495]</td>\n",
       "      <td>0.53 [0;0]</td>\n",
       "      <td>4.36 [0;10]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>2.0 [0;6]</td>\n",
       "      <td>0.21 [0;0]</td>\n",
       "      <td>19471.2 [0;55105]</td>\n",
       "      <td>18915.2 [0;55105]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>64047.6 [0;108144]</td>\n",
       "      <td>0.58 [0;0]</td>\n",
       "      <td>3.0 [0;9]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>4.6 [0;17]</td>\n",
       "      <td>0.09 [0;0]</td>\n",
       "      <td>5037.4 [0;14811]</td>\n",
       "      <td>4911.6 [0;14811]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>59772.4 [0;104297]</td>\n",
       "      <td>0.54 [0;0]</td>\n",
       "      <td>12.4 [0;33]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.4 [0;2]</td>\n",
       "      <td>0.11 [0;0]</td>\n",
       "      <td>6241.8 [0;28384]</td>\n",
       "      <td>5676.8 [0;28384]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>32630.2 [0;107808]</td>\n",
       "      <td>0.3 [0;0]</td>\n",
       "      <td>8.2 [0;26]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>2.2 [0;6]</td>\n",
       "      <td>0.21 [0;0]</td>\n",
       "      <td>14364.53 [0;28541]</td>\n",
       "      <td>13607.0 [0;28541]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>65125.07 [0;109262]</td>\n",
       "      <td>0.59 [0;0]</td>\n",
       "      <td>3.73 [0;9]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.2 [0;1]</td>\n",
       "      <td>0.1 [0;0]</td>\n",
       "      <td>10999.4 [0;54997]</td>\n",
       "      <td>10586.4 [0;52932]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>21585.8 [0;107929]</td>\n",
       "      <td>0.2 [0;0]</td>\n",
       "      <td>0.4 [0;2]</td>\n",
       "      <td>0.2 [0;0]</td>\n",
       "      <td>0.2 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.2 [0;1]</td>\n",
       "      <td>0.2 [0;1]</td>\n",
       "      <td>0.2 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>1.4 [0;4]</td>\n",
       "      <td>0.12 [0;0]</td>\n",
       "      <td>9814.0 [0;27962]</td>\n",
       "      <td>9814.0 [0;27962]</td>\n",
       "      <td>205.8 [0;835]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>42935.6 [0;107377]</td>\n",
       "      <td>0.39 [0;0]</td>\n",
       "      <td>2.0 [0;6]</td>\n",
       "      <td>0.4 [0;0]</td>\n",
       "      <td>0.4 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm         L90         LSA  \\\n",
       "Assembler                                                                   \n",
       "ABySS              Multiple k-mer De Bruijn graph  0.17 [0;1]  0.11 [0;0]   \n",
       "BCALM2               Single k-mer De Bruijn graph  3.6 [0;18]  0.04 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph   1.8 [0;7]  0.16 [0;0]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph   3.0 [0;8]  0.16 [0;0]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph   2.0 [0;6]  0.21 [0;0]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph  2.73 [0;7]  0.15 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph   2.0 [0;6]  0.21 [0;0]   \n",
       "MINIA                Single k-mer De Bruijn graph  4.6 [0;17]  0.09 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph   0.4 [0;2]  0.11 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph   2.2 [0;6]  0.21 [0;0]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph   0.2 [0;1]   0.1 [0;0]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph   1.4 [0;4]  0.12 [0;0]   \n",
       "\n",
       "                                 NA50                NG50             Ns  \\\n",
       "Assembler                                                                  \n",
       "ABySS              10283.17 [0;55639]   9599.17 [0;55639]   78.0 [0;204]   \n",
       "BCALM2                2061.4 [0;7117]     1949.4 [0;6638]      0.0 [0;0]   \n",
       "GATBMiniaPipeline   10523.6 [0;28587]   10400.0 [0;28587]      0.0 [0;0]   \n",
       "IDBA-UD             13735.8 [0;28584]   13588.2 [0;28584]      0.0 [0;0]   \n",
       "MEGAHIT             20031.0 [0;55277]   19260.0 [0;55277]      0.0 [0;0]   \n",
       "MetaHipMer2        11549.45 [0;28267]  11139.55 [0;28267]      0.0 [0;0]   \n",
       "metaSPAdes          19471.2 [0;55105]   18915.2 [0;55105]      0.0 [0;0]   \n",
       "MINIA                5037.4 [0;14811]    4911.6 [0;14811]      0.0 [0;0]   \n",
       "SKESA                6241.8 [0;28384]    5676.8 [0;28384]      0.0 [0;0]   \n",
       "SPAdes             14364.53 [0;28541]   13607.0 [0;28541]      0.0 [0;0]   \n",
       "Unicycler           10999.4 [0;54997]   10586.4 [0;52932]      0.0 [0;0]   \n",
       "VelvetOptimiser      9814.0 [0;27962]    9814.0 [0;27962]  205.8 [0;835]   \n",
       "\n",
       "                          Type            basepairs breadth_of_coverage  \\\n",
       "Assembler                                                                 \n",
       "ABySS              Metagenomic  35982.58 [0;109424]          0.33 [0;0]   \n",
       "BCALM2                 Genomic   38728.2 [0;102586]          0.35 [0;0]   \n",
       "GATBMiniaPipeline  Metagenomic   62504.0 [0;109450]          0.57 [0;0]   \n",
       "IDBA-UD            Metagenomic  64103.93 [0;108159]          0.58 [0;0]   \n",
       "MEGAHIT            Metagenomic  65170.07 [0;109516]          0.59 [0;0]   \n",
       "MetaHipMer2        Metagenomic  58789.45 [0;107495]          0.53 [0;0]   \n",
       "metaSPAdes         Metagenomic   64047.6 [0;108144]          0.58 [0;0]   \n",
       "MINIA                  Genomic   59772.4 [0;104297]          0.54 [0;0]   \n",
       "SKESA                  Genomic   32630.2 [0;107808]           0.3 [0;0]   \n",
       "SPAdes                 Genomic  65125.07 [0;109262]          0.59 [0;0]   \n",
       "Unicycler              Genomic   21585.8 [0;107929]           0.2 [0;0]   \n",
       "VelvetOptimiser    Metagenomic   42935.6 [0;107377]          0.39 [0;0]   \n",
       "\n",
       "                       contigs    identity lowest_identity  \\\n",
       "Assembler                                                    \n",
       "ABySS               8.5 [0;26]  0.66 [0;0]      0.65 [0;0]   \n",
       "BCALM2             10.8 [0;33]   0.4 [0;1]       0.4 [0;1]   \n",
       "GATBMiniaPipeline   7.4 [0;24]   0.6 [0;1]       0.6 [0;1]   \n",
       "IDBA-UD            4.73 [0;11]  0.73 [0;1]      0.73 [0;1]   \n",
       "MEGAHIT             3.33 [0;9]  0.73 [0;1]      0.73 [0;1]   \n",
       "MetaHipMer2        4.36 [0;10]  0.82 [0;1]      0.82 [0;1]   \n",
       "metaSPAdes           3.0 [0;9]   0.6 [0;1]       0.6 [0;1]   \n",
       "MINIA              12.4 [0;33]   0.6 [0;1]       0.6 [0;1]   \n",
       "SKESA               8.2 [0;26]   0.6 [0;1]       0.6 [0;1]   \n",
       "SPAdes              3.73 [0;9]  0.73 [0;1]      0.73 [0;1]   \n",
       "Unicycler            0.4 [0;2]   0.2 [0;0]       0.2 [0;0]   \n",
       "VelvetOptimiser      2.0 [0;6]   0.4 [0;0]       0.4 [0;0]   \n",
       "\n",
       "                  misassembled contigs misassembly events multiplicity  \\\n",
       "Assembler                                                                \n",
       "ABySS                        0.0 [0;0]          0.0 [0;0]   0.67 [0;1]   \n",
       "BCALM2                       0.0 [0;0]          0.0 [0;0]    0.4 [0;1]   \n",
       "GATBMiniaPipeline            0.0 [0;0]          0.0 [0;0]    0.6 [0;1]   \n",
       "IDBA-UD                      0.0 [0;0]          0.0 [0;0]   0.73 [0;1]   \n",
       "MEGAHIT                      0.0 [0;0]          0.0 [0;0]   0.73 [0;1]   \n",
       "MetaHipMer2                  0.0 [0;0]          0.0 [0;0]   0.82 [0;1]   \n",
       "metaSPAdes                   0.0 [0;0]          0.0 [0;0]    0.6 [0;1]   \n",
       "MINIA                        0.0 [0;0]          0.0 [0;0]    0.6 [0;1]   \n",
       "SKESA                        0.0 [0;0]          0.0 [0;0]    0.6 [0;1]   \n",
       "SPAdes                       0.0 [0;0]          0.0 [0;0]   0.73 [0;1]   \n",
       "Unicycler                    0.0 [0;0]          0.0 [0;0]    0.2 [0;1]   \n",
       "VelvetOptimiser              0.0 [0;0]          0.0 [0;0]    0.4 [0;1]   \n",
       "\n",
       "                    parsimony    validity  \n",
       "Assembler                                  \n",
       "ABySS              0.67 [0;1]  0.67 [0;1]  \n",
       "BCALM2              0.4 [0;1]   0.4 [0;1]  \n",
       "GATBMiniaPipeline   0.6 [0;1]   0.6 [0;1]  \n",
       "IDBA-UD            0.73 [0;1]  0.73 [0;1]  \n",
       "MEGAHIT            0.73 [0;1]  0.73 [0;1]  \n",
       "MetaHipMer2        0.82 [0;1]  0.82 [0;1]  \n",
       "metaSPAdes          0.6 [0;1]   0.6 [0;1]  \n",
       "MINIA               0.6 [0;1]   0.6 [0;1]  \n",
       "SKESA               0.6 [0;1]   0.6 [0;1]  \n",
       "SPAdes             0.73 [0;1]  0.73 [0;1]  \n",
       "Unicycler           0.2 [0;1]   0.2 [0;1]  \n",
       "VelvetOptimiser     0.4 [0;1]   0.4 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Escherichia_coli\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>62.42 [0;133]</td>\n",
       "      <td>0.03 [0;0]</td>\n",
       "      <td>43541.17 [0;100101]</td>\n",
       "      <td>43308.58 [0;100101]</td>\n",
       "      <td>11.25 [0;45]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3125898.33 [0;4706885]</td>\n",
       "      <td>0.66 [0;0]</td>\n",
       "      <td>111.75 [0;234]</td>\n",
       "      <td>0.67 [0;0]</td>\n",
       "      <td>0.58 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.68 [0;1]</td>\n",
       "      <td>0.68 [0;1]</td>\n",
       "      <td>0.67 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1341.8 [0;4465]</td>\n",
       "      <td>937.8 [0;3200]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1371834.67 [0;3766804]</td>\n",
       "      <td>0.29 [0;0]</td>\n",
       "      <td>523.0 [0;1495]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>62.2 [0;112]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>35065.27 [0;59656]</td>\n",
       "      <td>34770.2 [0;59530]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2775301.33 [0;4632389]</td>\n",
       "      <td>0.58 [0;0]</td>\n",
       "      <td>113.13 [0;199]</td>\n",
       "      <td>0.73 [0;0]</td>\n",
       "      <td>0.72 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.74 [0;1]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>149.6 [0;283]</td>\n",
       "      <td>0.01 [0;0]</td>\n",
       "      <td>16529.8 [1146;30548]</td>\n",
       "      <td>11945.6 [0;22537]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2723855.87 [2287;4589511]</td>\n",
       "      <td>0.57 [0;0]</td>\n",
       "      <td>251.73 [2;449]</td>\n",
       "      <td>0.96 [0;0]</td>\n",
       "      <td>0.49 [0;0]</td>\n",
       "      <td>3.8 [0;8]</td>\n",
       "      <td>6.73 [0;15]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.64 [1;5]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>30.4 [0;55]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>48259.0 [1049;90017]</td>\n",
       "      <td>50420.4 [0;102135]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2803150.47 [4237;4673019]</td>\n",
       "      <td>0.59 [0;0]</td>\n",
       "      <td>84.67 [3;141]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.91 [0;0]</td>\n",
       "      <td>2.0 [0;4]</td>\n",
       "      <td>3.4 [0;7]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.01 [0;1]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>59.18 [0;110]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>32175.36 [1155;59122]</td>\n",
       "      <td>30858.0 [0;56794]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2550895.64 [1352;4613132]</td>\n",
       "      <td>0.54 [0;0]</td>\n",
       "      <td>134.0 [1;193]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>32.4 [0;90]</td>\n",
       "      <td>0.05 [0;0]</td>\n",
       "      <td>62955.07 [1088;138793]</td>\n",
       "      <td>61514.6 [0;138793]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2794750.93 [4384;4666787]</td>\n",
       "      <td>0.59 [0;0]</td>\n",
       "      <td>69.4 [3;167]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>0.92 [0;0]</td>\n",
       "      <td>0.73 [0;2]</td>\n",
       "      <td>1.93 [0;6]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.01 [0;1]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.01 [0;0]</td>\n",
       "      <td>2668.33 [0;4469]</td>\n",
       "      <td>1897.13 [0;3211]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2258385.33 [0;3773077]</td>\n",
       "      <td>0.47 [0;0]</td>\n",
       "      <td>678.2 [0;1147]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>0.59 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>70.2 [0;152]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>27914.6 [0;62623]</td>\n",
       "      <td>28329.8 [0;66235]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2743117.0 [0;4579934]</td>\n",
       "      <td>0.58 [0;0]</td>\n",
       "      <td>131.8 [0;270]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>0.42 [0;0]</td>\n",
       "      <td>1.0 [0;3]</td>\n",
       "      <td>1.6 [0;4]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>34.6 [0;79]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>50754.93 [1094;98812]</td>\n",
       "      <td>50237.4 [0;96999]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2783937.53 [6019;4643850]</td>\n",
       "      <td>0.58 [0;0]</td>\n",
       "      <td>74.87 [4;143]</td>\n",
       "      <td>0.97 [0;0]</td>\n",
       "      <td>0.74 [0;0]</td>\n",
       "      <td>0.8 [0;3]</td>\n",
       "      <td>1.47 [0;5]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.03 [0;1]</td>\n",
       "      <td>0.97 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>27.2 [0;57]</td>\n",
       "      <td>0.05 [0;0]</td>\n",
       "      <td>65418.0 [0;124073]</td>\n",
       "      <td>64069.2 [0;119183]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2771130.4 [0;4632276]</td>\n",
       "      <td>0.58 [0;0]</td>\n",
       "      <td>56.2 [0;111]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>0.59 [0;0]</td>\n",
       "      <td>0.2 [0;1]</td>\n",
       "      <td>0.4 [0;2]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>2003.4 [0;4598]</td>\n",
       "      <td>1339.6 [0;3363]</td>\n",
       "      <td>404.4 [0;6066]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1524975.47 [0;3810326]</td>\n",
       "      <td>0.32 [0;0]</td>\n",
       "      <td>440.47 [0;1098]</td>\n",
       "      <td>0.52 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.27 [0;4]</td>\n",
       "      <td>0.6 [0;9]</td>\n",
       "      <td>0.54 [0;1]</td>\n",
       "      <td>0.54 [0;1]</td>\n",
       "      <td>0.53 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm            L90         LSA  \\\n",
       "Assembler                                                                      \n",
       "ABySS              Multiple k-mer De Bruijn graph  62.42 [0;133]  0.03 [0;0]   \n",
       "BCALM2               Single k-mer De Bruijn graph      0.0 [0;0]   0.0 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph   62.2 [0;112]  0.02 [0;0]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  149.6 [0;283]  0.01 [0;0]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph    30.4 [0;55]  0.04 [0;0]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph  59.18 [0;110]  0.02 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph    32.4 [0;90]  0.05 [0;0]   \n",
       "MINIA                Single k-mer De Bruijn graph      0.0 [0;0]  0.01 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph   70.2 [0;152]  0.02 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph    34.6 [0;79]  0.04 [0;0]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph    27.2 [0;57]  0.05 [0;0]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph      0.0 [0;0]   0.0 [0;0]   \n",
       "\n",
       "                                     NA50                 NG50  \\\n",
       "Assembler                                                        \n",
       "ABySS                 43541.17 [0;100101]  43308.58 [0;100101]   \n",
       "BCALM2                    1341.8 [0;4465]       937.8 [0;3200]   \n",
       "GATBMiniaPipeline      35065.27 [0;59656]    34770.2 [0;59530]   \n",
       "IDBA-UD              16529.8 [1146;30548]    11945.6 [0;22537]   \n",
       "MEGAHIT              48259.0 [1049;90017]   50420.4 [0;102135]   \n",
       "MetaHipMer2         32175.36 [1155;59122]    30858.0 [0;56794]   \n",
       "metaSPAdes         62955.07 [1088;138793]   61514.6 [0;138793]   \n",
       "MINIA                    2668.33 [0;4469]     1897.13 [0;3211]   \n",
       "SKESA                   27914.6 [0;62623]    28329.8 [0;66235]   \n",
       "SPAdes              50754.93 [1094;98812]    50237.4 [0;96999]   \n",
       "Unicycler              65418.0 [0;124073]   64069.2 [0;119183]   \n",
       "VelvetOptimiser           2003.4 [0;4598]      1339.6 [0;3363]   \n",
       "\n",
       "                               Ns         Type                  basepairs  \\\n",
       "Assembler                                                                   \n",
       "ABySS                11.25 [0;45]  Metagenomic     3125898.33 [0;4706885]   \n",
       "BCALM2                  0.0 [0;0]      Genomic     1371834.67 [0;3766804]   \n",
       "GATBMiniaPipeline       0.0 [0;0]  Metagenomic     2775301.33 [0;4632389]   \n",
       "IDBA-UD                 0.0 [0;0]  Metagenomic  2723855.87 [2287;4589511]   \n",
       "MEGAHIT                 0.0 [0;0]  Metagenomic  2803150.47 [4237;4673019]   \n",
       "MetaHipMer2             0.0 [0;0]  Metagenomic  2550895.64 [1352;4613132]   \n",
       "metaSPAdes              0.0 [0;0]  Metagenomic  2794750.93 [4384;4666787]   \n",
       "MINIA                   0.0 [0;0]      Genomic     2258385.33 [0;3773077]   \n",
       "SKESA                   0.0 [0;0]      Genomic      2743117.0 [0;4579934]   \n",
       "SPAdes                  0.0 [0;0]      Genomic  2783937.53 [6019;4643850]   \n",
       "Unicycler               0.0 [0;0]      Genomic      2771130.4 [0;4632276]   \n",
       "VelvetOptimiser    404.4 [0;6066]  Metagenomic     1524975.47 [0;3810326]   \n",
       "\n",
       "                  breadth_of_coverage          contigs    identity  \\\n",
       "Assembler                                                            \n",
       "ABySS                      0.66 [0;0]   111.75 [0;234]  0.67 [0;0]   \n",
       "BCALM2                     0.29 [0;0]   523.0 [0;1495]   0.4 [0;1]   \n",
       "GATBMiniaPipeline          0.58 [0;0]   113.13 [0;199]  0.73 [0;0]   \n",
       "IDBA-UD                    0.57 [0;0]   251.73 [2;449]  0.96 [0;0]   \n",
       "MEGAHIT                    0.59 [0;0]    84.67 [3;141]  0.99 [0;0]   \n",
       "MetaHipMer2                0.54 [0;0]    134.0 [1;193]   1.0 [0;1]   \n",
       "metaSPAdes                 0.59 [0;0]     69.4 [3;167]  0.98 [0;0]   \n",
       "MINIA                      0.47 [0;0]   678.2 [0;1147]   0.6 [0;0]   \n",
       "SKESA                      0.58 [0;0]    131.8 [0;270]   0.6 [0;0]   \n",
       "SPAdes                     0.58 [0;0]    74.87 [4;143]  0.97 [0;0]   \n",
       "Unicycler                  0.58 [0;0]     56.2 [0;111]   0.6 [0;0]   \n",
       "VelvetOptimiser            0.32 [0;0]  440.47 [0;1098]  0.52 [0;1]   \n",
       "\n",
       "                  lowest_identity misassembled contigs misassembly events  \\\n",
       "Assembler                                                                   \n",
       "ABySS                  0.58 [0;0]            0.0 [0;0]          0.0 [0;0]   \n",
       "BCALM2                  0.4 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "GATBMiniaPipeline      0.72 [0;0]            0.0 [0;0]          0.0 [0;0]   \n",
       "IDBA-UD                0.49 [0;0]            3.8 [0;8]        6.73 [0;15]   \n",
       "MEGAHIT                0.91 [0;0]            2.0 [0;4]          3.4 [0;7]   \n",
       "MetaHipMer2            0.99 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "metaSPAdes             0.92 [0;0]           0.73 [0;2]         1.93 [0;6]   \n",
       "MINIA                  0.59 [0;0]            0.0 [0;0]          0.0 [0;0]   \n",
       "SKESA                  0.42 [0;0]            1.0 [0;3]          1.6 [0;4]   \n",
       "SPAdes                 0.74 [0;0]            0.8 [0;3]         1.47 [0;5]   \n",
       "Unicycler              0.59 [0;0]            0.2 [0;1]          0.4 [0;2]   \n",
       "VelvetOptimiser         0.5 [0;1]           0.27 [0;4]          0.6 [0;9]   \n",
       "\n",
       "                  multiplicity   parsimony    validity  \n",
       "Assembler                                               \n",
       "ABySS               0.68 [0;1]  0.68 [0;1]  0.67 [0;0]  \n",
       "BCALM2               0.4 [0;1]   0.4 [0;1]   0.4 [0;1]  \n",
       "GATBMiniaPipeline   0.73 [0;1]  0.74 [0;1]  0.73 [0;1]  \n",
       "IDBA-UD              1.0 [0;1]  1.64 [1;5]  0.87 [0;1]  \n",
       "MEGAHIT              1.0 [0;1]  1.01 [0;1]  0.99 [0;0]  \n",
       "MetaHipMer2          1.0 [1;1]   1.0 [1;1]   1.0 [0;1]  \n",
       "metaSPAdes           1.0 [0;1]  1.01 [0;1]  0.99 [0;1]  \n",
       "MINIA                0.6 [0;1]   0.6 [0;1]   0.6 [0;1]  \n",
       "SKESA                0.6 [0;1]   0.6 [0;1]   0.6 [0;0]  \n",
       "SPAdes               1.0 [0;1]  1.03 [0;1]  0.97 [0;0]  \n",
       "Unicycler            0.6 [0;1]   0.6 [0;1]   0.6 [0;1]  \n",
       "VelvetOptimiser     0.54 [0;1]  0.54 [0;1]  0.53 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Lactobacillus_fermentum\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>38.5 [0;106]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>24508.5 [0;42368]</td>\n",
       "      <td>24303.58 [0;42368]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1229075.58 [0;1873053]</td>\n",
       "      <td>0.65 [0;0]</td>\n",
       "      <td>58.83 [0;137]</td>\n",
       "      <td>0.67 [0;0]</td>\n",
       "      <td>0.66 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.68 [0;1]</td>\n",
       "      <td>0.68 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>41.6 [0;208]</td>\n",
       "      <td>0.01 [0;0]</td>\n",
       "      <td>2403.27 [0;11209]</td>\n",
       "      <td>2046.8 [0;10234]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>350032.73 [0;1748625]</td>\n",
       "      <td>0.18 [0;0]</td>\n",
       "      <td>47.47 [0;236]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>32.4 [0;55]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>24567.33 [0;40488]</td>\n",
       "      <td>23554.2 [0;40151]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1078082.93 [0;1798516]</td>\n",
       "      <td>0.57 [0;0]</td>\n",
       "      <td>52.47 [0;89]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>49.0 [0;131]</td>\n",
       "      <td>0.03 [0;0]</td>\n",
       "      <td>19620.4 [0;40364]</td>\n",
       "      <td>18106.6 [0;37467]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1072046.93 [0;1790018]</td>\n",
       "      <td>0.56 [0;0]</td>\n",
       "      <td>69.87 [0;171]</td>\n",
       "      <td>0.73 [0;0]</td>\n",
       "      <td>0.73 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>25.6 [0;53]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>26859.4 [0;46932]</td>\n",
       "      <td>27325.6 [0;52972]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1085207.6 [0;1808544]</td>\n",
       "      <td>0.57 [0;0]</td>\n",
       "      <td>50.47 [0;88]</td>\n",
       "      <td>0.73 [0;0]</td>\n",
       "      <td>0.66 [0;0]</td>\n",
       "      <td>1.4 [0;3]</td>\n",
       "      <td>3.13 [0;6]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.73 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>29.73 [0;55]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>22337.18 [0;40362]</td>\n",
       "      <td>21854.73 [0;40067]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>977484.91 [0;1787622]</td>\n",
       "      <td>0.51 [0;0]</td>\n",
       "      <td>49.36 [0;87]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>32.0 [0;64]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>26123.67 [0;46069]</td>\n",
       "      <td>23903.0 [0;45077]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1076339.8 [0;1796429]</td>\n",
       "      <td>0.56 [0;0]</td>\n",
       "      <td>51.27 [0;95]</td>\n",
       "      <td>0.87 [0;0]</td>\n",
       "      <td>0.86 [0;0]</td>\n",
       "      <td>0.2 [0;1]</td>\n",
       "      <td>0.2 [0;1]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>135.07 [0;263]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>6518.53 [0;11293]</td>\n",
       "      <td>5976.2 [0;10836]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1048619.47 [0;1754083]</td>\n",
       "      <td>0.55 [0;0]</td>\n",
       "      <td>151.53 [0;282]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.72 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>37.8 [0;69]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>22395.8 [0;40164]</td>\n",
       "      <td>20728.6 [0;39869]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1061666.2 [0;1776186]</td>\n",
       "      <td>0.56 [0;0]</td>\n",
       "      <td>53.2 [0;94]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>36.2 [0;76]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>23888.6 [1408;46123]</td>\n",
       "      <td>22329.4 [0;45008]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1074757.8 [1368;1792828]</td>\n",
       "      <td>0.56 [0;0]</td>\n",
       "      <td>58.27 [1;109]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>0.8 [0;4]</td>\n",
       "      <td>1.6 [0;8]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>31.0 [0;55]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>26162.8 [0;45959]</td>\n",
       "      <td>24141.8 [0;40306]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1064324.2 [0;1775427]</td>\n",
       "      <td>0.56 [0;0]</td>\n",
       "      <td>47.2 [0;80]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>81.0 [0;207]</td>\n",
       "      <td>0.01 [0;0]</td>\n",
       "      <td>4863.6 [0;11784]</td>\n",
       "      <td>4416.2 [0;11122]</td>\n",
       "      <td>6.53 [0;98]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>700557.47 [0;1751517]</td>\n",
       "      <td>0.37 [0;0]</td>\n",
       "      <td>92.67 [0;235]</td>\n",
       "      <td>0.53 [0;1]</td>\n",
       "      <td>0.53 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.53 [0;1]</td>\n",
       "      <td>0.54 [0;1]</td>\n",
       "      <td>0.53 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm             L90         LSA  \\\n",
       "Assembler                                                                       \n",
       "ABySS              Multiple k-mer De Bruijn graph    38.5 [0;106]  0.04 [0;0]   \n",
       "BCALM2               Single k-mer De Bruijn graph    41.6 [0;208]  0.01 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph     32.4 [0;55]  0.04 [0;0]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph    49.0 [0;131]  0.03 [0;0]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph     25.6 [0;53]  0.04 [0;0]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph    29.73 [0;55]  0.04 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph     32.0 [0;64]  0.04 [0;0]   \n",
       "MINIA                Single k-mer De Bruijn graph  135.07 [0;263]  0.02 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph     37.8 [0;69]  0.04 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph     36.2 [0;76]  0.04 [0;0]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph     31.0 [0;55]  0.04 [0;0]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph    81.0 [0;207]  0.01 [0;0]   \n",
       "\n",
       "                                   NA50                NG50           Ns  \\\n",
       "Assembler                                                                  \n",
       "ABySS                 24508.5 [0;42368]  24303.58 [0;42368]    0.0 [0;0]   \n",
       "BCALM2                2403.27 [0;11209]    2046.8 [0;10234]    0.0 [0;0]   \n",
       "GATBMiniaPipeline    24567.33 [0;40488]   23554.2 [0;40151]    0.0 [0;0]   \n",
       "IDBA-UD               19620.4 [0;40364]   18106.6 [0;37467]    0.0 [0;0]   \n",
       "MEGAHIT               26859.4 [0;46932]   27325.6 [0;52972]    0.0 [0;0]   \n",
       "MetaHipMer2          22337.18 [0;40362]  21854.73 [0;40067]    0.0 [0;0]   \n",
       "metaSPAdes           26123.67 [0;46069]   23903.0 [0;45077]    0.0 [0;0]   \n",
       "MINIA                 6518.53 [0;11293]    5976.2 [0;10836]    0.0 [0;0]   \n",
       "SKESA                 22395.8 [0;40164]   20728.6 [0;39869]    0.0 [0;0]   \n",
       "SPAdes             23888.6 [1408;46123]   22329.4 [0;45008]    0.0 [0;0]   \n",
       "Unicycler             26162.8 [0;45959]   24141.8 [0;40306]    0.0 [0;0]   \n",
       "VelvetOptimiser        4863.6 [0;11784]    4416.2 [0;11122]  6.53 [0;98]   \n",
       "\n",
       "                          Type                 basepairs breadth_of_coverage  \\\n",
       "Assembler                                                                      \n",
       "ABySS              Metagenomic    1229075.58 [0;1873053]          0.65 [0;0]   \n",
       "BCALM2                 Genomic     350032.73 [0;1748625]          0.18 [0;0]   \n",
       "GATBMiniaPipeline  Metagenomic    1078082.93 [0;1798516]          0.57 [0;0]   \n",
       "IDBA-UD            Metagenomic    1072046.93 [0;1790018]          0.56 [0;0]   \n",
       "MEGAHIT            Metagenomic     1085207.6 [0;1808544]          0.57 [0;0]   \n",
       "MetaHipMer2        Metagenomic     977484.91 [0;1787622]          0.51 [0;0]   \n",
       "metaSPAdes         Metagenomic     1076339.8 [0;1796429]          0.56 [0;0]   \n",
       "MINIA                  Genomic    1048619.47 [0;1754083]          0.55 [0;0]   \n",
       "SKESA                  Genomic     1061666.2 [0;1776186]          0.56 [0;0]   \n",
       "SPAdes                 Genomic  1074757.8 [1368;1792828]          0.56 [0;0]   \n",
       "Unicycler              Genomic     1064324.2 [0;1775427]          0.56 [0;0]   \n",
       "VelvetOptimiser    Metagenomic     700557.47 [0;1751517]          0.37 [0;0]   \n",
       "\n",
       "                          contigs    identity lowest_identity  \\\n",
       "Assembler                                                       \n",
       "ABySS               58.83 [0;137]  0.67 [0;0]      0.66 [0;0]   \n",
       "BCALM2              47.47 [0;236]  0.33 [0;1]      0.33 [0;1]   \n",
       "GATBMiniaPipeline    52.47 [0;89]  0.73 [0;1]      0.73 [0;1]   \n",
       "IDBA-UD             69.87 [0;171]  0.73 [0;0]      0.73 [0;0]   \n",
       "MEGAHIT              50.47 [0;88]  0.73 [0;0]      0.66 [0;0]   \n",
       "MetaHipMer2          49.36 [0;87]  0.82 [0;1]      0.82 [0;1]   \n",
       "metaSPAdes           51.27 [0;95]  0.87 [0;0]      0.86 [0;0]   \n",
       "MINIA              151.53 [0;282]  0.73 [0;1]      0.72 [0;1]   \n",
       "SKESA                 53.2 [0;94]   0.6 [0;1]       0.6 [0;1]   \n",
       "SPAdes              58.27 [1;109]  0.99 [0;0]      0.98 [0;0]   \n",
       "Unicycler             47.2 [0;80]   0.6 [0;1]       0.6 [0;1]   \n",
       "VelvetOptimiser     92.67 [0;235]  0.53 [0;1]      0.53 [0;1]   \n",
       "\n",
       "                  misassembled contigs misassembly events multiplicity  \\\n",
       "Assembler                                                                \n",
       "ABySS                        0.0 [0;0]          0.0 [0;0]   0.68 [0;1]   \n",
       "BCALM2                       0.0 [0;0]          0.0 [0;0]   0.33 [0;1]   \n",
       "GATBMiniaPipeline            0.0 [0;0]          0.0 [0;0]   0.73 [0;1]   \n",
       "IDBA-UD                      0.0 [0;0]          0.0 [0;0]   0.73 [0;1]   \n",
       "MEGAHIT                      1.4 [0;3]         3.13 [0;6]   0.73 [0;1]   \n",
       "MetaHipMer2                  0.0 [0;0]          0.0 [0;0]   0.82 [0;1]   \n",
       "metaSPAdes                   0.2 [0;1]          0.2 [0;1]   0.87 [0;1]   \n",
       "MINIA                        0.0 [0;0]          0.0 [0;0]   0.73 [0;1]   \n",
       "SKESA                        0.0 [0;0]          0.0 [0;0]    0.6 [0;1]   \n",
       "SPAdes                       0.8 [0;4]          1.6 [0;8]   0.99 [0;1]   \n",
       "Unicycler                    0.0 [0;0]          0.0 [0;0]    0.6 [0;1]   \n",
       "VelvetOptimiser              0.0 [0;0]          0.0 [0;0]   0.53 [0;1]   \n",
       "\n",
       "                    parsimony    validity  \n",
       "Assembler                                  \n",
       "ABySS              0.68 [0;1]  0.67 [0;1]  \n",
       "BCALM2             0.33 [0;1]  0.33 [0;1]  \n",
       "GATBMiniaPipeline  0.73 [0;1]  0.73 [0;1]  \n",
       "IDBA-UD            0.73 [0;1]  0.73 [0;1]  \n",
       "MEGAHIT            0.73 [0;1]  0.73 [0;0]  \n",
       "MetaHipMer2        0.82 [0;1]  0.82 [0;1]  \n",
       "metaSPAdes         0.87 [0;1]  0.87 [0;1]  \n",
       "MINIA              0.73 [0;1]  0.73 [0;1]  \n",
       "SKESA               0.6 [0;1]   0.6 [0;1]  \n",
       "SPAdes             1.01 [1;1]  0.99 [0;1]  \n",
       "Unicycler           0.6 [0;1]   0.6 [0;1]  \n",
       "VelvetOptimiser    0.54 [0;1]  0.53 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Listeria_monocytogenes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>35.17 [8;147]</td>\n",
       "      <td>0.18 [0;0]</td>\n",
       "      <td>253854.25 [21261;351511]</td>\n",
       "      <td>253672.92 [20173;351511]</td>\n",
       "      <td>0.25 [0;1]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2955765.83 [2941416;2961911]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>55.42 [13;226]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>21.33 [0;75]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>18158.53 [0;64777]</td>\n",
       "      <td>17203.0 [0;62493]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1060470.27 [0;2921297]</td>\n",
       "      <td>0.35 [0;0]</td>\n",
       "      <td>108.67 [0;359]</td>\n",
       "      <td>0.53 [0;1]</td>\n",
       "      <td>0.53 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.53 [0;1]</td>\n",
       "      <td>0.53 [0;1]</td>\n",
       "      <td>0.53 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>10.87 [10;12]</td>\n",
       "      <td>0.2 [0;0]</td>\n",
       "      <td>358527.73 [254631;452364]</td>\n",
       "      <td>299075.93 [155105;351186]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2944550.0 [2915611;2957674]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>20.93 [18;26]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>30.8 [9;108]</td>\n",
       "      <td>0.16 [0;0]</td>\n",
       "      <td>251447.4 [27259;475377]</td>\n",
       "      <td>251106.6 [25555;475377]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2938972.07 [2915464;2956853]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>52.8 [19;171]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>7.6 [6;9]</td>\n",
       "      <td>0.26 [0;0]</td>\n",
       "      <td>455760.2 [269400;543128]</td>\n",
       "      <td>456450.07 [274574;543128]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2951515.87 [2924395;2962451]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>19.53 [17;26]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.83 [0;0]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>0.67 [0;2]</td>\n",
       "      <td>1.01 [0;1]</td>\n",
       "      <td>1.01 [0;1]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>7.09 [0;13]</td>\n",
       "      <td>0.09 [0;0]</td>\n",
       "      <td>123175.09 [1221;254627]</td>\n",
       "      <td>122798.09 [0;254627]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2197140.64 [450667;2949910]</td>\n",
       "      <td>0.73 [0;0]</td>\n",
       "      <td>305.55 [21;1074]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.97 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>8.93 [5;13]</td>\n",
       "      <td>0.22 [0;0]</td>\n",
       "      <td>372172.07 [235115;593167]</td>\n",
       "      <td>372172.07 [235115;593167]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2945879.0 [2920851;2964879]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>18.13 [12;27]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>122.07 [0;461]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>34150.53 [3362;64777]</td>\n",
       "      <td>32985.53 [2969;64473]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2843160.27 [2548083;2924618]</td>\n",
       "      <td>0.95 [0;0]</td>\n",
       "      <td>300.6 [87;919]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.86 [0;1]</td>\n",
       "      <td>0.13 [0;1]</td>\n",
       "      <td>0.13 [0;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>40.27 [0;232]</td>\n",
       "      <td>0.15 [0;0]</td>\n",
       "      <td>228614.27 [2534;351027]</td>\n",
       "      <td>228469.2 [1926;351027]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2823224.73 [2259852;2950392]</td>\n",
       "      <td>0.94 [0;0]</td>\n",
       "      <td>194.13 [19;1001]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>120.2 [0;835]</td>\n",
       "      <td>0.13 [0;0]</td>\n",
       "      <td>188038.87 [1964;351104]</td>\n",
       "      <td>188092.47 [1651;351104]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2726050.53 [1899833;2957750]</td>\n",
       "      <td>0.91 [0;0]</td>\n",
       "      <td>433.27 [13;1839]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.93 [0;1]</td>\n",
       "      <td>75.33 [0;565]</td>\n",
       "      <td>151.2 [0;1134]</td>\n",
       "      <td>1.05 [0;1]</td>\n",
       "      <td>1.05 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>11.13 [6;33]</td>\n",
       "      <td>0.23 [0;0]</td>\n",
       "      <td>345918.67 [126634;475179]</td>\n",
       "      <td>345918.67 [126634;475179]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2937182.33 [2914588;2953853]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>22.33 [15;64]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>37.67 [0;145]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>21138.4 [0;52654]</td>\n",
       "      <td>20739.47 [0;51199]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1528222.4 [0;2918755]</td>\n",
       "      <td>0.51 [0;0]</td>\n",
       "      <td>127.07 [0;957]</td>\n",
       "      <td>0.53 [0;1]</td>\n",
       "      <td>0.53 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.53 [0;1]</td>\n",
       "      <td>0.53 [0;1]</td>\n",
       "      <td>0.53 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm             L90         LSA  \\\n",
       "Assembler                                                                       \n",
       "ABySS              Multiple k-mer De Bruijn graph   35.17 [8;147]  0.18 [0;0]   \n",
       "BCALM2               Single k-mer De Bruijn graph    21.33 [0;75]  0.02 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph   10.87 [10;12]   0.2 [0;0]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph    30.8 [9;108]  0.16 [0;0]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph       7.6 [6;9]  0.26 [0;0]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph     7.09 [0;13]  0.09 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph     8.93 [5;13]  0.22 [0;0]   \n",
       "MINIA                Single k-mer De Bruijn graph  122.07 [0;461]  0.04 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph   40.27 [0;232]  0.15 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph   120.2 [0;835]  0.13 [0;0]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph    11.13 [6;33]  0.23 [0;0]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph   37.67 [0;145]  0.02 [0;0]   \n",
       "\n",
       "                                        NA50                       NG50  \\\n",
       "Assembler                                                                 \n",
       "ABySS               253854.25 [21261;351511]   253672.92 [20173;351511]   \n",
       "BCALM2                    18158.53 [0;64777]          17203.0 [0;62493]   \n",
       "GATBMiniaPipeline  358527.73 [254631;452364]  299075.93 [155105;351186]   \n",
       "IDBA-UD              251447.4 [27259;475377]    251106.6 [25555;475377]   \n",
       "MEGAHIT             455760.2 [269400;543128]  456450.07 [274574;543128]   \n",
       "MetaHipMer2          123175.09 [1221;254627]       122798.09 [0;254627]   \n",
       "metaSPAdes         372172.07 [235115;593167]  372172.07 [235115;593167]   \n",
       "MINIA                  34150.53 [3362;64777]      32985.53 [2969;64473]   \n",
       "SKESA                228614.27 [2534;351027]     228469.2 [1926;351027]   \n",
       "SPAdes               188038.87 [1964;351104]    188092.47 [1651;351104]   \n",
       "Unicycler          345918.67 [126634;475179]  345918.67 [126634;475179]   \n",
       "VelvetOptimiser            21138.4 [0;52654]         20739.47 [0;51199]   \n",
       "\n",
       "                           Ns         Type                     basepairs  \\\n",
       "Assembler                                                                  \n",
       "ABySS              0.25 [0;1]  Metagenomic  2955765.83 [2941416;2961911]   \n",
       "BCALM2              0.0 [0;0]      Genomic        1060470.27 [0;2921297]   \n",
       "GATBMiniaPipeline   0.0 [0;0]  Metagenomic   2944550.0 [2915611;2957674]   \n",
       "IDBA-UD             0.0 [0;0]  Metagenomic  2938972.07 [2915464;2956853]   \n",
       "MEGAHIT             0.0 [0;0]  Metagenomic  2951515.87 [2924395;2962451]   \n",
       "MetaHipMer2         0.0 [0;0]  Metagenomic   2197140.64 [450667;2949910]   \n",
       "metaSPAdes          0.0 [0;0]  Metagenomic   2945879.0 [2920851;2964879]   \n",
       "MINIA               0.0 [0;0]      Genomic  2843160.27 [2548083;2924618]   \n",
       "SKESA               0.0 [0;0]      Genomic  2823224.73 [2259852;2950392]   \n",
       "SPAdes              0.0 [0;0]      Genomic  2726050.53 [1899833;2957750]   \n",
       "Unicycler           0.0 [0;0]      Genomic  2937182.33 [2914588;2953853]   \n",
       "VelvetOptimiser     0.0 [0;0]  Metagenomic         1528222.4 [0;2918755]   \n",
       "\n",
       "                  breadth_of_coverage           contigs    identity  \\\n",
       "Assembler                                                             \n",
       "ABySS                      0.99 [0;0]    55.42 [13;226]   1.0 [0;0]   \n",
       "BCALM2                     0.35 [0;0]    108.67 [0;359]  0.53 [0;1]   \n",
       "GATBMiniaPipeline          0.98 [0;0]     20.93 [18;26]   1.0 [0;1]   \n",
       "IDBA-UD                    0.98 [0;0]     52.8 [19;171]   1.0 [0;0]   \n",
       "MEGAHIT                    0.99 [0;0]     19.53 [17;26]  0.99 [0;0]   \n",
       "MetaHipMer2                0.73 [0;0]  305.55 [21;1074]   1.0 [0;1]   \n",
       "metaSPAdes                 0.98 [0;0]     18.13 [12;27]   1.0 [0;0]   \n",
       "MINIA                      0.95 [0;0]    300.6 [87;919]   1.0 [0;1]   \n",
       "SKESA                      0.94 [0;0]  194.13 [19;1001]   1.0 [0;1]   \n",
       "SPAdes                     0.91 [0;0]  433.27 [13;1839]   1.0 [0;1]   \n",
       "Unicycler                  0.98 [0;0]     22.33 [15;64]   1.0 [0;1]   \n",
       "VelvetOptimiser            0.51 [0;0]    127.07 [0;957]  0.53 [0;1]   \n",
       "\n",
       "                  lowest_identity misassembled contigs misassembly events  \\\n",
       "Assembler                                                                   \n",
       "ABySS                   1.0 [0;0]            0.0 [0;0]          0.0 [0;0]   \n",
       "BCALM2                 0.53 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "GATBMiniaPipeline      0.99 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "IDBA-UD                0.99 [0;0]            0.0 [0;0]          0.0 [0;0]   \n",
       "MEGAHIT                0.83 [0;0]           0.33 [0;1]         0.67 [0;2]   \n",
       "MetaHipMer2            0.97 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "metaSPAdes             0.99 [0;0]            0.0 [0;0]          0.0 [0;0]   \n",
       "MINIA                  0.86 [0;1]           0.13 [0;1]         0.13 [0;1]   \n",
       "SKESA                   1.0 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "SPAdes                 0.93 [0;1]        75.33 [0;565]     151.2 [0;1134]   \n",
       "Unicycler              0.98 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "VelvetOptimiser        0.53 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "\n",
       "                  multiplicity   parsimony    validity  \n",
       "Assembler                                               \n",
       "ABySS               1.01 [1;1]  1.01 [1;1]   1.0 [0;1]  \n",
       "BCALM2              0.53 [0;1]  0.53 [0;1]  0.53 [0;1]  \n",
       "GATBMiniaPipeline    1.0 [0;1]   1.0 [0;1]   1.0 [0;1]  \n",
       "IDBA-UD              1.0 [0;1]   1.0 [0;1]   1.0 [0;1]  \n",
       "MEGAHIT             1.01 [0;1]  1.01 [0;1]   1.0 [0;0]  \n",
       "MetaHipMer2          1.0 [1;1]   1.0 [1;1]   1.0 [0;1]  \n",
       "metaSPAdes           1.0 [1;1]   1.0 [1;1]   1.0 [0;1]  \n",
       "MINIA                1.0 [1;1]   1.0 [1;1]   1.0 [0;1]  \n",
       "SKESA                1.0 [0;1]   1.0 [0;1]   1.0 [1;1]  \n",
       "SPAdes              1.05 [0;1]  1.05 [0;1]   1.0 [0;1]  \n",
       "Unicycler            1.0 [0;1]   1.0 [0;1]   1.0 [0;1]  \n",
       "VelvetOptimiser     0.53 [0;1]  0.53 [0;1]  0.53 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pseudomonas_aeruginosa\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>295.58 [0;1004]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>74053.42 [1421;245341]</td>\n",
       "      <td>74015.0 [0;245341]</td>\n",
       "      <td>617.0 [0;2170]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>5612753.83 [179111;6781605]</td>\n",
       "      <td>0.83 [0;0]</td>\n",
       "      <td>462.75 [58;1404]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.88 [0;0]</td>\n",
       "      <td>2.33 [0;8]</td>\n",
       "      <td>3.92 [0;14]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.02 [1;1]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>293.67 [0;921]</td>\n",
       "      <td>0.01 [0;0]</td>\n",
       "      <td>9221.53 [0;23566]</td>\n",
       "      <td>8785.67 [0;22511]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>3450896.67 [0;6515102]</td>\n",
       "      <td>0.51 [0;0]</td>\n",
       "      <td>398.33 [0;1142]</td>\n",
       "      <td>0.53 [0;1]</td>\n",
       "      <td>0.53 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.53 [0;1]</td>\n",
       "      <td>0.53 [0;1]</td>\n",
       "      <td>0.53 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>53.67 [36;78]</td>\n",
       "      <td>0.06 [0;0]</td>\n",
       "      <td>155708.8 [88774;211558]</td>\n",
       "      <td>154839.2 [88774;211019]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>6695432.93 [6646876;6716522]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>123.33 [98;166]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.97 [0;1]</td>\n",
       "      <td>0.6 [0;3]</td>\n",
       "      <td>1.2 [0;6]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>68.47 [48;90]</td>\n",
       "      <td>0.05 [0;0]</td>\n",
       "      <td>110417.93 [84458;171408]</td>\n",
       "      <td>107730.8 [83502;171408]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>6699911.07 [6694930;6710005]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>151.73 [119;185]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.93 [0;0]</td>\n",
       "      <td>0.73 [0;2]</td>\n",
       "      <td>1.13 [0;4]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>28.33 [24;31]</td>\n",
       "      <td>0.07 [0;0]</td>\n",
       "      <td>251345.27 [215542;286027]</td>\n",
       "      <td>253092.47 [215542;294763]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>6733131.47 [6729060;6738936]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>80.13 [75;82]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.97 [0;0]</td>\n",
       "      <td>0.2 [0;1]</td>\n",
       "      <td>0.4 [0;2]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>21.55 [0;42]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>109490.09 [1126;211554]</td>\n",
       "      <td>108943.27 [0;211015]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>4374306.0 [93618;6717767]</td>\n",
       "      <td>0.64 [0;0]</td>\n",
       "      <td>514.27 [81;2404]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>29.2 [24;44]</td>\n",
       "      <td>0.11 [0;0]</td>\n",
       "      <td>230830.27 [151807;285156]</td>\n",
       "      <td>226615.73 [151807;257869]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>6727170.67 [6721688;6733500]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>67.0 [56;100]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>337.07 [291;407]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>22111.87 [18690;24665]</td>\n",
       "      <td>21444.33 [17851;24029]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>6510535.4 [6489664;6524118]</td>\n",
       "      <td>0.96 [0;0]</td>\n",
       "      <td>519.6 [463;598]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>180.4 [0;670]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>32504.4 [0;111919]</td>\n",
       "      <td>31006.47 [0;106185]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>4660954.93 [0;6673601]</td>\n",
       "      <td>0.69 [0;0]</td>\n",
       "      <td>670.73 [0;1877]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "      <td>0.54 [0;1]</td>\n",
       "      <td>4.2 [0;12]</td>\n",
       "      <td>8.47 [0;23]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>37.13 [26;60]</td>\n",
       "      <td>0.1 [0;0]</td>\n",
       "      <td>201079.13 [122703;264371]</td>\n",
       "      <td>201079.13 [122703;264371]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>6712051.8 [6682762;6728888]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>83.4 [63;129]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>0.53 [0;2]</td>\n",
       "      <td>0.87 [0;3]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>23.6 [20;25]</td>\n",
       "      <td>0.12 [0;0]</td>\n",
       "      <td>275007.27 [256472;337495]</td>\n",
       "      <td>277936.53 [239075;337495]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>6691221.33 [6551562;6723626]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>66.8 [54;77]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.67 [0;5]</td>\n",
       "      <td>1.33 [0;10]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>5.8 [0;15]</td>\n",
       "      <td>0.03 [0;0]</td>\n",
       "      <td>121245.53 [0;338923]</td>\n",
       "      <td>161122.8 [0;464977]</td>\n",
       "      <td>3665.2 [0;12098]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3708331.07 [0;6708956]</td>\n",
       "      <td>0.55 [0;0]</td>\n",
       "      <td>499.87 [0;2156]</td>\n",
       "      <td>0.66 [0;1]</td>\n",
       "      <td>0.61 [0;1]</td>\n",
       "      <td>2.8 [0;8]</td>\n",
       "      <td>4.4 [0;11]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm               L90  \\\n",
       "Assembler                                                             \n",
       "ABySS              Multiple k-mer De Bruijn graph   295.58 [0;1004]   \n",
       "BCALM2               Single k-mer De Bruijn graph    293.67 [0;921]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph     53.67 [36;78]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph     68.47 [48;90]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph     28.33 [24;31]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph      21.55 [0;42]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph      29.2 [24;44]   \n",
       "MINIA                Single k-mer De Bruijn graph  337.07 [291;407]   \n",
       "SKESA              Multiple k-mer De Bruijn graph     180.4 [0;670]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph     37.13 [26;60]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph      23.6 [20;25]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph        5.8 [0;15]   \n",
       "\n",
       "                          LSA                       NA50  \\\n",
       "Assembler                                                  \n",
       "ABySS              0.04 [0;0]     74053.42 [1421;245341]   \n",
       "BCALM2             0.01 [0;0]          9221.53 [0;23566]   \n",
       "GATBMiniaPipeline  0.06 [0;0]    155708.8 [88774;211558]   \n",
       "IDBA-UD            0.05 [0;0]   110417.93 [84458;171408]   \n",
       "MEGAHIT            0.07 [0;0]  251345.27 [215542;286027]   \n",
       "MetaHipMer2        0.04 [0;0]    109490.09 [1126;211554]   \n",
       "metaSPAdes         0.11 [0;0]  230830.27 [151807;285156]   \n",
       "MINIA              0.02 [0;0]     22111.87 [18690;24665]   \n",
       "SKESA              0.02 [0;0]         32504.4 [0;111919]   \n",
       "SPAdes              0.1 [0;0]  201079.13 [122703;264371]   \n",
       "Unicycler          0.12 [0;0]  275007.27 [256472;337495]   \n",
       "VelvetOptimiser    0.03 [0;0]       121245.53 [0;338923]   \n",
       "\n",
       "                                        NG50                Ns         Type  \\\n",
       "Assembler                                                                     \n",
       "ABySS                     74015.0 [0;245341]    617.0 [0;2170]  Metagenomic   \n",
       "BCALM2                     8785.67 [0;22511]         0.0 [0;0]      Genomic   \n",
       "GATBMiniaPipeline    154839.2 [88774;211019]         0.0 [0;0]  Metagenomic   \n",
       "IDBA-UD              107730.8 [83502;171408]         0.0 [0;0]  Metagenomic   \n",
       "MEGAHIT            253092.47 [215542;294763]         0.0 [0;0]  Metagenomic   \n",
       "MetaHipMer2             108943.27 [0;211015]         0.0 [0;0]  Metagenomic   \n",
       "metaSPAdes         226615.73 [151807;257869]         0.0 [0;0]  Metagenomic   \n",
       "MINIA                 21444.33 [17851;24029]         0.0 [0;0]      Genomic   \n",
       "SKESA                    31006.47 [0;106185]         0.0 [0;0]      Genomic   \n",
       "SPAdes             201079.13 [122703;264371]         0.0 [0;0]      Genomic   \n",
       "Unicycler          277936.53 [239075;337495]         0.0 [0;0]      Genomic   \n",
       "VelvetOptimiser          161122.8 [0;464977]  3665.2 [0;12098]  Metagenomic   \n",
       "\n",
       "                                      basepairs breadth_of_coverage  \\\n",
       "Assembler                                                             \n",
       "ABySS               5612753.83 [179111;6781605]          0.83 [0;0]   \n",
       "BCALM2                   3450896.67 [0;6515102]          0.51 [0;0]   \n",
       "GATBMiniaPipeline  6695432.93 [6646876;6716522]          0.99 [0;0]   \n",
       "IDBA-UD            6699911.07 [6694930;6710005]          0.99 [0;0]   \n",
       "MEGAHIT            6733131.47 [6729060;6738936]          0.99 [0;0]   \n",
       "MetaHipMer2           4374306.0 [93618;6717767]          0.64 [0;0]   \n",
       "metaSPAdes         6727170.67 [6721688;6733500]          0.99 [0;0]   \n",
       "MINIA               6510535.4 [6489664;6524118]          0.96 [0;0]   \n",
       "SKESA                    4660954.93 [0;6673601]          0.69 [0;0]   \n",
       "SPAdes              6712051.8 [6682762;6728888]          0.99 [0;0]   \n",
       "Unicycler          6691221.33 [6551562;6723626]          0.99 [0;0]   \n",
       "VelvetOptimiser          3708331.07 [0;6708956]          0.55 [0;0]   \n",
       "\n",
       "                            contigs    identity lowest_identity  \\\n",
       "Assembler                                                         \n",
       "ABySS              462.75 [58;1404]   1.0 [0;0]      0.88 [0;0]   \n",
       "BCALM2              398.33 [0;1142]  0.53 [0;1]      0.53 [0;1]   \n",
       "GATBMiniaPipeline   123.33 [98;166]   1.0 [0;1]      0.97 [0;1]   \n",
       "IDBA-UD            151.73 [119;185]   1.0 [0;0]      0.93 [0;0]   \n",
       "MEGAHIT               80.13 [75;82]   1.0 [0;0]      0.97 [0;0]   \n",
       "MetaHipMer2        514.27 [81;2404]   1.0 [0;1]      0.99 [0;1]   \n",
       "metaSPAdes            67.0 [56;100]   1.0 [0;0]      0.98 [0;0]   \n",
       "MINIA               519.6 [463;598]   1.0 [0;0]      0.99 [0;0]   \n",
       "SKESA               670.73 [0;1877]  0.87 [0;1]      0.54 [0;1]   \n",
       "SPAdes                83.4 [63;129]   1.0 [0;0]      0.98 [0;0]   \n",
       "Unicycler              66.8 [54;77]   1.0 [0;1]      0.95 [0;1]   \n",
       "VelvetOptimiser     499.87 [0;2156]  0.66 [0;1]      0.61 [0;1]   \n",
       "\n",
       "                  misassembled contigs misassembly events multiplicity  \\\n",
       "Assembler                                                                \n",
       "ABySS                       2.33 [0;8]        3.92 [0;14]   1.01 [1;1]   \n",
       "BCALM2                       0.0 [0;0]          0.0 [0;0]   0.53 [0;1]   \n",
       "GATBMiniaPipeline            0.6 [0;3]          1.2 [0;6]    1.0 [1;1]   \n",
       "IDBA-UD                     0.73 [0;2]         1.13 [0;4]    1.0 [1;1]   \n",
       "MEGAHIT                      0.2 [0;1]          0.4 [0;2]    1.0 [1;1]   \n",
       "MetaHipMer2                  0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "metaSPAdes                   0.0 [0;0]          0.0 [0;0]   1.01 [1;1]   \n",
       "MINIA                        0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "SKESA                       4.2 [0;12]        8.47 [0;23]   0.87 [0;1]   \n",
       "SPAdes                      0.53 [0;2]         0.87 [0;3]    1.0 [0;1]   \n",
       "Unicycler                   0.67 [0;5]        1.33 [0;10]    1.0 [0;1]   \n",
       "VelvetOptimiser              2.8 [0;8]         4.4 [0;11]   0.67 [0;1]   \n",
       "\n",
       "                    parsimony    validity  \n",
       "Assembler                                  \n",
       "ABySS              1.02 [1;1]   1.0 [0;0]  \n",
       "BCALM2             0.53 [0;1]  0.53 [0;1]  \n",
       "GATBMiniaPipeline   1.0 [1;1]   1.0 [0;1]  \n",
       "IDBA-UD             1.0 [1;1]   1.0 [0;1]  \n",
       "MEGAHIT             1.0 [1;1]   1.0 [0;1]  \n",
       "MetaHipMer2         1.0 [1;1]   1.0 [0;1]  \n",
       "metaSPAdes         1.01 [1;1]   1.0 [0;1]  \n",
       "MINIA               1.0 [1;1]   1.0 [0;1]  \n",
       "SKESA              0.87 [0;1]  0.87 [0;1]  \n",
       "SPAdes              1.0 [0;1]   1.0 [0;1]  \n",
       "Unicycler           1.0 [0;1]   1.0 [0;1]  \n",
       "VelvetOptimiser    0.67 [0;1]  0.67 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salmonella_enterica\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>38.08 [0;118]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>76069.5 [0;157988]</td>\n",
       "      <td>75866.08 [0;157988]</td>\n",
       "      <td>1.08 [0;2]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3127326.33 [0;4703394]</td>\n",
       "      <td>0.66 [0;0]</td>\n",
       "      <td>71.92 [0;214]</td>\n",
       "      <td>0.67 [0;0]</td>\n",
       "      <td>0.62 [0;0]</td>\n",
       "      <td>0.17 [0;1]</td>\n",
       "      <td>0.5 [0;4]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1465.2 [0;5009]</td>\n",
       "      <td>1010.0 [0;3551]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1383060.2 [0;3817265]</td>\n",
       "      <td>0.29 [0;0]</td>\n",
       "      <td>511.6 [0;1479]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>52.6 [0;96]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>35662.87 [0;64572]</td>\n",
       "      <td>34177.0 [0;60374]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2788737.47 [0;4652574]</td>\n",
       "      <td>0.59 [0;0]</td>\n",
       "      <td>100.93 [0;178]</td>\n",
       "      <td>0.87 [0;0]</td>\n",
       "      <td>0.78 [0;0]</td>\n",
       "      <td>0.2 [0;1]</td>\n",
       "      <td>0.2 [0;1]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>109.33 [0;200]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>16812.53 [1117;29331]</td>\n",
       "      <td>16476.2 [0;28547]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2752150.13 [1463;4618887]</td>\n",
       "      <td>0.58 [0;0]</td>\n",
       "      <td>219.07 [1;397]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>0.68 [0;0]</td>\n",
       "      <td>3.6 [0;9]</td>\n",
       "      <td>5.67 [0;14]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.02 [1;1]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>26.4 [0;47]</td>\n",
       "      <td>0.06 [0;0]</td>\n",
       "      <td>66355.13 [1086;120785]</td>\n",
       "      <td>66523.4 [0;120785]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2806238.73 [4780;4671620]</td>\n",
       "      <td>0.59 [0;0]</td>\n",
       "      <td>65.2 [2;105]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.93 [0;0]</td>\n",
       "      <td>0.8 [0;3]</td>\n",
       "      <td>1.6 [0;6]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.02 [1;1]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>46.91 [0;86]</td>\n",
       "      <td>0.03 [0;0]</td>\n",
       "      <td>34361.55 [0;64529]</td>\n",
       "      <td>32532.82 [0;60329]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2559742.73 [0;4641887]</td>\n",
       "      <td>0.54 [0;0]</td>\n",
       "      <td>111.55 [0;162]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "      <td>0.79 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>26.0 [0;77]</td>\n",
       "      <td>0.05 [0;0]</td>\n",
       "      <td>87215.33 [1223;198784]</td>\n",
       "      <td>85403.8 [0;198784]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2806770.87 [2586;4680545]</td>\n",
       "      <td>0.59 [0;0]</td>\n",
       "      <td>55.33 [2;145]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>0.85 [0;0]</td>\n",
       "      <td>0.13 [0;1]</td>\n",
       "      <td>0.13 [0;1]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>1.02 [0;1]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.01 [0;0]</td>\n",
       "      <td>3501.47 [1313;4996]</td>\n",
       "      <td>2091.4 [0;3551]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2291336.87 [1312;3826699]</td>\n",
       "      <td>0.48 [0;0]</td>\n",
       "      <td>654.8 [1;1106]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.96 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>67.2 [0;144]</td>\n",
       "      <td>0.03 [0;0]</td>\n",
       "      <td>29176.2 [0;60912]</td>\n",
       "      <td>28863.4 [0;60131]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2751052.6 [0;4612070]</td>\n",
       "      <td>0.58 [0;0]</td>\n",
       "      <td>119.0 [0;245]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>0.37 [0;0]</td>\n",
       "      <td>0.8 [0;2]</td>\n",
       "      <td>1.0 [0;3]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>24.6 [0;50]</td>\n",
       "      <td>0.06 [0;0]</td>\n",
       "      <td>75469.27 [1251;164404]</td>\n",
       "      <td>73530.2 [0;157954]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2794301.87 [1463;4666050]</td>\n",
       "      <td>0.59 [0;0]</td>\n",
       "      <td>59.0 [1;110]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>0.76 [0;0]</td>\n",
       "      <td>0.47 [0;2]</td>\n",
       "      <td>0.47 [0;2]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.03 [1;1]</td>\n",
       "      <td>0.97 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>17.4 [0;35]</td>\n",
       "      <td>0.06 [0;0]</td>\n",
       "      <td>107912.4 [0;223960]</td>\n",
       "      <td>107360.0 [0;223960]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2789200.4 [0;4660116]</td>\n",
       "      <td>0.59 [0;0]</td>\n",
       "      <td>41.8 [0;89]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>2313.6 [0;5223]</td>\n",
       "      <td>1498.0 [0;3753]</td>\n",
       "      <td>231.73 [0;3476]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1547544.4 [0;3867436]</td>\n",
       "      <td>0.33 [0;0]</td>\n",
       "      <td>422.87 [0;1058]</td>\n",
       "      <td>0.59 [0;1]</td>\n",
       "      <td>0.54 [0;1]</td>\n",
       "      <td>0.13 [0;2]</td>\n",
       "      <td>0.13 [0;2]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.62 [0;1]</td>\n",
       "      <td>0.59 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm             L90         LSA  \\\n",
       "Assembler                                                                       \n",
       "ABySS              Multiple k-mer De Bruijn graph   38.08 [0;118]  0.04 [0;0]   \n",
       "BCALM2               Single k-mer De Bruijn graph       0.0 [0;0]   0.0 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph     52.6 [0;96]  0.04 [0;0]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  109.33 [0;200]  0.02 [0;0]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph     26.4 [0;47]  0.06 [0;0]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph    46.91 [0;86]  0.03 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph     26.0 [0;77]  0.05 [0;0]   \n",
       "MINIA                Single k-mer De Bruijn graph       0.0 [0;0]  0.01 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph    67.2 [0;144]  0.03 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph     24.6 [0;50]  0.06 [0;0]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph     17.4 [0;35]  0.06 [0;0]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph       0.0 [0;0]   0.0 [0;0]   \n",
       "\n",
       "                                     NA50                 NG50  \\\n",
       "Assembler                                                        \n",
       "ABySS                  76069.5 [0;157988]  75866.08 [0;157988]   \n",
       "BCALM2                    1465.2 [0;5009]      1010.0 [0;3551]   \n",
       "GATBMiniaPipeline      35662.87 [0;64572]    34177.0 [0;60374]   \n",
       "IDBA-UD             16812.53 [1117;29331]    16476.2 [0;28547]   \n",
       "MEGAHIT            66355.13 [1086;120785]   66523.4 [0;120785]   \n",
       "MetaHipMer2            34361.55 [0;64529]   32532.82 [0;60329]   \n",
       "metaSPAdes         87215.33 [1223;198784]   85403.8 [0;198784]   \n",
       "MINIA                 3501.47 [1313;4996]      2091.4 [0;3551]   \n",
       "SKESA                   29176.2 [0;60912]    28863.4 [0;60131]   \n",
       "SPAdes             75469.27 [1251;164404]   73530.2 [0;157954]   \n",
       "Unicycler             107912.4 [0;223960]  107360.0 [0;223960]   \n",
       "VelvetOptimiser           2313.6 [0;5223]      1498.0 [0;3753]   \n",
       "\n",
       "                                Ns         Type                  basepairs  \\\n",
       "Assembler                                                                    \n",
       "ABySS                   1.08 [0;2]  Metagenomic     3127326.33 [0;4703394]   \n",
       "BCALM2                   0.0 [0;0]      Genomic      1383060.2 [0;3817265]   \n",
       "GATBMiniaPipeline        0.0 [0;0]  Metagenomic     2788737.47 [0;4652574]   \n",
       "IDBA-UD                  0.0 [0;0]  Metagenomic  2752150.13 [1463;4618887]   \n",
       "MEGAHIT                  0.0 [0;0]  Metagenomic  2806238.73 [4780;4671620]   \n",
       "MetaHipMer2              0.0 [0;0]  Metagenomic     2559742.73 [0;4641887]   \n",
       "metaSPAdes               0.0 [0;0]  Metagenomic  2806770.87 [2586;4680545]   \n",
       "MINIA                    0.0 [0;0]      Genomic  2291336.87 [1312;3826699]   \n",
       "SKESA                    0.0 [0;0]      Genomic      2751052.6 [0;4612070]   \n",
       "SPAdes                   0.0 [0;0]      Genomic  2794301.87 [1463;4666050]   \n",
       "Unicycler                0.0 [0;0]      Genomic      2789200.4 [0;4660116]   \n",
       "VelvetOptimiser    231.73 [0;3476]  Metagenomic      1547544.4 [0;3867436]   \n",
       "\n",
       "                  breadth_of_coverage          contigs    identity  \\\n",
       "Assembler                                                            \n",
       "ABySS                      0.66 [0;0]    71.92 [0;214]  0.67 [0;0]   \n",
       "BCALM2                     0.29 [0;0]   511.6 [0;1479]   0.4 [0;1]   \n",
       "GATBMiniaPipeline          0.59 [0;0]   100.93 [0;178]  0.87 [0;0]   \n",
       "IDBA-UD                    0.58 [0;0]   219.07 [1;397]  0.98 [0;0]   \n",
       "MEGAHIT                    0.59 [0;0]     65.2 [2;105]  0.99 [0;0]   \n",
       "MetaHipMer2                0.54 [0;0]   111.55 [0;162]  0.82 [0;1]   \n",
       "metaSPAdes                 0.59 [0;0]    55.33 [2;145]  0.98 [0;0]   \n",
       "MINIA                      0.48 [0;0]   654.8 [1;1106]   1.0 [0;1]   \n",
       "SKESA                      0.58 [0;0]    119.0 [0;245]   0.6 [0;0]   \n",
       "SPAdes                     0.59 [0;0]     59.0 [1;110]  0.98 [0;0]   \n",
       "Unicycler                  0.59 [0;0]      41.8 [0;89]   0.6 [0;0]   \n",
       "VelvetOptimiser            0.33 [0;0]  422.87 [0;1058]  0.59 [0;1]   \n",
       "\n",
       "                  lowest_identity misassembled contigs misassembly events  \\\n",
       "Assembler                                                                   \n",
       "ABySS                  0.62 [0;0]           0.17 [0;1]          0.5 [0;4]   \n",
       "BCALM2                  0.4 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "GATBMiniaPipeline      0.78 [0;0]            0.2 [0;1]          0.2 [0;1]   \n",
       "IDBA-UD                0.68 [0;0]            3.6 [0;9]        5.67 [0;14]   \n",
       "MEGAHIT                0.93 [0;0]            0.8 [0;3]          1.6 [0;6]   \n",
       "MetaHipMer2            0.79 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "metaSPAdes             0.85 [0;0]           0.13 [0;1]         0.13 [0;1]   \n",
       "MINIA                  0.96 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "SKESA                  0.37 [0;0]            0.8 [0;2]          1.0 [0;3]   \n",
       "SPAdes                 0.76 [0;0]           0.47 [0;2]         0.47 [0;2]   \n",
       "Unicycler               0.6 [0;0]            0.0 [0;0]          0.0 [0;0]   \n",
       "VelvetOptimiser        0.54 [0;1]           0.13 [0;2]         0.13 [0;2]   \n",
       "\n",
       "                  multiplicity   parsimony    validity  \n",
       "Assembler                                               \n",
       "ABySS               0.67 [0;1]  0.67 [0;1]  0.67 [0;0]  \n",
       "BCALM2               0.4 [0;1]   0.4 [0;1]   0.4 [0;1]  \n",
       "GATBMiniaPipeline   0.87 [0;1]  0.87 [0;1]  0.87 [0;1]  \n",
       "IDBA-UD              1.0 [0;1]  1.02 [1;1]  0.98 [0;0]  \n",
       "MEGAHIT              1.0 [0;1]  1.02 [1;1]  0.99 [0;0]  \n",
       "MetaHipMer2         0.82 [0;1]  0.82 [0;1]  0.82 [0;1]  \n",
       "metaSPAdes          0.99 [0;1]  1.02 [0;1]  0.98 [0;1]  \n",
       "MINIA                1.0 [1;1]   1.0 [1;1]   1.0 [0;1]  \n",
       "SKESA                0.6 [0;1]   0.6 [0;1]   0.6 [0;0]  \n",
       "SPAdes               1.0 [0;1]  1.03 [1;1]  0.97 [0;0]  \n",
       "Unicycler            0.6 [0;1]   0.6 [0;1]   0.6 [0;1]  \n",
       "VelvetOptimiser      0.6 [0;1]  0.62 [0;1]  0.59 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Staphylococcus_aureus\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>14.17 [0;48]</td>\n",
       "      <td>0.18 [0;0]</td>\n",
       "      <td>146373.58 [0;281333]</td>\n",
       "      <td>146373.58 [0;281333]</td>\n",
       "      <td>17.75 [0;71]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1792639.0 [0;2693907]</td>\n",
       "      <td>0.66 [0;0]</td>\n",
       "      <td>25.17 [0;80]</td>\n",
       "      <td>0.67 [0;0]</td>\n",
       "      <td>0.66 [0;0]</td>\n",
       "      <td>0.17 [0;1]</td>\n",
       "      <td>0.33 [0;2]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>22.4 [0;112]</td>\n",
       "      <td>0.01 [0;0]</td>\n",
       "      <td>6162.2 [0;29647]</td>\n",
       "      <td>5833.0 [0;29165]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>546308.2 [0;2600423]</td>\n",
       "      <td>0.2 [0;0]</td>\n",
       "      <td>56.8 [0;175]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>19.2 [0;39]</td>\n",
       "      <td>0.07 [0;0]</td>\n",
       "      <td>53510.0 [0;112739]</td>\n",
       "      <td>50514.0 [0;108199]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1605229.13 [0;2679055]</td>\n",
       "      <td>0.59 [0;0]</td>\n",
       "      <td>31.8 [0;65]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>39.4 [0;112]</td>\n",
       "      <td>0.03 [0;0]</td>\n",
       "      <td>26606.6 [0;53248]</td>\n",
       "      <td>26311.0 [0;53248]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1598390.8 [0;2665812]</td>\n",
       "      <td>0.59 [0;0]</td>\n",
       "      <td>64.0 [0;174]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>0.51 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>9.6 [0;19]</td>\n",
       "      <td>0.09 [0;0]</td>\n",
       "      <td>84298.6 [0;181811]</td>\n",
       "      <td>98279.6 [0;213210]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1608652.4 [0;2681291]</td>\n",
       "      <td>0.59 [0;0]</td>\n",
       "      <td>24.0 [0;43]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>0.51 [0;0]</td>\n",
       "      <td>1.4 [0;3]</td>\n",
       "      <td>3.2 [0;7]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>22.73 [0;42]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>29395.0 [0;53350]</td>\n",
       "      <td>28681.55 [0;53350]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1537141.0 [0;2667635]</td>\n",
       "      <td>0.57 [0;0]</td>\n",
       "      <td>103.55 [0;247]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>10.2 [0;30]</td>\n",
       "      <td>0.13 [0;0]</td>\n",
       "      <td>128284.4 [0;360908]</td>\n",
       "      <td>128284.4 [0;360908]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1607602.2 [0;2682662]</td>\n",
       "      <td>0.59 [0;0]</td>\n",
       "      <td>18.2 [0;49]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>72.73 [0;140]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>16545.4 [0;29647]</td>\n",
       "      <td>16263.2 [0;29165]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1559474.6 [0;2600576]</td>\n",
       "      <td>0.57 [0;0]</td>\n",
       "      <td>111.33 [0;207]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>24.6 [0;43]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>35869.4 [0;73147]</td>\n",
       "      <td>33913.0 [0;63469]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1593903.4 [0;2661968]</td>\n",
       "      <td>0.59 [0;0]</td>\n",
       "      <td>41.0 [0;73]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>11.0 [0;30]</td>\n",
       "      <td>0.1 [0;0]</td>\n",
       "      <td>113799.0 [0;280950]</td>\n",
       "      <td>113799.0 [0;280950]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1604400.4 [0;2678585]</td>\n",
       "      <td>0.59 [0;0]</td>\n",
       "      <td>18.6 [0;48]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>0.59 [0;0]</td>\n",
       "      <td>0.2 [0;1]</td>\n",
       "      <td>0.6 [0;3]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>5.4 [0;14]</td>\n",
       "      <td>0.22 [0;0]</td>\n",
       "      <td>197639.8 [0;394879]</td>\n",
       "      <td>185790.4 [0;394879]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1604154.4 [0;2678241]</td>\n",
       "      <td>0.59 [0;0]</td>\n",
       "      <td>13.0 [0;25]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>0.2 [0;1]</td>\n",
       "      <td>0.4 [0;2]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>43.4 [0;113]</td>\n",
       "      <td>0.01 [0;0]</td>\n",
       "      <td>12442.4 [0;32297]</td>\n",
       "      <td>11508.2 [0;30086]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1041153.4 [0;2603433]</td>\n",
       "      <td>0.38 [0;0]</td>\n",
       "      <td>67.8 [0;175]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm            L90         LSA  \\\n",
       "Assembler                                                                      \n",
       "ABySS              Multiple k-mer De Bruijn graph   14.17 [0;48]  0.18 [0;0]   \n",
       "BCALM2               Single k-mer De Bruijn graph   22.4 [0;112]  0.01 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph    19.2 [0;39]  0.07 [0;0]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph   39.4 [0;112]  0.03 [0;0]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph     9.6 [0;19]  0.09 [0;0]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph   22.73 [0;42]  0.04 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph    10.2 [0;30]  0.13 [0;0]   \n",
       "MINIA                Single k-mer De Bruijn graph  72.73 [0;140]  0.02 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph    24.6 [0;43]  0.04 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph    11.0 [0;30]   0.1 [0;0]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph     5.4 [0;14]  0.22 [0;0]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph   43.4 [0;113]  0.01 [0;0]   \n",
       "\n",
       "                                   NA50                  NG50            Ns  \\\n",
       "Assembler                                                                     \n",
       "ABySS              146373.58 [0;281333]  146373.58 [0;281333]  17.75 [0;71]   \n",
       "BCALM2                 6162.2 [0;29647]      5833.0 [0;29165]     0.0 [0;0]   \n",
       "GATBMiniaPipeline    53510.0 [0;112739]    50514.0 [0;108199]     0.0 [0;0]   \n",
       "IDBA-UD               26606.6 [0;53248]     26311.0 [0;53248]     0.0 [0;0]   \n",
       "MEGAHIT              84298.6 [0;181811]    98279.6 [0;213210]     0.0 [0;0]   \n",
       "MetaHipMer2           29395.0 [0;53350]    28681.55 [0;53350]     0.0 [0;0]   \n",
       "metaSPAdes          128284.4 [0;360908]   128284.4 [0;360908]     0.0 [0;0]   \n",
       "MINIA                 16545.4 [0;29647]     16263.2 [0;29165]     0.0 [0;0]   \n",
       "SKESA                 35869.4 [0;73147]     33913.0 [0;63469]     0.0 [0;0]   \n",
       "SPAdes              113799.0 [0;280950]   113799.0 [0;280950]     0.0 [0;0]   \n",
       "Unicycler           197639.8 [0;394879]   185790.4 [0;394879]     0.0 [0;0]   \n",
       "VelvetOptimiser       12442.4 [0;32297]     11508.2 [0;30086]     0.0 [0;0]   \n",
       "\n",
       "                          Type               basepairs breadth_of_coverage  \\\n",
       "Assembler                                                                    \n",
       "ABySS              Metagenomic   1792639.0 [0;2693907]          0.66 [0;0]   \n",
       "BCALM2                 Genomic    546308.2 [0;2600423]           0.2 [0;0]   \n",
       "GATBMiniaPipeline  Metagenomic  1605229.13 [0;2679055]          0.59 [0;0]   \n",
       "IDBA-UD            Metagenomic   1598390.8 [0;2665812]          0.59 [0;0]   \n",
       "MEGAHIT            Metagenomic   1608652.4 [0;2681291]          0.59 [0;0]   \n",
       "MetaHipMer2        Metagenomic   1537141.0 [0;2667635]          0.57 [0;0]   \n",
       "metaSPAdes         Metagenomic   1607602.2 [0;2682662]          0.59 [0;0]   \n",
       "MINIA                  Genomic   1559474.6 [0;2600576]          0.57 [0;0]   \n",
       "SKESA                  Genomic   1593903.4 [0;2661968]          0.59 [0;0]   \n",
       "SPAdes                 Genomic   1604400.4 [0;2678585]          0.59 [0;0]   \n",
       "Unicycler              Genomic   1604154.4 [0;2678241]          0.59 [0;0]   \n",
       "VelvetOptimiser    Metagenomic   1041153.4 [0;2603433]          0.38 [0;0]   \n",
       "\n",
       "                          contigs    identity lowest_identity  \\\n",
       "Assembler                                                       \n",
       "ABySS                25.17 [0;80]  0.67 [0;0]      0.66 [0;0]   \n",
       "BCALM2               56.8 [0;175]   0.4 [0;1]       0.4 [0;1]   \n",
       "GATBMiniaPipeline     31.8 [0;65]   0.6 [0;1]       0.6 [0;1]   \n",
       "IDBA-UD              64.0 [0;174]   0.6 [0;0]      0.51 [0;0]   \n",
       "MEGAHIT               24.0 [0;43]   0.6 [0;0]      0.51 [0;0]   \n",
       "MetaHipMer2        103.55 [0;247]  0.82 [0;1]      0.82 [0;1]   \n",
       "metaSPAdes            18.2 [0;49]   0.6 [0;0]       0.6 [0;0]   \n",
       "MINIA              111.33 [0;207]   0.6 [0;0]       0.6 [0;0]   \n",
       "SKESA                 41.0 [0;73]   0.6 [0;1]       0.6 [0;1]   \n",
       "SPAdes                18.6 [0;48]   0.6 [0;0]      0.59 [0;0]   \n",
       "Unicycler             13.0 [0;25]   0.6 [0;0]       0.6 [0;0]   \n",
       "VelvetOptimiser      67.8 [0;175]   0.4 [0;1]       0.4 [0;1]   \n",
       "\n",
       "                  misassembled contigs misassembly events multiplicity  \\\n",
       "Assembler                                                                \n",
       "ABySS                       0.17 [0;1]         0.33 [0;2]   0.67 [0;1]   \n",
       "BCALM2                       0.0 [0;0]          0.0 [0;0]    0.4 [0;1]   \n",
       "GATBMiniaPipeline            0.0 [0;0]          0.0 [0;0]    0.6 [0;1]   \n",
       "IDBA-UD                      0.0 [0;0]          0.0 [0;0]    0.6 [0;1]   \n",
       "MEGAHIT                      1.4 [0;3]          3.2 [0;7]    0.6 [0;1]   \n",
       "MetaHipMer2                  0.0 [0;0]          0.0 [0;0]   0.82 [0;1]   \n",
       "metaSPAdes                   0.0 [0;0]          0.0 [0;0]    0.6 [0;1]   \n",
       "MINIA                        0.0 [0;0]          0.0 [0;0]    0.6 [0;1]   \n",
       "SKESA                        0.0 [0;0]          0.0 [0;0]    0.6 [0;1]   \n",
       "SPAdes                       0.2 [0;1]          0.6 [0;3]    0.6 [0;1]   \n",
       "Unicycler                    0.2 [0;1]          0.4 [0;2]    0.6 [0;0]   \n",
       "VelvetOptimiser              0.0 [0;0]          0.0 [0;0]    0.4 [0;1]   \n",
       "\n",
       "                    parsimony    validity  \n",
       "Assembler                                  \n",
       "ABySS              0.67 [0;1]  0.67 [0;1]  \n",
       "BCALM2              0.4 [0;1]   0.4 [0;1]  \n",
       "GATBMiniaPipeline   0.6 [0;1]   0.6 [0;1]  \n",
       "IDBA-UD             0.6 [0;1]   0.6 [0;1]  \n",
       "MEGAHIT             0.6 [0;1]   0.6 [0;0]  \n",
       "MetaHipMer2        0.82 [0;1]  0.82 [0;1]  \n",
       "metaSPAdes          0.6 [0;1]   0.6 [0;1]  \n",
       "MINIA               0.6 [0;1]   0.6 [0;0]  \n",
       "SKESA               0.6 [0;1]   0.6 [0;1]  \n",
       "SPAdes              0.6 [0;1]   0.6 [0;1]  \n",
       "Unicycler           0.6 [0;0]   0.6 [0;1]  \n",
       "VelvetOptimiser     0.4 [0;1]   0.4 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Staphylococcus_aureus_plasmid1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.5 [0;0]</td>\n",
       "      <td>3167.25 [0;6335]</td>\n",
       "      <td>3167.25 [0;6335]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3167.25 [0;6335]</td>\n",
       "      <td>0.5 [0;0]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.2 [0;0]</td>\n",
       "      <td>1267.2 [0;6336]</td>\n",
       "      <td>1267.2 [0;6336]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1267.2 [0;6336]</td>\n",
       "      <td>0.2 [0;0]</td>\n",
       "      <td>0.2 [0;1]</td>\n",
       "      <td>0.2 [0;1]</td>\n",
       "      <td>0.2 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.2 [0;1]</td>\n",
       "      <td>0.2 [0;1]</td>\n",
       "      <td>0.2 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>3837.27 [0;6522]</td>\n",
       "      <td>3837.27 [0;6522]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3801.4 [0;6337]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.61 [0;1]</td>\n",
       "      <td>0.61 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.55 [0;0]</td>\n",
       "      <td>3487.4 [0;6336]</td>\n",
       "      <td>3487.4 [0;6336]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3481.4 [0;6336]</td>\n",
       "      <td>0.55 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>3887.8 [0;6795]</td>\n",
       "      <td>3887.8 [0;6795]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3796.2 [0;6337]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.59 [0;1]</td>\n",
       "      <td>0.59 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.61 [0;1]</td>\n",
       "      <td>0.59 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.55 [0;0]</td>\n",
       "      <td>3454.91 [0;6334]</td>\n",
       "      <td>3454.91 [0;6334]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3454.91 [0;6334]</td>\n",
       "      <td>0.55 [0;0]</td>\n",
       "      <td>0.55 [0;1]</td>\n",
       "      <td>0.55 [0;1]</td>\n",
       "      <td>0.55 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.55 [0;1]</td>\n",
       "      <td>0.55 [0;1]</td>\n",
       "      <td>0.55 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.2 [0;1]</td>\n",
       "      <td>0.53 [0;0]</td>\n",
       "      <td>3381.8 [0;6336]</td>\n",
       "      <td>3381.8 [0;6336]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3775.2 [0;6336]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>0.8 [0;2]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.4 [0;0]</td>\n",
       "      <td>2528.6 [0;6336]</td>\n",
       "      <td>2528.6 [0;6336]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2528.6 [0;6336]</td>\n",
       "      <td>0.4 [0;0]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.43 [0;0]</td>\n",
       "      <td>2736.6 [0;6044]</td>\n",
       "      <td>2736.6 [0;6044]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>3374.8 [0;6044]</td>\n",
       "      <td>0.53 [0;0]</td>\n",
       "      <td>1.0 [0;3]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.59 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.4 [0;0]</td>\n",
       "      <td>2534.4 [0;6336]</td>\n",
       "      <td>2534.4 [0;6336]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2534.4 [0;6336]</td>\n",
       "      <td>0.4 [0;0]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.59 [0;1]</td>\n",
       "      <td>3768.2 [0;6367]</td>\n",
       "      <td>3768.2 [0;6367]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>3762.2 [0;6337]</td>\n",
       "      <td>0.59 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.4 [0;0]</td>\n",
       "      <td>2534.4 [0;6336]</td>\n",
       "      <td>2534.4 [0;6336]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2534.4 [0;6336]</td>\n",
       "      <td>0.4 [0;0]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm        L90         LSA  \\\n",
       "Assembler                                                                  \n",
       "ABySS              Multiple k-mer De Bruijn graph  0.0 [0;0]   0.5 [0;0]   \n",
       "BCALM2               Single k-mer De Bruijn graph  0.0 [0;0]   0.2 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph  0.0 [0;0]   0.6 [0;1]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  0.0 [0;0]  0.55 [0;0]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph  0.0 [0;0]   0.6 [0;1]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph  0.0 [0;0]  0.55 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph  0.2 [0;1]  0.53 [0;0]   \n",
       "MINIA                Single k-mer De Bruijn graph  0.0 [0;0]   0.4 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph  0.0 [0;0]  0.43 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph  0.0 [0;0]   0.4 [0;0]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph  0.0 [0;0]  0.59 [0;1]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph  0.0 [0;0]   0.4 [0;0]   \n",
       "\n",
       "                               NA50              NG50         Ns         Type  \\\n",
       "Assembler                                                                       \n",
       "ABySS              3167.25 [0;6335]  3167.25 [0;6335]  0.0 [0;0]  Metagenomic   \n",
       "BCALM2              1267.2 [0;6336]   1267.2 [0;6336]  0.0 [0;0]      Genomic   \n",
       "GATBMiniaPipeline  3837.27 [0;6522]  3837.27 [0;6522]  0.0 [0;0]  Metagenomic   \n",
       "IDBA-UD             3487.4 [0;6336]   3487.4 [0;6336]  0.0 [0;0]  Metagenomic   \n",
       "MEGAHIT             3887.8 [0;6795]   3887.8 [0;6795]  0.0 [0;0]  Metagenomic   \n",
       "MetaHipMer2        3454.91 [0;6334]  3454.91 [0;6334]  0.0 [0;0]  Metagenomic   \n",
       "metaSPAdes          3381.8 [0;6336]   3381.8 [0;6336]  0.0 [0;0]  Metagenomic   \n",
       "MINIA               2528.6 [0;6336]   2528.6 [0;6336]  0.0 [0;0]      Genomic   \n",
       "SKESA               2736.6 [0;6044]   2736.6 [0;6044]  0.0 [0;0]      Genomic   \n",
       "SPAdes              2534.4 [0;6336]   2534.4 [0;6336]  0.0 [0;0]      Genomic   \n",
       "Unicycler           3768.2 [0;6367]   3768.2 [0;6367]  0.0 [0;0]      Genomic   \n",
       "VelvetOptimiser     2534.4 [0;6336]   2534.4 [0;6336]  0.0 [0;0]  Metagenomic   \n",
       "\n",
       "                          basepairs breadth_of_coverage     contigs  \\\n",
       "Assembler                                                             \n",
       "ABySS              3167.25 [0;6335]           0.5 [0;0]   0.5 [0;1]   \n",
       "BCALM2              1267.2 [0;6336]           0.2 [0;0]   0.2 [0;1]   \n",
       "GATBMiniaPipeline   3801.4 [0;6337]           0.6 [0;1]   0.6 [0;1]   \n",
       "IDBA-UD             3481.4 [0;6336]          0.55 [0;0]   0.6 [0;1]   \n",
       "MEGAHIT             3796.2 [0;6337]           0.6 [0;1]   0.6 [0;1]   \n",
       "MetaHipMer2        3454.91 [0;6334]          0.55 [0;0]  0.55 [0;1]   \n",
       "metaSPAdes          3775.2 [0;6336]           0.6 [0;0]   0.8 [0;2]   \n",
       "MINIA               2528.6 [0;6336]           0.4 [0;0]   0.4 [0;1]   \n",
       "SKESA               3374.8 [0;6044]          0.53 [0;0]   1.0 [0;3]   \n",
       "SPAdes              2534.4 [0;6336]           0.4 [0;0]   0.4 [0;1]   \n",
       "Unicycler           3762.2 [0;6337]          0.59 [0;1]   0.6 [0;1]   \n",
       "VelvetOptimiser     2534.4 [0;6336]           0.4 [0;0]   0.4 [0;1]   \n",
       "\n",
       "                     identity lowest_identity misassembled contigs  \\\n",
       "Assembler                                                            \n",
       "ABySS               0.5 [0;1]       0.5 [0;1]            0.0 [0;0]   \n",
       "BCALM2              0.2 [0;1]       0.2 [0;1]            0.0 [0;0]   \n",
       "GATBMiniaPipeline   0.6 [0;1]       0.6 [0;1]            0.0 [0;0]   \n",
       "IDBA-UD             0.6 [0;1]       0.6 [0;1]            0.0 [0;0]   \n",
       "MEGAHIT            0.59 [0;1]      0.59 [0;1]            0.0 [0;0]   \n",
       "MetaHipMer2        0.55 [0;1]      0.55 [0;1]            0.0 [0;0]   \n",
       "metaSPAdes          0.6 [0;1]       0.6 [0;1]            0.0 [0;0]   \n",
       "MINIA               0.4 [0;1]       0.4 [0;1]            0.0 [0;0]   \n",
       "SKESA               0.6 [0;1]      0.59 [0;1]            0.0 [0;0]   \n",
       "SPAdes              0.4 [0;1]       0.4 [0;1]            0.0 [0;0]   \n",
       "Unicycler           0.6 [0;1]       0.6 [0;1]            0.0 [0;0]   \n",
       "VelvetOptimiser     0.4 [0;1]       0.4 [0;1]            0.0 [0;0]   \n",
       "\n",
       "                  misassembly events multiplicity   parsimony    validity  \n",
       "Assembler                                                                  \n",
       "ABySS                      0.0 [0;0]    0.5 [0;1]   0.5 [0;1]   0.5 [0;1]  \n",
       "BCALM2                     0.0 [0;0]    0.2 [0;1]   0.2 [0;1]   0.2 [0;1]  \n",
       "GATBMiniaPipeline          0.0 [0;0]   0.61 [0;1]  0.61 [0;1]   0.6 [0;1]  \n",
       "IDBA-UD                    0.0 [0;0]    0.6 [0;1]   0.6 [0;1]   0.6 [0;1]  \n",
       "MEGAHIT                    0.0 [0;0]    0.6 [0;1]  0.61 [0;1]  0.59 [0;1]  \n",
       "MetaHipMer2                0.0 [0;0]   0.55 [0;1]  0.55 [0;1]  0.55 [0;1]  \n",
       "metaSPAdes                 0.0 [0;0]    0.6 [0;1]   0.6 [0;1]   0.6 [0;1]  \n",
       "MINIA                      0.0 [0;0]    0.4 [0;1]   0.4 [0;1]   0.4 [0;1]  \n",
       "SKESA                      0.0 [0;0]    0.6 [0;1]   0.6 [0;1]   0.6 [0;1]  \n",
       "SPAdes                     0.0 [0;0]    0.4 [0;1]   0.4 [0;1]   0.4 [0;1]  \n",
       "Unicycler                  0.0 [0;0]    0.6 [0;1]   0.6 [0;1]   0.6 [0;1]  \n",
       "VelvetOptimiser            0.0 [0;0]    0.4 [0;1]   0.4 [0;1]   0.4 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Staphylococcus_aureus_plasmid2\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>1336.67 [0;2215]</td>\n",
       "      <td>1336.67 [0;2215]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1336.67 [0;2215]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.61 [0;1]</td>\n",
       "      <td>1355.2 [0;2397]</td>\n",
       "      <td>1355.2 [0;2397]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1321.4 [0;2216]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.62 [0;1]</td>\n",
       "      <td>0.62 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.41 [0;1]</td>\n",
       "      <td>906.0 [0;2315]</td>\n",
       "      <td>906.0 [0;2315]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>886.2 [0;2216]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.41 [0;1]</td>\n",
       "      <td>0.41 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.61 [0;1]</td>\n",
       "      <td>1357.4 [0;2357]</td>\n",
       "      <td>1357.4 [0;2357]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1329.2 [0;2216]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.61 [0;1]</td>\n",
       "      <td>0.61 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.54 [0;0]</td>\n",
       "      <td>1207.09 [0;2213]</td>\n",
       "      <td>1207.09 [0;2213]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1207.09 [0;2213]</td>\n",
       "      <td>0.54 [0;0]</td>\n",
       "      <td>0.55 [0;1]</td>\n",
       "      <td>0.55 [0;1]</td>\n",
       "      <td>0.55 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.55 [0;1]</td>\n",
       "      <td>0.55 [0;1]</td>\n",
       "      <td>0.55 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "      <td>1284.6 [0;2271]</td>\n",
       "      <td>1284.6 [0;2271]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1273.6 [0;2216]</td>\n",
       "      <td>0.57 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.48 [0;0]</td>\n",
       "      <td>1074.4 [0;2005]</td>\n",
       "      <td>1074.4 [0;2005]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1074.4 [0;2005]</td>\n",
       "      <td>0.48 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.52 [0;1]</td>\n",
       "      <td>1357.2 [0;3246]</td>\n",
       "      <td>1357.2 [0;3246]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1151.2 [0;2216]</td>\n",
       "      <td>0.52 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.54 [0;1]</td>\n",
       "      <td>0.54 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.69 [0;1]</td>\n",
       "      <td>0.54 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "      <td>1289.6 [0;2216]</td>\n",
       "      <td>1289.6 [0;2216]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1289.6 [0;2216]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm        L90         LSA  \\\n",
       "Assembler                                                                  \n",
       "ABySS              Multiple k-mer De Bruijn graph  0.0 [0;0]   0.6 [0;0]   \n",
       "BCALM2               Single k-mer De Bruijn graph  0.0 [0;0]   0.0 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph  0.0 [0;0]  0.61 [0;1]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  0.0 [0;0]  0.41 [0;1]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph  0.0 [0;0]  0.61 [0;1]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph  0.0 [0;0]  0.54 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph  0.0 [0;0]  0.58 [0;1]   \n",
       "MINIA                Single k-mer De Bruijn graph  0.0 [0;0]   0.0 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph  0.0 [0;0]  0.48 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph  0.0 [0;0]  0.52 [0;1]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph  0.0 [0;0]  0.58 [0;1]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph  0.0 [0;0]   0.0 [0;0]   \n",
       "\n",
       "                               NA50              NG50         Ns         Type  \\\n",
       "Assembler                                                                       \n",
       "ABySS              1336.67 [0;2215]  1336.67 [0;2215]  0.0 [0;0]  Metagenomic   \n",
       "BCALM2                    0.0 [0;0]         0.0 [0;0]  0.0 [0;0]      Genomic   \n",
       "GATBMiniaPipeline   1355.2 [0;2397]   1355.2 [0;2397]  0.0 [0;0]  Metagenomic   \n",
       "IDBA-UD              906.0 [0;2315]    906.0 [0;2315]  0.0 [0;0]  Metagenomic   \n",
       "MEGAHIT             1357.4 [0;2357]   1357.4 [0;2357]  0.0 [0;0]  Metagenomic   \n",
       "MetaHipMer2        1207.09 [0;2213]  1207.09 [0;2213]  0.0 [0;0]  Metagenomic   \n",
       "metaSPAdes          1284.6 [0;2271]   1284.6 [0;2271]  0.0 [0;0]  Metagenomic   \n",
       "MINIA                     0.0 [0;0]         0.0 [0;0]  0.0 [0;0]      Genomic   \n",
       "SKESA               1074.4 [0;2005]   1074.4 [0;2005]  0.0 [0;0]      Genomic   \n",
       "SPAdes              1357.2 [0;3246]   1357.2 [0;3246]  0.0 [0;0]      Genomic   \n",
       "Unicycler           1289.6 [0;2216]   1289.6 [0;2216]  0.0 [0;0]      Genomic   \n",
       "VelvetOptimiser           0.0 [0;0]         0.0 [0;0]  0.0 [0;0]  Metagenomic   \n",
       "\n",
       "                          basepairs breadth_of_coverage     contigs  \\\n",
       "Assembler                                                             \n",
       "ABySS              1336.67 [0;2215]           0.6 [0;0]  0.67 [0;1]   \n",
       "BCALM2                    0.0 [0;0]           0.0 [0;0]   0.0 [0;0]   \n",
       "GATBMiniaPipeline   1321.4 [0;2216]           0.6 [0;1]   0.6 [0;1]   \n",
       "IDBA-UD              886.2 [0;2216]           0.4 [0;1]   0.4 [0;1]   \n",
       "MEGAHIT             1329.2 [0;2216]           0.6 [0;1]   0.6 [0;1]   \n",
       "MetaHipMer2        1207.09 [0;2213]          0.54 [0;0]  0.55 [0;1]   \n",
       "metaSPAdes          1273.6 [0;2216]          0.57 [0;1]   0.6 [0;1]   \n",
       "MINIA                     0.0 [0;0]           0.0 [0;0]   0.0 [0;0]   \n",
       "SKESA               1074.4 [0;2005]          0.48 [0;0]   0.6 [0;1]   \n",
       "SPAdes              1151.2 [0;2216]          0.52 [0;1]   0.6 [0;1]   \n",
       "Unicycler           1289.6 [0;2216]          0.58 [0;1]   0.6 [0;1]   \n",
       "VelvetOptimiser           0.0 [0;0]           0.0 [0;0]   0.0 [0;0]   \n",
       "\n",
       "                     identity lowest_identity misassembled contigs  \\\n",
       "Assembler                                                            \n",
       "ABySS              0.67 [0;1]      0.67 [0;1]            0.0 [0;0]   \n",
       "BCALM2              0.0 [0;0]       0.0 [0;0]            0.0 [0;0]   \n",
       "GATBMiniaPipeline   0.6 [0;1]       0.6 [0;1]            0.0 [0;0]   \n",
       "IDBA-UD             0.4 [0;1]       0.4 [0;1]            0.0 [0;0]   \n",
       "MEGAHIT             0.6 [0;1]       0.6 [0;1]            0.0 [0;0]   \n",
       "MetaHipMer2        0.55 [0;1]      0.55 [0;1]            0.0 [0;0]   \n",
       "metaSPAdes          0.6 [0;0]       0.6 [0;0]            0.0 [0;0]   \n",
       "MINIA               0.0 [0;0]       0.0 [0;0]            0.0 [0;0]   \n",
       "SKESA               0.6 [0;1]       0.6 [0;1]            0.0 [0;0]   \n",
       "SPAdes             0.54 [0;1]      0.54 [0;1]            0.0 [0;0]   \n",
       "Unicycler           0.6 [0;1]       0.6 [0;1]            0.0 [0;0]   \n",
       "VelvetOptimiser     0.0 [0;0]       0.0 [0;0]            0.0 [0;0]   \n",
       "\n",
       "                  misassembly events multiplicity   parsimony    validity  \n",
       "Assembler                                                                  \n",
       "ABySS                      0.0 [0;0]   0.67 [0;1]  0.67 [0;1]  0.67 [0;1]  \n",
       "BCALM2                     0.0 [0;0]    0.0 [0;0]   0.0 [0;0]   0.0 [0;0]  \n",
       "GATBMiniaPipeline          0.0 [0;0]   0.62 [0;1]  0.62 [0;1]   0.6 [0;1]  \n",
       "IDBA-UD                    0.0 [0;0]   0.41 [0;1]  0.41 [0;1]   0.4 [0;1]  \n",
       "MEGAHIT                    0.0 [0;0]   0.61 [0;1]  0.61 [0;1]   0.6 [0;1]  \n",
       "MetaHipMer2                0.0 [0;0]   0.55 [0;1]  0.55 [0;1]  0.55 [0;1]  \n",
       "metaSPAdes                 0.0 [0;0]    0.6 [0;1]   0.6 [0;1]   0.6 [0;1]  \n",
       "MINIA                      0.0 [0;0]    0.0 [0;0]   0.0 [0;0]   0.0 [0;0]  \n",
       "SKESA                      0.0 [0;0]    0.6 [0;1]   0.6 [0;1]   0.6 [0;1]  \n",
       "SPAdes                     0.0 [0;0]    0.6 [0;1]  0.69 [0;1]  0.54 [0;1]  \n",
       "Unicycler                  0.0 [0;0]    0.6 [0;1]   0.6 [0;1]   0.6 [0;1]  \n",
       "VelvetOptimiser            0.0 [0;0]    0.0 [0;0]   0.0 [0;0]   0.0 [0;0]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Staphylococcus_aureus_plasmid3\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.61 [0;0]</td>\n",
       "      <td>1821.58 [0;2992]</td>\n",
       "      <td>1821.58 [0;2992]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1821.58 [0;2992]</td>\n",
       "      <td>0.61 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>1785.8 [0;3162]</td>\n",
       "      <td>1785.8 [0;3162]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1754.8 [0;2993]</td>\n",
       "      <td>0.59 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.61 [0;1]</td>\n",
       "      <td>0.61 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.2 [0;1]</td>\n",
       "      <td>0.49 [0;0]</td>\n",
       "      <td>1476.4 [0;2847]</td>\n",
       "      <td>1476.4 [0;2847]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1690.2 [0;2847]</td>\n",
       "      <td>0.56 [0;0]</td>\n",
       "      <td>0.8 [0;2]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.61 [0;1]</td>\n",
       "      <td>0.61 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.61 [0;1]</td>\n",
       "      <td>1823.6 [0;3134]</td>\n",
       "      <td>1823.6 [0;3134]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1795.4 [0;2993]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.61 [0;1]</td>\n",
       "      <td>0.61 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.69 [0;0]</td>\n",
       "      <td>2065.36 [0;2990]</td>\n",
       "      <td>2065.36 [0;2990]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2065.36 [0;2990]</td>\n",
       "      <td>0.69 [0;0]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.48 [0;1]</td>\n",
       "      <td>1442.2 [0;3857]</td>\n",
       "      <td>1442.2 [0;3857]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1766.4 [0;2993]</td>\n",
       "      <td>0.59 [0;1]</td>\n",
       "      <td>1.0 [0;2]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.51 [0;0]</td>\n",
       "      <td>1522.0 [0;2782]</td>\n",
       "      <td>1522.0 [0;2782]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1522.0 [0;2782]</td>\n",
       "      <td>0.51 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.31 [0;0]</td>\n",
       "      <td>933.8 [0;2992]</td>\n",
       "      <td>933.8 [0;2992]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>933.8 [0;2992]</td>\n",
       "      <td>0.31 [0;0]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "      <td>0.4 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.59 [0;1]</td>\n",
       "      <td>1755.8 [0;2993]</td>\n",
       "      <td>1755.8 [0;2993]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1755.8 [0;2993]</td>\n",
       "      <td>0.59 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm        L90         LSA  \\\n",
       "Assembler                                                                  \n",
       "ABySS              Multiple k-mer De Bruijn graph  0.0 [0;0]  0.61 [0;0]   \n",
       "BCALM2               Single k-mer De Bruijn graph  0.0 [0;0]   0.0 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph  0.0 [0;0]   0.6 [0;1]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  0.2 [0;1]  0.49 [0;0]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph  0.0 [0;0]  0.61 [0;1]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph  0.0 [0;0]  0.69 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph  0.4 [0;1]  0.48 [0;1]   \n",
       "MINIA                Single k-mer De Bruijn graph  0.0 [0;0]   0.0 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph  0.0 [0;0]  0.51 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph  0.0 [0;0]  0.31 [0;0]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph  0.0 [0;0]  0.59 [0;1]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph  0.0 [0;0]   0.0 [0;0]   \n",
       "\n",
       "                               NA50              NG50         Ns         Type  \\\n",
       "Assembler                                                                       \n",
       "ABySS              1821.58 [0;2992]  1821.58 [0;2992]  0.0 [0;0]  Metagenomic   \n",
       "BCALM2                    0.0 [0;0]         0.0 [0;0]  0.0 [0;0]      Genomic   \n",
       "GATBMiniaPipeline   1785.8 [0;3162]   1785.8 [0;3162]  0.0 [0;0]  Metagenomic   \n",
       "IDBA-UD             1476.4 [0;2847]   1476.4 [0;2847]  0.0 [0;0]  Metagenomic   \n",
       "MEGAHIT             1823.6 [0;3134]   1823.6 [0;3134]  0.0 [0;0]  Metagenomic   \n",
       "MetaHipMer2        2065.36 [0;2990]  2065.36 [0;2990]  0.0 [0;0]  Metagenomic   \n",
       "metaSPAdes          1442.2 [0;3857]   1442.2 [0;3857]  0.0 [0;0]  Metagenomic   \n",
       "MINIA                     0.0 [0;0]         0.0 [0;0]  0.0 [0;0]      Genomic   \n",
       "SKESA               1522.0 [0;2782]   1522.0 [0;2782]  0.0 [0;0]      Genomic   \n",
       "SPAdes               933.8 [0;2992]    933.8 [0;2992]  0.0 [0;0]      Genomic   \n",
       "Unicycler           1755.8 [0;2993]   1755.8 [0;2993]  0.0 [0;0]      Genomic   \n",
       "VelvetOptimiser           0.0 [0;0]         0.0 [0;0]  0.0 [0;0]  Metagenomic   \n",
       "\n",
       "                          basepairs breadth_of_coverage     contigs  \\\n",
       "Assembler                                                             \n",
       "ABySS              1821.58 [0;2992]          0.61 [0;0]  0.67 [0;1]   \n",
       "BCALM2                    0.0 [0;0]           0.0 [0;0]   0.0 [0;0]   \n",
       "GATBMiniaPipeline   1754.8 [0;2993]          0.59 [0;1]   0.6 [0;1]   \n",
       "IDBA-UD             1690.2 [0;2847]          0.56 [0;0]   0.8 [0;2]   \n",
       "MEGAHIT             1795.4 [0;2993]           0.6 [0;1]   0.6 [0;1]   \n",
       "MetaHipMer2        2065.36 [0;2990]          0.69 [0;0]  0.82 [0;1]   \n",
       "metaSPAdes          1766.4 [0;2993]          0.59 [0;1]   1.0 [0;2]   \n",
       "MINIA                     0.0 [0;0]           0.0 [0;0]   0.0 [0;0]   \n",
       "SKESA               1522.0 [0;2782]          0.51 [0;0]   0.6 [0;1]   \n",
       "SPAdes               933.8 [0;2992]          0.31 [0;0]   0.4 [0;1]   \n",
       "Unicycler           1755.8 [0;2993]          0.59 [0;1]   0.6 [0;1]   \n",
       "VelvetOptimiser           0.0 [0;0]           0.0 [0;0]   0.0 [0;0]   \n",
       "\n",
       "                     identity lowest_identity misassembled contigs  \\\n",
       "Assembler                                                            \n",
       "ABySS              0.67 [0;1]      0.67 [0;1]            0.0 [0;0]   \n",
       "BCALM2              0.0 [0;0]       0.0 [0;0]            0.0 [0;0]   \n",
       "GATBMiniaPipeline   0.6 [0;1]       0.6 [0;1]            0.0 [0;0]   \n",
       "IDBA-UD             0.6 [0;1]       0.6 [0;1]            0.0 [0;0]   \n",
       "MEGAHIT             0.6 [0;1]       0.6 [0;1]            0.0 [0;0]   \n",
       "MetaHipMer2        0.82 [0;1]      0.82 [0;1]            0.0 [0;0]   \n",
       "metaSPAdes          0.6 [0;1]       0.6 [0;1]            0.0 [0;0]   \n",
       "MINIA               0.0 [0;0]       0.0 [0;0]            0.0 [0;0]   \n",
       "SKESA               0.6 [0;1]       0.6 [0;1]            0.0 [0;0]   \n",
       "SPAdes              0.4 [0;1]       0.4 [0;1]            0.0 [0;0]   \n",
       "Unicycler           0.6 [0;1]       0.6 [0;1]            0.0 [0;0]   \n",
       "VelvetOptimiser     0.0 [0;0]       0.0 [0;0]            0.0 [0;0]   \n",
       "\n",
       "                  misassembly events multiplicity   parsimony    validity  \n",
       "Assembler                                                                  \n",
       "ABySS                      0.0 [0;0]   0.67 [0;1]  0.67 [0;1]  0.67 [0;1]  \n",
       "BCALM2                     0.0 [0;0]    0.0 [0;0]   0.0 [0;0]   0.0 [0;0]  \n",
       "GATBMiniaPipeline          0.0 [0;0]   0.61 [0;1]  0.61 [0;1]   0.6 [0;1]  \n",
       "IDBA-UD                    0.0 [0;0]   0.61 [0;1]  0.61 [0;1]   0.6 [0;1]  \n",
       "MEGAHIT                    0.0 [0;0]   0.61 [0;1]  0.61 [0;1]   0.6 [0;1]  \n",
       "MetaHipMer2                0.0 [0;0]   0.82 [0;1]  0.82 [0;1]  0.82 [0;1]  \n",
       "metaSPAdes                 0.0 [0;0]   0.67 [0;1]  0.67 [0;1]   0.6 [0;1]  \n",
       "MINIA                      0.0 [0;0]    0.0 [0;0]   0.0 [0;0]   0.0 [0;0]  \n",
       "SKESA                      0.0 [0;0]    0.6 [0;1]   0.6 [0;1]   0.6 [0;1]  \n",
       "SPAdes                     0.0 [0;0]    0.4 [0;1]   0.4 [0;1]   0.4 [0;1]  \n",
       "Unicycler                  0.0 [0;0]    0.6 [0;1]   0.6 [0;1]   0.6 [0;1]  \n",
       "VelvetOptimiser            0.0 [0;0]    0.0 [0;0]   0.0 [0;0]   0.0 [0;0]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "for reference in reference_pipeline_metrics_df.reference.unique():\n",
    "    ref_stats_per_reference_per_assembler = pd.DataFrame()\n",
    "    for assembler in reference_pipeline_metrics_df.assembler.unique():\n",
    "        row={'Assembler': assembler, \n",
    "             'Type':reference_pipeline_metrics_df[reference_pipeline_metrics_df.assembler == assembler].type.unique()[0],\n",
    "             'Algorithm': reference_pipeline_metrics_df[reference_pipeline_metrics_df.assembler == assembler].algorythm.unique()[0]}\n",
    "        lala=reference_pipeline_metrics_df[(reference_pipeline_metrics_df.assembler == assembler) & (reference_pipeline_metrics_df.reference == reference)].describe()\n",
    "        for column in lala.columns:\n",
    "            mean = lala.loc['mean',column]\n",
    "            minimum = lala.loc['min',column]\n",
    "            maximum = lala.loc['max',column]\n",
    "            row[column] = \"{} [{};{}]\".format(round(mean, 2),int(minimum), int(maximum))\n",
    "        ref_stats_per_reference_per_assembler=ref_stats_per_reference_per_assembler.append(row, ignore_index=True)\n",
    "    ref_stats_per_reference_per_assembler = ref_stats_per_reference_per_assembler.set_index('Assembler')\n",
    "    ref_stats_per_reference_per_assembler.to_csv(\"Tables/Results/{} - Reference metrics per assembler.csv\".format(reference))\n",
    "    print(reference)\n",
    "    display(ref_stats_per_reference_per_assembler)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Per reference replicon for all assemblers - Even only!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Reference</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Bacillus_subtilis</th>\n",
       "      <td>25.64 [0;120]</td>\n",
       "      <td>0.1 [0;0]</td>\n",
       "      <td>196509.03 [0;343843]</td>\n",
       "      <td>194589.61 [0;343843]</td>\n",
       "      <td>1.32 [0;47]</td>\n",
       "      <td>3614434.53 [0;3991152]</td>\n",
       "      <td>0.89 [0;0]</td>\n",
       "      <td>91.85 [0;1467]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "      <td>0.08 [0;2]</td>\n",
       "      <td>0.2 [0;4]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Enterococcus_faecalis</th>\n",
       "      <td>36.26 [0;149]</td>\n",
       "      <td>0.13 [0;0]</td>\n",
       "      <td>134153.57 [0;377104]</td>\n",
       "      <td>132615.19 [0;377104]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>2491484.12 [0;2828460]</td>\n",
       "      <td>0.88 [0;0]</td>\n",
       "      <td>72.49 [0;231]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.91 [0;1]</td>\n",
       "      <td>0.34 [0;3]</td>\n",
       "      <td>0.62 [0;6]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Escherichia_coli_plasmid</th>\n",
       "      <td>3.27 [0;18]</td>\n",
       "      <td>0.22 [0;0]</td>\n",
       "      <td>18003.78 [0;55639]</td>\n",
       "      <td>17461.25 [0;55639]</td>\n",
       "      <td>37.6 [0;835]</td>\n",
       "      <td>82423.2 [0;109516]</td>\n",
       "      <td>0.75 [0;0]</td>\n",
       "      <td>9.2 [0;33]</td>\n",
       "      <td>0.89 [0;1]</td>\n",
       "      <td>0.88 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.89 [0;1]</td>\n",
       "      <td>0.89 [0;1]</td>\n",
       "      <td>0.89 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Escherichia_coli</th>\n",
       "      <td>70.08 [0;283]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>51151.78 [0;138793]</td>\n",
       "      <td>50851.09 [0;138793]</td>\n",
       "      <td>1.26 [0;45]</td>\n",
       "      <td>4051935.78 [0;4706885]</td>\n",
       "      <td>0.85 [0;0]</td>\n",
       "      <td>363.8 [0;1495]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.84 [0;1]</td>\n",
       "      <td>1.07 [0;8]</td>\n",
       "      <td>1.99 [0;15]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Lactobacillus_fermentum</th>\n",
       "      <td>77.7 [0;263]</td>\n",
       "      <td>0.05 [0;0]</td>\n",
       "      <td>30397.63 [0;46932]</td>\n",
       "      <td>29158.61 [0;52972]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1585069.36 [0;1873053]</td>\n",
       "      <td>0.83 [0;0]</td>\n",
       "      <td>105.8 [0;282]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "      <td>0.34 [0;4]</td>\n",
       "      <td>0.69 [0;8]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Listeria_monocytogenes</th>\n",
       "      <td>24.36 [0;122]</td>\n",
       "      <td>0.15 [0;0]</td>\n",
       "      <td>236971.75 [0;542872]</td>\n",
       "      <td>228275.5 [0;542872]</td>\n",
       "      <td>0.03 [0;1]</td>\n",
       "      <td>2625730.37 [0;2961845]</td>\n",
       "      <td>0.88 [0;0]</td>\n",
       "      <td>62.41 [0;361]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.93 [0;1]</td>\n",
       "      <td>0.03 [0;1]</td>\n",
       "      <td>0.06 [0;2]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pseudomonas_aeruginosa</th>\n",
       "      <td>147.13 [0;1004]</td>\n",
       "      <td>0.06 [0;0]</td>\n",
       "      <td>141661.78 [0;338923]</td>\n",
       "      <td>146649.92 [0;464977]</td>\n",
       "      <td>583.01 [0;12098]</td>\n",
       "      <td>6111755.96 [0;6780808]</td>\n",
       "      <td>0.9 [0;0]</td>\n",
       "      <td>240.31 [0;1404]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.85 [0;1]</td>\n",
       "      <td>1.27 [0;8]</td>\n",
       "      <td>2.26 [0;15]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Salmonella_enterica</th>\n",
       "      <td>54.45 [0;200]</td>\n",
       "      <td>0.05 [0;0]</td>\n",
       "      <td>71283.69 [0;223960]</td>\n",
       "      <td>70301.45 [0;223960]</td>\n",
       "      <td>0.12 [0;2]</td>\n",
       "      <td>4074863.96 [0;4703394]</td>\n",
       "      <td>0.86 [0;0]</td>\n",
       "      <td>332.36 [0;1479]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "      <td>0.79 [0;9]</td>\n",
       "      <td>1.25 [0;14]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Staphylococcus_aureus</th>\n",
       "      <td>40.08 [0;140]</td>\n",
       "      <td>0.12 [0;0]</td>\n",
       "      <td>114085.93 [0;394879]</td>\n",
       "      <td>113359.04 [0;394879]</td>\n",
       "      <td>1.99 [0;71]</td>\n",
       "      <td>2373450.96 [0;2693907]</td>\n",
       "      <td>0.87 [0;0]</td>\n",
       "      <td>76.07 [0;247]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.91 [0;1]</td>\n",
       "      <td>0.27 [0;3]</td>\n",
       "      <td>0.63 [0;7]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Staphylococcus_aureus_plasmid1</th>\n",
       "      <td>0.03 [0;1]</td>\n",
       "      <td>0.77 [0;1]</td>\n",
       "      <td>4910.9 [0;6795]</td>\n",
       "      <td>4910.9 [0;6795]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>5035.96 [0;6337]</td>\n",
       "      <td>0.79 [0;1]</td>\n",
       "      <td>0.9 [0;3]</td>\n",
       "      <td>0.81 [0;1]</td>\n",
       "      <td>0.81 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "      <td>0.81 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Staphylococcus_aureus_plasmid2</th>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.66 [0;1]</td>\n",
       "      <td>1483.03 [0;3246]</td>\n",
       "      <td>1483.03 [0;3246]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1441.14 [0;2216]</td>\n",
       "      <td>0.65 [0;1]</td>\n",
       "      <td>0.69 [0;1]</td>\n",
       "      <td>0.68 [0;1]</td>\n",
       "      <td>0.68 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.7 [0;1]</td>\n",
       "      <td>0.71 [0;1]</td>\n",
       "      <td>0.68 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Staphylococcus_aureus_plasmid3</th>\n",
       "      <td>0.08 [0;1]</td>\n",
       "      <td>0.64 [0;1]</td>\n",
       "      <td>1922.17 [0;3857]</td>\n",
       "      <td>1922.17 [0;3857]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1989.29 [0;2993]</td>\n",
       "      <td>0.66 [0;1]</td>\n",
       "      <td>0.8 [0;2]</td>\n",
       "      <td>0.72 [0;1]</td>\n",
       "      <td>0.72 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.72 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            L90         LSA  \\\n",
       "Reference                                                     \n",
       "Bacillus_subtilis                 25.64 [0;120]   0.1 [0;0]   \n",
       "Enterococcus_faecalis             36.26 [0;149]  0.13 [0;0]   \n",
       "Escherichia_coli_plasmid            3.27 [0;18]  0.22 [0;0]   \n",
       "Escherichia_coli                  70.08 [0;283]  0.04 [0;0]   \n",
       "Lactobacillus_fermentum            77.7 [0;263]  0.05 [0;0]   \n",
       "Listeria_monocytogenes            24.36 [0;122]  0.15 [0;0]   \n",
       "Pseudomonas_aeruginosa          147.13 [0;1004]  0.06 [0;0]   \n",
       "Salmonella_enterica               54.45 [0;200]  0.05 [0;0]   \n",
       "Staphylococcus_aureus             40.08 [0;140]  0.12 [0;0]   \n",
       "Staphylococcus_aureus_plasmid1       0.03 [0;1]  0.77 [0;1]   \n",
       "Staphylococcus_aureus_plasmid2        0.0 [0;0]  0.66 [0;1]   \n",
       "Staphylococcus_aureus_plasmid3       0.08 [0;1]  0.64 [0;1]   \n",
       "\n",
       "                                                NA50                  NG50  \\\n",
       "Reference                                                                    \n",
       "Bacillus_subtilis               196509.03 [0;343843]  194589.61 [0;343843]   \n",
       "Enterococcus_faecalis           134153.57 [0;377104]  132615.19 [0;377104]   \n",
       "Escherichia_coli_plasmid          18003.78 [0;55639]    17461.25 [0;55639]   \n",
       "Escherichia_coli                 51151.78 [0;138793]   50851.09 [0;138793]   \n",
       "Lactobacillus_fermentum           30397.63 [0;46932]    29158.61 [0;52972]   \n",
       "Listeria_monocytogenes          236971.75 [0;542872]   228275.5 [0;542872]   \n",
       "Pseudomonas_aeruginosa          141661.78 [0;338923]  146649.92 [0;464977]   \n",
       "Salmonella_enterica              71283.69 [0;223960]   70301.45 [0;223960]   \n",
       "Staphylococcus_aureus           114085.93 [0;394879]  113359.04 [0;394879]   \n",
       "Staphylococcus_aureus_plasmid1       4910.9 [0;6795]       4910.9 [0;6795]   \n",
       "Staphylococcus_aureus_plasmid2      1483.03 [0;3246]      1483.03 [0;3246]   \n",
       "Staphylococcus_aureus_plasmid3      1922.17 [0;3857]      1922.17 [0;3857]   \n",
       "\n",
       "                                              Ns               basepairs  \\\n",
       "Reference                                                                  \n",
       "Bacillus_subtilis                    1.32 [0;47]  3614434.53 [0;3991152]   \n",
       "Enterococcus_faecalis                  0.0 [0;0]  2491484.12 [0;2828460]   \n",
       "Escherichia_coli_plasmid            37.6 [0;835]      82423.2 [0;109516]   \n",
       "Escherichia_coli                     1.26 [0;45]  4051935.78 [0;4706885]   \n",
       "Lactobacillus_fermentum                0.0 [0;0]  1585069.36 [0;1873053]   \n",
       "Listeria_monocytogenes                0.03 [0;1]  2625730.37 [0;2961845]   \n",
       "Pseudomonas_aeruginosa          583.01 [0;12098]  6111755.96 [0;6780808]   \n",
       "Salmonella_enterica                   0.12 [0;2]  4074863.96 [0;4703394]   \n",
       "Staphylococcus_aureus                1.99 [0;71]  2373450.96 [0;2693907]   \n",
       "Staphylococcus_aureus_plasmid1         0.0 [0;0]        5035.96 [0;6337]   \n",
       "Staphylococcus_aureus_plasmid2         0.0 [0;0]        1441.14 [0;2216]   \n",
       "Staphylococcus_aureus_plasmid3         0.0 [0;0]        1989.29 [0;2993]   \n",
       "\n",
       "                               breadth_of_coverage          contigs  \\\n",
       "Reference                                                             \n",
       "Bacillus_subtilis                       0.89 [0;0]   91.85 [0;1467]   \n",
       "Enterococcus_faecalis                   0.88 [0;0]    72.49 [0;231]   \n",
       "Escherichia_coli_plasmid                0.75 [0;0]       9.2 [0;33]   \n",
       "Escherichia_coli                        0.85 [0;0]   363.8 [0;1495]   \n",
       "Lactobacillus_fermentum                 0.83 [0;0]    105.8 [0;282]   \n",
       "Listeria_monocytogenes                  0.88 [0;0]    62.41 [0;361]   \n",
       "Pseudomonas_aeruginosa                   0.9 [0;0]  240.31 [0;1404]   \n",
       "Salmonella_enterica                     0.86 [0;0]  332.36 [0;1479]   \n",
       "Staphylococcus_aureus                   0.87 [0;0]    76.07 [0;247]   \n",
       "Staphylococcus_aureus_plasmid1          0.79 [0;1]        0.9 [0;3]   \n",
       "Staphylococcus_aureus_plasmid2          0.65 [0;1]       0.69 [0;1]   \n",
       "Staphylococcus_aureus_plasmid3          0.66 [0;1]        0.8 [0;2]   \n",
       "\n",
       "                                  identity lowest_identity  \\\n",
       "Reference                                                    \n",
       "Bacillus_subtilis               0.94 [0;1]      0.92 [0;1]   \n",
       "Enterococcus_faecalis           0.94 [0;1]      0.91 [0;1]   \n",
       "Escherichia_coli_plasmid        0.89 [0;1]      0.88 [0;1]   \n",
       "Escherichia_coli                0.94 [0;1]      0.84 [0;1]   \n",
       "Lactobacillus_fermentum         0.92 [0;1]       0.9 [0;1]   \n",
       "Listeria_monocytogenes          0.94 [0;1]      0.93 [0;1]   \n",
       "Pseudomonas_aeruginosa          0.94 [0;1]      0.85 [0;1]   \n",
       "Salmonella_enterica             0.94 [0;1]      0.83 [0;1]   \n",
       "Staphylococcus_aureus           0.94 [0;1]      0.91 [0;1]   \n",
       "Staphylococcus_aureus_plasmid1  0.81 [0;1]      0.81 [0;1]   \n",
       "Staphylococcus_aureus_plasmid2  0.68 [0;1]      0.68 [0;1]   \n",
       "Staphylococcus_aureus_plasmid3  0.72 [0;1]      0.72 [0;1]   \n",
       "\n",
       "                               misassembled contigs misassembly events  \\\n",
       "Reference                                                                \n",
       "Bacillus_subtilis                        0.08 [0;2]          0.2 [0;4]   \n",
       "Enterococcus_faecalis                    0.34 [0;3]         0.62 [0;6]   \n",
       "Escherichia_coli_plasmid                  0.0 [0;0]          0.0 [0;0]   \n",
       "Escherichia_coli                         1.07 [0;8]        1.99 [0;15]   \n",
       "Lactobacillus_fermentum                  0.34 [0;4]         0.69 [0;8]   \n",
       "Listeria_monocytogenes                   0.03 [0;1]         0.06 [0;2]   \n",
       "Pseudomonas_aeruginosa                   1.27 [0;8]        2.26 [0;15]   \n",
       "Salmonella_enterica                      0.79 [0;9]        1.25 [0;14]   \n",
       "Staphylococcus_aureus                    0.27 [0;3]         0.63 [0;7]   \n",
       "Staphylococcus_aureus_plasmid1            0.0 [0;0]          0.0 [0;0]   \n",
       "Staphylococcus_aureus_plasmid2            0.0 [0;0]          0.0 [0;0]   \n",
       "Staphylococcus_aureus_plasmid3            0.0 [0;0]          0.0 [0;0]   \n",
       "\n",
       "                               multiplicity   parsimony    validity  \n",
       "Reference                                                            \n",
       "Bacillus_subtilis                0.94 [0;1]  0.94 [0;1]  0.94 [0;1]  \n",
       "Enterococcus_faecalis            0.95 [0;1]  0.95 [0;1]  0.94 [0;1]  \n",
       "Escherichia_coli_plasmid         0.89 [0;1]  0.89 [0;1]  0.89 [0;1]  \n",
       "Escherichia_coli                 0.95 [0;1]  0.95 [0;1]  0.94 [0;1]  \n",
       "Lactobacillus_fermentum          0.92 [0;1]  0.92 [0;1]  0.92 [0;1]  \n",
       "Listeria_monocytogenes           0.95 [0;1]  0.95 [0;1]  0.94 [0;1]  \n",
       "Pseudomonas_aeruginosa           0.95 [0;1]  0.95 [0;1]  0.94 [0;1]  \n",
       "Salmonella_enterica              0.95 [0;1]  0.95 [0;1]  0.94 [0;1]  \n",
       "Staphylococcus_aureus            0.94 [0;1]  0.95 [0;1]  0.94 [0;1]  \n",
       "Staphylococcus_aureus_plasmid1   0.82 [0;1]  0.82 [0;1]  0.81 [0;1]  \n",
       "Staphylococcus_aureus_plasmid2    0.7 [0;1]  0.71 [0;1]  0.68 [0;1]  \n",
       "Staphylococcus_aureus_plasmid3   0.73 [0;1]  0.73 [0;1]  0.72 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ref_stats_per_reference_all_assemblers = pd.DataFrame()\n",
    "for reference in reference_pipeline_metrics_df.reference.unique():\n",
    "    row={'Reference': reference}\n",
    "    lala=reference_pipeline_metrics_df[(reference_pipeline_metrics_df.reference == reference) & (reference_pipeline_metrics_df.distribution == 'Even')].describe()\n",
    "    for column in lala.columns:\n",
    "        mean = lala.loc['mean',column]\n",
    "        minimum = lala.loc['min',column]\n",
    "        maximum = lala.loc['max',column]\n",
    "        row[column] = \"{} [{};{}]\".format(round(mean, 2),int(minimum), int(maximum))\n",
    "    ref_stats_per_reference_all_assemblers=ref_stats_per_reference_all_assemblers.append(row, ignore_index=True)\n",
    "ref_stats_per_reference_all_assemblers = ref_stats_per_reference_all_assemblers.set_index('Reference')\n",
    "ref_stats_per_reference_all_assemblers.to_csv(\"Tables/Results/Reference metrics per replicon.csv\")\n",
    "display(ref_stats_per_reference_all_assemblers)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reference stats per distribution"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>50.54 [0;1033]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>25986.46 [0;351511]</td>\n",
       "      <td>25866.25 [0;351511]</td>\n",
       "      <td>16.67 [0;400]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>690335.21 [0;6781605]</td>\n",
       "      <td>0.16 [0;0]</td>\n",
       "      <td>65.12 [0;1141]</td>\n",
       "      <td>0.21 [0;0]</td>\n",
       "      <td>0.2 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.21 [0;1]</td>\n",
       "      <td>0.21 [0;1]</td>\n",
       "      <td>0.21 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>13.58 [0;311]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>3455.14 [0;64777]</td>\n",
       "      <td>3263.39 [0;62493]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>390195.64 [0;6515102]</td>\n",
       "      <td>0.09 [0;0]</td>\n",
       "      <td>35.86 [0;496]</td>\n",
       "      <td>0.17 [0;1]</td>\n",
       "      <td>0.17 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.17 [0;1]</td>\n",
       "      <td>0.17 [0;1]</td>\n",
       "      <td>0.17 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>23.6 [0;459]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>41255.35 [0;351149]</td>\n",
       "      <td>40885.01 [0;351149]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1131019.22 [0;6716046]</td>\n",
       "      <td>0.25 [0;0]</td>\n",
       "      <td>37.54 [0;604]</td>\n",
       "      <td>0.36 [0;1]</td>\n",
       "      <td>0.36 [0;1]</td>\n",
       "      <td>0.08 [0;2]</td>\n",
       "      <td>0.25 [0;6]</td>\n",
       "      <td>0.36 [0;1]</td>\n",
       "      <td>0.36 [0;1]</td>\n",
       "      <td>0.36 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>16.33 [0;292]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>48013.69 [0;475377]</td>\n",
       "      <td>46431.78 [0;475377]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1135250.6 [0;6700232]</td>\n",
       "      <td>0.25 [0;0]</td>\n",
       "      <td>29.56 [0;418]</td>\n",
       "      <td>0.46 [0;0]</td>\n",
       "      <td>0.38 [0;0]</td>\n",
       "      <td>0.25 [0;2]</td>\n",
       "      <td>0.39 [0;5]</td>\n",
       "      <td>0.47 [0;1]</td>\n",
       "      <td>0.61 [0;5]</td>\n",
       "      <td>0.44 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>8.42 [0;147]</td>\n",
       "      <td>0.03 [0;0]</td>\n",
       "      <td>66650.44 [0;543128]</td>\n",
       "      <td>66455.53 [0;543128]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1141247.44 [0;6738936]</td>\n",
       "      <td>0.25 [0;0]</td>\n",
       "      <td>18.11 [0;223]</td>\n",
       "      <td>0.46 [0;0]</td>\n",
       "      <td>0.42 [0;0]</td>\n",
       "      <td>0.06 [0;1]</td>\n",
       "      <td>0.11 [0;2]</td>\n",
       "      <td>0.47 [0;1]</td>\n",
       "      <td>0.48 [0;1]</td>\n",
       "      <td>0.47 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>903.29 [0;2986]</td>\n",
       "      <td>529.12 [0;2697]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>825097.88 [0;3766506]</td>\n",
       "      <td>0.19 [0;0]</td>\n",
       "      <td>407.83 [0;2404]</td>\n",
       "      <td>0.42 [0;1]</td>\n",
       "      <td>0.39 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.42 [0;1]</td>\n",
       "      <td>0.42 [0;1]</td>\n",
       "      <td>0.42 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>7.19 [0;66]</td>\n",
       "      <td>0.03 [0;0]</td>\n",
       "      <td>60717.78 [0;593167]</td>\n",
       "      <td>59453.89 [0;593167]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1140707.44 [0;6733500]</td>\n",
       "      <td>0.25 [0;0]</td>\n",
       "      <td>15.53 [0;110]</td>\n",
       "      <td>0.47 [0;0]</td>\n",
       "      <td>0.43 [0;0]</td>\n",
       "      <td>0.17 [0;2]</td>\n",
       "      <td>0.39 [0;6]</td>\n",
       "      <td>0.47 [0;1]</td>\n",
       "      <td>0.48 [0;1]</td>\n",
       "      <td>0.47 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>65.1 [0;649]</td>\n",
       "      <td>0.01 [0;0]</td>\n",
       "      <td>6383.53 [0;64777]</td>\n",
       "      <td>5783.75 [0;64473]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1099789.64 [0;6524118]</td>\n",
       "      <td>0.24 [0;0]</td>\n",
       "      <td>119.54 [0;919]</td>\n",
       "      <td>0.42 [0;1]</td>\n",
       "      <td>0.39 [0;1]</td>\n",
       "      <td>0.14 [0;2]</td>\n",
       "      <td>0.22 [0;4]</td>\n",
       "      <td>0.42 [0;1]</td>\n",
       "      <td>0.42 [0;1]</td>\n",
       "      <td>0.42 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>6.72 [0;232]</td>\n",
       "      <td>0.01 [0;0]</td>\n",
       "      <td>10421.97 [0;351027]</td>\n",
       "      <td>10210.64 [0;351027]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>366940.31 [0;3512871]</td>\n",
       "      <td>0.1 [0;0]</td>\n",
       "      <td>120.94 [0;1877]</td>\n",
       "      <td>0.19 [0;1]</td>\n",
       "      <td>0.19 [0;1]</td>\n",
       "      <td>0.33 [0;12]</td>\n",
       "      <td>0.64 [0;23]</td>\n",
       "      <td>0.19 [0;1]</td>\n",
       "      <td>0.19 [0;1]</td>\n",
       "      <td>0.19 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>30.83 [0;835]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>36831.08 [0;351104]</td>\n",
       "      <td>37693.72 [0;351104]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1093338.33 [0;6710057]</td>\n",
       "      <td>0.23 [0;0]</td>\n",
       "      <td>105.17 [0;1839]</td>\n",
       "      <td>0.56 [0;1]</td>\n",
       "      <td>0.49 [0;1]</td>\n",
       "      <td>16.08 [0;565]</td>\n",
       "      <td>32.11 [0;1134]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.63 [0;1]</td>\n",
       "      <td>0.56 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>7.17 [0;143]</td>\n",
       "      <td>0.03 [0;0]</td>\n",
       "      <td>51507.03 [0;351002]</td>\n",
       "      <td>49210.58 [0;351002]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>963132.78 [0;6712831]</td>\n",
       "      <td>0.2 [0;0]</td>\n",
       "      <td>13.72 [0;179]</td>\n",
       "      <td>0.22 [0;1]</td>\n",
       "      <td>0.22 [0;1]</td>\n",
       "      <td>0.14 [0;5]</td>\n",
       "      <td>0.28 [0;10]</td>\n",
       "      <td>0.22 [0;1]</td>\n",
       "      <td>0.22 [0;1]</td>\n",
       "      <td>0.22 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>2.18 [0;145]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>4610.54 [0;269136]</td>\n",
       "      <td>4628.78 [0;294404]</td>\n",
       "      <td>211.79 [0;6066]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>486043.26 [0;5336634]</td>\n",
       "      <td>0.11 [0;0]</td>\n",
       "      <td>165.1 [0;2156]</td>\n",
       "      <td>0.29 [0;1]</td>\n",
       "      <td>0.27 [0;1]</td>\n",
       "      <td>0.12 [0;4]</td>\n",
       "      <td>0.22 [0;9]</td>\n",
       "      <td>0.29 [0;1]</td>\n",
       "      <td>0.3 [0;1]</td>\n",
       "      <td>0.29 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm             L90         LSA  \\\n",
       "Assembler                                                                       \n",
       "ABySS              Multiple k-mer De Bruijn graph  50.54 [0;1033]  0.02 [0;0]   \n",
       "BCALM2               Single k-mer De Bruijn graph   13.58 [0;311]   0.0 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph    23.6 [0;459]  0.02 [0;0]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph   16.33 [0;292]  0.02 [0;0]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph    8.42 [0;147]  0.03 [0;0]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph       0.0 [0;0]   0.0 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph     7.19 [0;66]  0.03 [0;0]   \n",
       "MINIA                Single k-mer De Bruijn graph    65.1 [0;649]  0.01 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph    6.72 [0;232]  0.01 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph   30.83 [0;835]  0.02 [0;0]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph    7.17 [0;143]  0.03 [0;0]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph    2.18 [0;145]   0.0 [0;0]   \n",
       "\n",
       "                                  NA50                 NG50               Ns  \\\n",
       "Assembler                                                                      \n",
       "ABySS              25986.46 [0;351511]  25866.25 [0;351511]    16.67 [0;400]   \n",
       "BCALM2               3455.14 [0;64777]    3263.39 [0;62493]        0.0 [0;0]   \n",
       "GATBMiniaPipeline  41255.35 [0;351149]  40885.01 [0;351149]        0.0 [0;0]   \n",
       "IDBA-UD            48013.69 [0;475377]  46431.78 [0;475377]        0.0 [0;0]   \n",
       "MEGAHIT            66650.44 [0;543128]  66455.53 [0;543128]        0.0 [0;0]   \n",
       "MetaHipMer2            903.29 [0;2986]      529.12 [0;2697]        0.0 [0;0]   \n",
       "metaSPAdes         60717.78 [0;593167]  59453.89 [0;593167]        0.0 [0;0]   \n",
       "MINIA                6383.53 [0;64777]    5783.75 [0;64473]        0.0 [0;0]   \n",
       "SKESA              10421.97 [0;351027]  10210.64 [0;351027]        0.0 [0;0]   \n",
       "SPAdes             36831.08 [0;351104]  37693.72 [0;351104]        0.0 [0;0]   \n",
       "Unicycler          51507.03 [0;351002]  49210.58 [0;351002]        0.0 [0;0]   \n",
       "VelvetOptimiser     4610.54 [0;269136]   4628.78 [0;294404]  211.79 [0;6066]   \n",
       "\n",
       "                          Type               basepairs breadth_of_coverage  \\\n",
       "Assembler                                                                    \n",
       "ABySS              Metagenomic   690335.21 [0;6781605]          0.16 [0;0]   \n",
       "BCALM2                 Genomic   390195.64 [0;6515102]          0.09 [0;0]   \n",
       "GATBMiniaPipeline  Metagenomic  1131019.22 [0;6716046]          0.25 [0;0]   \n",
       "IDBA-UD            Metagenomic   1135250.6 [0;6700232]          0.25 [0;0]   \n",
       "MEGAHIT            Metagenomic  1141247.44 [0;6738936]          0.25 [0;0]   \n",
       "MetaHipMer2        Metagenomic   825097.88 [0;3766506]          0.19 [0;0]   \n",
       "metaSPAdes         Metagenomic  1140707.44 [0;6733500]          0.25 [0;0]   \n",
       "MINIA                  Genomic  1099789.64 [0;6524118]          0.24 [0;0]   \n",
       "SKESA                  Genomic   366940.31 [0;3512871]           0.1 [0;0]   \n",
       "SPAdes                 Genomic  1093338.33 [0;6710057]          0.23 [0;0]   \n",
       "Unicycler              Genomic   963132.78 [0;6712831]           0.2 [0;0]   \n",
       "VelvetOptimiser    Metagenomic   486043.26 [0;5336634]          0.11 [0;0]   \n",
       "\n",
       "                           contigs    identity lowest_identity  \\\n",
       "Assembler                                                        \n",
       "ABySS               65.12 [0;1141]  0.21 [0;0]       0.2 [0;0]   \n",
       "BCALM2               35.86 [0;496]  0.17 [0;1]      0.17 [0;1]   \n",
       "GATBMiniaPipeline    37.54 [0;604]  0.36 [0;1]      0.36 [0;1]   \n",
       "IDBA-UD              29.56 [0;418]  0.46 [0;0]      0.38 [0;0]   \n",
       "MEGAHIT              18.11 [0;223]  0.46 [0;0]      0.42 [0;0]   \n",
       "MetaHipMer2        407.83 [0;2404]  0.42 [0;1]      0.39 [0;1]   \n",
       "metaSPAdes           15.53 [0;110]  0.47 [0;0]      0.43 [0;0]   \n",
       "MINIA               119.54 [0;919]  0.42 [0;1]      0.39 [0;1]   \n",
       "SKESA              120.94 [0;1877]  0.19 [0;1]      0.19 [0;1]   \n",
       "SPAdes             105.17 [0;1839]  0.56 [0;1]      0.49 [0;1]   \n",
       "Unicycler            13.72 [0;179]  0.22 [0;1]      0.22 [0;1]   \n",
       "VelvetOptimiser     165.1 [0;2156]  0.29 [0;1]      0.27 [0;1]   \n",
       "\n",
       "                  misassembled contigs misassembly events multiplicity  \\\n",
       "Assembler                                                                \n",
       "ABySS                        0.0 [0;0]          0.0 [0;0]   0.21 [0;1]   \n",
       "BCALM2                       0.0 [0;0]          0.0 [0;0]   0.17 [0;1]   \n",
       "GATBMiniaPipeline           0.08 [0;2]         0.25 [0;6]   0.36 [0;1]   \n",
       "IDBA-UD                     0.25 [0;2]         0.39 [0;5]   0.47 [0;1]   \n",
       "MEGAHIT                     0.06 [0;1]         0.11 [0;2]   0.47 [0;1]   \n",
       "MetaHipMer2                  0.0 [0;0]          0.0 [0;0]   0.42 [0;1]   \n",
       "metaSPAdes                  0.17 [0;2]         0.39 [0;6]   0.47 [0;1]   \n",
       "MINIA                       0.14 [0;2]         0.22 [0;4]   0.42 [0;1]   \n",
       "SKESA                      0.33 [0;12]        0.64 [0;23]   0.19 [0;1]   \n",
       "SPAdes                   16.08 [0;565]     32.11 [0;1134]    0.6 [0;1]   \n",
       "Unicycler                   0.14 [0;5]        0.28 [0;10]   0.22 [0;1]   \n",
       "VelvetOptimiser             0.12 [0;4]         0.22 [0;9]   0.29 [0;1]   \n",
       "\n",
       "                    parsimony    validity  \n",
       "Assembler                                  \n",
       "ABySS              0.21 [0;1]  0.21 [0;1]  \n",
       "BCALM2             0.17 [0;1]  0.17 [0;1]  \n",
       "GATBMiniaPipeline  0.36 [0;1]  0.36 [0;1]  \n",
       "IDBA-UD            0.61 [0;5]  0.44 [0;1]  \n",
       "MEGAHIT            0.48 [0;1]  0.47 [0;1]  \n",
       "MetaHipMer2        0.42 [0;1]  0.42 [0;1]  \n",
       "metaSPAdes         0.48 [0;1]  0.47 [0;1]  \n",
       "MINIA              0.42 [0;1]  0.42 [0;1]  \n",
       "SKESA              0.19 [0;1]  0.19 [0;1]  \n",
       "SPAdes             0.63 [0;1]  0.56 [0;1]  \n",
       "Unicycler          0.22 [0;1]  0.22 [0;1]  \n",
       "VelvetOptimiser     0.3 [0;1]  0.29 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Even\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>60.4 [0;1004]</td>\n",
       "      <td>0.3 [0;0]</td>\n",
       "      <td>101808.53 [0;351502]</td>\n",
       "      <td>101667.72 [0;351502]</td>\n",
       "      <td>92.14 [0;2170]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2534947.7 [0;6780808]</td>\n",
       "      <td>0.91 [0;0]</td>\n",
       "      <td>97.11 [0;1404]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.33 [0;8]</td>\n",
       "      <td>0.59 [0;14]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>51.2 [0;921]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>6932.33 [0;65138]</td>\n",
       "      <td>6539.94 [0;64934]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1236062.67 [0;6510588]</td>\n",
       "      <td>0.37 [0;0]</td>\n",
       "      <td>272.97 [0;1495]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>31.33 [0;112]</td>\n",
       "      <td>0.35 [0;1]</td>\n",
       "      <td>106093.76 [2176;452364]</td>\n",
       "      <td>96782.73 [2176;351186]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2523829.08 [2176;6716522]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "      <td>59.92 [1;199]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "      <td>0.14 [0;3]</td>\n",
       "      <td>0.25 [0;6]</td>\n",
       "      <td>1.01 [0;1]</td>\n",
       "      <td>1.01 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>68.82 [0;283]</td>\n",
       "      <td>0.27 [0;1]</td>\n",
       "      <td>55836.03 [0;254629]</td>\n",
       "      <td>54554.11 [0;254629]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2507600.54 [0;6710005]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>121.12 [0;449]</td>\n",
       "      <td>0.97 [0;1]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "      <td>1.12 [0;9]</td>\n",
       "      <td>1.94 [0;15]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "      <td>0.97 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>18.31 [0;55]</td>\n",
       "      <td>0.37 [0;1]</td>\n",
       "      <td>138989.33 [2215;542872]</td>\n",
       "      <td>141793.83 [2215;542872]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2534425.59 [2215;6737986]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>44.39 [1;141]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.93 [0;1]</td>\n",
       "      <td>0.97 [0;4]</td>\n",
       "      <td>1.99 [0;7]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.01 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>21.7 [0;110]</td>\n",
       "      <td>0.23 [0;0]</td>\n",
       "      <td>58464.9 [0;257424]</td>\n",
       "      <td>57747.71 [0;257424]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1722404.01 [0;6717767]</td>\n",
       "      <td>0.68 [0;0]</td>\n",
       "      <td>74.19 [0;361]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>20.14 [0;90]</td>\n",
       "      <td>0.34 [0;1]</td>\n",
       "      <td>133559.97 [1677;475510]</td>\n",
       "      <td>132927.86 [1677;475510]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2530047.11 [2076;6726884]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "      <td>39.33 [1;167]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>0.14 [0;2]</td>\n",
       "      <td>0.33 [0;6]</td>\n",
       "      <td>1.01 [0;1]</td>\n",
       "      <td>1.01 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>83.41 [0;407]</td>\n",
       "      <td>0.08 [0;0]</td>\n",
       "      <td>17271.52 [0;65138]</td>\n",
       "      <td>16622.5 [0;64960]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2345242.24 [0;6518607]</td>\n",
       "      <td>0.74 [0;0]</td>\n",
       "      <td>307.84 [0;1147]</td>\n",
       "      <td>0.81 [0;1]</td>\n",
       "      <td>0.79 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.81 [0;1]</td>\n",
       "      <td>0.81 [0;1]</td>\n",
       "      <td>0.81 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>57.92 [0;670]</td>\n",
       "      <td>0.27 [0;0]</td>\n",
       "      <td>76053.75 [1404;351014]</td>\n",
       "      <td>75134.92 [0;351014]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2496709.31 [1460;6673601]</td>\n",
       "      <td>0.9 [0;0]</td>\n",
       "      <td>96.06 [1;933]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "      <td>0.67 [0;7]</td>\n",
       "      <td>1.17 [0;15]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>21.06 [0;79]</td>\n",
       "      <td>0.29 [0;1]</td>\n",
       "      <td>117650.97 [0;315090]</td>\n",
       "      <td>117061.69 [0;315090]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2524122.0 [0;6728888]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "      <td>41.19 [0;143]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "      <td>0.36 [0;4]</td>\n",
       "      <td>0.69 [0;8]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.96 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>15.75 [0;57]</td>\n",
       "      <td>0.38 [0;1]</td>\n",
       "      <td>162270.42 [0;475179]</td>\n",
       "      <td>158177.44 [0;475179]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2515166.11 [0;6723626]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "      <td>32.86 [0;111]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.06 [0;1]</td>\n",
       "      <td>0.11 [0;2]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>31.33 [0;207]</td>\n",
       "      <td>0.09 [0;0]</td>\n",
       "      <td>29632.83 [0;338923]</td>\n",
       "      <td>34682.42 [0;464977]</td>\n",
       "      <td>537.64 [0;12098]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1580537.53 [0;6708956]</td>\n",
       "      <td>0.52 [0;0]</td>\n",
       "      <td>167.92 [0;1098]</td>\n",
       "      <td>0.56 [0;1]</td>\n",
       "      <td>0.55 [0;1]</td>\n",
       "      <td>0.39 [0;8]</td>\n",
       "      <td>0.61 [0;11]</td>\n",
       "      <td>0.56 [0;1]</td>\n",
       "      <td>0.56 [0;1]</td>\n",
       "      <td>0.56 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm            L90         LSA  \\\n",
       "Assembler                                                                      \n",
       "ABySS              Multiple k-mer De Bruijn graph  60.4 [0;1004]   0.3 [0;0]   \n",
       "BCALM2               Single k-mer De Bruijn graph   51.2 [0;921]  0.04 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph  31.33 [0;112]  0.35 [0;1]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  68.82 [0;283]  0.27 [0;1]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph   18.31 [0;55]  0.37 [0;1]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph   21.7 [0;110]  0.23 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph   20.14 [0;90]  0.34 [0;1]   \n",
       "MINIA                Single k-mer De Bruijn graph  83.41 [0;407]  0.08 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph  57.92 [0;670]  0.27 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph   21.06 [0;79]  0.29 [0;1]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph   15.75 [0;57]  0.38 [0;1]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph  31.33 [0;207]  0.09 [0;0]   \n",
       "\n",
       "                                      NA50                     NG50  \\\n",
       "Assembler                                                             \n",
       "ABySS                 101808.53 [0;351502]     101667.72 [0;351502]   \n",
       "BCALM2                   6932.33 [0;65138]        6539.94 [0;64934]   \n",
       "GATBMiniaPipeline  106093.76 [2176;452364]   96782.73 [2176;351186]   \n",
       "IDBA-UD                55836.03 [0;254629]      54554.11 [0;254629]   \n",
       "MEGAHIT            138989.33 [2215;542872]  141793.83 [2215;542872]   \n",
       "MetaHipMer2             58464.9 [0;257424]      57747.71 [0;257424]   \n",
       "metaSPAdes         133559.97 [1677;475510]  132927.86 [1677;475510]   \n",
       "MINIA                   17271.52 [0;65138]        16622.5 [0;64960]   \n",
       "SKESA               76053.75 [1404;351014]      75134.92 [0;351014]   \n",
       "SPAdes                117650.97 [0;315090]     117061.69 [0;315090]   \n",
       "Unicycler             162270.42 [0;475179]     158177.44 [0;475179]   \n",
       "VelvetOptimiser        29632.83 [0;338923]      34682.42 [0;464977]   \n",
       "\n",
       "                                 Ns         Type                  basepairs  \\\n",
       "Assembler                                                                     \n",
       "ABySS                92.14 [0;2170]  Metagenomic      2534947.7 [0;6780808]   \n",
       "BCALM2                    0.0 [0;0]      Genomic     1236062.67 [0;6510588]   \n",
       "GATBMiniaPipeline         0.0 [0;0]  Metagenomic  2523829.08 [2176;6716522]   \n",
       "IDBA-UD                   0.0 [0;0]  Metagenomic     2507600.54 [0;6710005]   \n",
       "MEGAHIT                   0.0 [0;0]  Metagenomic  2534425.59 [2215;6737986]   \n",
       "MetaHipMer2               0.0 [0;0]  Metagenomic     1722404.01 [0;6717767]   \n",
       "metaSPAdes                0.0 [0;0]  Metagenomic  2530047.11 [2076;6726884]   \n",
       "MINIA                     0.0 [0;0]      Genomic     2345242.24 [0;6518607]   \n",
       "SKESA                     0.0 [0;0]      Genomic  2496709.31 [1460;6673601]   \n",
       "SPAdes                    0.0 [0;0]      Genomic      2524122.0 [0;6728888]   \n",
       "Unicycler                 0.0 [0;0]      Genomic     2515166.11 [0;6723626]   \n",
       "VelvetOptimiser    537.64 [0;12098]  Metagenomic     1580537.53 [0;6708956]   \n",
       "\n",
       "                  breadth_of_coverage          contigs    identity  \\\n",
       "Assembler                                                            \n",
       "ABySS                      0.91 [0;0]   97.11 [0;1404]  0.98 [0;1]   \n",
       "BCALM2                     0.37 [0;0]  272.97 [0;1495]   0.5 [0;1]   \n",
       "GATBMiniaPipeline          0.98 [0;1]    59.92 [1;199]   1.0 [0;1]   \n",
       "IDBA-UD                    0.94 [0;1]   121.12 [0;449]  0.97 [0;1]   \n",
       "MEGAHIT                    0.99 [0;1]    44.39 [1;141]   1.0 [0;1]   \n",
       "MetaHipMer2                0.68 [0;0]    74.19 [0;361]  0.94 [0;1]   \n",
       "metaSPAdes                 0.98 [0;1]    39.33 [1;167]   1.0 [0;1]   \n",
       "MINIA                      0.74 [0;0]  307.84 [0;1147]  0.81 [0;1]   \n",
       "SKESA                       0.9 [0;0]    96.06 [1;933]   1.0 [0;1]   \n",
       "SPAdes                      0.9 [0;1]    41.19 [0;143]  0.94 [0;1]   \n",
       "Unicycler                  0.92 [0;1]    32.86 [0;111]  0.94 [0;1]   \n",
       "VelvetOptimiser            0.52 [0;0]  167.92 [0;1098]  0.56 [0;1]   \n",
       "\n",
       "                  lowest_identity misassembled contigs misassembly events  \\\n",
       "Assembler                                                                   \n",
       "ABySS                  0.94 [0;1]           0.33 [0;8]        0.59 [0;14]   \n",
       "BCALM2                  0.5 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "GATBMiniaPipeline      0.98 [0;1]           0.14 [0;3]         0.25 [0;6]   \n",
       "IDBA-UD                0.87 [0;1]           1.12 [0;9]        1.94 [0;15]   \n",
       "MEGAHIT                0.93 [0;1]           0.97 [0;4]         1.99 [0;7]   \n",
       "MetaHipMer2            0.94 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "metaSPAdes             0.99 [0;1]           0.14 [0;2]         0.33 [0;6]   \n",
       "MINIA                  0.79 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "SKESA                  0.87 [0;1]           0.67 [0;7]        1.17 [0;15]   \n",
       "SPAdes                  0.9 [0;1]           0.36 [0;4]         0.69 [0;8]   \n",
       "Unicycler              0.94 [0;1]           0.06 [0;1]         0.11 [0;2]   \n",
       "VelvetOptimiser        0.55 [0;1]           0.39 [0;8]        0.61 [0;11]   \n",
       "\n",
       "                  multiplicity   parsimony    validity  \n",
       "Assembler                                               \n",
       "ABySS               0.99 [0;1]  0.99 [0;1]  0.98 [0;1]  \n",
       "BCALM2               0.5 [0;1]   0.5 [0;1]   0.5 [0;1]  \n",
       "GATBMiniaPipeline   1.01 [0;1]  1.01 [0;1]   1.0 [0;1]  \n",
       "IDBA-UD             0.98 [0;1]  0.98 [0;1]  0.97 [0;1]  \n",
       "MEGAHIT              1.0 [0;1]  1.01 [0;1]   1.0 [0;1]  \n",
       "MetaHipMer2         0.94 [0;1]  0.94 [0;1]  0.94 [0;1]  \n",
       "metaSPAdes          1.01 [0;1]  1.01 [0;1]   1.0 [0;1]  \n",
       "MINIA               0.81 [0;1]  0.81 [0;1]  0.81 [0;1]  \n",
       "SKESA                1.0 [0;1]   1.0 [0;1]   1.0 [0;1]  \n",
       "SPAdes              0.95 [0;1]  0.96 [0;1]  0.94 [0;1]  \n",
       "Unicycler           0.94 [0;1]  0.94 [0;1]  0.94 [0;1]  \n",
       "VelvetOptimiser     0.56 [0;1]  0.56 [0;1]  0.56 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "for distribution in reference_pipeline_metrics_df.distribution.unique():\n",
    "    ref_stats_per_reference_per_assembler = pd.DataFrame()\n",
    "    for assembler in reference_pipeline_metrics_df.assembler.unique():\n",
    "        row={'Assembler': assembler, \n",
    "             'Type':reference_pipeline_metrics_df[reference_pipeline_metrics_df.assembler == assembler].type.unique()[0],\n",
    "             'Algorithm': reference_pipeline_metrics_df[reference_pipeline_metrics_df.assembler == assembler].algorythm.unique()[0]}\n",
    "        lala=reference_pipeline_metrics_df[(reference_pipeline_metrics_df.assembler == assembler) & (reference_pipeline_metrics_df.distribution == distribution)].describe()\n",
    "        for column in lala.columns:\n",
    "            mean = lala.loc['mean',column]\n",
    "            minimum = lala.loc['min',column]\n",
    "            maximum = lala.loc['max',column]\n",
    "            row[column] = \"{} [{};{}]\".format(round(mean, 2),int(minimum), int(maximum))\n",
    "        ref_stats_per_reference_per_assembler=ref_stats_per_reference_per_assembler.append(row, ignore_index=True)\n",
    "    ref_stats_per_reference_per_assembler = ref_stats_per_reference_per_assembler.set_index('Assembler')\n",
    "    ref_stats_per_reference_per_assembler.to_csv(\"Tables/Results/{} - Reference metrics per assembler.csv\".format(distribution))\n",
    "    print(distribution)\n",
    "    display(ref_stats_per_reference_per_assembler)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Genomes vs Plasmids"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Replicon</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Genome</th>\n",
       "      <td>47.95(± 107.54)</td>\n",
       "      <td>0.06(± 0.08)</td>\n",
       "      <td>93543.9(± 121264.69)</td>\n",
       "      <td>92430.74(± 121287.75)</td>\n",
       "      <td>57.03(± 683.46)</td>\n",
       "      <td>2585647.14(± 2155710.78)</td>\n",
       "      <td>0.65(± 0.45)</td>\n",
       "      <td>147.71(± 307.44)</td>\n",
       "      <td>0.78(± 0.41)</td>\n",
       "      <td>0.73(± 0.41)</td>\n",
       "      <td>1.23(± 21.49)</td>\n",
       "      <td>2.4(± 43.13)</td>\n",
       "      <td>0.78(± 0.41)</td>\n",
       "      <td>0.79(± 0.46)</td>\n",
       "      <td>0.78(± 0.41)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Plasmid</th>\n",
       "      <td>0.52 [0;18]</td>\n",
       "      <td>0.36 [0;1]</td>\n",
       "      <td>4081.6 [0;55639]</td>\n",
       "      <td>3985.8 [0;55639]</td>\n",
       "      <td>5.81 [0;835]</td>\n",
       "      <td>14065.64 [0;109516]</td>\n",
       "      <td>0.44 [0;1]</td>\n",
       "      <td>1.8 [0;33]</td>\n",
       "      <td>0.49 [0;1]</td>\n",
       "      <td>0.49 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.49 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.49 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      L90           LSA                  NA50  \\\n",
       "Replicon                                                        \n",
       "Genome    47.95(± 107.54)  0.06(± 0.08)  93543.9(± 121264.69)   \n",
       "Plasmid       0.52 [0;18]    0.36 [0;1]      4081.6 [0;55639]   \n",
       "\n",
       "                           NG50               Ns                 basepairs  \\\n",
       "Replicon                                                                     \n",
       "Genome    92430.74(± 121287.75)  57.03(± 683.46)  2585647.14(± 2155710.78)   \n",
       "Plasmid        3985.8 [0;55639]     5.81 [0;835]       14065.64 [0;109516]   \n",
       "\n",
       "         breadth_of_coverage           contigs      identity lowest_identity  \\\n",
       "Replicon                                                                       \n",
       "Genome          0.65(± 0.45)  147.71(± 307.44)  0.78(± 0.41)    0.73(± 0.41)   \n",
       "Plasmid           0.44 [0;1]        1.8 [0;33]    0.49 [0;1]      0.49 [0;1]   \n",
       "\n",
       "         misassembled contigs misassembly events  multiplicity     parsimony  \\\n",
       "Replicon                                                                       \n",
       "Genome          1.23(± 21.49)       2.4(± 43.13)  0.78(± 0.41)  0.79(± 0.46)   \n",
       "Plasmid             0.0 [0;0]          0.0 [0;0]    0.49 [0;1]     0.5 [0;1]   \n",
       "\n",
       "              validity  \n",
       "Replicon                \n",
       "Genome    0.78(± 0.41)  \n",
       "Plasmid     0.49 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ref_stats_per_replicon_type = pd.DataFrame()\n",
    "# Genome\n",
    "row={'Replicon': \"Genome\"}\n",
    "lala=reference_pipeline_metrics_df[~reference_pipeline_metrics_df['reference'].str.contains(\"plasmid\")].describe()\n",
    "for column in lala.columns:\n",
    "    mean = lala.loc['mean',column]\n",
    "    std = lala.loc['std',column]\n",
    "    row[column] = \"{}(± {})\".format(round(mean, 2), round(std, 2))\n",
    "ref_stats_per_replicon_type =ref_stats_per_replicon_type .append(row, ignore_index=True)\n",
    "\n",
    "# Plasmid\n",
    "row={'Replicon': \"Plasmid\"}\n",
    "lala=reference_pipeline_metrics_df[reference_pipeline_metrics_df['reference'].str.contains(\"plasmid\")].describe()\n",
    "for column in lala.columns:\n",
    "    mean = lala.loc['mean',column]\n",
    "    minimum = lala.loc['min',column]\n",
    "    maximum = lala.loc['max',column]\n",
    "    row[column] = \"{} [{};{}]\".format(round(mean, 2),int(minimum), int(maximum))\n",
    "    \n",
    "ref_stats_per_replicon_type =ref_stats_per_replicon_type .append(row, ignore_index=True)\n",
    "ref_stats_per_replicon_type  = ref_stats_per_replicon_type .set_index('Replicon')\n",
    "ref_stats_per_replicon_type.to_csv(\"Tables/Results/Reference metrics per replicon type.csv\")\n",
    "display(ref_stats_per_replicon_type)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
