{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 1. Compute Results Overview Tables"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The success of an assembly is evaluated by the computation of metrics in two defined ways: globally through statistics inherent to the complete set of sequences that were assembled, and relative to the replicons present in the sample. \n",
    "\n",
    "The computation of the global metrics is done through custom python code from the complete set of sequences assembled. \n",
    "\n",
    "The following metrics are computed for the complete and filtered set of assembled sequences, restricted to contigs of length above a specified minimum size: \n",
    "\n",
    "- **Contig sizes**\n",
    "    - **Contigs:** The total number of contigs in the assembly;\n",
    "    - **Basepairs:** The total number of bases in the assembly;\n",
    "    - **Maximum sequence length:** The length of the largest contig in the assembly.\n",
    "    - **Number of ‘N’s:** Number of uncalled bases (N's) \n",
    "- **Contiguity**\n",
    "    - **Nx (where 0  < x  ⩽ 100):** Length for which the collection of all assembled sequences of that length or longer in an assembly covers at least a given percentage of the total length of the assembly\n",
    "- **Misassembly**\n",
    "    - **Misassemblies** - Number of aligned contigs that contain a misassembly event\n",
    "\n",
    "After filtering, the sequences are mapped with the reference replicons and the metrics are computed through custom python code. The metrics are calculated, filtered for a minimum length, for each replicon in the file provided input references. \n",
    "\n",
    "- **Contig sizes**\n",
    "    - **Contigs:** The total number of contigs in the assembly;\n",
    "    - **Basepairs:** The total number of bases in the assembly;\n",
    "    - **Number of ‘N’s:** Number of uncalled bases (N's) \n",
    "- **COMPASS**\n",
    "    - **(Breadth of) Coverage:** Ratio of covered sequence on the reference by aligned contigs;\n",
    "    - **Multiplicity:** Ratio of the length of alignable assembled sequence to covered sequence on the reference;\n",
    "    - **Validity:** Ratio of the length of the alignable assembled sequence to total basepairs in the aligned contigs;\n",
    "    - **Parsimony:** Cost of the assembly (multiplicity over validity);\n",
    "- **Contiguity**\n",
    "    - **Contiguity:** longest single alignment between the assembly and the reference, relative to the reference length;\n",
    "    - **NAx (where 0  < x  ⩽ 100):** Length for which the collection of aligned assembled sequences of that length or longer in an assembly covers at least a given percentage of the total length of the reference replicon;\n",
    "    - **NGx (where 0  < x  ⩽ 100):** Length for which the collection of aligned contigs of that length or longer covers at least a given percentage of the sequence of the reference.\n",
    "    - **Lx (where 0  < x  ⩽ 100):** Minimal number of contigs that cover x % of the sequence of the reference;\n",
    "- **Identity**\n",
    "    - **Identity:** Ratio of identical basepairs in all aligned contigs to the reference;\n",
    "    - **Lowest identity:** Identity of the lowest scoring contig to the reference.\n",
    "    - **PLS**: Phred-like score per contig, per assembler.\n",
    "- **Misassembly**\n",
    "    - **Misassemblies** - Number of aligned contigs that contain a misassembly event\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "from plotly.offline import plot\n",
    "import glob\n",
    "import fnmatch\n",
    "import plotly.graph_objects as go\n",
    "from plotly.subplots import make_subplots\n",
    "import json\n",
    "import pandas as pd\n",
    "from itertools import groupby\n",
    "import csv\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Global variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [],
   "source": [
    "ASSEMBLER_PROCESS_LIST = [\"ABYSS\", \"BCALM2\", \"GATBMINIAPIPELINE\", \"METAHIPMER2\", \"MINIA\", \"MEGAHIT\", \"METASPADES\", \"UNICYCLER\", \"SPADES\",\n",
    "                          \"SKESA\", \"VELVETOPTIMIZER\", \"IDBA\"]\n",
    "PROCESS_TO_NAME = {\"ABYSS\": \"AbYSS\",\n",
    "                   \"BCALM2\": \"BCALM2\", \n",
    "                   \"GATBMINIAPIPELINE\": \"GATBMiniaPipeline\",\n",
    "                   \"METAHIPMER2\": \"MetaHipMer2\",\n",
    "                   \"MINIA\": \"MINIA\",\n",
    "                   \"MEGAHIT\": \"MEGAHIT\", \n",
    "                   \"METASPADES\": \"metaSPAdes\", \n",
    "                   \"UNICYCLER\": \"Unicycler\", \n",
    "                   \"SPADES\": \"SPAdes\",\n",
    "                    \"SKESA\": \"SKESA\",\n",
    "                   \"VELVETOPTIMIZER\": \"VelvetOptimizer\",\n",
    "                   \"IDBA\": \"IDBA-UD\"}\n",
    "\n",
    "log_distributed = ['LHS', 'LNN', 'ERR2935805']\n",
    "even_distribution = ['EMS', 'LNN', 'ERR2984773']\n",
    "\n",
    "genomic_assemblers = ['AbYSS', 'BCALM2', 'MINIA', 'SKESA', 'SPAdes', 'Unicycler', 'VelvetOptimizer']\n",
    "metagenomic_assemblers = ['MetaHipMer2','GATBMiniaPipeline', 'IDBA-UD', 'MEGAHIT', 'metaSPAdes']\n",
    "single_kmer = ['BCALM2', 'MINIA', 'AbYSS']\n",
    "multiple_kmer = ['SKESA', 'SPAdes', 'Unicycler', 'VelvetOptimizer', 'GATBMiniaPipeline', \n",
    "                 'IDBA-UD', 'MEGAHIT', 'metaSPAdes', 'MetaHipMer2']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Global metrics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pipeline_report_tables.json data from run1...\n",
      "Processing pipeline_report_tables.json data from run2...\n",
      "Processing pipeline_report_tables.json data from run3...\n"
     ]
    }
   ],
   "source": [
    "report_glob = glob.glob('../Results/*/*/report/pipeline_report_tables.json')\n",
    "global_pipeline_metrics_df = pd.DataFrame()\n",
    "\n",
    "for pipeline_report_file in report_glob:\n",
    "    report_file_name = pipeline_report_file.split('/')[-1]\n",
    "    stats_run = pipeline_report_file.split('/')[-3]\n",
    "    print('Processing {0} data from {1}...'.format(report_file_name, stats_run))\n",
    "    \n",
    "    with open(pipeline_report_file) as _fh:\n",
    "        json_report = json.load(_fh)\n",
    "        for sample in json_report.keys():\n",
    "            for line in json_report[sample]['GlobalTable']:\n",
    "                assembler = line['assembler']\n",
    "                global_pipeline_metrics_df = global_pipeline_metrics_df.append({'run': stats_run,\n",
    "                                                                                'sample': sample,\n",
    "                                                                                'assembler': line['assembler'],\n",
    "                                                                                'contigs': int(line['original']['contigs']),\n",
    "                                                                                'basepairs': int(line['original']['basepairs']),\n",
    "                                                                                'max_contig': int(line['original']['max_contig_size']),\n",
    "                                                                                'n50': int(line['original']['N50']),\n",
    "                                                                                'mapped_reads': line['original']['mapped_reads'],\n",
    "                                                                                'Ns': int(line['original']['Ns']),\n",
    "                                                                                'misassembled contigs': line['filtered']['misassembled_contigs'],\n",
    "                                                                                'misassembly events': line['filtered']['misassembly_events'],\n",
    "                                                                                'filtered_contigs': line['filtered']['contigs'],\n",
    "                                                                                'filtered_basepairs': line['filtered']['basepairs'],\n",
    "                                                                                'filtered_n50': line['filtered']['N50'],\n",
    "                                                                                'filtered_Ns': line['filtered']['Ns'],\n",
    "                                                                                'filtered_mapped_reads': line['filtered']['mapped_reads'],\n",
    "                                                                               },\n",
    "                                                                               ignore_index=True)\n",
    "global_pipeline_metrics_df['distribution'] = np.where(global_pipeline_metrics_df['sample'].isin(log_distributed), 'Log', 'Even')\n",
    "global_pipeline_metrics_df['type'] = np.where(global_pipeline_metrics_df['assembler'].isin(genomic_assemblers), 'Genomic', 'Metagenomic')\n",
    "global_pipeline_metrics_df['algorythm'] = np.where(global_pipeline_metrics_df['assembler'].isin(single_kmer), 'Single k-mer De Bruijn graph', 'Multiple k-mer De Bruijn graph')\n",
    "global_pipeline_metrics_df[['contigs','basepairs','max_contig', 'Ns','n50', 'filtered_n50','misassembled contigs', 'misassembly events']] = global_pipeline_metrics_df[['contigs','basepairs','max_contig', 'Ns','n50','filtered_n50','misassembled contigs', 'misassembly events']].apply(pd.to_numeric)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Preview data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Ns                       float64\n",
       "assembler                 object\n",
       "basepairs                float64\n",
       "contigs                  float64\n",
       "filtered_Ns              float64\n",
       "filtered_basepairs       float64\n",
       "filtered_contigs         float64\n",
       "filtered_mapped_reads    float64\n",
       "filtered_n50             float64\n",
       "mapped_reads             float64\n",
       "max_contig               float64\n",
       "misassembled contigs     float64\n",
       "misassembly events       float64\n",
       "n50                      float64\n",
       "run                       object\n",
       "sample                    object\n",
       "distribution              object\n",
       "type                      object\n",
       "algorythm                 object\n",
       "dtype: object"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "global_pipeline_metrics_df.dtypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ns</th>\n",
       "      <th>assembler</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>contigs</th>\n",
       "      <th>filtered_Ns</th>\n",
       "      <th>filtered_basepairs</th>\n",
       "      <th>filtered_contigs</th>\n",
       "      <th>filtered_mapped_reads</th>\n",
       "      <th>filtered_n50</th>\n",
       "      <th>mapped_reads</th>\n",
       "      <th>max_contig</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>n50</th>\n",
       "      <th>run</th>\n",
       "      <th>sample</th>\n",
       "      <th>distribution</th>\n",
       "      <th>type</th>\n",
       "      <th>algorythm</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>147.0</td>\n",
       "      <td>ABySS</td>\n",
       "      <td>91720949.0</td>\n",
       "      <td>497783.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>30657287.0</td>\n",
       "      <td>1196.0</td>\n",
       "      <td>82.019612</td>\n",
       "      <td>52869.0</td>\n",
       "      <td>92.343671</td>\n",
       "      <td>298093.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>139.0</td>\n",
       "      <td>run1</td>\n",
       "      <td>ERR2984773</td>\n",
       "      <td>Even</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>BCALM2</td>\n",
       "      <td>170402348.0</td>\n",
       "      <td>3508016.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8.151058</td>\n",
       "      <td>994.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>run1</td>\n",
       "      <td>ERR2984773</td>\n",
       "      <td>Even</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.0</td>\n",
       "      <td>GATBMiniaPipeline</td>\n",
       "      <td>32809112.0</td>\n",
       "      <td>5397.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30564302.0</td>\n",
       "      <td>761.0</td>\n",
       "      <td>82.840039</td>\n",
       "      <td>113162.0</td>\n",
       "      <td>84.068286</td>\n",
       "      <td>768462.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>102115.0</td>\n",
       "      <td>run1</td>\n",
       "      <td>ERR2984773</td>\n",
       "      <td>Even</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.0</td>\n",
       "      <td>IDBA-UD</td>\n",
       "      <td>39773020.0</td>\n",
       "      <td>20164.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>31562690.0</td>\n",
       "      <td>2654.0</td>\n",
       "      <td>81.596739</td>\n",
       "      <td>29134.0</td>\n",
       "      <td>84.336773</td>\n",
       "      <td>303329.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>20993.0</td>\n",
       "      <td>run1</td>\n",
       "      <td>ERR2984773</td>\n",
       "      <td>Even</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.0</td>\n",
       "      <td>MEGAHIT</td>\n",
       "      <td>41166407.0</td>\n",
       "      <td>20614.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>32021766.0</td>\n",
       "      <td>1447.0</td>\n",
       "      <td>84.138912</td>\n",
       "      <td>149841.0</td>\n",
       "      <td>86.166393</td>\n",
       "      <td>1205839.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>88308.0</td>\n",
       "      <td>run1</td>\n",
       "      <td>ERR2984773</td>\n",
       "      <td>Even</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>0.0</td>\n",
       "      <td>MINIA</td>\n",
       "      <td>41187149.0</td>\n",
       "      <td>67998.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28276844.0</td>\n",
       "      <td>3973.0</td>\n",
       "      <td>74.390822</td>\n",
       "      <td>15912.0</td>\n",
       "      <td>80.663396</td>\n",
       "      <td>117945.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7106.0</td>\n",
       "      <td>run3</td>\n",
       "      <td>ERR2984773</td>\n",
       "      <td>Even</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>0.0</td>\n",
       "      <td>SKESA</td>\n",
       "      <td>30430077.0</td>\n",
       "      <td>1352.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30211740.0</td>\n",
       "      <td>846.0</td>\n",
       "      <td>80.664130</td>\n",
       "      <td>77870.0</td>\n",
       "      <td>81.479714</td>\n",
       "      <td>446163.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>76924.0</td>\n",
       "      <td>run3</td>\n",
       "      <td>ERR2984773</td>\n",
       "      <td>Even</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>0.0</td>\n",
       "      <td>SPAdes</td>\n",
       "      <td>46535940.0</td>\n",
       "      <td>39819.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>32500826.0</td>\n",
       "      <td>2049.0</td>\n",
       "      <td>82.045580</td>\n",
       "      <td>99299.0</td>\n",
       "      <td>85.446436</td>\n",
       "      <td>745370.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>53292.0</td>\n",
       "      <td>run3</td>\n",
       "      <td>ERR2984773</td>\n",
       "      <td>Even</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>0.0</td>\n",
       "      <td>Unicycler</td>\n",
       "      <td>30592960.0</td>\n",
       "      <td>895.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30432114.0</td>\n",
       "      <td>425.0</td>\n",
       "      <td>81.941639</td>\n",
       "      <td>191424.0</td>\n",
       "      <td>82.823086</td>\n",
       "      <td>882182.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>191424.0</td>\n",
       "      <td>run3</td>\n",
       "      <td>ERR2984773</td>\n",
       "      <td>Even</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>551.0</td>\n",
       "      <td>VelvetOptimiser</td>\n",
       "      <td>54970900.0</td>\n",
       "      <td>595080.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>35406.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>0.005321</td>\n",
       "      <td>1080.0</td>\n",
       "      <td>2.028585</td>\n",
       "      <td>1488.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>88.0</td>\n",
       "      <td>run3</td>\n",
       "      <td>ERR2984773</td>\n",
       "      <td>Even</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>106 rows × 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Ns          assembler    basepairs    contigs  filtered_Ns  \\\n",
       "0    147.0              ABySS   91720949.0   497783.0          5.0   \n",
       "1      0.0             BCALM2  170402348.0  3508016.0          0.0   \n",
       "2      0.0  GATBMiniaPipeline   32809112.0     5397.0          0.0   \n",
       "3      0.0            IDBA-UD   39773020.0    20164.0          0.0   \n",
       "4      0.0            MEGAHIT   41166407.0    20614.0          0.0   \n",
       "..     ...                ...          ...        ...          ...   \n",
       "101    0.0              MINIA   41187149.0    67998.0          0.0   \n",
       "102    0.0              SKESA   30430077.0     1352.0          0.0   \n",
       "103    0.0             SPAdes   46535940.0    39819.0          0.0   \n",
       "104    0.0          Unicycler   30592960.0      895.0          0.0   \n",
       "105  551.0    VelvetOptimiser   54970900.0   595080.0          0.0   \n",
       "\n",
       "     filtered_basepairs  filtered_contigs  filtered_mapped_reads  \\\n",
       "0            30657287.0            1196.0              82.019612   \n",
       "1                   0.0               0.0               0.000000   \n",
       "2            30564302.0             761.0              82.840039   \n",
       "3            31562690.0            2654.0              81.596739   \n",
       "4            32021766.0            1447.0              84.138912   \n",
       "..                  ...               ...                    ...   \n",
       "101          28276844.0            3973.0              74.390822   \n",
       "102          30211740.0             846.0              80.664130   \n",
       "103          32500826.0            2049.0              82.045580   \n",
       "104          30432114.0             425.0              81.941639   \n",
       "105             35406.0              32.0               0.005321   \n",
       "\n",
       "     filtered_n50  mapped_reads  max_contig  misassembled contigs  \\\n",
       "0         52869.0     92.343671    298093.0                   2.0   \n",
       "1             0.0      8.151058       994.0                   0.0   \n",
       "2        113162.0     84.068286    768462.0                   1.0   \n",
       "3         29134.0     84.336773    303329.0                   4.0   \n",
       "4        149841.0     86.166393   1205839.0                  15.0   \n",
       "..            ...           ...         ...                   ...   \n",
       "101       15912.0     80.663396    117945.0                   0.0   \n",
       "102       77870.0     81.479714    446163.0                   0.0   \n",
       "103       99299.0     85.446436    745370.0                   9.0   \n",
       "104      191424.0     82.823086    882182.0                   2.0   \n",
       "105        1080.0      2.028585      1488.0                   0.0   \n",
       "\n",
       "     misassembly events       n50   run      sample distribution         type  \\\n",
       "0                   6.0     139.0  run1  ERR2984773         Even  Metagenomic   \n",
       "1                   0.0      49.0  run1  ERR2984773         Even      Genomic   \n",
       "2                   2.0  102115.0  run1  ERR2984773         Even  Metagenomic   \n",
       "3                   7.0   20993.0  run1  ERR2984773         Even  Metagenomic   \n",
       "4                  32.0   88308.0  run1  ERR2984773         Even  Metagenomic   \n",
       "..                  ...       ...   ...         ...          ...          ...   \n",
       "101                 0.0    7106.0  run3  ERR2984773         Even      Genomic   \n",
       "102                 0.0   76924.0  run3  ERR2984773         Even      Genomic   \n",
       "103                18.0   53292.0  run3  ERR2984773         Even      Genomic   \n",
       "104                 4.0  191424.0  run3  ERR2984773         Even      Genomic   \n",
       "105                 0.0      88.0  run3  ERR2984773         Even  Metagenomic   \n",
       "\n",
       "                          algorythm  \n",
       "0    Multiple k-mer De Bruijn graph  \n",
       "1      Single k-mer De Bruijn graph  \n",
       "2    Multiple k-mer De Bruijn graph  \n",
       "3    Multiple k-mer De Bruijn graph  \n",
       "4    Multiple k-mer De Bruijn graph  \n",
       "..                              ...  \n",
       "101    Single k-mer De Bruijn graph  \n",
       "102  Multiple k-mer De Bruijn graph  \n",
       "103  Multiple k-mer De Bruijn graph  \n",
       "104  Multiple k-mer De Bruijn graph  \n",
       "105  Multiple k-mer De Bruijn graph  \n",
       "\n",
       "[106 rows x 19 columns]"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "global_pipeline_metrics_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Save data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "global_pipeline_metrics_df.to_csv(\"Tables/Results/Global metrics.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Global statistics per assembler type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [],
   "source": [
    "stats_per_assembler_type = pd.DataFrame()\n",
    "for assembly_type in global_pipeline_metrics_df.type.unique():\n",
    "    row={'Assembler Type': assembly_type}\n",
    "    lala=global_pipeline_metrics_df[global_pipeline_metrics_df.type == assembly_type].describe()\n",
    "    for column in lala.columns:\n",
    "        mean = lala.loc['mean',column]\n",
    "        minimum = lala.loc['min',column]\n",
    "        maximum = lala.loc['max',column]\n",
    "        row[column] = \"{} [{};{}]\".format(round(mean, 2), int(minimum), int(maximum))\n",
    "    stats_per_assembler_type=stats_per_assembler_type.append(row, ignore_index=True)\n",
    "stats_per_assembler_type = stats_per_assembler_type.set_index('Assembler Type')\n",
    "stats_per_assembler_type.to_csv(\"Tables/Results/Global metrics per assembler type.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ns</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>contigs</th>\n",
       "      <th>filtered_Ns</th>\n",
       "      <th>filtered_basepairs</th>\n",
       "      <th>filtered_contigs</th>\n",
       "      <th>filtered_mapped_reads</th>\n",
       "      <th>filtered_n50</th>\n",
       "      <th>mapped_reads</th>\n",
       "      <th>max_contig</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>n50</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler Type</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Metagenomic</th>\n",
       "      <td>899.21 [0;8788]</td>\n",
       "      <td>34000575.0 [13174826;91720949]</td>\n",
       "      <td>45733.13 [640;595080]</td>\n",
       "      <td>767.56 [0;8283]</td>\n",
       "      <td>27525314.28 [13883;32021766]</td>\n",
       "      <td>1171.89 [13;3035]</td>\n",
       "      <td>86.1 [0;107]</td>\n",
       "      <td>100984.79 [1039;238611]</td>\n",
       "      <td>88.84 [2;108]</td>\n",
       "      <td>643814.77 [1157;1205839]</td>\n",
       "      <td>5.05 [0;18]</td>\n",
       "      <td>9.62 [0;36]</td>\n",
       "      <td>91833.59 [88;238611]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic</th>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>42206039.13 [30110758;170402348]</td>\n",
       "      <td>262526.38 [656;3508016]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>26882765.51 [0;32500826]</td>\n",
       "      <td>1902.33 [0;6250]</td>\n",
       "      <td>80.88 [0;99]</td>\n",
       "      <td>95630.53 [0;245050]</td>\n",
       "      <td>85.91 [8;100]</td>\n",
       "      <td>522015.4 [994;1303875]</td>\n",
       "      <td>2.2 [0;10]</td>\n",
       "      <td>4.33 [0;19]</td>\n",
       "      <td>90605.67 [49;245050]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             Ns                         basepairs  \\\n",
       "Assembler Type                                                      \n",
       "Metagenomic     899.21 [0;8788]    34000575.0 [13174826;91720949]   \n",
       "Genomic               0.0 [0;0]  42206039.13 [30110758;170402348]   \n",
       "\n",
       "                                contigs      filtered_Ns  \\\n",
       "Assembler Type                                             \n",
       "Metagenomic       45733.13 [640;595080]  767.56 [0;8283]   \n",
       "Genomic         262526.38 [656;3508016]        0.0 [0;0]   \n",
       "\n",
       "                          filtered_basepairs   filtered_contigs  \\\n",
       "Assembler Type                                                    \n",
       "Metagenomic     27525314.28 [13883;32021766]  1171.89 [13;3035]   \n",
       "Genomic             26882765.51 [0;32500826]   1902.33 [0;6250]   \n",
       "\n",
       "               filtered_mapped_reads             filtered_n50   mapped_reads  \\\n",
       "Assembler Type                                                                 \n",
       "Metagenomic             86.1 [0;107]  100984.79 [1039;238611]  88.84 [2;108]   \n",
       "Genomic                 80.88 [0;99]      95630.53 [0;245050]  85.91 [8;100]   \n",
       "\n",
       "                              max_contig misassembled contigs  \\\n",
       "Assembler Type                                                  \n",
       "Metagenomic     643814.77 [1157;1205839]          5.05 [0;18]   \n",
       "Genomic           522015.4 [994;1303875]           2.2 [0;10]   \n",
       "\n",
       "               misassembly events                   n50  \n",
       "Assembler Type                                           \n",
       "Metagenomic           9.62 [0;36]  91833.59 [88;238611]  \n",
       "Genomic               4.33 [0;19]  90605.67 [49;245050]  "
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "stats_per_assembler_type"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Global stats per assembly algorithm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "stats_per_assembler_algo = pd.DataFrame()\n",
    "for assembly_algo in global_pipeline_metrics_df.algorythm.unique():\n",
    "    row={'Assembler Algorithm': assembly_algo}\n",
    "    lala=global_pipeline_metrics_df[global_pipeline_metrics_df.algorythm == assembly_algo].describe()\n",
    "    for column in lala.columns:\n",
    "        mean = lala.loc['mean',column]\n",
    "        minimum = lala.loc['min',column]\n",
    "        maximum = lala.loc['max',column]\n",
    "        row[column] = \"{} [{};{}]\".format(round(mean, 2), int(minimum), int(maximum))\n",
    "    stats_per_assembler_algo=stats_per_assembler_algo.append(row, ignore_index=True)\n",
    "stats_per_assembler_algo = stats_per_assembler_algo.set_index('Assembler Algorithm')\n",
    "stats_per_assembler_algo.to_csv(\"Tables/Results/Global metrics per assembler algorithm.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ns</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>contigs</th>\n",
       "      <th>filtered_Ns</th>\n",
       "      <th>filtered_basepairs</th>\n",
       "      <th>filtered_contigs</th>\n",
       "      <th>filtered_mapped_reads</th>\n",
       "      <th>filtered_n50</th>\n",
       "      <th>mapped_reads</th>\n",
       "      <th>max_contig</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>n50</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler Algorithm</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Multiple k-mer De Bruijn graph</th>\n",
       "      <td>623.32 [0;8788]</td>\n",
       "      <td>33441027.89 [13174826;91720949]</td>\n",
       "      <td>33440.59 [640;595080]</td>\n",
       "      <td>532.06 [0;8283]</td>\n",
       "      <td>28419512.4 [13883;32500826]</td>\n",
       "      <td>1070.4 [13;3035]</td>\n",
       "      <td>88.11 [0;107]</td>\n",
       "      <td>116138.57 [1039;245050]</td>\n",
       "      <td>90.38 [2;108]</td>\n",
       "      <td>691320.52 [1157;1303875]</td>\n",
       "      <td>4.62 [0;18]</td>\n",
       "      <td>8.89 [0;36]</td>\n",
       "      <td>107910.89 [88;245050]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Single k-mer De Bruijn graph</th>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>57249799.0 [30641333;170402348]</td>\n",
       "      <td>647813.11 [16570;3508016]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>21547307.11 [0;28279162]</td>\n",
       "      <td>3494.17 [0;6250]</td>\n",
       "      <td>63.2 [0;92]</td>\n",
       "      <td>13514.0 [0;20782]</td>\n",
       "      <td>73.97 [8;97]</td>\n",
       "      <td>107066.0 [994;161812]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>10163.67 [49;17967]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             Ns  \\\n",
       "Assembler Algorithm                               \n",
       "Multiple k-mer De Bruijn graph  623.32 [0;8788]   \n",
       "Single k-mer De Bruijn graph          0.0 [0;0]   \n",
       "\n",
       "                                                      basepairs  \\\n",
       "Assembler Algorithm                                               \n",
       "Multiple k-mer De Bruijn graph  33441027.89 [13174826;91720949]   \n",
       "Single k-mer De Bruijn graph    57249799.0 [30641333;170402348]   \n",
       "\n",
       "                                                  contigs      filtered_Ns  \\\n",
       "Assembler Algorithm                                                          \n",
       "Multiple k-mer De Bruijn graph      33440.59 [640;595080]  532.06 [0;8283]   \n",
       "Single k-mer De Bruijn graph    647813.11 [16570;3508016]        0.0 [0;0]   \n",
       "\n",
       "                                         filtered_basepairs  filtered_contigs  \\\n",
       "Assembler Algorithm                                                             \n",
       "Multiple k-mer De Bruijn graph  28419512.4 [13883;32500826]  1070.4 [13;3035]   \n",
       "Single k-mer De Bruijn graph       21547307.11 [0;28279162]  3494.17 [0;6250]   \n",
       "\n",
       "                               filtered_mapped_reads             filtered_n50  \\\n",
       "Assembler Algorithm                                                             \n",
       "Multiple k-mer De Bruijn graph         88.11 [0;107]  116138.57 [1039;245050]   \n",
       "Single k-mer De Bruijn graph             63.2 [0;92]        13514.0 [0;20782]   \n",
       "\n",
       "                                 mapped_reads                max_contig  \\\n",
       "Assembler Algorithm                                                       \n",
       "Multiple k-mer De Bruijn graph  90.38 [2;108]  691320.52 [1157;1303875]   \n",
       "Single k-mer De Bruijn graph     73.97 [8;97]     107066.0 [994;161812]   \n",
       "\n",
       "                               misassembled contigs misassembly events  \\\n",
       "Assembler Algorithm                                                      \n",
       "Multiple k-mer De Bruijn graph          4.62 [0;18]        8.89 [0;36]   \n",
       "Single k-mer De Bruijn graph              0.0 [0;0]          0.0 [0;0]   \n",
       "\n",
       "                                                  n50  \n",
       "Assembler Algorithm                                    \n",
       "Multiple k-mer De Bruijn graph  107910.89 [88;245050]  \n",
       "Single k-mer De Bruijn graph      10163.67 [49;17967]  "
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "stats_per_assembler_algo"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Global stats per assembler"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "stats_per_assembler = pd.DataFrame()\n",
    "for assembler in global_pipeline_metrics_df.assembler.unique():\n",
    "    row={'Assembler': assembler, \n",
    "         'Type':global_pipeline_metrics_df[global_pipeline_metrics_df.assembler == assembler].type.unique()[0],\n",
    "         'Algorithm': global_pipeline_metrics_df[global_pipeline_metrics_df.assembler == assembler].algorythm.unique()[0]}\n",
    "    lala=global_pipeline_metrics_df[global_pipeline_metrics_df.assembler == assembler].describe()\n",
    "    for column in lala.columns:\n",
    "        mean = lala.loc['mean',column]\n",
    "        minimum = lala.loc['min',column]\n",
    "        maximum = lala.loc['max',column]\n",
    "        row[column] = \"{} [{};{}]\".format(round(mean, 2), int(minimum), int(maximum))\n",
    "    stats_per_assembler=stats_per_assembler.append(row, ignore_index=True)\n",
    "stats_per_assembler = stats_per_assembler.set_index('Assembler')\n",
    "stats_per_assembler.to_csv(\"Tables/Results/Global metrics per assembler.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>contigs</th>\n",
       "      <th>filtered_Ns</th>\n",
       "      <th>filtered_basepairs</th>\n",
       "      <th>filtered_contigs</th>\n",
       "      <th>filtered_mapped_reads</th>\n",
       "      <th>filtered_n50</th>\n",
       "      <th>mapped_reads</th>\n",
       "      <th>max_contig</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>n50</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>2062.43 [147;3884]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>41418778.14 [31502565;91720949]</td>\n",
       "      <td>87055.86 [4770;497783]</td>\n",
       "      <td>1260.43 [5;2328]</td>\n",
       "      <td>30833848.86 [30657287;30988173]</td>\n",
       "      <td>1165.43 [497;1825]</td>\n",
       "      <td>102.83 [82;107]</td>\n",
       "      <td>105261.86 [52869;138706]</td>\n",
       "      <td>104.9 [92;108]</td>\n",
       "      <td>811320.71 [298093;952618]</td>\n",
       "      <td>4.43 [2;8]</td>\n",
       "      <td>7.86 [3;14]</td>\n",
       "      <td>90370.86 [139;124246]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>80340494.0 [31047581;170402348]</td>\n",
       "      <td>1261858.33 [25377;3508016]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>14860681.0 [0;28174748]</td>\n",
       "      <td>3275.0 [0;6250]</td>\n",
       "      <td>39.88 [0;92]</td>\n",
       "      <td>7918.67 [0;20700]</td>\n",
       "      <td>56.66 [8;96]</td>\n",
       "      <td>66942.33 [994;161812]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>6029.67 [49;17395]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>31253502.89 [30440762;32811080]</td>\n",
       "      <td>2480.89 [943;5402]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>30393672.89 [30273320;30564398]</td>\n",
       "      <td>744.0 [657;814]</td>\n",
       "      <td>93.59 [82;98]</td>\n",
       "      <td>108898.33 [90988;122545]</td>\n",
       "      <td>94.75 [84;100]</td>\n",
       "      <td>723087.33 [632215;768585]</td>\n",
       "      <td>1.67 [1;3]</td>\n",
       "      <td>3.0 [1;6]</td>\n",
       "      <td>104806.0 [90346;121957]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>33525481.22 [30390357;39779451]</td>\n",
       "      <td>7944.11 [1821;20192]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>30612932.33 [30129709;31562690]</td>\n",
       "      <td>1728.89 [1254;2654]</td>\n",
       "      <td>92.47 [81;98]</td>\n",
       "      <td>52473.33 [29134;65108]</td>\n",
       "      <td>94.54 [84;100]</td>\n",
       "      <td>452420.67 [303329;585610]</td>\n",
       "      <td>11.44 [4;18]</td>\n",
       "      <td>21.22 [7;36]</td>\n",
       "      <td>48807.33 [20993;63725]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>34103641.33 [30545419;41166407]</td>\n",
       "      <td>7469.0 [814;20614]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>30953572.33 [30418624;32021766]</td>\n",
       "      <td>830.0 [518;1447]</td>\n",
       "      <td>94.89 [84;100]</td>\n",
       "      <td>167083.67 [149841;179870]</td>\n",
       "      <td>96.22 [86;101]</td>\n",
       "      <td>868917.67 [632274;1205839]</td>\n",
       "      <td>11.33 [9;15]</td>\n",
       "      <td>23.56 [18;32]</td>\n",
       "      <td>146076.33 [88308;178381]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>24645723.78 [13174826;30381792]</td>\n",
       "      <td>7040.22 [881;19361]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>20709046.78 [1597681;30264953]</td>\n",
       "      <td>912.67 [712;1309]</td>\n",
       "      <td>66.63 [3;98]</td>\n",
       "      <td>70222.0 [1175;106637]</td>\n",
       "      <td>74.44 [25;99]</td>\n",
       "      <td>313232.67 [2964;468375]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>69239.33 [658;106383]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>36633480.0 [30453778;48964620]</td>\n",
       "      <td>21015.0 [640;61644]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>30917818.67 [30383186;31967753]</td>\n",
       "      <td>804.33 [359;1689]</td>\n",
       "      <td>94.17 [83;100]</td>\n",
       "      <td>180893.67 [95101;238611]</td>\n",
       "      <td>95.36 [85;100]</td>\n",
       "      <td>809315.0 [473277;1036942]</td>\n",
       "      <td>1.67 [1;2]</td>\n",
       "      <td>4.0 [3;6]</td>\n",
       "      <td>163304.0 [42332;238611]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>34159104.0 [30641333;41187149]</td>\n",
       "      <td>33767.89 [16570;67998]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>28233933.22 [28205937;28279162]</td>\n",
       "      <td>3713.33 [3574;3980]</td>\n",
       "      <td>86.52 [74;92]</td>\n",
       "      <td>19109.33 [15912;20782]</td>\n",
       "      <td>91.29 [80;97]</td>\n",
       "      <td>147189.67 [117945;161812]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>14297.67 [7106;17967]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>30263322.33 [30110758;30430077]</td>\n",
       "      <td>1647.67 [1269;2322]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>30037221.0 [29835112;30211740]</td>\n",
       "      <td>1154.33 [846;1726]</td>\n",
       "      <td>91.9 [80;97]</td>\n",
       "      <td>65058.67 [43168;77870]</td>\n",
       "      <td>92.76 [81;98]</td>\n",
       "      <td>569520.0 [446163;632013]</td>\n",
       "      <td>6.33 [0;10]</td>\n",
       "      <td>12.33 [0;19]</td>\n",
       "      <td>64292.67 [42115;76924]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>35881180.33 [30442718;46535940]</td>\n",
       "      <td>14529.0 [1140;39819]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>31043946.33 [30276227;32500826]</td>\n",
       "      <td>968.33 [412;2049]</td>\n",
       "      <td>93.29 [82;99]</td>\n",
       "      <td>170923.67 [99299;208991]</td>\n",
       "      <td>95.02 [85;100]</td>\n",
       "      <td>815228.0 [745370;850176]</td>\n",
       "      <td>4.0 [0;9]</td>\n",
       "      <td>8.0 [0;18]</td>\n",
       "      <td>154762.67 [53292;206515]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>30386095.0 [30263458;30592960]</td>\n",
       "      <td>829.0 [656;936]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>30238046.0 [30093081;30432114]</td>\n",
       "      <td>400.67 [334;443]</td>\n",
       "      <td>92.78 [81;98]</td>\n",
       "      <td>215142.33 [191424;245050]</td>\n",
       "      <td>93.83 [82;99]</td>\n",
       "      <td>1011197.0 [847534;1303875]</td>\n",
       "      <td>0.67 [0;2]</td>\n",
       "      <td>1.33 [0;4]</td>\n",
       "      <td>213645.67 [191424;245050]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>4490.56 [105;8788]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>38071907.22 [30526291;54970900]</td>\n",
       "      <td>196309.67 [10924;595080]</td>\n",
       "      <td>4222.0 [0;8283]</td>\n",
       "      <td>18991538.0 [13883;28485306]</td>\n",
       "      <td>2016.44 [13;3035]</td>\n",
       "      <td>61.81 [0;93]</td>\n",
       "      <td>23011.11 [1039;34664]</td>\n",
       "      <td>65.23 [2;97]</td>\n",
       "      <td>565632.89 [1157;847809]</td>\n",
       "      <td>4.67 [0;8]</td>\n",
       "      <td>7.33 [0;11]</td>\n",
       "      <td>19906.22 [88;30453]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm                  Ns  \\\n",
       "Assembler                                                               \n",
       "ABySS              Multiple k-mer De Bruijn graph  2062.43 [147;3884]   \n",
       "BCALM2               Single k-mer De Bruijn graph           0.0 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "MINIA                Single k-mer De Bruijn graph           0.0 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph  4490.56 [105;8788]   \n",
       "\n",
       "                          Type                        basepairs  \\\n",
       "Assembler                                                         \n",
       "ABySS              Metagenomic  41418778.14 [31502565;91720949]   \n",
       "BCALM2                 Genomic  80340494.0 [31047581;170402348]   \n",
       "GATBMiniaPipeline  Metagenomic  31253502.89 [30440762;32811080]   \n",
       "IDBA-UD            Metagenomic  33525481.22 [30390357;39779451]   \n",
       "MEGAHIT            Metagenomic  34103641.33 [30545419;41166407]   \n",
       "MetaHipMer2        Metagenomic  24645723.78 [13174826;30381792]   \n",
       "metaSPAdes         Metagenomic   36633480.0 [30453778;48964620]   \n",
       "MINIA                  Genomic   34159104.0 [30641333;41187149]   \n",
       "SKESA                  Genomic  30263322.33 [30110758;30430077]   \n",
       "SPAdes                 Genomic  35881180.33 [30442718;46535940]   \n",
       "Unicycler              Genomic   30386095.0 [30263458;30592960]   \n",
       "VelvetOptimiser    Metagenomic  38071907.22 [30526291;54970900]   \n",
       "\n",
       "                                      contigs       filtered_Ns  \\\n",
       "Assembler                                                         \n",
       "ABySS                  87055.86 [4770;497783]  1260.43 [5;2328]   \n",
       "BCALM2             1261858.33 [25377;3508016]         0.0 [0;0]   \n",
       "GATBMiniaPipeline          2480.89 [943;5402]         0.0 [0;0]   \n",
       "IDBA-UD                  7944.11 [1821;20192]         0.0 [0;0]   \n",
       "MEGAHIT                    7469.0 [814;20614]         0.0 [0;0]   \n",
       "MetaHipMer2               7040.22 [881;19361]         0.0 [0;0]   \n",
       "metaSPAdes                21015.0 [640;61644]         0.0 [0;0]   \n",
       "MINIA                  33767.89 [16570;67998]         0.0 [0;0]   \n",
       "SKESA                     1647.67 [1269;2322]         0.0 [0;0]   \n",
       "SPAdes                   14529.0 [1140;39819]         0.0 [0;0]   \n",
       "Unicycler                     829.0 [656;936]         0.0 [0;0]   \n",
       "VelvetOptimiser      196309.67 [10924;595080]   4222.0 [0;8283]   \n",
       "\n",
       "                                filtered_basepairs     filtered_contigs  \\\n",
       "Assembler                                                                 \n",
       "ABySS              30833848.86 [30657287;30988173]   1165.43 [497;1825]   \n",
       "BCALM2                     14860681.0 [0;28174748]      3275.0 [0;6250]   \n",
       "GATBMiniaPipeline  30393672.89 [30273320;30564398]      744.0 [657;814]   \n",
       "IDBA-UD            30612932.33 [30129709;31562690]  1728.89 [1254;2654]   \n",
       "MEGAHIT            30953572.33 [30418624;32021766]     830.0 [518;1447]   \n",
       "MetaHipMer2         20709046.78 [1597681;30264953]    912.67 [712;1309]   \n",
       "metaSPAdes         30917818.67 [30383186;31967753]    804.33 [359;1689]   \n",
       "MINIA              28233933.22 [28205937;28279162]  3713.33 [3574;3980]   \n",
       "SKESA               30037221.0 [29835112;30211740]   1154.33 [846;1726]   \n",
       "SPAdes             31043946.33 [30276227;32500826]    968.33 [412;2049]   \n",
       "Unicycler           30238046.0 [30093081;30432114]     400.67 [334;443]   \n",
       "VelvetOptimiser        18991538.0 [13883;28485306]    2016.44 [13;3035]   \n",
       "\n",
       "                  filtered_mapped_reads               filtered_n50  \\\n",
       "Assembler                                                            \n",
       "ABySS                   102.83 [82;107]   105261.86 [52869;138706]   \n",
       "BCALM2                     39.88 [0;92]          7918.67 [0;20700]   \n",
       "GATBMiniaPipeline         93.59 [82;98]   108898.33 [90988;122545]   \n",
       "IDBA-UD                   92.47 [81;98]     52473.33 [29134;65108]   \n",
       "MEGAHIT                  94.89 [84;100]  167083.67 [149841;179870]   \n",
       "MetaHipMer2                66.63 [3;98]      70222.0 [1175;106637]   \n",
       "metaSPAdes               94.17 [83;100]   180893.67 [95101;238611]   \n",
       "MINIA                     86.52 [74;92]     19109.33 [15912;20782]   \n",
       "SKESA                      91.9 [80;97]     65058.67 [43168;77870]   \n",
       "SPAdes                    93.29 [82;99]   170923.67 [99299;208991]   \n",
       "Unicycler                 92.78 [81;98]  215142.33 [191424;245050]   \n",
       "VelvetOptimiser            61.81 [0;93]      23011.11 [1039;34664]   \n",
       "\n",
       "                     mapped_reads                  max_contig  \\\n",
       "Assembler                                                       \n",
       "ABySS              104.9 [92;108]   811320.71 [298093;952618]   \n",
       "BCALM2               56.66 [8;96]       66942.33 [994;161812]   \n",
       "GATBMiniaPipeline  94.75 [84;100]   723087.33 [632215;768585]   \n",
       "IDBA-UD            94.54 [84;100]   452420.67 [303329;585610]   \n",
       "MEGAHIT            96.22 [86;101]  868917.67 [632274;1205839]   \n",
       "MetaHipMer2         74.44 [25;99]     313232.67 [2964;468375]   \n",
       "metaSPAdes         95.36 [85;100]   809315.0 [473277;1036942]   \n",
       "MINIA               91.29 [80;97]   147189.67 [117945;161812]   \n",
       "SKESA               92.76 [81;98]    569520.0 [446163;632013]   \n",
       "SPAdes             95.02 [85;100]    815228.0 [745370;850176]   \n",
       "Unicycler           93.83 [82;99]  1011197.0 [847534;1303875]   \n",
       "VelvetOptimiser      65.23 [2;97]     565632.89 [1157;847809]   \n",
       "\n",
       "                  misassembled contigs misassembly events  \\\n",
       "Assembler                                                   \n",
       "ABySS                       4.43 [2;8]        7.86 [3;14]   \n",
       "BCALM2                       0.0 [0;0]          0.0 [0;0]   \n",
       "GATBMiniaPipeline           1.67 [1;3]          3.0 [1;6]   \n",
       "IDBA-UD                   11.44 [4;18]       21.22 [7;36]   \n",
       "MEGAHIT                   11.33 [9;15]      23.56 [18;32]   \n",
       "MetaHipMer2                  0.0 [0;0]          0.0 [0;0]   \n",
       "metaSPAdes                  1.67 [1;2]          4.0 [3;6]   \n",
       "MINIA                        0.0 [0;0]          0.0 [0;0]   \n",
       "SKESA                      6.33 [0;10]       12.33 [0;19]   \n",
       "SPAdes                       4.0 [0;9]         8.0 [0;18]   \n",
       "Unicycler                   0.67 [0;2]         1.33 [0;4]   \n",
       "VelvetOptimiser             4.67 [0;8]        7.33 [0;11]   \n",
       "\n",
       "                                         n50  \n",
       "Assembler                                     \n",
       "ABySS                  90370.86 [139;124246]  \n",
       "BCALM2                    6029.67 [49;17395]  \n",
       "GATBMiniaPipeline    104806.0 [90346;121957]  \n",
       "IDBA-UD               48807.33 [20993;63725]  \n",
       "MEGAHIT             146076.33 [88308;178381]  \n",
       "MetaHipMer2            69239.33 [658;106383]  \n",
       "metaSPAdes           163304.0 [42332;238611]  \n",
       "MINIA                  14297.67 [7106;17967]  \n",
       "SKESA                 64292.67 [42115;76924]  \n",
       "SPAdes              154762.67 [53292;206515]  \n",
       "Unicycler          213645.67 [191424;245050]  \n",
       "VelvetOptimiser          19906.22 [88;30453]  "
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "stats_per_assembler"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Global stats per distribution"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Even\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>contigs</th>\n",
       "      <th>filtered_Ns</th>\n",
       "      <th>filtered_basepairs</th>\n",
       "      <th>filtered_contigs</th>\n",
       "      <th>filtered_mapped_reads</th>\n",
       "      <th>filtered_n50</th>\n",
       "      <th>mapped_reads</th>\n",
       "      <th>max_contig</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>n50</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>2062.43 [147;3884]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>41418778.14 [31502565;91720949]</td>\n",
       "      <td>87055.86 [4770;497783]</td>\n",
       "      <td>1260.43 [5;2328]</td>\n",
       "      <td>30833848.86 [30657287;30988173]</td>\n",
       "      <td>1165.43 [497;1825]</td>\n",
       "      <td>102.83 [82;107]</td>\n",
       "      <td>105261.86 [52869;138706]</td>\n",
       "      <td>104.9 [92;108]</td>\n",
       "      <td>811320.71 [298093;952618]</td>\n",
       "      <td>4.43 [2;8]</td>\n",
       "      <td>7.86 [3;14]</td>\n",
       "      <td>90370.86 [139;124246]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>80340494.0 [31047581;170402348]</td>\n",
       "      <td>1261858.33 [25377;3508016]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>14860681.0 [0;28174748]</td>\n",
       "      <td>3275.0 [0;6250]</td>\n",
       "      <td>39.88 [0;92]</td>\n",
       "      <td>7918.67 [0;20700]</td>\n",
       "      <td>56.66 [8;96]</td>\n",
       "      <td>66942.33 [994;161812]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>6029.67 [49;17395]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>31253502.89 [30440762;32811080]</td>\n",
       "      <td>2480.89 [943;5402]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>30393672.89 [30273320;30564398]</td>\n",
       "      <td>744.0 [657;814]</td>\n",
       "      <td>93.59 [82;98]</td>\n",
       "      <td>108898.33 [90988;122545]</td>\n",
       "      <td>94.75 [84;100]</td>\n",
       "      <td>723087.33 [632215;768585]</td>\n",
       "      <td>1.67 [1;3]</td>\n",
       "      <td>3.0 [1;6]</td>\n",
       "      <td>104806.0 [90346;121957]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>33525481.22 [30390357;39779451]</td>\n",
       "      <td>7944.11 [1821;20192]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>30612932.33 [30129709;31562690]</td>\n",
       "      <td>1728.89 [1254;2654]</td>\n",
       "      <td>92.47 [81;98]</td>\n",
       "      <td>52473.33 [29134;65108]</td>\n",
       "      <td>94.54 [84;100]</td>\n",
       "      <td>452420.67 [303329;585610]</td>\n",
       "      <td>11.44 [4;18]</td>\n",
       "      <td>21.22 [7;36]</td>\n",
       "      <td>48807.33 [20993;63725]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>34103641.33 [30545419;41166407]</td>\n",
       "      <td>7469.0 [814;20614]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>30953572.33 [30418624;32021766]</td>\n",
       "      <td>830.0 [518;1447]</td>\n",
       "      <td>94.89 [84;100]</td>\n",
       "      <td>167083.67 [149841;179870]</td>\n",
       "      <td>96.22 [86;101]</td>\n",
       "      <td>868917.67 [632274;1205839]</td>\n",
       "      <td>11.33 [9;15]</td>\n",
       "      <td>23.56 [18;32]</td>\n",
       "      <td>146076.33 [88308;178381]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>24645723.78 [13174826;30381792]</td>\n",
       "      <td>7040.22 [881;19361]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>20709046.78 [1597681;30264953]</td>\n",
       "      <td>912.67 [712;1309]</td>\n",
       "      <td>66.63 [3;98]</td>\n",
       "      <td>70222.0 [1175;106637]</td>\n",
       "      <td>74.44 [25;99]</td>\n",
       "      <td>313232.67 [2964;468375]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>69239.33 [658;106383]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>36633480.0 [30453778;48964620]</td>\n",
       "      <td>21015.0 [640;61644]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>30917818.67 [30383186;31967753]</td>\n",
       "      <td>804.33 [359;1689]</td>\n",
       "      <td>94.17 [83;100]</td>\n",
       "      <td>180893.67 [95101;238611]</td>\n",
       "      <td>95.36 [85;100]</td>\n",
       "      <td>809315.0 [473277;1036942]</td>\n",
       "      <td>1.67 [1;2]</td>\n",
       "      <td>4.0 [3;6]</td>\n",
       "      <td>163304.0 [42332;238611]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>34159104.0 [30641333;41187149]</td>\n",
       "      <td>33767.89 [16570;67998]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>28233933.22 [28205937;28279162]</td>\n",
       "      <td>3713.33 [3574;3980]</td>\n",
       "      <td>86.52 [74;92]</td>\n",
       "      <td>19109.33 [15912;20782]</td>\n",
       "      <td>91.29 [80;97]</td>\n",
       "      <td>147189.67 [117945;161812]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>14297.67 [7106;17967]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>30263322.33 [30110758;30430077]</td>\n",
       "      <td>1647.67 [1269;2322]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>30037221.0 [29835112;30211740]</td>\n",
       "      <td>1154.33 [846;1726]</td>\n",
       "      <td>91.9 [80;97]</td>\n",
       "      <td>65058.67 [43168;77870]</td>\n",
       "      <td>92.76 [81;98]</td>\n",
       "      <td>569520.0 [446163;632013]</td>\n",
       "      <td>6.33 [0;10]</td>\n",
       "      <td>12.33 [0;19]</td>\n",
       "      <td>64292.67 [42115;76924]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>35881180.33 [30442718;46535940]</td>\n",
       "      <td>14529.0 [1140;39819]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>31043946.33 [30276227;32500826]</td>\n",
       "      <td>968.33 [412;2049]</td>\n",
       "      <td>93.29 [82;99]</td>\n",
       "      <td>170923.67 [99299;208991]</td>\n",
       "      <td>95.02 [85;100]</td>\n",
       "      <td>815228.0 [745370;850176]</td>\n",
       "      <td>4.0 [0;9]</td>\n",
       "      <td>8.0 [0;18]</td>\n",
       "      <td>154762.67 [53292;206515]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>30386095.0 [30263458;30592960]</td>\n",
       "      <td>829.0 [656;936]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>30238046.0 [30093081;30432114]</td>\n",
       "      <td>400.67 [334;443]</td>\n",
       "      <td>92.78 [81;98]</td>\n",
       "      <td>215142.33 [191424;245050]</td>\n",
       "      <td>93.83 [82;99]</td>\n",
       "      <td>1011197.0 [847534;1303875]</td>\n",
       "      <td>0.67 [0;2]</td>\n",
       "      <td>1.33 [0;4]</td>\n",
       "      <td>213645.67 [191424;245050]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>4490.56 [105;8788]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>38071907.22 [30526291;54970900]</td>\n",
       "      <td>196309.67 [10924;595080]</td>\n",
       "      <td>4222.0 [0;8283]</td>\n",
       "      <td>18991538.0 [13883;28485306]</td>\n",
       "      <td>2016.44 [13;3035]</td>\n",
       "      <td>61.81 [0;93]</td>\n",
       "      <td>23011.11 [1039;34664]</td>\n",
       "      <td>65.23 [2;97]</td>\n",
       "      <td>565632.89 [1157;847809]</td>\n",
       "      <td>4.67 [0;8]</td>\n",
       "      <td>7.33 [0;11]</td>\n",
       "      <td>19906.22 [88;30453]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm                  Ns  \\\n",
       "Assembler                                                               \n",
       "ABySS              Multiple k-mer De Bruijn graph  2062.43 [147;3884]   \n",
       "BCALM2               Single k-mer De Bruijn graph           0.0 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "MINIA                Single k-mer De Bruijn graph           0.0 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph  4490.56 [105;8788]   \n",
       "\n",
       "                          Type                        basepairs  \\\n",
       "Assembler                                                         \n",
       "ABySS              Metagenomic  41418778.14 [31502565;91720949]   \n",
       "BCALM2                 Genomic  80340494.0 [31047581;170402348]   \n",
       "GATBMiniaPipeline  Metagenomic  31253502.89 [30440762;32811080]   \n",
       "IDBA-UD            Metagenomic  33525481.22 [30390357;39779451]   \n",
       "MEGAHIT            Metagenomic  34103641.33 [30545419;41166407]   \n",
       "MetaHipMer2        Metagenomic  24645723.78 [13174826;30381792]   \n",
       "metaSPAdes         Metagenomic   36633480.0 [30453778;48964620]   \n",
       "MINIA                  Genomic   34159104.0 [30641333;41187149]   \n",
       "SKESA                  Genomic  30263322.33 [30110758;30430077]   \n",
       "SPAdes                 Genomic  35881180.33 [30442718;46535940]   \n",
       "Unicycler              Genomic   30386095.0 [30263458;30592960]   \n",
       "VelvetOptimiser    Metagenomic  38071907.22 [30526291;54970900]   \n",
       "\n",
       "                                      contigs       filtered_Ns  \\\n",
       "Assembler                                                         \n",
       "ABySS                  87055.86 [4770;497783]  1260.43 [5;2328]   \n",
       "BCALM2             1261858.33 [25377;3508016]         0.0 [0;0]   \n",
       "GATBMiniaPipeline          2480.89 [943;5402]         0.0 [0;0]   \n",
       "IDBA-UD                  7944.11 [1821;20192]         0.0 [0;0]   \n",
       "MEGAHIT                    7469.0 [814;20614]         0.0 [0;0]   \n",
       "MetaHipMer2               7040.22 [881;19361]         0.0 [0;0]   \n",
       "metaSPAdes                21015.0 [640;61644]         0.0 [0;0]   \n",
       "MINIA                  33767.89 [16570;67998]         0.0 [0;0]   \n",
       "SKESA                     1647.67 [1269;2322]         0.0 [0;0]   \n",
       "SPAdes                   14529.0 [1140;39819]         0.0 [0;0]   \n",
       "Unicycler                     829.0 [656;936]         0.0 [0;0]   \n",
       "VelvetOptimiser      196309.67 [10924;595080]   4222.0 [0;8283]   \n",
       "\n",
       "                                filtered_basepairs     filtered_contigs  \\\n",
       "Assembler                                                                 \n",
       "ABySS              30833848.86 [30657287;30988173]   1165.43 [497;1825]   \n",
       "BCALM2                     14860681.0 [0;28174748]      3275.0 [0;6250]   \n",
       "GATBMiniaPipeline  30393672.89 [30273320;30564398]      744.0 [657;814]   \n",
       "IDBA-UD            30612932.33 [30129709;31562690]  1728.89 [1254;2654]   \n",
       "MEGAHIT            30953572.33 [30418624;32021766]     830.0 [518;1447]   \n",
       "MetaHipMer2         20709046.78 [1597681;30264953]    912.67 [712;1309]   \n",
       "metaSPAdes         30917818.67 [30383186;31967753]    804.33 [359;1689]   \n",
       "MINIA              28233933.22 [28205937;28279162]  3713.33 [3574;3980]   \n",
       "SKESA               30037221.0 [29835112;30211740]   1154.33 [846;1726]   \n",
       "SPAdes             31043946.33 [30276227;32500826]    968.33 [412;2049]   \n",
       "Unicycler           30238046.0 [30093081;30432114]     400.67 [334;443]   \n",
       "VelvetOptimiser        18991538.0 [13883;28485306]    2016.44 [13;3035]   \n",
       "\n",
       "                  filtered_mapped_reads               filtered_n50  \\\n",
       "Assembler                                                            \n",
       "ABySS                   102.83 [82;107]   105261.86 [52869;138706]   \n",
       "BCALM2                     39.88 [0;92]          7918.67 [0;20700]   \n",
       "GATBMiniaPipeline         93.59 [82;98]   108898.33 [90988;122545]   \n",
       "IDBA-UD                   92.47 [81;98]     52473.33 [29134;65108]   \n",
       "MEGAHIT                  94.89 [84;100]  167083.67 [149841;179870]   \n",
       "MetaHipMer2                66.63 [3;98]      70222.0 [1175;106637]   \n",
       "metaSPAdes               94.17 [83;100]   180893.67 [95101;238611]   \n",
       "MINIA                     86.52 [74;92]     19109.33 [15912;20782]   \n",
       "SKESA                      91.9 [80;97]     65058.67 [43168;77870]   \n",
       "SPAdes                    93.29 [82;99]   170923.67 [99299;208991]   \n",
       "Unicycler                 92.78 [81;98]  215142.33 [191424;245050]   \n",
       "VelvetOptimiser            61.81 [0;93]      23011.11 [1039;34664]   \n",
       "\n",
       "                     mapped_reads                  max_contig  \\\n",
       "Assembler                                                       \n",
       "ABySS              104.9 [92;108]   811320.71 [298093;952618]   \n",
       "BCALM2               56.66 [8;96]       66942.33 [994;161812]   \n",
       "GATBMiniaPipeline  94.75 [84;100]   723087.33 [632215;768585]   \n",
       "IDBA-UD            94.54 [84;100]   452420.67 [303329;585610]   \n",
       "MEGAHIT            96.22 [86;101]  868917.67 [632274;1205839]   \n",
       "MetaHipMer2         74.44 [25;99]     313232.67 [2964;468375]   \n",
       "metaSPAdes         95.36 [85;100]   809315.0 [473277;1036942]   \n",
       "MINIA               91.29 [80;97]   147189.67 [117945;161812]   \n",
       "SKESA               92.76 [81;98]    569520.0 [446163;632013]   \n",
       "SPAdes             95.02 [85;100]    815228.0 [745370;850176]   \n",
       "Unicycler           93.83 [82;99]  1011197.0 [847534;1303875]   \n",
       "VelvetOptimiser      65.23 [2;97]     565632.89 [1157;847809]   \n",
       "\n",
       "                  misassembled contigs misassembly events  \\\n",
       "Assembler                                                   \n",
       "ABySS                       4.43 [2;8]        7.86 [3;14]   \n",
       "BCALM2                       0.0 [0;0]          0.0 [0;0]   \n",
       "GATBMiniaPipeline           1.67 [1;3]          3.0 [1;6]   \n",
       "IDBA-UD                   11.44 [4;18]       21.22 [7;36]   \n",
       "MEGAHIT                   11.33 [9;15]      23.56 [18;32]   \n",
       "MetaHipMer2                  0.0 [0;0]          0.0 [0;0]   \n",
       "metaSPAdes                  1.67 [1;2]          4.0 [3;6]   \n",
       "MINIA                        0.0 [0;0]          0.0 [0;0]   \n",
       "SKESA                      6.33 [0;10]       12.33 [0;19]   \n",
       "SPAdes                       4.0 [0;9]         8.0 [0;18]   \n",
       "Unicycler                   0.67 [0;2]         1.33 [0;4]   \n",
       "VelvetOptimiser             4.67 [0;8]        7.33 [0;11]   \n",
       "\n",
       "                                         n50  \n",
       "Assembler                                     \n",
       "ABySS                  90370.86 [139;124246]  \n",
       "BCALM2                    6029.67 [49;17395]  \n",
       "GATBMiniaPipeline    104806.0 [90346;121957]  \n",
       "IDBA-UD               48807.33 [20993;63725]  \n",
       "MEGAHIT             146076.33 [88308;178381]  \n",
       "MetaHipMer2            69239.33 [658;106383]  \n",
       "metaSPAdes           163304.0 [42332;238611]  \n",
       "MINIA                  14297.67 [7106;17967]  \n",
       "SKESA                 64292.67 [42115;76924]  \n",
       "SPAdes              154762.67 [53292;206515]  \n",
       "Unicycler          213645.67 [191424;245050]  \n",
       "VelvetOptimiser          19906.22 [88;30453]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "for distribution in global_pipeline_metrics_df.distribution.unique():\n",
    "    stats_per_distribution = pd.DataFrame()\n",
    "    for assembler in global_pipeline_metrics_df.assembler.unique():\n",
    "        row={'Assembler': assembler, \n",
    "             'Type':global_pipeline_metrics_df[(global_pipeline_metrics_df.assembler == assembler) & (global_pipeline_metrics_df.distribution == distribution)].type.unique()[0],\n",
    "             'Algorithm': global_pipeline_metrics_df[global_pipeline_metrics_df.assembler == assembler].algorythm.unique()[0]}\n",
    "        lala=global_pipeline_metrics_df[(global_pipeline_metrics_df.assembler == assembler) & (global_pipeline_metrics_df.distribution == distribution)].describe()\n",
    "        for column in lala.columns:\n",
    "            mean = lala.loc['mean',column]\n",
    "            minimum = lala.loc['min',column]\n",
    "            maximum = lala.loc['max',column]\n",
    "            row[column] = \"{} [{};{}]\".format(round(mean, 2),int(minimum), int(maximum))\n",
    "        stats_per_distribution=stats_per_distribution.append(row, ignore_index=True)\n",
    "    stats_per_distribution = stats_per_distribution.set_index('Assembler')\n",
    "    stats_per_distribution.to_csv(\"Tables/Results/{} - Global metrics per assembler.csv\".format(distribution))\n",
    "    print(distribution)\n",
    "    display(stats_per_distribution)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Reference-based metrics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pipeline_report_tables.json data from run1...\n",
      "Processing pipeline_report_tables.json data from run2...\n",
      "Processing pipeline_report_tables.json data from run3...\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>assembler</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>...</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>reference</th>\n",
       "      <th>run</th>\n",
       "      <th>sample</th>\n",
       "      <th>validity</th>\n",
       "      <th>distribution</th>\n",
       "      <th>type</th>\n",
       "      <th>algorythm</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>39.0</td>\n",
       "      <td>0.073679</td>\n",
       "      <td>98440.0</td>\n",
       "      <td>98440.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ABySS</td>\n",
       "      <td>3977886.0</td>\n",
       "      <td>0.983244</td>\n",
       "      <td>66.0</td>\n",
       "      <td>0.999639</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.001060</td>\n",
       "      <td>1.001159</td>\n",
       "      <td>Bacillus_subtilis</td>\n",
       "      <td>run1</td>\n",
       "      <td>ERR2984773</td>\n",
       "      <td>0.999901</td>\n",
       "      <td>Even</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>BCALM2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Bacillus_subtilis</td>\n",
       "      <td>run1</td>\n",
       "      <td>ERR2984773</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Even</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>15.0</td>\n",
       "      <td>0.189942</td>\n",
       "      <td>263420.0</td>\n",
       "      <td>257358.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GATBMiniaPipeline</td>\n",
       "      <td>3982010.0</td>\n",
       "      <td>0.984263</td>\n",
       "      <td>30.0</td>\n",
       "      <td>0.999809</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.000079</td>\n",
       "      <td>1.000083</td>\n",
       "      <td>Bacillus_subtilis</td>\n",
       "      <td>run1</td>\n",
       "      <td>ERR2984773</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>Even</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>120.0</td>\n",
       "      <td>0.031121</td>\n",
       "      <td>34330.0</td>\n",
       "      <td>34330.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IDBA-UD</td>\n",
       "      <td>3976919.0</td>\n",
       "      <td>0.983005</td>\n",
       "      <td>184.0</td>\n",
       "      <td>0.999756</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.003298</td>\n",
       "      <td>1.003311</td>\n",
       "      <td>Bacillus_subtilis</td>\n",
       "      <td>run1</td>\n",
       "      <td>ERR2984773</td>\n",
       "      <td>0.999986</td>\n",
       "      <td>Even</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>14.0</td>\n",
       "      <td>0.103094</td>\n",
       "      <td>294549.0</td>\n",
       "      <td>294549.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>MEGAHIT</td>\n",
       "      <td>3989518.0</td>\n",
       "      <td>0.986119</td>\n",
       "      <td>31.0</td>\n",
       "      <td>0.978636</td>\n",
       "      <td>...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.998972</td>\n",
       "      <td>1.011678</td>\n",
       "      <td>Bacillus_subtilis</td>\n",
       "      <td>run1</td>\n",
       "      <td>ERR2984773</td>\n",
       "      <td>0.987440</td>\n",
       "      <td>Even</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1267</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>MINIA</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Staphylococcus_aureus_plasmid3</td>\n",
       "      <td>run3</td>\n",
       "      <td>ERR2984773</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Even</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1268</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.929502</td>\n",
       "      <td>2782.0</td>\n",
       "      <td>2782.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>SKESA</td>\n",
       "      <td>2782.0</td>\n",
       "      <td>0.929502</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>Staphylococcus_aureus_plasmid3</td>\n",
       "      <td>run3</td>\n",
       "      <td>ERR2984773</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>Even</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1269</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>SPAdes</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Staphylococcus_aureus_plasmid3</td>\n",
       "      <td>run3</td>\n",
       "      <td>ERR2984773</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Even</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1270</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>2993.0</td>\n",
       "      <td>2993.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Unicycler</td>\n",
       "      <td>2993.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>Staphylococcus_aureus_plasmid3</td>\n",
       "      <td>run3</td>\n",
       "      <td>ERR2984773</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>Even</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1271</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>VelvetOptimiser</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Staphylococcus_aureus_plasmid3</td>\n",
       "      <td>run3</td>\n",
       "      <td>ERR2984773</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Even</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1272 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        L90       LSA      NA50      NG50   Ns          assembler  basepairs  \\\n",
       "0      39.0  0.073679   98440.0   98440.0  0.0              ABySS  3977886.0   \n",
       "1       0.0  0.000000       0.0       0.0  0.0             BCALM2        0.0   \n",
       "2      15.0  0.189942  263420.0  257358.0  0.0  GATBMiniaPipeline  3982010.0   \n",
       "3     120.0  0.031121   34330.0   34330.0  0.0            IDBA-UD  3976919.0   \n",
       "4      14.0  0.103094  294549.0  294549.0  0.0            MEGAHIT  3989518.0   \n",
       "...     ...       ...       ...       ...  ...                ...        ...   \n",
       "1267    0.0  0.000000       0.0       0.0  0.0              MINIA        0.0   \n",
       "1268    0.0  0.929502    2782.0    2782.0  0.0              SKESA     2782.0   \n",
       "1269    0.0  0.000000       0.0       0.0  0.0             SPAdes        0.0   \n",
       "1270    0.0  1.000000    2993.0    2993.0  0.0          Unicycler     2993.0   \n",
       "1271    0.0  0.000000       0.0       0.0  0.0    VelvetOptimiser        0.0   \n",
       "\n",
       "      breadth_of_coverage  contigs  identity  ...  misassembly events  \\\n",
       "0                0.983244     66.0  0.999639  ...                 0.0   \n",
       "1                0.000000      0.0  0.000000  ...                 0.0   \n",
       "2                0.984263     30.0  0.999809  ...                 0.0   \n",
       "3                0.983005    184.0  0.999756  ...                 0.0   \n",
       "4                0.986119     31.0  0.978636  ...                 4.0   \n",
       "...                   ...      ...       ...  ...                 ...   \n",
       "1267             0.000000      0.0  0.000000  ...                 0.0   \n",
       "1268             0.929502      1.0  1.000000  ...                 0.0   \n",
       "1269             0.000000      0.0  0.000000  ...                 0.0   \n",
       "1270             1.000000      1.0  1.000000  ...                 0.0   \n",
       "1271             0.000000      0.0  0.000000  ...                 0.0   \n",
       "\n",
       "      multiplicity  parsimony                       reference   run  \\\n",
       "0         1.001060   1.001159               Bacillus_subtilis  run1   \n",
       "1         0.000000   0.000000               Bacillus_subtilis  run1   \n",
       "2         1.000079   1.000083               Bacillus_subtilis  run1   \n",
       "3         1.003298   1.003311               Bacillus_subtilis  run1   \n",
       "4         0.998972   1.011678               Bacillus_subtilis  run1   \n",
       "...            ...        ...                             ...   ...   \n",
       "1267      0.000000   0.000000  Staphylococcus_aureus_plasmid3  run3   \n",
       "1268      1.000000   1.000000  Staphylococcus_aureus_plasmid3  run3   \n",
       "1269      0.000000   0.000000  Staphylococcus_aureus_plasmid3  run3   \n",
       "1270      1.000000   1.000000  Staphylococcus_aureus_plasmid3  run3   \n",
       "1271      0.000000   0.000000  Staphylococcus_aureus_plasmid3  run3   \n",
       "\n",
       "          sample  validity distribution         type  \\\n",
       "0     ERR2984773  0.999901         Even  Metagenomic   \n",
       "1     ERR2984773  0.000000         Even      Genomic   \n",
       "2     ERR2984773  0.999995         Even  Metagenomic   \n",
       "3     ERR2984773  0.999986         Even  Metagenomic   \n",
       "4     ERR2984773  0.987440         Even  Metagenomic   \n",
       "...          ...       ...          ...          ...   \n",
       "1267  ERR2984773  0.000000         Even      Genomic   \n",
       "1268  ERR2984773  1.000000         Even      Genomic   \n",
       "1269  ERR2984773  0.000000         Even      Genomic   \n",
       "1270  ERR2984773  1.000000         Even      Genomic   \n",
       "1271  ERR2984773  0.000000         Even  Metagenomic   \n",
       "\n",
       "                           algorythm  \n",
       "0     Multiple k-mer De Bruijn graph  \n",
       "1       Single k-mer De Bruijn graph  \n",
       "2     Multiple k-mer De Bruijn graph  \n",
       "3     Multiple k-mer De Bruijn graph  \n",
       "4     Multiple k-mer De Bruijn graph  \n",
       "...                              ...  \n",
       "1267    Single k-mer De Bruijn graph  \n",
       "1268  Multiple k-mer De Bruijn graph  \n",
       "1269  Multiple k-mer De Bruijn graph  \n",
       "1270  Multiple k-mer De Bruijn graph  \n",
       "1271  Multiple k-mer De Bruijn graph  \n",
       "\n",
       "[1272 rows x 22 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "report_glob = glob.glob('../Results/*/*/report/pipeline_report_tables.json')\n",
    "reference_pipeline_metrics_df = pd.DataFrame()\n",
    "\n",
    "for pipeline_report_file in report_glob:\n",
    "    report_file_name = pipeline_report_file.split('/')[-1]\n",
    "    stats_run = pipeline_report_file.split('/')[-3]\n",
    "    print('Processing {0} data from {1}...'.format(report_file_name, stats_run))\n",
    "    \n",
    "    with open(pipeline_report_file) as _fh:\n",
    "        json_report = json.load(_fh)\n",
    "        for sample in json_report.keys():\n",
    "            for reference, data in json_report[sample]['ReferenceTables'].items():\n",
    "                for row in data:\n",
    "                    for item in row:\n",
    "                        reference_pipeline_metrics_df = reference_pipeline_metrics_df.append({'run': stats_run,\n",
    "                                                                                        'sample': sample,\n",
    "                                                                                        'assembler': item['assembler'],\n",
    "                                                                                        'reference': reference,\n",
    "                                                                                        'LSA': item['contiguity'],\n",
    "                                                                                        'breadth_of_coverage': item['breadth_of_coverage'],\n",
    "                                                                                        'multiplicity': item['multiplicity'],\n",
    "                                                                                        'validity': item['validity'],\n",
    "                                                                                        'parsimony': item['parsimony'],\n",
    "                                                                                        'identity': item['identity'],\n",
    "                                                                                        'lowest_identity': item['lowest_identity'],\n",
    "                                                                                        'L90': item['L90'],\n",
    "                                                                                        'contigs': item['aligned_contigs'],\n",
    "                                                                                        'NA50': item['NA50'],\n",
    "                                                                                        'NG50': item['NG50'],\n",
    "                                                                                        'basepairs': item['aligned_basepairs'],\n",
    "                                                                                        'Ns': item['Ns'],\n",
    "                                                                                        'misassembled contigs': item['misassembled_contigs'],\n",
    "                                                                                        'misassembly events': item['misassembly_events']},\n",
    "                                                                                       ignore_index=True)\n",
    "\n",
    "reference_pipeline_metrics_df['distribution'] = np.where(reference_pipeline_metrics_df['sample'].isin(log_distributed), 'Log', 'Even')\n",
    "reference_pipeline_metrics_df['type'] = np.where(reference_pipeline_metrics_df['assembler'].isin(genomic_assemblers), 'Genomic', 'Metagenomic')\n",
    "reference_pipeline_metrics_df['algorythm'] = np.where(reference_pipeline_metrics_df['assembler'].isin(single_kmer), 'Single k-mer De Bruijn graph', 'Multiple k-mer De Bruijn graph')\n",
    "reference_pipeline_metrics_df[['contigs','basepairs','L90','Ns','NA50','NG50','misassembled contigs', 'misassembly events', 'multiplicity','validity','parsimony','identity','lowest_identity']] = reference_pipeline_metrics_df[['contigs','basepairs','L90','Ns','NA50','NG50','misassembled contigs', 'misassembly events','multiplicity','validity','parsimony','identity','lowest_identity']].apply(pd.to_numeric)\n",
    "display(reference_pipeline_metrics_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "L90                     float64\n",
       "LSA                     float64\n",
       "NA50                    float64\n",
       "NG50                    float64\n",
       "Ns                      float64\n",
       "assembler                object\n",
       "basepairs               float64\n",
       "breadth_of_coverage     float64\n",
       "contigs                 float64\n",
       "identity                float64\n",
       "lowest_identity         float64\n",
       "misassembled contigs    float64\n",
       "misassembly events      float64\n",
       "multiplicity            float64\n",
       "parsimony               float64\n",
       "reference                object\n",
       "run                      object\n",
       "sample                   object\n",
       "validity                float64\n",
       "distribution             object\n",
       "type                     object\n",
       "algorythm                object\n",
       "dtype: object"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reference_pipeline_metrics_df.dtypes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Save data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [],
   "source": [
    "reference_pipeline_metrics_df.to_csv(\"Tables/Results/Reference metrics.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reference statistics per assembler type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_stats_per_assembler_type = pd.DataFrame()\n",
    "for assembly_type in reference_pipeline_metrics_df.type.unique():\n",
    "    row={'Assembler Type': assembly_type}\n",
    "    lala=reference_pipeline_metrics_df[reference_pipeline_metrics_df.type == assembly_type].describe()\n",
    "    for column in lala.columns:\n",
    "        mean = lala.loc['mean',column]\n",
    "        minimum = lala.loc['min',column]\n",
    "        maximum = lala.loc['max',column]\n",
    "        row[column] = \"{} [{};{}]\".format(round(mean, 2),int(minimum), int(maximum))\n",
    "    ref_stats_per_assembler_type=ref_stats_per_assembler_type.append(row, ignore_index=True)\n",
    "ref_stats_per_assembler_type = ref_stats_per_assembler_type.set_index('Assembler Type')\n",
    "ref_stats_per_assembler_type.to_csv(\"Tables/Results/Reference metrics per assembler type.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler Type</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Metagenomic</th>\n",
       "      <td>35.26 [0;1004]</td>\n",
       "      <td>0.28 [0;1]</td>\n",
       "      <td>89746.41 [0;542872]</td>\n",
       "      <td>89131.67 [0;542872]</td>\n",
       "      <td>91.4 [0;12098]</td>\n",
       "      <td>2267909.42 [0;6780808]</td>\n",
       "      <td>0.86 [0;1]</td>\n",
       "      <td>85.92 [0;1404]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "      <td>0.89 [0;1]</td>\n",
       "      <td>0.45 [0;9]</td>\n",
       "      <td>0.83 [0;15]</td>\n",
       "      <td>0.93 [0;1]</td>\n",
       "      <td>0.93 [0;1]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic</th>\n",
       "      <td>45.86 [0;921]</td>\n",
       "      <td>0.21 [0;1]</td>\n",
       "      <td>76036.23 [0;475179]</td>\n",
       "      <td>74709.01 [0;475179]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>2223458.03 [0;6728888]</td>\n",
       "      <td>0.77 [0;1]</td>\n",
       "      <td>150.17 [0;1495]</td>\n",
       "      <td>0.84 [0;1]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>0.22 [0;7]</td>\n",
       "      <td>0.39 [0;15]</td>\n",
       "      <td>0.84 [0;1]</td>\n",
       "      <td>0.84 [0;1]</td>\n",
       "      <td>0.84 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           L90         LSA                 NA50  \\\n",
       "Assembler Type                                                    \n",
       "Metagenomic     35.26 [0;1004]  0.28 [0;1]  89746.41 [0;542872]   \n",
       "Genomic          45.86 [0;921]  0.21 [0;1]  76036.23 [0;475179]   \n",
       "\n",
       "                               NG50              Ns               basepairs  \\\n",
       "Assembler Type                                                                \n",
       "Metagenomic     89131.67 [0;542872]  91.4 [0;12098]  2267909.42 [0;6780808]   \n",
       "Genomic         74709.01 [0;475179]       0.0 [0;0]  2223458.03 [0;6728888]   \n",
       "\n",
       "               breadth_of_coverage          contigs    identity  \\\n",
       "Assembler Type                                                    \n",
       "Metagenomic             0.86 [0;1]   85.92 [0;1404]  0.92 [0;1]   \n",
       "Genomic                 0.77 [0;1]  150.17 [0;1495]  0.84 [0;1]   \n",
       "\n",
       "               lowest_identity misassembled contigs misassembly events  \\\n",
       "Assembler Type                                                           \n",
       "Metagenomic         0.89 [0;1]           0.45 [0;9]        0.83 [0;15]   \n",
       "Genomic              0.8 [0;1]           0.22 [0;7]        0.39 [0;15]   \n",
       "\n",
       "               multiplicity   parsimony    validity  \n",
       "Assembler Type                                       \n",
       "Metagenomic      0.93 [0;1]  0.93 [0;1]  0.92 [0;1]  \n",
       "Genomic          0.84 [0;1]  0.84 [0;1]  0.84 [0;1]  "
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ref_stats_per_assembler_type"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reference stats per assembly algorithm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_stats_per_assembler_algo = pd.DataFrame()\n",
    "for assembly_algo in reference_pipeline_metrics_df.algorythm.unique():\n",
    "    row={'Assembler Algorithm': assembly_algo}\n",
    "    lala=reference_pipeline_metrics_df[reference_pipeline_metrics_df.algorythm == assembly_algo].describe()\n",
    "    for column in lala.columns:\n",
    "        mean = lala.loc['mean',column]\n",
    "        minimum = lala.loc['min',column]\n",
    "        maximum = lala.loc['max',column]\n",
    "        row[column] = \"{} [{};{}]\".format(round(mean, 2),int(minimum), int(maximum))\n",
    "    ref_stats_per_assembler_algo=ref_stats_per_assembler_algo.append(row, ignore_index=True)\n",
    "ref_stats_per_assembler_algo = ref_stats_per_assembler_algo.set_index('Assembler Algorithm')\n",
    "ref_stats_per_assembler_algo.to_csv(\"Tables/Results/Reference metrics per assembler algorithm.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler Algorithm</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Multiple k-mer De Bruijn graph</th>\n",
       "      <td>34.13 [0;1004]</td>\n",
       "      <td>0.29 [0;1]</td>\n",
       "      <td>98617.13 [0;542872]</td>\n",
       "      <td>97618.16 [0;542872]</td>\n",
       "      <td>63.36 [0;12098]</td>\n",
       "      <td>2342800.58 [0;6780808]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "      <td>76.95 [0;1404]</td>\n",
       "      <td>0.93 [0;1]</td>\n",
       "      <td>0.89 [0;1]</td>\n",
       "      <td>0.42 [0;9]</td>\n",
       "      <td>0.78 [0;15]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.93 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Single k-mer De Bruijn graph</th>\n",
       "      <td>67.3 [0;921]</td>\n",
       "      <td>0.06 [0;0]</td>\n",
       "      <td>12103.01 [0;65138]</td>\n",
       "      <td>11585.5 [0;64960]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1790646.37 [0;6518607]</td>\n",
       "      <td>0.56 [0;0]</td>\n",
       "      <td>290.36 [0;1495]</td>\n",
       "      <td>0.65 [0;1]</td>\n",
       "      <td>0.65 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.65 [0;1]</td>\n",
       "      <td>0.65 [0;1]</td>\n",
       "      <td>0.65 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           L90         LSA  \\\n",
       "Assembler Algorithm                                          \n",
       "Multiple k-mer De Bruijn graph  34.13 [0;1004]  0.29 [0;1]   \n",
       "Single k-mer De Bruijn graph      67.3 [0;921]  0.06 [0;0]   \n",
       "\n",
       "                                               NA50                 NG50  \\\n",
       "Assembler Algorithm                                                        \n",
       "Multiple k-mer De Bruijn graph  98617.13 [0;542872]  97618.16 [0;542872]   \n",
       "Single k-mer De Bruijn graph     12103.01 [0;65138]    11585.5 [0;64960]   \n",
       "\n",
       "                                             Ns               basepairs  \\\n",
       "Assembler Algorithm                                                       \n",
       "Multiple k-mer De Bruijn graph  63.36 [0;12098]  2342800.58 [0;6780808]   \n",
       "Single k-mer De Bruijn graph          0.0 [0;0]  1790646.37 [0;6518607]   \n",
       "\n",
       "                               breadth_of_coverage          contigs  \\\n",
       "Assembler Algorithm                                                   \n",
       "Multiple k-mer De Bruijn graph          0.87 [0;1]   76.95 [0;1404]   \n",
       "Single k-mer De Bruijn graph            0.56 [0;0]  290.36 [0;1495]   \n",
       "\n",
       "                                  identity lowest_identity  \\\n",
       "Assembler Algorithm                                          \n",
       "Multiple k-mer De Bruijn graph  0.93 [0;1]      0.89 [0;1]   \n",
       "Single k-mer De Bruijn graph    0.65 [0;1]      0.65 [0;1]   \n",
       "\n",
       "                               misassembled contigs misassembly events  \\\n",
       "Assembler Algorithm                                                      \n",
       "Multiple k-mer De Bruijn graph           0.42 [0;9]        0.78 [0;15]   \n",
       "Single k-mer De Bruijn graph              0.0 [0;0]          0.0 [0;0]   \n",
       "\n",
       "                               multiplicity   parsimony    validity  \n",
       "Assembler Algorithm                                                  \n",
       "Multiple k-mer De Bruijn graph   0.94 [0;1]  0.94 [0;1]  0.93 [0;1]  \n",
       "Single k-mer De Bruijn graph     0.65 [0;1]  0.65 [0;1]  0.65 [0;1]  "
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ref_stats_per_assembler_algo"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reference stats per assembler"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_stats_per_assembler = pd.DataFrame()\n",
    "for assembler in reference_pipeline_metrics_df.assembler.unique():\n",
    "    row={'Assembler': assembler, \n",
    "         'Type':reference_pipeline_metrics_df[reference_pipeline_metrics_df.assembler == assembler].type.unique()[0],\n",
    "         'Algorithm': reference_pipeline_metrics_df[reference_pipeline_metrics_df.assembler == assembler].algorythm.unique()[0]}\n",
    "    lala=reference_pipeline_metrics_df[reference_pipeline_metrics_df.assembler == assembler].describe()\n",
    "    for column in lala.columns:\n",
    "        mean = lala.loc['mean',column]\n",
    "        minimum = lala.loc['min',column]\n",
    "        maximum = lala.loc['max',column]\n",
    "        row[column] = \"{} [{};{}]\".format(round(mean, 2),int(minimum), int(maximum))\n",
    "    ref_stats_per_assembler=ref_stats_per_assembler.append(row, ignore_index=True)\n",
    "ref_stats_per_assembler = ref_stats_per_assembler.set_index('Assembler')\n",
    "ref_stats_per_assembler.to_csv(\"Tables/Results/Reference metrics per assembler.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>60.88 [0;1004]</td>\n",
       "      <td>0.32 [0;0]</td>\n",
       "      <td>110190.23 [0;351502]</td>\n",
       "      <td>110089.38 [0;351502]</td>\n",
       "      <td>105.21 [0;2170]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2536099.3 [0;6780808]</td>\n",
       "      <td>0.92 [0;0]</td>\n",
       "      <td>97.05 [0;1404]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.37 [0;8]</td>\n",
       "      <td>0.65 [0;14]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>51.2 [0;921]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>6932.33 [0;65138]</td>\n",
       "      <td>6539.94 [0;64934]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1236061.67 [0;6510588]</td>\n",
       "      <td>0.37 [0;0]</td>\n",
       "      <td>272.97 [0;1495]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>31.33 [0;112]</td>\n",
       "      <td>0.35 [0;1]</td>\n",
       "      <td>106093.36 [2176;452364]</td>\n",
       "      <td>96782.33 [2176;351186]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2523831.42 [2176;6716518]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "      <td>59.92 [1;199]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "      <td>0.14 [0;3]</td>\n",
       "      <td>0.25 [0;6]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>68.82 [0;283]</td>\n",
       "      <td>0.27 [0;1]</td>\n",
       "      <td>55837.36 [0;254629]</td>\n",
       "      <td>54555.43 [0;254629]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2507606.06 [0;6710104]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>121.11 [0;449]</td>\n",
       "      <td>0.97 [0;1]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "      <td>1.12 [0;9]</td>\n",
       "      <td>1.94 [0;15]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "      <td>0.97 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>18.31 [0;55]</td>\n",
       "      <td>0.37 [0;1]</td>\n",
       "      <td>138989.33 [2215;542872]</td>\n",
       "      <td>141793.83 [2215;542872]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2534425.75 [2215;6737986]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>44.39 [1;141]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.93 [0;1]</td>\n",
       "      <td>0.97 [0;4]</td>\n",
       "      <td>1.99 [0;7]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.01 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>21.69 [0;110]</td>\n",
       "      <td>0.23 [0;0]</td>\n",
       "      <td>58464.84 [0;257424]</td>\n",
       "      <td>57747.71 [0;257424]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1722416.51 [0;6717767]</td>\n",
       "      <td>0.68 [0;0]</td>\n",
       "      <td>74.19 [0;361]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>20.14 [0;90]</td>\n",
       "      <td>0.34 [0;1]</td>\n",
       "      <td>133559.97 [1677;475510]</td>\n",
       "      <td>132927.86 [1677;475510]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2530047.11 [2076;6726884]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "      <td>39.33 [1;167]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>0.14 [0;2]</td>\n",
       "      <td>0.33 [0;6]</td>\n",
       "      <td>1.01 [0;1]</td>\n",
       "      <td>1.01 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>83.4 [0;406]</td>\n",
       "      <td>0.08 [0;0]</td>\n",
       "      <td>17273.69 [0;65138]</td>\n",
       "      <td>16631.06 [0;64960]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2345231.06 [0;6518607]</td>\n",
       "      <td>0.74 [0;0]</td>\n",
       "      <td>307.75 [0;1146]</td>\n",
       "      <td>0.81 [0;1]</td>\n",
       "      <td>0.79 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.81 [0;1]</td>\n",
       "      <td>0.81 [0;1]</td>\n",
       "      <td>0.81 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>57.92 [0;670]</td>\n",
       "      <td>0.27 [0;0]</td>\n",
       "      <td>76053.75 [1404;351014]</td>\n",
       "      <td>75134.92 [0;351014]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2496709.31 [1460;6673601]</td>\n",
       "      <td>0.9 [0;0]</td>\n",
       "      <td>96.06 [1;933]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "      <td>0.67 [0;7]</td>\n",
       "      <td>1.17 [0;15]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>21.06 [0;79]</td>\n",
       "      <td>0.29 [0;1]</td>\n",
       "      <td>117650.97 [0;315090]</td>\n",
       "      <td>117061.69 [0;315090]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2524122.0 [0;6728888]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "      <td>41.19 [0;143]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "      <td>0.36 [0;4]</td>\n",
       "      <td>0.69 [0;8]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.96 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>15.75 [0;57]</td>\n",
       "      <td>0.38 [0;1]</td>\n",
       "      <td>162270.42 [0;475179]</td>\n",
       "      <td>158177.44 [0;475179]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2515166.11 [0;6723626]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "      <td>32.86 [0;111]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.06 [0;1]</td>\n",
       "      <td>0.11 [0;2]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>31.33 [0;207]</td>\n",
       "      <td>0.09 [0;0]</td>\n",
       "      <td>29632.83 [0;338923]</td>\n",
       "      <td>34682.42 [0;464977]</td>\n",
       "      <td>537.64 [0;12098]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1580537.53 [0;6708956]</td>\n",
       "      <td>0.52 [0;0]</td>\n",
       "      <td>167.92 [0;1098]</td>\n",
       "      <td>0.56 [0;1]</td>\n",
       "      <td>0.55 [0;1]</td>\n",
       "      <td>0.39 [0;8]</td>\n",
       "      <td>0.61 [0;11]</td>\n",
       "      <td>0.56 [0;1]</td>\n",
       "      <td>0.56 [0;1]</td>\n",
       "      <td>0.56 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm             L90         LSA  \\\n",
       "Assembler                                                                       \n",
       "ABySS              Multiple k-mer De Bruijn graph  60.88 [0;1004]  0.32 [0;0]   \n",
       "BCALM2               Single k-mer De Bruijn graph    51.2 [0;921]  0.04 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph   31.33 [0;112]  0.35 [0;1]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph   68.82 [0;283]  0.27 [0;1]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph    18.31 [0;55]  0.37 [0;1]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph   21.69 [0;110]  0.23 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph    20.14 [0;90]  0.34 [0;1]   \n",
       "MINIA                Single k-mer De Bruijn graph    83.4 [0;406]  0.08 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph   57.92 [0;670]  0.27 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph    21.06 [0;79]  0.29 [0;1]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph    15.75 [0;57]  0.38 [0;1]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph   31.33 [0;207]  0.09 [0;0]   \n",
       "\n",
       "                                      NA50                     NG50  \\\n",
       "Assembler                                                             \n",
       "ABySS                 110190.23 [0;351502]     110089.38 [0;351502]   \n",
       "BCALM2                   6932.33 [0;65138]        6539.94 [0;64934]   \n",
       "GATBMiniaPipeline  106093.36 [2176;452364]   96782.33 [2176;351186]   \n",
       "IDBA-UD                55837.36 [0;254629]      54555.43 [0;254629]   \n",
       "MEGAHIT            138989.33 [2215;542872]  141793.83 [2215;542872]   \n",
       "MetaHipMer2            58464.84 [0;257424]      57747.71 [0;257424]   \n",
       "metaSPAdes         133559.97 [1677;475510]  132927.86 [1677;475510]   \n",
       "MINIA                   17273.69 [0;65138]       16631.06 [0;64960]   \n",
       "SKESA               76053.75 [1404;351014]      75134.92 [0;351014]   \n",
       "SPAdes                117650.97 [0;315090]     117061.69 [0;315090]   \n",
       "Unicycler             162270.42 [0;475179]     158177.44 [0;475179]   \n",
       "VelvetOptimiser        29632.83 [0;338923]      34682.42 [0;464977]   \n",
       "\n",
       "                                 Ns         Type                  basepairs  \\\n",
       "Assembler                                                                     \n",
       "ABySS               105.21 [0;2170]  Metagenomic      2536099.3 [0;6780808]   \n",
       "BCALM2                    0.0 [0;0]      Genomic     1236061.67 [0;6510588]   \n",
       "GATBMiniaPipeline         0.0 [0;0]  Metagenomic  2523831.42 [2176;6716518]   \n",
       "IDBA-UD                   0.0 [0;0]  Metagenomic     2507606.06 [0;6710104]   \n",
       "MEGAHIT                   0.0 [0;0]  Metagenomic  2534425.75 [2215;6737986]   \n",
       "MetaHipMer2               0.0 [0;0]  Metagenomic     1722416.51 [0;6717767]   \n",
       "metaSPAdes                0.0 [0;0]  Metagenomic  2530047.11 [2076;6726884]   \n",
       "MINIA                     0.0 [0;0]      Genomic     2345231.06 [0;6518607]   \n",
       "SKESA                     0.0 [0;0]      Genomic  2496709.31 [1460;6673601]   \n",
       "SPAdes                    0.0 [0;0]      Genomic      2524122.0 [0;6728888]   \n",
       "Unicycler                 0.0 [0;0]      Genomic     2515166.11 [0;6723626]   \n",
       "VelvetOptimiser    537.64 [0;12098]  Metagenomic     1580537.53 [0;6708956]   \n",
       "\n",
       "                  breadth_of_coverage          contigs    identity  \\\n",
       "Assembler                                                            \n",
       "ABySS                      0.92 [0;0]   97.05 [0;1404]  0.99 [0;1]   \n",
       "BCALM2                     0.37 [0;0]  272.97 [0;1495]   0.5 [0;1]   \n",
       "GATBMiniaPipeline          0.98 [0;1]    59.92 [1;199]   1.0 [0;1]   \n",
       "IDBA-UD                    0.94 [0;1]   121.11 [0;449]  0.97 [0;1]   \n",
       "MEGAHIT                    0.99 [0;1]    44.39 [1;141]   1.0 [0;1]   \n",
       "MetaHipMer2                0.68 [0;0]    74.19 [0;361]  0.94 [0;1]   \n",
       "metaSPAdes                 0.98 [0;1]    39.33 [1;167]   1.0 [0;1]   \n",
       "MINIA                      0.74 [0;0]  307.75 [0;1146]  0.81 [0;1]   \n",
       "SKESA                       0.9 [0;0]    96.06 [1;933]   1.0 [0;1]   \n",
       "SPAdes                      0.9 [0;1]    41.19 [0;143]  0.94 [0;1]   \n",
       "Unicycler                  0.92 [0;1]    32.86 [0;111]  0.94 [0;1]   \n",
       "VelvetOptimiser            0.52 [0;0]  167.92 [0;1098]  0.56 [0;1]   \n",
       "\n",
       "                  lowest_identity misassembled contigs misassembly events  \\\n",
       "Assembler                                                                   \n",
       "ABySS                  0.95 [0;1]           0.37 [0;8]        0.65 [0;14]   \n",
       "BCALM2                  0.5 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "GATBMiniaPipeline      0.98 [0;1]           0.14 [0;3]         0.25 [0;6]   \n",
       "IDBA-UD                0.87 [0;1]           1.12 [0;9]        1.94 [0;15]   \n",
       "MEGAHIT                0.93 [0;1]           0.97 [0;4]         1.99 [0;7]   \n",
       "MetaHipMer2            0.94 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "metaSPAdes             0.99 [0;1]           0.14 [0;2]         0.33 [0;6]   \n",
       "MINIA                  0.79 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "SKESA                  0.87 [0;1]           0.67 [0;7]        1.17 [0;15]   \n",
       "SPAdes                  0.9 [0;1]           0.36 [0;4]         0.69 [0;8]   \n",
       "Unicycler              0.94 [0;1]           0.06 [0;1]         0.11 [0;2]   \n",
       "VelvetOptimiser        0.55 [0;1]           0.39 [0;8]        0.61 [0;11]   \n",
       "\n",
       "                  multiplicity   parsimony    validity  \n",
       "Assembler                                               \n",
       "ABySS                1.0 [0;1]   1.0 [0;1]  0.99 [0;1]  \n",
       "BCALM2               0.5 [0;1]   0.5 [0;1]   0.5 [0;1]  \n",
       "GATBMiniaPipeline    1.0 [0;1]   1.0 [0;1]   1.0 [0;1]  \n",
       "IDBA-UD             0.98 [0;1]  0.98 [0;1]  0.97 [0;1]  \n",
       "MEGAHIT              1.0 [0;1]  1.01 [0;1]   1.0 [0;1]  \n",
       "MetaHipMer2         0.94 [0;1]  0.94 [0;1]  0.94 [0;1]  \n",
       "metaSPAdes          1.01 [0;1]  1.01 [0;1]   1.0 [0;1]  \n",
       "MINIA               0.81 [0;1]  0.81 [0;1]  0.81 [0;1]  \n",
       "SKESA                1.0 [0;1]   1.0 [0;1]   1.0 [0;1]  \n",
       "SPAdes              0.95 [0;1]  0.96 [0;1]  0.94 [0;1]  \n",
       "Unicycler           0.94 [0;1]  0.94 [0;1]  0.94 [0;1]  \n",
       "VelvetOptimiser     0.56 [0;1]  0.56 [0;1]  0.56 [0;1]  "
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ref_stats_per_assembler"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reference stats per reference replicon"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Per assembler for each reference replicon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Bacillus_subtilis', 'Enterococcus_faecalis',\n",
       "       'Escherichia_coli_plasmid', 'Escherichia_coli',\n",
       "       'Lactobacillus_fermentum', 'Listeria_monocytogenes',\n",
       "       'Pseudomonas_aeruginosa', 'Salmonella_enterica',\n",
       "       'Staphylococcus_aureus', 'Staphylococcus_aureus_plasmid1',\n",
       "       'Staphylococcus_aureus_plasmid2', 'Staphylococcus_aureus_plasmid3'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reference_pipeline_metrics_df.reference.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Bacillus_subtilis\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>18.0 [13;39]</td>\n",
       "      <td>0.1 [0;0]</td>\n",
       "      <td>250023.14 [98440;295233]</td>\n",
       "      <td>250023.14 [98440;295233]</td>\n",
       "      <td>20.14 [0;47]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3988493.71 [3977886;3991152]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>30.0 [22;66]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>22.0 [0;66]</td>\n",
       "      <td>0.01 [0;0]</td>\n",
       "      <td>22400.0 [0;65138]</td>\n",
       "      <td>22174.33 [0;64934]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2265479.33 [0;3951974]</td>\n",
       "      <td>0.56 [0;0]</td>\n",
       "      <td>527.67 [0;1467]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>14.0 [13;15]</td>\n",
       "      <td>0.16 [0;0]</td>\n",
       "      <td>265520.0 [263420;269695]</td>\n",
       "      <td>261389.67 [257358;263445]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3981896.67 [3979577;3984103]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>28.33 [27;30]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>52.33 [18;120]</td>\n",
       "      <td>0.07 [0;0]</td>\n",
       "      <td>166650.67 [34330;252653]</td>\n",
       "      <td>153422.67 [34330;212969]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3977018.67 [3976292;3977845]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>85.33 [34;184]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.94 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>13.67 [13;14]</td>\n",
       "      <td>0.13 [0;0]</td>\n",
       "      <td>274090.0 [263586;294549]</td>\n",
       "      <td>274090.0 [263586;294549]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3988399.89 [3986471;3989518]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>28.33 [27;31]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.79 [0;0]</td>\n",
       "      <td>1.0 [0;2]</td>\n",
       "      <td>2.33 [0;4]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>10.33 [0;16]</td>\n",
       "      <td>0.07 [0;0]</td>\n",
       "      <td>171945.22 [1145;257424]</td>\n",
       "      <td>171562.67 [0;257424]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2721019.89 [205828;3976387]</td>\n",
       "      <td>0.67 [0;0]</td>\n",
       "      <td>77.11 [27;180]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>14.67 [9;24]</td>\n",
       "      <td>0.18 [0;0]</td>\n",
       "      <td>272233.67 [164894;343843]</td>\n",
       "      <td>272233.67 [164894;343843]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3981566.67 [3975594;3987932]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>30.67 [18;51]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>78.0 [66;102]</td>\n",
       "      <td>0.03 [0;0]</td>\n",
       "      <td>58810.33 [46155;65138]</td>\n",
       "      <td>57658.67 [43056;64960]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>3951436.44 [3949080;3953200]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>134.11 [116;170]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>17.0 [14;20]</td>\n",
       "      <td>0.1 [0;0]</td>\n",
       "      <td>242427.67 [200881;269493]</td>\n",
       "      <td>240318.0 [200881;263164]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>3969836.33 [3967464;3971432]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>30.0 [26;34]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>12.67 [11;16]</td>\n",
       "      <td>0.18 [0;0]</td>\n",
       "      <td>297790.67 [263236;315090]</td>\n",
       "      <td>297790.67 [263236;315090]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>3980233.67 [3976703;3982668]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>25.67 [22;33]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>11.0 [10;12]</td>\n",
       "      <td>0.19 [0;0]</td>\n",
       "      <td>311365.0 [294271;324887]</td>\n",
       "      <td>311365.0 [294271;324887]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>3974877.67 [3973744;3975910]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>20.0 [18;22]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>41.0 [0;63]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>46308.0 [0;69914]</td>\n",
       "      <td>44716.0 [0;69010]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2635788.33 [0;3953983]</td>\n",
       "      <td>0.65 [0;0]</td>\n",
       "      <td>74.0 [0;113]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm             L90         LSA  \\\n",
       "Assembler                                                                       \n",
       "ABySS              Multiple k-mer De Bruijn graph    18.0 [13;39]   0.1 [0;0]   \n",
       "BCALM2               Single k-mer De Bruijn graph     22.0 [0;66]  0.01 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph    14.0 [13;15]  0.16 [0;0]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  52.33 [18;120]  0.07 [0;0]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph   13.67 [13;14]  0.13 [0;0]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph    10.33 [0;16]  0.07 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph    14.67 [9;24]  0.18 [0;0]   \n",
       "MINIA                Single k-mer De Bruijn graph   78.0 [66;102]  0.03 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph    17.0 [14;20]   0.1 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph   12.67 [11;16]  0.18 [0;0]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph    11.0 [10;12]  0.19 [0;0]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph     41.0 [0;63]  0.04 [0;0]   \n",
       "\n",
       "                                        NA50                       NG50  \\\n",
       "Assembler                                                                 \n",
       "ABySS               250023.14 [98440;295233]   250023.14 [98440;295233]   \n",
       "BCALM2                     22400.0 [0;65138]         22174.33 [0;64934]   \n",
       "GATBMiniaPipeline   265520.0 [263420;269695]  261389.67 [257358;263445]   \n",
       "IDBA-UD             166650.67 [34330;252653]   153422.67 [34330;212969]   \n",
       "MEGAHIT             274090.0 [263586;294549]   274090.0 [263586;294549]   \n",
       "MetaHipMer2          171945.22 [1145;257424]       171562.67 [0;257424]   \n",
       "metaSPAdes         272233.67 [164894;343843]  272233.67 [164894;343843]   \n",
       "MINIA                 58810.33 [46155;65138]     57658.67 [43056;64960]   \n",
       "SKESA              242427.67 [200881;269493]   240318.0 [200881;263164]   \n",
       "SPAdes             297790.67 [263236;315090]  297790.67 [263236;315090]   \n",
       "Unicycler           311365.0 [294271;324887]   311365.0 [294271;324887]   \n",
       "VelvetOptimiser            46308.0 [0;69914]          44716.0 [0;69010]   \n",
       "\n",
       "                             Ns         Type                     basepairs  \\\n",
       "Assembler                                                                    \n",
       "ABySS              20.14 [0;47]  Metagenomic  3988493.71 [3977886;3991152]   \n",
       "BCALM2                0.0 [0;0]      Genomic        2265479.33 [0;3951974]   \n",
       "GATBMiniaPipeline     0.0 [0;0]  Metagenomic  3981896.67 [3979577;3984103]   \n",
       "IDBA-UD               0.0 [0;0]  Metagenomic  3977018.67 [3976292;3977845]   \n",
       "MEGAHIT               0.0 [0;0]  Metagenomic  3988399.89 [3986471;3989518]   \n",
       "MetaHipMer2           0.0 [0;0]  Metagenomic   2721019.89 [205828;3976387]   \n",
       "metaSPAdes            0.0 [0;0]  Metagenomic  3981566.67 [3975594;3987932]   \n",
       "MINIA                 0.0 [0;0]      Genomic  3951436.44 [3949080;3953200]   \n",
       "SKESA                 0.0 [0;0]      Genomic  3969836.33 [3967464;3971432]   \n",
       "SPAdes                0.0 [0;0]      Genomic  3980233.67 [3976703;3982668]   \n",
       "Unicycler             0.0 [0;0]      Genomic  3974877.67 [3973744;3975910]   \n",
       "VelvetOptimiser       0.0 [0;0]  Metagenomic        2635788.33 [0;3953983]   \n",
       "\n",
       "                  breadth_of_coverage           contigs    identity  \\\n",
       "Assembler                                                             \n",
       "ABySS                      0.99 [0;0]      30.0 [22;66]   1.0 [0;0]   \n",
       "BCALM2                     0.56 [0;0]   527.67 [0;1467]  0.67 [0;1]   \n",
       "GATBMiniaPipeline          0.98 [0;0]     28.33 [27;30]   1.0 [0;1]   \n",
       "IDBA-UD                    0.98 [0;0]    85.33 [34;184]   1.0 [0;0]   \n",
       "MEGAHIT                    0.99 [0;0]     28.33 [27;31]  0.99 [0;0]   \n",
       "MetaHipMer2                0.67 [0;0]    77.11 [27;180]   1.0 [0;1]   \n",
       "metaSPAdes                 0.98 [0;0]     30.67 [18;51]   1.0 [0;0]   \n",
       "MINIA                      0.98 [0;0]  134.11 [116;170]   1.0 [0;1]   \n",
       "SKESA                      0.98 [0;0]      30.0 [26;34]   1.0 [0;1]   \n",
       "SPAdes                     0.98 [0;0]     25.67 [22;33]   1.0 [0;0]   \n",
       "Unicycler                  0.98 [0;0]      20.0 [18;22]   1.0 [0;1]   \n",
       "VelvetOptimiser            0.65 [0;0]      74.0 [0;113]  0.67 [0;1]   \n",
       "\n",
       "                  lowest_identity misassembled contigs misassembly events  \\\n",
       "Assembler                                                                   \n",
       "ABySS                   1.0 [0;0]            0.0 [0;0]          0.0 [0;0]   \n",
       "BCALM2                 0.67 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "GATBMiniaPipeline       1.0 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "IDBA-UD                0.94 [0;0]            0.0 [0;0]          0.0 [0;0]   \n",
       "MEGAHIT                0.79 [0;0]            1.0 [0;2]         2.33 [0;4]   \n",
       "MetaHipMer2            0.99 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "metaSPAdes             0.98 [0;0]            0.0 [0;0]          0.0 [0;0]   \n",
       "MINIA                   1.0 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "SKESA                   1.0 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "SPAdes                  1.0 [0;0]            0.0 [0;0]          0.0 [0;0]   \n",
       "Unicycler               1.0 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "VelvetOptimiser        0.67 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "\n",
       "                  multiplicity   parsimony    validity  \n",
       "Assembler                                               \n",
       "ABySS                1.0 [1;1]   1.0 [1;1]   1.0 [0;0]  \n",
       "BCALM2              0.67 [0;1]  0.67 [0;1]  0.67 [0;1]  \n",
       "GATBMiniaPipeline    1.0 [1;1]   1.0 [1;1]   1.0 [0;1]  \n",
       "IDBA-UD              1.0 [1;1]   1.0 [1;1]   1.0 [0;0]  \n",
       "MEGAHIT              1.0 [0;0]   1.0 [0;1]   1.0 [0;0]  \n",
       "MetaHipMer2          1.0 [1;1]   1.0 [1;1]   1.0 [0;1]  \n",
       "metaSPAdes           1.0 [0;1]   1.0 [0;1]   1.0 [0;1]  \n",
       "MINIA                1.0 [1;1]   1.0 [1;1]   1.0 [0;1]  \n",
       "SKESA                1.0 [1;1]   1.0 [1;1]   1.0 [1;1]  \n",
       "SPAdes               1.0 [0;1]   1.0 [0;1]   1.0 [0;1]  \n",
       "Unicycler            1.0 [0;1]   1.0 [0;1]   1.0 [0;1]  \n",
       "VelvetOptimiser     0.67 [0;1]  0.67 [0;1]  0.67 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Enterococcus_faecalis\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>17.29 [9;61]</td>\n",
       "      <td>0.17 [0;0]</td>\n",
       "      <td>212767.0 [45769;241090]</td>\n",
       "      <td>212767.0 [45769;241090]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2823330.57 [2813974;2828460]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>32.71 [19;106]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>34.78 [0;107]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>10553.67 [0;30453]</td>\n",
       "      <td>10140.0 [0;30452]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>993772.0 [0;2768590]</td>\n",
       "      <td>0.35 [0;0]</td>\n",
       "      <td>115.22 [0;180]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>18.33 [13;24]</td>\n",
       "      <td>0.17 [0;0]</td>\n",
       "      <td>180947.33 [127741;235254]</td>\n",
       "      <td>180947.33 [127741;235254]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2798634.44 [2779793;2820351]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>37.44 [28;46]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>0.67 [0;2]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>65.67 [22;147]</td>\n",
       "      <td>0.1 [0;0]</td>\n",
       "      <td>81575.67 [18985;113206]</td>\n",
       "      <td>83696.0 [18985;118897]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2799260.67 [2777299;2811602]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>113.0 [53;231]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.93 [0;0]</td>\n",
       "      <td>1.0 [0;3]</td>\n",
       "      <td>2.0 [0;6]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>14.33 [13;17]</td>\n",
       "      <td>0.23 [0;0]</td>\n",
       "      <td>235337.67 [235316;235381]</td>\n",
       "      <td>236414.33 [235316;238546]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2802086.33 [2785847;2822833]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>32.44 [31;34]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.96 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>1.33 [0;2]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>17.89 [0;27]</td>\n",
       "      <td>0.09 [0;0]</td>\n",
       "      <td>79651.22 [1159;118895]</td>\n",
       "      <td>77900.67 [0;118895]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1936605.78 [207594;2811406]</td>\n",
       "      <td>0.68 [0;0]</td>\n",
       "      <td>88.67 [45;177]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>15.0 [9;25]</td>\n",
       "      <td>0.17 [0;0]</td>\n",
       "      <td>206754.33 [143041;238611]</td>\n",
       "      <td>206754.33 [143041;238611]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2806275.33 [2786440;2818677]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>31.67 [21;51]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>0.67 [0;2]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>119.11 [103;149]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>27073.11 [20287;30453]</td>\n",
       "      <td>26905.44 [19908;30453]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2749839.33 [2732359;2770364]</td>\n",
       "      <td>0.97 [0;0]</td>\n",
       "      <td>183.67 [166;219]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>30.33 [19;42]</td>\n",
       "      <td>0.12 [0;0]</td>\n",
       "      <td>105908.0 [76714;152709]</td>\n",
       "      <td>105785.67 [76714;152709]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2801849.33 [2797241;2810284]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>54.67 [40;72]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>0.72 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>16.33 [10;25]</td>\n",
       "      <td>0.2 [0;0]</td>\n",
       "      <td>193128.33 [104623;239100]</td>\n",
       "      <td>190834.33 [118250;239100]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2794852.33 [2781588;2812518]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>36.67 [21;63]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>1.0 [0;3]</td>\n",
       "      <td>2.0 [0;6]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>10.33 [8;14]</td>\n",
       "      <td>0.26 [0;0]</td>\n",
       "      <td>283526.33 [234974;377104]</td>\n",
       "      <td>267184.0 [185947;377104]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2803278.67 [2783327;2813747]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>23.33 [20;30]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>69.0 [0;106]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>20269.67 [0;30453]</td>\n",
       "      <td>19873.33 [0;30453]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1823316.67 [0;2736286]</td>\n",
       "      <td>0.64 [0;0]</td>\n",
       "      <td>106.67 [0;163]</td>\n",
       "      <td>0.67 [0;0]</td>\n",
       "      <td>0.67 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm               L90  \\\n",
       "Assembler                                                             \n",
       "ABySS              Multiple k-mer De Bruijn graph      17.29 [9;61]   \n",
       "BCALM2               Single k-mer De Bruijn graph     34.78 [0;107]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph     18.33 [13;24]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph    65.67 [22;147]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph     14.33 [13;17]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph      17.89 [0;27]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph       15.0 [9;25]   \n",
       "MINIA                Single k-mer De Bruijn graph  119.11 [103;149]   \n",
       "SKESA              Multiple k-mer De Bruijn graph     30.33 [19;42]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph     16.33 [10;25]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph      10.33 [8;14]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph      69.0 [0;106]   \n",
       "\n",
       "                          LSA                       NA50  \\\n",
       "Assembler                                                  \n",
       "ABySS              0.17 [0;0]    212767.0 [45769;241090]   \n",
       "BCALM2             0.02 [0;0]         10553.67 [0;30453]   \n",
       "GATBMiniaPipeline  0.17 [0;0]  180947.33 [127741;235254]   \n",
       "IDBA-UD             0.1 [0;0]    81575.67 [18985;113206]   \n",
       "MEGAHIT            0.23 [0;0]  235337.67 [235316;235381]   \n",
       "MetaHipMer2        0.09 [0;0]     79651.22 [1159;118895]   \n",
       "metaSPAdes         0.17 [0;0]  206754.33 [143041;238611]   \n",
       "MINIA              0.04 [0;0]     27073.11 [20287;30453]   \n",
       "SKESA              0.12 [0;0]    105908.0 [76714;152709]   \n",
       "SPAdes              0.2 [0;0]  193128.33 [104623;239100]   \n",
       "Unicycler          0.26 [0;0]  283526.33 [234974;377104]   \n",
       "VelvetOptimiser    0.02 [0;0]         20269.67 [0;30453]   \n",
       "\n",
       "                                        NG50         Ns         Type  \\\n",
       "Assembler                                                              \n",
       "ABySS                212767.0 [45769;241090]  0.0 [0;0]  Metagenomic   \n",
       "BCALM2                     10140.0 [0;30452]  0.0 [0;0]      Genomic   \n",
       "GATBMiniaPipeline  180947.33 [127741;235254]  0.0 [0;0]  Metagenomic   \n",
       "IDBA-UD               83696.0 [18985;118897]  0.0 [0;0]  Metagenomic   \n",
       "MEGAHIT            236414.33 [235316;238546]  0.0 [0;0]  Metagenomic   \n",
       "MetaHipMer2              77900.67 [0;118895]  0.0 [0;0]  Metagenomic   \n",
       "metaSPAdes         206754.33 [143041;238611]  0.0 [0;0]  Metagenomic   \n",
       "MINIA                 26905.44 [19908;30453]  0.0 [0;0]      Genomic   \n",
       "SKESA               105785.67 [76714;152709]  0.0 [0;0]      Genomic   \n",
       "SPAdes             190834.33 [118250;239100]  0.0 [0;0]      Genomic   \n",
       "Unicycler           267184.0 [185947;377104]  0.0 [0;0]      Genomic   \n",
       "VelvetOptimiser           19873.33 [0;30453]  0.0 [0;0]  Metagenomic   \n",
       "\n",
       "                                      basepairs breadth_of_coverage  \\\n",
       "Assembler                                                             \n",
       "ABySS              2823330.57 [2813974;2828460]          0.99 [0;0]   \n",
       "BCALM2                     993772.0 [0;2768590]          0.35 [0;0]   \n",
       "GATBMiniaPipeline  2798634.44 [2779793;2820351]          0.98 [0;0]   \n",
       "IDBA-UD            2799260.67 [2777299;2811602]          0.98 [0;0]   \n",
       "MEGAHIT            2802086.33 [2785847;2822833]          0.98 [0;0]   \n",
       "MetaHipMer2         1936605.78 [207594;2811406]          0.68 [0;0]   \n",
       "metaSPAdes         2806275.33 [2786440;2818677]          0.99 [0;0]   \n",
       "MINIA              2749839.33 [2732359;2770364]          0.97 [0;0]   \n",
       "SKESA              2801849.33 [2797241;2810284]          0.98 [0;0]   \n",
       "SPAdes             2794852.33 [2781588;2812518]          0.98 [0;0]   \n",
       "Unicycler          2803278.67 [2783327;2813747]          0.99 [0;0]   \n",
       "VelvetOptimiser          1823316.67 [0;2736286]          0.64 [0;0]   \n",
       "\n",
       "                            contigs    identity lowest_identity  \\\n",
       "Assembler                                                         \n",
       "ABySS                32.71 [19;106]   1.0 [0;0]       1.0 [0;0]   \n",
       "BCALM2               115.22 [0;180]  0.67 [0;1]      0.67 [0;1]   \n",
       "GATBMiniaPipeline     37.44 [28;46]   1.0 [0;1]      0.99 [0;1]   \n",
       "IDBA-UD              113.0 [53;231]   1.0 [0;0]      0.93 [0;0]   \n",
       "MEGAHIT               32.44 [31;34]   1.0 [0;0]      0.96 [0;0]   \n",
       "MetaHipMer2          88.67 [45;177]   1.0 [0;1]       1.0 [0;1]   \n",
       "metaSPAdes            31.67 [21;51]   1.0 [0;0]      0.98 [0;0]   \n",
       "MINIA              183.67 [166;219]   1.0 [0;0]       1.0 [0;0]   \n",
       "SKESA                 54.67 [40;72]  0.99 [0;1]      0.72 [0;1]   \n",
       "SPAdes                36.67 [21;63]   1.0 [0;0]       1.0 [0;0]   \n",
       "Unicycler             23.33 [20;30]   1.0 [0;0]       1.0 [0;0]   \n",
       "VelvetOptimiser      106.67 [0;163]  0.67 [0;0]      0.67 [0;0]   \n",
       "\n",
       "                  misassembled contigs misassembly events multiplicity  \\\n",
       "Assembler                                                                \n",
       "ABySS                        0.0 [0;0]          0.0 [0;0]   1.01 [1;1]   \n",
       "BCALM2                       0.0 [0;0]          0.0 [0;0]   0.67 [0;1]   \n",
       "GATBMiniaPipeline           0.33 [0;1]         0.67 [0;2]    1.0 [1;1]   \n",
       "IDBA-UD                      1.0 [0;3]          2.0 [0;6]    1.0 [0;1]   \n",
       "MEGAHIT                     0.67 [0;1]         1.33 [0;2]    1.0 [0;1]   \n",
       "MetaHipMer2                  0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "metaSPAdes                  0.33 [0;1]         0.67 [0;2]    1.0 [1;1]   \n",
       "MINIA                        0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "SKESA                       0.67 [0;1]         0.67 [0;1]    1.0 [1;1]   \n",
       "SPAdes                       1.0 [0;3]          2.0 [0;6]   1.01 [1;1]   \n",
       "Unicycler                    0.0 [0;0]          0.0 [0;0]    1.0 [0;0]   \n",
       "VelvetOptimiser              0.0 [0;0]          0.0 [0;0]   0.67 [0;1]   \n",
       "\n",
       "                    parsimony    validity  \n",
       "Assembler                                  \n",
       "ABySS              1.01 [1;1]   1.0 [0;1]  \n",
       "BCALM2             0.67 [0;1]  0.67 [0;1]  \n",
       "GATBMiniaPipeline   1.0 [1;1]   1.0 [0;1]  \n",
       "IDBA-UD             1.0 [1;1]   1.0 [0;0]  \n",
       "MEGAHIT             1.0 [1;1]   1.0 [0;0]  \n",
       "MetaHipMer2         1.0 [1;1]   1.0 [1;1]  \n",
       "metaSPAdes          1.0 [1;1]   1.0 [0;1]  \n",
       "MINIA               1.0 [1;1]   1.0 [0;1]  \n",
       "SKESA              1.01 [1;1]  0.99 [0;1]  \n",
       "SPAdes             1.01 [1;1]   1.0 [0;1]  \n",
       "Unicycler           1.0 [0;0]   1.0 [0;1]  \n",
       "VelvetOptimiser    0.67 [0;1]  0.67 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Escherichia_coli_plasmid\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.14 [0;1]</td>\n",
       "      <td>0.11 [0;0]</td>\n",
       "      <td>9679.86 [1638;55639]</td>\n",
       "      <td>8507.29 [0;55639]</td>\n",
       "      <td>133.71 [0;204]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>46052.43 [10505;109424]</td>\n",
       "      <td>0.42 [0;0]</td>\n",
       "      <td>14.14 [3;26]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.96 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>6.0 [0;18]</td>\n",
       "      <td>0.06 [0;0]</td>\n",
       "      <td>3435.67 [0;7117]</td>\n",
       "      <td>3249.0 [0;6638]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>64547.0 [0;102586]</td>\n",
       "      <td>0.59 [0;0]</td>\n",
       "      <td>18.0 [0;33]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>3.0 [0;7]</td>\n",
       "      <td>0.27 [0;0]</td>\n",
       "      <td>17539.33 [5269;28587]</td>\n",
       "      <td>17333.33 [4651;28587]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>104173.33 [97767;109450]</td>\n",
       "      <td>0.95 [0;0]</td>\n",
       "      <td>12.33 [3;24]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>5.0 [3;8]</td>\n",
       "      <td>0.27 [0;0]</td>\n",
       "      <td>22647.0 [12577;28584]</td>\n",
       "      <td>22647.0 [12577;28584]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>106594.33 [105631;108159]</td>\n",
       "      <td>0.97 [0;0]</td>\n",
       "      <td>7.67 [5;11]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>3.33 [1;6]</td>\n",
       "      <td>0.35 [0;0]</td>\n",
       "      <td>33051.0 [15523;55277]</td>\n",
       "      <td>32100.0 [12670;55277]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>108283.0 [107615;109516]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>5.33 [3;9]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>3.33 [0;7]</td>\n",
       "      <td>0.19 [0;0]</td>\n",
       "      <td>14116.0 [1503;28267]</td>\n",
       "      <td>13615.0 [0;28267]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>71853.78 [1503;107495]</td>\n",
       "      <td>0.65 [0;0]</td>\n",
       "      <td>5.33 [1;10]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>3.33 [1;6]</td>\n",
       "      <td>0.35 [0;0]</td>\n",
       "      <td>32452.0 [15437;55105]</td>\n",
       "      <td>31525.33 [12657;55105]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>106746.0 [105952;108144]</td>\n",
       "      <td>0.97 [0;0]</td>\n",
       "      <td>5.0 [2;9]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>7.67 [0;17]</td>\n",
       "      <td>0.14 [0;0]</td>\n",
       "      <td>8395.67 [3190;14811]</td>\n",
       "      <td>8186.0 [3109;14811]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>99620.67 [91055;104297]</td>\n",
       "      <td>0.91 [0;0]</td>\n",
       "      <td>20.67 [8;33]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.67 [0;2]</td>\n",
       "      <td>0.18 [0;0]</td>\n",
       "      <td>10403.0 [1404;28384]</td>\n",
       "      <td>9461.33 [0;28384]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>54383.67 [15416;107808]</td>\n",
       "      <td>0.49 [0;0]</td>\n",
       "      <td>13.67 [3;26]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>3.67 [2;6]</td>\n",
       "      <td>0.35 [0;0]</td>\n",
       "      <td>23605.33 [15438;28541]</td>\n",
       "      <td>22678.33 [12657;28541]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>108206.0 [107620;109262]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>6.0 [4;9]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>0.17 [0;0]</td>\n",
       "      <td>18332.33 [0;54997]</td>\n",
       "      <td>17644.0 [0;52932]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>35976.33 [0;107929]</td>\n",
       "      <td>0.33 [0;0]</td>\n",
       "      <td>0.67 [0;2]</td>\n",
       "      <td>0.33 [0;0]</td>\n",
       "      <td>0.33 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>2.33 [0;4]</td>\n",
       "      <td>0.2 [0;0]</td>\n",
       "      <td>16356.67 [0;27962]</td>\n",
       "      <td>16356.67 [0;27962]</td>\n",
       "      <td>343.0 [0;835]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>71559.33 [0;107377]</td>\n",
       "      <td>0.65 [0;0]</td>\n",
       "      <td>3.33 [0;6]</td>\n",
       "      <td>0.66 [0;0]</td>\n",
       "      <td>0.66 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm          L90         LSA  \\\n",
       "Assembler                                                                    \n",
       "ABySS              Multiple k-mer De Bruijn graph   0.14 [0;1]  0.11 [0;0]   \n",
       "BCALM2               Single k-mer De Bruijn graph   6.0 [0;18]  0.06 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph    3.0 [0;7]  0.27 [0;0]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph    5.0 [3;8]  0.27 [0;0]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph   3.33 [1;6]  0.35 [0;0]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph   3.33 [0;7]  0.19 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph   3.33 [1;6]  0.35 [0;0]   \n",
       "MINIA                Single k-mer De Bruijn graph  7.67 [0;17]  0.14 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph   0.67 [0;2]  0.18 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph   3.67 [2;6]  0.35 [0;0]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph   0.33 [0;1]  0.17 [0;0]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph   2.33 [0;4]   0.2 [0;0]   \n",
       "\n",
       "                                     NA50                    NG50  \\\n",
       "Assembler                                                           \n",
       "ABySS                9679.86 [1638;55639]       8507.29 [0;55639]   \n",
       "BCALM2                   3435.67 [0;7117]         3249.0 [0;6638]   \n",
       "GATBMiniaPipeline   17539.33 [5269;28587]   17333.33 [4651;28587]   \n",
       "IDBA-UD             22647.0 [12577;28584]   22647.0 [12577;28584]   \n",
       "MEGAHIT             33051.0 [15523;55277]   32100.0 [12670;55277]   \n",
       "MetaHipMer2          14116.0 [1503;28267]       13615.0 [0;28267]   \n",
       "metaSPAdes          32452.0 [15437;55105]  31525.33 [12657;55105]   \n",
       "MINIA                8395.67 [3190;14811]     8186.0 [3109;14811]   \n",
       "SKESA                10403.0 [1404;28384]       9461.33 [0;28384]   \n",
       "SPAdes             23605.33 [15438;28541]  22678.33 [12657;28541]   \n",
       "Unicycler              18332.33 [0;54997]       17644.0 [0;52932]   \n",
       "VelvetOptimiser        16356.67 [0;27962]      16356.67 [0;27962]   \n",
       "\n",
       "                               Ns         Type                  basepairs  \\\n",
       "Assembler                                                                   \n",
       "ABySS              133.71 [0;204]  Metagenomic    46052.43 [10505;109424]   \n",
       "BCALM2                  0.0 [0;0]      Genomic         64547.0 [0;102586]   \n",
       "GATBMiniaPipeline       0.0 [0;0]  Metagenomic   104173.33 [97767;109450]   \n",
       "IDBA-UD                 0.0 [0;0]  Metagenomic  106594.33 [105631;108159]   \n",
       "MEGAHIT                 0.0 [0;0]  Metagenomic   108283.0 [107615;109516]   \n",
       "MetaHipMer2             0.0 [0;0]  Metagenomic     71853.78 [1503;107495]   \n",
       "metaSPAdes              0.0 [0;0]  Metagenomic   106746.0 [105952;108144]   \n",
       "MINIA                   0.0 [0;0]      Genomic    99620.67 [91055;104297]   \n",
       "SKESA                   0.0 [0;0]      Genomic    54383.67 [15416;107808]   \n",
       "SPAdes                  0.0 [0;0]      Genomic   108206.0 [107620;109262]   \n",
       "Unicycler               0.0 [0;0]      Genomic        35976.33 [0;107929]   \n",
       "VelvetOptimiser     343.0 [0;835]  Metagenomic        71559.33 [0;107377]   \n",
       "\n",
       "                  breadth_of_coverage       contigs    identity  \\\n",
       "Assembler                                                         \n",
       "ABySS                      0.42 [0;0]  14.14 [3;26]  0.99 [0;0]   \n",
       "BCALM2                     0.59 [0;0]   18.0 [0;33]  0.67 [0;1]   \n",
       "GATBMiniaPipeline          0.95 [0;0]  12.33 [3;24]   1.0 [0;1]   \n",
       "IDBA-UD                    0.97 [0;0]   7.67 [5;11]   1.0 [0;1]   \n",
       "MEGAHIT                    0.98 [0;0]    5.33 [3;9]   1.0 [0;1]   \n",
       "MetaHipMer2                0.65 [0;0]   5.33 [1;10]   1.0 [1;1]   \n",
       "metaSPAdes                 0.97 [0;0]     5.0 [2;9]   1.0 [0;1]   \n",
       "MINIA                      0.91 [0;0]  20.67 [8;33]   1.0 [0;1]   \n",
       "SKESA                      0.49 [0;0]  13.67 [3;26]   1.0 [0;1]   \n",
       "SPAdes                     0.98 [0;0]     6.0 [4;9]   1.0 [0;1]   \n",
       "Unicycler                  0.33 [0;0]    0.67 [0;2]  0.33 [0;0]   \n",
       "VelvetOptimiser            0.65 [0;0]    3.33 [0;6]  0.66 [0;0]   \n",
       "\n",
       "                  lowest_identity misassembled contigs misassembly events  \\\n",
       "Assembler                                                                   \n",
       "ABySS                  0.96 [0;0]            0.0 [0;0]          0.0 [0;0]   \n",
       "BCALM2                 0.67 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "GATBMiniaPipeline       1.0 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "IDBA-UD                 1.0 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "MEGAHIT                 1.0 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "MetaHipMer2             1.0 [1;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "metaSPAdes              1.0 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "MINIA                   1.0 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "SKESA                   1.0 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "SPAdes                  1.0 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "Unicycler              0.33 [0;0]            0.0 [0;0]          0.0 [0;0]   \n",
       "VelvetOptimiser        0.66 [0;0]            0.0 [0;0]          0.0 [0;0]   \n",
       "\n",
       "                  multiplicity   parsimony    validity  \n",
       "Assembler                                               \n",
       "ABySS               1.01 [1;1]  1.01 [1;1]   1.0 [0;1]  \n",
       "BCALM2              0.67 [0;1]  0.67 [0;1]  0.67 [0;1]  \n",
       "GATBMiniaPipeline    1.0 [1;1]   1.0 [1;1]   1.0 [0;1]  \n",
       "IDBA-UD              1.0 [1;1]   1.0 [1;1]   1.0 [1;1]  \n",
       "MEGAHIT              1.0 [1;1]   1.0 [1;1]   1.0 [1;1]  \n",
       "MetaHipMer2          1.0 [1;1]   1.0 [1;1]   1.0 [1;1]  \n",
       "metaSPAdes           1.0 [1;1]   1.0 [1;1]   1.0 [1;1]  \n",
       "MINIA                1.0 [1;1]   1.0 [1;1]   1.0 [0;1]  \n",
       "SKESA                1.0 [1;1]   1.0 [1;1]   1.0 [1;1]  \n",
       "SPAdes               1.0 [1;1]   1.0 [1;1]   1.0 [1;1]  \n",
       "Unicycler           0.33 [0;1]  0.33 [0;1]  0.33 [0;1]  \n",
       "VelvetOptimiser     0.67 [0;1]  0.67 [0;1]  0.67 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Escherichia_coli\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>88.0 [50;133]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>68560.86 [43381;100101]</td>\n",
       "      <td>68444.71 [42568;100101]</td>\n",
       "      <td>19.29 [0;45]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>4695531.86 [4644860;4706885]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>158.14 [100;234]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.85 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.02 [1;1]</td>\n",
       "      <td>1.02 [1;1]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>2236.33 [0;4465]</td>\n",
       "      <td>1563.0 [0;3200]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2286381.11 [0;3766804]</td>\n",
       "      <td>0.48 [0;0]</td>\n",
       "      <td>871.67 [0;1495]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>103.67 [99;112]</td>\n",
       "      <td>0.03 [0;0]</td>\n",
       "      <td>58200.33 [55370;59656]</td>\n",
       "      <td>57950.33 [54964;59530]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>4624540.22 [4615491;4632389]</td>\n",
       "      <td>0.97 [0;0]</td>\n",
       "      <td>187.67 [180;199]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>249.22 [215;283]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>20242.33 [18743;22923]</td>\n",
       "      <td>19925.11 [18446;22537]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>4535757.44 [4497818;4589511]</td>\n",
       "      <td>0.95 [0;0]</td>\n",
       "      <td>415.22 [358;449]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.55 [0;0]</td>\n",
       "      <td>5.67 [3;8]</td>\n",
       "      <td>10.33 [6;15]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>50.67 [43;55]</td>\n",
       "      <td>0.06 [0;0]</td>\n",
       "      <td>79557.0 [74104;90017]</td>\n",
       "      <td>84034.0 [74550;102135]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>4666033.67 [4662300;4673019]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>136.67 [132;141]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.88 [0;0]</td>\n",
       "      <td>3.33 [3;4]</td>\n",
       "      <td>5.67 [5;7]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>72.33 [0;110]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>39025.78 [1161;59122]</td>\n",
       "      <td>37715.33 [0;56794]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3117359.11 [126763;4613118]</td>\n",
       "      <td>0.65 [0;0]</td>\n",
       "      <td>163.44 [106;193]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>54.0 [33;90]</td>\n",
       "      <td>0.09 [0;0]</td>\n",
       "      <td>104053.33 [65619;138793]</td>\n",
       "      <td>102524.33 [61032;138793]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>4653317.33 [4635986;4666787]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>111.67 [82;167]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>1.0 [0;2]</td>\n",
       "      <td>3.0 [0;6]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.01 [0;0]</td>\n",
       "      <td>4448.11 [4409;4469]</td>\n",
       "      <td>3161.89 [3072;3211]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>3763911.33 [3746226;3773077]</td>\n",
       "      <td>0.79 [0;0]</td>\n",
       "      <td>1130.22 [1122;1146]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>117.0 [70;152]</td>\n",
       "      <td>0.03 [0;0]</td>\n",
       "      <td>46524.33 [34752;62623]</td>\n",
       "      <td>47216.33 [34302;66235]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>4571861.67 [4560589;4579934]</td>\n",
       "      <td>0.96 [0;0]</td>\n",
       "      <td>219.67 [180;270]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.71 [0;0]</td>\n",
       "      <td>1.67 [0;3]</td>\n",
       "      <td>2.67 [0;4]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>57.67 [45;79]</td>\n",
       "      <td>0.07 [0;0]</td>\n",
       "      <td>83719.33 [66358;98812]</td>\n",
       "      <td>83729.0 [65182;96999]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>4630205.67 [4615956;4643850]</td>\n",
       "      <td>0.97 [0;0]</td>\n",
       "      <td>115.67 [84;143]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.78 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>1.33 [0;3]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>45.33 [35;57]</td>\n",
       "      <td>0.09 [0;0]</td>\n",
       "      <td>109030.0 [96845;124073]</td>\n",
       "      <td>106782.0 [94991;119183]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>4618550.67 [4594376;4632276]</td>\n",
       "      <td>0.97 [0;0]</td>\n",
       "      <td>93.67 [72;111]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>0.67 [0;2]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.01 [0;0]</td>\n",
       "      <td>3063.67 [0;4598]</td>\n",
       "      <td>2232.67 [0;3363]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2537083.67 [0;3810326]</td>\n",
       "      <td>0.53 [0;0]</td>\n",
       "      <td>730.33 [0;1098]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm               L90  \\\n",
       "Assembler                                                             \n",
       "ABySS              Multiple k-mer De Bruijn graph     88.0 [50;133]   \n",
       "BCALM2               Single k-mer De Bruijn graph         0.0 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph   103.67 [99;112]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  249.22 [215;283]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph     50.67 [43;55]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph     72.33 [0;110]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph      54.0 [33;90]   \n",
       "MINIA                Single k-mer De Bruijn graph         0.0 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph    117.0 [70;152]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph     57.67 [45;79]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph     45.33 [35;57]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph         0.0 [0;0]   \n",
       "\n",
       "                          LSA                      NA50  \\\n",
       "Assembler                                                 \n",
       "ABySS              0.04 [0;0]   68560.86 [43381;100101]   \n",
       "BCALM2              0.0 [0;0]          2236.33 [0;4465]   \n",
       "GATBMiniaPipeline  0.03 [0;0]    58200.33 [55370;59656]   \n",
       "IDBA-UD            0.02 [0;0]    20242.33 [18743;22923]   \n",
       "MEGAHIT            0.06 [0;0]     79557.0 [74104;90017]   \n",
       "MetaHipMer2        0.02 [0;0]     39025.78 [1161;59122]   \n",
       "metaSPAdes         0.09 [0;0]  104053.33 [65619;138793]   \n",
       "MINIA              0.01 [0;0]       4448.11 [4409;4469]   \n",
       "SKESA              0.03 [0;0]    46524.33 [34752;62623]   \n",
       "SPAdes             0.07 [0;0]    83719.33 [66358;98812]   \n",
       "Unicycler          0.09 [0;0]   109030.0 [96845;124073]   \n",
       "VelvetOptimiser    0.01 [0;0]          3063.67 [0;4598]   \n",
       "\n",
       "                                       NG50            Ns         Type  \\\n",
       "Assembler                                                                \n",
       "ABySS               68444.71 [42568;100101]  19.29 [0;45]  Metagenomic   \n",
       "BCALM2                      1563.0 [0;3200]     0.0 [0;0]      Genomic   \n",
       "GATBMiniaPipeline    57950.33 [54964;59530]     0.0 [0;0]  Metagenomic   \n",
       "IDBA-UD              19925.11 [18446;22537]     0.0 [0;0]  Metagenomic   \n",
       "MEGAHIT              84034.0 [74550;102135]     0.0 [0;0]  Metagenomic   \n",
       "MetaHipMer2              37715.33 [0;56794]     0.0 [0;0]  Metagenomic   \n",
       "metaSPAdes         102524.33 [61032;138793]     0.0 [0;0]  Metagenomic   \n",
       "MINIA                   3161.89 [3072;3211]     0.0 [0;0]      Genomic   \n",
       "SKESA                47216.33 [34302;66235]     0.0 [0;0]      Genomic   \n",
       "SPAdes                83729.0 [65182;96999]     0.0 [0;0]      Genomic   \n",
       "Unicycler           106782.0 [94991;119183]     0.0 [0;0]      Genomic   \n",
       "VelvetOptimiser            2232.67 [0;3363]     0.0 [0;0]  Metagenomic   \n",
       "\n",
       "                                      basepairs breadth_of_coverage  \\\n",
       "Assembler                                                             \n",
       "ABySS              4695531.86 [4644860;4706885]          0.99 [0;0]   \n",
       "BCALM2                   2286381.11 [0;3766804]          0.48 [0;0]   \n",
       "GATBMiniaPipeline  4624540.22 [4615491;4632389]          0.97 [0;0]   \n",
       "IDBA-UD            4535757.44 [4497818;4589511]          0.95 [0;0]   \n",
       "MEGAHIT            4666033.67 [4662300;4673019]          0.98 [0;0]   \n",
       "MetaHipMer2         3117359.11 [126763;4613118]          0.65 [0;0]   \n",
       "metaSPAdes         4653317.33 [4635986;4666787]          0.98 [0;0]   \n",
       "MINIA              3763911.33 [3746226;3773077]          0.79 [0;0]   \n",
       "SKESA              4571861.67 [4560589;4579934]          0.96 [0;0]   \n",
       "SPAdes             4630205.67 [4615956;4643850]          0.97 [0;0]   \n",
       "Unicycler          4618550.67 [4594376;4632276]          0.97 [0;0]   \n",
       "VelvetOptimiser          2537083.67 [0;3810326]          0.53 [0;0]   \n",
       "\n",
       "                               contigs    identity lowest_identity  \\\n",
       "Assembler                                                            \n",
       "ABySS                 158.14 [100;234]   1.0 [0;0]      0.85 [0;0]   \n",
       "BCALM2                 871.67 [0;1495]  0.67 [0;1]      0.67 [0;1]   \n",
       "GATBMiniaPipeline     187.67 [180;199]   1.0 [0;0]      0.99 [0;0]   \n",
       "IDBA-UD               415.22 [358;449]  0.99 [0;0]      0.55 [0;0]   \n",
       "MEGAHIT               136.67 [132;141]   1.0 [0;0]      0.88 [0;0]   \n",
       "MetaHipMer2           163.44 [106;193]   1.0 [0;1]      0.99 [0;1]   \n",
       "metaSPAdes             111.67 [82;167]   1.0 [0;0]      0.99 [0;0]   \n",
       "MINIA              1130.22 [1122;1146]   1.0 [0;0]      0.99 [0;0]   \n",
       "SKESA                 219.67 [180;270]   1.0 [0;0]      0.71 [0;0]   \n",
       "SPAdes                 115.67 [84;143]   1.0 [0;0]      0.78 [0;0]   \n",
       "Unicycler               93.67 [72;111]   1.0 [0;0]      0.99 [0;0]   \n",
       "VelvetOptimiser        730.33 [0;1098]  0.67 [0;1]      0.67 [0;1]   \n",
       "\n",
       "                  misassembled contigs misassembly events multiplicity  \\\n",
       "Assembler                                                                \n",
       "ABySS                        0.0 [0;0]          0.0 [0;0]   1.02 [1;1]   \n",
       "BCALM2                       0.0 [0;0]          0.0 [0;0]   0.67 [0;1]   \n",
       "GATBMiniaPipeline            0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "IDBA-UD                     5.67 [3;8]       10.33 [6;15]   1.01 [1;1]   \n",
       "MEGAHIT                     3.33 [3;4]         5.67 [5;7]   1.01 [1;1]   \n",
       "MetaHipMer2                  0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "metaSPAdes                   1.0 [0;2]          3.0 [0;6]    1.0 [0;1]   \n",
       "MINIA                        0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "SKESA                       1.67 [0;3]         2.67 [0;4]    1.0 [1;1]   \n",
       "SPAdes                      0.67 [0;1]         1.33 [0;3]    1.0 [1;1]   \n",
       "Unicycler                   0.33 [0;1]         0.67 [0;2]    1.0 [0;1]   \n",
       "VelvetOptimiser              0.0 [0;0]          0.0 [0;0]   0.67 [0;1]   \n",
       "\n",
       "                    parsimony    validity  \n",
       "Assembler                                  \n",
       "ABySS              1.02 [1;1]   1.0 [0;0]  \n",
       "BCALM2             0.67 [0;1]  0.67 [0;1]  \n",
       "GATBMiniaPipeline   1.0 [1;1]   1.0 [0;1]  \n",
       "IDBA-UD            1.01 [1;1]  0.99 [0;0]  \n",
       "MEGAHIT            1.01 [1;1]   1.0 [0;0]  \n",
       "MetaHipMer2         1.0 [1;1]   1.0 [0;1]  \n",
       "metaSPAdes          1.0 [0;1]   1.0 [0;1]  \n",
       "MINIA               1.0 [1;1]   1.0 [0;1]  \n",
       "SKESA              1.01 [1;1]   1.0 [0;0]  \n",
       "SPAdes              1.0 [1;1]   1.0 [0;0]  \n",
       "Unicycler           1.0 [0;1]   1.0 [0;1]  \n",
       "VelvetOptimiser    0.67 [0;1]  0.67 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Lactobacillus_fermentum\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>50.86 [41;104]</td>\n",
       "      <td>0.06 [0;0]</td>\n",
       "      <td>39094.0 [19450;42368]</td>\n",
       "      <td>38939.0 [18365;42368]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1852399.43 [1782430;1873053]</td>\n",
       "      <td>0.97 [0;0]</td>\n",
       "      <td>81.29 [72;137]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>69.33 [0;208]</td>\n",
       "      <td>0.01 [0;0]</td>\n",
       "      <td>3736.33 [0;11209]</td>\n",
       "      <td>3411.33 [0;10234]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>582868.44 [0;1748648]</td>\n",
       "      <td>0.31 [0;0]</td>\n",
       "      <td>78.67 [0;236]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>54.0 [53;55]</td>\n",
       "      <td>0.06 [0;0]</td>\n",
       "      <td>40433.33 [40366;40488]</td>\n",
       "      <td>39257.0 [37549;40151]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1795961.0 [1792170;1798443]</td>\n",
       "      <td>0.94 [0;0]</td>\n",
       "      <td>87.0 [83;89]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>81.67 [56;131]</td>\n",
       "      <td>0.06 [0;0]</td>\n",
       "      <td>32188.0 [16131;40364]</td>\n",
       "      <td>30177.67 [15599;37467]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1785823.56 [1779383;1789854]</td>\n",
       "      <td>0.94 [0;0]</td>\n",
       "      <td>116.0 [88;171]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>42.67 [35;53]</td>\n",
       "      <td>0.06 [0;0]</td>\n",
       "      <td>44246.33 [40505;46932]</td>\n",
       "      <td>45542.67 [38354;52972]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1807707.22 [1806816;1808544]</td>\n",
       "      <td>0.95 [0;0]</td>\n",
       "      <td>83.67 [81;88]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.88 [0;0]</td>\n",
       "      <td>2.33 [2;3]</td>\n",
       "      <td>5.22 [4;6]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>36.33 [0;55]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>27301.0 [1179;40362]</td>\n",
       "      <td>26711.33 [0;40067]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1194713.11 [9143;1787622]</td>\n",
       "      <td>0.63 [0;0]</td>\n",
       "      <td>60.33 [8;87]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>53.33 [48;64]</td>\n",
       "      <td>0.06 [0;0]</td>\n",
       "      <td>42817.67 [36315;46069]</td>\n",
       "      <td>39838.33 [33788;45077]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1792094.33 [1784123;1796429]</td>\n",
       "      <td>0.94 [0;0]</td>\n",
       "      <td>84.33 [78;95]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>225.11 [206;263]</td>\n",
       "      <td>0.03 [0;0]</td>\n",
       "      <td>10527.33 [9080;11293]</td>\n",
       "      <td>9960.33 [8209;10836]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1747076.11 [1733284;1754083]</td>\n",
       "      <td>0.92 [0;0]</td>\n",
       "      <td>252.11 [237;282]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>63.0 [56;69]</td>\n",
       "      <td>0.06 [0;0]</td>\n",
       "      <td>37326.33 [34729;40164]</td>\n",
       "      <td>34547.67 [30688;39869]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1769443.67 [1757511;1776186]</td>\n",
       "      <td>0.93 [0;0]</td>\n",
       "      <td>88.67 [86;94]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>60.33 [47;76]</td>\n",
       "      <td>0.06 [0;0]</td>\n",
       "      <td>38747.67 [32241;46123]</td>\n",
       "      <td>37215.67 [29397;45008]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1789419.0 [1782712;1792828]</td>\n",
       "      <td>0.94 [0;0]</td>\n",
       "      <td>96.0 [78;109]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>1.33 [0;4]</td>\n",
       "      <td>2.67 [0;8]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>51.67 [49;55]</td>\n",
       "      <td>0.06 [0;0]</td>\n",
       "      <td>43604.67 [40540;45959]</td>\n",
       "      <td>40236.33 [40168;40306]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1773873.67 [1771389;1775427]</td>\n",
       "      <td>0.93 [0;0]</td>\n",
       "      <td>78.67 [77;80]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>135.0 [0;207]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>7787.33 [0;11784]</td>\n",
       "      <td>7360.33 [0;11122]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1166876.0 [0;1751517]</td>\n",
       "      <td>0.61 [0;0]</td>\n",
       "      <td>154.0 [0;235]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm               L90  \\\n",
       "Assembler                                                             \n",
       "ABySS              Multiple k-mer De Bruijn graph    50.86 [41;104]   \n",
       "BCALM2               Single k-mer De Bruijn graph     69.33 [0;208]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph      54.0 [53;55]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph    81.67 [56;131]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph     42.67 [35;53]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph      36.33 [0;55]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph     53.33 [48;64]   \n",
       "MINIA                Single k-mer De Bruijn graph  225.11 [206;263]   \n",
       "SKESA              Multiple k-mer De Bruijn graph      63.0 [56;69]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph     60.33 [47;76]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph     51.67 [49;55]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph     135.0 [0;207]   \n",
       "\n",
       "                          LSA                    NA50                    NG50  \\\n",
       "Assembler                                                                       \n",
       "ABySS              0.06 [0;0]   39094.0 [19450;42368]   38939.0 [18365;42368]   \n",
       "BCALM2             0.01 [0;0]       3736.33 [0;11209]       3411.33 [0;10234]   \n",
       "GATBMiniaPipeline  0.06 [0;0]  40433.33 [40366;40488]   39257.0 [37549;40151]   \n",
       "IDBA-UD            0.06 [0;0]   32188.0 [16131;40364]  30177.67 [15599;37467]   \n",
       "MEGAHIT            0.06 [0;0]  44246.33 [40505;46932]  45542.67 [38354;52972]   \n",
       "MetaHipMer2        0.04 [0;0]    27301.0 [1179;40362]      26711.33 [0;40067]   \n",
       "metaSPAdes         0.06 [0;0]  42817.67 [36315;46069]  39838.33 [33788;45077]   \n",
       "MINIA              0.03 [0;0]   10527.33 [9080;11293]    9960.33 [8209;10836]   \n",
       "SKESA              0.06 [0;0]  37326.33 [34729;40164]  34547.67 [30688;39869]   \n",
       "SPAdes             0.06 [0;0]  38747.67 [32241;46123]  37215.67 [29397;45008]   \n",
       "Unicycler          0.06 [0;0]  43604.67 [40540;45959]  40236.33 [40168;40306]   \n",
       "VelvetOptimiser    0.02 [0;0]       7787.33 [0;11784]       7360.33 [0;11122]   \n",
       "\n",
       "                          Ns         Type                     basepairs  \\\n",
       "Assembler                                                                 \n",
       "ABySS              0.0 [0;0]  Metagenomic  1852399.43 [1782430;1873053]   \n",
       "BCALM2             0.0 [0;0]      Genomic         582868.44 [0;1748648]   \n",
       "GATBMiniaPipeline  0.0 [0;0]  Metagenomic   1795961.0 [1792170;1798443]   \n",
       "IDBA-UD            0.0 [0;0]  Metagenomic  1785823.56 [1779383;1789854]   \n",
       "MEGAHIT            0.0 [0;0]  Metagenomic  1807707.22 [1806816;1808544]   \n",
       "MetaHipMer2        0.0 [0;0]  Metagenomic     1194713.11 [9143;1787622]   \n",
       "metaSPAdes         0.0 [0;0]  Metagenomic  1792094.33 [1784123;1796429]   \n",
       "MINIA              0.0 [0;0]      Genomic  1747076.11 [1733284;1754083]   \n",
       "SKESA              0.0 [0;0]      Genomic  1769443.67 [1757511;1776186]   \n",
       "SPAdes             0.0 [0;0]      Genomic   1789419.0 [1782712;1792828]   \n",
       "Unicycler          0.0 [0;0]      Genomic  1773873.67 [1771389;1775427]   \n",
       "VelvetOptimiser    0.0 [0;0]  Metagenomic         1166876.0 [0;1751517]   \n",
       "\n",
       "                  breadth_of_coverage           contigs    identity  \\\n",
       "Assembler                                                             \n",
       "ABySS                      0.97 [0;0]    81.29 [72;137]   1.0 [0;0]   \n",
       "BCALM2                     0.31 [0;0]     78.67 [0;236]  0.33 [0;1]   \n",
       "GATBMiniaPipeline          0.94 [0;0]      87.0 [83;89]   1.0 [0;1]   \n",
       "IDBA-UD                    0.94 [0;0]    116.0 [88;171]   1.0 [0;0]   \n",
       "MEGAHIT                    0.95 [0;0]     83.67 [81;88]   1.0 [0;0]   \n",
       "MetaHipMer2                0.63 [0;0]      60.33 [8;87]   1.0 [1;1]   \n",
       "metaSPAdes                 0.94 [0;0]     84.33 [78;95]   1.0 [0;0]   \n",
       "MINIA                      0.92 [0;0]  252.11 [237;282]   1.0 [0;1]   \n",
       "SKESA                      0.93 [0;0]     88.67 [86;94]   1.0 [0;1]   \n",
       "SPAdes                     0.94 [0;0]     96.0 [78;109]   1.0 [0;0]   \n",
       "Unicycler                  0.93 [0;0]     78.67 [77;80]   1.0 [0;1]   \n",
       "VelvetOptimiser            0.61 [0;0]     154.0 [0;235]  0.67 [0;1]   \n",
       "\n",
       "                  lowest_identity misassembled contigs misassembly events  \\\n",
       "Assembler                                                                   \n",
       "ABySS                   1.0 [0;0]            0.0 [0;0]          0.0 [0;0]   \n",
       "BCALM2                 0.33 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "GATBMiniaPipeline      0.99 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "IDBA-UD                 1.0 [0;0]            0.0 [0;0]          0.0 [0;0]   \n",
       "MEGAHIT                0.88 [0;0]           2.33 [2;3]         5.22 [4;6]   \n",
       "MetaHipMer2             1.0 [1;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "metaSPAdes             0.99 [0;0]           0.33 [0;1]         0.33 [0;1]   \n",
       "MINIA                  0.98 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "SKESA                   1.0 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "SPAdes                  1.0 [0;0]           1.33 [0;4]         2.67 [0;8]   \n",
       "Unicycler               1.0 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "VelvetOptimiser        0.67 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "\n",
       "                  multiplicity   parsimony    validity  \n",
       "Assembler                                               \n",
       "ABySS               1.01 [1;1]  1.01 [1;1]   1.0 [0;1]  \n",
       "BCALM2              0.33 [0;1]  0.33 [0;1]  0.33 [0;1]  \n",
       "GATBMiniaPipeline    1.0 [1;1]   1.0 [1;1]   1.0 [0;1]  \n",
       "IDBA-UD              1.0 [1;1]   1.0 [1;1]   1.0 [1;1]  \n",
       "MEGAHIT              1.0 [0;1]   1.0 [0;1]   1.0 [0;0]  \n",
       "MetaHipMer2          1.0 [1;1]   1.0 [1;1]   1.0 [1;1]  \n",
       "metaSPAdes           1.0 [1;1]   1.0 [1;1]   1.0 [0;1]  \n",
       "MINIA                1.0 [1;1]   1.0 [1;1]   1.0 [0;1]  \n",
       "SKESA                1.0 [1;1]   1.0 [1;1]   1.0 [0;1]  \n",
       "SPAdes               1.0 [1;1]   1.0 [1;1]   1.0 [0;1]  \n",
       "Unicycler            1.0 [0;1]   1.0 [0;1]   1.0 [0;1]  \n",
       "VelvetOptimiser     0.67 [0;1]  0.67 [0;1]  0.67 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Listeria_monocytogenes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>11.71 [8;31]</td>\n",
       "      <td>0.22 [0;0]</td>\n",
       "      <td>314902.57 [96389;351502]</td>\n",
       "      <td>314902.57 [96389;351502]</td>\n",
       "      <td>0.43 [0;1]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2956884.43 [2941416;2961845]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>19.57 [14;50]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>24.67 [0;75]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>15869.33 [0;46368]</td>\n",
       "      <td>14784.33 [0;44353]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1118273.33 [0;2916831]</td>\n",
       "      <td>0.37 [0;0]</td>\n",
       "      <td>161.78 [0;359]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>11.0 [10;12]</td>\n",
       "      <td>0.2 [0;0]</td>\n",
       "      <td>384896.67 [351140;452364]</td>\n",
       "      <td>285810.33 [155105;351186]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2938753.56 [2915611;2957674]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>20.89 [19;24]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>44.67 [13;108]</td>\n",
       "      <td>0.14 [0;0]</td>\n",
       "      <td>178839.0 [27259;254629]</td>\n",
       "      <td>178271.0 [25555;254629]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2928181.67 [2915464;2952356]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>74.0 [25;171]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>7.33 [6;8]</td>\n",
       "      <td>0.28 [0;0]</td>\n",
       "      <td>479521.67 [447841;542872]</td>\n",
       "      <td>479521.67 [447841;542872]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2944849.56 [2925388;2961114]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>19.22 [19;20]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.95 [0;0]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>0.67 [0;2]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>8.67 [0;13]</td>\n",
       "      <td>0.1 [0;0]</td>\n",
       "      <td>149928.33 [1221;254627]</td>\n",
       "      <td>149520.67 [0;254627]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2102458.89 [454982;2949910]</td>\n",
       "      <td>0.7 [0;0]</td>\n",
       "      <td>135.56 [21;361]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>9.33 [6;13]</td>\n",
       "      <td>0.23 [0;0]</td>\n",
       "      <td>353907.0 [235115;475510]</td>\n",
       "      <td>353907.0 [235115;475510]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2935095.67 [2920851;2961206]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>18.0 [13;25]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>89.44 [72;122]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>40292.67 [28142;46368]</td>\n",
       "      <td>38618.0 [26654;44353]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2900580.22 [2882699;2918434]</td>\n",
       "      <td>0.97 [0;0]</td>\n",
       "      <td>146.67 [127;186]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>13.33 [11;17]</td>\n",
       "      <td>0.19 [0;0]</td>\n",
       "      <td>299330.0 [195967;351014]</td>\n",
       "      <td>299330.0 [195967;351014]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2911993.0 [2910021;2914137]</td>\n",
       "      <td>0.97 [0;0]</td>\n",
       "      <td>24.67 [19;31]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>13.0 [10;19]</td>\n",
       "      <td>0.16 [0;0]</td>\n",
       "      <td>234397.67 [147676;300425]</td>\n",
       "      <td>234397.67 [147676;300425]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2939672.0 [2916832;2957750]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>24.33 [16;37]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>8.33 [8;9]</td>\n",
       "      <td>0.23 [0;0]</td>\n",
       "      <td>392390.0 [350995;475179]</td>\n",
       "      <td>392390.0 [350995;475179]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2927893.0 [2914588;2953581]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>16.33 [16;17]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>46.67 [0;74]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>32324.67 [0;52654]</td>\n",
       "      <td>31839.67 [0;51199]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1945060.33 [0;2918755]</td>\n",
       "      <td>0.65 [0;0]</td>\n",
       "      <td>80.33 [0;125]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm             L90         LSA  \\\n",
       "Assembler                                                                       \n",
       "ABySS              Multiple k-mer De Bruijn graph    11.71 [8;31]  0.22 [0;0]   \n",
       "BCALM2               Single k-mer De Bruijn graph    24.67 [0;75]  0.02 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph    11.0 [10;12]   0.2 [0;0]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  44.67 [13;108]  0.14 [0;0]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph      7.33 [6;8]  0.28 [0;0]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph     8.67 [0;13]   0.1 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph     9.33 [6;13]  0.23 [0;0]   \n",
       "MINIA                Single k-mer De Bruijn graph  89.44 [72;122]  0.04 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph   13.33 [11;17]  0.19 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph    13.0 [10;19]  0.16 [0;0]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph      8.33 [8;9]  0.23 [0;0]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph    46.67 [0;74]  0.04 [0;0]   \n",
       "\n",
       "                                        NA50                       NG50  \\\n",
       "Assembler                                                                 \n",
       "ABySS               314902.57 [96389;351502]   314902.57 [96389;351502]   \n",
       "BCALM2                    15869.33 [0;46368]         14784.33 [0;44353]   \n",
       "GATBMiniaPipeline  384896.67 [351140;452364]  285810.33 [155105;351186]   \n",
       "IDBA-UD              178839.0 [27259;254629]    178271.0 [25555;254629]   \n",
       "MEGAHIT            479521.67 [447841;542872]  479521.67 [447841;542872]   \n",
       "MetaHipMer2          149928.33 [1221;254627]       149520.67 [0;254627]   \n",
       "metaSPAdes          353907.0 [235115;475510]   353907.0 [235115;475510]   \n",
       "MINIA                 40292.67 [28142;46368]      38618.0 [26654;44353]   \n",
       "SKESA               299330.0 [195967;351014]   299330.0 [195967;351014]   \n",
       "SPAdes             234397.67 [147676;300425]  234397.67 [147676;300425]   \n",
       "Unicycler           392390.0 [350995;475179]   392390.0 [350995;475179]   \n",
       "VelvetOptimiser           32324.67 [0;52654]         31839.67 [0;51199]   \n",
       "\n",
       "                           Ns         Type                     basepairs  \\\n",
       "Assembler                                                                  \n",
       "ABySS              0.43 [0;1]  Metagenomic  2956884.43 [2941416;2961845]   \n",
       "BCALM2              0.0 [0;0]      Genomic        1118273.33 [0;2916831]   \n",
       "GATBMiniaPipeline   0.0 [0;0]  Metagenomic  2938753.56 [2915611;2957674]   \n",
       "IDBA-UD             0.0 [0;0]  Metagenomic  2928181.67 [2915464;2952356]   \n",
       "MEGAHIT             0.0 [0;0]  Metagenomic  2944849.56 [2925388;2961114]   \n",
       "MetaHipMer2         0.0 [0;0]  Metagenomic   2102458.89 [454982;2949910]   \n",
       "metaSPAdes          0.0 [0;0]  Metagenomic  2935095.67 [2920851;2961206]   \n",
       "MINIA               0.0 [0;0]      Genomic  2900580.22 [2882699;2918434]   \n",
       "SKESA               0.0 [0;0]      Genomic   2911993.0 [2910021;2914137]   \n",
       "SPAdes              0.0 [0;0]      Genomic   2939672.0 [2916832;2957750]   \n",
       "Unicycler           0.0 [0;0]      Genomic   2927893.0 [2914588;2953581]   \n",
       "VelvetOptimiser     0.0 [0;0]  Metagenomic        1945060.33 [0;2918755]   \n",
       "\n",
       "                  breadth_of_coverage           contigs    identity  \\\n",
       "Assembler                                                             \n",
       "ABySS                      0.99 [0;0]     19.57 [14;50]   1.0 [0;0]   \n",
       "BCALM2                     0.37 [0;0]    161.78 [0;359]  0.67 [0;1]   \n",
       "GATBMiniaPipeline          0.98 [0;0]     20.89 [19;24]   1.0 [0;1]   \n",
       "IDBA-UD                    0.98 [0;0]     74.0 [25;171]   1.0 [0;0]   \n",
       "MEGAHIT                    0.98 [0;0]     19.22 [19;20]   1.0 [0;0]   \n",
       "MetaHipMer2                 0.7 [0;0]   135.56 [21;361]   1.0 [0;1]   \n",
       "metaSPAdes                 0.98 [0;0]      18.0 [13;25]   1.0 [0;0]   \n",
       "MINIA                      0.97 [0;0]  146.67 [127;186]   1.0 [0;0]   \n",
       "SKESA                      0.97 [0;0]     24.67 [19;31]   1.0 [0;1]   \n",
       "SPAdes                     0.98 [0;0]     24.33 [16;37]   1.0 [0;0]   \n",
       "Unicycler                  0.98 [0;0]     16.33 [16;17]   1.0 [0;0]   \n",
       "VelvetOptimiser            0.65 [0;0]     80.33 [0;125]  0.67 [0;1]   \n",
       "\n",
       "                  lowest_identity misassembled contigs misassembly events  \\\n",
       "Assembler                                                                   \n",
       "ABySS                   1.0 [0;0]            0.0 [0;0]          0.0 [0;0]   \n",
       "BCALM2                 0.67 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "GATBMiniaPipeline      0.99 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "IDBA-UD                0.99 [0;0]            0.0 [0;0]          0.0 [0;0]   \n",
       "MEGAHIT                0.95 [0;0]           0.33 [0;1]         0.67 [0;2]   \n",
       "MetaHipMer2             1.0 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "metaSPAdes             0.99 [0;0]            0.0 [0;0]          0.0 [0;0]   \n",
       "MINIA                  0.99 [0;0]            0.0 [0;0]          0.0 [0;0]   \n",
       "SKESA                   1.0 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "SPAdes                 0.99 [0;0]            0.0 [0;0]          0.0 [0;0]   \n",
       "Unicycler              0.98 [0;0]            0.0 [0;0]          0.0 [0;0]   \n",
       "VelvetOptimiser        0.67 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "\n",
       "                  multiplicity   parsimony    validity  \n",
       "Assembler                                               \n",
       "ABySS               1.01 [1;1]  1.01 [1;1]   1.0 [0;1]  \n",
       "BCALM2              0.67 [0;1]  0.67 [0;1]  0.67 [0;1]  \n",
       "GATBMiniaPipeline    1.0 [0;1]   1.0 [0;1]   1.0 [0;1]  \n",
       "IDBA-UD              1.0 [0;1]   1.0 [0;1]   1.0 [0;1]  \n",
       "MEGAHIT              1.0 [0;1]   1.0 [0;1]   1.0 [0;0]  \n",
       "MetaHipMer2          1.0 [1;1]   1.0 [1;1]   1.0 [0;1]  \n",
       "metaSPAdes           1.0 [1;1]   1.0 [1;1]   1.0 [0;1]  \n",
       "MINIA                1.0 [1;1]   1.0 [1;1]   1.0 [0;1]  \n",
       "SKESA                1.0 [0;1]   1.0 [0;1]   1.0 [1;1]  \n",
       "SPAdes               1.0 [0;1]   1.0 [0;1]   1.0 [0;1]  \n",
       "Unicycler            1.0 [0;1]   1.0 [0;1]   1.0 [0;0]  \n",
       "VelvetOptimiser     0.67 [0;1]  0.67 [0;1]  0.67 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pseudomonas_aeruginosa\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>478.71 [67;1004]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>49503.43 [6462;92433]</td>\n",
       "      <td>50037.57 [6430;93704]</td>\n",
       "      <td>1057.0 [3;2170]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>6674637.0 [6555420;6780808]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>701.29 [143;1404]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.81 [0;0]</td>\n",
       "      <td>4.14 [0;8]</td>\n",
       "      <td>7.0 [0;14]</td>\n",
       "      <td>1.02 [1;1]</td>\n",
       "      <td>1.02 [1;1]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>420.33 [0;921]</td>\n",
       "      <td>0.01 [0;0]</td>\n",
       "      <td>10132.33 [0;21972]</td>\n",
       "      <td>9640.33 [0;21139]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>4303694.0 [0;6510588]</td>\n",
       "      <td>0.63 [0;0]</td>\n",
       "      <td>555.0 [0;1142]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>52.33 [39;78]</td>\n",
       "      <td>0.06 [0;0]</td>\n",
       "      <td>165922.67 [88774;204691]</td>\n",
       "      <td>165922.67 [88774;204691]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>6708469.67 [6703986;6716518]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>122.0 [98;166]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.96 [0;0]</td>\n",
       "      <td>1.0 [0;3]</td>\n",
       "      <td>2.0 [0;6]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>79.0 [71;90]</td>\n",
       "      <td>0.05 [0;0]</td>\n",
       "      <td>86603.0 [84458;88071]</td>\n",
       "      <td>85430.67 [83502;87280]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>6700972.44 [6694930;6710104]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>170.0 [161;185]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.91 [0;0]</td>\n",
       "      <td>1.0 [0;2]</td>\n",
       "      <td>1.67 [0;4]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>27.67 [24;30]</td>\n",
       "      <td>0.07 [0;0]</td>\n",
       "      <td>260458.33 [244431;286027]</td>\n",
       "      <td>263370.33 [244431;294763]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>6733229.67 [6730124;6737986]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>80.67 [78;82]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.95 [0;0]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>0.67 [0;2]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>26.33 [0;42]</td>\n",
       "      <td>0.05 [0;0]</td>\n",
       "      <td>133463.67 [1126;211554]</td>\n",
       "      <td>132907.0 [0;211015]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>4508723.22 [92881;6717767]</td>\n",
       "      <td>0.66 [0;0]</td>\n",
       "      <td>94.0 [80;101]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>30.67 [24;44]</td>\n",
       "      <td>0.1 [0;0]</td>\n",
       "      <td>222023.33 [151807;257869]</td>\n",
       "      <td>222023.33 [151807;257869]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>6724940.0 [6721688;6726884]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>71.0 [56;100]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>360.11 [335;406]</td>\n",
       "      <td>0.01 [0;0]</td>\n",
       "      <td>20986.67 [18832;22207]</td>\n",
       "      <td>20277.0 [17958;21734]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>6508946.0 [6489816;6518607]</td>\n",
       "      <td>0.96 [0;0]</td>\n",
       "      <td>548.67 [523;598]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>300.67 [58;670]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>53428.0 [10114;111919]</td>\n",
       "      <td>51444.33 [9897;106185]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>6629598.67 [6545484;6673601]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>451.0 [117;933]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.47 [0;0]</td>\n",
       "      <td>4.33 [0;7]</td>\n",
       "      <td>9.0 [0;15]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>29.67 [26;37]</td>\n",
       "      <td>0.12 [0;0]</td>\n",
       "      <td>217837.0 [173647;252244]</td>\n",
       "      <td>217837.0 [173647;252244]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>6721289.0 [6716697;6728888]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>69.67 [63;79]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>0.67 [0;2]</td>\n",
       "      <td>1.0 [0;3]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>24.0 [22;25]</td>\n",
       "      <td>0.12 [0;0]</td>\n",
       "      <td>268387.0 [256472;283759]</td>\n",
       "      <td>262588.0 [239075;283759]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>6713939.33 [6704188;6723626]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>67.33 [60;77]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.95 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>9.67 [0;15]</td>\n",
       "      <td>0.06 [0;0]</td>\n",
       "      <td>201072.0 [0;338923]</td>\n",
       "      <td>267909.33 [0;464977]</td>\n",
       "      <td>6108.67 [0;12098]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>4470796.67 [0;6708956]</td>\n",
       "      <td>0.66 [0;0]</td>\n",
       "      <td>50.33 [0;82]</td>\n",
       "      <td>0.66 [0;0]</td>\n",
       "      <td>0.57 [0;0]</td>\n",
       "      <td>4.67 [0;8]</td>\n",
       "      <td>7.33 [0;11]</td>\n",
       "      <td>0.67 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;0]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm               L90  \\\n",
       "Assembler                                                             \n",
       "ABySS              Multiple k-mer De Bruijn graph  478.71 [67;1004]   \n",
       "BCALM2               Single k-mer De Bruijn graph    420.33 [0;921]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph     52.33 [39;78]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph      79.0 [71;90]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph     27.67 [24;30]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph      26.33 [0;42]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph     30.67 [24;44]   \n",
       "MINIA                Single k-mer De Bruijn graph  360.11 [335;406]   \n",
       "SKESA              Multiple k-mer De Bruijn graph   300.67 [58;670]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph     29.67 [26;37]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph      24.0 [22;25]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph       9.67 [0;15]   \n",
       "\n",
       "                          LSA                       NA50  \\\n",
       "Assembler                                                  \n",
       "ABySS              0.02 [0;0]      49503.43 [6462;92433]   \n",
       "BCALM2             0.01 [0;0]         10132.33 [0;21972]   \n",
       "GATBMiniaPipeline  0.06 [0;0]   165922.67 [88774;204691]   \n",
       "IDBA-UD            0.05 [0;0]      86603.0 [84458;88071]   \n",
       "MEGAHIT            0.07 [0;0]  260458.33 [244431;286027]   \n",
       "MetaHipMer2        0.05 [0;0]    133463.67 [1126;211554]   \n",
       "metaSPAdes          0.1 [0;0]  222023.33 [151807;257869]   \n",
       "MINIA              0.01 [0;0]     20986.67 [18832;22207]   \n",
       "SKESA              0.02 [0;0]     53428.0 [10114;111919]   \n",
       "SPAdes             0.12 [0;0]   217837.0 [173647;252244]   \n",
       "Unicycler          0.12 [0;0]   268387.0 [256472;283759]   \n",
       "VelvetOptimiser    0.06 [0;0]        201072.0 [0;338923]   \n",
       "\n",
       "                                        NG50                 Ns         Type  \\\n",
       "Assembler                                                                      \n",
       "ABySS                  50037.57 [6430;93704]    1057.0 [3;2170]  Metagenomic   \n",
       "BCALM2                     9640.33 [0;21139]          0.0 [0;0]      Genomic   \n",
       "GATBMiniaPipeline   165922.67 [88774;204691]          0.0 [0;0]  Metagenomic   \n",
       "IDBA-UD               85430.67 [83502;87280]          0.0 [0;0]  Metagenomic   \n",
       "MEGAHIT            263370.33 [244431;294763]          0.0 [0;0]  Metagenomic   \n",
       "MetaHipMer2              132907.0 [0;211015]          0.0 [0;0]  Metagenomic   \n",
       "metaSPAdes         222023.33 [151807;257869]          0.0 [0;0]  Metagenomic   \n",
       "MINIA                  20277.0 [17958;21734]          0.0 [0;0]      Genomic   \n",
       "SKESA                 51444.33 [9897;106185]          0.0 [0;0]      Genomic   \n",
       "SPAdes              217837.0 [173647;252244]          0.0 [0;0]      Genomic   \n",
       "Unicycler           262588.0 [239075;283759]          0.0 [0;0]      Genomic   \n",
       "VelvetOptimiser         267909.33 [0;464977]  6108.67 [0;12098]  Metagenomic   \n",
       "\n",
       "                                      basepairs breadth_of_coverage  \\\n",
       "Assembler                                                             \n",
       "ABySS               6674637.0 [6555420;6780808]          0.98 [0;0]   \n",
       "BCALM2                    4303694.0 [0;6510588]          0.63 [0;0]   \n",
       "GATBMiniaPipeline  6708469.67 [6703986;6716518]          0.99 [0;0]   \n",
       "IDBA-UD            6700972.44 [6694930;6710104]          0.99 [0;0]   \n",
       "MEGAHIT            6733229.67 [6730124;6737986]          0.99 [0;0]   \n",
       "MetaHipMer2          4508723.22 [92881;6717767]          0.66 [0;0]   \n",
       "metaSPAdes          6724940.0 [6721688;6726884]          0.99 [0;0]   \n",
       "MINIA               6508946.0 [6489816;6518607]          0.96 [0;0]   \n",
       "SKESA              6629598.67 [6545484;6673601]          0.98 [0;0]   \n",
       "SPAdes              6721289.0 [6716697;6728888]          0.99 [0;0]   \n",
       "Unicycler          6713939.33 [6704188;6723626]          0.99 [0;0]   \n",
       "VelvetOptimiser          4470796.67 [0;6708956]          0.66 [0;0]   \n",
       "\n",
       "                             contigs    identity lowest_identity  \\\n",
       "Assembler                                                          \n",
       "ABySS              701.29 [143;1404]   1.0 [0;0]      0.81 [0;0]   \n",
       "BCALM2                555.0 [0;1142]  0.67 [0;1]      0.67 [0;1]   \n",
       "GATBMiniaPipeline     122.0 [98;166]   1.0 [0;0]      0.96 [0;0]   \n",
       "IDBA-UD              170.0 [161;185]   1.0 [0;0]      0.91 [0;0]   \n",
       "MEGAHIT                80.67 [78;82]   1.0 [0;0]      0.95 [0;0]   \n",
       "MetaHipMer2            94.0 [80;101]   1.0 [0;1]       1.0 [0;1]   \n",
       "metaSPAdes             71.0 [56;100]   1.0 [0;0]      0.98 [0;0]   \n",
       "MINIA               548.67 [523;598]   1.0 [0;0]      0.99 [0;0]   \n",
       "SKESA                451.0 [117;933]   1.0 [0;0]      0.47 [0;0]   \n",
       "SPAdes                 69.67 [63;79]   1.0 [0;0]      0.98 [0;0]   \n",
       "Unicycler              67.33 [60;77]   1.0 [0;0]      0.95 [0;0]   \n",
       "VelvetOptimiser         50.33 [0;82]  0.66 [0;0]      0.57 [0;0]   \n",
       "\n",
       "                  misassembled contigs misassembly events multiplicity  \\\n",
       "Assembler                                                                \n",
       "ABySS                       4.14 [0;8]         7.0 [0;14]   1.02 [1;1]   \n",
       "BCALM2                       0.0 [0;0]          0.0 [0;0]   0.67 [0;1]   \n",
       "GATBMiniaPipeline            1.0 [0;3]          2.0 [0;6]    1.0 [1;1]   \n",
       "IDBA-UD                      1.0 [0;2]         1.67 [0;4]    1.0 [1;1]   \n",
       "MEGAHIT                     0.33 [0;1]         0.67 [0;2]    1.0 [1;1]   \n",
       "MetaHipMer2                  0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "metaSPAdes                   0.0 [0;0]          0.0 [0;0]   1.01 [1;1]   \n",
       "MINIA                        0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "SKESA                       4.33 [0;7]         9.0 [0;15]    1.0 [0;1]   \n",
       "SPAdes                      0.67 [0;2]          1.0 [0;3]    1.0 [0;1]   \n",
       "Unicycler                    0.0 [0;0]          0.0 [0;0]    1.0 [0;1]   \n",
       "VelvetOptimiser             4.67 [0;8]        7.33 [0;11]   0.67 [0;0]   \n",
       "\n",
       "                    parsimony    validity  \n",
       "Assembler                                  \n",
       "ABySS              1.02 [1;1]   1.0 [0;0]  \n",
       "BCALM2             0.67 [0;1]  0.67 [0;1]  \n",
       "GATBMiniaPipeline   1.0 [1;1]   1.0 [0;0]  \n",
       "IDBA-UD             1.0 [1;1]   1.0 [0;0]  \n",
       "MEGAHIT             1.0 [1;1]   1.0 [0;1]  \n",
       "MetaHipMer2         1.0 [1;1]   1.0 [0;1]  \n",
       "metaSPAdes         1.01 [1;1]   1.0 [0;1]  \n",
       "MINIA               1.0 [1;1]   1.0 [0;1]  \n",
       "SKESA               1.0 [1;1]   1.0 [0;0]  \n",
       "SPAdes              1.0 [0;1]   1.0 [0;0]  \n",
       "Unicycler           1.0 [0;1]   1.0 [0;1]  \n",
       "VelvetOptimiser    0.67 [0;1]  0.67 [0;0]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salmonella_enterica\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>48.43 [33;116]</td>\n",
       "      <td>0.06 [0;0]</td>\n",
       "      <td>124063.86 [44050;157988]</td>\n",
       "      <td>123763.29 [44050;157988]</td>\n",
       "      <td>1.57 [1;2]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>4695624.71 [4657217;4703394]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>92.86 [68;214]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.93 [0;0]</td>\n",
       "      <td>0.14 [0;1]</td>\n",
       "      <td>0.57 [0;4]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>2442.0 [0;5009]</td>\n",
       "      <td>1683.33 [0;3551]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2305099.11 [0;3817265]</td>\n",
       "      <td>0.48 [0;0]</td>\n",
       "      <td>852.67 [0;1479]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>87.67 [82;96]</td>\n",
       "      <td>0.06 [0;0]</td>\n",
       "      <td>58851.67 [52959;64572]</td>\n",
       "      <td>56961.67 [51487;60374]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>4646702.33 [4643641;4652686]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>167.33 [158;178]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.86 [0;0]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>182.33 [171;200]</td>\n",
       "      <td>0.03 [0;0]</td>\n",
       "      <td>27175.33 [25757;29331]</td>\n",
       "      <td>27460.33 [26117;28547]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>4583582.89 [4550744;4618887]</td>\n",
       "      <td>0.96 [0;0]</td>\n",
       "      <td>362.44 [305;397]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.63 [0;0]</td>\n",
       "      <td>5.78 [1;9]</td>\n",
       "      <td>9.22 [1;14]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>44.0 [42;47]</td>\n",
       "      <td>0.09 [0;0]</td>\n",
       "      <td>109331.0 [89627;120785]</td>\n",
       "      <td>110872.33 [94251;120785]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>4669897.67 [4668709;4671620]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>103.33 [102;105]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.95 [0;0]</td>\n",
       "      <td>1.33 [0;3]</td>\n",
       "      <td>2.67 [0;6]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>57.33 [0;86]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>41997.22 [1125;64529]</td>\n",
       "      <td>39762.33 [0;60329]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3129016.11 [103085;4641887]</td>\n",
       "      <td>0.66 [0;0]</td>\n",
       "      <td>136.67 [86;162]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.97 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>43.33 [25;77]</td>\n",
       "      <td>0.08 [0;0]</td>\n",
       "      <td>144490.0 [70327;198784]</td>\n",
       "      <td>142339.67 [70327;198784]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>4669834.33 [4654552;4680545]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>85.33 [55;145]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.96 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.01 [0;0]</td>\n",
       "      <td>4960.44 [4884;4996]</td>\n",
       "      <td>3485.67 [3355;3551]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>3818024.22 [3801964;3827013]</td>\n",
       "      <td>0.8 [0;0]</td>\n",
       "      <td>1090.56 [1083;1106]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.93 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>112.0 [83;144]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>48627.0 [37980;60912]</td>\n",
       "      <td>48105.67 [37661;60131]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>4585087.67 [4566475;4612070]</td>\n",
       "      <td>0.96 [0;0]</td>\n",
       "      <td>198.33 [157;245]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.62 [0;0]</td>\n",
       "      <td>1.33 [0;2]</td>\n",
       "      <td>1.67 [0;3]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>41.0 [27;50]</td>\n",
       "      <td>0.09 [0;0]</td>\n",
       "      <td>124878.33 [103636;164404]</td>\n",
       "      <td>122550.33 [103102;157954]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>4643886.67 [4614007;4666050]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>87.0 [60;110]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.83 [0;0]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>29.0 [20;35]</td>\n",
       "      <td>0.09 [0;0]</td>\n",
       "      <td>179854.0 [157800;223960]</td>\n",
       "      <td>178933.33 [155040;223960]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>4648667.33 [4637757;4660116]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>69.67 [50;89]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.01 [0;0]</td>\n",
       "      <td>3450.67 [0;5223]</td>\n",
       "      <td>2496.67 [0;3753]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2576489.67 [0;3867436]</td>\n",
       "      <td>0.54 [0;0]</td>\n",
       "      <td>702.33 [0;1058]</td>\n",
       "      <td>0.67 [0;0]</td>\n",
       "      <td>0.66 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm               L90  \\\n",
       "Assembler                                                             \n",
       "ABySS              Multiple k-mer De Bruijn graph    48.43 [33;116]   \n",
       "BCALM2               Single k-mer De Bruijn graph         0.0 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph     87.67 [82;96]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  182.33 [171;200]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph      44.0 [42;47]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph      57.33 [0;86]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph     43.33 [25;77]   \n",
       "MINIA                Single k-mer De Bruijn graph         0.0 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph    112.0 [83;144]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph      41.0 [27;50]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph      29.0 [20;35]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph         0.0 [0;0]   \n",
       "\n",
       "                          LSA                       NA50  \\\n",
       "Assembler                                                  \n",
       "ABySS              0.06 [0;0]   124063.86 [44050;157988]   \n",
       "BCALM2              0.0 [0;0]            2442.0 [0;5009]   \n",
       "GATBMiniaPipeline  0.06 [0;0]     58851.67 [52959;64572]   \n",
       "IDBA-UD            0.03 [0;0]     27175.33 [25757;29331]   \n",
       "MEGAHIT            0.09 [0;0]    109331.0 [89627;120785]   \n",
       "MetaHipMer2        0.04 [0;0]      41997.22 [1125;64529]   \n",
       "metaSPAdes         0.08 [0;0]    144490.0 [70327;198784]   \n",
       "MINIA              0.01 [0;0]        4960.44 [4884;4996]   \n",
       "SKESA              0.04 [0;0]      48627.0 [37980;60912]   \n",
       "SPAdes             0.09 [0;0]  124878.33 [103636;164404]   \n",
       "Unicycler          0.09 [0;0]   179854.0 [157800;223960]   \n",
       "VelvetOptimiser    0.01 [0;0]           3450.67 [0;5223]   \n",
       "\n",
       "                                        NG50          Ns         Type  \\\n",
       "Assembler                                                               \n",
       "ABySS               123763.29 [44050;157988]  1.57 [1;2]  Metagenomic   \n",
       "BCALM2                      1683.33 [0;3551]   0.0 [0;0]      Genomic   \n",
       "GATBMiniaPipeline     56961.67 [51487;60374]   0.0 [0;0]  Metagenomic   \n",
       "IDBA-UD               27460.33 [26117;28547]   0.0 [0;0]  Metagenomic   \n",
       "MEGAHIT             110872.33 [94251;120785]   0.0 [0;0]  Metagenomic   \n",
       "MetaHipMer2               39762.33 [0;60329]   0.0 [0;0]  Metagenomic   \n",
       "metaSPAdes          142339.67 [70327;198784]   0.0 [0;0]  Metagenomic   \n",
       "MINIA                    3485.67 [3355;3551]   0.0 [0;0]      Genomic   \n",
       "SKESA                 48105.67 [37661;60131]   0.0 [0;0]      Genomic   \n",
       "SPAdes             122550.33 [103102;157954]   0.0 [0;0]      Genomic   \n",
       "Unicycler          178933.33 [155040;223960]   0.0 [0;0]      Genomic   \n",
       "VelvetOptimiser             2496.67 [0;3753]   0.0 [0;0]  Metagenomic   \n",
       "\n",
       "                                      basepairs breadth_of_coverage  \\\n",
       "Assembler                                                             \n",
       "ABySS              4695624.71 [4657217;4703394]          0.99 [0;0]   \n",
       "BCALM2                   2305099.11 [0;3817265]          0.48 [0;0]   \n",
       "GATBMiniaPipeline  4646702.33 [4643641;4652686]          0.98 [0;0]   \n",
       "IDBA-UD            4583582.89 [4550744;4618887]          0.96 [0;0]   \n",
       "MEGAHIT            4669897.67 [4668709;4671620]          0.98 [0;0]   \n",
       "MetaHipMer2         3129016.11 [103085;4641887]          0.66 [0;0]   \n",
       "metaSPAdes         4669834.33 [4654552;4680545]          0.98 [0;0]   \n",
       "MINIA              3818024.22 [3801964;3827013]           0.8 [0;0]   \n",
       "SKESA              4585087.67 [4566475;4612070]          0.96 [0;0]   \n",
       "SPAdes             4643886.67 [4614007;4666050]          0.98 [0;0]   \n",
       "Unicycler          4648667.33 [4637757;4660116]          0.98 [0;0]   \n",
       "VelvetOptimiser          2576489.67 [0;3867436]          0.54 [0;0]   \n",
       "\n",
       "                               contigs    identity lowest_identity  \\\n",
       "Assembler                                                            \n",
       "ABySS                   92.86 [68;214]   1.0 [0;0]      0.93 [0;0]   \n",
       "BCALM2                 852.67 [0;1479]  0.67 [0;1]      0.67 [0;1]   \n",
       "GATBMiniaPipeline     167.33 [158;178]   1.0 [0;0]      0.86 [0;0]   \n",
       "IDBA-UD               362.44 [305;397]  0.99 [0;0]      0.63 [0;0]   \n",
       "MEGAHIT               103.33 [102;105]   1.0 [0;0]      0.95 [0;0]   \n",
       "MetaHipMer2            136.67 [86;162]   1.0 [0;1]      0.97 [0;1]   \n",
       "metaSPAdes              85.33 [55;145]   1.0 [0;0]      0.96 [0;0]   \n",
       "MINIA              1090.56 [1083;1106]   1.0 [0;0]      0.93 [0;0]   \n",
       "SKESA                 198.33 [157;245]   1.0 [0;0]      0.62 [0;0]   \n",
       "SPAdes                   87.0 [60;110]   1.0 [0;0]      0.83 [0;0]   \n",
       "Unicycler                69.67 [50;89]   1.0 [0;0]      0.99 [0;0]   \n",
       "VelvetOptimiser        702.33 [0;1058]  0.67 [0;0]      0.66 [0;0]   \n",
       "\n",
       "                  misassembled contigs misassembly events multiplicity  \\\n",
       "Assembler                                                                \n",
       "ABySS                       0.14 [0;1]         0.57 [0;4]   1.01 [1;1]   \n",
       "BCALM2                       0.0 [0;0]          0.0 [0;0]   0.67 [0;1]   \n",
       "GATBMiniaPipeline           0.33 [0;1]         0.33 [0;1]    1.0 [1;1]   \n",
       "IDBA-UD                     5.78 [1;9]        9.22 [1;14]   1.01 [1;1]   \n",
       "MEGAHIT                     1.33 [0;3]         2.67 [0;6]   1.01 [1;1]   \n",
       "MetaHipMer2                  0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "metaSPAdes                   0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "MINIA                        0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "SKESA                       1.33 [0;2]         1.67 [0;3]    1.0 [1;1]   \n",
       "SPAdes                      0.33 [0;1]         0.33 [0;1]    1.0 [1;1]   \n",
       "Unicycler                    0.0 [0;0]          0.0 [0;0]    1.0 [0;1]   \n",
       "VelvetOptimiser              0.0 [0;0]          0.0 [0;0]   0.67 [0;1]   \n",
       "\n",
       "                    parsimony    validity  \n",
       "Assembler                                  \n",
       "ABySS              1.01 [1;1]   1.0 [0;0]  \n",
       "BCALM2             0.67 [0;1]  0.67 [0;1]  \n",
       "GATBMiniaPipeline   1.0 [1;1]   1.0 [0;1]  \n",
       "IDBA-UD            1.01 [1;1]  0.99 [0;0]  \n",
       "MEGAHIT            1.01 [1;1]   1.0 [0;0]  \n",
       "MetaHipMer2         1.0 [1;1]   1.0 [0;1]  \n",
       "metaSPAdes          1.0 [1;1]   1.0 [0;1]  \n",
       "MINIA               1.0 [1;1]   1.0 [0;1]  \n",
       "SKESA              1.01 [1;1]  0.99 [0;0]  \n",
       "SPAdes              1.0 [1;1]   1.0 [0;0]  \n",
       "Unicycler           1.0 [1;1]   1.0 [0;1]  \n",
       "VelvetOptimiser    0.67 [0;1]  0.67 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Staphylococcus_aureus\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>17.43 [10;47]</td>\n",
       "      <td>0.29 [0;0]</td>\n",
       "      <td>243320.43 [53240;281333]</td>\n",
       "      <td>243320.43 [53240;281333]</td>\n",
       "      <td>30.43 [0;71]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2689869.86 [2681667;2693907]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>31.71 [20;78]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.14 [0;1]</td>\n",
       "      <td>0.29 [0;2]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>37.33 [0;112]</td>\n",
       "      <td>0.01 [0;0]</td>\n",
       "      <td>10270.33 [0;29647]</td>\n",
       "      <td>9721.67 [0;29165]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>910513.67 [0;2600423]</td>\n",
       "      <td>0.33 [0;0]</td>\n",
       "      <td>94.67 [0;175]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>32.0 [24;39]</td>\n",
       "      <td>0.11 [0;0]</td>\n",
       "      <td>89183.33 [63844;112739]</td>\n",
       "      <td>84190.0 [53404;108199]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2675383.11 [2670675;2679055]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>53.0 [41;65]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>65.67 [42;112]</td>\n",
       "      <td>0.05 [0;0]</td>\n",
       "      <td>44344.33 [26537;53248]</td>\n",
       "      <td>43851.67 [26420;53248]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2663984.67 [2661504;2665812]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>106.67 [72;174]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.84 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>16.0 [14;19]</td>\n",
       "      <td>0.15 [0;0]</td>\n",
       "      <td>140497.67 [112741;181811]</td>\n",
       "      <td>163799.33 [126941;213210]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2681087.33 [2680839;2681291]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>40.0 [37;43]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.85 [0;0]</td>\n",
       "      <td>2.33 [1;3]</td>\n",
       "      <td>5.33 [3;7]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>27.78 [0;42]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>35927.33 [1220;53350]</td>\n",
       "      <td>35055.22 [0;53350]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1879025.89 [298938;2667635]</td>\n",
       "      <td>0.69 [0;0]</td>\n",
       "      <td>126.78 [67;249]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>17.0 [8;30]</td>\n",
       "      <td>0.21 [0;0]</td>\n",
       "      <td>213807.33 [76042;360908]</td>\n",
       "      <td>213807.33 [76042;360908]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2679337.0 [2673904;2682662]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>30.33 [19;49]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>121.33 [112;140]</td>\n",
       "      <td>0.03 [0;0]</td>\n",
       "      <td>27575.67 [23433;29647]</td>\n",
       "      <td>27105.33 [22986;29165]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2599124.11 [2596281;2600576]</td>\n",
       "      <td>0.96 [0;0]</td>\n",
       "      <td>185.67 [175;207]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>41.0 [38;43]</td>\n",
       "      <td>0.07 [0;0]</td>\n",
       "      <td>59782.33 [53048;73147]</td>\n",
       "      <td>56521.67 [53048;63469]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2656505.67 [2652026;2661968]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>68.33 [63;73]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>18.33 [11;30]</td>\n",
       "      <td>0.17 [0;0]</td>\n",
       "      <td>189665.0 [73613;280950]</td>\n",
       "      <td>189665.0 [73613;280950]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2674000.67 [2670466;2678585]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>31.0 [21;48]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>1.0 [0;3]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>9.0 [5;14]</td>\n",
       "      <td>0.37 [0;0]</td>\n",
       "      <td>329399.67 [223263;394879]</td>\n",
       "      <td>309650.67 [164016;394879]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2673590.67 [2670223;2678241]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>21.67 [16;25]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>0.67 [0;2]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>72.33 [0;113]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>20737.33 [0;32297]</td>\n",
       "      <td>19180.33 [0;30086]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1735255.67 [0;2603433]</td>\n",
       "      <td>0.64 [0;0]</td>\n",
       "      <td>113.0 [0;175]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm               L90  \\\n",
       "Assembler                                                             \n",
       "ABySS              Multiple k-mer De Bruijn graph     17.43 [10;47]   \n",
       "BCALM2               Single k-mer De Bruijn graph     37.33 [0;112]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph      32.0 [24;39]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph    65.67 [42;112]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph      16.0 [14;19]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph      27.78 [0;42]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph       17.0 [8;30]   \n",
       "MINIA                Single k-mer De Bruijn graph  121.33 [112;140]   \n",
       "SKESA              Multiple k-mer De Bruijn graph      41.0 [38;43]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph     18.33 [11;30]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph        9.0 [5;14]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph     72.33 [0;113]   \n",
       "\n",
       "                          LSA                       NA50  \\\n",
       "Assembler                                                  \n",
       "ABySS              0.29 [0;0]   243320.43 [53240;281333]   \n",
       "BCALM2             0.01 [0;0]         10270.33 [0;29647]   \n",
       "GATBMiniaPipeline  0.11 [0;0]    89183.33 [63844;112739]   \n",
       "IDBA-UD            0.05 [0;0]     44344.33 [26537;53248]   \n",
       "MEGAHIT            0.15 [0;0]  140497.67 [112741;181811]   \n",
       "MetaHipMer2        0.04 [0;0]      35927.33 [1220;53350]   \n",
       "metaSPAdes         0.21 [0;0]   213807.33 [76042;360908]   \n",
       "MINIA              0.03 [0;0]     27575.67 [23433;29647]   \n",
       "SKESA              0.07 [0;0]     59782.33 [53048;73147]   \n",
       "SPAdes             0.17 [0;0]    189665.0 [73613;280950]   \n",
       "Unicycler          0.37 [0;0]  329399.67 [223263;394879]   \n",
       "VelvetOptimiser    0.02 [0;0]         20737.33 [0;32297]   \n",
       "\n",
       "                                        NG50            Ns         Type  \\\n",
       "Assembler                                                                 \n",
       "ABySS               243320.43 [53240;281333]  30.43 [0;71]  Metagenomic   \n",
       "BCALM2                     9721.67 [0;29165]     0.0 [0;0]      Genomic   \n",
       "GATBMiniaPipeline     84190.0 [53404;108199]     0.0 [0;0]  Metagenomic   \n",
       "IDBA-UD               43851.67 [26420;53248]     0.0 [0;0]  Metagenomic   \n",
       "MEGAHIT            163799.33 [126941;213210]     0.0 [0;0]  Metagenomic   \n",
       "MetaHipMer2               35055.22 [0;53350]     0.0 [0;0]  Metagenomic   \n",
       "metaSPAdes          213807.33 [76042;360908]     0.0 [0;0]  Metagenomic   \n",
       "MINIA                 27105.33 [22986;29165]     0.0 [0;0]      Genomic   \n",
       "SKESA                 56521.67 [53048;63469]     0.0 [0;0]      Genomic   \n",
       "SPAdes               189665.0 [73613;280950]     0.0 [0;0]      Genomic   \n",
       "Unicycler          309650.67 [164016;394879]     0.0 [0;0]      Genomic   \n",
       "VelvetOptimiser           19180.33 [0;30086]     0.0 [0;0]  Metagenomic   \n",
       "\n",
       "                                      basepairs breadth_of_coverage  \\\n",
       "Assembler                                                             \n",
       "ABySS              2689869.86 [2681667;2693907]          0.99 [0;0]   \n",
       "BCALM2                    910513.67 [0;2600423]          0.33 [0;0]   \n",
       "GATBMiniaPipeline  2675383.11 [2670675;2679055]          0.98 [0;0]   \n",
       "IDBA-UD            2663984.67 [2661504;2665812]          0.98 [0;0]   \n",
       "MEGAHIT            2681087.33 [2680839;2681291]          0.99 [0;0]   \n",
       "MetaHipMer2         1879025.89 [298938;2667635]          0.69 [0;0]   \n",
       "metaSPAdes          2679337.0 [2673904;2682662]          0.99 [0;0]   \n",
       "MINIA              2599124.11 [2596281;2600576]          0.96 [0;0]   \n",
       "SKESA              2656505.67 [2652026;2661968]          0.98 [0;0]   \n",
       "SPAdes             2674000.67 [2670466;2678585]          0.98 [0;0]   \n",
       "Unicycler          2673590.67 [2670223;2678241]          0.98 [0;0]   \n",
       "VelvetOptimiser          1735255.67 [0;2603433]          0.64 [0;0]   \n",
       "\n",
       "                            contigs    identity lowest_identity  \\\n",
       "Assembler                                                         \n",
       "ABySS                 31.71 [20;78]   1.0 [0;0]       1.0 [0;0]   \n",
       "BCALM2                94.67 [0;175]  0.67 [0;1]      0.67 [0;1]   \n",
       "GATBMiniaPipeline      53.0 [41;65]   1.0 [0;1]       1.0 [0;1]   \n",
       "IDBA-UD             106.67 [72;174]   1.0 [0;0]      0.84 [0;0]   \n",
       "MEGAHIT                40.0 [37;43]   1.0 [0;0]      0.85 [0;0]   \n",
       "MetaHipMer2         126.78 [67;249]   1.0 [0;1]       1.0 [0;1]   \n",
       "metaSPAdes            30.33 [19;49]   1.0 [0;0]      0.99 [0;0]   \n",
       "MINIA              185.67 [175;207]   1.0 [0;0]       1.0 [0;0]   \n",
       "SKESA                 68.33 [63;73]   1.0 [0;1]       1.0 [0;1]   \n",
       "SPAdes                 31.0 [21;48]   1.0 [0;0]      0.98 [0;0]   \n",
       "Unicycler             21.67 [16;25]   1.0 [0;0]       1.0 [0;0]   \n",
       "VelvetOptimiser       113.0 [0;175]  0.67 [0;1]      0.67 [0;1]   \n",
       "\n",
       "                  misassembled contigs misassembly events multiplicity  \\\n",
       "Assembler                                                                \n",
       "ABySS                       0.14 [0;1]         0.29 [0;2]   1.01 [1;1]   \n",
       "BCALM2                       0.0 [0;0]          0.0 [0;0]   0.67 [0;1]   \n",
       "GATBMiniaPipeline            0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "IDBA-UD                      0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "MEGAHIT                     2.33 [1;3]         5.33 [3;7]    1.0 [0;1]   \n",
       "MetaHipMer2                  0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "metaSPAdes                   0.0 [0;0]          0.0 [0;0]    1.0 [0;1]   \n",
       "MINIA                        0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "SKESA                        0.0 [0;0]          0.0 [0;0]    1.0 [0;1]   \n",
       "SPAdes                      0.33 [0;1]          1.0 [0;3]    1.0 [0;1]   \n",
       "Unicycler                   0.33 [0;1]         0.67 [0;2]    1.0 [0;0]   \n",
       "VelvetOptimiser              0.0 [0;0]          0.0 [0;0]   0.67 [0;1]   \n",
       "\n",
       "                    parsimony    validity  \n",
       "Assembler                                  \n",
       "ABySS              1.01 [1;1]   1.0 [0;1]  \n",
       "BCALM2             0.67 [0;1]  0.67 [0;1]  \n",
       "GATBMiniaPipeline   1.0 [1;1]   1.0 [1;1]  \n",
       "IDBA-UD             1.0 [1;1]   1.0 [0;1]  \n",
       "MEGAHIT             1.0 [1;1]   1.0 [0;0]  \n",
       "MetaHipMer2         1.0 [1;1]   1.0 [0;1]  \n",
       "metaSPAdes          1.0 [0;1]   1.0 [0;1]  \n",
       "MINIA               1.0 [1;1]   1.0 [0;0]  \n",
       "SKESA               1.0 [0;1]   1.0 [1;1]  \n",
       "SPAdes              1.0 [0;1]   1.0 [0;1]  \n",
       "Unicycler           1.0 [0;0]   1.0 [1;1]  \n",
       "VelvetOptimiser    0.67 [0;1]  0.67 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Staphylococcus_aureus_plasmid1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.86 [0;0]</td>\n",
       "      <td>5429.57 [0;6335]</td>\n",
       "      <td>5429.57 [0;6335]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>5429.57 [0;6335]</td>\n",
       "      <td>0.86 [0;0]</td>\n",
       "      <td>0.86 [0;1]</td>\n",
       "      <td>0.86 [0;1]</td>\n",
       "      <td>0.86 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.86 [0;1]</td>\n",
       "      <td>0.86 [0;1]</td>\n",
       "      <td>0.86 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.33 [0;0]</td>\n",
       "      <td>2112.0 [0;6336]</td>\n",
       "      <td>2112.0 [0;6336]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2112.0 [0;6336]</td>\n",
       "      <td>0.33 [0;0]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.01 [0;1]</td>\n",
       "      <td>6394.67 [6334;6515]</td>\n",
       "      <td>6394.67 [6334;6515]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>6335.67 [6334;6337]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.92 [0;0]</td>\n",
       "      <td>5812.33 [4793;6336]</td>\n",
       "      <td>5812.33 [4793;6336]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>5802.33 [4763;6336]</td>\n",
       "      <td>0.92 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>6479.67 [6308;6795]</td>\n",
       "      <td>6479.67 [6308;6795]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>6327.0 [6308;6337]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.02 [1;1]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.67 [0;0]</td>\n",
       "      <td>4222.67 [0;6334]</td>\n",
       "      <td>4222.67 [0;6334]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>4222.67 [0;6334]</td>\n",
       "      <td>0.67 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>0.89 [0;0]</td>\n",
       "      <td>5636.33 [4237;6336]</td>\n",
       "      <td>5636.33 [4237;6336]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>6292.0 [6204;6336]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>1.33 [1;2]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.67 [0;0]</td>\n",
       "      <td>4214.33 [0;6336]</td>\n",
       "      <td>4214.33 [0;6336]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>4214.33 [0;6336]</td>\n",
       "      <td>0.67 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.72 [0;0]</td>\n",
       "      <td>4561.0 [1672;6044]</td>\n",
       "      <td>4561.0 [1672;6044]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>5624.67 [4863;6044]</td>\n",
       "      <td>0.89 [0;0]</td>\n",
       "      <td>1.67 [1;3]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.67 [0;0]</td>\n",
       "      <td>4224.0 [0;6336]</td>\n",
       "      <td>4224.0 [0;6336]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>4224.0 [0;6336]</td>\n",
       "      <td>0.67 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>6280.33 [6230;6367]</td>\n",
       "      <td>6280.33 [6230;6367]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>6270.33 [6230;6337]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.67 [0;0]</td>\n",
       "      <td>4224.0 [0;6336]</td>\n",
       "      <td>4224.0 [0;6336]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>4224.0 [0;6336]</td>\n",
       "      <td>0.67 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm         L90         LSA  \\\n",
       "Assembler                                                                   \n",
       "ABySS              Multiple k-mer De Bruijn graph   0.0 [0;0]  0.86 [0;0]   \n",
       "BCALM2               Single k-mer De Bruijn graph   0.0 [0;0]  0.33 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph   0.0 [0;0]  1.01 [0;1]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph   0.0 [0;0]  0.92 [0;0]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph   0.0 [0;0]   1.0 [0;1]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph   0.0 [0;0]  0.67 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph  0.33 [0;1]  0.89 [0;0]   \n",
       "MINIA                Single k-mer De Bruijn graph   0.0 [0;0]  0.67 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph   0.0 [0;0]  0.72 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph   0.0 [0;0]  0.67 [0;0]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph   0.0 [0;0]  0.99 [0;1]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph   0.0 [0;0]  0.67 [0;0]   \n",
       "\n",
       "                                  NA50                 NG50         Ns  \\\n",
       "Assembler                                                                \n",
       "ABySS                 5429.57 [0;6335]     5429.57 [0;6335]  0.0 [0;0]   \n",
       "BCALM2                 2112.0 [0;6336]      2112.0 [0;6336]  0.0 [0;0]   \n",
       "GATBMiniaPipeline  6394.67 [6334;6515]  6394.67 [6334;6515]  0.0 [0;0]   \n",
       "IDBA-UD            5812.33 [4793;6336]  5812.33 [4793;6336]  0.0 [0;0]   \n",
       "MEGAHIT            6479.67 [6308;6795]  6479.67 [6308;6795]  0.0 [0;0]   \n",
       "MetaHipMer2           4222.67 [0;6334]     4222.67 [0;6334]  0.0 [0;0]   \n",
       "metaSPAdes         5636.33 [4237;6336]  5636.33 [4237;6336]  0.0 [0;0]   \n",
       "MINIA                 4214.33 [0;6336]     4214.33 [0;6336]  0.0 [0;0]   \n",
       "SKESA               4561.0 [1672;6044]   4561.0 [1672;6044]  0.0 [0;0]   \n",
       "SPAdes                 4224.0 [0;6336]      4224.0 [0;6336]  0.0 [0;0]   \n",
       "Unicycler          6280.33 [6230;6367]  6280.33 [6230;6367]  0.0 [0;0]   \n",
       "VelvetOptimiser        4224.0 [0;6336]      4224.0 [0;6336]  0.0 [0;0]   \n",
       "\n",
       "                          Type            basepairs breadth_of_coverage  \\\n",
       "Assembler                                                                 \n",
       "ABySS              Metagenomic     5429.57 [0;6335]          0.86 [0;0]   \n",
       "BCALM2                 Genomic      2112.0 [0;6336]          0.33 [0;0]   \n",
       "GATBMiniaPipeline  Metagenomic  6335.67 [6334;6337]           1.0 [0;1]   \n",
       "IDBA-UD            Metagenomic  5802.33 [4763;6336]          0.92 [0;0]   \n",
       "MEGAHIT            Metagenomic   6327.0 [6308;6337]           1.0 [0;1]   \n",
       "MetaHipMer2        Metagenomic     4222.67 [0;6334]          0.67 [0;0]   \n",
       "metaSPAdes         Metagenomic   6292.0 [6204;6336]          0.99 [0;0]   \n",
       "MINIA                  Genomic     4214.33 [0;6336]          0.67 [0;0]   \n",
       "SKESA                  Genomic  5624.67 [4863;6044]          0.89 [0;0]   \n",
       "SPAdes                 Genomic      4224.0 [0;6336]          0.67 [0;0]   \n",
       "Unicycler              Genomic  6270.33 [6230;6337]          0.99 [0;1]   \n",
       "VelvetOptimiser    Metagenomic      4224.0 [0;6336]          0.67 [0;0]   \n",
       "\n",
       "                      contigs    identity lowest_identity  \\\n",
       "Assembler                                                   \n",
       "ABySS              0.86 [0;1]  0.86 [0;1]      0.86 [0;1]   \n",
       "BCALM2             0.33 [0;1]  0.33 [0;1]      0.33 [0;1]   \n",
       "GATBMiniaPipeline   1.0 [1;1]   1.0 [0;1]       1.0 [0;1]   \n",
       "IDBA-UD             1.0 [1;1]   1.0 [0;1]       1.0 [0;1]   \n",
       "MEGAHIT             1.0 [1;1]  0.98 [0;1]      0.98 [0;1]   \n",
       "MetaHipMer2        0.67 [0;1]  0.67 [0;1]      0.67 [0;1]   \n",
       "metaSPAdes         1.33 [1;2]   1.0 [0;1]       1.0 [0;1]   \n",
       "MINIA              0.67 [0;1]  0.67 [0;1]      0.67 [0;1]   \n",
       "SKESA              1.67 [1;3]   1.0 [0;1]      0.99 [0;1]   \n",
       "SPAdes             0.67 [0;1]  0.67 [0;1]      0.67 [0;1]   \n",
       "Unicycler           1.0 [1;1]   1.0 [0;1]       1.0 [0;1]   \n",
       "VelvetOptimiser    0.67 [0;1]  0.67 [0;1]      0.67 [0;1]   \n",
       "\n",
       "                  misassembled contigs misassembly events multiplicity  \\\n",
       "Assembler                                                                \n",
       "ABySS                        0.0 [0;0]          0.0 [0;0]   0.86 [0;1]   \n",
       "BCALM2                       0.0 [0;0]          0.0 [0;0]   0.33 [0;1]   \n",
       "GATBMiniaPipeline            0.0 [0;0]          0.0 [0;0]   1.01 [1;1]   \n",
       "IDBA-UD                      0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "MEGAHIT                      0.0 [0;0]          0.0 [0;0]   1.01 [1;1]   \n",
       "MetaHipMer2                  0.0 [0;0]          0.0 [0;0]   0.67 [0;1]   \n",
       "metaSPAdes                   0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "MINIA                        0.0 [0;0]          0.0 [0;0]   0.67 [0;1]   \n",
       "SKESA                        0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "SPAdes                       0.0 [0;0]          0.0 [0;0]   0.67 [0;1]   \n",
       "Unicycler                    0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "VelvetOptimiser              0.0 [0;0]          0.0 [0;0]   0.67 [0;1]   \n",
       "\n",
       "                    parsimony    validity  \n",
       "Assembler                                  \n",
       "ABySS              0.86 [0;1]  0.86 [0;1]  \n",
       "BCALM2             0.33 [0;1]  0.33 [0;1]  \n",
       "GATBMiniaPipeline  1.01 [1;1]   1.0 [0;1]  \n",
       "IDBA-UD             1.0 [1;1]   1.0 [1;1]  \n",
       "MEGAHIT            1.02 [1;1]  0.98 [0;1]  \n",
       "MetaHipMer2        0.67 [0;1]  0.67 [0;1]  \n",
       "metaSPAdes          1.0 [1;1]   1.0 [1;1]  \n",
       "MINIA              0.67 [0;1]  0.67 [0;1]  \n",
       "SKESA               1.0 [1;1]   1.0 [0;1]  \n",
       "SPAdes             0.67 [0;1]  0.67 [0;1]  \n",
       "Unicycler           1.0 [1;1]   1.0 [1;1]  \n",
       "VelvetOptimiser    0.67 [0;1]  0.67 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Staphylococcus_aureus_plasmid2\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.95 [0;0]</td>\n",
       "      <td>2094.57 [1378;2215]</td>\n",
       "      <td>2094.57 [1378;2215]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2094.57 [1378;2215]</td>\n",
       "      <td>0.95 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.02 [0;1]</td>\n",
       "      <td>2255.11 [2176;2389]</td>\n",
       "      <td>2255.11 [2176;2389]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2202.33 [2176;2216]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.02 [1;1]</td>\n",
       "      <td>1.02 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.68 [0;1]</td>\n",
       "      <td>1510.0 [0;2315]</td>\n",
       "      <td>1510.0 [0;2315]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1477.0 [0;2216]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.68 [0;1]</td>\n",
       "      <td>0.68 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.02 [0;1]</td>\n",
       "      <td>2262.33 [2215;2357]</td>\n",
       "      <td>2262.33 [2215;2357]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2215.33 [2215;2216]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.02 [1;1]</td>\n",
       "      <td>1.02 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.67 [0;0]</td>\n",
       "      <td>1475.33 [0;2213]</td>\n",
       "      <td>1475.33 [0;2213]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1475.33 [0;2213]</td>\n",
       "      <td>0.67 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.97 [0;1]</td>\n",
       "      <td>2141.0 [2076;2271]</td>\n",
       "      <td>2141.0 [2076;2271]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2122.67 [2076;2216]</td>\n",
       "      <td>0.96 [0;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.81 [0;0]</td>\n",
       "      <td>1790.67 [1460;2005]</td>\n",
       "      <td>1790.67 [1460;2005]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1790.67 [1460;2005]</td>\n",
       "      <td>0.81 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "      <td>2262.0 [1579;3246]</td>\n",
       "      <td>2262.0 [1579;3246]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1918.67 [1579;2216]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>0.89 [0;1]</td>\n",
       "      <td>0.89 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.15 [1;1]</td>\n",
       "      <td>0.89 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.97 [0;1]</td>\n",
       "      <td>2149.33 [2109;2216]</td>\n",
       "      <td>2149.33 [2109;2216]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2149.33 [2109;2216]</td>\n",
       "      <td>0.97 [0;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm        L90         LSA  \\\n",
       "Assembler                                                                  \n",
       "ABySS              Multiple k-mer De Bruijn graph  0.0 [0;0]  0.95 [0;0]   \n",
       "BCALM2               Single k-mer De Bruijn graph  0.0 [0;0]   0.0 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph  0.0 [0;0]  1.02 [0;1]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  0.0 [0;0]  0.68 [0;1]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph  0.0 [0;0]  1.02 [0;1]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph  0.0 [0;0]  0.67 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph  0.0 [0;0]  0.97 [0;1]   \n",
       "MINIA                Single k-mer De Bruijn graph  0.0 [0;0]   0.0 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph  0.0 [0;0]  0.81 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph  0.0 [0;0]  0.87 [0;1]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph  0.0 [0;0]  0.97 [0;1]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph  0.0 [0;0]   0.0 [0;0]   \n",
       "\n",
       "                                  NA50                 NG50         Ns  \\\n",
       "Assembler                                                                \n",
       "ABySS              2094.57 [1378;2215]  2094.57 [1378;2215]  0.0 [0;0]   \n",
       "BCALM2                       0.0 [0;0]            0.0 [0;0]  0.0 [0;0]   \n",
       "GATBMiniaPipeline  2255.11 [2176;2389]  2255.11 [2176;2389]  0.0 [0;0]   \n",
       "IDBA-UD                1510.0 [0;2315]      1510.0 [0;2315]  0.0 [0;0]   \n",
       "MEGAHIT            2262.33 [2215;2357]  2262.33 [2215;2357]  0.0 [0;0]   \n",
       "MetaHipMer2           1475.33 [0;2213]     1475.33 [0;2213]  0.0 [0;0]   \n",
       "metaSPAdes          2141.0 [2076;2271]   2141.0 [2076;2271]  0.0 [0;0]   \n",
       "MINIA                        0.0 [0;0]            0.0 [0;0]  0.0 [0;0]   \n",
       "SKESA              1790.67 [1460;2005]  1790.67 [1460;2005]  0.0 [0;0]   \n",
       "SPAdes              2262.0 [1579;3246]   2262.0 [1579;3246]  0.0 [0;0]   \n",
       "Unicycler          2149.33 [2109;2216]  2149.33 [2109;2216]  0.0 [0;0]   \n",
       "VelvetOptimiser              0.0 [0;0]            0.0 [0;0]  0.0 [0;0]   \n",
       "\n",
       "                          Type            basepairs breadth_of_coverage  \\\n",
       "Assembler                                                                 \n",
       "ABySS              Metagenomic  2094.57 [1378;2215]          0.95 [0;0]   \n",
       "BCALM2                 Genomic            0.0 [0;0]           0.0 [0;0]   \n",
       "GATBMiniaPipeline  Metagenomic  2202.33 [2176;2216]          0.99 [0;1]   \n",
       "IDBA-UD            Metagenomic      1477.0 [0;2216]          0.67 [0;1]   \n",
       "MEGAHIT            Metagenomic  2215.33 [2215;2216]           1.0 [0;1]   \n",
       "MetaHipMer2        Metagenomic     1475.33 [0;2213]          0.67 [0;0]   \n",
       "metaSPAdes         Metagenomic  2122.67 [2076;2216]          0.96 [0;1]   \n",
       "MINIA                  Genomic            0.0 [0;0]           0.0 [0;0]   \n",
       "SKESA                  Genomic  1790.67 [1460;2005]          0.81 [0;0]   \n",
       "SPAdes                 Genomic  1918.67 [1579;2216]          0.87 [0;1]   \n",
       "Unicycler              Genomic  2149.33 [2109;2216]          0.97 [0;1]   \n",
       "VelvetOptimiser    Metagenomic            0.0 [0;0]           0.0 [0;0]   \n",
       "\n",
       "                      contigs    identity lowest_identity  \\\n",
       "Assembler                                                   \n",
       "ABySS               1.0 [1;1]   1.0 [1;1]       1.0 [1;1]   \n",
       "BCALM2              0.0 [0;0]   0.0 [0;0]       0.0 [0;0]   \n",
       "GATBMiniaPipeline   1.0 [1;1]   1.0 [1;1]       1.0 [1;1]   \n",
       "IDBA-UD            0.67 [0;1]  0.67 [0;1]      0.67 [0;1]   \n",
       "MEGAHIT             1.0 [1;1]   1.0 [1;1]       1.0 [1;1]   \n",
       "MetaHipMer2        0.67 [0;1]  0.67 [0;1]      0.67 [0;1]   \n",
       "metaSPAdes          1.0 [1;1]   1.0 [0;0]       1.0 [0;0]   \n",
       "MINIA               0.0 [0;0]   0.0 [0;0]       0.0 [0;0]   \n",
       "SKESA               1.0 [1;1]   1.0 [1;1]       1.0 [1;1]   \n",
       "SPAdes              1.0 [1;1]  0.89 [0;1]      0.89 [0;1]   \n",
       "Unicycler           1.0 [1;1]   1.0 [1;1]       1.0 [1;1]   \n",
       "VelvetOptimiser     0.0 [0;0]   0.0 [0;0]       0.0 [0;0]   \n",
       "\n",
       "                  misassembled contigs misassembly events multiplicity  \\\n",
       "Assembler                                                                \n",
       "ABySS                        0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "BCALM2                       0.0 [0;0]          0.0 [0;0]    0.0 [0;0]   \n",
       "GATBMiniaPipeline            0.0 [0;0]          0.0 [0;0]   1.02 [1;1]   \n",
       "IDBA-UD                      0.0 [0;0]          0.0 [0;0]   0.68 [0;1]   \n",
       "MEGAHIT                      0.0 [0;0]          0.0 [0;0]   1.02 [1;1]   \n",
       "MetaHipMer2                  0.0 [0;0]          0.0 [0;0]   0.67 [0;1]   \n",
       "metaSPAdes                   0.0 [0;0]          0.0 [0;0]   1.01 [1;1]   \n",
       "MINIA                        0.0 [0;0]          0.0 [0;0]    0.0 [0;0]   \n",
       "SKESA                        0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "SPAdes                       0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "Unicycler                    0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "VelvetOptimiser              0.0 [0;0]          0.0 [0;0]    0.0 [0;0]   \n",
       "\n",
       "                    parsimony    validity  \n",
       "Assembler                                  \n",
       "ABySS               1.0 [1;1]   1.0 [1;1]  \n",
       "BCALM2              0.0 [0;0]   0.0 [0;0]  \n",
       "GATBMiniaPipeline  1.02 [1;1]   1.0 [1;1]  \n",
       "IDBA-UD            0.68 [0;1]  0.67 [0;1]  \n",
       "MEGAHIT            1.02 [1;1]   1.0 [1;1]  \n",
       "MetaHipMer2        0.67 [0;1]  0.67 [0;1]  \n",
       "metaSPAdes         1.01 [1;1]   1.0 [1;1]  \n",
       "MINIA               0.0 [0;0]   0.0 [0;0]  \n",
       "SKESA               1.0 [1;1]   1.0 [1;1]  \n",
       "SPAdes             1.15 [1;1]  0.89 [0;1]  \n",
       "Unicycler           1.0 [1;1]   1.0 [1;1]  \n",
       "VelvetOptimiser     0.0 [0;0]   0.0 [0;0]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Staphylococcus_aureus_plasmid3\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.95 [0;0]</td>\n",
       "      <td>2843.43 [1955;2992]</td>\n",
       "      <td>2843.43 [1955;2992]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2843.43 [1955;2992]</td>\n",
       "      <td>0.95 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>2975.89 [2789;3158]</td>\n",
       "      <td>2975.89 [2789;3158]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2924.67 [2789;2993]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.02 [1;1]</td>\n",
       "      <td>1.02 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>0.82 [0;0]</td>\n",
       "      <td>2460.67 [1688;2847]</td>\n",
       "      <td>2460.67 [1688;2847]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2817.0 [2757;2847]</td>\n",
       "      <td>0.94 [0;0]</td>\n",
       "      <td>1.33 [1;2]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.02 [0;1]</td>\n",
       "      <td>3039.33 [2992;3134]</td>\n",
       "      <td>3039.33 [2992;3134]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2992.33 [2992;2993]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.02 [1;1]</td>\n",
       "      <td>1.02 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.84 [0;0]</td>\n",
       "      <td>2524.33 [1593;2990]</td>\n",
       "      <td>2524.33 [1593;2990]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2524.33 [1593;2990]</td>\n",
       "      <td>0.84 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>2403.67 [1677;3857]</td>\n",
       "      <td>2403.67 [1677;3857]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2944.0 [2847;2993]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "      <td>1.67 [1;2]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.11 [1;1]</td>\n",
       "      <td>1.11 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.85 [0;0]</td>\n",
       "      <td>2536.67 [2110;2782]</td>\n",
       "      <td>2536.67 [2110;2782]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2536.67 [2110;2782]</td>\n",
       "      <td>0.85 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.52 [0;0]</td>\n",
       "      <td>1556.33 [0;2992]</td>\n",
       "      <td>1556.33 [0;2992]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1556.33 [0;2992]</td>\n",
       "      <td>0.52 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "      <td>2926.33 [2886;2993]</td>\n",
       "      <td>2926.33 [2886;2993]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2926.33 [2886;2993]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm         L90         LSA  \\\n",
       "Assembler                                                                   \n",
       "ABySS              Multiple k-mer De Bruijn graph   0.0 [0;0]  0.95 [0;0]   \n",
       "BCALM2               Single k-mer De Bruijn graph   0.0 [0;0]   0.0 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph   0.0 [0;0]  0.99 [0;1]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  0.33 [0;1]  0.82 [0;0]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph   0.0 [0;0]  1.02 [0;1]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph   0.0 [0;0]  0.84 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph  0.67 [0;1]   0.8 [0;1]   \n",
       "MINIA                Single k-mer De Bruijn graph   0.0 [0;0]   0.0 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph   0.0 [0;0]  0.85 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph   0.0 [0;0]  0.52 [0;0]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph   0.0 [0;0]  0.98 [0;1]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph   0.0 [0;0]   0.0 [0;0]   \n",
       "\n",
       "                                  NA50                 NG50         Ns  \\\n",
       "Assembler                                                                \n",
       "ABySS              2843.43 [1955;2992]  2843.43 [1955;2992]  0.0 [0;0]   \n",
       "BCALM2                       0.0 [0;0]            0.0 [0;0]  0.0 [0;0]   \n",
       "GATBMiniaPipeline  2975.89 [2789;3158]  2975.89 [2789;3158]  0.0 [0;0]   \n",
       "IDBA-UD            2460.67 [1688;2847]  2460.67 [1688;2847]  0.0 [0;0]   \n",
       "MEGAHIT            3039.33 [2992;3134]  3039.33 [2992;3134]  0.0 [0;0]   \n",
       "MetaHipMer2        2524.33 [1593;2990]  2524.33 [1593;2990]  0.0 [0;0]   \n",
       "metaSPAdes         2403.67 [1677;3857]  2403.67 [1677;3857]  0.0 [0;0]   \n",
       "MINIA                        0.0 [0;0]            0.0 [0;0]  0.0 [0;0]   \n",
       "SKESA              2536.67 [2110;2782]  2536.67 [2110;2782]  0.0 [0;0]   \n",
       "SPAdes                1556.33 [0;2992]     1556.33 [0;2992]  0.0 [0;0]   \n",
       "Unicycler          2926.33 [2886;2993]  2926.33 [2886;2993]  0.0 [0;0]   \n",
       "VelvetOptimiser              0.0 [0;0]            0.0 [0;0]  0.0 [0;0]   \n",
       "\n",
       "                          Type            basepairs breadth_of_coverage  \\\n",
       "Assembler                                                                 \n",
       "ABySS              Metagenomic  2843.43 [1955;2992]          0.95 [0;0]   \n",
       "BCALM2                 Genomic            0.0 [0;0]           0.0 [0;0]   \n",
       "GATBMiniaPipeline  Metagenomic  2924.67 [2789;2993]          0.98 [0;1]   \n",
       "IDBA-UD            Metagenomic   2817.0 [2757;2847]          0.94 [0;0]   \n",
       "MEGAHIT            Metagenomic  2992.33 [2992;2993]           1.0 [0;1]   \n",
       "MetaHipMer2        Metagenomic  2524.33 [1593;2990]          0.84 [0;0]   \n",
       "metaSPAdes         Metagenomic   2944.0 [2847;2993]          0.98 [0;1]   \n",
       "MINIA                  Genomic            0.0 [0;0]           0.0 [0;0]   \n",
       "SKESA                  Genomic  2536.67 [2110;2782]          0.85 [0;0]   \n",
       "SPAdes                 Genomic     1556.33 [0;2992]          0.52 [0;0]   \n",
       "Unicycler              Genomic  2926.33 [2886;2993]          0.98 [0;1]   \n",
       "VelvetOptimiser    Metagenomic            0.0 [0;0]           0.0 [0;0]   \n",
       "\n",
       "                      contigs    identity lowest_identity  \\\n",
       "Assembler                                                   \n",
       "ABySS               1.0 [1;1]   1.0 [1;1]       1.0 [1;1]   \n",
       "BCALM2              0.0 [0;0]   0.0 [0;0]       0.0 [0;0]   \n",
       "GATBMiniaPipeline   1.0 [1;1]   1.0 [1;1]       1.0 [1;1]   \n",
       "IDBA-UD            1.33 [1;2]   1.0 [1;1]       1.0 [1;1]   \n",
       "MEGAHIT             1.0 [1;1]   1.0 [1;1]       1.0 [1;1]   \n",
       "MetaHipMer2         1.0 [1;1]   1.0 [1;1]       1.0 [1;1]   \n",
       "metaSPAdes         1.67 [1;2]   1.0 [1;1]       1.0 [1;1]   \n",
       "MINIA               0.0 [0;0]   0.0 [0;0]       0.0 [0;0]   \n",
       "SKESA               1.0 [1;1]   1.0 [1;1]       1.0 [1;1]   \n",
       "SPAdes             0.67 [0;1]  0.67 [0;1]      0.67 [0;1]   \n",
       "Unicycler           1.0 [1;1]   1.0 [1;1]       1.0 [1;1]   \n",
       "VelvetOptimiser     0.0 [0;0]   0.0 [0;0]       0.0 [0;0]   \n",
       "\n",
       "                  misassembled contigs misassembly events multiplicity  \\\n",
       "Assembler                                                                \n",
       "ABySS                        0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "BCALM2                       0.0 [0;0]          0.0 [0;0]    0.0 [0;0]   \n",
       "GATBMiniaPipeline            0.0 [0;0]          0.0 [0;0]   1.02 [1;1]   \n",
       "IDBA-UD                      0.0 [0;0]          0.0 [0;0]   1.01 [1;1]   \n",
       "MEGAHIT                      0.0 [0;0]          0.0 [0;0]   1.02 [1;1]   \n",
       "MetaHipMer2                  0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "metaSPAdes                   0.0 [0;0]          0.0 [0;0]   1.11 [1;1]   \n",
       "MINIA                        0.0 [0;0]          0.0 [0;0]    0.0 [0;0]   \n",
       "SKESA                        0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "SPAdes                       0.0 [0;0]          0.0 [0;0]   0.67 [0;1]   \n",
       "Unicycler                    0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "VelvetOptimiser              0.0 [0;0]          0.0 [0;0]    0.0 [0;0]   \n",
       "\n",
       "                    parsimony    validity  \n",
       "Assembler                                  \n",
       "ABySS               1.0 [1;1]   1.0 [1;1]  \n",
       "BCALM2              0.0 [0;0]   0.0 [0;0]  \n",
       "GATBMiniaPipeline  1.02 [1;1]   1.0 [1;1]  \n",
       "IDBA-UD            1.01 [1;1]   1.0 [1;1]  \n",
       "MEGAHIT            1.02 [1;1]   1.0 [1;1]  \n",
       "MetaHipMer2         1.0 [1;1]   1.0 [1;1]  \n",
       "metaSPAdes         1.11 [1;1]   1.0 [1;1]  \n",
       "MINIA               0.0 [0;0]   0.0 [0;0]  \n",
       "SKESA               1.0 [1;1]   1.0 [1;1]  \n",
       "SPAdes             0.67 [0;1]  0.67 [0;1]  \n",
       "Unicycler           1.0 [1;1]   1.0 [1;1]  \n",
       "VelvetOptimiser     0.0 [0;0]   0.0 [0;0]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "for reference in reference_pipeline_metrics_df.reference.unique():\n",
    "    ref_stats_per_reference_per_assembler = pd.DataFrame()\n",
    "    for assembler in reference_pipeline_metrics_df.assembler.unique():\n",
    "        row={'Assembler': assembler, \n",
    "             'Type':reference_pipeline_metrics_df[reference_pipeline_metrics_df.assembler == assembler].type.unique()[0],\n",
    "             'Algorithm': reference_pipeline_metrics_df[reference_pipeline_metrics_df.assembler == assembler].algorythm.unique()[0]}\n",
    "        lala=reference_pipeline_metrics_df[(reference_pipeline_metrics_df.assembler == assembler) & (reference_pipeline_metrics_df.reference == reference)].describe()\n",
    "        for column in lala.columns:\n",
    "            mean = lala.loc['mean',column]\n",
    "            minimum = lala.loc['min',column]\n",
    "            maximum = lala.loc['max',column]\n",
    "            row[column] = \"{} [{};{}]\".format(round(mean, 2),int(minimum), int(maximum))\n",
    "        ref_stats_per_reference_per_assembler=ref_stats_per_reference_per_assembler.append(row, ignore_index=True)\n",
    "    ref_stats_per_reference_per_assembler = ref_stats_per_reference_per_assembler.set_index('Assembler')\n",
    "    ref_stats_per_reference_per_assembler.to_csv(\"Tables/Results/{} - Reference metrics per assembler.csv\".format(reference))\n",
    "    print(reference)\n",
    "    display(ref_stats_per_reference_per_assembler)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Per reference replicon for all assemblers - Even only!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Reference</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Bacillus_subtilis</th>\n",
       "      <td>25.53 [0;120]</td>\n",
       "      <td>0.1 [0;0]</td>\n",
       "      <td>197321.07 [0;343843]</td>\n",
       "      <td>195383.53 [0;343843]</td>\n",
       "      <td>1.33 [0;47]</td>\n",
       "      <td>3611013.57 [0;3991152]</td>\n",
       "      <td>0.89 [0;0]</td>\n",
       "      <td>92.08 [0;1467]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "      <td>0.08 [0;2]</td>\n",
       "      <td>0.2 [0;4]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Enterococcus_faecalis</th>\n",
       "      <td>36.05 [0;149]</td>\n",
       "      <td>0.13 [0;0]</td>\n",
       "      <td>135017.9 [0;377104]</td>\n",
       "      <td>133464.98 [0;377104]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>2488219.36 [0;2828460]</td>\n",
       "      <td>0.87 [0;0]</td>\n",
       "      <td>72.08 [0;231]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.91 [0;1]</td>\n",
       "      <td>0.34 [0;3]</td>\n",
       "      <td>0.62 [0;6]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Escherichia_coli_plasmid</th>\n",
       "      <td>3.29 [0;18]</td>\n",
       "      <td>0.22 [0;0]</td>\n",
       "      <td>17648.73 [0;55639]</td>\n",
       "      <td>17101.08 [0;55639]</td>\n",
       "      <td>37.95 [0;835]</td>\n",
       "      <td>82168.47 [0;109516]</td>\n",
       "      <td>0.75 [0;0]</td>\n",
       "      <td>9.25 [0;33]</td>\n",
       "      <td>0.89 [0;1]</td>\n",
       "      <td>0.88 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.89 [0;1]</td>\n",
       "      <td>0.89 [0;1]</td>\n",
       "      <td>0.89 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Escherichia_coli</th>\n",
       "      <td>69.48 [0;283]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>51234.25 [0;138793]</td>\n",
       "      <td>50949.24 [0;138793]</td>\n",
       "      <td>1.27 [0;45]</td>\n",
       "      <td>4046356.04 [0;4706885]</td>\n",
       "      <td>0.85 [0;0]</td>\n",
       "      <td>365.0 [0;1495]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.84 [0;1]</td>\n",
       "      <td>1.08 [0;8]</td>\n",
       "      <td>2.01 [0;15]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Lactobacillus_fermentum</th>\n",
       "      <td>77.43 [0;263]</td>\n",
       "      <td>0.05 [0;0]</td>\n",
       "      <td>30491.53 [0;46932]</td>\n",
       "      <td>29253.78 [0;52972]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1583202.84 [0;1873053]</td>\n",
       "      <td>0.83 [0;0]</td>\n",
       "      <td>105.51 [0;282]</td>\n",
       "      <td>0.91 [0;1]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "      <td>0.34 [0;4]</td>\n",
       "      <td>0.7 [0;8]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Listeria_monocytogenes</th>\n",
       "      <td>24.25 [0;122]</td>\n",
       "      <td>0.15 [0;0]</td>\n",
       "      <td>238298.03 [0;542872]</td>\n",
       "      <td>229526.71 [0;542872]</td>\n",
       "      <td>0.03 [0;1]</td>\n",
       "      <td>2622957.47 [0;2961845]</td>\n",
       "      <td>0.88 [0;0]</td>\n",
       "      <td>62.58 [0;361]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.93 [0;1]</td>\n",
       "      <td>0.03 [0;1]</td>\n",
       "      <td>0.06 [0;2]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pseudomonas_aeruginosa</th>\n",
       "      <td>147.12 [0;1004]</td>\n",
       "      <td>0.06 [0;0]</td>\n",
       "      <td>142541.04 [0;338923]</td>\n",
       "      <td>147588.8 [0;464977]</td>\n",
       "      <td>588.46 [0;12098]</td>\n",
       "      <td>6106074.03 [0;6780808]</td>\n",
       "      <td>0.9 [0;0]</td>\n",
       "      <td>239.87 [0;1404]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.85 [0;1]</td>\n",
       "      <td>1.29 [0;8]</td>\n",
       "      <td>2.3 [0;15]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Salmonella_enterica</th>\n",
       "      <td>53.86 [0;200]</td>\n",
       "      <td>0.05 [0;0]</td>\n",
       "      <td>71537.42 [0;223960]</td>\n",
       "      <td>70549.1 [0;223960]</td>\n",
       "      <td>0.1 [0;2]</td>\n",
       "      <td>4069395.9 [0;4703394]</td>\n",
       "      <td>0.85 [0;0]</td>\n",
       "      <td>333.5 [0;1479]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "      <td>0.78 [0;9]</td>\n",
       "      <td>1.25 [0;14]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Staphylococcus_aureus</th>\n",
       "      <td>40.02 [0;140]</td>\n",
       "      <td>0.12 [0;0]</td>\n",
       "      <td>114659.96 [0;394879]</td>\n",
       "      <td>113926.2 [0;394879]</td>\n",
       "      <td>2.01 [0;71]</td>\n",
       "      <td>2370560.05 [0;2693907]</td>\n",
       "      <td>0.87 [0;0]</td>\n",
       "      <td>76.06 [0;249]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.91 [0;1]</td>\n",
       "      <td>0.26 [0;3]</td>\n",
       "      <td>0.61 [0;7]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Staphylococcus_aureus_plasmid1</th>\n",
       "      <td>0.03 [0;1]</td>\n",
       "      <td>0.78 [0;1]</td>\n",
       "      <td>4957.16 [0;6795]</td>\n",
       "      <td>4957.16 [0;6795]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>5083.47 [0;6337]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>0.91 [0;3]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Staphylococcus_aureus_plasmid2</th>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.66 [0;1]</td>\n",
       "      <td>1483.72 [0;3246]</td>\n",
       "      <td>1483.72 [0;3246]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1441.74 [0;2216]</td>\n",
       "      <td>0.65 [0;1]</td>\n",
       "      <td>0.69 [0;1]</td>\n",
       "      <td>0.68 [0;1]</td>\n",
       "      <td>0.68 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.69 [0;1]</td>\n",
       "      <td>0.71 [0;1]</td>\n",
       "      <td>0.68 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Staphylococcus_aureus_plasmid3</th>\n",
       "      <td>0.08 [0;1]</td>\n",
       "      <td>0.64 [0;1]</td>\n",
       "      <td>1921.82 [0;3857]</td>\n",
       "      <td>1921.82 [0;3857]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1989.61 [0;2993]</td>\n",
       "      <td>0.66 [0;1]</td>\n",
       "      <td>0.8 [0;2]</td>\n",
       "      <td>0.72 [0;1]</td>\n",
       "      <td>0.72 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.72 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            L90         LSA  \\\n",
       "Reference                                                     \n",
       "Bacillus_subtilis                 25.53 [0;120]   0.1 [0;0]   \n",
       "Enterococcus_faecalis             36.05 [0;149]  0.13 [0;0]   \n",
       "Escherichia_coli_plasmid            3.29 [0;18]  0.22 [0;0]   \n",
       "Escherichia_coli                  69.48 [0;283]  0.04 [0;0]   \n",
       "Lactobacillus_fermentum           77.43 [0;263]  0.05 [0;0]   \n",
       "Listeria_monocytogenes            24.25 [0;122]  0.15 [0;0]   \n",
       "Pseudomonas_aeruginosa          147.12 [0;1004]  0.06 [0;0]   \n",
       "Salmonella_enterica               53.86 [0;200]  0.05 [0;0]   \n",
       "Staphylococcus_aureus             40.02 [0;140]  0.12 [0;0]   \n",
       "Staphylococcus_aureus_plasmid1       0.03 [0;1]  0.78 [0;1]   \n",
       "Staphylococcus_aureus_plasmid2        0.0 [0;0]  0.66 [0;1]   \n",
       "Staphylococcus_aureus_plasmid3       0.08 [0;1]  0.64 [0;1]   \n",
       "\n",
       "                                                NA50                  NG50  \\\n",
       "Reference                                                                    \n",
       "Bacillus_subtilis               197321.07 [0;343843]  195383.53 [0;343843]   \n",
       "Enterococcus_faecalis            135017.9 [0;377104]  133464.98 [0;377104]   \n",
       "Escherichia_coli_plasmid          17648.73 [0;55639]    17101.08 [0;55639]   \n",
       "Escherichia_coli                 51234.25 [0;138793]   50949.24 [0;138793]   \n",
       "Lactobacillus_fermentum           30491.53 [0;46932]    29253.78 [0;52972]   \n",
       "Listeria_monocytogenes          238298.03 [0;542872]  229526.71 [0;542872]   \n",
       "Pseudomonas_aeruginosa          142541.04 [0;338923]   147588.8 [0;464977]   \n",
       "Salmonella_enterica              71537.42 [0;223960]    70549.1 [0;223960]   \n",
       "Staphylococcus_aureus           114659.96 [0;394879]   113926.2 [0;394879]   \n",
       "Staphylococcus_aureus_plasmid1      4957.16 [0;6795]      4957.16 [0;6795]   \n",
       "Staphylococcus_aureus_plasmid2      1483.72 [0;3246]      1483.72 [0;3246]   \n",
       "Staphylococcus_aureus_plasmid3      1921.82 [0;3857]      1921.82 [0;3857]   \n",
       "\n",
       "                                              Ns               basepairs  \\\n",
       "Reference                                                                  \n",
       "Bacillus_subtilis                    1.33 [0;47]  3611013.57 [0;3991152]   \n",
       "Enterococcus_faecalis                  0.0 [0;0]  2488219.36 [0;2828460]   \n",
       "Escherichia_coli_plasmid           37.95 [0;835]     82168.47 [0;109516]   \n",
       "Escherichia_coli                     1.27 [0;45]  4046356.04 [0;4706885]   \n",
       "Lactobacillus_fermentum                0.0 [0;0]  1583202.84 [0;1873053]   \n",
       "Listeria_monocytogenes                0.03 [0;1]  2622957.47 [0;2961845]   \n",
       "Pseudomonas_aeruginosa          588.46 [0;12098]  6106074.03 [0;6780808]   \n",
       "Salmonella_enterica                    0.1 [0;2]   4069395.9 [0;4703394]   \n",
       "Staphylococcus_aureus                2.01 [0;71]  2370560.05 [0;2693907]   \n",
       "Staphylococcus_aureus_plasmid1         0.0 [0;0]        5083.47 [0;6337]   \n",
       "Staphylococcus_aureus_plasmid2         0.0 [0;0]        1441.74 [0;2216]   \n",
       "Staphylococcus_aureus_plasmid3         0.0 [0;0]        1989.61 [0;2993]   \n",
       "\n",
       "                               breadth_of_coverage          contigs  \\\n",
       "Reference                                                             \n",
       "Bacillus_subtilis                       0.89 [0;0]   92.08 [0;1467]   \n",
       "Enterococcus_faecalis                   0.87 [0;0]    72.08 [0;231]   \n",
       "Escherichia_coli_plasmid                0.75 [0;0]      9.25 [0;33]   \n",
       "Escherichia_coli                        0.85 [0;0]   365.0 [0;1495]   \n",
       "Lactobacillus_fermentum                 0.83 [0;0]   105.51 [0;282]   \n",
       "Listeria_monocytogenes                  0.88 [0;0]    62.58 [0;361]   \n",
       "Pseudomonas_aeruginosa                   0.9 [0;0]  239.87 [0;1404]   \n",
       "Salmonella_enterica                     0.85 [0;0]   333.5 [0;1479]   \n",
       "Staphylococcus_aureus                   0.87 [0;0]    76.06 [0;249]   \n",
       "Staphylococcus_aureus_plasmid1           0.8 [0;1]       0.91 [0;3]   \n",
       "Staphylococcus_aureus_plasmid2          0.65 [0;1]       0.69 [0;1]   \n",
       "Staphylococcus_aureus_plasmid3          0.66 [0;1]        0.8 [0;2]   \n",
       "\n",
       "                                  identity lowest_identity  \\\n",
       "Reference                                                    \n",
       "Bacillus_subtilis               0.94 [0;1]      0.92 [0;1]   \n",
       "Enterococcus_faecalis           0.94 [0;1]      0.91 [0;1]   \n",
       "Escherichia_coli_plasmid        0.89 [0;1]      0.88 [0;1]   \n",
       "Escherichia_coli                0.94 [0;1]      0.84 [0;1]   \n",
       "Lactobacillus_fermentum         0.91 [0;1]       0.9 [0;1]   \n",
       "Listeria_monocytogenes          0.94 [0;1]      0.93 [0;1]   \n",
       "Pseudomonas_aeruginosa          0.94 [0;1]      0.85 [0;1]   \n",
       "Salmonella_enterica             0.94 [0;1]      0.83 [0;1]   \n",
       "Staphylococcus_aureus           0.94 [0;1]      0.91 [0;1]   \n",
       "Staphylococcus_aureus_plasmid1  0.82 [0;1]      0.82 [0;1]   \n",
       "Staphylococcus_aureus_plasmid2  0.68 [0;1]      0.68 [0;1]   \n",
       "Staphylococcus_aureus_plasmid3  0.72 [0;1]      0.72 [0;1]   \n",
       "\n",
       "                               misassembled contigs misassembly events  \\\n",
       "Reference                                                                \n",
       "Bacillus_subtilis                        0.08 [0;2]          0.2 [0;4]   \n",
       "Enterococcus_faecalis                    0.34 [0;3]         0.62 [0;6]   \n",
       "Escherichia_coli_plasmid                  0.0 [0;0]          0.0 [0;0]   \n",
       "Escherichia_coli                         1.08 [0;8]        2.01 [0;15]   \n",
       "Lactobacillus_fermentum                  0.34 [0;4]          0.7 [0;8]   \n",
       "Listeria_monocytogenes                   0.03 [0;1]         0.06 [0;2]   \n",
       "Pseudomonas_aeruginosa                   1.29 [0;8]         2.3 [0;15]   \n",
       "Salmonella_enterica                      0.78 [0;9]        1.25 [0;14]   \n",
       "Staphylococcus_aureus                    0.26 [0;3]         0.61 [0;7]   \n",
       "Staphylococcus_aureus_plasmid1            0.0 [0;0]          0.0 [0;0]   \n",
       "Staphylococcus_aureus_plasmid2            0.0 [0;0]          0.0 [0;0]   \n",
       "Staphylococcus_aureus_plasmid3            0.0 [0;0]          0.0 [0;0]   \n",
       "\n",
       "                               multiplicity   parsimony    validity  \n",
       "Reference                                                            \n",
       "Bacillus_subtilis                0.94 [0;1]  0.94 [0;1]  0.94 [0;1]  \n",
       "Enterococcus_faecalis            0.95 [0;1]  0.95 [0;1]  0.94 [0;1]  \n",
       "Escherichia_coli_plasmid         0.89 [0;1]  0.89 [0;1]  0.89 [0;1]  \n",
       "Escherichia_coli                 0.95 [0;1]  0.95 [0;1]  0.94 [0;1]  \n",
       "Lactobacillus_fermentum          0.92 [0;1]  0.92 [0;1]  0.92 [0;1]  \n",
       "Listeria_monocytogenes           0.94 [0;1]  0.94 [0;1]  0.94 [0;1]  \n",
       "Pseudomonas_aeruginosa           0.95 [0;1]  0.95 [0;1]  0.94 [0;1]  \n",
       "Salmonella_enterica              0.95 [0;1]  0.95 [0;1]  0.94 [0;1]  \n",
       "Staphylococcus_aureus            0.94 [0;1]  0.94 [0;1]  0.94 [0;1]  \n",
       "Staphylococcus_aureus_plasmid1   0.82 [0;1]  0.82 [0;1]  0.82 [0;1]  \n",
       "Staphylococcus_aureus_plasmid2   0.69 [0;1]  0.71 [0;1]  0.68 [0;1]  \n",
       "Staphylococcus_aureus_plasmid3   0.73 [0;1]  0.73 [0;1]  0.72 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ref_stats_per_reference_all_assemblers = pd.DataFrame()\n",
    "for reference in reference_pipeline_metrics_df.reference.unique():\n",
    "    row={'Reference': reference}\n",
    "    lala=reference_pipeline_metrics_df[(reference_pipeline_metrics_df.reference == reference) & (reference_pipeline_metrics_df.distribution == 'Even')].describe()\n",
    "    for column in lala.columns:\n",
    "        mean = lala.loc['mean',column]\n",
    "        minimum = lala.loc['min',column]\n",
    "        maximum = lala.loc['max',column]\n",
    "        row[column] = \"{} [{};{}]\".format(round(mean, 2),int(minimum), int(maximum))\n",
    "    ref_stats_per_reference_all_assemblers=ref_stats_per_reference_all_assemblers.append(row, ignore_index=True)\n",
    "ref_stats_per_reference_all_assemblers = ref_stats_per_reference_all_assemblers.set_index('Reference')\n",
    "ref_stats_per_reference_all_assemblers.to_csv(\"Tables/Results/Reference metrics per replicon.csv\")\n",
    "display(ref_stats_per_reference_all_assemblers)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reference stats per distribution"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Even\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>60.88 [0;1004]</td>\n",
       "      <td>0.32 [0;0]</td>\n",
       "      <td>110190.23 [0;351502]</td>\n",
       "      <td>110089.38 [0;351502]</td>\n",
       "      <td>105.21 [0;2170]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2536099.3 [0;6780808]</td>\n",
       "      <td>0.92 [0;0]</td>\n",
       "      <td>97.05 [0;1404]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.37 [0;8]</td>\n",
       "      <td>0.65 [0;14]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>51.2 [0;921]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>6932.33 [0;65138]</td>\n",
       "      <td>6539.94 [0;64934]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1236061.67 [0;6510588]</td>\n",
       "      <td>0.37 [0;0]</td>\n",
       "      <td>272.97 [0;1495]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>31.33 [0;112]</td>\n",
       "      <td>0.35 [0;1]</td>\n",
       "      <td>106093.36 [2176;452364]</td>\n",
       "      <td>96782.33 [2176;351186]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2523831.42 [2176;6716518]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "      <td>59.92 [1;199]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "      <td>0.14 [0;3]</td>\n",
       "      <td>0.25 [0;6]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>68.82 [0;283]</td>\n",
       "      <td>0.27 [0;1]</td>\n",
       "      <td>55837.36 [0;254629]</td>\n",
       "      <td>54555.43 [0;254629]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2507606.06 [0;6710104]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>121.11 [0;449]</td>\n",
       "      <td>0.97 [0;1]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "      <td>1.12 [0;9]</td>\n",
       "      <td>1.94 [0;15]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "      <td>0.97 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>18.31 [0;55]</td>\n",
       "      <td>0.37 [0;1]</td>\n",
       "      <td>138989.33 [2215;542872]</td>\n",
       "      <td>141793.83 [2215;542872]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2534425.75 [2215;6737986]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>44.39 [1;141]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.93 [0;1]</td>\n",
       "      <td>0.97 [0;4]</td>\n",
       "      <td>1.99 [0;7]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.01 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>21.69 [0;110]</td>\n",
       "      <td>0.23 [0;0]</td>\n",
       "      <td>58464.84 [0;257424]</td>\n",
       "      <td>57747.71 [0;257424]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1722416.51 [0;6717767]</td>\n",
       "      <td>0.68 [0;0]</td>\n",
       "      <td>74.19 [0;361]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>20.14 [0;90]</td>\n",
       "      <td>0.34 [0;1]</td>\n",
       "      <td>133559.97 [1677;475510]</td>\n",
       "      <td>132927.86 [1677;475510]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2530047.11 [2076;6726884]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "      <td>39.33 [1;167]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>0.14 [0;2]</td>\n",
       "      <td>0.33 [0;6]</td>\n",
       "      <td>1.01 [0;1]</td>\n",
       "      <td>1.01 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>83.4 [0;406]</td>\n",
       "      <td>0.08 [0;0]</td>\n",
       "      <td>17273.69 [0;65138]</td>\n",
       "      <td>16631.06 [0;64960]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2345231.06 [0;6518607]</td>\n",
       "      <td>0.74 [0;0]</td>\n",
       "      <td>307.75 [0;1146]</td>\n",
       "      <td>0.81 [0;1]</td>\n",
       "      <td>0.79 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.81 [0;1]</td>\n",
       "      <td>0.81 [0;1]</td>\n",
       "      <td>0.81 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>57.92 [0;670]</td>\n",
       "      <td>0.27 [0;0]</td>\n",
       "      <td>76053.75 [1404;351014]</td>\n",
       "      <td>75134.92 [0;351014]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2496709.31 [1460;6673601]</td>\n",
       "      <td>0.9 [0;0]</td>\n",
       "      <td>96.06 [1;933]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "      <td>0.67 [0;7]</td>\n",
       "      <td>1.17 [0;15]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>21.06 [0;79]</td>\n",
       "      <td>0.29 [0;1]</td>\n",
       "      <td>117650.97 [0;315090]</td>\n",
       "      <td>117061.69 [0;315090]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2524122.0 [0;6728888]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "      <td>41.19 [0;143]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "      <td>0.36 [0;4]</td>\n",
       "      <td>0.69 [0;8]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.96 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>15.75 [0;57]</td>\n",
       "      <td>0.38 [0;1]</td>\n",
       "      <td>162270.42 [0;475179]</td>\n",
       "      <td>158177.44 [0;475179]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2515166.11 [0;6723626]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "      <td>32.86 [0;111]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.06 [0;1]</td>\n",
       "      <td>0.11 [0;2]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>31.33 [0;207]</td>\n",
       "      <td>0.09 [0;0]</td>\n",
       "      <td>29632.83 [0;338923]</td>\n",
       "      <td>34682.42 [0;464977]</td>\n",
       "      <td>537.64 [0;12098]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1580537.53 [0;6708956]</td>\n",
       "      <td>0.52 [0;0]</td>\n",
       "      <td>167.92 [0;1098]</td>\n",
       "      <td>0.56 [0;1]</td>\n",
       "      <td>0.55 [0;1]</td>\n",
       "      <td>0.39 [0;8]</td>\n",
       "      <td>0.61 [0;11]</td>\n",
       "      <td>0.56 [0;1]</td>\n",
       "      <td>0.56 [0;1]</td>\n",
       "      <td>0.56 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm             L90         LSA  \\\n",
       "Assembler                                                                       \n",
       "ABySS              Multiple k-mer De Bruijn graph  60.88 [0;1004]  0.32 [0;0]   \n",
       "BCALM2               Single k-mer De Bruijn graph    51.2 [0;921]  0.04 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph   31.33 [0;112]  0.35 [0;1]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph   68.82 [0;283]  0.27 [0;1]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph    18.31 [0;55]  0.37 [0;1]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph   21.69 [0;110]  0.23 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph    20.14 [0;90]  0.34 [0;1]   \n",
       "MINIA                Single k-mer De Bruijn graph    83.4 [0;406]  0.08 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph   57.92 [0;670]  0.27 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph    21.06 [0;79]  0.29 [0;1]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph    15.75 [0;57]  0.38 [0;1]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph   31.33 [0;207]  0.09 [0;0]   \n",
       "\n",
       "                                      NA50                     NG50  \\\n",
       "Assembler                                                             \n",
       "ABySS                 110190.23 [0;351502]     110089.38 [0;351502]   \n",
       "BCALM2                   6932.33 [0;65138]        6539.94 [0;64934]   \n",
       "GATBMiniaPipeline  106093.36 [2176;452364]   96782.33 [2176;351186]   \n",
       "IDBA-UD                55837.36 [0;254629]      54555.43 [0;254629]   \n",
       "MEGAHIT            138989.33 [2215;542872]  141793.83 [2215;542872]   \n",
       "MetaHipMer2            58464.84 [0;257424]      57747.71 [0;257424]   \n",
       "metaSPAdes         133559.97 [1677;475510]  132927.86 [1677;475510]   \n",
       "MINIA                   17273.69 [0;65138]       16631.06 [0;64960]   \n",
       "SKESA               76053.75 [1404;351014]      75134.92 [0;351014]   \n",
       "SPAdes                117650.97 [0;315090]     117061.69 [0;315090]   \n",
       "Unicycler             162270.42 [0;475179]     158177.44 [0;475179]   \n",
       "VelvetOptimiser        29632.83 [0;338923]      34682.42 [0;464977]   \n",
       "\n",
       "                                 Ns         Type                  basepairs  \\\n",
       "Assembler                                                                     \n",
       "ABySS               105.21 [0;2170]  Metagenomic      2536099.3 [0;6780808]   \n",
       "BCALM2                    0.0 [0;0]      Genomic     1236061.67 [0;6510588]   \n",
       "GATBMiniaPipeline         0.0 [0;0]  Metagenomic  2523831.42 [2176;6716518]   \n",
       "IDBA-UD                   0.0 [0;0]  Metagenomic     2507606.06 [0;6710104]   \n",
       "MEGAHIT                   0.0 [0;0]  Metagenomic  2534425.75 [2215;6737986]   \n",
       "MetaHipMer2               0.0 [0;0]  Metagenomic     1722416.51 [0;6717767]   \n",
       "metaSPAdes                0.0 [0;0]  Metagenomic  2530047.11 [2076;6726884]   \n",
       "MINIA                     0.0 [0;0]      Genomic     2345231.06 [0;6518607]   \n",
       "SKESA                     0.0 [0;0]      Genomic  2496709.31 [1460;6673601]   \n",
       "SPAdes                    0.0 [0;0]      Genomic      2524122.0 [0;6728888]   \n",
       "Unicycler                 0.0 [0;0]      Genomic     2515166.11 [0;6723626]   \n",
       "VelvetOptimiser    537.64 [0;12098]  Metagenomic     1580537.53 [0;6708956]   \n",
       "\n",
       "                  breadth_of_coverage          contigs    identity  \\\n",
       "Assembler                                                            \n",
       "ABySS                      0.92 [0;0]   97.05 [0;1404]  0.99 [0;1]   \n",
       "BCALM2                     0.37 [0;0]  272.97 [0;1495]   0.5 [0;1]   \n",
       "GATBMiniaPipeline          0.98 [0;1]    59.92 [1;199]   1.0 [0;1]   \n",
       "IDBA-UD                    0.94 [0;1]   121.11 [0;449]  0.97 [0;1]   \n",
       "MEGAHIT                    0.99 [0;1]    44.39 [1;141]   1.0 [0;1]   \n",
       "MetaHipMer2                0.68 [0;0]    74.19 [0;361]  0.94 [0;1]   \n",
       "metaSPAdes                 0.98 [0;1]    39.33 [1;167]   1.0 [0;1]   \n",
       "MINIA                      0.74 [0;0]  307.75 [0;1146]  0.81 [0;1]   \n",
       "SKESA                       0.9 [0;0]    96.06 [1;933]   1.0 [0;1]   \n",
       "SPAdes                      0.9 [0;1]    41.19 [0;143]  0.94 [0;1]   \n",
       "Unicycler                  0.92 [0;1]    32.86 [0;111]  0.94 [0;1]   \n",
       "VelvetOptimiser            0.52 [0;0]  167.92 [0;1098]  0.56 [0;1]   \n",
       "\n",
       "                  lowest_identity misassembled contigs misassembly events  \\\n",
       "Assembler                                                                   \n",
       "ABySS                  0.95 [0;1]           0.37 [0;8]        0.65 [0;14]   \n",
       "BCALM2                  0.5 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "GATBMiniaPipeline      0.98 [0;1]           0.14 [0;3]         0.25 [0;6]   \n",
       "IDBA-UD                0.87 [0;1]           1.12 [0;9]        1.94 [0;15]   \n",
       "MEGAHIT                0.93 [0;1]           0.97 [0;4]         1.99 [0;7]   \n",
       "MetaHipMer2            0.94 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "metaSPAdes             0.99 [0;1]           0.14 [0;2]         0.33 [0;6]   \n",
       "MINIA                  0.79 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "SKESA                  0.87 [0;1]           0.67 [0;7]        1.17 [0;15]   \n",
       "SPAdes                  0.9 [0;1]           0.36 [0;4]         0.69 [0;8]   \n",
       "Unicycler              0.94 [0;1]           0.06 [0;1]         0.11 [0;2]   \n",
       "VelvetOptimiser        0.55 [0;1]           0.39 [0;8]        0.61 [0;11]   \n",
       "\n",
       "                  multiplicity   parsimony    validity  \n",
       "Assembler                                               \n",
       "ABySS                1.0 [0;1]   1.0 [0;1]  0.99 [0;1]  \n",
       "BCALM2               0.5 [0;1]   0.5 [0;1]   0.5 [0;1]  \n",
       "GATBMiniaPipeline    1.0 [0;1]   1.0 [0;1]   1.0 [0;1]  \n",
       "IDBA-UD             0.98 [0;1]  0.98 [0;1]  0.97 [0;1]  \n",
       "MEGAHIT              1.0 [0;1]  1.01 [0;1]   1.0 [0;1]  \n",
       "MetaHipMer2         0.94 [0;1]  0.94 [0;1]  0.94 [0;1]  \n",
       "metaSPAdes          1.01 [0;1]  1.01 [0;1]   1.0 [0;1]  \n",
       "MINIA               0.81 [0;1]  0.81 [0;1]  0.81 [0;1]  \n",
       "SKESA                1.0 [0;1]   1.0 [0;1]   1.0 [0;1]  \n",
       "SPAdes              0.95 [0;1]  0.96 [0;1]  0.94 [0;1]  \n",
       "Unicycler           0.94 [0;1]  0.94 [0;1]  0.94 [0;1]  \n",
       "VelvetOptimiser     0.56 [0;1]  0.56 [0;1]  0.56 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "for distribution in reference_pipeline_metrics_df.distribution.unique():\n",
    "    ref_stats_per_reference_per_assembler = pd.DataFrame()\n",
    "    for assembler in reference_pipeline_metrics_df.assembler.unique():\n",
    "        row={'Assembler': assembler, \n",
    "             'Type':reference_pipeline_metrics_df[reference_pipeline_metrics_df.assembler == assembler].type.unique()[0],\n",
    "             'Algorithm': reference_pipeline_metrics_df[reference_pipeline_metrics_df.assembler == assembler].algorythm.unique()[0]}\n",
    "        lala=reference_pipeline_metrics_df[(reference_pipeline_metrics_df.assembler == assembler) & (reference_pipeline_metrics_df.distribution == distribution)].describe()\n",
    "        for column in lala.columns:\n",
    "            mean = lala.loc['mean',column]\n",
    "            minimum = lala.loc['min',column]\n",
    "            maximum = lala.loc['max',column]\n",
    "            row[column] = \"{} [{};{}]\".format(round(mean, 2),int(minimum), int(maximum))\n",
    "        ref_stats_per_reference_per_assembler=ref_stats_per_reference_per_assembler.append(row, ignore_index=True)\n",
    "    ref_stats_per_reference_per_assembler = ref_stats_per_reference_per_assembler.set_index('Assembler')\n",
    "    ref_stats_per_reference_per_assembler.to_csv(\"Tables/Results/{} - Reference metrics per assembler.csv\".format(distribution))\n",
    "    print(distribution)\n",
    "    display(ref_stats_per_reference_per_assembler)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Genomes vs Plasmids"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Replicon</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Genome</th>\n",
       "      <td>59.22(± 105.72)</td>\n",
       "      <td>0.09(± 0.08)</td>\n",
       "      <td>122637.65(± 120855.79)</td>\n",
       "      <td>121330.29(± 120588.4)</td>\n",
       "      <td>74.15(± 816.88)</td>\n",
       "      <td>3362222.41(± 1746712.85)</td>\n",
       "      <td>0.87(± 0.28)</td>\n",
       "      <td>168.33(± 281.98)</td>\n",
       "      <td>0.94(± 0.24)</td>\n",
       "      <td>0.89(± 0.26)</td>\n",
       "      <td>0.53(± 1.45)</td>\n",
       "      <td>0.97(± 2.63)</td>\n",
       "      <td>0.94(± 0.24)</td>\n",
       "      <td>0.94(± 0.24)</td>\n",
       "      <td>0.94(± 0.24)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Plasmid</th>\n",
       "      <td>0.85 [0;18]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "      <td>6502.86 [0;55639]</td>\n",
       "      <td>6365.95 [0;55639]</td>\n",
       "      <td>9.49 [0;835]</td>\n",
       "      <td>22670.82 [0;109516]</td>\n",
       "      <td>0.72 [0;1]</td>\n",
       "      <td>2.91 [0;33]</td>\n",
       "      <td>0.78 [0;1]</td>\n",
       "      <td>0.77 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.78 [0;1]</td>\n",
       "      <td>0.79 [0;1]</td>\n",
       "      <td>0.78 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      L90           LSA                    NA50  \\\n",
       "Replicon                                                          \n",
       "Genome    59.22(± 105.72)  0.09(± 0.08)  122637.65(± 120855.79)   \n",
       "Plasmid       0.85 [0;18]    0.58 [0;1]       6502.86 [0;55639]   \n",
       "\n",
       "                           NG50               Ns                 basepairs  \\\n",
       "Replicon                                                                     \n",
       "Genome    121330.29(± 120588.4)  74.15(± 816.88)  3362222.41(± 1746712.85)   \n",
       "Plasmid       6365.95 [0;55639]     9.49 [0;835]       22670.82 [0;109516]   \n",
       "\n",
       "         breadth_of_coverage           contigs      identity lowest_identity  \\\n",
       "Replicon                                                                       \n",
       "Genome          0.87(± 0.28)  168.33(± 281.98)  0.94(± 0.24)    0.89(± 0.26)   \n",
       "Plasmid           0.72 [0;1]       2.91 [0;33]    0.78 [0;1]      0.77 [0;1]   \n",
       "\n",
       "         misassembled contigs misassembly events  multiplicity     parsimony  \\\n",
       "Replicon                                                                       \n",
       "Genome           0.53(± 1.45)       0.97(± 2.63)  0.94(± 0.24)  0.94(± 0.24)   \n",
       "Plasmid             0.0 [0;0]          0.0 [0;0]    0.78 [0;1]    0.79 [0;1]   \n",
       "\n",
       "              validity  \n",
       "Replicon                \n",
       "Genome    0.94(± 0.24)  \n",
       "Plasmid     0.78 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ref_stats_per_replicon_type = pd.DataFrame()\n",
    "# Genome\n",
    "row={'Replicon': \"Genome\"}\n",
    "lala=reference_pipeline_metrics_df[~reference_pipeline_metrics_df['reference'].str.contains(\"plasmid\")].describe()\n",
    "for column in lala.columns:\n",
    "    mean = lala.loc['mean',column]\n",
    "    std = lala.loc['std',column]\n",
    "    row[column] = \"{}(± {})\".format(round(mean, 2), round(std, 2))\n",
    "ref_stats_per_replicon_type =ref_stats_per_replicon_type .append(row, ignore_index=True)\n",
    "\n",
    "# Plasmid\n",
    "row={'Replicon': \"Plasmid\"}\n",
    "lala=reference_pipeline_metrics_df[reference_pipeline_metrics_df['reference'].str.contains(\"plasmid\")].describe()\n",
    "for column in lala.columns:\n",
    "    mean = lala.loc['mean',column]\n",
    "    minimum = lala.loc['min',column]\n",
    "    maximum = lala.loc['max',column]\n",
    "    row[column] = \"{} [{};{}]\".format(round(mean, 2),int(minimum), int(maximum))\n",
    "    \n",
    "ref_stats_per_replicon_type =ref_stats_per_replicon_type .append(row, ignore_index=True)\n",
    "ref_stats_per_replicon_type  = ref_stats_per_replicon_type .set_index('Replicon')\n",
    "ref_stats_per_replicon_type.to_csv(\"Tables/Results/Reference metrics per replicon type.csv\")\n",
    "display(ref_stats_per_replicon_type)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
