{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 1. Compute Results Overview Tables"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The success of an assembly is evaluated by the computation of metrics in two defined ways: globally through statistics inherent to the complete set of sequences that were assembled, and relative to the replicons present in the sample. \n",
    "\n",
    "The computation of the global metrics is done through custom python code from the complete set of sequences assembled. \n",
    "\n",
    "The following metrics are computed for the complete and filtered set of assembled sequences, restricted to contigs of length above a specified minimum size: \n",
    "\n",
    "- **Contig sizes**\n",
    "    - **Contigs:** The total number of contigs in the assembly;\n",
    "    - **Basepairs:** The total number of bases in the assembly;\n",
    "    - **Maximum sequence length:** The length of the largest contig in the assembly.\n",
    "    - **Number of ‘N’s:** Number of uncalled bases (N's) \n",
    "- **Contiguity**\n",
    "    - **Nx (where 0  < x  ⩽ 100):** Length for which the collection of all assembled sequences of that length or longer in an assembly covers at least a given percentage of the total length of the assembly\n",
    "- **Misassembly**\n",
    "    - **Misassemblies** - Number of aligned contigs that contain a misassembly event\n",
    "\n",
    "After filtering, the sequences are mapped with the reference replicons and the metrics are computed through custom python code. The metrics are calculated, filtered for a minimum length, for each replicon in the file provided input references. \n",
    "\n",
    "- **Contig sizes**\n",
    "    - **Contigs:** The total number of contigs in the assembly;\n",
    "    - **Basepairs:** The total number of bases in the assembly;\n",
    "    - **Number of ‘N’s:** Number of uncalled bases (N's) \n",
    "- **COMPASS**\n",
    "    - **(Breadth of) Coverage:** Ratio of covered sequence on the reference by aligned contigs;\n",
    "    - **Multiplicity:** Ratio of the length of alignable assembled sequence to covered sequence on the reference;\n",
    "    - **Validity:** Ratio of the length of the alignable assembled sequence to total basepairs in the aligned contigs;\n",
    "    - **Parsimony:** Cost of the assembly (multiplicity over validity);\n",
    "- **Contiguity**\n",
    "    - **Contiguity:** longest single alignment between the assembly and the reference, relative to the reference length;\n",
    "    - **NAx (where 0  < x  ⩽ 100):** Length for which the collection of aligned assembled sequences of that length or longer in an assembly covers at least a given percentage of the total length of the reference replicon;\n",
    "    - **NGx (where 0  < x  ⩽ 100):** Length for which the collection of aligned contigs of that length or longer covers at least a given percentage of the sequence of the reference.\n",
    "    - **Lx (where 0  < x  ⩽ 100):** Minimal number of contigs that cover x % of the sequence of the reference;\n",
    "- **Identity**\n",
    "    - **Identity:** Ratio of identical basepairs in all aligned contigs to the reference;\n",
    "    - **Lowest identity:** Identity of the lowest scoring contig to the reference.\n",
    "    - **PLS**: Phred-like score per contig, per assembler.\n",
    "- **Misassembly**\n",
    "    - **Misassemblies** - Number of aligned contigs that contain a misassembly event\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "from plotly.offline import plot\n",
    "import glob\n",
    "import fnmatch\n",
    "import plotly.graph_objects as go\n",
    "from plotly.subplots import make_subplots\n",
    "import json\n",
    "import pandas as pd\n",
    "from itertools import groupby\n",
    "import csv\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Global variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [],
   "source": [
    "ASSEMBLER_PROCESS_LIST = [\"ABYSS\", \"BCALM2\", \"GATBMINIAPIPELINE\", \"METAHIPMER2\", \"MINIA\", \"MEGAHIT\", \"METASPADES\", \"UNICYCLER\", \"SPADES\",\n",
    "                          \"SKESA\", \"VELVETOPTIMIZER\", \"IDBA\"]\n",
    "PROCESS_TO_NAME = {\"ABYSS\": \"AbYSS\",\n",
    "                   \"BCALM2\": \"BCALM2\", \n",
    "                   \"GATBMINIAPIPELINE\": \"GATBMiniaPipeline\",\n",
    "                   \"METAHIPMER2\": \"MetaHipMer2\",\n",
    "                   \"MINIA\": \"MINIA\",\n",
    "                   \"MEGAHIT\": \"MEGAHIT\", \n",
    "                   \"METASPADES\": \"metaSPAdes\", \n",
    "                   \"UNICYCLER\": \"Unicycler\", \n",
    "                   \"SPADES\": \"SPAdes\",\n",
    "                    \"SKESA\": \"SKESA\",\n",
    "                   \"VELVETOPTIMIZER\": \"VelvetOptimizer\",\n",
    "                   \"IDBA\": \"IDBA-UD\"}\n",
    "\n",
    "log_distributed = ['LHS', 'LNN', 'ERR2935805']\n",
    "even_distribution = ['EMS', 'LNN', 'ERR2984773']\n",
    "\n",
    "genomic_assemblers = ['AbYSS', 'BCALM2', 'MINIA', 'SKESA', 'SPAdes', 'Unicycler', 'VelvetOptimizer']\n",
    "metagenomic_assemblers = ['MetaHipMer2','GATBMiniaPipeline', 'IDBA-UD', 'MEGAHIT', 'metaSPAdes']\n",
    "single_kmer = ['BCALM2', 'MINIA', 'AbYSS']\n",
    "multiple_kmer = ['SKESA', 'SPAdes', 'Unicycler', 'VelvetOptimizer', 'GATBMiniaPipeline', \n",
    "                 'IDBA-UD', 'MEGAHIT', 'metaSPAdes', 'MetaHipMer2']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Global metrics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pipeline_report_tables.json data from run2...\n",
      "Processing pipeline_report_tables.json data from run3...\n",
      "Processing pipeline_report_tables.json data from run1...\n"
     ]
    }
   ],
   "source": [
    "report_glob = glob.glob('../Results/*/*/report/pipeline_report_tables.json')\n",
    "global_pipeline_metrics_df = pd.DataFrame()\n",
    "\n",
    "for pipeline_report_file in report_glob:\n",
    "    report_file_name = pipeline_report_file.split('/')[-1]\n",
    "    stats_run = pipeline_report_file.split('/')[-3]\n",
    "    print('Processing {0} data from {1}...'.format(report_file_name, stats_run))\n",
    "    \n",
    "    with open(pipeline_report_file) as _fh:\n",
    "        json_report = json.load(_fh)\n",
    "        for sample in json_report.keys():\n",
    "            for line in json_report[sample]['GlobalTable']:\n",
    "                assembler = line['assembler']\n",
    "                global_pipeline_metrics_df = global_pipeline_metrics_df.append({'run': stats_run,\n",
    "                                                                                'sample': sample,\n",
    "                                                                                'assembler': line['assembler'],\n",
    "                                                                                'contigs': int(line['original']['contigs']),\n",
    "                                                                                'basepairs': int(line['original']['basepairs']),\n",
    "                                                                                'max_contig': int(line['original']['max_contig_size']),\n",
    "                                                                                'n50': int(line['original']['N50']),\n",
    "                                                                                'mapped_reads': line['original']['mapped_reads'],\n",
    "                                                                                'Ns': int(line['original']['Ns']),\n",
    "                                                                                'misassembled contigs': line['filtered']['misassembled_contigs'],\n",
    "                                                                                'misassembly events': line['filtered']['misassembly_events'],\n",
    "                                                                                'filtered_contigs': line['filtered']['contigs'],\n",
    "                                                                                'filtered_basepairs': line['filtered']['basepairs'],\n",
    "                                                                                'filtered_n50': line['filtered']['N50'],\n",
    "                                                                                'filtered_Ns': line['filtered']['Ns'],\n",
    "                                                                                'filtered_mapped_reads': line['filtered']['mapped_reads'],\n",
    "                                                                               },\n",
    "                                                                               ignore_index=True)\n",
    "global_pipeline_metrics_df['distribution'] = np.where(global_pipeline_metrics_df['sample'].isin(log_distributed), 'Log', 'Even')\n",
    "global_pipeline_metrics_df['type'] = np.where(global_pipeline_metrics_df['assembler'].isin(genomic_assemblers), 'Genomic', 'Metagenomic')\n",
    "global_pipeline_metrics_df['algorythm'] = np.where(global_pipeline_metrics_df['assembler'].isin(single_kmer), 'Single k-mer De Bruijn graph', 'Multiple k-mer De Bruijn graph')\n",
    "global_pipeline_metrics_df[['contigs','basepairs','max_contig', 'Ns','n50', 'filtered_n50','misassembled contigs', 'misassembly events']] = global_pipeline_metrics_df[['contigs','basepairs','max_contig', 'Ns','n50','filtered_n50','misassembled contigs', 'misassembly events']].apply(pd.to_numeric)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Preview data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Ns                       float64\n",
       "assembler                 object\n",
       "basepairs                float64\n",
       "contigs                  float64\n",
       "filtered_Ns              float64\n",
       "filtered_basepairs       float64\n",
       "filtered_contigs         float64\n",
       "filtered_mapped_reads    float64\n",
       "filtered_n50             float64\n",
       "mapped_reads             float64\n",
       "max_contig               float64\n",
       "misassembled contigs     float64\n",
       "misassembly events       float64\n",
       "n50                      float64\n",
       "run                       object\n",
       "sample                    object\n",
       "distribution              object\n",
       "type                      object\n",
       "algorythm                 object\n",
       "dtype: object"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "global_pipeline_metrics_df.dtypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ns</th>\n",
       "      <th>assembler</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>contigs</th>\n",
       "      <th>filtered_Ns</th>\n",
       "      <th>filtered_basepairs</th>\n",
       "      <th>filtered_contigs</th>\n",
       "      <th>filtered_mapped_reads</th>\n",
       "      <th>filtered_n50</th>\n",
       "      <th>mapped_reads</th>\n",
       "      <th>max_contig</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>n50</th>\n",
       "      <th>run</th>\n",
       "      <th>sample</th>\n",
       "      <th>distribution</th>\n",
       "      <th>type</th>\n",
       "      <th>algorythm</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>147.0</td>\n",
       "      <td>ABySS</td>\n",
       "      <td>91720949.0</td>\n",
       "      <td>497783.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>30657287.0</td>\n",
       "      <td>1196.0</td>\n",
       "      <td>82.019612</td>\n",
       "      <td>52869.0</td>\n",
       "      <td>92.343671</td>\n",
       "      <td>298093.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>139.0</td>\n",
       "      <td>run2</td>\n",
       "      <td>ERR2984773</td>\n",
       "      <td>Even</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>BCALM2</td>\n",
       "      <td>170402348.0</td>\n",
       "      <td>3508016.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8.151058</td>\n",
       "      <td>994.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>run2</td>\n",
       "      <td>ERR2984773</td>\n",
       "      <td>Even</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.0</td>\n",
       "      <td>GATBMiniaPipeline</td>\n",
       "      <td>32809112.0</td>\n",
       "      <td>5397.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30564302.0</td>\n",
       "      <td>761.0</td>\n",
       "      <td>82.840039</td>\n",
       "      <td>113162.0</td>\n",
       "      <td>84.068286</td>\n",
       "      <td>768462.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>102115.0</td>\n",
       "      <td>run2</td>\n",
       "      <td>ERR2984773</td>\n",
       "      <td>Even</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.0</td>\n",
       "      <td>IDBA-UD</td>\n",
       "      <td>39773020.0</td>\n",
       "      <td>20164.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>31562690.0</td>\n",
       "      <td>2654.0</td>\n",
       "      <td>81.596739</td>\n",
       "      <td>29134.0</td>\n",
       "      <td>84.336773</td>\n",
       "      <td>303329.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>20993.0</td>\n",
       "      <td>run2</td>\n",
       "      <td>ERR2984773</td>\n",
       "      <td>Even</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.0</td>\n",
       "      <td>MEGAHIT</td>\n",
       "      <td>41166407.0</td>\n",
       "      <td>20614.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>32021766.0</td>\n",
       "      <td>1447.0</td>\n",
       "      <td>84.138912</td>\n",
       "      <td>149841.0</td>\n",
       "      <td>86.166393</td>\n",
       "      <td>1205839.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>88308.0</td>\n",
       "      <td>run2</td>\n",
       "      <td>ERR2984773</td>\n",
       "      <td>Even</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135</th>\n",
       "      <td>0.0</td>\n",
       "      <td>MINIA</td>\n",
       "      <td>15334799.0</td>\n",
       "      <td>24654.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>12920975.0</td>\n",
       "      <td>2171.0</td>\n",
       "      <td>81.993164</td>\n",
       "      <td>10582.0</td>\n",
       "      <td>91.155743</td>\n",
       "      <td>163543.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7727.0</td>\n",
       "      <td>run1</td>\n",
       "      <td>LHS</td>\n",
       "      <td>Log</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>0.0</td>\n",
       "      <td>SKESA</td>\n",
       "      <td>2957060.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2950393.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>92.094062</td>\n",
       "      <td>351027.0</td>\n",
       "      <td>92.347172</td>\n",
       "      <td>632013.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>351027.0</td>\n",
       "      <td>run1</td>\n",
       "      <td>LHS</td>\n",
       "      <td>Log</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137</th>\n",
       "      <td>0.0</td>\n",
       "      <td>SPAdes</td>\n",
       "      <td>16214591.0</td>\n",
       "      <td>31040.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>13087777.0</td>\n",
       "      <td>1526.0</td>\n",
       "      <td>77.136681</td>\n",
       "      <td>69361.0</td>\n",
       "      <td>99.314996</td>\n",
       "      <td>263645.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>46478.0</td>\n",
       "      <td>run1</td>\n",
       "      <td>LHS</td>\n",
       "      <td>Log</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>0.0</td>\n",
       "      <td>Unicycler</td>\n",
       "      <td>13324149.0</td>\n",
       "      <td>264.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>13315624.0</td>\n",
       "      <td>243.0</td>\n",
       "      <td>96.651358</td>\n",
       "      <td>192888.0</td>\n",
       "      <td>97.201237</td>\n",
       "      <td>847572.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>192888.0</td>\n",
       "      <td>run1</td>\n",
       "      <td>LHS</td>\n",
       "      <td>Log</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>0.0</td>\n",
       "      <td>VelvetOptimiser</td>\n",
       "      <td>22242266.0</td>\n",
       "      <td>185614.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6701756.0</td>\n",
       "      <td>3237.0</td>\n",
       "      <td>2.558256</td>\n",
       "      <td>2177.0</td>\n",
       "      <td>9.675074</td>\n",
       "      <td>17566.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>175.0</td>\n",
       "      <td>run1</td>\n",
       "      <td>LHS</td>\n",
       "      <td>Log</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>140 rows × 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Ns          assembler    basepairs    contigs  filtered_Ns  \\\n",
       "0    147.0              ABySS   91720949.0   497783.0          5.0   \n",
       "1      0.0             BCALM2  170402348.0  3508016.0          0.0   \n",
       "2      0.0  GATBMiniaPipeline   32809112.0     5397.0          0.0   \n",
       "3      0.0            IDBA-UD   39773020.0    20164.0          0.0   \n",
       "4      0.0            MEGAHIT   41166407.0    20614.0          0.0   \n",
       "..     ...                ...          ...        ...          ...   \n",
       "135    0.0              MINIA   15334799.0    24654.0          0.0   \n",
       "136    0.0              SKESA    2957060.0       36.0          0.0   \n",
       "137    0.0             SPAdes   16214591.0    31040.0          0.0   \n",
       "138    0.0          Unicycler   13324149.0      264.0          0.0   \n",
       "139    0.0    VelvetOptimiser   22242266.0   185614.0          0.0   \n",
       "\n",
       "     filtered_basepairs  filtered_contigs  filtered_mapped_reads  \\\n",
       "0            30657287.0            1196.0              82.019612   \n",
       "1                   0.0               0.0               0.000000   \n",
       "2            30564302.0             761.0              82.840039   \n",
       "3            31562690.0            2654.0              81.596739   \n",
       "4            32021766.0            1447.0              84.138912   \n",
       "..                  ...               ...                    ...   \n",
       "135          12920975.0            2171.0              81.993164   \n",
       "136           2950393.0              21.0              92.094062   \n",
       "137          13087777.0            1526.0              77.136681   \n",
       "138          13315624.0             243.0              96.651358   \n",
       "139           6701756.0            3237.0               2.558256   \n",
       "\n",
       "     filtered_n50  mapped_reads  max_contig  misassembled contigs  \\\n",
       "0         52869.0     92.343671    298093.0                   2.0   \n",
       "1             0.0      8.151058       994.0                   0.0   \n",
       "2        113162.0     84.068286    768462.0                   1.0   \n",
       "3         29134.0     84.336773    303329.0                   4.0   \n",
       "4        149841.0     86.166393   1205839.0                  15.0   \n",
       "..            ...           ...         ...                   ...   \n",
       "135       10582.0     91.155743    163543.0                   0.0   \n",
       "136      351027.0     92.347172    632013.0                   0.0   \n",
       "137       69361.0     99.314996    263645.0                   1.0   \n",
       "138      192888.0     97.201237    847572.0                   5.0   \n",
       "139        2177.0      9.675074     17566.0                   0.0   \n",
       "\n",
       "     misassembly events       n50   run      sample distribution         type  \\\n",
       "0                   6.0     139.0  run2  ERR2984773         Even  Metagenomic   \n",
       "1                   0.0      49.0  run2  ERR2984773         Even      Genomic   \n",
       "2                   2.0  102115.0  run2  ERR2984773         Even  Metagenomic   \n",
       "3                   7.0   20993.0  run2  ERR2984773         Even  Metagenomic   \n",
       "4                  32.0   88308.0  run2  ERR2984773         Even  Metagenomic   \n",
       "..                  ...       ...   ...         ...          ...          ...   \n",
       "135                 0.0    7727.0  run1         LHS          Log      Genomic   \n",
       "136                 0.0  351027.0  run1         LHS          Log      Genomic   \n",
       "137                 2.0   46478.0  run1         LHS          Log      Genomic   \n",
       "138                10.0  192888.0  run1         LHS          Log      Genomic   \n",
       "139                 0.0     175.0  run1         LHS          Log  Metagenomic   \n",
       "\n",
       "                          algorythm  \n",
       "0    Multiple k-mer De Bruijn graph  \n",
       "1      Single k-mer De Bruijn graph  \n",
       "2    Multiple k-mer De Bruijn graph  \n",
       "3    Multiple k-mer De Bruijn graph  \n",
       "4    Multiple k-mer De Bruijn graph  \n",
       "..                              ...  \n",
       "135    Single k-mer De Bruijn graph  \n",
       "136  Multiple k-mer De Bruijn graph  \n",
       "137  Multiple k-mer De Bruijn graph  \n",
       "138  Multiple k-mer De Bruijn graph  \n",
       "139  Multiple k-mer De Bruijn graph  \n",
       "\n",
       "[140 rows x 19 columns]"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "global_pipeline_metrics_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Save data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [],
   "source": [
    "global_pipeline_metrics_df.to_csv(\"Tables/Results/Global metrics.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Global statistics per assembler type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [],
   "source": [
    "stats_per_assembler_type = pd.DataFrame()\n",
    "for assembly_type in global_pipeline_metrics_df.type.unique():\n",
    "    row={'Assembler Type': assembly_type}\n",
    "    lala=global_pipeline_metrics_df[global_pipeline_metrics_df.type == assembly_type].describe()\n",
    "    for column in lala.columns:\n",
    "        mean = lala.loc['mean',column]\n",
    "        minimum = lala.loc['min',column]\n",
    "        maximum = lala.loc['max',column]\n",
    "        row[column] = \"{} [{};{}]\".format(round(mean, 2), int(minimum), int(maximum))\n",
    "    stats_per_assembler_type=stats_per_assembler_type.append(row, ignore_index=True)\n",
    "stats_per_assembler_type = stats_per_assembler_type.set_index('Assembler Type')\n",
    "stats_per_assembler_type.to_csv(\"Tables/Results/Global metrics per assembler type.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ns</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>contigs</th>\n",
       "      <th>filtered_Ns</th>\n",
       "      <th>filtered_basepairs</th>\n",
       "      <th>filtered_contigs</th>\n",
       "      <th>filtered_mapped_reads</th>\n",
       "      <th>filtered_n50</th>\n",
       "      <th>mapped_reads</th>\n",
       "      <th>max_contig</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>n50</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler Type</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Metagenomic</th>\n",
       "      <td>768.38 [0;8788]</td>\n",
       "      <td>32228239.44 [13174826;91720949]</td>\n",
       "      <td>61715.1 [640;677860]</td>\n",
       "      <td>590.35 [0;8283]</td>\n",
       "      <td>24113385.45 [13883;32021766]</td>\n",
       "      <td>1228.86 [13;5038]</td>\n",
       "      <td>85.24 [0;107]</td>\n",
       "      <td>102483.75 [1039;239185]</td>\n",
       "      <td>90.7 [1;297]</td>\n",
       "      <td>613615.18 [1157;1205839]</td>\n",
       "      <td>4.08 [0;18]</td>\n",
       "      <td>7.84 [0;36]</td>\n",
       "      <td>87849.68 [85;238611]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic</th>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>46130651.25 [2957060;364761201]</td>\n",
       "      <td>470678.18 [36;8175890]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>22632362.1 [0;32500826]</td>\n",
       "      <td>1715.12 [0;6250]</td>\n",
       "      <td>79.7 [0;99]</td>\n",
       "      <td>95228.97 [0;351027]</td>\n",
       "      <td>85.17 [1;107]</td>\n",
       "      <td>486162.63 [994;1303875]</td>\n",
       "      <td>11.55 [0;572]</td>\n",
       "      <td>23.08 [0;1148]</td>\n",
       "      <td>89248.1 [35;351027]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             Ns                        basepairs  \\\n",
       "Assembler Type                                                     \n",
       "Metagenomic     768.38 [0;8788]  32228239.44 [13174826;91720949]   \n",
       "Genomic               0.0 [0;0]  46130651.25 [2957060;364761201]   \n",
       "\n",
       "                               contigs      filtered_Ns  \\\n",
       "Assembler Type                                            \n",
       "Metagenomic       61715.1 [640;677860]  590.35 [0;8283]   \n",
       "Genomic         470678.18 [36;8175890]        0.0 [0;0]   \n",
       "\n",
       "                          filtered_basepairs   filtered_contigs  \\\n",
       "Assembler Type                                                    \n",
       "Metagenomic     24113385.45 [13883;32021766]  1228.86 [13;5038]   \n",
       "Genomic              22632362.1 [0;32500826]   1715.12 [0;6250]   \n",
       "\n",
       "               filtered_mapped_reads             filtered_n50   mapped_reads  \\\n",
       "Assembler Type                                                                 \n",
       "Metagenomic            85.24 [0;107]  102483.75 [1039;239185]   90.7 [1;297]   \n",
       "Genomic                  79.7 [0;99]      95228.97 [0;351027]  85.17 [1;107]   \n",
       "\n",
       "                              max_contig misassembled contigs  \\\n",
       "Assembler Type                                                  \n",
       "Metagenomic     613615.18 [1157;1205839]          4.08 [0;18]   \n",
       "Genomic          486162.63 [994;1303875]        11.55 [0;572]   \n",
       "\n",
       "               misassembly events                   n50  \n",
       "Assembler Type                                           \n",
       "Metagenomic           7.84 [0;36]  87849.68 [85;238611]  \n",
       "Genomic            23.08 [0;1148]   89248.1 [35;351027]  "
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "stats_per_assembler_type"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Global stats per assembly algorithm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [],
   "source": [
    "stats_per_assembler_algo = pd.DataFrame()\n",
    "for assembly_algo in global_pipeline_metrics_df.algorythm.unique():\n",
    "    row={'Assembler Algorithm': assembly_algo}\n",
    "    lala=global_pipeline_metrics_df[global_pipeline_metrics_df.algorythm == assembly_algo].describe()\n",
    "    for column in lala.columns:\n",
    "        mean = lala.loc['mean',column]\n",
    "        minimum = lala.loc['min',column]\n",
    "        maximum = lala.loc['max',column]\n",
    "        row[column] = \"{} [{};{}]\".format(round(mean, 2), int(minimum), int(maximum))\n",
    "    stats_per_assembler_algo=stats_per_assembler_algo.append(row, ignore_index=True)\n",
    "stats_per_assembler_algo = stats_per_assembler_algo.set_index('Assembler Algorithm')\n",
    "stats_per_assembler_algo.to_csv(\"Tables/Results/Global metrics per assembler algorithm.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ns</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>contigs</th>\n",
       "      <th>filtered_Ns</th>\n",
       "      <th>filtered_basepairs</th>\n",
       "      <th>filtered_contigs</th>\n",
       "      <th>filtered_mapped_reads</th>\n",
       "      <th>filtered_n50</th>\n",
       "      <th>mapped_reads</th>\n",
       "      <th>max_contig</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>n50</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler Algorithm</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Multiple k-mer De Bruijn graph</th>\n",
       "      <td>529.91 [0;8788]</td>\n",
       "      <td>30716745.31 [2957060;91720949]</td>\n",
       "      <td>44688.53 [36;677860]</td>\n",
       "      <td>407.14 [0;8283]</td>\n",
       "      <td>24527772.38 [13883;32500826]</td>\n",
       "      <td>1143.22 [13;5038]</td>\n",
       "      <td>87.09 [0;107]</td>\n",
       "      <td>117011.65 [1039;351027]</td>\n",
       "      <td>91.8 [1;297]</td>\n",
       "      <td>652239.76 [1157;1303875]</td>\n",
       "      <td>8.75 [0;572]</td>\n",
       "      <td>17.28 [0;1148]</td>\n",
       "      <td>104650.93 [85;351027]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Single k-mer De Bruijn graph</th>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>74289823.92 [15334799;364761201]</td>\n",
       "      <td>1166417.88 [15001;8175890]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>18407956.92 [0;28279162]</td>\n",
       "      <td>2858.46 [0;6250]</td>\n",
       "      <td>62.45 [0;97]</td>\n",
       "      <td>14128.62 [0;33550]</td>\n",
       "      <td>71.58 [1;99]</td>\n",
       "      <td>108298.33 [994;177038]</td>\n",
       "      <td>0.17 [0;2]</td>\n",
       "      <td>0.29 [0;4]</td>\n",
       "      <td>10139.67 [35;26474]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             Ns  \\\n",
       "Assembler Algorithm                               \n",
       "Multiple k-mer De Bruijn graph  529.91 [0;8788]   \n",
       "Single k-mer De Bruijn graph          0.0 [0;0]   \n",
       "\n",
       "                                                       basepairs  \\\n",
       "Assembler Algorithm                                                \n",
       "Multiple k-mer De Bruijn graph    30716745.31 [2957060;91720949]   \n",
       "Single k-mer De Bruijn graph    74289823.92 [15334799;364761201]   \n",
       "\n",
       "                                                   contigs      filtered_Ns  \\\n",
       "Assembler Algorithm                                                           \n",
       "Multiple k-mer De Bruijn graph        44688.53 [36;677860]  407.14 [0;8283]   \n",
       "Single k-mer De Bruijn graph    1166417.88 [15001;8175890]        0.0 [0;0]   \n",
       "\n",
       "                                          filtered_basepairs  \\\n",
       "Assembler Algorithm                                            \n",
       "Multiple k-mer De Bruijn graph  24527772.38 [13883;32500826]   \n",
       "Single k-mer De Bruijn graph        18407956.92 [0;28279162]   \n",
       "\n",
       "                                 filtered_contigs filtered_mapped_reads  \\\n",
       "Assembler Algorithm                                                       \n",
       "Multiple k-mer De Bruijn graph  1143.22 [13;5038]         87.09 [0;107]   \n",
       "Single k-mer De Bruijn graph     2858.46 [0;6250]          62.45 [0;97]   \n",
       "\n",
       "                                           filtered_n50  mapped_reads  \\\n",
       "Assembler Algorithm                                                     \n",
       "Multiple k-mer De Bruijn graph  117011.65 [1039;351027]  91.8 [1;297]   \n",
       "Single k-mer De Bruijn graph         14128.62 [0;33550]  71.58 [1;99]   \n",
       "\n",
       "                                              max_contig misassembled contigs  \\\n",
       "Assembler Algorithm                                                             \n",
       "Multiple k-mer De Bruijn graph  652239.76 [1157;1303875]         8.75 [0;572]   \n",
       "Single k-mer De Bruijn graph      108298.33 [994;177038]           0.17 [0;2]   \n",
       "\n",
       "                               misassembly events                    n50  \n",
       "Assembler Algorithm                                                       \n",
       "Multiple k-mer De Bruijn graph     17.28 [0;1148]  104650.93 [85;351027]  \n",
       "Single k-mer De Bruijn graph           0.29 [0;4]    10139.67 [35;26474]  "
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "stats_per_assembler_algo"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Global stats per assembler"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [],
   "source": [
    "stats_per_assembler = pd.DataFrame()\n",
    "for assembler in global_pipeline_metrics_df.assembler.unique():\n",
    "    row={'Assembler': assembler, \n",
    "         'Type':global_pipeline_metrics_df[global_pipeline_metrics_df.assembler == assembler].type.unique()[0],\n",
    "         'Algorithm': global_pipeline_metrics_df[global_pipeline_metrics_df.assembler == assembler].algorythm.unique()[0]}\n",
    "    lala=global_pipeline_metrics_df[global_pipeline_metrics_df.assembler == assembler].describe()\n",
    "    for column in lala.columns:\n",
    "        mean = lala.loc['mean',column]\n",
    "        minimum = lala.loc['min',column]\n",
    "        maximum = lala.loc['max',column]\n",
    "        row[column] = \"{} [{};{}]\".format(round(mean, 2), int(minimum), int(maximum))\n",
    "    stats_per_assembler=stats_per_assembler.append(row, ignore_index=True)\n",
    "stats_per_assembler = stats_per_assembler.set_index('Assembler')\n",
    "stats_per_assembler.to_csv(\"Tables/Results/Global metrics per assembler.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>contigs</th>\n",
       "      <th>filtered_Ns</th>\n",
       "      <th>filtered_basepairs</th>\n",
       "      <th>filtered_contigs</th>\n",
       "      <th>filtered_mapped_reads</th>\n",
       "      <th>filtered_n50</th>\n",
       "      <th>mapped_reads</th>\n",
       "      <th>max_contig</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>n50</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>1594.1 [9;3884]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>48050192.0 [20132631;91720949]</td>\n",
       "      <td>182966.7 [4769;677860]</td>\n",
       "      <td>923.0 [0;2328]</td>\n",
       "      <td>26320647.2 [3150980;30988200]</td>\n",
       "      <td>1093.3 [356;1825]</td>\n",
       "      <td>98.09 [79;107]</td>\n",
       "      <td>100040.5 [18568;192306]</td>\n",
       "      <td>122.57 [92;297]</td>\n",
       "      <td>691710.5 [88665;952618]</td>\n",
       "      <td>3.2 [0;8]</td>\n",
       "      <td>5.7 [0;14]</td>\n",
       "      <td>64240.5 [101;124246]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>118585717.58 [19635512;364761201]</td>\n",
       "      <td>2301095.5 [25377;8175890]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>12317748.75 [0;28174748]</td>\n",
       "      <td>2564.5 [0;6250]</td>\n",
       "      <td>38.02 [0;97]</td>\n",
       "      <td>8752.33 [0;31436]</td>\n",
       "      <td>51.53 [1;98]</td>\n",
       "      <td>64194.08 [994;163543]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>6086.08 [35;18682]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>27162999.67 [13638625;32810830]</td>\n",
       "      <td>2565.25 [693;6777]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>26216699.17 [13507672;30564396]</td>\n",
       "      <td>690.0 [219;814]</td>\n",
       "      <td>93.95 [82;98]</td>\n",
       "      <td>108613.08 [90988;140561]</td>\n",
       "      <td>94.99 [84;100]</td>\n",
       "      <td>692934.42 [542997;768585]</td>\n",
       "      <td>1.5 [0;3]</td>\n",
       "      <td>3.0 [0;6]</td>\n",
       "      <td>102311.75 [56759;136741]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>29452731.0 [13791669;39779384]</td>\n",
       "      <td>7388.92 [1025;20194]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>26691259.58 [13601126;31562690]</td>\n",
       "      <td>1592.92 [261;2725]</td>\n",
       "      <td>93.16 [81;99]</td>\n",
       "      <td>66708.58 [29134;113853]</td>\n",
       "      <td>94.88 [84;100]</td>\n",
       "      <td>493485.17 [303329;632213]</td>\n",
       "      <td>9.17 [2;18]</td>\n",
       "      <td>16.92 [2;36]</td>\n",
       "      <td>58013.67 [20993;105808]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>29973124.08 [13843596;41166407]</td>\n",
       "      <td>7000.17 [767;20614]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>27042637.5 [13671964;32021766]</td>\n",
       "      <td>917.75 [172;3044]</td>\n",
       "      <td>95.43 [84;100]</td>\n",
       "      <td>174773.25 [149841;210993]</td>\n",
       "      <td>96.57 [86;101]</td>\n",
       "      <td>858411.0 [632250;1205839]</td>\n",
       "      <td>8.67 [0;15]</td>\n",
       "      <td>18.0 [0;32]</td>\n",
       "      <td>147653.58 [82249;203548]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>23684949.7 [13174826;30381792]</td>\n",
       "      <td>7533.5 [882;19361]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>19665016.4 [1599587;30264953]</td>\n",
       "      <td>1325.7 [712;5038]</td>\n",
       "      <td>67.41 [3;98]</td>\n",
       "      <td>63175.6 [1175;106637]</td>\n",
       "      <td>75.53 [25;99]</td>\n",
       "      <td>283385.4 [2964;468375]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>62261.0 [658;106383]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>32266278.08 [14157309;48964620]</td>\n",
       "      <td>18684.67 [640;61644]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>27034872.42 [13672885;31967753]</td>\n",
       "      <td>912.83 [139;3392]</td>\n",
       "      <td>94.58 [83;100]</td>\n",
       "      <td>179333.08 [87983;239185]</td>\n",
       "      <td>95.56 [85;100]</td>\n",
       "      <td>806147.42 [473277;1036942]</td>\n",
       "      <td>1.67 [1;3]</td>\n",
       "      <td>4.08 [3;7]</td>\n",
       "      <td>157272.92 [3674;238611]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>29993930.25 [15334799;41187744]</td>\n",
       "      <td>31740.25 [15001;68012]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>24498165.08 [12920975;28279162]</td>\n",
       "      <td>3152.42 [762;3983]</td>\n",
       "      <td>86.87 [74;97]</td>\n",
       "      <td>19504.92 [10582;33550]</td>\n",
       "      <td>91.63 [80;99]</td>\n",
       "      <td>152402.58 [117945;177038]</td>\n",
       "      <td>0.33 [0;2]</td>\n",
       "      <td>0.58 [0;4]</td>\n",
       "      <td>14193.25 [7060;26474]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>24362387.0 [2957060;30430077]</td>\n",
       "      <td>2766.75 [36;9514]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>23629742.75 [2950393;30211740]</td>\n",
       "      <td>1227.67 [21;2889]</td>\n",
       "      <td>89.47 [62;97]</td>\n",
       "      <td>78840.67 [2148;351027]</td>\n",
       "      <td>91.75 [78;98]</td>\n",
       "      <td>487176.67 [29155;632013]</td>\n",
       "      <td>5.75 [0;12]</td>\n",
       "      <td>11.17 [0;23]</td>\n",
       "      <td>77687.25 [1155;351027]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>32022403.0 [15606257;46535940]</td>\n",
       "      <td>17112.17 [1140;39819]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>27143287.25 [13087777;32500826]</td>\n",
       "      <td>1288.92 [227;4999]</td>\n",
       "      <td>90.97 [75;99]</td>\n",
       "      <td>149985.25 [55465;208991]</td>\n",
       "      <td>96.85 [85;107]</td>\n",
       "      <td>756751.58 [263645;850176]</td>\n",
       "      <td>50.75 [0;572]</td>\n",
       "      <td>101.83 [0;1148]</td>\n",
       "      <td>130334.75 [1972;206515]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>25688818.42 [9747587;30592960]</td>\n",
       "      <td>676.25 [185;936]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>25572866.67 [9725301;30432114]</td>\n",
       "      <td>342.08 [112;443]</td>\n",
       "      <td>93.17 [81;98]</td>\n",
       "      <td>219061.67 [191424;256884]</td>\n",
       "      <td>94.11 [82;99]</td>\n",
       "      <td>970288.25 [847534;1303875]</td>\n",
       "      <td>0.92 [0;5]</td>\n",
       "      <td>1.83 [0;10]</td>\n",
       "      <td>217939.17 [191424;256884]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>3794.08 [0;8788]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>36220512.0 [19638608;57130377]</td>\n",
       "      <td>217044.83 [10924;652127]</td>\n",
       "      <td>3166.5 [0;8283]</td>\n",
       "      <td>15449048.0 [13883;28485306]</td>\n",
       "      <td>2063.08 [13;3237]</td>\n",
       "      <td>53.24 [0;93]</td>\n",
       "      <td>17783.58 [1039;34664]</td>\n",
       "      <td>57.59 [1;97]</td>\n",
       "      <td>427209.92 [1157;847809]</td>\n",
       "      <td>3.5 [0;8]</td>\n",
       "      <td>5.5 [0;11]</td>\n",
       "      <td>14994.67 [85;30453]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm                Ns  \\\n",
       "Assembler                                                             \n",
       "ABySS              Multiple k-mer De Bruijn graph   1594.1 [9;3884]   \n",
       "BCALM2               Single k-mer De Bruijn graph         0.0 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph         0.0 [0;0]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph         0.0 [0;0]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph         0.0 [0;0]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph         0.0 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph         0.0 [0;0]   \n",
       "MINIA                Single k-mer De Bruijn graph         0.0 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph         0.0 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph         0.0 [0;0]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph         0.0 [0;0]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph  3794.08 [0;8788]   \n",
       "\n",
       "                          Type                          basepairs  \\\n",
       "Assembler                                                           \n",
       "ABySS              Metagenomic     48050192.0 [20132631;91720949]   \n",
       "BCALM2                 Genomic  118585717.58 [19635512;364761201]   \n",
       "GATBMiniaPipeline  Metagenomic    27162999.67 [13638625;32810830]   \n",
       "IDBA-UD            Metagenomic     29452731.0 [13791669;39779384]   \n",
       "MEGAHIT            Metagenomic    29973124.08 [13843596;41166407]   \n",
       "MetaHipMer2        Metagenomic     23684949.7 [13174826;30381792]   \n",
       "metaSPAdes         Metagenomic    32266278.08 [14157309;48964620]   \n",
       "MINIA                  Genomic    29993930.25 [15334799;41187744]   \n",
       "SKESA                  Genomic      24362387.0 [2957060;30430077]   \n",
       "SPAdes                 Genomic     32022403.0 [15606257;46535940]   \n",
       "Unicycler              Genomic     25688818.42 [9747587;30592960]   \n",
       "VelvetOptimiser    Metagenomic     36220512.0 [19638608;57130377]   \n",
       "\n",
       "                                     contigs      filtered_Ns  \\\n",
       "Assembler                                                       \n",
       "ABySS                 182966.7 [4769;677860]   923.0 [0;2328]   \n",
       "BCALM2             2301095.5 [25377;8175890]        0.0 [0;0]   \n",
       "GATBMiniaPipeline         2565.25 [693;6777]        0.0 [0;0]   \n",
       "IDBA-UD                 7388.92 [1025;20194]        0.0 [0;0]   \n",
       "MEGAHIT                  7000.17 [767;20614]        0.0 [0;0]   \n",
       "MetaHipMer2               7533.5 [882;19361]        0.0 [0;0]   \n",
       "metaSPAdes              18684.67 [640;61644]        0.0 [0;0]   \n",
       "MINIA                 31740.25 [15001;68012]        0.0 [0;0]   \n",
       "SKESA                      2766.75 [36;9514]        0.0 [0;0]   \n",
       "SPAdes                 17112.17 [1140;39819]        0.0 [0;0]   \n",
       "Unicycler                   676.25 [185;936]        0.0 [0;0]   \n",
       "VelvetOptimiser     217044.83 [10924;652127]  3166.5 [0;8283]   \n",
       "\n",
       "                                filtered_basepairs    filtered_contigs  \\\n",
       "Assembler                                                                \n",
       "ABySS                26320647.2 [3150980;30988200]   1093.3 [356;1825]   \n",
       "BCALM2                    12317748.75 [0;28174748]     2564.5 [0;6250]   \n",
       "GATBMiniaPipeline  26216699.17 [13507672;30564396]     690.0 [219;814]   \n",
       "IDBA-UD            26691259.58 [13601126;31562690]  1592.92 [261;2725]   \n",
       "MEGAHIT             27042637.5 [13671964;32021766]   917.75 [172;3044]   \n",
       "MetaHipMer2          19665016.4 [1599587;30264953]   1325.7 [712;5038]   \n",
       "metaSPAdes         27034872.42 [13672885;31967753]   912.83 [139;3392]   \n",
       "MINIA              24498165.08 [12920975;28279162]  3152.42 [762;3983]   \n",
       "SKESA               23629742.75 [2950393;30211740]   1227.67 [21;2889]   \n",
       "SPAdes             27143287.25 [13087777;32500826]  1288.92 [227;4999]   \n",
       "Unicycler           25572866.67 [9725301;30432114]    342.08 [112;443]   \n",
       "VelvetOptimiser        15449048.0 [13883;28485306]   2063.08 [13;3237]   \n",
       "\n",
       "                  filtered_mapped_reads               filtered_n50  \\\n",
       "Assembler                                                            \n",
       "ABySS                    98.09 [79;107]    100040.5 [18568;192306]   \n",
       "BCALM2                     38.02 [0;97]          8752.33 [0;31436]   \n",
       "GATBMiniaPipeline         93.95 [82;98]   108613.08 [90988;140561]   \n",
       "IDBA-UD                   93.16 [81;99]    66708.58 [29134;113853]   \n",
       "MEGAHIT                  95.43 [84;100]  174773.25 [149841;210993]   \n",
       "MetaHipMer2                67.41 [3;98]      63175.6 [1175;106637]   \n",
       "metaSPAdes               94.58 [83;100]   179333.08 [87983;239185]   \n",
       "MINIA                     86.87 [74;97]     19504.92 [10582;33550]   \n",
       "SKESA                     89.47 [62;97]     78840.67 [2148;351027]   \n",
       "SPAdes                    90.97 [75;99]   149985.25 [55465;208991]   \n",
       "Unicycler                 93.17 [81;98]  219061.67 [191424;256884]   \n",
       "VelvetOptimiser            53.24 [0;93]      17783.58 [1039;34664]   \n",
       "\n",
       "                      mapped_reads                  max_contig  \\\n",
       "Assembler                                                        \n",
       "ABySS              122.57 [92;297]     691710.5 [88665;952618]   \n",
       "BCALM2                51.53 [1;98]       64194.08 [994;163543]   \n",
       "GATBMiniaPipeline   94.99 [84;100]   692934.42 [542997;768585]   \n",
       "IDBA-UD             94.88 [84;100]   493485.17 [303329;632213]   \n",
       "MEGAHIT             96.57 [86;101]   858411.0 [632250;1205839]   \n",
       "MetaHipMer2          75.53 [25;99]      283385.4 [2964;468375]   \n",
       "metaSPAdes          95.56 [85;100]  806147.42 [473277;1036942]   \n",
       "MINIA                91.63 [80;99]   152402.58 [117945;177038]   \n",
       "SKESA                91.75 [78;98]    487176.67 [29155;632013]   \n",
       "SPAdes              96.85 [85;107]   756751.58 [263645;850176]   \n",
       "Unicycler            94.11 [82;99]  970288.25 [847534;1303875]   \n",
       "VelvetOptimiser       57.59 [1;97]     427209.92 [1157;847809]   \n",
       "\n",
       "                  misassembled contigs misassembly events  \\\n",
       "Assembler                                                   \n",
       "ABySS                        3.2 [0;8]         5.7 [0;14]   \n",
       "BCALM2                       0.0 [0;0]          0.0 [0;0]   \n",
       "GATBMiniaPipeline            1.5 [0;3]          3.0 [0;6]   \n",
       "IDBA-UD                    9.17 [2;18]       16.92 [2;36]   \n",
       "MEGAHIT                    8.67 [0;15]        18.0 [0;32]   \n",
       "MetaHipMer2                  0.0 [0;0]          0.0 [0;0]   \n",
       "metaSPAdes                  1.67 [1;3]         4.08 [3;7]   \n",
       "MINIA                       0.33 [0;2]         0.58 [0;4]   \n",
       "SKESA                      5.75 [0;12]       11.17 [0;23]   \n",
       "SPAdes                   50.75 [0;572]    101.83 [0;1148]   \n",
       "Unicycler                   0.92 [0;5]        1.83 [0;10]   \n",
       "VelvetOptimiser              3.5 [0;8]         5.5 [0;11]   \n",
       "\n",
       "                                         n50  \n",
       "Assembler                                     \n",
       "ABySS                   64240.5 [101;124246]  \n",
       "BCALM2                    6086.08 [35;18682]  \n",
       "GATBMiniaPipeline   102311.75 [56759;136741]  \n",
       "IDBA-UD              58013.67 [20993;105808]  \n",
       "MEGAHIT             147653.58 [82249;203548]  \n",
       "MetaHipMer2             62261.0 [658;106383]  \n",
       "metaSPAdes           157272.92 [3674;238611]  \n",
       "MINIA                  14193.25 [7060;26474]  \n",
       "SKESA                 77687.25 [1155;351027]  \n",
       "SPAdes               130334.75 [1972;206515]  \n",
       "Unicycler          217939.17 [191424;256884]  \n",
       "VelvetOptimiser          14994.67 [85;30453]  "
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "stats_per_assembler"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Global stats per distribution"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Even\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>contigs</th>\n",
       "      <th>filtered_Ns</th>\n",
       "      <th>filtered_basepairs</th>\n",
       "      <th>filtered_contigs</th>\n",
       "      <th>filtered_mapped_reads</th>\n",
       "      <th>filtered_n50</th>\n",
       "      <th>mapped_reads</th>\n",
       "      <th>max_contig</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>n50</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>1823.12 [147;3884]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>47705094.62 [31503127;91720949]</td>\n",
       "      <td>138398.25 [4769;497797]</td>\n",
       "      <td>1103.75 [5;2328]</td>\n",
       "      <td>30810219.75 [30632246;30988200]</td>\n",
       "      <td>1170.62 [497;1825]</td>\n",
       "      <td>100.22 [81;107]</td>\n",
       "      <td>98691.38 [52698;138706]</td>\n",
       "      <td>103.33 [92;108]</td>\n",
       "      <td>747140.75 [297881;952618]</td>\n",
       "      <td>4.0 [2;8]</td>\n",
       "      <td>7.12 [3;14]</td>\n",
       "      <td>79091.88 [139;124246]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>80340494.0 [31047581;170402348]</td>\n",
       "      <td>1261858.33 [25377;3508016]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>14860681.0 [0;28174748]</td>\n",
       "      <td>3275.0 [0;6250]</td>\n",
       "      <td>39.88 [0;92]</td>\n",
       "      <td>7918.67 [0;20700]</td>\n",
       "      <td>56.66 [8;96]</td>\n",
       "      <td>66942.33 [994;161812]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>6029.67 [49;17395]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>31253352.33 [30440762;32810830]</td>\n",
       "      <td>2480.44 [943;5400]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>30393672.67 [30273320;30564396]</td>\n",
       "      <td>744.0 [657;814]</td>\n",
       "      <td>93.59 [82;98]</td>\n",
       "      <td>108898.33 [90988;122545]</td>\n",
       "      <td>94.75 [84;100]</td>\n",
       "      <td>723087.33 [632215;768585]</td>\n",
       "      <td>1.67 [1;3]</td>\n",
       "      <td>3.0 [1;6]</td>\n",
       "      <td>104806.0 [90346;121957]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>33525497.0 [30390566;39779384]</td>\n",
       "      <td>7944.44 [1822;20194]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>30613010.56 [30130413;31562690]</td>\n",
       "      <td>1729.0 [1254;2654]</td>\n",
       "      <td>92.47 [81;98]</td>\n",
       "      <td>52473.33 [29134;65108]</td>\n",
       "      <td>94.54 [84;100]</td>\n",
       "      <td>452420.67 [303329;585610]</td>\n",
       "      <td>11.44 [4;18]</td>\n",
       "      <td>21.22 [7;36]</td>\n",
       "      <td>48829.89 [20993;63725]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>34103641.33 [30545419;41166407]</td>\n",
       "      <td>7469.0 [814;20614]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>30953572.33 [30418624;32021766]</td>\n",
       "      <td>830.0 [518;1447]</td>\n",
       "      <td>94.89 [84;100]</td>\n",
       "      <td>167083.67 [149841;179870]</td>\n",
       "      <td>96.22 [86;101]</td>\n",
       "      <td>868917.67 [632274;1205839]</td>\n",
       "      <td>11.33 [9;15]</td>\n",
       "      <td>23.56 [18;32]</td>\n",
       "      <td>146076.33 [88308;178381]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>24646614.11 [13174826;30381792]</td>\n",
       "      <td>7041.0 [882;19361]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>20709564.33 [1599587;30264953]</td>\n",
       "      <td>913.22 [712;1310]</td>\n",
       "      <td>66.63 [3;98]</td>\n",
       "      <td>69954.0 [1175;106637]</td>\n",
       "      <td>74.44 [25;99]</td>\n",
       "      <td>313232.67 [2964;468375]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>69010.56 [658;106383]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>36633480.0 [30453778;48964620]</td>\n",
       "      <td>21015.0 [640;61644]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>30917818.67 [30383186;31967753]</td>\n",
       "      <td>804.33 [359;1689]</td>\n",
       "      <td>94.17 [83;100]</td>\n",
       "      <td>180893.67 [95101;238611]</td>\n",
       "      <td>95.36 [85;100]</td>\n",
       "      <td>809315.0 [473277;1036942]</td>\n",
       "      <td>1.67 [1;2]</td>\n",
       "      <td>4.0 [3;6]</td>\n",
       "      <td>163304.0 [42332;238611]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>34159156.67 [30641333;41187744]</td>\n",
       "      <td>33769.78 [16570;68012]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>28234093.56 [28205937;28279162]</td>\n",
       "      <td>3714.44 [3574;3983]</td>\n",
       "      <td>86.52 [74;92]</td>\n",
       "      <td>19098.78 [15817;20782]</td>\n",
       "      <td>91.29 [80;97]</td>\n",
       "      <td>147189.67 [117945;161812]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>14292.56 [7060;17967]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>30263322.33 [30110758;30430077]</td>\n",
       "      <td>1647.67 [1269;2322]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>30037221.0 [29835112;30211740]</td>\n",
       "      <td>1154.33 [846;1726]</td>\n",
       "      <td>91.9 [80;97]</td>\n",
       "      <td>65058.67 [43168;77870]</td>\n",
       "      <td>92.76 [81;98]</td>\n",
       "      <td>569520.0 [446163;632013]</td>\n",
       "      <td>6.33 [0;10]</td>\n",
       "      <td>12.33 [0;19]</td>\n",
       "      <td>64292.67 [42115;76924]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>35881180.33 [30442718;46535940]</td>\n",
       "      <td>14529.0 [1140;39819]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>31043946.33 [30276227;32500826]</td>\n",
       "      <td>968.33 [412;2049]</td>\n",
       "      <td>93.29 [82;99]</td>\n",
       "      <td>170923.67 [99299;208991]</td>\n",
       "      <td>95.02 [85;100]</td>\n",
       "      <td>815228.0 [745370;850176]</td>\n",
       "      <td>4.0 [0;9]</td>\n",
       "      <td>8.0 [0;18]</td>\n",
       "      <td>154762.67 [53292;206515]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>30386095.0 [30263458;30592960]</td>\n",
       "      <td>829.0 [656;936]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>30238046.0 [30093081;30432114]</td>\n",
       "      <td>400.67 [334;443]</td>\n",
       "      <td>92.78 [81;98]</td>\n",
       "      <td>215142.33 [191424;245050]</td>\n",
       "      <td>93.83 [82;99]</td>\n",
       "      <td>1011197.0 [847534;1303875]</td>\n",
       "      <td>0.67 [0;2]</td>\n",
       "      <td>1.33 [0;4]</td>\n",
       "      <td>213645.67 [191424;245050]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>4510.78 [105;8788]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>38311849.11 [30526291;57130377]</td>\n",
       "      <td>202648.22 [10924;652127]</td>\n",
       "      <td>4222.0 [0;8283]</td>\n",
       "      <td>18993730.33 [13883;28485306]</td>\n",
       "      <td>2018.22 [13;3035]</td>\n",
       "      <td>61.81 [0;93]</td>\n",
       "      <td>23014.78 [1039;34664]</td>\n",
       "      <td>65.2 [1;97]</td>\n",
       "      <td>565656.67 [1157;847809]</td>\n",
       "      <td>4.67 [0;8]</td>\n",
       "      <td>7.33 [0;11]</td>\n",
       "      <td>19905.89 [85;30453]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm                  Ns  \\\n",
       "Assembler                                                               \n",
       "ABySS              Multiple k-mer De Bruijn graph  1823.12 [147;3884]   \n",
       "BCALM2               Single k-mer De Bruijn graph           0.0 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "MINIA                Single k-mer De Bruijn graph           0.0 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph           0.0 [0;0]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph  4510.78 [105;8788]   \n",
       "\n",
       "                          Type                        basepairs  \\\n",
       "Assembler                                                         \n",
       "ABySS              Metagenomic  47705094.62 [31503127;91720949]   \n",
       "BCALM2                 Genomic  80340494.0 [31047581;170402348]   \n",
       "GATBMiniaPipeline  Metagenomic  31253352.33 [30440762;32810830]   \n",
       "IDBA-UD            Metagenomic   33525497.0 [30390566;39779384]   \n",
       "MEGAHIT            Metagenomic  34103641.33 [30545419;41166407]   \n",
       "MetaHipMer2        Metagenomic  24646614.11 [13174826;30381792]   \n",
       "metaSPAdes         Metagenomic   36633480.0 [30453778;48964620]   \n",
       "MINIA                  Genomic  34159156.67 [30641333;41187744]   \n",
       "SKESA                  Genomic  30263322.33 [30110758;30430077]   \n",
       "SPAdes                 Genomic  35881180.33 [30442718;46535940]   \n",
       "Unicycler              Genomic   30386095.0 [30263458;30592960]   \n",
       "VelvetOptimiser    Metagenomic  38311849.11 [30526291;57130377]   \n",
       "\n",
       "                                      contigs       filtered_Ns  \\\n",
       "Assembler                                                         \n",
       "ABySS                 138398.25 [4769;497797]  1103.75 [5;2328]   \n",
       "BCALM2             1261858.33 [25377;3508016]         0.0 [0;0]   \n",
       "GATBMiniaPipeline          2480.44 [943;5400]         0.0 [0;0]   \n",
       "IDBA-UD                  7944.44 [1822;20194]         0.0 [0;0]   \n",
       "MEGAHIT                    7469.0 [814;20614]         0.0 [0;0]   \n",
       "MetaHipMer2                7041.0 [882;19361]         0.0 [0;0]   \n",
       "metaSPAdes                21015.0 [640;61644]         0.0 [0;0]   \n",
       "MINIA                  33769.78 [16570;68012]         0.0 [0;0]   \n",
       "SKESA                     1647.67 [1269;2322]         0.0 [0;0]   \n",
       "SPAdes                   14529.0 [1140;39819]         0.0 [0;0]   \n",
       "Unicycler                     829.0 [656;936]         0.0 [0;0]   \n",
       "VelvetOptimiser      202648.22 [10924;652127]   4222.0 [0;8283]   \n",
       "\n",
       "                                filtered_basepairs     filtered_contigs  \\\n",
       "Assembler                                                                 \n",
       "ABySS              30810219.75 [30632246;30988200]   1170.62 [497;1825]   \n",
       "BCALM2                     14860681.0 [0;28174748]      3275.0 [0;6250]   \n",
       "GATBMiniaPipeline  30393672.67 [30273320;30564396]      744.0 [657;814]   \n",
       "IDBA-UD            30613010.56 [30130413;31562690]   1729.0 [1254;2654]   \n",
       "MEGAHIT            30953572.33 [30418624;32021766]     830.0 [518;1447]   \n",
       "MetaHipMer2         20709564.33 [1599587;30264953]    913.22 [712;1310]   \n",
       "metaSPAdes         30917818.67 [30383186;31967753]    804.33 [359;1689]   \n",
       "MINIA              28234093.56 [28205937;28279162]  3714.44 [3574;3983]   \n",
       "SKESA               30037221.0 [29835112;30211740]   1154.33 [846;1726]   \n",
       "SPAdes             31043946.33 [30276227;32500826]    968.33 [412;2049]   \n",
       "Unicycler           30238046.0 [30093081;30432114]     400.67 [334;443]   \n",
       "VelvetOptimiser       18993730.33 [13883;28485306]    2018.22 [13;3035]   \n",
       "\n",
       "                  filtered_mapped_reads               filtered_n50  \\\n",
       "Assembler                                                            \n",
       "ABySS                   100.22 [81;107]    98691.38 [52698;138706]   \n",
       "BCALM2                     39.88 [0;92]          7918.67 [0;20700]   \n",
       "GATBMiniaPipeline         93.59 [82;98]   108898.33 [90988;122545]   \n",
       "IDBA-UD                   92.47 [81;98]     52473.33 [29134;65108]   \n",
       "MEGAHIT                  94.89 [84;100]  167083.67 [149841;179870]   \n",
       "MetaHipMer2                66.63 [3;98]      69954.0 [1175;106637]   \n",
       "metaSPAdes               94.17 [83;100]   180893.67 [95101;238611]   \n",
       "MINIA                     86.52 [74;92]     19098.78 [15817;20782]   \n",
       "SKESA                      91.9 [80;97]     65058.67 [43168;77870]   \n",
       "SPAdes                    93.29 [82;99]   170923.67 [99299;208991]   \n",
       "Unicycler                 92.78 [81;98]  215142.33 [191424;245050]   \n",
       "VelvetOptimiser            61.81 [0;93]      23014.78 [1039;34664]   \n",
       "\n",
       "                      mapped_reads                  max_contig  \\\n",
       "Assembler                                                        \n",
       "ABySS              103.33 [92;108]   747140.75 [297881;952618]   \n",
       "BCALM2                56.66 [8;96]       66942.33 [994;161812]   \n",
       "GATBMiniaPipeline   94.75 [84;100]   723087.33 [632215;768585]   \n",
       "IDBA-UD             94.54 [84;100]   452420.67 [303329;585610]   \n",
       "MEGAHIT             96.22 [86;101]  868917.67 [632274;1205839]   \n",
       "MetaHipMer2          74.44 [25;99]     313232.67 [2964;468375]   \n",
       "metaSPAdes          95.36 [85;100]   809315.0 [473277;1036942]   \n",
       "MINIA                91.29 [80;97]   147189.67 [117945;161812]   \n",
       "SKESA                92.76 [81;98]    569520.0 [446163;632013]   \n",
       "SPAdes              95.02 [85;100]    815228.0 [745370;850176]   \n",
       "Unicycler            93.83 [82;99]  1011197.0 [847534;1303875]   \n",
       "VelvetOptimiser        65.2 [1;97]     565656.67 [1157;847809]   \n",
       "\n",
       "                  misassembled contigs misassembly events  \\\n",
       "Assembler                                                   \n",
       "ABySS                        4.0 [2;8]        7.12 [3;14]   \n",
       "BCALM2                       0.0 [0;0]          0.0 [0;0]   \n",
       "GATBMiniaPipeline           1.67 [1;3]          3.0 [1;6]   \n",
       "IDBA-UD                   11.44 [4;18]       21.22 [7;36]   \n",
       "MEGAHIT                   11.33 [9;15]      23.56 [18;32]   \n",
       "MetaHipMer2                  0.0 [0;0]          0.0 [0;0]   \n",
       "metaSPAdes                  1.67 [1;2]          4.0 [3;6]   \n",
       "MINIA                        0.0 [0;0]          0.0 [0;0]   \n",
       "SKESA                      6.33 [0;10]       12.33 [0;19]   \n",
       "SPAdes                       4.0 [0;9]         8.0 [0;18]   \n",
       "Unicycler                   0.67 [0;2]         1.33 [0;4]   \n",
       "VelvetOptimiser             4.67 [0;8]        7.33 [0;11]   \n",
       "\n",
       "                                         n50  \n",
       "Assembler                                     \n",
       "ABySS                  79091.88 [139;124246]  \n",
       "BCALM2                    6029.67 [49;17395]  \n",
       "GATBMiniaPipeline    104806.0 [90346;121957]  \n",
       "IDBA-UD               48829.89 [20993;63725]  \n",
       "MEGAHIT             146076.33 [88308;178381]  \n",
       "MetaHipMer2            69010.56 [658;106383]  \n",
       "metaSPAdes           163304.0 [42332;238611]  \n",
       "MINIA                  14292.56 [7060;17967]  \n",
       "SKESA                 64292.67 [42115;76924]  \n",
       "SPAdes              154762.67 [53292;206515]  \n",
       "Unicycler          213645.67 [191424;245050]  \n",
       "VelvetOptimiser          19905.89 [85;30453]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>contigs</th>\n",
       "      <th>filtered_Ns</th>\n",
       "      <th>filtered_basepairs</th>\n",
       "      <th>filtered_contigs</th>\n",
       "      <th>filtered_mapped_reads</th>\n",
       "      <th>filtered_n50</th>\n",
       "      <th>mapped_reads</th>\n",
       "      <th>max_contig</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>n50</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>678.0 [9;1347]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>49430581.5 [20132631;78728532]</td>\n",
       "      <td>361240.5 [44621;677860]</td>\n",
       "      <td>200.0 [0;400]</td>\n",
       "      <td>8362357.0 [3150980;13573734]</td>\n",
       "      <td>784.0 [356;1212]</td>\n",
       "      <td>89.57 [79;99]</td>\n",
       "      <td>105437.0 [18568;192306]</td>\n",
       "      <td>199.55 [101;297]</td>\n",
       "      <td>469989.5 [88665;851314]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>4835.0 [101;9569]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>233321388.33 [19635512;364761201]</td>\n",
       "      <td>5418807.0 [44825;8175890]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>4688952.0 [11967;13397815]</td>\n",
       "      <td>433.0 [11;784]</td>\n",
       "      <td>32.44 [0;97]</td>\n",
       "      <td>11253.33 [1062;31436]</td>\n",
       "      <td>36.14 [1;98]</td>\n",
       "      <td>55949.33 [1212;163543]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>6255.33 [35;18682]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>14891941.67 [13638625;17126362]</td>\n",
       "      <td>2819.67 [693;6777]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>13685778.67 [13507672;13891328]</td>\n",
       "      <td>528.0 [219;762]</td>\n",
       "      <td>95.04 [89;98]</td>\n",
       "      <td>107757.33 [90989;140561]</td>\n",
       "      <td>95.72 [89;100]</td>\n",
       "      <td>602475.67 [542997;632215]</td>\n",
       "      <td>1.0 [0;2]</td>\n",
       "      <td>3.0 [0;6]</td>\n",
       "      <td>94829.0 [56759;136741]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>17234433.0 [13791669;23586883]</td>\n",
       "      <td>5722.33 [1025;14155]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>14926006.67 [13601126;17534580]</td>\n",
       "      <td>1184.67 [261;2725]</td>\n",
       "      <td>95.23 [89;99]</td>\n",
       "      <td>109414.33 [106347;113853]</td>\n",
       "      <td>95.91 [90;99]</td>\n",
       "      <td>616678.67 [585610;632213]</td>\n",
       "      <td>2.33 [2;3]</td>\n",
       "      <td>4.0 [2;5]</td>\n",
       "      <td>85565.0 [51161;105808]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>17581572.33 [13843596;24498566]</td>\n",
       "      <td>5593.67 [767;14172]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>15309833.0 [13671964;18552996]</td>\n",
       "      <td>1181.0 [172;3044]</td>\n",
       "      <td>97.05 [93;100]</td>\n",
       "      <td>197842.0 [171540;210993]</td>\n",
       "      <td>97.63 [94;100]</td>\n",
       "      <td>826891.0 [632250;1080013]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>1.33 [0;2]</td>\n",
       "      <td>152385.33 [82249;203548]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>15029970.0 [15029970;15029970]</td>\n",
       "      <td>11966.0 [11966;11966]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>10264085.0 [10264085;10264085]</td>\n",
       "      <td>5038.0 [5038;5038]</td>\n",
       "      <td>74.39 [74;74]</td>\n",
       "      <td>2170.0 [2170;2170]</td>\n",
       "      <td>85.28 [85;85]</td>\n",
       "      <td>14760.0 [14760;14760]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1515.0 [1515;1515]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>19164672.33 [14157309;27822010]</td>\n",
       "      <td>11693.67 [2099;26282]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>15386033.67 [13672885;18792639]</td>\n",
       "      <td>1238.33 [139;3392]</td>\n",
       "      <td>95.81 [89;99]</td>\n",
       "      <td>174651.33 [87983;239185]</td>\n",
       "      <td>96.17 [90;100]</td>\n",
       "      <td>796644.67 [622107;917728]</td>\n",
       "      <td>1.67 [1;3]</td>\n",
       "      <td>4.33 [3;7]</td>\n",
       "      <td>139179.67 [3674;220609]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>17498251.0 [15334799;20758864]</td>\n",
       "      <td>25651.67 [15001;37300]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>13290379.67 [12920975;13501498]</td>\n",
       "      <td>1466.33 [762;2171]</td>\n",
       "      <td>87.94 [81;97]</td>\n",
       "      <td>20723.33 [10582;33550]</td>\n",
       "      <td>92.64 [87;99]</td>\n",
       "      <td>168041.33 [163543;177038]</td>\n",
       "      <td>1.33 [0;2]</td>\n",
       "      <td>2.33 [0;4]</td>\n",
       "      <td>13895.33 [7485;26474]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>6659581.0 [2957060;8818742]</td>\n",
       "      <td>6124.0 [36;9514]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>4407308.0 [2950393;5801439]</td>\n",
       "      <td>1447.67 [21;2889]</td>\n",
       "      <td>82.19 [62;92]</td>\n",
       "      <td>120186.67 [2148;351027]</td>\n",
       "      <td>88.7 [78;95]</td>\n",
       "      <td>240146.67 [29155;632013]</td>\n",
       "      <td>4.0 [0;12]</td>\n",
       "      <td>7.67 [0;23]</td>\n",
       "      <td>117871.0 [1155;351027]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>20446071.0 [15606257;29517365]</td>\n",
       "      <td>24861.67 [7331;36214]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>15441310.0 [13087777;19595813]</td>\n",
       "      <td>2250.67 [227;4999]</td>\n",
       "      <td>84.02 [75;98]</td>\n",
       "      <td>87170.0 [55465;136684]</td>\n",
       "      <td>102.37 [99;107]</td>\n",
       "      <td>581322.33 [263645;847683]</td>\n",
       "      <td>191.0 [0;572]</td>\n",
       "      <td>383.33 [0;1148]</td>\n",
       "      <td>57051.0 [1972;122703]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>11596988.67 [9747587;13324149]</td>\n",
       "      <td>218.0 [185;264]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>11577328.67 [9725301;13315624]</td>\n",
       "      <td>166.33 [112;243]</td>\n",
       "      <td>94.33 [87;98]</td>\n",
       "      <td>230819.67 [192888;256884]</td>\n",
       "      <td>94.94 [88;99]</td>\n",
       "      <td>847562.0 [847534;847580]</td>\n",
       "      <td>1.67 [0;5]</td>\n",
       "      <td>3.33 [0;10]</td>\n",
       "      <td>230819.67 [192888;256884]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>1644.0 [0;4932]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>29946500.67 [19638608;47958628]</td>\n",
       "      <td>260234.67 [69092;525998]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>4815001.0 [409222;7334025]</td>\n",
       "      <td>2197.67 [340;3237]</td>\n",
       "      <td>27.53 [0;79]</td>\n",
       "      <td>2090.0 [1180;2913]</td>\n",
       "      <td>34.77 [1;93]</td>\n",
       "      <td>11869.67 [2044;17566]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>261.0 [88;520]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm               Ns  \\\n",
       "Assembler                                                            \n",
       "ABySS              Multiple k-mer De Bruijn graph   678.0 [9;1347]   \n",
       "BCALM2               Single k-mer De Bruijn graph        0.0 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph        0.0 [0;0]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph        0.0 [0;0]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph        0.0 [0;0]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph        0.0 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph        0.0 [0;0]   \n",
       "MINIA                Single k-mer De Bruijn graph        0.0 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph        0.0 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph        0.0 [0;0]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph        0.0 [0;0]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph  1644.0 [0;4932]   \n",
       "\n",
       "                          Type                          basepairs  \\\n",
       "Assembler                                                           \n",
       "ABySS              Metagenomic     49430581.5 [20132631;78728532]   \n",
       "BCALM2                 Genomic  233321388.33 [19635512;364761201]   \n",
       "GATBMiniaPipeline  Metagenomic    14891941.67 [13638625;17126362]   \n",
       "IDBA-UD            Metagenomic     17234433.0 [13791669;23586883]   \n",
       "MEGAHIT            Metagenomic    17581572.33 [13843596;24498566]   \n",
       "MetaHipMer2        Metagenomic     15029970.0 [15029970;15029970]   \n",
       "metaSPAdes         Metagenomic    19164672.33 [14157309;27822010]   \n",
       "MINIA                  Genomic     17498251.0 [15334799;20758864]   \n",
       "SKESA                  Genomic        6659581.0 [2957060;8818742]   \n",
       "SPAdes                 Genomic     20446071.0 [15606257;29517365]   \n",
       "Unicycler              Genomic     11596988.67 [9747587;13324149]   \n",
       "VelvetOptimiser    Metagenomic    29946500.67 [19638608;47958628]   \n",
       "\n",
       "                                     contigs    filtered_Ns  \\\n",
       "Assembler                                                     \n",
       "ABySS                361240.5 [44621;677860]  200.0 [0;400]   \n",
       "BCALM2             5418807.0 [44825;8175890]      0.0 [0;0]   \n",
       "GATBMiniaPipeline         2819.67 [693;6777]      0.0 [0;0]   \n",
       "IDBA-UD                 5722.33 [1025;14155]      0.0 [0;0]   \n",
       "MEGAHIT                  5593.67 [767;14172]      0.0 [0;0]   \n",
       "MetaHipMer2            11966.0 [11966;11966]      0.0 [0;0]   \n",
       "metaSPAdes             11693.67 [2099;26282]      0.0 [0;0]   \n",
       "MINIA                 25651.67 [15001;37300]      0.0 [0;0]   \n",
       "SKESA                       6124.0 [36;9514]      0.0 [0;0]   \n",
       "SPAdes                 24861.67 [7331;36214]      0.0 [0;0]   \n",
       "Unicycler                    218.0 [185;264]      0.0 [0;0]   \n",
       "VelvetOptimiser     260234.67 [69092;525998]      0.0 [0;0]   \n",
       "\n",
       "                                filtered_basepairs    filtered_contigs  \\\n",
       "Assembler                                                                \n",
       "ABySS                 8362357.0 [3150980;13573734]    784.0 [356;1212]   \n",
       "BCALM2                  4688952.0 [11967;13397815]      433.0 [11;784]   \n",
       "GATBMiniaPipeline  13685778.67 [13507672;13891328]     528.0 [219;762]   \n",
       "IDBA-UD            14926006.67 [13601126;17534580]  1184.67 [261;2725]   \n",
       "MEGAHIT             15309833.0 [13671964;18552996]   1181.0 [172;3044]   \n",
       "MetaHipMer2         10264085.0 [10264085;10264085]  5038.0 [5038;5038]   \n",
       "metaSPAdes         15386033.67 [13672885;18792639]  1238.33 [139;3392]   \n",
       "MINIA              13290379.67 [12920975;13501498]  1466.33 [762;2171]   \n",
       "SKESA                  4407308.0 [2950393;5801439]   1447.67 [21;2889]   \n",
       "SPAdes              15441310.0 [13087777;19595813]  2250.67 [227;4999]   \n",
       "Unicycler           11577328.67 [9725301;13315624]    166.33 [112;243]   \n",
       "VelvetOptimiser         4815001.0 [409222;7334025]  2197.67 [340;3237]   \n",
       "\n",
       "                  filtered_mapped_reads               filtered_n50  \\\n",
       "Assembler                                                            \n",
       "ABySS                     89.57 [79;99]    105437.0 [18568;192306]   \n",
       "BCALM2                     32.44 [0;97]      11253.33 [1062;31436]   \n",
       "GATBMiniaPipeline         95.04 [89;98]   107757.33 [90989;140561]   \n",
       "IDBA-UD                   95.23 [89;99]  109414.33 [106347;113853]   \n",
       "MEGAHIT                  97.05 [93;100]   197842.0 [171540;210993]   \n",
       "MetaHipMer2               74.39 [74;74]         2170.0 [2170;2170]   \n",
       "metaSPAdes                95.81 [89;99]   174651.33 [87983;239185]   \n",
       "MINIA                     87.94 [81;97]     20723.33 [10582;33550]   \n",
       "SKESA                     82.19 [62;92]    120186.67 [2148;351027]   \n",
       "SPAdes                    84.02 [75;98]     87170.0 [55465;136684]   \n",
       "Unicycler                 94.33 [87;98]  230819.67 [192888;256884]   \n",
       "VelvetOptimiser            27.53 [0;79]         2090.0 [1180;2913]   \n",
       "\n",
       "                       mapped_reads                 max_contig  \\\n",
       "Assembler                                                        \n",
       "ABySS              199.55 [101;297]    469989.5 [88665;851314]   \n",
       "BCALM2                 36.14 [1;98]     55949.33 [1212;163543]   \n",
       "GATBMiniaPipeline    95.72 [89;100]  602475.67 [542997;632215]   \n",
       "IDBA-UD               95.91 [90;99]  616678.67 [585610;632213]   \n",
       "MEGAHIT              97.63 [94;100]  826891.0 [632250;1080013]   \n",
       "MetaHipMer2           85.28 [85;85]      14760.0 [14760;14760]   \n",
       "metaSPAdes           96.17 [90;100]  796644.67 [622107;917728]   \n",
       "MINIA                 92.64 [87;99]  168041.33 [163543;177038]   \n",
       "SKESA                  88.7 [78;95]   240146.67 [29155;632013]   \n",
       "SPAdes              102.37 [99;107]  581322.33 [263645;847683]   \n",
       "Unicycler             94.94 [88;99]   847562.0 [847534;847580]   \n",
       "VelvetOptimiser        34.77 [1;93]      11869.67 [2044;17566]   \n",
       "\n",
       "                  misassembled contigs misassembly events  \\\n",
       "Assembler                                                   \n",
       "ABySS                        0.0 [0;0]          0.0 [0;0]   \n",
       "BCALM2                       0.0 [0;0]          0.0 [0;0]   \n",
       "GATBMiniaPipeline            1.0 [0;2]          3.0 [0;6]   \n",
       "IDBA-UD                     2.33 [2;3]          4.0 [2;5]   \n",
       "MEGAHIT                     0.67 [0;1]         1.33 [0;2]   \n",
       "MetaHipMer2                  0.0 [0;0]          0.0 [0;0]   \n",
       "metaSPAdes                  1.67 [1;3]         4.33 [3;7]   \n",
       "MINIA                       1.33 [0;2]         2.33 [0;4]   \n",
       "SKESA                       4.0 [0;12]        7.67 [0;23]   \n",
       "SPAdes                   191.0 [0;572]    383.33 [0;1148]   \n",
       "Unicycler                   1.67 [0;5]        3.33 [0;10]   \n",
       "VelvetOptimiser              0.0 [0;0]          0.0 [0;0]   \n",
       "\n",
       "                                         n50  \n",
       "Assembler                                     \n",
       "ABySS                      4835.0 [101;9569]  \n",
       "BCALM2                    6255.33 [35;18682]  \n",
       "GATBMiniaPipeline     94829.0 [56759;136741]  \n",
       "IDBA-UD               85565.0 [51161;105808]  \n",
       "MEGAHIT             152385.33 [82249;203548]  \n",
       "MetaHipMer2               1515.0 [1515;1515]  \n",
       "metaSPAdes           139179.67 [3674;220609]  \n",
       "MINIA                  13895.33 [7485;26474]  \n",
       "SKESA                 117871.0 [1155;351027]  \n",
       "SPAdes                 57051.0 [1972;122703]  \n",
       "Unicycler          230819.67 [192888;256884]  \n",
       "VelvetOptimiser               261.0 [88;520]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "for distribution in global_pipeline_metrics_df.distribution.unique():\n",
    "    stats_per_distribution = pd.DataFrame()\n",
    "    for assembler in global_pipeline_metrics_df.assembler.unique():\n",
    "        row={'Assembler': assembler, \n",
    "             'Type':global_pipeline_metrics_df[(global_pipeline_metrics_df.assembler == assembler) & (global_pipeline_metrics_df.distribution == distribution)].type.unique()[0],\n",
    "             'Algorithm': global_pipeline_metrics_df[global_pipeline_metrics_df.assembler == assembler].algorythm.unique()[0]}\n",
    "        lala=global_pipeline_metrics_df[(global_pipeline_metrics_df.assembler == assembler) & (global_pipeline_metrics_df.distribution == distribution)].describe()\n",
    "        for column in lala.columns:\n",
    "            mean = lala.loc['mean',column]\n",
    "            minimum = lala.loc['min',column]\n",
    "            maximum = lala.loc['max',column]\n",
    "            row[column] = \"{} [{};{}]\".format(round(mean, 2),int(minimum), int(maximum))\n",
    "        stats_per_distribution=stats_per_distribution.append(row, ignore_index=True)\n",
    "    stats_per_distribution = stats_per_distribution.set_index('Assembler')\n",
    "    stats_per_distribution.to_csv(\"Tables/Results/{} - Global metrics per assembler.csv\".format(distribution))\n",
    "    print(distribution)\n",
    "    display(stats_per_distribution)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Reference-based metrics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pipeline_report_tables.json data from run2...\n",
      "Processing pipeline_report_tables.json data from run3...\n",
      "Processing pipeline_report_tables.json data from run1...\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>assembler</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>...</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>reference</th>\n",
       "      <th>run</th>\n",
       "      <th>sample</th>\n",
       "      <th>validity</th>\n",
       "      <th>distribution</th>\n",
       "      <th>type</th>\n",
       "      <th>algorythm</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>39.0</td>\n",
       "      <td>0.073679</td>\n",
       "      <td>98440.0</td>\n",
       "      <td>98440.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ABySS</td>\n",
       "      <td>3977886.0</td>\n",
       "      <td>0.983244</td>\n",
       "      <td>66.0</td>\n",
       "      <td>0.999639</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.001060</td>\n",
       "      <td>1.001159</td>\n",
       "      <td>Bacillus_subtilis</td>\n",
       "      <td>run2</td>\n",
       "      <td>ERR2984773</td>\n",
       "      <td>0.999901</td>\n",
       "      <td>Even</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>BCALM2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Bacillus_subtilis</td>\n",
       "      <td>run2</td>\n",
       "      <td>ERR2984773</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Even</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>15.0</td>\n",
       "      <td>0.189942</td>\n",
       "      <td>263420.0</td>\n",
       "      <td>257358.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GATBMiniaPipeline</td>\n",
       "      <td>3982010.0</td>\n",
       "      <td>0.984263</td>\n",
       "      <td>30.0</td>\n",
       "      <td>0.999809</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.000079</td>\n",
       "      <td>1.000083</td>\n",
       "      <td>Bacillus_subtilis</td>\n",
       "      <td>run2</td>\n",
       "      <td>ERR2984773</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>Even</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>120.0</td>\n",
       "      <td>0.031121</td>\n",
       "      <td>34330.0</td>\n",
       "      <td>34330.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IDBA-UD</td>\n",
       "      <td>3976919.0</td>\n",
       "      <td>0.983005</td>\n",
       "      <td>184.0</td>\n",
       "      <td>0.999756</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.003298</td>\n",
       "      <td>1.003311</td>\n",
       "      <td>Bacillus_subtilis</td>\n",
       "      <td>run2</td>\n",
       "      <td>ERR2984773</td>\n",
       "      <td>0.999986</td>\n",
       "      <td>Even</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>14.0</td>\n",
       "      <td>0.103094</td>\n",
       "      <td>294549.0</td>\n",
       "      <td>294549.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>MEGAHIT</td>\n",
       "      <td>3989518.0</td>\n",
       "      <td>0.986119</td>\n",
       "      <td>31.0</td>\n",
       "      <td>0.978636</td>\n",
       "      <td>...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.998972</td>\n",
       "      <td>1.011678</td>\n",
       "      <td>Bacillus_subtilis</td>\n",
       "      <td>run2</td>\n",
       "      <td>ERR2984773</td>\n",
       "      <td>0.987440</td>\n",
       "      <td>Even</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1675</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>MINIA</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Staphylococcus_aureus_plasmid3</td>\n",
       "      <td>run1</td>\n",
       "      <td>LHS</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Log</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1676</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>SKESA</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Staphylococcus_aureus_plasmid3</td>\n",
       "      <td>run1</td>\n",
       "      <td>LHS</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Log</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1677</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>SPAdes</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Staphylococcus_aureus_plasmid3</td>\n",
       "      <td>run1</td>\n",
       "      <td>LHS</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Log</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1678</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Unicycler</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Staphylococcus_aureus_plasmid3</td>\n",
       "      <td>run1</td>\n",
       "      <td>LHS</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Log</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1679</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>VelvetOptimiser</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Staphylococcus_aureus_plasmid3</td>\n",
       "      <td>run1</td>\n",
       "      <td>LHS</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Log</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1680 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        L90       LSA      NA50      NG50   Ns          assembler  basepairs  \\\n",
       "0      39.0  0.073679   98440.0   98440.0  0.0              ABySS  3977886.0   \n",
       "1       0.0  0.000000       0.0       0.0  0.0             BCALM2        0.0   \n",
       "2      15.0  0.189942  263420.0  257358.0  0.0  GATBMiniaPipeline  3982010.0   \n",
       "3     120.0  0.031121   34330.0   34330.0  0.0            IDBA-UD  3976919.0   \n",
       "4      14.0  0.103094  294549.0  294549.0  0.0            MEGAHIT  3989518.0   \n",
       "...     ...       ...       ...       ...  ...                ...        ...   \n",
       "1675    0.0  0.000000       0.0       0.0  0.0              MINIA        0.0   \n",
       "1676    0.0  0.000000       0.0       0.0  0.0              SKESA        0.0   \n",
       "1677    0.0  0.000000       0.0       0.0  0.0             SPAdes        0.0   \n",
       "1678    0.0  0.000000       0.0       0.0  0.0          Unicycler        0.0   \n",
       "1679    0.0  0.000000       0.0       0.0  0.0    VelvetOptimiser        0.0   \n",
       "\n",
       "      breadth_of_coverage  contigs  identity  ...  misassembly events  \\\n",
       "0                0.983244     66.0  0.999639  ...                 0.0   \n",
       "1                0.000000      0.0  0.000000  ...                 0.0   \n",
       "2                0.984263     30.0  0.999809  ...                 0.0   \n",
       "3                0.983005    184.0  0.999756  ...                 0.0   \n",
       "4                0.986119     31.0  0.978636  ...                 4.0   \n",
       "...                   ...      ...       ...  ...                 ...   \n",
       "1675             0.000000      0.0  0.000000  ...                 0.0   \n",
       "1676             0.000000      0.0  0.000000  ...                 0.0   \n",
       "1677             0.000000      0.0  0.000000  ...                 0.0   \n",
       "1678             0.000000      0.0  0.000000  ...                 0.0   \n",
       "1679             0.000000      0.0  0.000000  ...                 0.0   \n",
       "\n",
       "      multiplicity  parsimony                       reference   run  \\\n",
       "0         1.001060   1.001159               Bacillus_subtilis  run2   \n",
       "1         0.000000   0.000000               Bacillus_subtilis  run2   \n",
       "2         1.000079   1.000083               Bacillus_subtilis  run2   \n",
       "3         1.003298   1.003311               Bacillus_subtilis  run2   \n",
       "4         0.998972   1.011678               Bacillus_subtilis  run2   \n",
       "...            ...        ...                             ...   ...   \n",
       "1675      0.000000   0.000000  Staphylococcus_aureus_plasmid3  run1   \n",
       "1676      0.000000   0.000000  Staphylococcus_aureus_plasmid3  run1   \n",
       "1677      0.000000   0.000000  Staphylococcus_aureus_plasmid3  run1   \n",
       "1678      0.000000   0.000000  Staphylococcus_aureus_plasmid3  run1   \n",
       "1679      0.000000   0.000000  Staphylococcus_aureus_plasmid3  run1   \n",
       "\n",
       "          sample  validity distribution         type  \\\n",
       "0     ERR2984773  0.999901         Even  Metagenomic   \n",
       "1     ERR2984773  0.000000         Even      Genomic   \n",
       "2     ERR2984773  0.999995         Even  Metagenomic   \n",
       "3     ERR2984773  0.999986         Even  Metagenomic   \n",
       "4     ERR2984773  0.987440         Even  Metagenomic   \n",
       "...          ...       ...          ...          ...   \n",
       "1675         LHS  0.000000          Log      Genomic   \n",
       "1676         LHS  0.000000          Log      Genomic   \n",
       "1677         LHS  0.000000          Log      Genomic   \n",
       "1678         LHS  0.000000          Log      Genomic   \n",
       "1679         LHS  0.000000          Log  Metagenomic   \n",
       "\n",
       "                           algorythm  \n",
       "0     Multiple k-mer De Bruijn graph  \n",
       "1       Single k-mer De Bruijn graph  \n",
       "2     Multiple k-mer De Bruijn graph  \n",
       "3     Multiple k-mer De Bruijn graph  \n",
       "4     Multiple k-mer De Bruijn graph  \n",
       "...                              ...  \n",
       "1675    Single k-mer De Bruijn graph  \n",
       "1676  Multiple k-mer De Bruijn graph  \n",
       "1677  Multiple k-mer De Bruijn graph  \n",
       "1678  Multiple k-mer De Bruijn graph  \n",
       "1679  Multiple k-mer De Bruijn graph  \n",
       "\n",
       "[1680 rows x 22 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "report_glob = glob.glob('../Results/*/*/report/pipeline_report_tables.json')\n",
    "reference_pipeline_metrics_df = pd.DataFrame()\n",
    "\n",
    "for pipeline_report_file in report_glob:\n",
    "    report_file_name = pipeline_report_file.split('/')[-1]\n",
    "    stats_run = pipeline_report_file.split('/')[-3]\n",
    "    print('Processing {0} data from {1}...'.format(report_file_name, stats_run))\n",
    "    \n",
    "    with open(pipeline_report_file) as _fh:\n",
    "        json_report = json.load(_fh)\n",
    "        for sample in json_report.keys():\n",
    "            for reference, data in json_report[sample]['ReferenceTables'].items():\n",
    "                for row in data:\n",
    "                    for item in row:\n",
    "                        reference_pipeline_metrics_df = reference_pipeline_metrics_df.append({'run': stats_run,\n",
    "                                                                                        'sample': sample,\n",
    "                                                                                        'assembler': item['assembler'],\n",
    "                                                                                        'reference': reference,\n",
    "                                                                                        'LSA': item['contiguity'],\n",
    "                                                                                        'breadth_of_coverage': item['breadth_of_coverage'],\n",
    "                                                                                        'multiplicity': item['multiplicity'],\n",
    "                                                                                        'validity': item['validity'],\n",
    "                                                                                        'parsimony': item['parsimony'],\n",
    "                                                                                        'identity': item['identity'],\n",
    "                                                                                        'lowest_identity': item['lowest_identity'],\n",
    "                                                                                        'L90': item['L90'],\n",
    "                                                                                        'contigs': item['aligned_contigs'],\n",
    "                                                                                        'NA50': item['NA50'],\n",
    "                                                                                        'NG50': item['NG50'],\n",
    "                                                                                        'basepairs': item['aligned_basepairs'],\n",
    "                                                                                        'Ns': item['Ns'],\n",
    "                                                                                        'misassembled contigs': item['misassembled_contigs'],\n",
    "                                                                                        'misassembly events': item['misassembly_events']},\n",
    "                                                                                       ignore_index=True)\n",
    "\n",
    "reference_pipeline_metrics_df['distribution'] = np.where(reference_pipeline_metrics_df['sample'].isin(log_distributed), 'Log', 'Even')\n",
    "reference_pipeline_metrics_df['type'] = np.where(reference_pipeline_metrics_df['assembler'].isin(genomic_assemblers), 'Genomic', 'Metagenomic')\n",
    "reference_pipeline_metrics_df['algorythm'] = np.where(reference_pipeline_metrics_df['assembler'].isin(single_kmer), 'Single k-mer De Bruijn graph', 'Multiple k-mer De Bruijn graph')\n",
    "reference_pipeline_metrics_df[['contigs','basepairs','L90','Ns','NA50','NG50','misassembled contigs', 'misassembly events', 'multiplicity','validity','parsimony','identity','lowest_identity']] = reference_pipeline_metrics_df[['contigs','basepairs','L90','Ns','NA50','NG50','misassembled contigs', 'misassembly events','multiplicity','validity','parsimony','identity','lowest_identity']].apply(pd.to_numeric)\n",
    "display(reference_pipeline_metrics_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "L90                     float64\n",
       "LSA                     float64\n",
       "NA50                    float64\n",
       "NG50                    float64\n",
       "Ns                      float64\n",
       "assembler                object\n",
       "basepairs               float64\n",
       "breadth_of_coverage     float64\n",
       "contigs                 float64\n",
       "identity                float64\n",
       "lowest_identity         float64\n",
       "misassembled contigs    float64\n",
       "misassembly events      float64\n",
       "multiplicity            float64\n",
       "parsimony               float64\n",
       "reference                object\n",
       "run                      object\n",
       "sample                   object\n",
       "validity                float64\n",
       "distribution             object\n",
       "type                     object\n",
       "algorythm                object\n",
       "dtype: object"
      ]
     },
     "execution_count": 110,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reference_pipeline_metrics_df.dtypes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Save data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [],
   "source": [
    "reference_pipeline_metrics_df.to_csv(\"Tables/Results/Reference metrics.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reference statistics per assembler type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_stats_per_assembler_type = pd.DataFrame()\n",
    "for assembly_type in reference_pipeline_metrics_df.type.unique():\n",
    "    row={'Assembler Type': assembly_type}\n",
    "    lala=reference_pipeline_metrics_df[reference_pipeline_metrics_df.type == assembly_type].describe()\n",
    "    for column in lala.columns:\n",
    "        mean = lala.loc['mean',column]\n",
    "        minimum = lala.loc['min',column]\n",
    "        maximum = lala.loc['max',column]\n",
    "        row[column] = \"{} [{};{}]\".format(round(mean, 2),int(minimum), int(maximum))\n",
    "    ref_stats_per_assembler_type=ref_stats_per_assembler_type.append(row, ignore_index=True)\n",
    "ref_stats_per_assembler_type = ref_stats_per_assembler_type.set_index('Assembler Type')\n",
    "ref_stats_per_assembler_type.to_csv(\"Tables/Results/Reference metrics per assembler type.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler Type</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Metagenomic</th>\n",
       "      <td>30.94 [0;1033]</td>\n",
       "      <td>0.22 [0;1]</td>\n",
       "      <td>77775.8 [0;593167]</td>\n",
       "      <td>77156.14 [0;593167]</td>\n",
       "      <td>70.11 [0;12098]</td>\n",
       "      <td>1973596.73 [0;6781605]</td>\n",
       "      <td>0.71 [0;1]</td>\n",
       "      <td>83.76 [0;2402]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>0.77 [0;1]</td>\n",
       "      <td>0.36 [0;9]</td>\n",
       "      <td>0.68 [0;15]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>0.81 [0;5]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic</th>\n",
       "      <td>40.57 [0;921]</td>\n",
       "      <td>0.16 [0;1]</td>\n",
       "      <td>62460.97 [0;475179]</td>\n",
       "      <td>61338.27 [0;475179]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1863265.6 [0;6728888]</td>\n",
       "      <td>0.62 [0;1]</td>\n",
       "      <td>132.4 [0;1877]</td>\n",
       "      <td>0.71 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>1.0 [0;565]</td>\n",
       "      <td>1.96 [0;1134]</td>\n",
       "      <td>0.71 [0;1]</td>\n",
       "      <td>0.71 [0;1]</td>\n",
       "      <td>0.71 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           L90         LSA                 NA50  \\\n",
       "Assembler Type                                                    \n",
       "Metagenomic     30.94 [0;1033]  0.22 [0;1]   77775.8 [0;593167]   \n",
       "Genomic          40.57 [0;921]  0.16 [0;1]  62460.97 [0;475179]   \n",
       "\n",
       "                               NG50               Ns               basepairs  \\\n",
       "Assembler Type                                                                 \n",
       "Metagenomic     77156.14 [0;593167]  70.11 [0;12098]  1973596.73 [0;6781605]   \n",
       "Genomic         61338.27 [0;475179]        0.0 [0;0]   1863265.6 [0;6728888]   \n",
       "\n",
       "               breadth_of_coverage         contigs    identity  \\\n",
       "Assembler Type                                                   \n",
       "Metagenomic             0.71 [0;1]  83.76 [0;2402]   0.8 [0;1]   \n",
       "Genomic                 0.62 [0;1]  132.4 [0;1877]  0.71 [0;1]   \n",
       "\n",
       "               lowest_identity misassembled contigs misassembly events  \\\n",
       "Assembler Type                                                           \n",
       "Metagenomic         0.77 [0;1]           0.36 [0;9]        0.68 [0;15]   \n",
       "Genomic             0.67 [0;1]          1.0 [0;565]      1.96 [0;1134]   \n",
       "\n",
       "               multiplicity   parsimony    validity  \n",
       "Assembler Type                                       \n",
       "Metagenomic       0.8 [0;1]  0.81 [0;5]   0.8 [0;1]  \n",
       "Genomic          0.71 [0;1]  0.71 [0;1]  0.71 [0;1]  "
      ]
     },
     "execution_count": 113,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ref_stats_per_assembler_type"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reference stats per assembly algorithm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_stats_per_assembler_algo = pd.DataFrame()\n",
    "for assembly_algo in reference_pipeline_metrics_df.algorythm.unique():\n",
    "    row={'Assembler Algorithm': assembly_algo}\n",
    "    lala=reference_pipeline_metrics_df[reference_pipeline_metrics_df.algorythm == assembly_algo].describe()\n",
    "    for column in lala.columns:\n",
    "        mean = lala.loc['mean',column]\n",
    "        minimum = lala.loc['min',column]\n",
    "        maximum = lala.loc['max',column]\n",
    "        row[column] = \"{} [{};{}]\".format(round(mean, 2),int(minimum), int(maximum))\n",
    "    ref_stats_per_assembler_algo=ref_stats_per_assembler_algo.append(row, ignore_index=True)\n",
    "ref_stats_per_assembler_algo = ref_stats_per_assembler_algo.set_index('Assembler Algorithm')\n",
    "ref_stats_per_assembler_algo.to_csv(\"Tables/Results/Reference metrics per assembler algorithm.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler Algorithm</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Multiple k-mer De Bruijn graph</th>\n",
       "      <td>29.85 [0;1033]</td>\n",
       "      <td>0.23 [0;1]</td>\n",
       "      <td>83811.38 [0;593167]</td>\n",
       "      <td>82906.92 [0;593167]</td>\n",
       "      <td>48.35 [0;12098]</td>\n",
       "      <td>2008465.08 [0;6781605]</td>\n",
       "      <td>0.71 [0;1]</td>\n",
       "      <td>77.17 [0;2402]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>0.76 [0;1]</td>\n",
       "      <td>0.76 [0;565]</td>\n",
       "      <td>1.47 [0;1134]</td>\n",
       "      <td>0.81 [0;1]</td>\n",
       "      <td>0.81 [0;5]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Single k-mer De Bruijn graph</th>\n",
       "      <td>60.32 [0;921]</td>\n",
       "      <td>0.05 [0;0]</td>\n",
       "      <td>10316.74 [0;65138]</td>\n",
       "      <td>9816.02 [0;64960]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1529238.55 [0;6524118]</td>\n",
       "      <td>0.46 [0;0]</td>\n",
       "      <td>237.24 [0;1495]</td>\n",
       "      <td>0.56 [0;1]</td>\n",
       "      <td>0.55 [0;1]</td>\n",
       "      <td>0.02 [0;2]</td>\n",
       "      <td>0.03 [0;4]</td>\n",
       "      <td>0.56 [0;1]</td>\n",
       "      <td>0.56 [0;1]</td>\n",
       "      <td>0.56 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           L90         LSA  \\\n",
       "Assembler Algorithm                                          \n",
       "Multiple k-mer De Bruijn graph  29.85 [0;1033]  0.23 [0;1]   \n",
       "Single k-mer De Bruijn graph     60.32 [0;921]  0.05 [0;0]   \n",
       "\n",
       "                                               NA50                 NG50  \\\n",
       "Assembler Algorithm                                                        \n",
       "Multiple k-mer De Bruijn graph  83811.38 [0;593167]  82906.92 [0;593167]   \n",
       "Single k-mer De Bruijn graph     10316.74 [0;65138]    9816.02 [0;64960]   \n",
       "\n",
       "                                             Ns               basepairs  \\\n",
       "Assembler Algorithm                                                       \n",
       "Multiple k-mer De Bruijn graph  48.35 [0;12098]  2008465.08 [0;6781605]   \n",
       "Single k-mer De Bruijn graph          0.0 [0;0]  1529238.55 [0;6524118]   \n",
       "\n",
       "                               breadth_of_coverage          contigs  \\\n",
       "Assembler Algorithm                                                   \n",
       "Multiple k-mer De Bruijn graph          0.71 [0;1]   77.17 [0;2402]   \n",
       "Single k-mer De Bruijn graph            0.46 [0;0]  237.24 [0;1495]   \n",
       "\n",
       "                                  identity lowest_identity  \\\n",
       "Assembler Algorithm                                          \n",
       "Multiple k-mer De Bruijn graph   0.8 [0;1]      0.76 [0;1]   \n",
       "Single k-mer De Bruijn graph    0.56 [0;1]      0.55 [0;1]   \n",
       "\n",
       "                               misassembled contigs misassembly events  \\\n",
       "Assembler Algorithm                                                      \n",
       "Multiple k-mer De Bruijn graph         0.76 [0;565]      1.47 [0;1134]   \n",
       "Single k-mer De Bruijn graph             0.02 [0;2]         0.03 [0;4]   \n",
       "\n",
       "                               multiplicity   parsimony    validity  \n",
       "Assembler Algorithm                                                  \n",
       "Multiple k-mer De Bruijn graph   0.81 [0;1]  0.81 [0;5]   0.8 [0;1]  \n",
       "Single k-mer De Bruijn graph     0.56 [0;1]  0.56 [0;1]  0.56 [0;1]  "
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ref_stats_per_assembler_algo"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reference stats per assembler"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_stats_per_assembler = pd.DataFrame()\n",
    "for assembler in reference_pipeline_metrics_df.assembler.unique():\n",
    "    row={'Assembler': assembler, \n",
    "         'Type':reference_pipeline_metrics_df[reference_pipeline_metrics_df.assembler == assembler].type.unique()[0],\n",
    "         'Algorithm': reference_pipeline_metrics_df[reference_pipeline_metrics_df.assembler == assembler].algorythm.unique()[0]}\n",
    "    lala=reference_pipeline_metrics_df[reference_pipeline_metrics_df.assembler == assembler].describe()\n",
    "    for column in lala.columns:\n",
    "        mean = lala.loc['mean',column]\n",
    "        minimum = lala.loc['min',column]\n",
    "        maximum = lala.loc['max',column]\n",
    "        row[column] = \"{} [{};{}]\".format(round(mean, 2),int(minimum), int(maximum))\n",
    "    ref_stats_per_assembler=ref_stats_per_assembler.append(row, ignore_index=True)\n",
    "ref_stats_per_assembler = ref_stats_per_assembler.set_index('Assembler')\n",
    "ref_stats_per_assembler.to_csv(\"Tables/Results/Reference metrics per assembler.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>58.42 [0;1033]</td>\n",
       "      <td>0.25 [0;0]</td>\n",
       "      <td>86644.12 [0;351511]</td>\n",
       "      <td>86507.42 [0;351511]</td>\n",
       "      <td>77.04 [0;2170]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2166028.81 [0;6781605]</td>\n",
       "      <td>0.76 [0;0]</td>\n",
       "      <td>90.72 [0;1404]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>0.27 [0;8]</td>\n",
       "      <td>0.48 [0;14]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>41.8 [0;921]</td>\n",
       "      <td>0.03 [0;0]</td>\n",
       "      <td>6063.03 [0;65138]</td>\n",
       "      <td>5720.81 [0;64934]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1024596.01 [0;6515102]</td>\n",
       "      <td>0.3 [0;0]</td>\n",
       "      <td>213.69 [0;1495]</td>\n",
       "      <td>0.42 [0;1]</td>\n",
       "      <td>0.42 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.42 [0;1]</td>\n",
       "      <td>0.42 [0;1]</td>\n",
       "      <td>0.42 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>29.4 [0;459]</td>\n",
       "      <td>0.27 [0;1]</td>\n",
       "      <td>89885.56 [0;452364]</td>\n",
       "      <td>82812.65 [0;351186]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2175625.31 [0;6716522]</td>\n",
       "      <td>0.79 [0;1]</td>\n",
       "      <td>54.33 [0;604]</td>\n",
       "      <td>0.84 [0;1]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "      <td>0.12 [0;3]</td>\n",
       "      <td>0.25 [0;6]</td>\n",
       "      <td>0.84 [0;1]</td>\n",
       "      <td>0.84 [0;1]</td>\n",
       "      <td>0.84 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>55.7 [0;292]</td>\n",
       "      <td>0.21 [0;1]</td>\n",
       "      <td>53880.44 [0;475377]</td>\n",
       "      <td>52523.53 [0;475377]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2164508.2 [0;6710005]</td>\n",
       "      <td>0.77 [0;1]</td>\n",
       "      <td>98.23 [0;449]</td>\n",
       "      <td>0.84 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.9 [0;9]</td>\n",
       "      <td>1.55 [0;15]</td>\n",
       "      <td>0.85 [0;1]</td>\n",
       "      <td>0.89 [0;5]</td>\n",
       "      <td>0.84 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>15.83 [0;147]</td>\n",
       "      <td>0.29 [0;1]</td>\n",
       "      <td>120904.61 [0;543128]</td>\n",
       "      <td>122959.26 [0;543128]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2186131.55 [0;6738936]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>37.82 [0;223]</td>\n",
       "      <td>0.86 [0;1]</td>\n",
       "      <td>0.81 [0;1]</td>\n",
       "      <td>0.74 [0;4]</td>\n",
       "      <td>1.52 [0;7]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>19.53 [0;110]</td>\n",
       "      <td>0.21 [0;0]</td>\n",
       "      <td>52708.73 [0;257424]</td>\n",
       "      <td>52025.82 [0;257424]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1632657.63 [0;6717767]</td>\n",
       "      <td>0.63 [0;0]</td>\n",
       "      <td>107.52 [0;2402]</td>\n",
       "      <td>0.89 [0;1]</td>\n",
       "      <td>0.88 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.89 [0;1]</td>\n",
       "      <td>0.89 [0;1]</td>\n",
       "      <td>0.89 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>16.9 [0;90]</td>\n",
       "      <td>0.27 [0;1]</td>\n",
       "      <td>115349.42 [0;593167]</td>\n",
       "      <td>114559.37 [0;593167]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2182712.19 [0;6733500]</td>\n",
       "      <td>0.79 [0;1]</td>\n",
       "      <td>33.38 [0;167]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "      <td>0.85 [0;1]</td>\n",
       "      <td>0.15 [0;2]</td>\n",
       "      <td>0.35 [0;6]</td>\n",
       "      <td>0.88 [0;1]</td>\n",
       "      <td>0.88 [0;1]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>78.83 [0;649]</td>\n",
       "      <td>0.07 [0;0]</td>\n",
       "      <td>14570.44 [0;65138]</td>\n",
       "      <td>13911.24 [0;64960]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2033881.08 [0;6524118]</td>\n",
       "      <td>0.62 [0;0]</td>\n",
       "      <td>260.78 [0;1147]</td>\n",
       "      <td>0.71 [0;1]</td>\n",
       "      <td>0.69 [0;1]</td>\n",
       "      <td>0.03 [0;2]</td>\n",
       "      <td>0.06 [0;4]</td>\n",
       "      <td>0.71 [0;1]</td>\n",
       "      <td>0.71 [0;1]</td>\n",
       "      <td>0.71 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>45.12 [0;670]</td>\n",
       "      <td>0.2 [0;0]</td>\n",
       "      <td>59645.81 [0;351027]</td>\n",
       "      <td>58903.85 [0;351027]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1964267.06 [0;6673601]</td>\n",
       "      <td>0.7 [0;0]</td>\n",
       "      <td>102.28 [0;1877]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>0.7 [0;1]</td>\n",
       "      <td>0.58 [0;12]</td>\n",
       "      <td>1.03 [0;23]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>23.5 [0;835]</td>\n",
       "      <td>0.22 [0;1]</td>\n",
       "      <td>97446.0 [0;351104]</td>\n",
       "      <td>97219.7 [0;351104]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2166426.08 [0;6728888]</td>\n",
       "      <td>0.74 [0;1]</td>\n",
       "      <td>57.19 [0;1839]</td>\n",
       "      <td>0.84 [0;1]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>4.29 [0;565]</td>\n",
       "      <td>8.55 [0;1134]</td>\n",
       "      <td>0.86 [0;1]</td>\n",
       "      <td>0.88 [0;1]</td>\n",
       "      <td>0.84 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>13.6 [0;143]</td>\n",
       "      <td>0.29 [0;1]</td>\n",
       "      <td>134579.57 [0;475179]</td>\n",
       "      <td>130935.73 [0;475179]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2127157.78 [0;6723626]</td>\n",
       "      <td>0.74 [0;1]</td>\n",
       "      <td>28.08 [0;179]</td>\n",
       "      <td>0.76 [0;1]</td>\n",
       "      <td>0.76 [0;1]</td>\n",
       "      <td>0.08 [0;5]</td>\n",
       "      <td>0.15 [0;10]</td>\n",
       "      <td>0.76 [0;1]</td>\n",
       "      <td>0.76 [0;1]</td>\n",
       "      <td>0.76 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>23.5 [0;207]</td>\n",
       "      <td>0.07 [0;0]</td>\n",
       "      <td>22357.9 [0;338923]</td>\n",
       "      <td>26075.07 [0;464977]</td>\n",
       "      <td>403.23 [0;12098]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1282762.22 [0;6708956]</td>\n",
       "      <td>0.41 [0;0]</td>\n",
       "      <td>169.47 [0;1988]</td>\n",
       "      <td>0.49 [0;1]</td>\n",
       "      <td>0.48 [0;1]</td>\n",
       "      <td>0.29 [0;8]</td>\n",
       "      <td>0.46 [0;11]</td>\n",
       "      <td>0.49 [0;1]</td>\n",
       "      <td>0.49 [0;1]</td>\n",
       "      <td>0.49 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm             L90         LSA  \\\n",
       "Assembler                                                                       \n",
       "ABySS              Multiple k-mer De Bruijn graph  58.42 [0;1033]  0.25 [0;0]   \n",
       "BCALM2               Single k-mer De Bruijn graph    41.8 [0;921]  0.03 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph    29.4 [0;459]  0.27 [0;1]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph    55.7 [0;292]  0.21 [0;1]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph   15.83 [0;147]  0.29 [0;1]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph   19.53 [0;110]  0.21 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph     16.9 [0;90]  0.27 [0;1]   \n",
       "MINIA                Single k-mer De Bruijn graph   78.83 [0;649]  0.07 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph   45.12 [0;670]   0.2 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph    23.5 [0;835]  0.22 [0;1]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph    13.6 [0;143]  0.29 [0;1]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph    23.5 [0;207]  0.07 [0;0]   \n",
       "\n",
       "                                   NA50                  NG50  \\\n",
       "Assembler                                                       \n",
       "ABySS               86644.12 [0;351511]   86507.42 [0;351511]   \n",
       "BCALM2                6063.03 [0;65138]     5720.81 [0;64934]   \n",
       "GATBMiniaPipeline   89885.56 [0;452364]   82812.65 [0;351186]   \n",
       "IDBA-UD             53880.44 [0;475377]   52523.53 [0;475377]   \n",
       "MEGAHIT            120904.61 [0;543128]  122959.26 [0;543128]   \n",
       "MetaHipMer2         52708.73 [0;257424]   52025.82 [0;257424]   \n",
       "metaSPAdes         115349.42 [0;593167]  114559.37 [0;593167]   \n",
       "MINIA                14570.44 [0;65138]    13911.24 [0;64960]   \n",
       "SKESA               59645.81 [0;351027]   58903.85 [0;351027]   \n",
       "SPAdes               97446.0 [0;351104]    97219.7 [0;351104]   \n",
       "Unicycler          134579.57 [0;475179]  130935.73 [0;475179]   \n",
       "VelvetOptimiser      22357.9 [0;338923]   26075.07 [0;464977]   \n",
       "\n",
       "                                 Ns         Type               basepairs  \\\n",
       "Assembler                                                                  \n",
       "ABySS                77.04 [0;2170]  Metagenomic  2166028.81 [0;6781605]   \n",
       "BCALM2                    0.0 [0;0]      Genomic  1024596.01 [0;6515102]   \n",
       "GATBMiniaPipeline         0.0 [0;0]  Metagenomic  2175625.31 [0;6716522]   \n",
       "IDBA-UD                   0.0 [0;0]  Metagenomic   2164508.2 [0;6710005]   \n",
       "MEGAHIT                   0.0 [0;0]  Metagenomic  2186131.55 [0;6738936]   \n",
       "MetaHipMer2               0.0 [0;0]  Metagenomic  1632657.63 [0;6717767]   \n",
       "metaSPAdes                0.0 [0;0]  Metagenomic  2182712.19 [0;6733500]   \n",
       "MINIA                     0.0 [0;0]      Genomic  2033881.08 [0;6524118]   \n",
       "SKESA                     0.0 [0;0]      Genomic  1964267.06 [0;6673601]   \n",
       "SPAdes                    0.0 [0;0]      Genomic  2166426.08 [0;6728888]   \n",
       "Unicycler                 0.0 [0;0]      Genomic  2127157.78 [0;6723626]   \n",
       "VelvetOptimiser    403.23 [0;12098]  Metagenomic  1282762.22 [0;6708956]   \n",
       "\n",
       "                  breadth_of_coverage          contigs    identity  \\\n",
       "Assembler                                                            \n",
       "ABySS                      0.76 [0;0]   90.72 [0;1404]  0.82 [0;1]   \n",
       "BCALM2                      0.3 [0;0]  213.69 [0;1495]  0.42 [0;1]   \n",
       "GATBMiniaPipeline          0.79 [0;1]    54.33 [0;604]  0.84 [0;1]   \n",
       "IDBA-UD                    0.77 [0;1]    98.23 [0;449]  0.84 [0;1]   \n",
       "MEGAHIT                     0.8 [0;1]    37.82 [0;223]  0.86 [0;1]   \n",
       "MetaHipMer2                0.63 [0;0]  107.52 [0;2402]  0.89 [0;1]   \n",
       "metaSPAdes                 0.79 [0;1]    33.38 [0;167]  0.87 [0;1]   \n",
       "MINIA                      0.62 [0;0]  260.78 [0;1147]  0.71 [0;1]   \n",
       "SKESA                       0.7 [0;0]  102.28 [0;1877]   0.8 [0;1]   \n",
       "SPAdes                     0.74 [0;1]   57.19 [0;1839]  0.84 [0;1]   \n",
       "Unicycler                  0.74 [0;1]    28.08 [0;179]  0.76 [0;1]   \n",
       "VelvetOptimiser            0.41 [0;0]  169.47 [0;1988]  0.49 [0;1]   \n",
       "\n",
       "                  lowest_identity misassembled contigs misassembly events  \\\n",
       "Assembler                                                                   \n",
       "ABySS                   0.8 [0;1]           0.27 [0;8]        0.48 [0;14]   \n",
       "BCALM2                 0.42 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "GATBMiniaPipeline      0.83 [0;1]           0.12 [0;3]         0.25 [0;6]   \n",
       "IDBA-UD                0.75 [0;1]            0.9 [0;9]        1.55 [0;15]   \n",
       "MEGAHIT                0.81 [0;1]           0.74 [0;4]         1.52 [0;7]   \n",
       "MetaHipMer2            0.88 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "metaSPAdes             0.85 [0;1]           0.15 [0;2]         0.35 [0;6]   \n",
       "MINIA                  0.69 [0;1]           0.03 [0;2]         0.06 [0;4]   \n",
       "SKESA                   0.7 [0;1]          0.58 [0;12]        1.03 [0;23]   \n",
       "SPAdes                  0.8 [0;1]         4.29 [0;565]      8.55 [0;1134]   \n",
       "Unicycler              0.76 [0;1]           0.08 [0;5]        0.15 [0;10]   \n",
       "VelvetOptimiser        0.48 [0;1]           0.29 [0;8]        0.46 [0;11]   \n",
       "\n",
       "                  multiplicity   parsimony    validity  \n",
       "Assembler                                               \n",
       "ABySS               0.83 [0;1]  0.83 [0;1]  0.82 [0;1]  \n",
       "BCALM2              0.42 [0;1]  0.42 [0;1]  0.42 [0;1]  \n",
       "GATBMiniaPipeline   0.84 [0;1]  0.84 [0;1]  0.84 [0;1]  \n",
       "IDBA-UD             0.85 [0;1]  0.89 [0;5]  0.84 [0;1]  \n",
       "MEGAHIT             0.87 [0;1]  0.87 [0;1]  0.87 [0;1]  \n",
       "MetaHipMer2         0.89 [0;1]  0.89 [0;1]  0.89 [0;1]  \n",
       "metaSPAdes          0.88 [0;1]  0.88 [0;1]  0.87 [0;1]  \n",
       "MINIA               0.71 [0;1]  0.71 [0;1]  0.71 [0;1]  \n",
       "SKESA                0.8 [0;1]   0.8 [0;1]   0.8 [0;1]  \n",
       "SPAdes              0.86 [0;1]  0.88 [0;1]  0.84 [0;1]  \n",
       "Unicycler           0.76 [0;1]  0.76 [0;1]  0.76 [0;1]  \n",
       "VelvetOptimiser     0.49 [0;1]  0.49 [0;1]  0.49 [0;1]  "
      ]
     },
     "execution_count": 117,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ref_stats_per_assembler"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reference stats per reference replicon"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Per assembler for each reference replicon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Bacillus_subtilis', 'Enterococcus_faecalis',\n",
       "       'Escherichia_coli_plasmid', 'Escherichia_coli',\n",
       "       'Lactobacillus_fermentum', 'Listeria_monocytogenes',\n",
       "       'Pseudomonas_aeruginosa', 'Salmonella_enterica',\n",
       "       'Staphylococcus_aureus', 'Staphylococcus_aureus_plasmid1',\n",
       "       'Staphylococcus_aureus_plasmid2', 'Staphylococcus_aureus_plasmid3'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reference_pipeline_metrics_df.reference.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Bacillus_subtilis\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>119.7 [0;1033]</td>\n",
       "      <td>0.08 [0;0]</td>\n",
       "      <td>186473.7 [0;295233]</td>\n",
       "      <td>186436.1 [0;295233]</td>\n",
       "      <td>54.1 [0;400]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3559593.4 [0;3991152]</td>\n",
       "      <td>0.88 [0;0]</td>\n",
       "      <td>141.7 [0;1141]</td>\n",
       "      <td>0.9 [0;0]</td>\n",
       "      <td>0.89 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "      <td>0.9 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>27.25 [0;129]</td>\n",
       "      <td>0.01 [0;0]</td>\n",
       "      <td>19702.58 [0;65138]</td>\n",
       "      <td>19337.25 [0;64934]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2083135.33 [0;3951974]</td>\n",
       "      <td>0.51 [0;0]</td>\n",
       "      <td>455.08 [0;1467]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>64.67 [13;459]</td>\n",
       "      <td>0.13 [0;0]</td>\n",
       "      <td>207579.58 [8915;269695]</td>\n",
       "      <td>204452.42 [8732;263445]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3971062.42 [3882084;3984103]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>96.5 [27;604]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>0.25 [0;2]</td>\n",
       "      <td>0.75 [0;6]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>72.58 [18;292]</td>\n",
       "      <td>0.06 [0;0]</td>\n",
       "      <td>139056.17 [14006;252653]</td>\n",
       "      <td>128977.08 [13214;212969]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3971930.75 [3925882;3977845]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>113.5 [34;418]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.91 [0;0]</td>\n",
       "      <td>0.33 [0;2]</td>\n",
       "      <td>0.67 [0;5]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>26.17 [13;147]</td>\n",
       "      <td>0.12 [0;0]</td>\n",
       "      <td>239949.58 [26511;294549]</td>\n",
       "      <td>240054.5 [26209;294549]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3986036.5 [3969756;3989518]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>46.83 [27;223]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.83 [0;0]</td>\n",
       "      <td>0.83 [0;2]</td>\n",
       "      <td>1.92 [0;4]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>9.3 [0;16]</td>\n",
       "      <td>0.06 [0;0]</td>\n",
       "      <td>155049.2 [1145;257424]</td>\n",
       "      <td>154676.1 [0;257424]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2800013.9 [205541;3976387]</td>\n",
       "      <td>0.69 [0;0]</td>\n",
       "      <td>210.3 [27;1409]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>23.75 [9;66]</td>\n",
       "      <td>0.15 [0;0]</td>\n",
       "      <td>224502.58 [61490;343843]</td>\n",
       "      <td>224268.42 [62046;343843]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3979708.42 [3971865;3987932]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>45.25 [18;110]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>0.33 [0;2]</td>\n",
       "      <td>1.0 [0;6]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>135.83 [66;649]</td>\n",
       "      <td>0.03 [0;0]</td>\n",
       "      <td>49500.25 [6662;65138]</td>\n",
       "      <td>48602.83 [6313;64960]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>3939810.5 [3821823;3953200]</td>\n",
       "      <td>0.97 [0;0]</td>\n",
       "      <td>206.58 [116;783]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.33 [0;2]</td>\n",
       "      <td>0.58 [0;4]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>12.75 [0;20]</td>\n",
       "      <td>0.07 [0;0]</td>\n",
       "      <td>181820.75 [0;269493]</td>\n",
       "      <td>180238.5 [0;263164]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2977377.25 [0;3971432]</td>\n",
       "      <td>0.74 [0;0]</td>\n",
       "      <td>22.5 [0;34]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>19.67 [11;72]</td>\n",
       "      <td>0.16 [0;0]</td>\n",
       "      <td>258653.92 [56470;315090]</td>\n",
       "      <td>262779.92 [54127;315090]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>3977556.17 [3964395;3982668]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>37.42 [22;116]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.17 [0;2]</td>\n",
       "      <td>0.42 [0;5]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>20.17 [0;143]</td>\n",
       "      <td>0.15 [0;0]</td>\n",
       "      <td>247757.5 [0;324887]</td>\n",
       "      <td>236862.75 [0;324887]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>3468491.42 [0;3975910]</td>\n",
       "      <td>0.86 [0;0]</td>\n",
       "      <td>31.83 [0;179]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>30.75 [0;63]</td>\n",
       "      <td>0.03 [0;0]</td>\n",
       "      <td>35359.92 [0;69914]</td>\n",
       "      <td>33984.75 [0;69010]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2538310.0 [0;3953983]</td>\n",
       "      <td>0.63 [0;0]</td>\n",
       "      <td>262.08 [0;1254]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm              L90  \\\n",
       "Assembler                                                            \n",
       "ABySS              Multiple k-mer De Bruijn graph   119.7 [0;1033]   \n",
       "BCALM2               Single k-mer De Bruijn graph    27.25 [0;129]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph   64.67 [13;459]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph   72.58 [18;292]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph   26.17 [13;147]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph       9.3 [0;16]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph     23.75 [9;66]   \n",
       "MINIA                Single k-mer De Bruijn graph  135.83 [66;649]   \n",
       "SKESA              Multiple k-mer De Bruijn graph     12.75 [0;20]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph    19.67 [11;72]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph    20.17 [0;143]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph     30.75 [0;63]   \n",
       "\n",
       "                          LSA                      NA50  \\\n",
       "Assembler                                                 \n",
       "ABySS              0.08 [0;0]       186473.7 [0;295233]   \n",
       "BCALM2             0.01 [0;0]        19702.58 [0;65138]   \n",
       "GATBMiniaPipeline  0.13 [0;0]   207579.58 [8915;269695]   \n",
       "IDBA-UD            0.06 [0;0]  139056.17 [14006;252653]   \n",
       "MEGAHIT            0.12 [0;0]  239949.58 [26511;294549]   \n",
       "MetaHipMer2        0.06 [0;0]    155049.2 [1145;257424]   \n",
       "metaSPAdes         0.15 [0;0]  224502.58 [61490;343843]   \n",
       "MINIA              0.03 [0;0]     49500.25 [6662;65138]   \n",
       "SKESA              0.07 [0;0]      181820.75 [0;269493]   \n",
       "SPAdes             0.16 [0;0]  258653.92 [56470;315090]   \n",
       "Unicycler          0.15 [0;0]       247757.5 [0;324887]   \n",
       "VelvetOptimiser    0.03 [0;0]        35359.92 [0;69914]   \n",
       "\n",
       "                                       NG50            Ns         Type  \\\n",
       "Assembler                                                                \n",
       "ABySS                   186436.1 [0;295233]  54.1 [0;400]  Metagenomic   \n",
       "BCALM2                   19337.25 [0;64934]     0.0 [0;0]      Genomic   \n",
       "GATBMiniaPipeline   204452.42 [8732;263445]     0.0 [0;0]  Metagenomic   \n",
       "IDBA-UD            128977.08 [13214;212969]     0.0 [0;0]  Metagenomic   \n",
       "MEGAHIT             240054.5 [26209;294549]     0.0 [0;0]  Metagenomic   \n",
       "MetaHipMer2             154676.1 [0;257424]     0.0 [0;0]  Metagenomic   \n",
       "metaSPAdes         224268.42 [62046;343843]     0.0 [0;0]  Metagenomic   \n",
       "MINIA                 48602.83 [6313;64960]     0.0 [0;0]      Genomic   \n",
       "SKESA                   180238.5 [0;263164]     0.0 [0;0]      Genomic   \n",
       "SPAdes             262779.92 [54127;315090]     0.0 [0;0]      Genomic   \n",
       "Unicycler              236862.75 [0;324887]     0.0 [0;0]      Genomic   \n",
       "VelvetOptimiser          33984.75 [0;69010]     0.0 [0;0]  Metagenomic   \n",
       "\n",
       "                                      basepairs breadth_of_coverage  \\\n",
       "Assembler                                                             \n",
       "ABySS                     3559593.4 [0;3991152]          0.88 [0;0]   \n",
       "BCALM2                   2083135.33 [0;3951974]          0.51 [0;0]   \n",
       "GATBMiniaPipeline  3971062.42 [3882084;3984103]          0.98 [0;0]   \n",
       "IDBA-UD            3971930.75 [3925882;3977845]          0.98 [0;0]   \n",
       "MEGAHIT             3986036.5 [3969756;3989518]          0.99 [0;0]   \n",
       "MetaHipMer2          2800013.9 [205541;3976387]          0.69 [0;0]   \n",
       "metaSPAdes         3979708.42 [3971865;3987932]          0.98 [0;0]   \n",
       "MINIA               3939810.5 [3821823;3953200]          0.97 [0;0]   \n",
       "SKESA                    2977377.25 [0;3971432]          0.74 [0;0]   \n",
       "SPAdes             3977556.17 [3964395;3982668]          0.98 [0;0]   \n",
       "Unicycler                3468491.42 [0;3975910]          0.86 [0;0]   \n",
       "VelvetOptimiser           2538310.0 [0;3953983]          0.63 [0;0]   \n",
       "\n",
       "                            contigs    identity lowest_identity  \\\n",
       "Assembler                                                         \n",
       "ABySS                141.7 [0;1141]   0.9 [0;0]      0.89 [0;0]   \n",
       "BCALM2              455.08 [0;1467]  0.75 [0;1]      0.75 [0;1]   \n",
       "GATBMiniaPipeline     96.5 [27;604]   1.0 [0;1]      0.99 [0;1]   \n",
       "IDBA-UD              113.5 [34;418]   1.0 [0;0]      0.91 [0;0]   \n",
       "MEGAHIT              46.83 [27;223]  0.99 [0;0]      0.83 [0;0]   \n",
       "MetaHipMer2         210.3 [27;1409]   1.0 [0;1]      0.99 [0;1]   \n",
       "metaSPAdes           45.25 [18;110]   1.0 [0;0]      0.98 [0;0]   \n",
       "MINIA              206.58 [116;783]   1.0 [0;1]       1.0 [0;1]   \n",
       "SKESA                   22.5 [0;34]  0.75 [0;1]      0.75 [0;1]   \n",
       "SPAdes               37.42 [22;116]   1.0 [0;0]      0.99 [0;0]   \n",
       "Unicycler             31.83 [0;179]  0.92 [0;1]      0.92 [0;1]   \n",
       "VelvetOptimiser     262.08 [0;1254]  0.75 [0;1]      0.75 [0;1]   \n",
       "\n",
       "                  misassembled contigs misassembly events multiplicity  \\\n",
       "Assembler                                                                \n",
       "ABySS                        0.0 [0;0]          0.0 [0;0]    0.9 [0;1]   \n",
       "BCALM2                       0.0 [0;0]          0.0 [0;0]   0.75 [0;1]   \n",
       "GATBMiniaPipeline           0.25 [0;2]         0.75 [0;6]    1.0 [1;1]   \n",
       "IDBA-UD                     0.33 [0;2]         0.67 [0;5]    1.0 [1;1]   \n",
       "MEGAHIT                     0.83 [0;2]         1.92 [0;4]    1.0 [0;1]   \n",
       "MetaHipMer2                  0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "metaSPAdes                  0.33 [0;2]          1.0 [0;6]    1.0 [0;1]   \n",
       "MINIA                       0.33 [0;2]         0.58 [0;4]    1.0 [1;1]   \n",
       "SKESA                        0.0 [0;0]          0.0 [0;0]   0.75 [0;1]   \n",
       "SPAdes                      0.17 [0;2]         0.42 [0;5]    1.0 [0;1]   \n",
       "Unicycler                    0.0 [0;0]          0.0 [0;0]   0.92 [0;1]   \n",
       "VelvetOptimiser              0.0 [0;0]          0.0 [0;0]   0.75 [0;1]   \n",
       "\n",
       "                    parsimony    validity  \n",
       "Assembler                                  \n",
       "ABySS               0.9 [0;1]   0.9 [0;0]  \n",
       "BCALM2             0.75 [0;1]  0.75 [0;1]  \n",
       "GATBMiniaPipeline   1.0 [1;1]   1.0 [0;1]  \n",
       "IDBA-UD             1.0 [1;1]   1.0 [0;0]  \n",
       "MEGAHIT             1.0 [0;1]   1.0 [0;0]  \n",
       "MetaHipMer2         1.0 [1;1]   1.0 [0;1]  \n",
       "metaSPAdes          1.0 [0;1]   1.0 [0;1]  \n",
       "MINIA               1.0 [1;1]   1.0 [0;1]  \n",
       "SKESA              0.75 [0;1]  0.75 [0;1]  \n",
       "SPAdes              1.0 [0;1]   1.0 [0;1]  \n",
       "Unicycler          0.92 [0;1]  0.92 [0;1]  \n",
       "VelvetOptimiser    0.75 [0;1]  0.75 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Enterococcus_faecalis\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>18.5 [0;64]</td>\n",
       "      <td>0.12 [0;0]</td>\n",
       "      <td>153198.2 [0;241090]</td>\n",
       "      <td>153198.2 [0;241090]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2256766.3 [0;2828460]</td>\n",
       "      <td>0.79 [0;0]</td>\n",
       "      <td>33.7 [0;108]</td>\n",
       "      <td>0.8 [0;0]</td>\n",
       "      <td>0.79 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.81 [0;1]</td>\n",
       "      <td>0.81 [0;1]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>26.08 [0;107]</td>\n",
       "      <td>0.01 [0;0]</td>\n",
       "      <td>7915.25 [0;30453]</td>\n",
       "      <td>7605.0 [0;30452]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>745329.0 [0;2768590]</td>\n",
       "      <td>0.26 [0;0]</td>\n",
       "      <td>86.42 [0;180]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>13.75 [0;24]</td>\n",
       "      <td>0.13 [0;0]</td>\n",
       "      <td>135710.5 [0;235254]</td>\n",
       "      <td>135710.5 [0;235254]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2098979.17 [0;2820351]</td>\n",
       "      <td>0.74 [0;0]</td>\n",
       "      <td>28.08 [0;46]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.74 [0;1]</td>\n",
       "      <td>0.25 [0;1]</td>\n",
       "      <td>0.5 [0;2]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>49.25 [0;147]</td>\n",
       "      <td>0.08 [0;0]</td>\n",
       "      <td>61181.75 [0;113206]</td>\n",
       "      <td>62772.0 [0;118897]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2099448.75 [0;2811602]</td>\n",
       "      <td>0.74 [0;0]</td>\n",
       "      <td>84.75 [0;231]</td>\n",
       "      <td>0.75 [0;0]</td>\n",
       "      <td>0.7 [0;0]</td>\n",
       "      <td>0.75 [0;3]</td>\n",
       "      <td>1.5 [0;6]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>10.75 [0;17]</td>\n",
       "      <td>0.17 [0;0]</td>\n",
       "      <td>176503.25 [0;235381]</td>\n",
       "      <td>177310.75 [0;238546]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2101557.67 [0;2822833]</td>\n",
       "      <td>0.74 [0;0]</td>\n",
       "      <td>24.33 [0;35]</td>\n",
       "      <td>0.75 [0;0]</td>\n",
       "      <td>0.72 [0;0]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>1.0 [0;2]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>16.2 [0;27]</td>\n",
       "      <td>0.08 [0;0]</td>\n",
       "      <td>71897.8 [1166;118895]</td>\n",
       "      <td>70110.6 [0;118895]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1740712.0 [9974;2811369]</td>\n",
       "      <td>0.61 [0;0]</td>\n",
       "      <td>80.5 [5;179]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>11.25 [0;25]</td>\n",
       "      <td>0.13 [0;0]</td>\n",
       "      <td>155065.75 [0;238611]</td>\n",
       "      <td>155065.75 [0;238611]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2104706.5 [0;2818677]</td>\n",
       "      <td>0.74 [0;0]</td>\n",
       "      <td>23.75 [0;51]</td>\n",
       "      <td>0.75 [0;0]</td>\n",
       "      <td>0.73 [0;0]</td>\n",
       "      <td>0.25 [0;1]</td>\n",
       "      <td>0.5 [0;2]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>88.83 [0;149]</td>\n",
       "      <td>0.03 [0;0]</td>\n",
       "      <td>21664.83 [0;30453]</td>\n",
       "      <td>20172.75 [0;30453]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2070182.92 [0;2770364]</td>\n",
       "      <td>0.73 [0;0]</td>\n",
       "      <td>138.83 [0;222]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "      <td>0.91 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>22.75 [0;42]</td>\n",
       "      <td>0.09 [0;0]</td>\n",
       "      <td>79782.0 [0;152709]</td>\n",
       "      <td>79339.25 [0;152709]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2102692.5 [0;2810284]</td>\n",
       "      <td>0.74 [0;0]</td>\n",
       "      <td>41.67 [0;72]</td>\n",
       "      <td>0.91 [0;1]</td>\n",
       "      <td>0.7 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "      <td>0.91 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>12.25 [0;25]</td>\n",
       "      <td>0.15 [0;0]</td>\n",
       "      <td>145226.0 [0;239100]</td>\n",
       "      <td>143125.75 [0;239100]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2098387.33 [0;2812518]</td>\n",
       "      <td>0.74 [0;0]</td>\n",
       "      <td>29.0 [0;63]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "      <td>1.25 [0;6]</td>\n",
       "      <td>2.17 [0;8]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "      <td>0.89 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>7.75 [0;14]</td>\n",
       "      <td>0.19 [0;0]</td>\n",
       "      <td>212644.75 [0;377104]</td>\n",
       "      <td>200388.0 [0;377104]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2102459.0 [0;2813747]</td>\n",
       "      <td>0.74 [0;0]</td>\n",
       "      <td>17.5 [0;30]</td>\n",
       "      <td>0.75 [0;0]</td>\n",
       "      <td>0.75 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.75 [0;0]</td>\n",
       "      <td>0.75 [0;0]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>51.75 [0;106]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>15360.92 [0;30453]</td>\n",
       "      <td>14905.0 [0;30453]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1368705.17 [0;2736286]</td>\n",
       "      <td>0.48 [0;0]</td>\n",
       "      <td>80.58 [0;163]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm            L90         LSA  \\\n",
       "Assembler                                                                      \n",
       "ABySS              Multiple k-mer De Bruijn graph    18.5 [0;64]  0.12 [0;0]   \n",
       "BCALM2               Single k-mer De Bruijn graph  26.08 [0;107]  0.01 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph   13.75 [0;24]  0.13 [0;0]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  49.25 [0;147]  0.08 [0;0]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph   10.75 [0;17]  0.17 [0;0]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph    16.2 [0;27]  0.08 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph   11.25 [0;25]  0.13 [0;0]   \n",
       "MINIA                Single k-mer De Bruijn graph  88.83 [0;149]  0.03 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph   22.75 [0;42]  0.09 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph   12.25 [0;25]  0.15 [0;0]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph    7.75 [0;14]  0.19 [0;0]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph  51.75 [0;106]  0.02 [0;0]   \n",
       "\n",
       "                                    NA50                  NG50         Ns  \\\n",
       "Assembler                                                                   \n",
       "ABySS                153198.2 [0;241090]   153198.2 [0;241090]  0.0 [0;0]   \n",
       "BCALM2                 7915.25 [0;30453]      7605.0 [0;30452]  0.0 [0;0]   \n",
       "GATBMiniaPipeline    135710.5 [0;235254]   135710.5 [0;235254]  0.0 [0;0]   \n",
       "IDBA-UD              61181.75 [0;113206]    62772.0 [0;118897]  0.0 [0;0]   \n",
       "MEGAHIT             176503.25 [0;235381]  177310.75 [0;238546]  0.0 [0;0]   \n",
       "MetaHipMer2        71897.8 [1166;118895]    70110.6 [0;118895]  0.0 [0;0]   \n",
       "metaSPAdes          155065.75 [0;238611]  155065.75 [0;238611]  0.0 [0;0]   \n",
       "MINIA                 21664.83 [0;30453]    20172.75 [0;30453]  0.0 [0;0]   \n",
       "SKESA                 79782.0 [0;152709]   79339.25 [0;152709]  0.0 [0;0]   \n",
       "SPAdes               145226.0 [0;239100]  143125.75 [0;239100]  0.0 [0;0]   \n",
       "Unicycler           212644.75 [0;377104]   200388.0 [0;377104]  0.0 [0;0]   \n",
       "VelvetOptimiser       15360.92 [0;30453]     14905.0 [0;30453]  0.0 [0;0]   \n",
       "\n",
       "                          Type                 basepairs breadth_of_coverage  \\\n",
       "Assembler                                                                      \n",
       "ABySS              Metagenomic     2256766.3 [0;2828460]          0.79 [0;0]   \n",
       "BCALM2                 Genomic      745329.0 [0;2768590]          0.26 [0;0]   \n",
       "GATBMiniaPipeline  Metagenomic    2098979.17 [0;2820351]          0.74 [0;0]   \n",
       "IDBA-UD            Metagenomic    2099448.75 [0;2811602]          0.74 [0;0]   \n",
       "MEGAHIT            Metagenomic    2101557.67 [0;2822833]          0.74 [0;0]   \n",
       "MetaHipMer2        Metagenomic  1740712.0 [9974;2811369]          0.61 [0;0]   \n",
       "metaSPAdes         Metagenomic     2104706.5 [0;2818677]          0.74 [0;0]   \n",
       "MINIA                  Genomic    2070182.92 [0;2770364]          0.73 [0;0]   \n",
       "SKESA                  Genomic     2102692.5 [0;2810284]          0.74 [0;0]   \n",
       "SPAdes                 Genomic    2098387.33 [0;2812518]          0.74 [0;0]   \n",
       "Unicycler              Genomic     2102459.0 [0;2813747]          0.74 [0;0]   \n",
       "VelvetOptimiser    Metagenomic    1368705.17 [0;2736286]          0.48 [0;0]   \n",
       "\n",
       "                          contigs    identity lowest_identity  \\\n",
       "Assembler                                                       \n",
       "ABySS                33.7 [0;108]   0.8 [0;0]      0.79 [0;0]   \n",
       "BCALM2              86.42 [0;180]   0.5 [0;1]       0.5 [0;1]   \n",
       "GATBMiniaPipeline    28.08 [0;46]  0.75 [0;1]      0.74 [0;1]   \n",
       "IDBA-UD             84.75 [0;231]  0.75 [0;0]       0.7 [0;0]   \n",
       "MEGAHIT              24.33 [0;35]  0.75 [0;0]      0.72 [0;0]   \n",
       "MetaHipMer2          80.5 [5;179]   1.0 [0;1]       1.0 [0;1]   \n",
       "metaSPAdes           23.75 [0;51]  0.75 [0;0]      0.73 [0;0]   \n",
       "MINIA              138.83 [0;222]  0.92 [0;1]      0.91 [0;1]   \n",
       "SKESA                41.67 [0;72]  0.91 [0;1]       0.7 [0;1]   \n",
       "SPAdes                29.0 [0;63]   0.9 [0;1]      0.87 [0;1]   \n",
       "Unicycler             17.5 [0;30]  0.75 [0;0]      0.75 [0;0]   \n",
       "VelvetOptimiser     80.58 [0;163]  0.58 [0;1]      0.58 [0;1]   \n",
       "\n",
       "                  misassembled contigs misassembly events multiplicity  \\\n",
       "Assembler                                                                \n",
       "ABySS                        0.0 [0;0]          0.0 [0;0]   0.81 [0;1]   \n",
       "BCALM2                       0.0 [0;0]          0.0 [0;0]    0.5 [0;1]   \n",
       "GATBMiniaPipeline           0.25 [0;1]          0.5 [0;2]   0.75 [0;1]   \n",
       "IDBA-UD                     0.75 [0;3]          1.5 [0;6]   0.75 [0;1]   \n",
       "MEGAHIT                      0.5 [0;1]          1.0 [0;2]   0.75 [0;1]   \n",
       "MetaHipMer2                  0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "metaSPAdes                  0.25 [0;1]          0.5 [0;2]   0.75 [0;1]   \n",
       "MINIA                        0.0 [0;0]          0.0 [0;0]   0.92 [0;1]   \n",
       "SKESA                        0.5 [0;1]          0.5 [0;1]   0.92 [0;1]   \n",
       "SPAdes                      1.25 [0;6]         2.17 [0;8]   0.94 [0;1]   \n",
       "Unicycler                    0.0 [0;0]          0.0 [0;0]   0.75 [0;0]   \n",
       "VelvetOptimiser              0.0 [0;0]          0.0 [0;0]   0.58 [0;1]   \n",
       "\n",
       "                    parsimony    validity  \n",
       "Assembler                                  \n",
       "ABySS              0.81 [0;1]   0.8 [0;1]  \n",
       "BCALM2              0.5 [0;1]   0.5 [0;1]  \n",
       "GATBMiniaPipeline  0.75 [0;1]  0.75 [0;1]  \n",
       "IDBA-UD            0.75 [0;1]  0.75 [0;0]  \n",
       "MEGAHIT            0.75 [0;1]  0.75 [0;0]  \n",
       "MetaHipMer2         1.0 [1;1]   1.0 [1;1]  \n",
       "metaSPAdes         0.75 [0;1]  0.75 [0;1]  \n",
       "MINIA              0.92 [0;1]  0.92 [0;1]  \n",
       "SKESA              0.92 [0;1]  0.91 [0;1]  \n",
       "SPAdes             0.98 [0;1]  0.89 [0;1]  \n",
       "Unicycler          0.75 [0;0]  0.75 [0;1]  \n",
       "VelvetOptimiser    0.58 [0;1]  0.58 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Escherichia_coli_plasmid\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.2 [0;1]</td>\n",
       "      <td>0.13 [0;0]</td>\n",
       "      <td>12339.8 [0;55639]</td>\n",
       "      <td>11519.0 [0;55639]</td>\n",
       "      <td>93.6 [0;204]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>43179.1 [0;109424]</td>\n",
       "      <td>0.39 [0;0]</td>\n",
       "      <td>10.2 [0;26]</td>\n",
       "      <td>0.79 [0;0]</td>\n",
       "      <td>0.78 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>4.5 [0;18]</td>\n",
       "      <td>0.05 [0;0]</td>\n",
       "      <td>2576.75 [0;7117]</td>\n",
       "      <td>2436.75 [0;6638]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>48410.25 [0;102586]</td>\n",
       "      <td>0.44 [0;0]</td>\n",
       "      <td>13.5 [0;33]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>2.25 [0;7]</td>\n",
       "      <td>0.2 [0;0]</td>\n",
       "      <td>13154.5 [0;28587]</td>\n",
       "      <td>13000.0 [0;28587]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>78130.0 [0;109450]</td>\n",
       "      <td>0.71 [0;0]</td>\n",
       "      <td>9.25 [0;24]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>3.75 [0;8]</td>\n",
       "      <td>0.2 [0;0]</td>\n",
       "      <td>17077.5 [0;28584]</td>\n",
       "      <td>16985.25 [0;28584]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>80037.83 [0;108159]</td>\n",
       "      <td>0.73 [0;0]</td>\n",
       "      <td>5.83 [0;11]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>2.5 [0;6]</td>\n",
       "      <td>0.26 [0;0]</td>\n",
       "      <td>24913.5 [0;55277]</td>\n",
       "      <td>24075.0 [0;55277]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>81337.42 [0;109516]</td>\n",
       "      <td>0.74 [0;0]</td>\n",
       "      <td>4.08 [0;9]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>3.0 [0;7]</td>\n",
       "      <td>0.17 [0;0]</td>\n",
       "      <td>12704.4 [0;28267]</td>\n",
       "      <td>12253.5 [0;28267]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>64668.4 [0;107495]</td>\n",
       "      <td>0.59 [0;0]</td>\n",
       "      <td>4.8 [0;10]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>2.5 [0;6]</td>\n",
       "      <td>0.26 [0;0]</td>\n",
       "      <td>24339.0 [0;55105]</td>\n",
       "      <td>23644.0 [0;55105]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>80059.5 [0;108144]</td>\n",
       "      <td>0.73 [0;0]</td>\n",
       "      <td>3.75 [0;9]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>5.75 [0;17]</td>\n",
       "      <td>0.11 [0;0]</td>\n",
       "      <td>6296.75 [0;14811]</td>\n",
       "      <td>6139.5 [0;14811]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>74715.5 [0;104297]</td>\n",
       "      <td>0.68 [0;0]</td>\n",
       "      <td>15.5 [0;33]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.5 [0;2]</td>\n",
       "      <td>0.13 [0;0]</td>\n",
       "      <td>7802.25 [0;28384]</td>\n",
       "      <td>7096.0 [0;28384]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>40787.75 [0;107808]</td>\n",
       "      <td>0.37 [0;0]</td>\n",
       "      <td>10.25 [0;26]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>2.75 [0;6]</td>\n",
       "      <td>0.26 [0;0]</td>\n",
       "      <td>17829.83 [0;28541]</td>\n",
       "      <td>17008.75 [0;28541]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>81280.42 [0;109262]</td>\n",
       "      <td>0.74 [0;0]</td>\n",
       "      <td>4.58 [0;9]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.25 [0;1]</td>\n",
       "      <td>0.12 [0;0]</td>\n",
       "      <td>13749.25 [0;54997]</td>\n",
       "      <td>13233.0 [0;52932]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>26982.25 [0;107929]</td>\n",
       "      <td>0.25 [0;0]</td>\n",
       "      <td>0.5 [0;2]</td>\n",
       "      <td>0.25 [0;0]</td>\n",
       "      <td>0.25 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.25 [0;1]</td>\n",
       "      <td>0.25 [0;1]</td>\n",
       "      <td>0.25 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>1.75 [0;4]</td>\n",
       "      <td>0.15 [0;0]</td>\n",
       "      <td>12267.5 [0;27962]</td>\n",
       "      <td>12267.5 [0;27962]</td>\n",
       "      <td>257.25 [0;835]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>53669.5 [0;107377]</td>\n",
       "      <td>0.49 [0;0]</td>\n",
       "      <td>2.5 [0;6]</td>\n",
       "      <td>0.5 [0;0]</td>\n",
       "      <td>0.5 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm          L90         LSA  \\\n",
       "Assembler                                                                    \n",
       "ABySS              Multiple k-mer De Bruijn graph    0.2 [0;1]  0.13 [0;0]   \n",
       "BCALM2               Single k-mer De Bruijn graph   4.5 [0;18]  0.05 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph   2.25 [0;7]   0.2 [0;0]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph   3.75 [0;8]   0.2 [0;0]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph    2.5 [0;6]  0.26 [0;0]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph    3.0 [0;7]  0.17 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph    2.5 [0;6]  0.26 [0;0]   \n",
       "MINIA                Single k-mer De Bruijn graph  5.75 [0;17]  0.11 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph    0.5 [0;2]  0.13 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph   2.75 [0;6]  0.26 [0;0]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph   0.25 [0;1]  0.12 [0;0]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph   1.75 [0;4]  0.15 [0;0]   \n",
       "\n",
       "                                 NA50                NG50              Ns  \\\n",
       "Assembler                                                                   \n",
       "ABySS               12339.8 [0;55639]   11519.0 [0;55639]    93.6 [0;204]   \n",
       "BCALM2               2576.75 [0;7117]    2436.75 [0;6638]       0.0 [0;0]   \n",
       "GATBMiniaPipeline   13154.5 [0;28587]   13000.0 [0;28587]       0.0 [0;0]   \n",
       "IDBA-UD             17077.5 [0;28584]  16985.25 [0;28584]       0.0 [0;0]   \n",
       "MEGAHIT             24913.5 [0;55277]   24075.0 [0;55277]       0.0 [0;0]   \n",
       "MetaHipMer2         12704.4 [0;28267]   12253.5 [0;28267]       0.0 [0;0]   \n",
       "metaSPAdes          24339.0 [0;55105]   23644.0 [0;55105]       0.0 [0;0]   \n",
       "MINIA               6296.75 [0;14811]    6139.5 [0;14811]       0.0 [0;0]   \n",
       "SKESA               7802.25 [0;28384]    7096.0 [0;28384]       0.0 [0;0]   \n",
       "SPAdes             17829.83 [0;28541]  17008.75 [0;28541]       0.0 [0;0]   \n",
       "Unicycler          13749.25 [0;54997]   13233.0 [0;52932]       0.0 [0;0]   \n",
       "VelvetOptimiser     12267.5 [0;27962]   12267.5 [0;27962]  257.25 [0;835]   \n",
       "\n",
       "                          Type            basepairs breadth_of_coverage  \\\n",
       "Assembler                                                                 \n",
       "ABySS              Metagenomic   43179.1 [0;109424]          0.39 [0;0]   \n",
       "BCALM2                 Genomic  48410.25 [0;102586]          0.44 [0;0]   \n",
       "GATBMiniaPipeline  Metagenomic   78130.0 [0;109450]          0.71 [0;0]   \n",
       "IDBA-UD            Metagenomic  80037.83 [0;108159]          0.73 [0;0]   \n",
       "MEGAHIT            Metagenomic  81337.42 [0;109516]          0.74 [0;0]   \n",
       "MetaHipMer2        Metagenomic   64668.4 [0;107495]          0.59 [0;0]   \n",
       "metaSPAdes         Metagenomic   80059.5 [0;108144]          0.73 [0;0]   \n",
       "MINIA                  Genomic   74715.5 [0;104297]          0.68 [0;0]   \n",
       "SKESA                  Genomic  40787.75 [0;107808]          0.37 [0;0]   \n",
       "SPAdes                 Genomic  81280.42 [0;109262]          0.74 [0;0]   \n",
       "Unicycler              Genomic  26982.25 [0;107929]          0.25 [0;0]   \n",
       "VelvetOptimiser    Metagenomic   53669.5 [0;107377]          0.49 [0;0]   \n",
       "\n",
       "                        contigs    identity lowest_identity  \\\n",
       "Assembler                                                     \n",
       "ABySS               10.2 [0;26]  0.79 [0;0]      0.78 [0;0]   \n",
       "BCALM2              13.5 [0;33]   0.5 [0;1]       0.5 [0;1]   \n",
       "GATBMiniaPipeline   9.25 [0;24]  0.75 [0;1]      0.75 [0;1]   \n",
       "IDBA-UD             5.83 [0;11]  0.83 [0;1]      0.83 [0;1]   \n",
       "MEGAHIT              4.08 [0;9]  0.83 [0;1]      0.83 [0;1]   \n",
       "MetaHipMer2          4.8 [0;10]   0.9 [0;1]       0.9 [0;1]   \n",
       "metaSPAdes           3.75 [0;9]  0.75 [0;1]      0.75 [0;1]   \n",
       "MINIA               15.5 [0;33]  0.75 [0;1]      0.75 [0;1]   \n",
       "SKESA              10.25 [0;26]  0.75 [0;1]      0.75 [0;1]   \n",
       "SPAdes               4.58 [0;9]  0.83 [0;1]      0.83 [0;1]   \n",
       "Unicycler             0.5 [0;2]  0.25 [0;0]      0.25 [0;0]   \n",
       "VelvetOptimiser       2.5 [0;6]   0.5 [0;0]       0.5 [0;0]   \n",
       "\n",
       "                  misassembled contigs misassembly events multiplicity  \\\n",
       "Assembler                                                                \n",
       "ABySS                        0.0 [0;0]          0.0 [0;0]    0.8 [0;1]   \n",
       "BCALM2                       0.0 [0;0]          0.0 [0;0]    0.5 [0;1]   \n",
       "GATBMiniaPipeline            0.0 [0;0]          0.0 [0;0]   0.75 [0;1]   \n",
       "IDBA-UD                      0.0 [0;0]          0.0 [0;0]   0.83 [0;1]   \n",
       "MEGAHIT                      0.0 [0;0]          0.0 [0;0]   0.83 [0;1]   \n",
       "MetaHipMer2                  0.0 [0;0]          0.0 [0;0]    0.9 [0;1]   \n",
       "metaSPAdes                   0.0 [0;0]          0.0 [0;0]   0.75 [0;1]   \n",
       "MINIA                        0.0 [0;0]          0.0 [0;0]   0.75 [0;1]   \n",
       "SKESA                        0.0 [0;0]          0.0 [0;0]   0.75 [0;1]   \n",
       "SPAdes                       0.0 [0;0]          0.0 [0;0]   0.83 [0;1]   \n",
       "Unicycler                    0.0 [0;0]          0.0 [0;0]   0.25 [0;1]   \n",
       "VelvetOptimiser              0.0 [0;0]          0.0 [0;0]    0.5 [0;1]   \n",
       "\n",
       "                    parsimony    validity  \n",
       "Assembler                                  \n",
       "ABySS               0.8 [0;1]   0.8 [0;1]  \n",
       "BCALM2              0.5 [0;1]   0.5 [0;1]  \n",
       "GATBMiniaPipeline  0.75 [0;1]  0.75 [0;1]  \n",
       "IDBA-UD            0.83 [0;1]  0.83 [0;1]  \n",
       "MEGAHIT            0.83 [0;1]  0.83 [0;1]  \n",
       "MetaHipMer2         0.9 [0;1]   0.9 [0;1]  \n",
       "metaSPAdes         0.75 [0;1]  0.75 [0;1]  \n",
       "MINIA              0.75 [0;1]  0.75 [0;1]  \n",
       "SKESA              0.75 [0;1]  0.75 [0;1]  \n",
       "SPAdes             0.83 [0;1]  0.83 [0;1]  \n",
       "Unicycler          0.25 [0;1]  0.25 [0;1]  \n",
       "VelvetOptimiser     0.5 [0;1]   0.5 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Escherichia_coli\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>74.9 [0;133]</td>\n",
       "      <td>0.03 [0;0]</td>\n",
       "      <td>52249.4 [0;100101]</td>\n",
       "      <td>51970.3 [0;100101]</td>\n",
       "      <td>13.5 [0;45]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3751078.0 [0;4706885]</td>\n",
       "      <td>0.79 [0;0]</td>\n",
       "      <td>134.1 [0;234]</td>\n",
       "      <td>0.8 [0;0]</td>\n",
       "      <td>0.69 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.81 [0;1]</td>\n",
       "      <td>0.81 [0;1]</td>\n",
       "      <td>0.8 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1677.25 [0;4465]</td>\n",
       "      <td>1172.25 [0;3200]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1714793.33 [0;3766804]</td>\n",
       "      <td>0.36 [0;0]</td>\n",
       "      <td>653.75 [0;1495]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>77.75 [0;112]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>43740.92 [0;59656]</td>\n",
       "      <td>43462.75 [0;59530]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3468765.25 [0;4632389]</td>\n",
       "      <td>0.73 [0;0]</td>\n",
       "      <td>141.08 [0;199]</td>\n",
       "      <td>0.83 [0;0]</td>\n",
       "      <td>0.82 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "      <td>0.84 [0;1]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>187.0 [0;283]</td>\n",
       "      <td>0.01 [0;0]</td>\n",
       "      <td>17916.0 [1146;30548]</td>\n",
       "      <td>14932.0 [0;22537]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3403322.83 [2287;4589511]</td>\n",
       "      <td>0.71 [0;0]</td>\n",
       "      <td>313.08 [2;449]</td>\n",
       "      <td>0.97 [0;0]</td>\n",
       "      <td>0.51 [0;0]</td>\n",
       "      <td>4.5 [0;8]</td>\n",
       "      <td>8.08 [0;15]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.41 [1;5]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>38.0 [0;55]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>59995.75 [1049;90017]</td>\n",
       "      <td>63025.5 [0;102135]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3501717.92 [4237;4673019]</td>\n",
       "      <td>0.73 [0;0]</td>\n",
       "      <td>104.17 [3;141]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.9 [0;0]</td>\n",
       "      <td>2.5 [0;4]</td>\n",
       "      <td>4.25 [0;7]</td>\n",
       "      <td>1.01 [0;1]</td>\n",
       "      <td>1.01 [0;1]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>65.1 [0;110]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>35257.7 [1155;59122]</td>\n",
       "      <td>33943.8 [0;56794]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2805850.0 [1352;4613132]</td>\n",
       "      <td>0.59 [0;0]</td>\n",
       "      <td>147.3 [1;193]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>40.5 [0;90]</td>\n",
       "      <td>0.07 [0;0]</td>\n",
       "      <td>78366.92 [1088;138793]</td>\n",
       "      <td>76893.25 [0;138793]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3491713.33 [4384;4666787]</td>\n",
       "      <td>0.73 [0;0]</td>\n",
       "      <td>85.25 [3;167]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.94 [0;0]</td>\n",
       "      <td>0.83 [0;2]</td>\n",
       "      <td>2.33 [0;6]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.01 [0;1]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.01 [0;0]</td>\n",
       "      <td>3335.42 [0;4469]</td>\n",
       "      <td>2371.42 [0;3211]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2822981.67 [0;3773077]</td>\n",
       "      <td>0.59 [0;0]</td>\n",
       "      <td>847.75 [0;1147]</td>\n",
       "      <td>0.75 [0;0]</td>\n",
       "      <td>0.74 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>87.75 [0;152]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>34893.25 [0;62623]</td>\n",
       "      <td>35412.25 [0;66235]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>3428896.25 [0;4579934]</td>\n",
       "      <td>0.72 [0;0]</td>\n",
       "      <td>164.75 [0;270]</td>\n",
       "      <td>0.75 [0;0]</td>\n",
       "      <td>0.53 [0;0]</td>\n",
       "      <td>1.25 [0;3]</td>\n",
       "      <td>2.0 [0;4]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.76 [0;1]</td>\n",
       "      <td>0.75 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>43.25 [0;79]</td>\n",
       "      <td>0.06 [0;0]</td>\n",
       "      <td>63116.58 [1094;98812]</td>\n",
       "      <td>62796.75 [0;96999]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>3476288.08 [6019;4643850]</td>\n",
       "      <td>0.73 [0;0]</td>\n",
       "      <td>90.17 [4;143]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>0.75 [0;0]</td>\n",
       "      <td>0.75 [0;3]</td>\n",
       "      <td>1.42 [0;5]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.02 [0;1]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>34.0 [0;57]</td>\n",
       "      <td>0.06 [0;0]</td>\n",
       "      <td>81772.5 [0;124073]</td>\n",
       "      <td>80086.5 [0;119183]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>3463913.0 [0;4632276]</td>\n",
       "      <td>0.73 [0;0]</td>\n",
       "      <td>70.25 [0;111]</td>\n",
       "      <td>0.75 [0;0]</td>\n",
       "      <td>0.74 [0;0]</td>\n",
       "      <td>0.25 [0;1]</td>\n",
       "      <td>0.5 [0;2]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.01 [0;0]</td>\n",
       "      <td>2388.42 [0;4598]</td>\n",
       "      <td>1674.5 [0;3363]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1903172.0 [0;3810326]</td>\n",
       "      <td>0.4 [0;0]</td>\n",
       "      <td>548.08 [0;1098]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "      <td>0.59 [0;1]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm            L90         LSA  \\\n",
       "Assembler                                                                      \n",
       "ABySS              Multiple k-mer De Bruijn graph   74.9 [0;133]  0.03 [0;0]   \n",
       "BCALM2               Single k-mer De Bruijn graph      0.0 [0;0]   0.0 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph  77.75 [0;112]  0.02 [0;0]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  187.0 [0;283]  0.01 [0;0]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph    38.0 [0;55]  0.04 [0;0]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph   65.1 [0;110]  0.02 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph    40.5 [0;90]  0.07 [0;0]   \n",
       "MINIA                Single k-mer De Bruijn graph      0.0 [0;0]  0.01 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph  87.75 [0;152]  0.02 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph   43.25 [0;79]  0.06 [0;0]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph    34.0 [0;57]  0.06 [0;0]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph      0.0 [0;0]  0.01 [0;0]   \n",
       "\n",
       "                                     NA50                 NG50           Ns  \\\n",
       "Assembler                                                                     \n",
       "ABySS                  52249.4 [0;100101]   51970.3 [0;100101]  13.5 [0;45]   \n",
       "BCALM2                   1677.25 [0;4465]     1172.25 [0;3200]    0.0 [0;0]   \n",
       "GATBMiniaPipeline      43740.92 [0;59656]   43462.75 [0;59530]    0.0 [0;0]   \n",
       "IDBA-UD              17916.0 [1146;30548]    14932.0 [0;22537]    0.0 [0;0]   \n",
       "MEGAHIT             59995.75 [1049;90017]   63025.5 [0;102135]    0.0 [0;0]   \n",
       "MetaHipMer2          35257.7 [1155;59122]    33943.8 [0;56794]    0.0 [0;0]   \n",
       "metaSPAdes         78366.92 [1088;138793]  76893.25 [0;138793]    0.0 [0;0]   \n",
       "MINIA                    3335.42 [0;4469]     2371.42 [0;3211]    0.0 [0;0]   \n",
       "SKESA                  34893.25 [0;62623]   35412.25 [0;66235]    0.0 [0;0]   \n",
       "SPAdes              63116.58 [1094;98812]   62796.75 [0;96999]    0.0 [0;0]   \n",
       "Unicycler              81772.5 [0;124073]   80086.5 [0;119183]    0.0 [0;0]   \n",
       "VelvetOptimiser          2388.42 [0;4598]      1674.5 [0;3363]    0.0 [0;0]   \n",
       "\n",
       "                          Type                  basepairs breadth_of_coverage  \\\n",
       "Assembler                                                                       \n",
       "ABySS              Metagenomic      3751078.0 [0;4706885]          0.79 [0;0]   \n",
       "BCALM2                 Genomic     1714793.33 [0;3766804]          0.36 [0;0]   \n",
       "GATBMiniaPipeline  Metagenomic     3468765.25 [0;4632389]          0.73 [0;0]   \n",
       "IDBA-UD            Metagenomic  3403322.83 [2287;4589511]          0.71 [0;0]   \n",
       "MEGAHIT            Metagenomic  3501717.92 [4237;4673019]          0.73 [0;0]   \n",
       "MetaHipMer2        Metagenomic   2805850.0 [1352;4613132]          0.59 [0;0]   \n",
       "metaSPAdes         Metagenomic  3491713.33 [4384;4666787]          0.73 [0;0]   \n",
       "MINIA                  Genomic     2822981.67 [0;3773077]          0.59 [0;0]   \n",
       "SKESA                  Genomic     3428896.25 [0;4579934]          0.72 [0;0]   \n",
       "SPAdes                 Genomic  3476288.08 [6019;4643850]          0.73 [0;0]   \n",
       "Unicycler              Genomic      3463913.0 [0;4632276]          0.73 [0;0]   \n",
       "VelvetOptimiser    Metagenomic      1903172.0 [0;3810326]           0.4 [0;0]   \n",
       "\n",
       "                           contigs    identity lowest_identity  \\\n",
       "Assembler                                                        \n",
       "ABySS                134.1 [0;234]   0.8 [0;0]      0.69 [0;0]   \n",
       "BCALM2             653.75 [0;1495]   0.5 [0;1]       0.5 [0;1]   \n",
       "GATBMiniaPipeline   141.08 [0;199]  0.83 [0;0]      0.82 [0;0]   \n",
       "IDBA-UD             313.08 [2;449]  0.97 [0;0]      0.51 [0;0]   \n",
       "MEGAHIT             104.17 [3;141]  0.99 [0;0]       0.9 [0;0]   \n",
       "MetaHipMer2          147.3 [1;193]   1.0 [0;1]      0.99 [0;1]   \n",
       "metaSPAdes           85.25 [3;167]  0.99 [0;0]      0.94 [0;0]   \n",
       "MINIA              847.75 [0;1147]  0.75 [0;0]      0.74 [0;0]   \n",
       "SKESA               164.75 [0;270]  0.75 [0;0]      0.53 [0;0]   \n",
       "SPAdes               90.17 [4;143]  0.98 [0;0]      0.75 [0;0]   \n",
       "Unicycler            70.25 [0;111]  0.75 [0;0]      0.74 [0;0]   \n",
       "VelvetOptimiser    548.08 [0;1098]  0.58 [0;1]      0.58 [0;1]   \n",
       "\n",
       "                  misassembled contigs misassembly events multiplicity  \\\n",
       "Assembler                                                                \n",
       "ABySS                        0.0 [0;0]          0.0 [0;0]   0.81 [0;1]   \n",
       "BCALM2                       0.0 [0;0]          0.0 [0;0]    0.5 [0;1]   \n",
       "GATBMiniaPipeline            0.0 [0;0]          0.0 [0;0]   0.83 [0;1]   \n",
       "IDBA-UD                      4.5 [0;8]        8.08 [0;15]    1.0 [0;1]   \n",
       "MEGAHIT                      2.5 [0;4]         4.25 [0;7]   1.01 [0;1]   \n",
       "MetaHipMer2                  0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "metaSPAdes                  0.83 [0;2]         2.33 [0;6]    1.0 [0;1]   \n",
       "MINIA                        0.0 [0;0]          0.0 [0;0]   0.75 [0;1]   \n",
       "SKESA                       1.25 [0;3]          2.0 [0;4]   0.75 [0;1]   \n",
       "SPAdes                      0.75 [0;3]         1.42 [0;5]    1.0 [0;1]   \n",
       "Unicycler                   0.25 [0;1]          0.5 [0;2]   0.75 [0;1]   \n",
       "VelvetOptimiser              0.0 [0;0]          0.0 [0;0]   0.58 [0;1]   \n",
       "\n",
       "                    parsimony    validity  \n",
       "Assembler                                  \n",
       "ABySS              0.81 [0;1]   0.8 [0;0]  \n",
       "BCALM2              0.5 [0;1]   0.5 [0;1]  \n",
       "GATBMiniaPipeline  0.84 [0;1]  0.83 [0;1]  \n",
       "IDBA-UD            1.41 [1;5]  0.92 [0;1]  \n",
       "MEGAHIT            1.01 [0;1]   1.0 [0;0]  \n",
       "MetaHipMer2         1.0 [1;1]   1.0 [0;1]  \n",
       "metaSPAdes         1.01 [0;1]  0.99 [0;1]  \n",
       "MINIA              0.75 [0;1]  0.75 [0;1]  \n",
       "SKESA              0.76 [0;1]  0.75 [0;0]  \n",
       "SPAdes             1.02 [0;1]  0.98 [0;0]  \n",
       "Unicycler          0.75 [0;1]  0.75 [0;1]  \n",
       "VelvetOptimiser    0.59 [0;1]  0.58 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Lactobacillus_fermentum\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>46.2 [0;106]</td>\n",
       "      <td>0.05 [0;0]</td>\n",
       "      <td>29410.2 [0;42368]</td>\n",
       "      <td>29164.3 [0;42368]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1474890.7 [0;1873053]</td>\n",
       "      <td>0.77 [0;0]</td>\n",
       "      <td>70.6 [0;137]</td>\n",
       "      <td>0.8 [0;0]</td>\n",
       "      <td>0.8 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.81 [0;1]</td>\n",
       "      <td>0.81 [0;1]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>52.0 [0;208]</td>\n",
       "      <td>0.01 [0;0]</td>\n",
       "      <td>2903.17 [0;11209]</td>\n",
       "      <td>2558.5 [0;10234]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>437345.17 [0;1748625]</td>\n",
       "      <td>0.23 [0;0]</td>\n",
       "      <td>59.17 [0;236]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>40.5 [0;55]</td>\n",
       "      <td>0.05 [0;0]</td>\n",
       "      <td>30517.08 [0;40488]</td>\n",
       "      <td>29442.75 [0;40151]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1347280.58 [0;1798516]</td>\n",
       "      <td>0.71 [0;0]</td>\n",
       "      <td>65.42 [0;89]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>61.25 [0;131]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>24333.25 [0;40364]</td>\n",
       "      <td>22633.25 [0;37467]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1339735.58 [0;1790018]</td>\n",
       "      <td>0.7 [0;0]</td>\n",
       "      <td>87.17 [0;171]</td>\n",
       "      <td>0.83 [0;0]</td>\n",
       "      <td>0.83 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>32.0 [0;53]</td>\n",
       "      <td>0.05 [0;0]</td>\n",
       "      <td>33379.5 [0;46932]</td>\n",
       "      <td>34157.0 [0;52972]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1356161.0 [0;1808544]</td>\n",
       "      <td>0.71 [0;0]</td>\n",
       "      <td>62.92 [0;88]</td>\n",
       "      <td>0.83 [0;0]</td>\n",
       "      <td>0.74 [0;0]</td>\n",
       "      <td>1.75 [0;3]</td>\n",
       "      <td>3.92 [0;6]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "      <td>0.83 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>32.7 [0;55]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>24570.9 [0;40362]</td>\n",
       "      <td>24040.2 [0;40067]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1075233.4 [0;1787622]</td>\n",
       "      <td>0.56 [0;0]</td>\n",
       "      <td>54.3 [0;87]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>40.0 [0;64]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>32383.92 [0;46069]</td>\n",
       "      <td>29878.75 [0;45077]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1344747.75 [0;1796429]</td>\n",
       "      <td>0.71 [0;0]</td>\n",
       "      <td>63.67 [0;95]</td>\n",
       "      <td>0.92 [0;0]</td>\n",
       "      <td>0.91 [0;0]</td>\n",
       "      <td>0.25 [0;1]</td>\n",
       "      <td>0.25 [0;1]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>168.83 [0;263]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>8021.83 [0;11293]</td>\n",
       "      <td>7470.25 [0;10836]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1310547.08 [0;1754083]</td>\n",
       "      <td>0.69 [0;0]</td>\n",
       "      <td>189.25 [0;282]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>47.25 [0;69]</td>\n",
       "      <td>0.05 [0;0]</td>\n",
       "      <td>27994.75 [0;40164]</td>\n",
       "      <td>25910.75 [0;39869]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1327082.75 [0;1776186]</td>\n",
       "      <td>0.7 [0;0]</td>\n",
       "      <td>66.5 [0;94]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>45.25 [0;76]</td>\n",
       "      <td>0.05 [0;0]</td>\n",
       "      <td>29460.75 [1408;46123]</td>\n",
       "      <td>27911.75 [0;45008]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1342755.75 [1368;1792828]</td>\n",
       "      <td>0.7 [0;0]</td>\n",
       "      <td>72.42 [1;109]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>1.0 [0;4]</td>\n",
       "      <td>2.0 [0;8]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>38.75 [0;55]</td>\n",
       "      <td>0.05 [0;0]</td>\n",
       "      <td>32703.5 [0;45959]</td>\n",
       "      <td>30177.25 [0;40306]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1330405.25 [0;1775427]</td>\n",
       "      <td>0.7 [0;0]</td>\n",
       "      <td>59.0 [0;80]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>101.25 [0;207]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>5929.92 [0;11784]</td>\n",
       "      <td>5520.25 [0;11122]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>875246.42 [0;1751517]</td>\n",
       "      <td>0.46 [0;0]</td>\n",
       "      <td>115.58 [0;235]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm             L90         LSA  \\\n",
       "Assembler                                                                       \n",
       "ABySS              Multiple k-mer De Bruijn graph    46.2 [0;106]  0.05 [0;0]   \n",
       "BCALM2               Single k-mer De Bruijn graph    52.0 [0;208]  0.01 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph     40.5 [0;55]  0.05 [0;0]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph   61.25 [0;131]  0.04 [0;0]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph     32.0 [0;53]  0.05 [0;0]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph     32.7 [0;55]  0.04 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph     40.0 [0;64]  0.04 [0;0]   \n",
       "MINIA                Single k-mer De Bruijn graph  168.83 [0;263]  0.02 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph    47.25 [0;69]  0.05 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph    45.25 [0;76]  0.05 [0;0]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph    38.75 [0;55]  0.05 [0;0]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph  101.25 [0;207]  0.02 [0;0]   \n",
       "\n",
       "                                    NA50                NG50         Ns  \\\n",
       "Assembler                                                                 \n",
       "ABySS                  29410.2 [0;42368]   29164.3 [0;42368]  0.0 [0;0]   \n",
       "BCALM2                 2903.17 [0;11209]    2558.5 [0;10234]  0.0 [0;0]   \n",
       "GATBMiniaPipeline     30517.08 [0;40488]  29442.75 [0;40151]  0.0 [0;0]   \n",
       "IDBA-UD               24333.25 [0;40364]  22633.25 [0;37467]  0.0 [0;0]   \n",
       "MEGAHIT                33379.5 [0;46932]   34157.0 [0;52972]  0.0 [0;0]   \n",
       "MetaHipMer2            24570.9 [0;40362]   24040.2 [0;40067]  0.0 [0;0]   \n",
       "metaSPAdes            32383.92 [0;46069]  29878.75 [0;45077]  0.0 [0;0]   \n",
       "MINIA                  8021.83 [0;11293]   7470.25 [0;10836]  0.0 [0;0]   \n",
       "SKESA                 27994.75 [0;40164]  25910.75 [0;39869]  0.0 [0;0]   \n",
       "SPAdes             29460.75 [1408;46123]  27911.75 [0;45008]  0.0 [0;0]   \n",
       "Unicycler              32703.5 [0;45959]  30177.25 [0;40306]  0.0 [0;0]   \n",
       "VelvetOptimiser        5929.92 [0;11784]   5520.25 [0;11122]  0.0 [0;0]   \n",
       "\n",
       "                          Type                  basepairs breadth_of_coverage  \\\n",
       "Assembler                                                                       \n",
       "ABySS              Metagenomic      1474890.7 [0;1873053]          0.77 [0;0]   \n",
       "BCALM2                 Genomic      437345.17 [0;1748625]          0.23 [0;0]   \n",
       "GATBMiniaPipeline  Metagenomic     1347280.58 [0;1798516]          0.71 [0;0]   \n",
       "IDBA-UD            Metagenomic     1339735.58 [0;1790018]           0.7 [0;0]   \n",
       "MEGAHIT            Metagenomic      1356161.0 [0;1808544]          0.71 [0;0]   \n",
       "MetaHipMer2        Metagenomic      1075233.4 [0;1787622]          0.56 [0;0]   \n",
       "metaSPAdes         Metagenomic     1344747.75 [0;1796429]          0.71 [0;0]   \n",
       "MINIA                  Genomic     1310547.08 [0;1754083]          0.69 [0;0]   \n",
       "SKESA                  Genomic     1327082.75 [0;1776186]           0.7 [0;0]   \n",
       "SPAdes                 Genomic  1342755.75 [1368;1792828]           0.7 [0;0]   \n",
       "Unicycler              Genomic     1330405.25 [0;1775427]           0.7 [0;0]   \n",
       "VelvetOptimiser    Metagenomic      875246.42 [0;1751517]          0.46 [0;0]   \n",
       "\n",
       "                          contigs    identity lowest_identity  \\\n",
       "Assembler                                                       \n",
       "ABySS                70.6 [0;137]   0.8 [0;0]       0.8 [0;0]   \n",
       "BCALM2              59.17 [0;236]  0.33 [0;1]      0.33 [0;1]   \n",
       "GATBMiniaPipeline    65.42 [0;89]  0.83 [0;1]      0.83 [0;1]   \n",
       "IDBA-UD             87.17 [0;171]  0.83 [0;0]      0.83 [0;0]   \n",
       "MEGAHIT              62.92 [0;88]  0.83 [0;0]      0.74 [0;0]   \n",
       "MetaHipMer2           54.3 [0;87]   0.9 [0;1]       0.9 [0;1]   \n",
       "metaSPAdes           63.67 [0;95]  0.92 [0;0]      0.91 [0;0]   \n",
       "MINIA              189.25 [0;282]  0.83 [0;1]      0.82 [0;1]   \n",
       "SKESA                 66.5 [0;94]  0.75 [0;1]      0.75 [0;1]   \n",
       "SPAdes              72.42 [1;109]  0.99 [0;0]      0.99 [0;0]   \n",
       "Unicycler             59.0 [0;80]  0.75 [0;1]      0.75 [0;1]   \n",
       "VelvetOptimiser    115.58 [0;235]  0.58 [0;1]      0.58 [0;1]   \n",
       "\n",
       "                  misassembled contigs misassembly events multiplicity  \\\n",
       "Assembler                                                                \n",
       "ABySS                        0.0 [0;0]          0.0 [0;0]   0.81 [0;1]   \n",
       "BCALM2                       0.0 [0;0]          0.0 [0;0]   0.33 [0;1]   \n",
       "GATBMiniaPipeline            0.0 [0;0]          0.0 [0;0]   0.83 [0;1]   \n",
       "IDBA-UD                      0.0 [0;0]          0.0 [0;0]   0.83 [0;1]   \n",
       "MEGAHIT                     1.75 [0;3]         3.92 [0;6]   0.83 [0;1]   \n",
       "MetaHipMer2                  0.0 [0;0]          0.0 [0;0]    0.9 [0;1]   \n",
       "metaSPAdes                  0.25 [0;1]         0.25 [0;1]   0.92 [0;1]   \n",
       "MINIA                        0.0 [0;0]          0.0 [0;0]   0.83 [0;1]   \n",
       "SKESA                        0.0 [0;0]          0.0 [0;0]   0.75 [0;1]   \n",
       "SPAdes                       1.0 [0;4]          2.0 [0;8]    1.0 [0;1]   \n",
       "Unicycler                    0.0 [0;0]          0.0 [0;0]   0.75 [0;1]   \n",
       "VelvetOptimiser              0.0 [0;0]          0.0 [0;0]   0.58 [0;1]   \n",
       "\n",
       "                    parsimony    validity  \n",
       "Assembler                                  \n",
       "ABySS              0.81 [0;1]   0.8 [0;1]  \n",
       "BCALM2             0.33 [0;1]  0.33 [0;1]  \n",
       "GATBMiniaPipeline  0.83 [0;1]  0.83 [0;1]  \n",
       "IDBA-UD            0.83 [0;1]  0.83 [0;1]  \n",
       "MEGAHIT            0.83 [0;1]  0.83 [0;0]  \n",
       "MetaHipMer2         0.9 [0;1]   0.9 [0;1]  \n",
       "metaSPAdes         0.92 [0;1]  0.92 [0;1]  \n",
       "MINIA              0.83 [0;1]  0.83 [0;1]  \n",
       "SKESA              0.75 [0;1]  0.75 [0;1]  \n",
       "SPAdes              1.0 [1;1]  0.99 [0;1]  \n",
       "Unicycler          0.75 [0;1]  0.75 [0;1]  \n",
       "VelvetOptimiser    0.58 [0;1]  0.58 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Listeria_monocytogenes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>26.7 [8;147]</td>\n",
       "      <td>0.19 [0;0]</td>\n",
       "      <td>267347.9 [21261;351511]</td>\n",
       "      <td>267239.1 [20173;351511]</td>\n",
       "      <td>0.3 [0;1]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2956056.2 [2941416;2961911]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>42.6 [13;226]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>22.58 [0;75]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>17300.08 [0;64777]</td>\n",
       "      <td>16296.0 [0;62493]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1082146.42 [0;2921297]</td>\n",
       "      <td>0.36 [0;0]</td>\n",
       "      <td>128.58 [0;359]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>10.92 [10;12]</td>\n",
       "      <td>0.2 [0;0]</td>\n",
       "      <td>368416.08 [254631;452364]</td>\n",
       "      <td>294101.33 [155105;351186]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2942369.25 [2915611;2957674]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>20.92 [18;26]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>36.0 [9;108]</td>\n",
       "      <td>0.15 [0;0]</td>\n",
       "      <td>224219.25 [27259;475377]</td>\n",
       "      <td>223793.25 [25555;475377]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2934925.67 [2915464;2956853]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>60.75 [19;171]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>7.5 [6;9]</td>\n",
       "      <td>0.27 [0;0]</td>\n",
       "      <td>464670.75 [269400;543128]</td>\n",
       "      <td>465101.92 [274574;543128]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2949010.67 [2924395;2962451]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>19.42 [17;26]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.87 [0;0]</td>\n",
       "      <td>0.33 [0;1]</td>\n",
       "      <td>0.67 [0;2]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>7.8 [0;13]</td>\n",
       "      <td>0.09 [0;0]</td>\n",
       "      <td>135213.7 [1221;254627]</td>\n",
       "      <td>134822.8 [0;254627]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2156355.7 [450667;2949910]</td>\n",
       "      <td>0.72 [0;0]</td>\n",
       "      <td>229.1 [21;1074]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>9.08 [5;13]</td>\n",
       "      <td>0.22 [0;0]</td>\n",
       "      <td>365322.67 [235115;593167]</td>\n",
       "      <td>365322.67 [235115;593167]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2941835.25 [2920851;2964879]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>18.08 [12;27]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>110.08 [0;461]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>36451.17 [3362;64777]</td>\n",
       "      <td>35066.67 [2969;64473]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2862068.83 [2548083;2924618]</td>\n",
       "      <td>0.96 [0;0]</td>\n",
       "      <td>242.83 [87;918]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.91 [0;1]</td>\n",
       "      <td>0.08 [0;1]</td>\n",
       "      <td>0.08 [0;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>30.17 [0;232]</td>\n",
       "      <td>0.16 [0;0]</td>\n",
       "      <td>255132.67 [2534;351027]</td>\n",
       "      <td>255042.0 [1926;351027]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2856512.83 [2259852;2950392]</td>\n",
       "      <td>0.95 [0;0]</td>\n",
       "      <td>130.58 [19;1001]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>80.0 [0;835]</td>\n",
       "      <td>0.14 [0;0]</td>\n",
       "      <td>205423.42 [1964;351104]</td>\n",
       "      <td>205456.92 [1651;351104]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2806158.58 [1899833;2957750]</td>\n",
       "      <td>0.94 [0;0]</td>\n",
       "      <td>279.92 [13;1839]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>47.08 [0;565]</td>\n",
       "      <td>94.5 [0;1134]</td>\n",
       "      <td>1.03 [0;1]</td>\n",
       "      <td>1.03 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>10.08 [6;33]</td>\n",
       "      <td>0.23 [0;0]</td>\n",
       "      <td>363345.42 [126634;475179]</td>\n",
       "      <td>363345.42 [126634;475179]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2933698.83 [2914588;2953853]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>20.08 [15;64]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>35.0 [0;74]</td>\n",
       "      <td>0.03 [0;0]</td>\n",
       "      <td>24503.75 [0;52654]</td>\n",
       "      <td>24105.5 [0;51199]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1669912.5 [0;2918755]</td>\n",
       "      <td>0.56 [0;0]</td>\n",
       "      <td>140.0 [0;957]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm             L90         LSA  \\\n",
       "Assembler                                                                       \n",
       "ABySS              Multiple k-mer De Bruijn graph    26.7 [8;147]  0.19 [0;0]   \n",
       "BCALM2               Single k-mer De Bruijn graph    22.58 [0;75]  0.02 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph   10.92 [10;12]   0.2 [0;0]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph    36.0 [9;108]  0.15 [0;0]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph       7.5 [6;9]  0.27 [0;0]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph      7.8 [0;13]  0.09 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph     9.08 [5;13]  0.22 [0;0]   \n",
       "MINIA                Single k-mer De Bruijn graph  110.08 [0;461]  0.04 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph   30.17 [0;232]  0.16 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph    80.0 [0;835]  0.14 [0;0]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph    10.08 [6;33]  0.23 [0;0]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph     35.0 [0;74]  0.03 [0;0]   \n",
       "\n",
       "                                        NA50                       NG50  \\\n",
       "Assembler                                                                 \n",
       "ABySS                267347.9 [21261;351511]    267239.1 [20173;351511]   \n",
       "BCALM2                    17300.08 [0;64777]          16296.0 [0;62493]   \n",
       "GATBMiniaPipeline  368416.08 [254631;452364]  294101.33 [155105;351186]   \n",
       "IDBA-UD             224219.25 [27259;475377]   223793.25 [25555;475377]   \n",
       "MEGAHIT            464670.75 [269400;543128]  465101.92 [274574;543128]   \n",
       "MetaHipMer2           135213.7 [1221;254627]        134822.8 [0;254627]   \n",
       "metaSPAdes         365322.67 [235115;593167]  365322.67 [235115;593167]   \n",
       "MINIA                  36451.17 [3362;64777]      35066.67 [2969;64473]   \n",
       "SKESA                255132.67 [2534;351027]     255042.0 [1926;351027]   \n",
       "SPAdes               205423.42 [1964;351104]    205456.92 [1651;351104]   \n",
       "Unicycler          363345.42 [126634;475179]  363345.42 [126634;475179]   \n",
       "VelvetOptimiser           24503.75 [0;52654]          24105.5 [0;51199]   \n",
       "\n",
       "                          Ns         Type                     basepairs  \\\n",
       "Assembler                                                                 \n",
       "ABySS              0.3 [0;1]  Metagenomic   2956056.2 [2941416;2961911]   \n",
       "BCALM2             0.0 [0;0]      Genomic        1082146.42 [0;2921297]   \n",
       "GATBMiniaPipeline  0.0 [0;0]  Metagenomic  2942369.25 [2915611;2957674]   \n",
       "IDBA-UD            0.0 [0;0]  Metagenomic  2934925.67 [2915464;2956853]   \n",
       "MEGAHIT            0.0 [0;0]  Metagenomic  2949010.67 [2924395;2962451]   \n",
       "MetaHipMer2        0.0 [0;0]  Metagenomic    2156355.7 [450667;2949910]   \n",
       "metaSPAdes         0.0 [0;0]  Metagenomic  2941835.25 [2920851;2964879]   \n",
       "MINIA              0.0 [0;0]      Genomic  2862068.83 [2548083;2924618]   \n",
       "SKESA              0.0 [0;0]      Genomic  2856512.83 [2259852;2950392]   \n",
       "SPAdes             0.0 [0;0]      Genomic  2806158.58 [1899833;2957750]   \n",
       "Unicycler          0.0 [0;0]      Genomic  2933698.83 [2914588;2953853]   \n",
       "VelvetOptimiser    0.0 [0;0]  Metagenomic         1669912.5 [0;2918755]   \n",
       "\n",
       "                  breadth_of_coverage           contigs    identity  \\\n",
       "Assembler                                                             \n",
       "ABySS                      0.99 [0;0]     42.6 [13;226]   1.0 [0;0]   \n",
       "BCALM2                     0.36 [0;0]    128.58 [0;359]  0.58 [0;1]   \n",
       "GATBMiniaPipeline          0.98 [0;0]     20.92 [18;26]   1.0 [0;1]   \n",
       "IDBA-UD                    0.98 [0;0]    60.75 [19;171]   1.0 [0;0]   \n",
       "MEGAHIT                    0.99 [0;0]     19.42 [17;26]  0.99 [0;0]   \n",
       "MetaHipMer2                0.72 [0;0]   229.1 [21;1074]   1.0 [0;1]   \n",
       "metaSPAdes                 0.98 [0;0]     18.08 [12;27]   1.0 [0;0]   \n",
       "MINIA                      0.96 [0;0]   242.83 [87;918]   1.0 [0;1]   \n",
       "SKESA                      0.95 [0;0]  130.58 [19;1001]   1.0 [0;1]   \n",
       "SPAdes                     0.94 [0;0]  279.92 [13;1839]   1.0 [0;1]   \n",
       "Unicycler                  0.98 [0;0]     20.08 [15;64]   1.0 [0;1]   \n",
       "VelvetOptimiser            0.56 [0;0]     140.0 [0;957]  0.58 [0;1]   \n",
       "\n",
       "                  lowest_identity misassembled contigs misassembly events  \\\n",
       "Assembler                                                                   \n",
       "ABySS                   1.0 [0;0]            0.0 [0;0]          0.0 [0;0]   \n",
       "BCALM2                 0.58 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "GATBMiniaPipeline      0.99 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "IDBA-UD                0.99 [0;0]            0.0 [0;0]          0.0 [0;0]   \n",
       "MEGAHIT                0.87 [0;0]           0.33 [0;1]         0.67 [0;2]   \n",
       "MetaHipMer2            0.98 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "metaSPAdes             0.99 [0;0]            0.0 [0;0]          0.0 [0;0]   \n",
       "MINIA                  0.91 [0;1]           0.08 [0;1]         0.08 [0;1]   \n",
       "SKESA                   1.0 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "SPAdes                 0.95 [0;1]        47.08 [0;565]      94.5 [0;1134]   \n",
       "Unicycler              0.98 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "VelvetOptimiser        0.58 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "\n",
       "                  multiplicity   parsimony    validity  \n",
       "Assembler                                               \n",
       "ABySS               1.01 [1;1]  1.01 [1;1]   1.0 [0;1]  \n",
       "BCALM2              0.58 [0;1]  0.58 [0;1]  0.58 [0;1]  \n",
       "GATBMiniaPipeline    1.0 [0;1]   1.0 [0;1]   1.0 [0;1]  \n",
       "IDBA-UD              1.0 [0;1]   1.0 [0;1]   1.0 [0;1]  \n",
       "MEGAHIT              1.0 [0;1]   1.0 [0;1]   1.0 [0;0]  \n",
       "MetaHipMer2          1.0 [1;1]   1.0 [1;1]   1.0 [0;1]  \n",
       "metaSPAdes           1.0 [1;1]   1.0 [1;1]   1.0 [0;1]  \n",
       "MINIA                1.0 [1;1]   1.0 [1;1]   1.0 [0;1]  \n",
       "SKESA                1.0 [0;1]   1.0 [0;1]   1.0 [1;1]  \n",
       "SPAdes              1.03 [0;1]  1.03 [0;1]   1.0 [0;1]  \n",
       "Unicycler            1.0 [0;1]   1.0 [0;1]   1.0 [0;1]  \n",
       "VelvetOptimiser     0.58 [0;1]  0.58 [0;1]  0.58 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pseudomonas_aeruginosa\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>352.2 [0;1004]</td>\n",
       "      <td>0.03 [0;0]</td>\n",
       "      <td>64187.9 [1421;245341]</td>\n",
       "      <td>64283.9 [0;245341]</td>\n",
       "      <td>740.4 [0;2170]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>6039233.0 [179111;6781605]</td>\n",
       "      <td>0.89 [0;0]</td>\n",
       "      <td>537.0 [58;1404]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.86 [0;0]</td>\n",
       "      <td>2.8 [0;8]</td>\n",
       "      <td>4.7 [0;14]</td>\n",
       "      <td>1.02 [1;1]</td>\n",
       "      <td>1.02 [1;1]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>341.17 [0;921]</td>\n",
       "      <td>0.01 [0;0]</td>\n",
       "      <td>9563.08 [0;23566]</td>\n",
       "      <td>9106.17 [0;22511]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>3770698.17 [0;6515102]</td>\n",
       "      <td>0.56 [0;0]</td>\n",
       "      <td>457.08 [0;1142]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>53.17 [36;78]</td>\n",
       "      <td>0.06 [0;0]</td>\n",
       "      <td>159539.0 [88774;211558]</td>\n",
       "      <td>158995.5 [88774;211019]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>6700314.08 [6646876;6716522]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>122.83 [98;166]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.97 [0;1]</td>\n",
       "      <td>0.75 [0;3]</td>\n",
       "      <td>1.5 [0;6]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>72.42 [48;90]</td>\n",
       "      <td>0.05 [0;0]</td>\n",
       "      <td>101487.33 [84458;171408]</td>\n",
       "      <td>99368.25 [83502;171408]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>6700196.5 [6694930;6710005]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>158.58 [119;185]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.92 [0;0]</td>\n",
       "      <td>0.83 [0;2]</td>\n",
       "      <td>1.33 [0;4]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>28.08 [24;31]</td>\n",
       "      <td>0.07 [0;0]</td>\n",
       "      <td>254762.67 [215542;286027]</td>\n",
       "      <td>256946.67 [215542;294763]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>6733180.08 [6729202;6738936]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>80.33 [75;82]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.96 [0;0]</td>\n",
       "      <td>0.25 [0;1]</td>\n",
       "      <td>0.5 [0;2]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>23.7 [0;42]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>120278.8 [1126;211554]</td>\n",
       "      <td>119727.0 [0;211015]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>4435086.0 [93618;6717767]</td>\n",
       "      <td>0.65 [0;0]</td>\n",
       "      <td>325.3 [81;2402]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>29.75 [24;44]</td>\n",
       "      <td>0.11 [0;0]</td>\n",
       "      <td>227527.67 [151807;285156]</td>\n",
       "      <td>224893.58 [151807;257869]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>6726334.17 [6721688;6733500]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>68.5 [56;100]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>345.75 [291;407]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>21684.0 [18690;24665]</td>\n",
       "      <td>21007.5 [17851;24029]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>6509919.5 [6489664;6524118]</td>\n",
       "      <td>0.96 [0;0]</td>\n",
       "      <td>530.67 [463;598]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>225.5 [0;670]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>40350.75 [0;111919]</td>\n",
       "      <td>38670.67 [0;106185]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>5399196.33 [0;6673601]</td>\n",
       "      <td>0.79 [0;0]</td>\n",
       "      <td>588.33 [0;1877]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "      <td>0.51 [0;1]</td>\n",
       "      <td>4.25 [0;12]</td>\n",
       "      <td>8.67 [0;23]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>34.33 [26;60]</td>\n",
       "      <td>0.11 [0;0]</td>\n",
       "      <td>207363.33 [122703;264371]</td>\n",
       "      <td>207363.33 [122703;264371]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>6715515.75 [6682762;6728888]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>78.25 [63;129]</td>\n",
       "      <td>1.0 [0;0]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>0.58 [0;2]</td>\n",
       "      <td>0.92 [0;3]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>23.75 [20;25]</td>\n",
       "      <td>0.12 [0;0]</td>\n",
       "      <td>272524.67 [256472;337495]</td>\n",
       "      <td>272180.83 [239075;337495]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>6699740.58 [6551562;6723626]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>67.0 [54;77]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.42 [0;5]</td>\n",
       "      <td>0.83 [0;10]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>7.25 [0;15]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>151070.0 [0;338923]</td>\n",
       "      <td>201017.58 [0;464977]</td>\n",
       "      <td>4581.5 [0;12098]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3747048.67 [0;6708956]</td>\n",
       "      <td>0.55 [0;0]</td>\n",
       "      <td>272.67 [0;1988]</td>\n",
       "      <td>0.66 [0;0]</td>\n",
       "      <td>0.59 [0;0]</td>\n",
       "      <td>3.5 [0;8]</td>\n",
       "      <td>5.5 [0;11]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm               L90  \\\n",
       "Assembler                                                             \n",
       "ABySS              Multiple k-mer De Bruijn graph    352.2 [0;1004]   \n",
       "BCALM2               Single k-mer De Bruijn graph    341.17 [0;921]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph     53.17 [36;78]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph     72.42 [48;90]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph     28.08 [24;31]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph       23.7 [0;42]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph     29.75 [24;44]   \n",
       "MINIA                Single k-mer De Bruijn graph  345.75 [291;407]   \n",
       "SKESA              Multiple k-mer De Bruijn graph     225.5 [0;670]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph     34.33 [26;60]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph     23.75 [20;25]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph       7.25 [0;15]   \n",
       "\n",
       "                          LSA                       NA50  \\\n",
       "Assembler                                                  \n",
       "ABySS              0.03 [0;0]      64187.9 [1421;245341]   \n",
       "BCALM2             0.01 [0;0]          9563.08 [0;23566]   \n",
       "GATBMiniaPipeline  0.06 [0;0]    159539.0 [88774;211558]   \n",
       "IDBA-UD            0.05 [0;0]   101487.33 [84458;171408]   \n",
       "MEGAHIT            0.07 [0;0]  254762.67 [215542;286027]   \n",
       "MetaHipMer2        0.04 [0;0]     120278.8 [1126;211554]   \n",
       "metaSPAdes         0.11 [0;0]  227527.67 [151807;285156]   \n",
       "MINIA              0.02 [0;0]      21684.0 [18690;24665]   \n",
       "SKESA              0.02 [0;0]        40350.75 [0;111919]   \n",
       "SPAdes             0.11 [0;0]  207363.33 [122703;264371]   \n",
       "Unicycler          0.12 [0;0]  272524.67 [256472;337495]   \n",
       "VelvetOptimiser    0.04 [0;0]        151070.0 [0;338923]   \n",
       "\n",
       "                                        NG50                Ns         Type  \\\n",
       "Assembler                                                                     \n",
       "ABySS                     64283.9 [0;245341]    740.4 [0;2170]  Metagenomic   \n",
       "BCALM2                     9106.17 [0;22511]         0.0 [0;0]      Genomic   \n",
       "GATBMiniaPipeline    158995.5 [88774;211019]         0.0 [0;0]  Metagenomic   \n",
       "IDBA-UD              99368.25 [83502;171408]         0.0 [0;0]  Metagenomic   \n",
       "MEGAHIT            256946.67 [215542;294763]         0.0 [0;0]  Metagenomic   \n",
       "MetaHipMer2              119727.0 [0;211015]         0.0 [0;0]  Metagenomic   \n",
       "metaSPAdes         224893.58 [151807;257869]         0.0 [0;0]  Metagenomic   \n",
       "MINIA                  21007.5 [17851;24029]         0.0 [0;0]      Genomic   \n",
       "SKESA                    38670.67 [0;106185]         0.0 [0;0]      Genomic   \n",
       "SPAdes             207363.33 [122703;264371]         0.0 [0;0]      Genomic   \n",
       "Unicycler          272180.83 [239075;337495]         0.0 [0;0]      Genomic   \n",
       "VelvetOptimiser         201017.58 [0;464977]  4581.5 [0;12098]  Metagenomic   \n",
       "\n",
       "                                      basepairs breadth_of_coverage  \\\n",
       "Assembler                                                             \n",
       "ABySS                6039233.0 [179111;6781605]          0.89 [0;0]   \n",
       "BCALM2                   3770698.17 [0;6515102]          0.56 [0;0]   \n",
       "GATBMiniaPipeline  6700314.08 [6646876;6716522]          0.99 [0;0]   \n",
       "IDBA-UD             6700196.5 [6694930;6710005]          0.99 [0;0]   \n",
       "MEGAHIT            6733180.08 [6729202;6738936]          0.99 [0;0]   \n",
       "MetaHipMer2           4435086.0 [93618;6717767]          0.65 [0;0]   \n",
       "metaSPAdes         6726334.17 [6721688;6733500]          0.99 [0;0]   \n",
       "MINIA               6509919.5 [6489664;6524118]          0.96 [0;0]   \n",
       "SKESA                    5399196.33 [0;6673601]          0.79 [0;0]   \n",
       "SPAdes             6715515.75 [6682762;6728888]          0.99 [0;0]   \n",
       "Unicycler          6699740.58 [6551562;6723626]          0.99 [0;0]   \n",
       "VelvetOptimiser          3747048.67 [0;6708956]          0.55 [0;0]   \n",
       "\n",
       "                            contigs    identity lowest_identity  \\\n",
       "Assembler                                                         \n",
       "ABySS               537.0 [58;1404]   1.0 [0;0]      0.86 [0;0]   \n",
       "BCALM2              457.08 [0;1142]  0.58 [0;1]      0.58 [0;1]   \n",
       "GATBMiniaPipeline   122.83 [98;166]   1.0 [0;1]      0.97 [0;1]   \n",
       "IDBA-UD            158.58 [119;185]   1.0 [0;0]      0.92 [0;0]   \n",
       "MEGAHIT               80.33 [75;82]   1.0 [0;0]      0.96 [0;0]   \n",
       "MetaHipMer2         325.3 [81;2402]   1.0 [0;1]      0.99 [0;1]   \n",
       "metaSPAdes            68.5 [56;100]   1.0 [0;0]      0.98 [0;0]   \n",
       "MINIA              530.67 [463;598]   1.0 [0;0]      0.99 [0;0]   \n",
       "SKESA               588.33 [0;1877]  0.92 [0;1]      0.51 [0;1]   \n",
       "SPAdes               78.25 [63;129]   1.0 [0;0]      0.98 [0;0]   \n",
       "Unicycler              67.0 [54;77]   1.0 [0;1]      0.95 [0;1]   \n",
       "VelvetOptimiser     272.67 [0;1988]  0.66 [0;0]      0.59 [0;0]   \n",
       "\n",
       "                  misassembled contigs misassembly events multiplicity  \\\n",
       "Assembler                                                                \n",
       "ABySS                        2.8 [0;8]         4.7 [0;14]   1.02 [1;1]   \n",
       "BCALM2                       0.0 [0;0]          0.0 [0;0]   0.58 [0;1]   \n",
       "GATBMiniaPipeline           0.75 [0;3]          1.5 [0;6]    1.0 [1;1]   \n",
       "IDBA-UD                     0.83 [0;2]         1.33 [0;4]    1.0 [1;1]   \n",
       "MEGAHIT                     0.25 [0;1]          0.5 [0;2]    1.0 [1;1]   \n",
       "MetaHipMer2                  0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "metaSPAdes                   0.0 [0;0]          0.0 [0;0]   1.01 [1;1]   \n",
       "MINIA                        0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "SKESA                      4.25 [0;12]        8.67 [0;23]   0.92 [0;1]   \n",
       "SPAdes                      0.58 [0;2]         0.92 [0;3]    1.0 [0;1]   \n",
       "Unicycler                   0.42 [0;5]        0.83 [0;10]    1.0 [0;1]   \n",
       "VelvetOptimiser              3.5 [0;8]         5.5 [0;11]   0.67 [0;1]   \n",
       "\n",
       "                    parsimony    validity  \n",
       "Assembler                                  \n",
       "ABySS              1.02 [1;1]   1.0 [0;0]  \n",
       "BCALM2             0.58 [0;1]  0.58 [0;1]  \n",
       "GATBMiniaPipeline   1.0 [1;1]   1.0 [0;1]  \n",
       "IDBA-UD             1.0 [1;1]   1.0 [0;1]  \n",
       "MEGAHIT             1.0 [1;1]   1.0 [0;1]  \n",
       "MetaHipMer2         1.0 [1;1]   1.0 [0;1]  \n",
       "metaSPAdes         1.01 [1;1]   1.0 [0;1]  \n",
       "MINIA               1.0 [1;1]   1.0 [0;1]  \n",
       "SKESA              0.92 [0;1]  0.92 [0;1]  \n",
       "SPAdes              1.0 [0;1]   1.0 [0;1]  \n",
       "Unicycler           1.0 [0;1]   1.0 [0;1]  \n",
       "VelvetOptimiser    0.67 [0;1]  0.67 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salmonella_enterica\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>45.7 [0;118]</td>\n",
       "      <td>0.05 [0;0]</td>\n",
       "      <td>91283.4 [0;157988]</td>\n",
       "      <td>91039.3 [0;157988]</td>\n",
       "      <td>1.3 [0;2]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3752791.6 [0;4703394]</td>\n",
       "      <td>0.79 [0;0]</td>\n",
       "      <td>86.3 [0;214]</td>\n",
       "      <td>0.8 [0;0]</td>\n",
       "      <td>0.74 [0;0]</td>\n",
       "      <td>0.2 [0;1]</td>\n",
       "      <td>0.6 [0;4]</td>\n",
       "      <td>0.81 [0;1]</td>\n",
       "      <td>0.81 [0;1]</td>\n",
       "      <td>0.8 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1831.5 [0;5009]</td>\n",
       "      <td>1262.5 [0;3551]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1728825.25 [0;3817265]</td>\n",
       "      <td>0.36 [0;0]</td>\n",
       "      <td>639.5 [0;1479]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>65.75 [0;96]</td>\n",
       "      <td>0.05 [0;0]</td>\n",
       "      <td>44358.67 [0;64572]</td>\n",
       "      <td>42721.25 [0;60374]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3485469.5 [0;4652574]</td>\n",
       "      <td>0.73 [0;0]</td>\n",
       "      <td>125.83 [0;178]</td>\n",
       "      <td>0.92 [0;0]</td>\n",
       "      <td>0.81 [0;0]</td>\n",
       "      <td>0.25 [0;1]</td>\n",
       "      <td>0.25 [0;1]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>136.67 [0;200]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>20698.58 [1117;29331]</td>\n",
       "      <td>20595.25 [0;28547]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3438939.75 [1463;4618887]</td>\n",
       "      <td>0.72 [0;0]</td>\n",
       "      <td>272.83 [1;397]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "      <td>0.66 [0;0]</td>\n",
       "      <td>4.42 [0;9]</td>\n",
       "      <td>7.0 [0;14]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.02 [1;1]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>33.0 [0;47]</td>\n",
       "      <td>0.07 [0;0]</td>\n",
       "      <td>82471.08 [1086;120785]</td>\n",
       "      <td>83154.25 [0;120785]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3505110.83 [4780;4671620]</td>\n",
       "      <td>0.74 [0;0]</td>\n",
       "      <td>79.5 [2;105]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.94 [0;0]</td>\n",
       "      <td>1.0 [0;3]</td>\n",
       "      <td>2.0 [0;6]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.01 [1;1]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>51.6 [0;86]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>37797.7 [0;64529]</td>\n",
       "      <td>35786.1 [0;60329]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2815717.0 [0;4641887]</td>\n",
       "      <td>0.59 [0;0]</td>\n",
       "      <td>122.7 [0;162]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>32.5 [0;77]</td>\n",
       "      <td>0.06 [0;0]</td>\n",
       "      <td>108693.33 [1223;198784]</td>\n",
       "      <td>106754.75 [0;198784]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3505419.67 [2586;4680545]</td>\n",
       "      <td>0.74 [0;0]</td>\n",
       "      <td>66.58 [2;145]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.89 [0;0]</td>\n",
       "      <td>0.08 [0;1]</td>\n",
       "      <td>0.08 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.01 [0;1]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.01 [0;0]</td>\n",
       "      <td>4048.58 [1313;4996]</td>\n",
       "      <td>2614.25 [0;3551]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2863842.92 [1312;3826699]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>818.25 [1;1106]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [1;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>84.0 [0;144]</td>\n",
       "      <td>0.03 [0;0]</td>\n",
       "      <td>36470.25 [0;60912]</td>\n",
       "      <td>36079.25 [0;60131]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>3438815.75 [0;4612070]</td>\n",
       "      <td>0.72 [0;0]</td>\n",
       "      <td>148.75 [0;245]</td>\n",
       "      <td>0.75 [0;0]</td>\n",
       "      <td>0.46 [0;0]</td>\n",
       "      <td>1.0 [0;2]</td>\n",
       "      <td>1.25 [0;3]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.76 [0;1]</td>\n",
       "      <td>0.75 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>30.75 [0;50]</td>\n",
       "      <td>0.07 [0;0]</td>\n",
       "      <td>93997.67 [1251;164404]</td>\n",
       "      <td>91912.75 [0;157954]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>3487896.17 [1463;4666050]</td>\n",
       "      <td>0.73 [0;0]</td>\n",
       "      <td>69.5 [1;110]</td>\n",
       "      <td>0.99 [0;0]</td>\n",
       "      <td>0.79 [0;0]</td>\n",
       "      <td>0.42 [0;2]</td>\n",
       "      <td>0.42 [0;2]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.02 [1;1]</td>\n",
       "      <td>0.98 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>21.75 [0;35]</td>\n",
       "      <td>0.07 [0;0]</td>\n",
       "      <td>134890.5 [0;223960]</td>\n",
       "      <td>134200.0 [0;223960]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>3486500.5 [0;4660116]</td>\n",
       "      <td>0.73 [0;0]</td>\n",
       "      <td>52.25 [0;89]</td>\n",
       "      <td>0.75 [0;0]</td>\n",
       "      <td>0.75 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.01 [0;0]</td>\n",
       "      <td>2693.42 [0;5223]</td>\n",
       "      <td>1872.5 [0;3753]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1932472.67 [0;3867436]</td>\n",
       "      <td>0.41 [0;0]</td>\n",
       "      <td>526.83 [0;1058]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "      <td>0.58 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm             L90         LSA  \\\n",
       "Assembler                                                                       \n",
       "ABySS              Multiple k-mer De Bruijn graph    45.7 [0;118]  0.05 [0;0]   \n",
       "BCALM2               Single k-mer De Bruijn graph       0.0 [0;0]   0.0 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph    65.75 [0;96]  0.05 [0;0]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  136.67 [0;200]  0.02 [0;0]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph     33.0 [0;47]  0.07 [0;0]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph     51.6 [0;86]  0.04 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph     32.5 [0;77]  0.06 [0;0]   \n",
       "MINIA                Single k-mer De Bruijn graph       0.0 [0;0]  0.01 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph    84.0 [0;144]  0.03 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph    30.75 [0;50]  0.07 [0;0]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph    21.75 [0;35]  0.07 [0;0]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph       0.0 [0;0]  0.01 [0;0]   \n",
       "\n",
       "                                      NA50                  NG50         Ns  \\\n",
       "Assembler                                                                     \n",
       "ABySS                   91283.4 [0;157988]    91039.3 [0;157988]  1.3 [0;2]   \n",
       "BCALM2                     1831.5 [0;5009]       1262.5 [0;3551]  0.0 [0;0]   \n",
       "GATBMiniaPipeline       44358.67 [0;64572]    42721.25 [0;60374]  0.0 [0;0]   \n",
       "IDBA-UD              20698.58 [1117;29331]    20595.25 [0;28547]  0.0 [0;0]   \n",
       "MEGAHIT             82471.08 [1086;120785]   83154.25 [0;120785]  0.0 [0;0]   \n",
       "MetaHipMer2              37797.7 [0;64529]     35786.1 [0;60329]  0.0 [0;0]   \n",
       "metaSPAdes         108693.33 [1223;198784]  106754.75 [0;198784]  0.0 [0;0]   \n",
       "MINIA                  4048.58 [1313;4996]      2614.25 [0;3551]  0.0 [0;0]   \n",
       "SKESA                   36470.25 [0;60912]    36079.25 [0;60131]  0.0 [0;0]   \n",
       "SPAdes              93997.67 [1251;164404]   91912.75 [0;157954]  0.0 [0;0]   \n",
       "Unicycler              134890.5 [0;223960]   134200.0 [0;223960]  0.0 [0;0]   \n",
       "VelvetOptimiser           2693.42 [0;5223]       1872.5 [0;3753]  0.0 [0;0]   \n",
       "\n",
       "                          Type                  basepairs breadth_of_coverage  \\\n",
       "Assembler                                                                       \n",
       "ABySS              Metagenomic      3752791.6 [0;4703394]          0.79 [0;0]   \n",
       "BCALM2                 Genomic     1728825.25 [0;3817265]          0.36 [0;0]   \n",
       "GATBMiniaPipeline  Metagenomic      3485469.5 [0;4652574]          0.73 [0;0]   \n",
       "IDBA-UD            Metagenomic  3438939.75 [1463;4618887]          0.72 [0;0]   \n",
       "MEGAHIT            Metagenomic  3505110.83 [4780;4671620]          0.74 [0;0]   \n",
       "MetaHipMer2        Metagenomic      2815717.0 [0;4641887]          0.59 [0;0]   \n",
       "metaSPAdes         Metagenomic  3505419.67 [2586;4680545]          0.74 [0;0]   \n",
       "MINIA                  Genomic  2863842.92 [1312;3826699]           0.6 [0;0]   \n",
       "SKESA                  Genomic     3438815.75 [0;4612070]          0.72 [0;0]   \n",
       "SPAdes                 Genomic  3487896.17 [1463;4666050]          0.73 [0;0]   \n",
       "Unicycler              Genomic      3486500.5 [0;4660116]          0.73 [0;0]   \n",
       "VelvetOptimiser    Metagenomic     1932472.67 [0;3867436]          0.41 [0;0]   \n",
       "\n",
       "                           contigs    identity lowest_identity  \\\n",
       "Assembler                                                        \n",
       "ABySS                 86.3 [0;214]   0.8 [0;0]      0.74 [0;0]   \n",
       "BCALM2              639.5 [0;1479]   0.5 [0;1]       0.5 [0;1]   \n",
       "GATBMiniaPipeline   125.83 [0;178]  0.92 [0;0]      0.81 [0;0]   \n",
       "IDBA-UD             272.83 [1;397]  0.98 [0;0]      0.66 [0;0]   \n",
       "MEGAHIT               79.5 [2;105]  0.99 [0;0]      0.94 [0;0]   \n",
       "MetaHipMer2          122.7 [0;162]   0.9 [0;1]      0.87 [0;1]   \n",
       "metaSPAdes           66.58 [2;145]  0.99 [0;0]      0.89 [0;0]   \n",
       "MINIA              818.25 [1;1106]   1.0 [0;1]      0.95 [0;1]   \n",
       "SKESA               148.75 [0;245]  0.75 [0;0]      0.46 [0;0]   \n",
       "SPAdes                69.5 [1;110]  0.99 [0;0]      0.79 [0;0]   \n",
       "Unicycler             52.25 [0;89]  0.75 [0;0]      0.75 [0;0]   \n",
       "VelvetOptimiser    526.83 [0;1058]  0.58 [0;1]      0.58 [0;1]   \n",
       "\n",
       "                  misassembled contigs misassembly events multiplicity  \\\n",
       "Assembler                                                                \n",
       "ABySS                        0.2 [0;1]          0.6 [0;4]   0.81 [0;1]   \n",
       "BCALM2                       0.0 [0;0]          0.0 [0;0]    0.5 [0;1]   \n",
       "GATBMiniaPipeline           0.25 [0;1]         0.25 [0;1]   0.92 [0;1]   \n",
       "IDBA-UD                     4.42 [0;9]         7.0 [0;14]    1.0 [0;1]   \n",
       "MEGAHIT                      1.0 [0;3]          2.0 [0;6]    1.0 [0;1]   \n",
       "MetaHipMer2                  0.0 [0;0]          0.0 [0;0]    0.9 [0;1]   \n",
       "metaSPAdes                  0.08 [0;1]         0.08 [0;1]    1.0 [0;1]   \n",
       "MINIA                        0.0 [0;0]          0.0 [0;0]    1.0 [1;1]   \n",
       "SKESA                        1.0 [0;2]         1.25 [0;3]   0.75 [0;1]   \n",
       "SPAdes                      0.42 [0;2]         0.42 [0;2]    1.0 [0;1]   \n",
       "Unicycler                    0.0 [0;0]          0.0 [0;0]   0.75 [0;1]   \n",
       "VelvetOptimiser              0.0 [0;0]          0.0 [0;0]   0.58 [0;1]   \n",
       "\n",
       "                    parsimony    validity  \n",
       "Assembler                                  \n",
       "ABySS              0.81 [0;1]   0.8 [0;0]  \n",
       "BCALM2              0.5 [0;1]   0.5 [0;1]  \n",
       "GATBMiniaPipeline  0.92 [0;1]  0.92 [0;1]  \n",
       "IDBA-UD            1.02 [1;1]  0.99 [0;0]  \n",
       "MEGAHIT            1.01 [1;1]  0.99 [0;0]  \n",
       "MetaHipMer2         0.9 [0;1]   0.9 [0;1]  \n",
       "metaSPAdes         1.01 [0;1]  0.99 [0;1]  \n",
       "MINIA               1.0 [1;1]   1.0 [0;1]  \n",
       "SKESA              0.76 [0;1]  0.75 [0;0]  \n",
       "SPAdes             1.02 [1;1]  0.98 [0;0]  \n",
       "Unicycler          0.75 [0;1]  0.75 [0;1]  \n",
       "VelvetOptimiser    0.58 [0;1]  0.58 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Staphylococcus_aureus\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>17.0 [0;48]</td>\n",
       "      <td>0.21 [0;0]</td>\n",
       "      <td>175648.3 [0;281333]</td>\n",
       "      <td>175648.3 [0;281333]</td>\n",
       "      <td>21.3 [0;71]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2151166.8 [0;2693907]</td>\n",
       "      <td>0.79 [0;0]</td>\n",
       "      <td>30.2 [0;80]</td>\n",
       "      <td>0.8 [0;0]</td>\n",
       "      <td>0.8 [0;0]</td>\n",
       "      <td>0.2 [0;1]</td>\n",
       "      <td>0.4 [0;2]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>28.0 [0;112]</td>\n",
       "      <td>0.01 [0;0]</td>\n",
       "      <td>7702.75 [0;29647]</td>\n",
       "      <td>7291.25 [0;29165]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>682885.25 [0;2600423]</td>\n",
       "      <td>0.25 [0;0]</td>\n",
       "      <td>71.0 [0;175]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>24.0 [0;39]</td>\n",
       "      <td>0.08 [0;0]</td>\n",
       "      <td>66887.5 [0;112739]</td>\n",
       "      <td>63142.5 [0;108199]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2006536.42 [0;2679055]</td>\n",
       "      <td>0.74 [0;0]</td>\n",
       "      <td>39.75 [0;65]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>49.25 [0;112]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>33258.25 [0;53248]</td>\n",
       "      <td>32888.75 [0;53248]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1997988.5 [0;2665812]</td>\n",
       "      <td>0.73 [0;0]</td>\n",
       "      <td>80.0 [0;174]</td>\n",
       "      <td>0.75 [0;0]</td>\n",
       "      <td>0.63 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>12.0 [0;19]</td>\n",
       "      <td>0.11 [0;0]</td>\n",
       "      <td>105373.25 [0;181811]</td>\n",
       "      <td>122849.5 [0;213210]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2010815.5 [0;2681291]</td>\n",
       "      <td>0.74 [0;0]</td>\n",
       "      <td>30.0 [0;43]</td>\n",
       "      <td>0.75 [0;0]</td>\n",
       "      <td>0.64 [0;0]</td>\n",
       "      <td>1.75 [0;3]</td>\n",
       "      <td>4.0 [0;7]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>25.0 [0;42]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>32334.5 [0;53350]</td>\n",
       "      <td>31549.7 [0;53350]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1690855.1 [0;2667635]</td>\n",
       "      <td>0.62 [0;0]</td>\n",
       "      <td>113.9 [0;247]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>12.75 [0;30]</td>\n",
       "      <td>0.16 [0;0]</td>\n",
       "      <td>160355.5 [0;360908]</td>\n",
       "      <td>160355.5 [0;360908]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2009502.75 [0;2682662]</td>\n",
       "      <td>0.74 [0;0]</td>\n",
       "      <td>22.75 [0;49]</td>\n",
       "      <td>0.75 [0;0]</td>\n",
       "      <td>0.74 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>90.92 [0;140]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>20681.75 [0;29647]</td>\n",
       "      <td>20329.0 [0;29165]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1949343.25 [0;2600576]</td>\n",
       "      <td>0.72 [0;0]</td>\n",
       "      <td>139.17 [0;207]</td>\n",
       "      <td>0.75 [0;0]</td>\n",
       "      <td>0.75 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>30.75 [0;43]</td>\n",
       "      <td>0.05 [0;0]</td>\n",
       "      <td>44836.75 [0;73147]</td>\n",
       "      <td>42391.25 [0;63469]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1992379.25 [0;2661968]</td>\n",
       "      <td>0.73 [0;0]</td>\n",
       "      <td>51.25 [0;73]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>13.75 [0;30]</td>\n",
       "      <td>0.13 [0;0]</td>\n",
       "      <td>142248.75 [0;280950]</td>\n",
       "      <td>142248.75 [0;280950]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2005500.5 [0;2678585]</td>\n",
       "      <td>0.74 [0;0]</td>\n",
       "      <td>23.25 [0;48]</td>\n",
       "      <td>0.75 [0;0]</td>\n",
       "      <td>0.73 [0;0]</td>\n",
       "      <td>0.25 [0;1]</td>\n",
       "      <td>0.75 [0;3]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>6.75 [0;14]</td>\n",
       "      <td>0.27 [0;0]</td>\n",
       "      <td>247049.75 [0;394879]</td>\n",
       "      <td>232238.0 [0;394879]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2005193.0 [0;2678241]</td>\n",
       "      <td>0.74 [0;0]</td>\n",
       "      <td>16.25 [0;25]</td>\n",
       "      <td>0.75 [0;0]</td>\n",
       "      <td>0.75 [0;0]</td>\n",
       "      <td>0.25 [0;1]</td>\n",
       "      <td>0.5 [0;2]</td>\n",
       "      <td>0.75 [0;0]</td>\n",
       "      <td>0.75 [0;0]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>54.25 [0;113]</td>\n",
       "      <td>0.01 [0;0]</td>\n",
       "      <td>15553.0 [0;32297]</td>\n",
       "      <td>14385.25 [0;30086]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1301441.75 [0;2603433]</td>\n",
       "      <td>0.48 [0;0]</td>\n",
       "      <td>84.75 [0;175]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm            L90         LSA  \\\n",
       "Assembler                                                                      \n",
       "ABySS              Multiple k-mer De Bruijn graph    17.0 [0;48]  0.21 [0;0]   \n",
       "BCALM2               Single k-mer De Bruijn graph   28.0 [0;112]  0.01 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph    24.0 [0;39]  0.08 [0;0]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  49.25 [0;112]  0.04 [0;0]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph    12.0 [0;19]  0.11 [0;0]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph    25.0 [0;42]  0.04 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph   12.75 [0;30]  0.16 [0;0]   \n",
       "MINIA                Single k-mer De Bruijn graph  90.92 [0;140]  0.02 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph   30.75 [0;43]  0.05 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph   13.75 [0;30]  0.13 [0;0]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph    6.75 [0;14]  0.27 [0;0]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph  54.25 [0;113]  0.01 [0;0]   \n",
       "\n",
       "                                   NA50                  NG50           Ns  \\\n",
       "Assembler                                                                    \n",
       "ABySS               175648.3 [0;281333]   175648.3 [0;281333]  21.3 [0;71]   \n",
       "BCALM2                7702.75 [0;29647]     7291.25 [0;29165]    0.0 [0;0]   \n",
       "GATBMiniaPipeline    66887.5 [0;112739]    63142.5 [0;108199]    0.0 [0;0]   \n",
       "IDBA-UD              33258.25 [0;53248]    32888.75 [0;53248]    0.0 [0;0]   \n",
       "MEGAHIT            105373.25 [0;181811]   122849.5 [0;213210]    0.0 [0;0]   \n",
       "MetaHipMer2           32334.5 [0;53350]     31549.7 [0;53350]    0.0 [0;0]   \n",
       "metaSPAdes          160355.5 [0;360908]   160355.5 [0;360908]    0.0 [0;0]   \n",
       "MINIA                20681.75 [0;29647]     20329.0 [0;29165]    0.0 [0;0]   \n",
       "SKESA                44836.75 [0;73147]    42391.25 [0;63469]    0.0 [0;0]   \n",
       "SPAdes             142248.75 [0;280950]  142248.75 [0;280950]    0.0 [0;0]   \n",
       "Unicycler          247049.75 [0;394879]   232238.0 [0;394879]    0.0 [0;0]   \n",
       "VelvetOptimiser       15553.0 [0;32297]    14385.25 [0;30086]    0.0 [0;0]   \n",
       "\n",
       "                          Type               basepairs breadth_of_coverage  \\\n",
       "Assembler                                                                    \n",
       "ABySS              Metagenomic   2151166.8 [0;2693907]          0.79 [0;0]   \n",
       "BCALM2                 Genomic   682885.25 [0;2600423]          0.25 [0;0]   \n",
       "GATBMiniaPipeline  Metagenomic  2006536.42 [0;2679055]          0.74 [0;0]   \n",
       "IDBA-UD            Metagenomic   1997988.5 [0;2665812]          0.73 [0;0]   \n",
       "MEGAHIT            Metagenomic   2010815.5 [0;2681291]          0.74 [0;0]   \n",
       "MetaHipMer2        Metagenomic   1690855.1 [0;2667635]          0.62 [0;0]   \n",
       "metaSPAdes         Metagenomic  2009502.75 [0;2682662]          0.74 [0;0]   \n",
       "MINIA                  Genomic  1949343.25 [0;2600576]          0.72 [0;0]   \n",
       "SKESA                  Genomic  1992379.25 [0;2661968]          0.73 [0;0]   \n",
       "SPAdes                 Genomic   2005500.5 [0;2678585]          0.74 [0;0]   \n",
       "Unicycler              Genomic   2005193.0 [0;2678241]          0.74 [0;0]   \n",
       "VelvetOptimiser    Metagenomic  1301441.75 [0;2603433]          0.48 [0;0]   \n",
       "\n",
       "                          contigs    identity lowest_identity  \\\n",
       "Assembler                                                       \n",
       "ABySS                 30.2 [0;80]   0.8 [0;0]       0.8 [0;0]   \n",
       "BCALM2               71.0 [0;175]   0.5 [0;1]       0.5 [0;1]   \n",
       "GATBMiniaPipeline    39.75 [0;65]  0.75 [0;1]      0.75 [0;1]   \n",
       "IDBA-UD              80.0 [0;174]  0.75 [0;0]      0.63 [0;0]   \n",
       "MEGAHIT               30.0 [0;43]  0.75 [0;0]      0.64 [0;0]   \n",
       "MetaHipMer2         113.9 [0;247]   0.9 [0;1]       0.9 [0;1]   \n",
       "metaSPAdes           22.75 [0;49]  0.75 [0;0]      0.74 [0;0]   \n",
       "MINIA              139.17 [0;207]  0.75 [0;0]      0.75 [0;0]   \n",
       "SKESA                51.25 [0;73]  0.75 [0;1]      0.75 [0;1]   \n",
       "SPAdes               23.25 [0;48]  0.75 [0;0]      0.73 [0;0]   \n",
       "Unicycler            16.25 [0;25]  0.75 [0;0]      0.75 [0;0]   \n",
       "VelvetOptimiser     84.75 [0;175]   0.5 [0;1]       0.5 [0;1]   \n",
       "\n",
       "                  misassembled contigs misassembly events multiplicity  \\\n",
       "Assembler                                                                \n",
       "ABySS                        0.2 [0;1]          0.4 [0;2]    0.8 [0;1]   \n",
       "BCALM2                       0.0 [0;0]          0.0 [0;0]    0.5 [0;1]   \n",
       "GATBMiniaPipeline            0.0 [0;0]          0.0 [0;0]   0.75 [0;1]   \n",
       "IDBA-UD                      0.0 [0;0]          0.0 [0;0]   0.75 [0;1]   \n",
       "MEGAHIT                     1.75 [0;3]          4.0 [0;7]   0.75 [0;1]   \n",
       "MetaHipMer2                  0.0 [0;0]          0.0 [0;0]    0.9 [0;1]   \n",
       "metaSPAdes                   0.0 [0;0]          0.0 [0;0]   0.75 [0;1]   \n",
       "MINIA                        0.0 [0;0]          0.0 [0;0]   0.75 [0;1]   \n",
       "SKESA                        0.0 [0;0]          0.0 [0;0]   0.75 [0;1]   \n",
       "SPAdes                      0.25 [0;1]         0.75 [0;3]   0.75 [0;1]   \n",
       "Unicycler                   0.25 [0;1]          0.5 [0;2]   0.75 [0;0]   \n",
       "VelvetOptimiser              0.0 [0;0]          0.0 [0;0]    0.5 [0;1]   \n",
       "\n",
       "                    parsimony    validity  \n",
       "Assembler                                  \n",
       "ABySS               0.8 [0;1]   0.8 [0;1]  \n",
       "BCALM2              0.5 [0;1]   0.5 [0;1]  \n",
       "GATBMiniaPipeline  0.75 [0;1]  0.75 [0;1]  \n",
       "IDBA-UD            0.75 [0;1]  0.75 [0;1]  \n",
       "MEGAHIT            0.75 [0;1]  0.75 [0;0]  \n",
       "MetaHipMer2         0.9 [0;1]   0.9 [0;1]  \n",
       "metaSPAdes         0.75 [0;1]  0.75 [0;1]  \n",
       "MINIA              0.75 [0;1]  0.75 [0;0]  \n",
       "SKESA              0.75 [0;1]  0.75 [0;1]  \n",
       "SPAdes             0.75 [0;1]  0.75 [0;1]  \n",
       "Unicycler          0.75 [0;0]  0.75 [0;1]  \n",
       "VelvetOptimiser     0.5 [0;1]   0.5 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Staphylococcus_aureus_plasmid1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>3800.7 [0;6335]</td>\n",
       "      <td>3800.7 [0;6335]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3800.7 [0;6335]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.25 [0;0]</td>\n",
       "      <td>1584.0 [0;6336]</td>\n",
       "      <td>1584.0 [0;6336]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1584.0 [0;6336]</td>\n",
       "      <td>0.25 [0;0]</td>\n",
       "      <td>0.25 [0;1]</td>\n",
       "      <td>0.25 [0;1]</td>\n",
       "      <td>0.25 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.25 [0;1]</td>\n",
       "      <td>0.25 [0;1]</td>\n",
       "      <td>0.25 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.76 [0;1]</td>\n",
       "      <td>4796.58 [0;6522]</td>\n",
       "      <td>4796.58 [0;6522]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>4751.75 [0;6337]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.76 [0;1]</td>\n",
       "      <td>0.76 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.69 [0;0]</td>\n",
       "      <td>4359.25 [0;6336]</td>\n",
       "      <td>4359.25 [0;6336]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>4351.75 [0;6336]</td>\n",
       "      <td>0.69 [0;0]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>4859.75 [0;6795]</td>\n",
       "      <td>4859.75 [0;6795]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>4745.25 [0;6337]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.74 [0;1]</td>\n",
       "      <td>0.74 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.77 [0;1]</td>\n",
       "      <td>0.74 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>3800.4 [0;6334]</td>\n",
       "      <td>3800.4 [0;6334]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3800.4 [0;6334]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.25 [0;1]</td>\n",
       "      <td>0.67 [0;0]</td>\n",
       "      <td>4227.25 [0;6336]</td>\n",
       "      <td>4227.25 [0;6336]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>4719.0 [0;6336]</td>\n",
       "      <td>0.74 [0;0]</td>\n",
       "      <td>1.0 [0;2]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.5 [0;0]</td>\n",
       "      <td>3160.75 [0;6336]</td>\n",
       "      <td>3160.75 [0;6336]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>3160.75 [0;6336]</td>\n",
       "      <td>0.5 [0;0]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.54 [0;0]</td>\n",
       "      <td>3420.75 [0;6044]</td>\n",
       "      <td>3420.75 [0;6044]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>4218.5 [0;6044]</td>\n",
       "      <td>0.67 [0;0]</td>\n",
       "      <td>1.25 [0;3]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.74 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.5 [0;0]</td>\n",
       "      <td>3168.0 [0;6336]</td>\n",
       "      <td>3168.0 [0;6336]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>3168.0 [0;6336]</td>\n",
       "      <td>0.5 [0;0]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.74 [0;1]</td>\n",
       "      <td>4710.25 [0;6367]</td>\n",
       "      <td>4710.25 [0;6367]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>4702.75 [0;6337]</td>\n",
       "      <td>0.74 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.5 [0;0]</td>\n",
       "      <td>3168.0 [0;6336]</td>\n",
       "      <td>3168.0 [0;6336]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3168.0 [0;6336]</td>\n",
       "      <td>0.5 [0;0]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm         L90         LSA  \\\n",
       "Assembler                                                                   \n",
       "ABySS              Multiple k-mer De Bruijn graph   0.0 [0;0]   0.6 [0;0]   \n",
       "BCALM2               Single k-mer De Bruijn graph   0.0 [0;0]  0.25 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph   0.0 [0;0]  0.76 [0;1]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph   0.0 [0;0]  0.69 [0;0]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph   0.0 [0;0]  0.75 [0;1]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph   0.0 [0;0]   0.6 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph  0.25 [0;1]  0.67 [0;0]   \n",
       "MINIA                Single k-mer De Bruijn graph   0.0 [0;0]   0.5 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph   0.0 [0;0]  0.54 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph   0.0 [0;0]   0.5 [0;0]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph   0.0 [0;0]  0.74 [0;1]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph   0.0 [0;0]   0.5 [0;0]   \n",
       "\n",
       "                               NA50              NG50         Ns         Type  \\\n",
       "Assembler                                                                       \n",
       "ABySS               3800.7 [0;6335]   3800.7 [0;6335]  0.0 [0;0]  Metagenomic   \n",
       "BCALM2              1584.0 [0;6336]   1584.0 [0;6336]  0.0 [0;0]      Genomic   \n",
       "GATBMiniaPipeline  4796.58 [0;6522]  4796.58 [0;6522]  0.0 [0;0]  Metagenomic   \n",
       "IDBA-UD            4359.25 [0;6336]  4359.25 [0;6336]  0.0 [0;0]  Metagenomic   \n",
       "MEGAHIT            4859.75 [0;6795]  4859.75 [0;6795]  0.0 [0;0]  Metagenomic   \n",
       "MetaHipMer2         3800.4 [0;6334]   3800.4 [0;6334]  0.0 [0;0]  Metagenomic   \n",
       "metaSPAdes         4227.25 [0;6336]  4227.25 [0;6336]  0.0 [0;0]  Metagenomic   \n",
       "MINIA              3160.75 [0;6336]  3160.75 [0;6336]  0.0 [0;0]      Genomic   \n",
       "SKESA              3420.75 [0;6044]  3420.75 [0;6044]  0.0 [0;0]      Genomic   \n",
       "SPAdes              3168.0 [0;6336]   3168.0 [0;6336]  0.0 [0;0]      Genomic   \n",
       "Unicycler          4710.25 [0;6367]  4710.25 [0;6367]  0.0 [0;0]      Genomic   \n",
       "VelvetOptimiser     3168.0 [0;6336]   3168.0 [0;6336]  0.0 [0;0]  Metagenomic   \n",
       "\n",
       "                          basepairs breadth_of_coverage     contigs  \\\n",
       "Assembler                                                             \n",
       "ABySS               3800.7 [0;6335]           0.6 [0;0]   0.6 [0;1]   \n",
       "BCALM2              1584.0 [0;6336]          0.25 [0;0]  0.25 [0;1]   \n",
       "GATBMiniaPipeline  4751.75 [0;6337]          0.75 [0;1]  0.75 [0;1]   \n",
       "IDBA-UD            4351.75 [0;6336]          0.69 [0;0]  0.75 [0;1]   \n",
       "MEGAHIT            4745.25 [0;6337]          0.75 [0;1]  0.75 [0;1]   \n",
       "MetaHipMer2         3800.4 [0;6334]           0.6 [0;0]   0.6 [0;1]   \n",
       "metaSPAdes          4719.0 [0;6336]          0.74 [0;0]   1.0 [0;2]   \n",
       "MINIA              3160.75 [0;6336]           0.5 [0;0]   0.5 [0;1]   \n",
       "SKESA               4218.5 [0;6044]          0.67 [0;0]  1.25 [0;3]   \n",
       "SPAdes              3168.0 [0;6336]           0.5 [0;0]   0.5 [0;1]   \n",
       "Unicycler          4702.75 [0;6337]          0.74 [0;1]  0.75 [0;1]   \n",
       "VelvetOptimiser     3168.0 [0;6336]           0.5 [0;0]   0.5 [0;1]   \n",
       "\n",
       "                     identity lowest_identity misassembled contigs  \\\n",
       "Assembler                                                            \n",
       "ABySS               0.6 [0;1]       0.6 [0;1]            0.0 [0;0]   \n",
       "BCALM2             0.25 [0;1]      0.25 [0;1]            0.0 [0;0]   \n",
       "GATBMiniaPipeline  0.75 [0;1]      0.75 [0;1]            0.0 [0;0]   \n",
       "IDBA-UD            0.75 [0;1]      0.75 [0;1]            0.0 [0;0]   \n",
       "MEGAHIT            0.74 [0;1]      0.74 [0;1]            0.0 [0;0]   \n",
       "MetaHipMer2         0.6 [0;1]       0.6 [0;1]            0.0 [0;0]   \n",
       "metaSPAdes         0.75 [0;1]      0.75 [0;1]            0.0 [0;0]   \n",
       "MINIA               0.5 [0;1]       0.5 [0;1]            0.0 [0;0]   \n",
       "SKESA              0.75 [0;1]      0.74 [0;1]            0.0 [0;0]   \n",
       "SPAdes              0.5 [0;1]       0.5 [0;1]            0.0 [0;0]   \n",
       "Unicycler          0.75 [0;1]      0.75 [0;1]            0.0 [0;0]   \n",
       "VelvetOptimiser     0.5 [0;1]       0.5 [0;1]            0.0 [0;0]   \n",
       "\n",
       "                  misassembly events multiplicity   parsimony    validity  \n",
       "Assembler                                                                  \n",
       "ABySS                      0.0 [0;0]    0.6 [0;1]   0.6 [0;1]   0.6 [0;1]  \n",
       "BCALM2                     0.0 [0;0]   0.25 [0;1]  0.25 [0;1]  0.25 [0;1]  \n",
       "GATBMiniaPipeline          0.0 [0;0]   0.76 [0;1]  0.76 [0;1]  0.75 [0;1]  \n",
       "IDBA-UD                    0.0 [0;0]   0.75 [0;1]  0.75 [0;1]  0.75 [0;1]  \n",
       "MEGAHIT                    0.0 [0;0]   0.75 [0;1]  0.77 [0;1]  0.74 [0;1]  \n",
       "MetaHipMer2                0.0 [0;0]    0.6 [0;1]   0.6 [0;1]   0.6 [0;1]  \n",
       "metaSPAdes                 0.0 [0;0]   0.75 [0;1]  0.75 [0;1]  0.75 [0;1]  \n",
       "MINIA                      0.0 [0;0]    0.5 [0;1]   0.5 [0;1]   0.5 [0;1]  \n",
       "SKESA                      0.0 [0;0]   0.75 [0;1]  0.75 [0;1]  0.75 [0;1]  \n",
       "SPAdes                     0.0 [0;0]    0.5 [0;1]   0.5 [0;1]   0.5 [0;1]  \n",
       "Unicycler                  0.0 [0;0]   0.75 [0;1]  0.75 [0;1]  0.75 [0;1]  \n",
       "VelvetOptimiser            0.0 [0;0]    0.5 [0;1]   0.5 [0;1]   0.5 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Staphylococcus_aureus_plasmid2\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.72 [0;0]</td>\n",
       "      <td>1604.0 [0;2215]</td>\n",
       "      <td>1604.0 [0;2215]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1604.0 [0;2215]</td>\n",
       "      <td>0.72 [0;0]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.76 [0;1]</td>\n",
       "      <td>1694.0 [0;2397]</td>\n",
       "      <td>1694.0 [0;2397]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1651.75 [0;2216]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.77 [0;1]</td>\n",
       "      <td>0.77 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.51 [0;1]</td>\n",
       "      <td>1132.5 [0;2315]</td>\n",
       "      <td>1132.5 [0;2315]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1107.75 [0;2216]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.51 [0;1]</td>\n",
       "      <td>0.51 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.77 [0;1]</td>\n",
       "      <td>1696.75 [0;2357]</td>\n",
       "      <td>1696.75 [0;2357]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1661.5 [0;2216]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.77 [0;1]</td>\n",
       "      <td>0.77 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>1327.8 [0;2213]</td>\n",
       "      <td>1327.8 [0;2213]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1327.8 [0;2213]</td>\n",
       "      <td>0.6 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.72 [0;1]</td>\n",
       "      <td>1605.75 [0;2271]</td>\n",
       "      <td>1605.75 [0;2271]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1592.0 [0;2216]</td>\n",
       "      <td>0.72 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;0]</td>\n",
       "      <td>0.75 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.76 [0;1]</td>\n",
       "      <td>0.76 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.61 [0;0]</td>\n",
       "      <td>1343.0 [0;2005]</td>\n",
       "      <td>1343.0 [0;2005]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1343.0 [0;2005]</td>\n",
       "      <td>0.61 [0;0]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.65 [0;1]</td>\n",
       "      <td>1696.5 [0;3246]</td>\n",
       "      <td>1696.5 [0;3246]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1439.0 [0;2216]</td>\n",
       "      <td>0.65 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "      <td>0.67 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>1612.0 [0;2216]</td>\n",
       "      <td>1612.0 [0;2216]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1612.0 [0;2216]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm        L90         LSA  \\\n",
       "Assembler                                                                  \n",
       "ABySS              Multiple k-mer De Bruijn graph  0.0 [0;0]  0.72 [0;0]   \n",
       "BCALM2               Single k-mer De Bruijn graph  0.0 [0;0]   0.0 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph  0.0 [0;0]  0.76 [0;1]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  0.0 [0;0]  0.51 [0;1]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph  0.0 [0;0]  0.77 [0;1]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph  0.0 [0;0]   0.6 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph  0.0 [0;0]  0.72 [0;1]   \n",
       "MINIA                Single k-mer De Bruijn graph  0.0 [0;0]   0.0 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph  0.0 [0;0]  0.61 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph  0.0 [0;0]  0.65 [0;1]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph  0.0 [0;0]  0.73 [0;1]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph  0.0 [0;0]   0.0 [0;0]   \n",
       "\n",
       "                               NA50              NG50         Ns         Type  \\\n",
       "Assembler                                                                       \n",
       "ABySS               1604.0 [0;2215]   1604.0 [0;2215]  0.0 [0;0]  Metagenomic   \n",
       "BCALM2                    0.0 [0;0]         0.0 [0;0]  0.0 [0;0]      Genomic   \n",
       "GATBMiniaPipeline   1694.0 [0;2397]   1694.0 [0;2397]  0.0 [0;0]  Metagenomic   \n",
       "IDBA-UD             1132.5 [0;2315]   1132.5 [0;2315]  0.0 [0;0]  Metagenomic   \n",
       "MEGAHIT            1696.75 [0;2357]  1696.75 [0;2357]  0.0 [0;0]  Metagenomic   \n",
       "MetaHipMer2         1327.8 [0;2213]   1327.8 [0;2213]  0.0 [0;0]  Metagenomic   \n",
       "metaSPAdes         1605.75 [0;2271]  1605.75 [0;2271]  0.0 [0;0]  Metagenomic   \n",
       "MINIA                     0.0 [0;0]         0.0 [0;0]  0.0 [0;0]      Genomic   \n",
       "SKESA               1343.0 [0;2005]   1343.0 [0;2005]  0.0 [0;0]      Genomic   \n",
       "SPAdes              1696.5 [0;3246]   1696.5 [0;3246]  0.0 [0;0]      Genomic   \n",
       "Unicycler           1612.0 [0;2216]   1612.0 [0;2216]  0.0 [0;0]      Genomic   \n",
       "VelvetOptimiser           0.0 [0;0]         0.0 [0;0]  0.0 [0;0]  Metagenomic   \n",
       "\n",
       "                          basepairs breadth_of_coverage     contigs  \\\n",
       "Assembler                                                             \n",
       "ABySS               1604.0 [0;2215]          0.72 [0;0]   0.8 [0;1]   \n",
       "BCALM2                    0.0 [0;0]           0.0 [0;0]   0.0 [0;0]   \n",
       "GATBMiniaPipeline  1651.75 [0;2216]          0.75 [0;1]  0.75 [0;1]   \n",
       "IDBA-UD            1107.75 [0;2216]           0.5 [0;1]   0.5 [0;1]   \n",
       "MEGAHIT             1661.5 [0;2216]          0.75 [0;1]  0.75 [0;1]   \n",
       "MetaHipMer2         1327.8 [0;2213]           0.6 [0;0]   0.6 [0;1]   \n",
       "metaSPAdes          1592.0 [0;2216]          0.72 [0;1]  0.75 [0;1]   \n",
       "MINIA                     0.0 [0;0]           0.0 [0;0]   0.0 [0;0]   \n",
       "SKESA               1343.0 [0;2005]          0.61 [0;0]  0.75 [0;1]   \n",
       "SPAdes              1439.0 [0;2216]          0.65 [0;1]  0.75 [0;1]   \n",
       "Unicycler           1612.0 [0;2216]          0.73 [0;1]  0.75 [0;1]   \n",
       "VelvetOptimiser           0.0 [0;0]           0.0 [0;0]   0.0 [0;0]   \n",
       "\n",
       "                     identity lowest_identity misassembled contigs  \\\n",
       "Assembler                                                            \n",
       "ABySS               0.8 [0;1]       0.8 [0;1]            0.0 [0;0]   \n",
       "BCALM2              0.0 [0;0]       0.0 [0;0]            0.0 [0;0]   \n",
       "GATBMiniaPipeline  0.75 [0;1]      0.75 [0;1]            0.0 [0;0]   \n",
       "IDBA-UD             0.5 [0;1]       0.5 [0;1]            0.0 [0;0]   \n",
       "MEGAHIT            0.75 [0;1]      0.75 [0;1]            0.0 [0;0]   \n",
       "MetaHipMer2         0.6 [0;1]       0.6 [0;1]            0.0 [0;0]   \n",
       "metaSPAdes         0.75 [0;0]      0.75 [0;0]            0.0 [0;0]   \n",
       "MINIA               0.0 [0;0]       0.0 [0;0]            0.0 [0;0]   \n",
       "SKESA              0.75 [0;1]      0.75 [0;1]            0.0 [0;0]   \n",
       "SPAdes             0.67 [0;1]      0.67 [0;1]            0.0 [0;0]   \n",
       "Unicycler          0.75 [0;1]      0.75 [0;1]            0.0 [0;0]   \n",
       "VelvetOptimiser     0.0 [0;0]       0.0 [0;0]            0.0 [0;0]   \n",
       "\n",
       "                  misassembly events multiplicity   parsimony    validity  \n",
       "Assembler                                                                  \n",
       "ABySS                      0.0 [0;0]    0.8 [0;1]   0.8 [0;1]   0.8 [0;1]  \n",
       "BCALM2                     0.0 [0;0]    0.0 [0;0]   0.0 [0;0]   0.0 [0;0]  \n",
       "GATBMiniaPipeline          0.0 [0;0]   0.77 [0;1]  0.77 [0;1]  0.75 [0;1]  \n",
       "IDBA-UD                    0.0 [0;0]   0.51 [0;1]  0.51 [0;1]   0.5 [0;1]  \n",
       "MEGAHIT                    0.0 [0;0]   0.77 [0;1]  0.77 [0;1]  0.75 [0;1]  \n",
       "MetaHipMer2                0.0 [0;0]    0.6 [0;1]   0.6 [0;1]   0.6 [0;1]  \n",
       "metaSPAdes                 0.0 [0;0]   0.76 [0;1]  0.76 [0;1]  0.75 [0;1]  \n",
       "MINIA                      0.0 [0;0]    0.0 [0;0]   0.0 [0;0]   0.0 [0;0]  \n",
       "SKESA                      0.0 [0;0]   0.75 [0;1]  0.75 [0;1]  0.75 [0;1]  \n",
       "SPAdes                     0.0 [0;0]   0.75 [0;1]  0.87 [0;1]  0.67 [0;1]  \n",
       "Unicycler                  0.0 [0;0]   0.75 [0;1]  0.75 [0;1]  0.75 [0;1]  \n",
       "VelvetOptimiser            0.0 [0;0]    0.0 [0;0]   0.0 [0;0]   0.0 [0;0]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Staphylococcus_aureus_plasmid3\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.73 [0;0]</td>\n",
       "      <td>2185.9 [0;2992]</td>\n",
       "      <td>2185.9 [0;2992]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2185.9 [0;2992]</td>\n",
       "      <td>0.73 [0;0]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "      <td>0.8 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>2232.25 [0;3162]</td>\n",
       "      <td>2232.25 [0;3162]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2193.5 [0;2993]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.76 [0;1]</td>\n",
       "      <td>0.76 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.25 [0;1]</td>\n",
       "      <td>0.62 [0;0]</td>\n",
       "      <td>1845.5 [0;2847]</td>\n",
       "      <td>1845.5 [0;2847]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2112.75 [0;2847]</td>\n",
       "      <td>0.71 [0;0]</td>\n",
       "      <td>1.0 [0;2]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.76 [0;1]</td>\n",
       "      <td>0.76 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.76 [0;1]</td>\n",
       "      <td>2279.5 [0;3134]</td>\n",
       "      <td>2279.5 [0;3134]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2244.25 [0;2993]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.76 [0;1]</td>\n",
       "      <td>0.76 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.76 [0;0]</td>\n",
       "      <td>2271.9 [0;2990]</td>\n",
       "      <td>2271.9 [0;2990]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2271.9 [0;2990]</td>\n",
       "      <td>0.76 [0;0]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>1802.75 [0;3857]</td>\n",
       "      <td>1802.75 [0;3857]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2208.0 [0;2993]</td>\n",
       "      <td>0.74 [0;1]</td>\n",
       "      <td>1.25 [0;2]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.64 [0;0]</td>\n",
       "      <td>1902.5 [0;2782]</td>\n",
       "      <td>1902.5 [0;2782]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1902.5 [0;2782]</td>\n",
       "      <td>0.64 [0;0]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.39 [0;0]</td>\n",
       "      <td>1167.25 [0;2992]</td>\n",
       "      <td>1167.25 [0;2992]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1167.25 [0;2992]</td>\n",
       "      <td>0.39 [0;0]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>2194.75 [0;2993]</td>\n",
       "      <td>2194.75 [0;2993]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2194.75 [0;2993]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "      <td>0.75 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm         L90         LSA  \\\n",
       "Assembler                                                                   \n",
       "ABySS              Multiple k-mer De Bruijn graph   0.0 [0;0]  0.73 [0;0]   \n",
       "BCALM2               Single k-mer De Bruijn graph   0.0 [0;0]   0.0 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph   0.0 [0;0]  0.75 [0;1]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  0.25 [0;1]  0.62 [0;0]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph   0.0 [0;0]  0.76 [0;1]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph   0.0 [0;0]  0.76 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph   0.5 [0;1]   0.6 [0;1]   \n",
       "MINIA                Single k-mer De Bruijn graph   0.0 [0;0]   0.0 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph   0.0 [0;0]  0.64 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph   0.0 [0;0]  0.39 [0;0]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph   0.0 [0;0]  0.73 [0;1]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph   0.0 [0;0]   0.0 [0;0]   \n",
       "\n",
       "                               NA50              NG50         Ns         Type  \\\n",
       "Assembler                                                                       \n",
       "ABySS               2185.9 [0;2992]   2185.9 [0;2992]  0.0 [0;0]  Metagenomic   \n",
       "BCALM2                    0.0 [0;0]         0.0 [0;0]  0.0 [0;0]      Genomic   \n",
       "GATBMiniaPipeline  2232.25 [0;3162]  2232.25 [0;3162]  0.0 [0;0]  Metagenomic   \n",
       "IDBA-UD             1845.5 [0;2847]   1845.5 [0;2847]  0.0 [0;0]  Metagenomic   \n",
       "MEGAHIT             2279.5 [0;3134]   2279.5 [0;3134]  0.0 [0;0]  Metagenomic   \n",
       "MetaHipMer2         2271.9 [0;2990]   2271.9 [0;2990]  0.0 [0;0]  Metagenomic   \n",
       "metaSPAdes         1802.75 [0;3857]  1802.75 [0;3857]  0.0 [0;0]  Metagenomic   \n",
       "MINIA                     0.0 [0;0]         0.0 [0;0]  0.0 [0;0]      Genomic   \n",
       "SKESA               1902.5 [0;2782]   1902.5 [0;2782]  0.0 [0;0]      Genomic   \n",
       "SPAdes             1167.25 [0;2992]  1167.25 [0;2992]  0.0 [0;0]      Genomic   \n",
       "Unicycler          2194.75 [0;2993]  2194.75 [0;2993]  0.0 [0;0]      Genomic   \n",
       "VelvetOptimiser           0.0 [0;0]         0.0 [0;0]  0.0 [0;0]  Metagenomic   \n",
       "\n",
       "                          basepairs breadth_of_coverage     contigs  \\\n",
       "Assembler                                                             \n",
       "ABySS               2185.9 [0;2992]          0.73 [0;0]   0.8 [0;1]   \n",
       "BCALM2                    0.0 [0;0]           0.0 [0;0]   0.0 [0;0]   \n",
       "GATBMiniaPipeline   2193.5 [0;2993]          0.73 [0;1]  0.75 [0;1]   \n",
       "IDBA-UD            2112.75 [0;2847]          0.71 [0;0]   1.0 [0;2]   \n",
       "MEGAHIT            2244.25 [0;2993]          0.75 [0;1]  0.75 [0;1]   \n",
       "MetaHipMer2         2271.9 [0;2990]          0.76 [0;0]   0.9 [0;1]   \n",
       "metaSPAdes          2208.0 [0;2993]          0.74 [0;1]  1.25 [0;2]   \n",
       "MINIA                     0.0 [0;0]           0.0 [0;0]   0.0 [0;0]   \n",
       "SKESA               1902.5 [0;2782]          0.64 [0;0]  0.75 [0;1]   \n",
       "SPAdes             1167.25 [0;2992]          0.39 [0;0]   0.5 [0;1]   \n",
       "Unicycler          2194.75 [0;2993]          0.73 [0;1]  0.75 [0;1]   \n",
       "VelvetOptimiser           0.0 [0;0]           0.0 [0;0]   0.0 [0;0]   \n",
       "\n",
       "                     identity lowest_identity misassembled contigs  \\\n",
       "Assembler                                                            \n",
       "ABySS               0.8 [0;1]       0.8 [0;1]            0.0 [0;0]   \n",
       "BCALM2              0.0 [0;0]       0.0 [0;0]            0.0 [0;0]   \n",
       "GATBMiniaPipeline  0.75 [0;1]      0.75 [0;1]            0.0 [0;0]   \n",
       "IDBA-UD            0.75 [0;1]      0.75 [0;1]            0.0 [0;0]   \n",
       "MEGAHIT            0.75 [0;1]      0.75 [0;1]            0.0 [0;0]   \n",
       "MetaHipMer2         0.9 [0;1]       0.9 [0;1]            0.0 [0;0]   \n",
       "metaSPAdes         0.75 [0;1]      0.75 [0;1]            0.0 [0;0]   \n",
       "MINIA               0.0 [0;0]       0.0 [0;0]            0.0 [0;0]   \n",
       "SKESA              0.75 [0;1]      0.75 [0;1]            0.0 [0;0]   \n",
       "SPAdes              0.5 [0;1]       0.5 [0;1]            0.0 [0;0]   \n",
       "Unicycler          0.75 [0;1]      0.75 [0;1]            0.0 [0;0]   \n",
       "VelvetOptimiser     0.0 [0;0]       0.0 [0;0]            0.0 [0;0]   \n",
       "\n",
       "                  misassembly events multiplicity   parsimony    validity  \n",
       "Assembler                                                                  \n",
       "ABySS                      0.0 [0;0]    0.8 [0;1]   0.8 [0;1]   0.8 [0;1]  \n",
       "BCALM2                     0.0 [0;0]    0.0 [0;0]   0.0 [0;0]   0.0 [0;0]  \n",
       "GATBMiniaPipeline          0.0 [0;0]   0.76 [0;1]  0.76 [0;1]  0.75 [0;1]  \n",
       "IDBA-UD                    0.0 [0;0]   0.76 [0;1]  0.76 [0;1]  0.75 [0;1]  \n",
       "MEGAHIT                    0.0 [0;0]   0.76 [0;1]  0.76 [0;1]  0.75 [0;1]  \n",
       "MetaHipMer2                0.0 [0;0]    0.9 [0;1]   0.9 [0;1]   0.9 [0;1]  \n",
       "metaSPAdes                 0.0 [0;0]   0.83 [0;1]  0.83 [0;1]  0.75 [0;1]  \n",
       "MINIA                      0.0 [0;0]    0.0 [0;0]   0.0 [0;0]   0.0 [0;0]  \n",
       "SKESA                      0.0 [0;0]   0.75 [0;1]  0.75 [0;1]  0.75 [0;1]  \n",
       "SPAdes                     0.0 [0;0]    0.5 [0;1]   0.5 [0;1]   0.5 [0;1]  \n",
       "Unicycler                  0.0 [0;0]   0.75 [0;1]  0.75 [0;1]  0.75 [0;1]  \n",
       "VelvetOptimiser            0.0 [0;0]    0.0 [0;0]   0.0 [0;0]   0.0 [0;0]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "for reference in reference_pipeline_metrics_df.reference.unique():\n",
    "    ref_stats_per_reference_per_assembler = pd.DataFrame()\n",
    "    for assembler in reference_pipeline_metrics_df.assembler.unique():\n",
    "        row={'Assembler': assembler, \n",
    "             'Type':reference_pipeline_metrics_df[reference_pipeline_metrics_df.assembler == assembler].type.unique()[0],\n",
    "             'Algorithm': reference_pipeline_metrics_df[reference_pipeline_metrics_df.assembler == assembler].algorythm.unique()[0]}\n",
    "        lala=reference_pipeline_metrics_df[(reference_pipeline_metrics_df.assembler == assembler) & (reference_pipeline_metrics_df.reference == reference)].describe()\n",
    "        for column in lala.columns:\n",
    "            mean = lala.loc['mean',column]\n",
    "            minimum = lala.loc['min',column]\n",
    "            maximum = lala.loc['max',column]\n",
    "            row[column] = \"{} [{};{}]\".format(round(mean, 2),int(minimum), int(maximum))\n",
    "        ref_stats_per_reference_per_assembler=ref_stats_per_reference_per_assembler.append(row, ignore_index=True)\n",
    "    ref_stats_per_reference_per_assembler = ref_stats_per_reference_per_assembler.set_index('Assembler')\n",
    "    ref_stats_per_reference_per_assembler.to_csv(\"Tables/Results/{} - Reference metrics per assembler.csv\".format(reference))\n",
    "    print(reference)\n",
    "    display(ref_stats_per_reference_per_assembler)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Per reference replicon for all assemblers - Even only!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Reference</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Bacillus_subtilis</th>\n",
       "      <td>25.64 [0;120]</td>\n",
       "      <td>0.1 [0;0]</td>\n",
       "      <td>196509.03 [0;343843]</td>\n",
       "      <td>194589.61 [0;343843]</td>\n",
       "      <td>1.32 [0;47]</td>\n",
       "      <td>3614434.53 [0;3991152]</td>\n",
       "      <td>0.89 [0;0]</td>\n",
       "      <td>91.85 [0;1467]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "      <td>0.08 [0;2]</td>\n",
       "      <td>0.2 [0;4]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Enterococcus_faecalis</th>\n",
       "      <td>36.26 [0;149]</td>\n",
       "      <td>0.13 [0;0]</td>\n",
       "      <td>134153.57 [0;377104]</td>\n",
       "      <td>132615.19 [0;377104]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>2491484.12 [0;2828460]</td>\n",
       "      <td>0.88 [0;0]</td>\n",
       "      <td>72.49 [0;231]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.91 [0;1]</td>\n",
       "      <td>0.34 [0;3]</td>\n",
       "      <td>0.62 [0;6]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Escherichia_coli_plasmid</th>\n",
       "      <td>3.27 [0;18]</td>\n",
       "      <td>0.22 [0;0]</td>\n",
       "      <td>18003.78 [0;55639]</td>\n",
       "      <td>17461.25 [0;55639]</td>\n",
       "      <td>37.6 [0;835]</td>\n",
       "      <td>82423.2 [0;109516]</td>\n",
       "      <td>0.75 [0;0]</td>\n",
       "      <td>9.2 [0;33]</td>\n",
       "      <td>0.89 [0;1]</td>\n",
       "      <td>0.88 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.89 [0;1]</td>\n",
       "      <td>0.89 [0;1]</td>\n",
       "      <td>0.89 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Escherichia_coli</th>\n",
       "      <td>70.08 [0;283]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>51151.78 [0;138793]</td>\n",
       "      <td>50851.09 [0;138793]</td>\n",
       "      <td>1.26 [0;45]</td>\n",
       "      <td>4051935.78 [0;4706885]</td>\n",
       "      <td>0.85 [0;0]</td>\n",
       "      <td>363.8 [0;1495]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.84 [0;1]</td>\n",
       "      <td>1.07 [0;8]</td>\n",
       "      <td>1.99 [0;15]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Lactobacillus_fermentum</th>\n",
       "      <td>77.7 [0;263]</td>\n",
       "      <td>0.05 [0;0]</td>\n",
       "      <td>30397.63 [0;46932]</td>\n",
       "      <td>29158.61 [0;52972]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1585069.36 [0;1873053]</td>\n",
       "      <td>0.83 [0;0]</td>\n",
       "      <td>105.8 [0;282]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "      <td>0.34 [0;4]</td>\n",
       "      <td>0.69 [0;8]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Listeria_monocytogenes</th>\n",
       "      <td>24.36 [0;122]</td>\n",
       "      <td>0.15 [0;0]</td>\n",
       "      <td>236971.75 [0;542872]</td>\n",
       "      <td>228275.5 [0;542872]</td>\n",
       "      <td>0.03 [0;1]</td>\n",
       "      <td>2625730.37 [0;2961845]</td>\n",
       "      <td>0.88 [0;0]</td>\n",
       "      <td>62.41 [0;361]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.93 [0;1]</td>\n",
       "      <td>0.03 [0;1]</td>\n",
       "      <td>0.06 [0;2]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pseudomonas_aeruginosa</th>\n",
       "      <td>147.13 [0;1004]</td>\n",
       "      <td>0.06 [0;0]</td>\n",
       "      <td>141661.78 [0;338923]</td>\n",
       "      <td>146649.92 [0;464977]</td>\n",
       "      <td>583.01 [0;12098]</td>\n",
       "      <td>6111755.96 [0;6780808]</td>\n",
       "      <td>0.9 [0;0]</td>\n",
       "      <td>240.31 [0;1404]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.85 [0;1]</td>\n",
       "      <td>1.27 [0;8]</td>\n",
       "      <td>2.26 [0;15]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Salmonella_enterica</th>\n",
       "      <td>54.45 [0;200]</td>\n",
       "      <td>0.05 [0;0]</td>\n",
       "      <td>71283.69 [0;223960]</td>\n",
       "      <td>70301.45 [0;223960]</td>\n",
       "      <td>0.12 [0;2]</td>\n",
       "      <td>4074863.96 [0;4703394]</td>\n",
       "      <td>0.86 [0;0]</td>\n",
       "      <td>332.36 [0;1479]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.83 [0;1]</td>\n",
       "      <td>0.79 [0;9]</td>\n",
       "      <td>1.25 [0;14]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Staphylococcus_aureus</th>\n",
       "      <td>40.08 [0;140]</td>\n",
       "      <td>0.12 [0;0]</td>\n",
       "      <td>114085.93 [0;394879]</td>\n",
       "      <td>113359.04 [0;394879]</td>\n",
       "      <td>1.99 [0;71]</td>\n",
       "      <td>2373450.96 [0;2693907]</td>\n",
       "      <td>0.87 [0;0]</td>\n",
       "      <td>76.07 [0;247]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.91 [0;1]</td>\n",
       "      <td>0.27 [0;3]</td>\n",
       "      <td>0.63 [0;7]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Staphylococcus_aureus_plasmid1</th>\n",
       "      <td>0.03 [0;1]</td>\n",
       "      <td>0.77 [0;1]</td>\n",
       "      <td>4910.9 [0;6795]</td>\n",
       "      <td>4910.9 [0;6795]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>5035.96 [0;6337]</td>\n",
       "      <td>0.79 [0;1]</td>\n",
       "      <td>0.9 [0;3]</td>\n",
       "      <td>0.81 [0;1]</td>\n",
       "      <td>0.81 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "      <td>0.82 [0;1]</td>\n",
       "      <td>0.81 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Staphylococcus_aureus_plasmid2</th>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.66 [0;1]</td>\n",
       "      <td>1483.03 [0;3246]</td>\n",
       "      <td>1483.03 [0;3246]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1441.14 [0;2216]</td>\n",
       "      <td>0.65 [0;1]</td>\n",
       "      <td>0.69 [0;1]</td>\n",
       "      <td>0.68 [0;1]</td>\n",
       "      <td>0.68 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.7 [0;1]</td>\n",
       "      <td>0.71 [0;1]</td>\n",
       "      <td>0.68 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Staphylococcus_aureus_plasmid3</th>\n",
       "      <td>0.08 [0;1]</td>\n",
       "      <td>0.64 [0;1]</td>\n",
       "      <td>1922.17 [0;3857]</td>\n",
       "      <td>1922.17 [0;3857]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>1989.29 [0;2993]</td>\n",
       "      <td>0.66 [0;1]</td>\n",
       "      <td>0.8 [0;2]</td>\n",
       "      <td>0.72 [0;1]</td>\n",
       "      <td>0.72 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.73 [0;1]</td>\n",
       "      <td>0.72 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            L90         LSA  \\\n",
       "Reference                                                     \n",
       "Bacillus_subtilis                 25.64 [0;120]   0.1 [0;0]   \n",
       "Enterococcus_faecalis             36.26 [0;149]  0.13 [0;0]   \n",
       "Escherichia_coli_plasmid            3.27 [0;18]  0.22 [0;0]   \n",
       "Escherichia_coli                  70.08 [0;283]  0.04 [0;0]   \n",
       "Lactobacillus_fermentum            77.7 [0;263]  0.05 [0;0]   \n",
       "Listeria_monocytogenes            24.36 [0;122]  0.15 [0;0]   \n",
       "Pseudomonas_aeruginosa          147.13 [0;1004]  0.06 [0;0]   \n",
       "Salmonella_enterica               54.45 [0;200]  0.05 [0;0]   \n",
       "Staphylococcus_aureus             40.08 [0;140]  0.12 [0;0]   \n",
       "Staphylococcus_aureus_plasmid1       0.03 [0;1]  0.77 [0;1]   \n",
       "Staphylococcus_aureus_plasmid2        0.0 [0;0]  0.66 [0;1]   \n",
       "Staphylococcus_aureus_plasmid3       0.08 [0;1]  0.64 [0;1]   \n",
       "\n",
       "                                                NA50                  NG50  \\\n",
       "Reference                                                                    \n",
       "Bacillus_subtilis               196509.03 [0;343843]  194589.61 [0;343843]   \n",
       "Enterococcus_faecalis           134153.57 [0;377104]  132615.19 [0;377104]   \n",
       "Escherichia_coli_plasmid          18003.78 [0;55639]    17461.25 [0;55639]   \n",
       "Escherichia_coli                 51151.78 [0;138793]   50851.09 [0;138793]   \n",
       "Lactobacillus_fermentum           30397.63 [0;46932]    29158.61 [0;52972]   \n",
       "Listeria_monocytogenes          236971.75 [0;542872]   228275.5 [0;542872]   \n",
       "Pseudomonas_aeruginosa          141661.78 [0;338923]  146649.92 [0;464977]   \n",
       "Salmonella_enterica              71283.69 [0;223960]   70301.45 [0;223960]   \n",
       "Staphylococcus_aureus           114085.93 [0;394879]  113359.04 [0;394879]   \n",
       "Staphylococcus_aureus_plasmid1       4910.9 [0;6795]       4910.9 [0;6795]   \n",
       "Staphylococcus_aureus_plasmid2      1483.03 [0;3246]      1483.03 [0;3246]   \n",
       "Staphylococcus_aureus_plasmid3      1922.17 [0;3857]      1922.17 [0;3857]   \n",
       "\n",
       "                                              Ns               basepairs  \\\n",
       "Reference                                                                  \n",
       "Bacillus_subtilis                    1.32 [0;47]  3614434.53 [0;3991152]   \n",
       "Enterococcus_faecalis                  0.0 [0;0]  2491484.12 [0;2828460]   \n",
       "Escherichia_coli_plasmid            37.6 [0;835]      82423.2 [0;109516]   \n",
       "Escherichia_coli                     1.26 [0;45]  4051935.78 [0;4706885]   \n",
       "Lactobacillus_fermentum                0.0 [0;0]  1585069.36 [0;1873053]   \n",
       "Listeria_monocytogenes                0.03 [0;1]  2625730.37 [0;2961845]   \n",
       "Pseudomonas_aeruginosa          583.01 [0;12098]  6111755.96 [0;6780808]   \n",
       "Salmonella_enterica                   0.12 [0;2]  4074863.96 [0;4703394]   \n",
       "Staphylococcus_aureus                1.99 [0;71]  2373450.96 [0;2693907]   \n",
       "Staphylococcus_aureus_plasmid1         0.0 [0;0]        5035.96 [0;6337]   \n",
       "Staphylococcus_aureus_plasmid2         0.0 [0;0]        1441.14 [0;2216]   \n",
       "Staphylococcus_aureus_plasmid3         0.0 [0;0]        1989.29 [0;2993]   \n",
       "\n",
       "                               breadth_of_coverage          contigs  \\\n",
       "Reference                                                             \n",
       "Bacillus_subtilis                       0.89 [0;0]   91.85 [0;1467]   \n",
       "Enterococcus_faecalis                   0.88 [0;0]    72.49 [0;231]   \n",
       "Escherichia_coli_plasmid                0.75 [0;0]       9.2 [0;33]   \n",
       "Escherichia_coli                        0.85 [0;0]   363.8 [0;1495]   \n",
       "Lactobacillus_fermentum                 0.83 [0;0]    105.8 [0;282]   \n",
       "Listeria_monocytogenes                  0.88 [0;0]    62.41 [0;361]   \n",
       "Pseudomonas_aeruginosa                   0.9 [0;0]  240.31 [0;1404]   \n",
       "Salmonella_enterica                     0.86 [0;0]  332.36 [0;1479]   \n",
       "Staphylococcus_aureus                   0.87 [0;0]    76.07 [0;247]   \n",
       "Staphylococcus_aureus_plasmid1          0.79 [0;1]        0.9 [0;3]   \n",
       "Staphylococcus_aureus_plasmid2          0.65 [0;1]       0.69 [0;1]   \n",
       "Staphylococcus_aureus_plasmid3          0.66 [0;1]        0.8 [0;2]   \n",
       "\n",
       "                                  identity lowest_identity  \\\n",
       "Reference                                                    \n",
       "Bacillus_subtilis               0.94 [0;1]      0.92 [0;1]   \n",
       "Enterococcus_faecalis           0.94 [0;1]      0.91 [0;1]   \n",
       "Escherichia_coli_plasmid        0.89 [0;1]      0.88 [0;1]   \n",
       "Escherichia_coli                0.94 [0;1]      0.84 [0;1]   \n",
       "Lactobacillus_fermentum         0.92 [0;1]       0.9 [0;1]   \n",
       "Listeria_monocytogenes          0.94 [0;1]      0.93 [0;1]   \n",
       "Pseudomonas_aeruginosa          0.94 [0;1]      0.85 [0;1]   \n",
       "Salmonella_enterica             0.94 [0;1]      0.83 [0;1]   \n",
       "Staphylococcus_aureus           0.94 [0;1]      0.91 [0;1]   \n",
       "Staphylococcus_aureus_plasmid1  0.81 [0;1]      0.81 [0;1]   \n",
       "Staphylococcus_aureus_plasmid2  0.68 [0;1]      0.68 [0;1]   \n",
       "Staphylococcus_aureus_plasmid3  0.72 [0;1]      0.72 [0;1]   \n",
       "\n",
       "                               misassembled contigs misassembly events  \\\n",
       "Reference                                                                \n",
       "Bacillus_subtilis                        0.08 [0;2]          0.2 [0;4]   \n",
       "Enterococcus_faecalis                    0.34 [0;3]         0.62 [0;6]   \n",
       "Escherichia_coli_plasmid                  0.0 [0;0]          0.0 [0;0]   \n",
       "Escherichia_coli                         1.07 [0;8]        1.99 [0;15]   \n",
       "Lactobacillus_fermentum                  0.34 [0;4]         0.69 [0;8]   \n",
       "Listeria_monocytogenes                   0.03 [0;1]         0.06 [0;2]   \n",
       "Pseudomonas_aeruginosa                   1.27 [0;8]        2.26 [0;15]   \n",
       "Salmonella_enterica                      0.79 [0;9]        1.25 [0;14]   \n",
       "Staphylococcus_aureus                    0.27 [0;3]         0.63 [0;7]   \n",
       "Staphylococcus_aureus_plasmid1            0.0 [0;0]          0.0 [0;0]   \n",
       "Staphylococcus_aureus_plasmid2            0.0 [0;0]          0.0 [0;0]   \n",
       "Staphylococcus_aureus_plasmid3            0.0 [0;0]          0.0 [0;0]   \n",
       "\n",
       "                               multiplicity   parsimony    validity  \n",
       "Reference                                                            \n",
       "Bacillus_subtilis                0.94 [0;1]  0.94 [0;1]  0.94 [0;1]  \n",
       "Enterococcus_faecalis            0.95 [0;1]  0.95 [0;1]  0.94 [0;1]  \n",
       "Escherichia_coli_plasmid         0.89 [0;1]  0.89 [0;1]  0.89 [0;1]  \n",
       "Escherichia_coli                 0.95 [0;1]  0.95 [0;1]  0.94 [0;1]  \n",
       "Lactobacillus_fermentum          0.92 [0;1]  0.92 [0;1]  0.92 [0;1]  \n",
       "Listeria_monocytogenes           0.95 [0;1]  0.95 [0;1]  0.94 [0;1]  \n",
       "Pseudomonas_aeruginosa           0.95 [0;1]  0.95 [0;1]  0.94 [0;1]  \n",
       "Salmonella_enterica              0.95 [0;1]  0.95 [0;1]  0.94 [0;1]  \n",
       "Staphylococcus_aureus            0.94 [0;1]  0.95 [0;1]  0.94 [0;1]  \n",
       "Staphylococcus_aureus_plasmid1   0.82 [0;1]  0.82 [0;1]  0.81 [0;1]  \n",
       "Staphylococcus_aureus_plasmid2    0.7 [0;1]  0.71 [0;1]  0.68 [0;1]  \n",
       "Staphylococcus_aureus_plasmid3   0.73 [0;1]  0.73 [0;1]  0.72 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ref_stats_per_reference_all_assemblers = pd.DataFrame()\n",
    "for reference in reference_pipeline_metrics_df.reference.unique():\n",
    "    row={'Reference': reference}\n",
    "    lala=reference_pipeline_metrics_df[(reference_pipeline_metrics_df.reference == reference) & (reference_pipeline_metrics_df.distribution == 'Even')].describe()\n",
    "    for column in lala.columns:\n",
    "        mean = lala.loc['mean',column]\n",
    "        minimum = lala.loc['min',column]\n",
    "        maximum = lala.loc['max',column]\n",
    "        row[column] = \"{} [{};{}]\".format(round(mean, 2),int(minimum), int(maximum))\n",
    "    ref_stats_per_reference_all_assemblers=ref_stats_per_reference_all_assemblers.append(row, ignore_index=True)\n",
    "ref_stats_per_reference_all_assemblers = ref_stats_per_reference_all_assemblers.set_index('Reference')\n",
    "ref_stats_per_reference_all_assemblers.to_csv(\"Tables/Results/Reference metrics per replicon.csv\")\n",
    "display(ref_stats_per_reference_all_assemblers)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reference stats per distribution"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Even\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>60.4 [0;1004]</td>\n",
       "      <td>0.3 [0;0]</td>\n",
       "      <td>101808.53 [0;351502]</td>\n",
       "      <td>101667.72 [0;351502]</td>\n",
       "      <td>92.14 [0;2170]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2534947.7 [0;6780808]</td>\n",
       "      <td>0.91 [0;0]</td>\n",
       "      <td>97.11 [0;1404]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.33 [0;8]</td>\n",
       "      <td>0.59 [0;14]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>51.2 [0;921]</td>\n",
       "      <td>0.04 [0;0]</td>\n",
       "      <td>6932.33 [0;65138]</td>\n",
       "      <td>6539.94 [0;64934]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1236062.67 [0;6510588]</td>\n",
       "      <td>0.37 [0;0]</td>\n",
       "      <td>272.97 [0;1495]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "      <td>0.5 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>31.33 [0;112]</td>\n",
       "      <td>0.35 [0;1]</td>\n",
       "      <td>106093.76 [2176;452364]</td>\n",
       "      <td>96782.73 [2176;351186]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2523829.08 [2176;6716522]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "      <td>59.92 [1;199]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "      <td>0.14 [0;3]</td>\n",
       "      <td>0.25 [0;6]</td>\n",
       "      <td>1.01 [0;1]</td>\n",
       "      <td>1.01 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>68.82 [0;283]</td>\n",
       "      <td>0.27 [0;1]</td>\n",
       "      <td>55836.03 [0;254629]</td>\n",
       "      <td>54554.11 [0;254629]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2507600.54 [0;6710005]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>121.12 [0;449]</td>\n",
       "      <td>0.97 [0;1]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "      <td>1.12 [0;9]</td>\n",
       "      <td>1.94 [0;15]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "      <td>0.97 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>18.31 [0;55]</td>\n",
       "      <td>0.37 [0;1]</td>\n",
       "      <td>138989.33 [2215;542872]</td>\n",
       "      <td>141793.83 [2215;542872]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2534425.59 [2215;6737986]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>44.39 [1;141]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.93 [0;1]</td>\n",
       "      <td>0.97 [0;4]</td>\n",
       "      <td>1.99 [0;7]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.01 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>21.7 [0;110]</td>\n",
       "      <td>0.23 [0;0]</td>\n",
       "      <td>58464.9 [0;257424]</td>\n",
       "      <td>57747.71 [0;257424]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1722404.01 [0;6717767]</td>\n",
       "      <td>0.68 [0;0]</td>\n",
       "      <td>74.19 [0;361]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>20.14 [0;90]</td>\n",
       "      <td>0.34 [0;1]</td>\n",
       "      <td>133559.97 [1677;475510]</td>\n",
       "      <td>132927.86 [1677;475510]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2530047.11 [2076;6726884]</td>\n",
       "      <td>0.98 [0;1]</td>\n",
       "      <td>39.33 [1;167]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.99 [0;1]</td>\n",
       "      <td>0.14 [0;2]</td>\n",
       "      <td>0.33 [0;6]</td>\n",
       "      <td>1.01 [0;1]</td>\n",
       "      <td>1.01 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>83.41 [0;407]</td>\n",
       "      <td>0.08 [0;0]</td>\n",
       "      <td>17271.52 [0;65138]</td>\n",
       "      <td>16622.5 [0;64960]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2345242.24 [0;6518607]</td>\n",
       "      <td>0.74 [0;0]</td>\n",
       "      <td>307.84 [0;1147]</td>\n",
       "      <td>0.81 [0;1]</td>\n",
       "      <td>0.79 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.81 [0;1]</td>\n",
       "      <td>0.81 [0;1]</td>\n",
       "      <td>0.81 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>57.92 [0;670]</td>\n",
       "      <td>0.27 [0;0]</td>\n",
       "      <td>76053.75 [1404;351014]</td>\n",
       "      <td>75134.92 [0;351014]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2496709.31 [1460;6673601]</td>\n",
       "      <td>0.9 [0;0]</td>\n",
       "      <td>96.06 [1;933]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>0.87 [0;1]</td>\n",
       "      <td>0.67 [0;7]</td>\n",
       "      <td>1.17 [0;15]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "      <td>1.0 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>21.06 [0;79]</td>\n",
       "      <td>0.29 [0;1]</td>\n",
       "      <td>117650.97 [0;315090]</td>\n",
       "      <td>117061.69 [0;315090]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2524122.0 [0;6728888]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "      <td>41.19 [0;143]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.9 [0;1]</td>\n",
       "      <td>0.36 [0;4]</td>\n",
       "      <td>0.69 [0;8]</td>\n",
       "      <td>0.95 [0;1]</td>\n",
       "      <td>0.96 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>15.75 [0;57]</td>\n",
       "      <td>0.38 [0;1]</td>\n",
       "      <td>162270.42 [0;475179]</td>\n",
       "      <td>158177.44 [0;475179]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2515166.11 [0;6723626]</td>\n",
       "      <td>0.92 [0;1]</td>\n",
       "      <td>32.86 [0;111]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.06 [0;1]</td>\n",
       "      <td>0.11 [0;2]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "      <td>0.94 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>31.33 [0;207]</td>\n",
       "      <td>0.09 [0;0]</td>\n",
       "      <td>29632.83 [0;338923]</td>\n",
       "      <td>34682.42 [0;464977]</td>\n",
       "      <td>537.64 [0;12098]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1580537.53 [0;6708956]</td>\n",
       "      <td>0.52 [0;0]</td>\n",
       "      <td>167.92 [0;1098]</td>\n",
       "      <td>0.56 [0;1]</td>\n",
       "      <td>0.55 [0;1]</td>\n",
       "      <td>0.39 [0;8]</td>\n",
       "      <td>0.61 [0;11]</td>\n",
       "      <td>0.56 [0;1]</td>\n",
       "      <td>0.56 [0;1]</td>\n",
       "      <td>0.56 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm            L90         LSA  \\\n",
       "Assembler                                                                      \n",
       "ABySS              Multiple k-mer De Bruijn graph  60.4 [0;1004]   0.3 [0;0]   \n",
       "BCALM2               Single k-mer De Bruijn graph   51.2 [0;921]  0.04 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph  31.33 [0;112]  0.35 [0;1]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  68.82 [0;283]  0.27 [0;1]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph   18.31 [0;55]  0.37 [0;1]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph   21.7 [0;110]  0.23 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph   20.14 [0;90]  0.34 [0;1]   \n",
       "MINIA                Single k-mer De Bruijn graph  83.41 [0;407]  0.08 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph  57.92 [0;670]  0.27 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph   21.06 [0;79]  0.29 [0;1]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph   15.75 [0;57]  0.38 [0;1]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph  31.33 [0;207]  0.09 [0;0]   \n",
       "\n",
       "                                      NA50                     NG50  \\\n",
       "Assembler                                                             \n",
       "ABySS                 101808.53 [0;351502]     101667.72 [0;351502]   \n",
       "BCALM2                   6932.33 [0;65138]        6539.94 [0;64934]   \n",
       "GATBMiniaPipeline  106093.76 [2176;452364]   96782.73 [2176;351186]   \n",
       "IDBA-UD                55836.03 [0;254629]      54554.11 [0;254629]   \n",
       "MEGAHIT            138989.33 [2215;542872]  141793.83 [2215;542872]   \n",
       "MetaHipMer2             58464.9 [0;257424]      57747.71 [0;257424]   \n",
       "metaSPAdes         133559.97 [1677;475510]  132927.86 [1677;475510]   \n",
       "MINIA                   17271.52 [0;65138]        16622.5 [0;64960]   \n",
       "SKESA               76053.75 [1404;351014]      75134.92 [0;351014]   \n",
       "SPAdes                117650.97 [0;315090]     117061.69 [0;315090]   \n",
       "Unicycler             162270.42 [0;475179]     158177.44 [0;475179]   \n",
       "VelvetOptimiser        29632.83 [0;338923]      34682.42 [0;464977]   \n",
       "\n",
       "                                 Ns         Type                  basepairs  \\\n",
       "Assembler                                                                     \n",
       "ABySS                92.14 [0;2170]  Metagenomic      2534947.7 [0;6780808]   \n",
       "BCALM2                    0.0 [0;0]      Genomic     1236062.67 [0;6510588]   \n",
       "GATBMiniaPipeline         0.0 [0;0]  Metagenomic  2523829.08 [2176;6716522]   \n",
       "IDBA-UD                   0.0 [0;0]  Metagenomic     2507600.54 [0;6710005]   \n",
       "MEGAHIT                   0.0 [0;0]  Metagenomic  2534425.59 [2215;6737986]   \n",
       "MetaHipMer2               0.0 [0;0]  Metagenomic     1722404.01 [0;6717767]   \n",
       "metaSPAdes                0.0 [0;0]  Metagenomic  2530047.11 [2076;6726884]   \n",
       "MINIA                     0.0 [0;0]      Genomic     2345242.24 [0;6518607]   \n",
       "SKESA                     0.0 [0;0]      Genomic  2496709.31 [1460;6673601]   \n",
       "SPAdes                    0.0 [0;0]      Genomic      2524122.0 [0;6728888]   \n",
       "Unicycler                 0.0 [0;0]      Genomic     2515166.11 [0;6723626]   \n",
       "VelvetOptimiser    537.64 [0;12098]  Metagenomic     1580537.53 [0;6708956]   \n",
       "\n",
       "                  breadth_of_coverage          contigs    identity  \\\n",
       "Assembler                                                            \n",
       "ABySS                      0.91 [0;0]   97.11 [0;1404]  0.98 [0;1]   \n",
       "BCALM2                     0.37 [0;0]  272.97 [0;1495]   0.5 [0;1]   \n",
       "GATBMiniaPipeline          0.98 [0;1]    59.92 [1;199]   1.0 [0;1]   \n",
       "IDBA-UD                    0.94 [0;1]   121.12 [0;449]  0.97 [0;1]   \n",
       "MEGAHIT                    0.99 [0;1]    44.39 [1;141]   1.0 [0;1]   \n",
       "MetaHipMer2                0.68 [0;0]    74.19 [0;361]  0.94 [0;1]   \n",
       "metaSPAdes                 0.98 [0;1]    39.33 [1;167]   1.0 [0;1]   \n",
       "MINIA                      0.74 [0;0]  307.84 [0;1147]  0.81 [0;1]   \n",
       "SKESA                       0.9 [0;0]    96.06 [1;933]   1.0 [0;1]   \n",
       "SPAdes                      0.9 [0;1]    41.19 [0;143]  0.94 [0;1]   \n",
       "Unicycler                  0.92 [0;1]    32.86 [0;111]  0.94 [0;1]   \n",
       "VelvetOptimiser            0.52 [0;0]  167.92 [0;1098]  0.56 [0;1]   \n",
       "\n",
       "                  lowest_identity misassembled contigs misassembly events  \\\n",
       "Assembler                                                                   \n",
       "ABySS                  0.94 [0;1]           0.33 [0;8]        0.59 [0;14]   \n",
       "BCALM2                  0.5 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "GATBMiniaPipeline      0.98 [0;1]           0.14 [0;3]         0.25 [0;6]   \n",
       "IDBA-UD                0.87 [0;1]           1.12 [0;9]        1.94 [0;15]   \n",
       "MEGAHIT                0.93 [0;1]           0.97 [0;4]         1.99 [0;7]   \n",
       "MetaHipMer2            0.94 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "metaSPAdes             0.99 [0;1]           0.14 [0;2]         0.33 [0;6]   \n",
       "MINIA                  0.79 [0;1]            0.0 [0;0]          0.0 [0;0]   \n",
       "SKESA                  0.87 [0;1]           0.67 [0;7]        1.17 [0;15]   \n",
       "SPAdes                  0.9 [0;1]           0.36 [0;4]         0.69 [0;8]   \n",
       "Unicycler              0.94 [0;1]           0.06 [0;1]         0.11 [0;2]   \n",
       "VelvetOptimiser        0.55 [0;1]           0.39 [0;8]        0.61 [0;11]   \n",
       "\n",
       "                  multiplicity   parsimony    validity  \n",
       "Assembler                                               \n",
       "ABySS               0.99 [0;1]  0.99 [0;1]  0.98 [0;1]  \n",
       "BCALM2               0.5 [0;1]   0.5 [0;1]   0.5 [0;1]  \n",
       "GATBMiniaPipeline   1.01 [0;1]  1.01 [0;1]   1.0 [0;1]  \n",
       "IDBA-UD             0.98 [0;1]  0.98 [0;1]  0.97 [0;1]  \n",
       "MEGAHIT              1.0 [0;1]  1.01 [0;1]   1.0 [0;1]  \n",
       "MetaHipMer2         0.94 [0;1]  0.94 [0;1]  0.94 [0;1]  \n",
       "metaSPAdes          1.01 [0;1]  1.01 [0;1]   1.0 [0;1]  \n",
       "MINIA               0.81 [0;1]  0.81 [0;1]  0.81 [0;1]  \n",
       "SKESA                1.0 [0;1]   1.0 [0;1]   1.0 [0;1]  \n",
       "SPAdes              0.95 [0;1]  0.96 [0;1]  0.94 [0;1]  \n",
       "Unicycler           0.94 [0;1]  0.94 [0;1]  0.94 [0;1]  \n",
       "VelvetOptimiser     0.56 [0;1]  0.56 [0;1]  0.56 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABySS</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>50.54 [0;1033]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>25986.46 [0;351511]</td>\n",
       "      <td>25866.25 [0;351511]</td>\n",
       "      <td>16.67 [0;400]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>690353.25 [0;6781605]</td>\n",
       "      <td>0.16 [0;0]</td>\n",
       "      <td>65.12 [0;1141]</td>\n",
       "      <td>0.21 [0;0]</td>\n",
       "      <td>0.2 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.21 [0;1]</td>\n",
       "      <td>0.21 [0;1]</td>\n",
       "      <td>0.21 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>13.58 [0;311]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>3455.14 [0;64777]</td>\n",
       "      <td>3263.39 [0;62493]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>390196.06 [0;6515102]</td>\n",
       "      <td>0.09 [0;0]</td>\n",
       "      <td>35.86 [0;496]</td>\n",
       "      <td>0.17 [0;1]</td>\n",
       "      <td>0.17 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.17 [0;1]</td>\n",
       "      <td>0.17 [0;1]</td>\n",
       "      <td>0.17 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>23.58 [0;459]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>41260.94 [0;351149]</td>\n",
       "      <td>40902.42 [0;351149]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1131013.97 [0;6715846]</td>\n",
       "      <td>0.25 [0;0]</td>\n",
       "      <td>37.56 [0;604]</td>\n",
       "      <td>0.36 [0;1]</td>\n",
       "      <td>0.36 [0;1]</td>\n",
       "      <td>0.08 [0;2]</td>\n",
       "      <td>0.25 [0;6]</td>\n",
       "      <td>0.36 [0;1]</td>\n",
       "      <td>0.36 [0;1]</td>\n",
       "      <td>0.36 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>16.33 [0;292]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>48013.69 [0;475377]</td>\n",
       "      <td>46431.78 [0;475377]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1135231.19 [0;6700232]</td>\n",
       "      <td>0.25 [0;0]</td>\n",
       "      <td>29.56 [0;418]</td>\n",
       "      <td>0.46 [0;0]</td>\n",
       "      <td>0.38 [0;0]</td>\n",
       "      <td>0.25 [0;2]</td>\n",
       "      <td>0.39 [0;5]</td>\n",
       "      <td>0.47 [0;1]</td>\n",
       "      <td>0.61 [0;5]</td>\n",
       "      <td>0.44 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>8.42 [0;147]</td>\n",
       "      <td>0.03 [0;0]</td>\n",
       "      <td>66650.44 [0;543128]</td>\n",
       "      <td>66455.53 [0;543128]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1141249.42 [0;6738936]</td>\n",
       "      <td>0.25 [0;0]</td>\n",
       "      <td>18.11 [0;223]</td>\n",
       "      <td>0.46 [0;0]</td>\n",
       "      <td>0.42 [0;0]</td>\n",
       "      <td>0.06 [0;1]</td>\n",
       "      <td>0.11 [0;2]</td>\n",
       "      <td>0.47 [0;1]</td>\n",
       "      <td>0.48 [0;1]</td>\n",
       "      <td>0.47 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MetaHipMer2</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>903.25 [0;2986]</td>\n",
       "      <td>528.83 [0;2697]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>824940.25 [0;3766165]</td>\n",
       "      <td>0.19 [0;0]</td>\n",
       "      <td>407.58 [0;2402]</td>\n",
       "      <td>0.42 [0;1]</td>\n",
       "      <td>0.39 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.42 [0;1]</td>\n",
       "      <td>0.42 [0;1]</td>\n",
       "      <td>0.42 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>7.19 [0;66]</td>\n",
       "      <td>0.03 [0;0]</td>\n",
       "      <td>60717.78 [0;593167]</td>\n",
       "      <td>59453.89 [0;593167]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1140707.44 [0;6733500]</td>\n",
       "      <td>0.25 [0;0]</td>\n",
       "      <td>15.53 [0;110]</td>\n",
       "      <td>0.47 [0;0]</td>\n",
       "      <td>0.43 [0;0]</td>\n",
       "      <td>0.17 [0;2]</td>\n",
       "      <td>0.39 [0;6]</td>\n",
       "      <td>0.47 [0;1]</td>\n",
       "      <td>0.48 [0;1]</td>\n",
       "      <td>0.47 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>65.11 [0;649]</td>\n",
       "      <td>0.01 [0;0]</td>\n",
       "      <td>6467.22 [0;64777]</td>\n",
       "      <td>5777.47 [0;64473]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1099797.58 [0;6524118]</td>\n",
       "      <td>0.24 [0;0]</td>\n",
       "      <td>119.58 [0;918]</td>\n",
       "      <td>0.42 [0;1]</td>\n",
       "      <td>0.39 [0;1]</td>\n",
       "      <td>0.14 [0;2]</td>\n",
       "      <td>0.22 [0;4]</td>\n",
       "      <td>0.42 [0;1]</td>\n",
       "      <td>0.42 [0;1]</td>\n",
       "      <td>0.42 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>6.72 [0;232]</td>\n",
       "      <td>0.01 [0;0]</td>\n",
       "      <td>10421.97 [0;351027]</td>\n",
       "      <td>10210.64 [0;351027]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>366940.31 [0;3512871]</td>\n",
       "      <td>0.1 [0;0]</td>\n",
       "      <td>120.94 [0;1877]</td>\n",
       "      <td>0.19 [0;1]</td>\n",
       "      <td>0.19 [0;1]</td>\n",
       "      <td>0.33 [0;12]</td>\n",
       "      <td>0.64 [0;23]</td>\n",
       "      <td>0.19 [0;1]</td>\n",
       "      <td>0.19 [0;1]</td>\n",
       "      <td>0.19 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>30.83 [0;835]</td>\n",
       "      <td>0.02 [0;0]</td>\n",
       "      <td>36831.08 [0;351104]</td>\n",
       "      <td>37693.72 [0;351104]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1093338.33 [0;6710057]</td>\n",
       "      <td>0.23 [0;0]</td>\n",
       "      <td>105.17 [0;1839]</td>\n",
       "      <td>0.56 [0;1]</td>\n",
       "      <td>0.49 [0;1]</td>\n",
       "      <td>16.08 [0;565]</td>\n",
       "      <td>32.11 [0;1134]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.63 [0;1]</td>\n",
       "      <td>0.56 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>7.17 [0;143]</td>\n",
       "      <td>0.03 [0;0]</td>\n",
       "      <td>51507.03 [0;351002]</td>\n",
       "      <td>49210.58 [0;351002]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>963132.78 [0;6712831]</td>\n",
       "      <td>0.2 [0;0]</td>\n",
       "      <td>13.72 [0;179]</td>\n",
       "      <td>0.22 [0;1]</td>\n",
       "      <td>0.22 [0;1]</td>\n",
       "      <td>0.14 [0;5]</td>\n",
       "      <td>0.28 [0;10]</td>\n",
       "      <td>0.22 [0;1]</td>\n",
       "      <td>0.22 [0;1]</td>\n",
       "      <td>0.22 [0;1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimiser</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>533.11 [0;3489]</td>\n",
       "      <td>253.03 [0;3015]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>389436.31 [0;3508763]</td>\n",
       "      <td>0.09 [0;0]</td>\n",
       "      <td>174.11 [0;1988]</td>\n",
       "      <td>0.28 [0;1]</td>\n",
       "      <td>0.28 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.28 [0;1]</td>\n",
       "      <td>0.28 [0;1]</td>\n",
       "      <td>0.28 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm             L90         LSA  \\\n",
       "Assembler                                                                       \n",
       "ABySS              Multiple k-mer De Bruijn graph  50.54 [0;1033]  0.02 [0;0]   \n",
       "BCALM2               Single k-mer De Bruijn graph   13.58 [0;311]   0.0 [0;0]   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph   23.58 [0;459]  0.02 [0;0]   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph   16.33 [0;292]  0.02 [0;0]   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph    8.42 [0;147]  0.03 [0;0]   \n",
       "MetaHipMer2        Multiple k-mer De Bruijn graph       0.0 [0;0]   0.0 [0;0]   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph     7.19 [0;66]  0.03 [0;0]   \n",
       "MINIA                Single k-mer De Bruijn graph   65.11 [0;649]  0.01 [0;0]   \n",
       "SKESA              Multiple k-mer De Bruijn graph    6.72 [0;232]  0.01 [0;0]   \n",
       "SPAdes             Multiple k-mer De Bruijn graph   30.83 [0;835]  0.02 [0;0]   \n",
       "Unicycler          Multiple k-mer De Bruijn graph    7.17 [0;143]  0.03 [0;0]   \n",
       "VelvetOptimiser    Multiple k-mer De Bruijn graph       0.0 [0;0]   0.0 [0;0]   \n",
       "\n",
       "                                  NA50                 NG50             Ns  \\\n",
       "Assembler                                                                    \n",
       "ABySS              25986.46 [0;351511]  25866.25 [0;351511]  16.67 [0;400]   \n",
       "BCALM2               3455.14 [0;64777]    3263.39 [0;62493]      0.0 [0;0]   \n",
       "GATBMiniaPipeline  41260.94 [0;351149]  40902.42 [0;351149]      0.0 [0;0]   \n",
       "IDBA-UD            48013.69 [0;475377]  46431.78 [0;475377]      0.0 [0;0]   \n",
       "MEGAHIT            66650.44 [0;543128]  66455.53 [0;543128]      0.0 [0;0]   \n",
       "MetaHipMer2            903.25 [0;2986]      528.83 [0;2697]      0.0 [0;0]   \n",
       "metaSPAdes         60717.78 [0;593167]  59453.89 [0;593167]      0.0 [0;0]   \n",
       "MINIA                6467.22 [0;64777]    5777.47 [0;64473]      0.0 [0;0]   \n",
       "SKESA              10421.97 [0;351027]  10210.64 [0;351027]      0.0 [0;0]   \n",
       "SPAdes             36831.08 [0;351104]  37693.72 [0;351104]      0.0 [0;0]   \n",
       "Unicycler          51507.03 [0;351002]  49210.58 [0;351002]      0.0 [0;0]   \n",
       "VelvetOptimiser        533.11 [0;3489]      253.03 [0;3015]      0.0 [0;0]   \n",
       "\n",
       "                          Type               basepairs breadth_of_coverage  \\\n",
       "Assembler                                                                    \n",
       "ABySS              Metagenomic   690353.25 [0;6781605]          0.16 [0;0]   \n",
       "BCALM2                 Genomic   390196.06 [0;6515102]          0.09 [0;0]   \n",
       "GATBMiniaPipeline  Metagenomic  1131013.97 [0;6715846]          0.25 [0;0]   \n",
       "IDBA-UD            Metagenomic  1135231.19 [0;6700232]          0.25 [0;0]   \n",
       "MEGAHIT            Metagenomic  1141249.42 [0;6738936]          0.25 [0;0]   \n",
       "MetaHipMer2        Metagenomic   824940.25 [0;3766165]          0.19 [0;0]   \n",
       "metaSPAdes         Metagenomic  1140707.44 [0;6733500]          0.25 [0;0]   \n",
       "MINIA                  Genomic  1099797.58 [0;6524118]          0.24 [0;0]   \n",
       "SKESA                  Genomic   366940.31 [0;3512871]           0.1 [0;0]   \n",
       "SPAdes                 Genomic  1093338.33 [0;6710057]          0.23 [0;0]   \n",
       "Unicycler              Genomic   963132.78 [0;6712831]           0.2 [0;0]   \n",
       "VelvetOptimiser    Metagenomic   389436.31 [0;3508763]          0.09 [0;0]   \n",
       "\n",
       "                           contigs    identity lowest_identity  \\\n",
       "Assembler                                                        \n",
       "ABySS               65.12 [0;1141]  0.21 [0;0]       0.2 [0;0]   \n",
       "BCALM2               35.86 [0;496]  0.17 [0;1]      0.17 [0;1]   \n",
       "GATBMiniaPipeline    37.56 [0;604]  0.36 [0;1]      0.36 [0;1]   \n",
       "IDBA-UD              29.56 [0;418]  0.46 [0;0]      0.38 [0;0]   \n",
       "MEGAHIT              18.11 [0;223]  0.46 [0;0]      0.42 [0;0]   \n",
       "MetaHipMer2        407.58 [0;2402]  0.42 [0;1]      0.39 [0;1]   \n",
       "metaSPAdes           15.53 [0;110]  0.47 [0;0]      0.43 [0;0]   \n",
       "MINIA               119.58 [0;918]  0.42 [0;1]      0.39 [0;1]   \n",
       "SKESA              120.94 [0;1877]  0.19 [0;1]      0.19 [0;1]   \n",
       "SPAdes             105.17 [0;1839]  0.56 [0;1]      0.49 [0;1]   \n",
       "Unicycler            13.72 [0;179]  0.22 [0;1]      0.22 [0;1]   \n",
       "VelvetOptimiser    174.11 [0;1988]  0.28 [0;1]      0.28 [0;1]   \n",
       "\n",
       "                  misassembled contigs misassembly events multiplicity  \\\n",
       "Assembler                                                                \n",
       "ABySS                        0.0 [0;0]          0.0 [0;0]   0.21 [0;1]   \n",
       "BCALM2                       0.0 [0;0]          0.0 [0;0]   0.17 [0;1]   \n",
       "GATBMiniaPipeline           0.08 [0;2]         0.25 [0;6]   0.36 [0;1]   \n",
       "IDBA-UD                     0.25 [0;2]         0.39 [0;5]   0.47 [0;1]   \n",
       "MEGAHIT                     0.06 [0;1]         0.11 [0;2]   0.47 [0;1]   \n",
       "MetaHipMer2                  0.0 [0;0]          0.0 [0;0]   0.42 [0;1]   \n",
       "metaSPAdes                  0.17 [0;2]         0.39 [0;6]   0.47 [0;1]   \n",
       "MINIA                       0.14 [0;2]         0.22 [0;4]   0.42 [0;1]   \n",
       "SKESA                      0.33 [0;12]        0.64 [0;23]   0.19 [0;1]   \n",
       "SPAdes                   16.08 [0;565]     32.11 [0;1134]    0.6 [0;1]   \n",
       "Unicycler                   0.14 [0;5]        0.28 [0;10]   0.22 [0;1]   \n",
       "VelvetOptimiser              0.0 [0;0]          0.0 [0;0]   0.28 [0;1]   \n",
       "\n",
       "                    parsimony    validity  \n",
       "Assembler                                  \n",
       "ABySS              0.21 [0;1]  0.21 [0;1]  \n",
       "BCALM2             0.17 [0;1]  0.17 [0;1]  \n",
       "GATBMiniaPipeline  0.36 [0;1]  0.36 [0;1]  \n",
       "IDBA-UD            0.61 [0;5]  0.44 [0;1]  \n",
       "MEGAHIT            0.48 [0;1]  0.47 [0;1]  \n",
       "MetaHipMer2        0.42 [0;1]  0.42 [0;1]  \n",
       "metaSPAdes         0.48 [0;1]  0.47 [0;1]  \n",
       "MINIA              0.42 [0;1]  0.42 [0;1]  \n",
       "SKESA              0.19 [0;1]  0.19 [0;1]  \n",
       "SPAdes             0.63 [0;1]  0.56 [0;1]  \n",
       "Unicycler          0.22 [0;1]  0.22 [0;1]  \n",
       "VelvetOptimiser    0.28 [0;1]  0.28 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "for distribution in reference_pipeline_metrics_df.distribution.unique():\n",
    "    ref_stats_per_reference_per_assembler = pd.DataFrame()\n",
    "    for assembler in reference_pipeline_metrics_df.assembler.unique():\n",
    "        row={'Assembler': assembler, \n",
    "             'Type':reference_pipeline_metrics_df[reference_pipeline_metrics_df.assembler == assembler].type.unique()[0],\n",
    "             'Algorithm': reference_pipeline_metrics_df[reference_pipeline_metrics_df.assembler == assembler].algorythm.unique()[0]}\n",
    "        lala=reference_pipeline_metrics_df[(reference_pipeline_metrics_df.assembler == assembler) & (reference_pipeline_metrics_df.distribution == distribution)].describe()\n",
    "        for column in lala.columns:\n",
    "            mean = lala.loc['mean',column]\n",
    "            minimum = lala.loc['min',column]\n",
    "            maximum = lala.loc['max',column]\n",
    "            row[column] = \"{} [{};{}]\".format(round(mean, 2),int(minimum), int(maximum))\n",
    "        ref_stats_per_reference_per_assembler=ref_stats_per_reference_per_assembler.append(row, ignore_index=True)\n",
    "    ref_stats_per_reference_per_assembler = ref_stats_per_reference_per_assembler.set_index('Assembler')\n",
    "    ref_stats_per_reference_per_assembler.to_csv(\"Tables/Results/{} - Reference metrics per assembler.csv\".format(distribution))\n",
    "    print(distribution)\n",
    "    display(ref_stats_per_reference_per_assembler)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Genomes vs Plasmids"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>L90</th>\n",
       "      <th>LSA</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassembled contigs</th>\n",
       "      <th>misassembly events</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Replicon</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Genome</th>\n",
       "      <td>52.28(± 106.83)</td>\n",
       "      <td>0.07(± 0.08)</td>\n",
       "      <td>104300.28(± 121733.48)</td>\n",
       "      <td>103102.92(± 121619.06)</td>\n",
       "      <td>56.51(± 711.48)</td>\n",
       "      <td>2880781.06(± 2043121.66)</td>\n",
       "      <td>0.73(± 0.41)</td>\n",
       "      <td>155.8(± 296.42)</td>\n",
       "      <td>0.84(± 0.36)</td>\n",
       "      <td>0.79(± 0.37)</td>\n",
       "      <td>0.95(± 16.92)</td>\n",
       "      <td>1.84(± 33.95)</td>\n",
       "      <td>0.84(± 0.37)</td>\n",
       "      <td>0.85(± 0.39)</td>\n",
       "      <td>0.84(± 0.36)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Plasmid</th>\n",
       "      <td>0.65 [0;18]</td>\n",
       "      <td>0.44 [0;1]</td>\n",
       "      <td>5036.33 [0;55639]</td>\n",
       "      <td>4925.31 [0;55639]</td>\n",
       "      <td>7.18 [0;835]</td>\n",
       "      <td>17373.76 [0;109516]</td>\n",
       "      <td>0.55 [0;1]</td>\n",
       "      <td>2.22 [0;33]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.0 [0;0]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "      <td>0.61 [0;1]</td>\n",
       "      <td>0.6 [0;1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      L90           LSA                    NA50  \\\n",
       "Replicon                                                          \n",
       "Genome    52.28(± 106.83)  0.07(± 0.08)  104300.28(± 121733.48)   \n",
       "Plasmid       0.65 [0;18]    0.44 [0;1]       5036.33 [0;55639]   \n",
       "\n",
       "                            NG50               Ns                 basepairs  \\\n",
       "Replicon                                                                      \n",
       "Genome    103102.92(± 121619.06)  56.51(± 711.48)  2880781.06(± 2043121.66)   \n",
       "Plasmid        4925.31 [0;55639]     7.18 [0;835]       17373.76 [0;109516]   \n",
       "\n",
       "         breadth_of_coverage          contigs      identity lowest_identity  \\\n",
       "Replicon                                                                      \n",
       "Genome          0.73(± 0.41)  155.8(± 296.42)  0.84(± 0.36)    0.79(± 0.37)   \n",
       "Plasmid           0.55 [0;1]      2.22 [0;33]     0.6 [0;1]       0.6 [0;1]   \n",
       "\n",
       "         misassembled contigs misassembly events  multiplicity     parsimony  \\\n",
       "Replicon                                                                       \n",
       "Genome          0.95(± 16.92)      1.84(± 33.95)  0.84(± 0.37)  0.85(± 0.39)   \n",
       "Plasmid             0.0 [0;0]          0.0 [0;0]     0.6 [0;1]    0.61 [0;1]   \n",
       "\n",
       "              validity  \n",
       "Replicon                \n",
       "Genome    0.84(± 0.36)  \n",
       "Plasmid      0.6 [0;1]  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ref_stats_per_replicon_type = pd.DataFrame()\n",
    "# Genome\n",
    "row={'Replicon': \"Genome\"}\n",
    "lala=reference_pipeline_metrics_df[~reference_pipeline_metrics_df['reference'].str.contains(\"plasmid\")].describe()\n",
    "for column in lala.columns:\n",
    "    mean = lala.loc['mean',column]\n",
    "    std = lala.loc['std',column]\n",
    "    row[column] = \"{}(± {})\".format(round(mean, 2), round(std, 2))\n",
    "ref_stats_per_replicon_type =ref_stats_per_replicon_type .append(row, ignore_index=True)\n",
    "\n",
    "# Plasmid\n",
    "row={'Replicon': \"Plasmid\"}\n",
    "lala=reference_pipeline_metrics_df[reference_pipeline_metrics_df['reference'].str.contains(\"plasmid\")].describe()\n",
    "for column in lala.columns:\n",
    "    mean = lala.loc['mean',column]\n",
    "    minimum = lala.loc['min',column]\n",
    "    maximum = lala.loc['max',column]\n",
    "    row[column] = \"{} [{};{}]\".format(round(mean, 2),int(minimum), int(maximum))\n",
    "    \n",
    "ref_stats_per_replicon_type =ref_stats_per_replicon_type .append(row, ignore_index=True)\n",
    "ref_stats_per_replicon_type  = ref_stats_per_replicon_type .set_index('Replicon')\n",
    "ref_stats_per_replicon_type.to_csv(\"Tables/Results/Reference metrics per replicon type.csv\")\n",
    "display(ref_stats_per_replicon_type)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
