{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Results Tables"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The success of an assembly is evaluated by the computation of metrics in two defined ways: globally through statistics inherent to the complete set of sequences that were assembled, and relative to the replicons present in the sample. \n",
    "\n",
    "The computation of the global metrics is done through custom python code from the complete set of sequences assembled. \n",
    "\n",
    "The following metrics are computed for the complete and filtered set of assembled sequences, restricted to contigs of length above a specified minimum size: \n",
    "\n",
    "- **Contig sizes**\n",
    "    - **Contigs:** The total number of contigs in the assembly;\n",
    "    - **Basepairs:** The total number of bases in the assembly;\n",
    "    - **Maximum sequence length:** The length of the largest contig in the assembly.\n",
    "    - **Number of ‘N’s:** Number of uncalled bases (N's) \n",
    "- **Contiguity**\n",
    "    - **Nx (where 0  < x  ⩽ 100):** Length for which the collection of all assembled sequences of that length or longer in an assembly covers at least a given percentage of the total length of the assembly\n",
    "- **Misassembly**\n",
    "    - **Misassemblies** - Number of aligned contigs that contain a misassembly event\n",
    "\n",
    "After filtering, the sequences are mapped with the reference replicons and the metrics are computed through custom python code. The metrics are calculated, filtered for a minimum length, for each replicon in the file provided input references. \n",
    "\n",
    "- **Contig sizes**\n",
    "    - **Contigs:** The total number of contigs in the assembly;\n",
    "    - **Basepairs:** The total number of bases in the assembly;\n",
    "    - **Number of ‘N’s:** Number of uncalled bases (N's) \n",
    "- **COMPASS**\n",
    "    - **(Breadth of) Coverage:** Ratio of covered sequence on the reference by aligned contigs;\n",
    "    - **Multiplicity:** Ratio of the length of alignable assembled sequence to covered sequence on the reference;\n",
    "    - **Validity:** Ratio of the length of the alignable assembled sequence to total basepairs in the aligned contigs;\n",
    "    - **Parsimony:** Cost of the assembly (multiplicity over validity);\n",
    "- **Contiguity**\n",
    "    - **Contiguity:** longest single alignment between the assembly and the reference, relative to the reference length;\n",
    "    - **NAx (where 0  < x  ⩽ 100):** Length for which the collection of aligned assembled sequences of that length or longer in an assembly covers at least a given percentage of the total length of the reference replicon;\n",
    "    - **NGx (where 0  < x  ⩽ 100):** Length for which the collection of aligned contigs of that length or longer covers at least a given percentage of the sequence of the reference.\n",
    "    - **Lx (where 0  < x  ⩽ 100):** Minimal number of contigs that cover x % of the sequence of the reference;\n",
    "- **Identity**\n",
    "    - **Identity:** Ratio of identical basepairs in all aligned contigs to the reference;\n",
    "    - **Lowest identity:** Identity of the lowest scoring contig to the reference.\n",
    "    - **PLS**: Phred-like score per contig, per assembler.\n",
    "- **Misassembly**\n",
    "    - **Misassemblies** - Number of aligned contigs that contain a misassembly event\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "from plotly.offline import plot\n",
    "import glob\n",
    "import fnmatch\n",
    "import plotly.graph_objects as go\n",
    "from plotly.subplots import make_subplots\n",
    "import json\n",
    "import pandas as pd\n",
    "from itertools import groupby\n",
    "import csv\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Global variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "METRICS_COLUMNS_GLOBAL = ['run','sample','assembler', 'contigs', 'basepairs', 'max_contig', 'Ns', 'n50', \n",
    "                          'misassemblies', 'mapped_reads', 'filtered_contigs', 'filtered_basepairs',\n",
    "                          'filtered_n50', 'filtered_Ns', 'type', 'algorythm']\n",
    "METRICS_COLUMNS_REFERENCE = ['run','sample','assembler', 'reference', 'contiguity', 'breadth_of_coverage', 'multiplicity', 'validity', \n",
    "                             'parsimony', 'identity', 'lowest_identity', 'L90', 'contigs', 'NA50', 'NG50', 'basepairs', 'Ns',\n",
    "                             'misassemblies', 'type', 'algorythm']\n",
    "\n",
    "ASSEMBLER_PROCESS_LIST = [\"BCALM2\", \"GATBMINIAPIPELINE\", \"MINIA\", \"MEGAHIT\", \"METASPADES\", \"UNICYCLER\", \"SPADES\",\n",
    "                          \"SKESA\", \"VELVETOPTIMIZER\", \"IDBA\"]\n",
    "PROCESS_TO_NAME = {\"BCALM2\": \"BCALM2\", \n",
    "                   \"GATBMINIAPIPELINE\": \"GATBMiniaPipeline\",\n",
    "                   \"MINIA\": \"MINIA\",\n",
    "                   \"MEGAHIT\": \"MEGAHIT\", \n",
    "                   \"METASPADES\": \"metaSPAdes\", \n",
    "                   \"UNICYCLER\": \"Unicycler\", \n",
    "                   \"SPADES\": \"SPAdes\",\n",
    "                    \"SKESA\": \"SKESA\",\n",
    "                   \"VELVETOPTIMIZER\": \"VelvetOptimizer\",\n",
    "                   \"IDBA\": \"IDBA-UD\"}\n",
    "\n",
    "log_distributed = ['LHS', 'LNN', 'ERR2935805']\n",
    "even_distribution = ['EHS', 'LNN', 'ERR2984773']\n",
    "\n",
    "genomic_assemblers = ['BCALM2', 'MINIA', 'SKESA', 'SPAdes', 'Unicycler', 'VelvetOptimizer']\n",
    "metagenomic_assemblers = ['GATBMiniaPipeline', 'IDBA-UD', 'MEGAHIT', 'metaSPAdes']\n",
    "single_kmer = ['BCALM2', 'MINIA']\n",
    "multiple_kmer = ['SKESA', 'SPAdes', 'Unicycler', 'VelvetOptimizer', 'GATBMiniaPipeline', \n",
    "                 'IDBA-UD', 'MEGAHIT', 'metaSPAdes']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Global metrics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pipeline_report_tables.json data from run1...\n",
      "Processing pipeline_report_tables.json data from run3...\n",
      "Processing pipeline_report_tables.json data from run2...\n"
     ]
    }
   ],
   "source": [
    "report_glob = glob.glob('../Results/*/report/pipeline_report_tables.json')\n",
    "global_pipeline_metrics_df = pd.DataFrame(columns=METRICS_COLUMNS_GLOBAL)\n",
    "\n",
    "for pipeline_report_file in report_glob:\n",
    "    report_file_name = pipeline_report_file.split('/')[-1]\n",
    "    stats_run = pipeline_report_file.split('/')[-3]\n",
    "    print('Processing {0} data from {1}...'.format(report_file_name, stats_run))\n",
    "    \n",
    "    with open(pipeline_report_file) as _fh:\n",
    "        json_report = json.load(_fh)\n",
    "        for sample in json_report.keys():\n",
    "            for line in json_report[sample]['GlobalTable']:\n",
    "                assembler = line['assembler']\n",
    "                global_pipeline_metrics_df = global_pipeline_metrics_df.append({'run': stats_run,\n",
    "                                                                                'sample': sample,\n",
    "                                                                                'assembler': line['assembler'],\n",
    "                                                                                'contigs': int(line['original']['contigs']),\n",
    "                                                                                'basepairs': int(line['original']['basepairs']),\n",
    "                                                                                'max_contig': int(line['original']['max_contig_size']),\n",
    "                                                                                'n50': int(line['original']['N50']),\n",
    "                                                                                'mapped_reads': line['original']['mapped_reads'],\n",
    "                                                                                'Ns': int(line['original']['Ns']),\n",
    "                                                                                'misassemblies': line['filtered']['misassembled_contigs'],\n",
    "                                                                                'filtered_contigs': line['filtered']['contigs'],\n",
    "                                                                                'filtered_basepairs': line['filtered']['basepairs'],\n",
    "                                                                                'filtered_n50': line['filtered']['N50'],\n",
    "                                                                                'filtered_Ns': line['filtered']['Ns']\n",
    "                                                                               },\n",
    "                                                                               ignore_index=True)\n",
    "global_pipeline_metrics_df['distribution'] = np.where(global_pipeline_metrics_df['sample'].isin(log_distributed), 'Log', 'Even')\n",
    "global_pipeline_metrics_df['type'] = np.where(global_pipeline_metrics_df['assembler'].isin(genomic_assemblers), 'Genomic', 'Metagenomic')\n",
    "global_pipeline_metrics_df['algorythm'] = np.where(global_pipeline_metrics_df['assembler'].isin(single_kmer), 'Single k-mer De Bruijn graph', 'Multiple k-mer De Bruijn graph')\n",
    "global_pipeline_metrics_df[['contigs','basepairs','max_contig', 'Ns','n50','misassemblies']] = global_pipeline_metrics_df[['contigs','basepairs','max_contig', 'Ns','n50','misassemblies']].apply(pd.to_numeric)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Preview data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "run                    object\n",
       "sample                 object\n",
       "assembler              object\n",
       "contigs                 int64\n",
       "basepairs               int64\n",
       "max_contig              int64\n",
       "Ns                      int64\n",
       "n50                     int64\n",
       "misassemblies           int64\n",
       "mapped_reads          float64\n",
       "filtered_contigs       object\n",
       "filtered_basepairs     object\n",
       "filtered_n50           object\n",
       "filtered_Ns            object\n",
       "type                   object\n",
       "algorythm              object\n",
       "distribution           object\n",
       "dtype: object"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "global_pipeline_metrics_df.dtypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>run</th>\n",
       "      <th>sample</th>\n",
       "      <th>assembler</th>\n",
       "      <th>contigs</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>max_contig</th>\n",
       "      <th>Ns</th>\n",
       "      <th>n50</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>mapped_reads</th>\n",
       "      <th>filtered_contigs</th>\n",
       "      <th>filtered_basepairs</th>\n",
       "      <th>filtered_n50</th>\n",
       "      <th>filtered_Ns</th>\n",
       "      <th>type</th>\n",
       "      <th>algorythm</th>\n",
       "      <th>distribution</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>run1</td>\n",
       "      <td>ENN</td>\n",
       "      <td>BCALM2</td>\n",
       "      <td>25258</td>\n",
       "      <td>31040696</td>\n",
       "      <td>161812</td>\n",
       "      <td>0</td>\n",
       "      <td>17766</td>\n",
       "      <td>0</td>\n",
       "      <td>96.832853</td>\n",
       "      <td>3542</td>\n",
       "      <td>28188554</td>\n",
       "      <td>21029</td>\n",
       "      <td>0</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>Even</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>run1</td>\n",
       "      <td>ENN</td>\n",
       "      <td>GATBMiniaPipeline</td>\n",
       "      <td>1008</td>\n",
       "      <td>30445418</td>\n",
       "      <td>632215</td>\n",
       "      <td>0</td>\n",
       "      <td>106963</td>\n",
       "      <td>0</td>\n",
       "      <td>99.480322</td>\n",
       "      <td>727</td>\n",
       "      <td>30282639</td>\n",
       "      <td>106963</td>\n",
       "      <td>0</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>Even</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>run1</td>\n",
       "      <td>ENN</td>\n",
       "      <td>IDBA-UD</td>\n",
       "      <td>1825</td>\n",
       "      <td>30390287</td>\n",
       "      <td>468605</td>\n",
       "      <td>0</td>\n",
       "      <td>62331</td>\n",
       "      <td>13</td>\n",
       "      <td>99.414101</td>\n",
       "      <td>1279</td>\n",
       "      <td>30136999</td>\n",
       "      <td>63253</td>\n",
       "      <td>0</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>Even</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>run1</td>\n",
       "      <td>ENN</td>\n",
       "      <td>MEGAHIT</td>\n",
       "      <td>810</td>\n",
       "      <td>30541296</td>\n",
       "      <td>632274</td>\n",
       "      <td>0</td>\n",
       "      <td>170162</td>\n",
       "      <td>9</td>\n",
       "      <td>99.731491</td>\n",
       "      <td>517</td>\n",
       "      <td>30415305</td>\n",
       "      <td>171359</td>\n",
       "      <td>0</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>Even</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>run1</td>\n",
       "      <td>ENN</td>\n",
       "      <td>metaSPAdes</td>\n",
       "      <td>633</td>\n",
       "      <td>30448137</td>\n",
       "      <td>1036942</td>\n",
       "      <td>0</td>\n",
       "      <td>234919</td>\n",
       "      <td>1</td>\n",
       "      <td>99.596146</td>\n",
       "      <td>350</td>\n",
       "      <td>30376419</td>\n",
       "      <td>234919</td>\n",
       "      <td>0</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>Even</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>175</th>\n",
       "      <td>run2</td>\n",
       "      <td>LHS</td>\n",
       "      <td>MINIA</td>\n",
       "      <td>24662</td>\n",
       "      <td>15334974</td>\n",
       "      <td>163543</td>\n",
       "      <td>0</td>\n",
       "      <td>7727</td>\n",
       "      <td>0</td>\n",
       "      <td>91.098929</td>\n",
       "      <td>2172</td>\n",
       "      <td>12922094</td>\n",
       "      <td>10582</td>\n",
       "      <td>0</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>Log</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>176</th>\n",
       "      <td>run2</td>\n",
       "      <td>LHS</td>\n",
       "      <td>SKESA</td>\n",
       "      <td>36</td>\n",
       "      <td>2957060</td>\n",
       "      <td>632013</td>\n",
       "      <td>0</td>\n",
       "      <td>351027</td>\n",
       "      <td>0</td>\n",
       "      <td>92.228441</td>\n",
       "      <td>21</td>\n",
       "      <td>2950393</td>\n",
       "      <td>351027</td>\n",
       "      <td>0</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>Log</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>177</th>\n",
       "      <td>run2</td>\n",
       "      <td>LHS</td>\n",
       "      <td>SPAdes</td>\n",
       "      <td>31225</td>\n",
       "      <td>16224766</td>\n",
       "      <td>263645</td>\n",
       "      <td>0</td>\n",
       "      <td>46478</td>\n",
       "      <td>1</td>\n",
       "      <td>95.503607</td>\n",
       "      <td>1526</td>\n",
       "      <td>13088051</td>\n",
       "      <td>69361</td>\n",
       "      <td>0</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>Log</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>178</th>\n",
       "      <td>run2</td>\n",
       "      <td>LHS</td>\n",
       "      <td>Unicycler</td>\n",
       "      <td>276</td>\n",
       "      <td>13345058</td>\n",
       "      <td>632062</td>\n",
       "      <td>0</td>\n",
       "      <td>187479</td>\n",
       "      <td>3</td>\n",
       "      <td>96.948570</td>\n",
       "      <td>247</td>\n",
       "      <td>13333597</td>\n",
       "      <td>187479</td>\n",
       "      <td>0</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>Log</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>179</th>\n",
       "      <td>run2</td>\n",
       "      <td>LHS</td>\n",
       "      <td>VelvetOptimizer</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>Log</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>180 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      run sample          assembler  contigs  basepairs  max_contig  Ns  \\\n",
       "0    run1    ENN             BCALM2    25258   31040696      161812   0   \n",
       "1    run1    ENN  GATBMiniaPipeline     1008   30445418      632215   0   \n",
       "2    run1    ENN            IDBA-UD     1825   30390287      468605   0   \n",
       "3    run1    ENN            MEGAHIT      810   30541296      632274   0   \n",
       "4    run1    ENN         metaSPAdes      633   30448137     1036942   0   \n",
       "..    ...    ...                ...      ...        ...         ...  ..   \n",
       "175  run2    LHS              MINIA    24662   15334974      163543   0   \n",
       "176  run2    LHS              SKESA       36    2957060      632013   0   \n",
       "177  run2    LHS             SPAdes    31225   16224766      263645   0   \n",
       "178  run2    LHS          Unicycler      276   13345058      632062   0   \n",
       "179  run2    LHS    VelvetOptimizer        0          0           0   0   \n",
       "\n",
       "        n50  misassemblies  mapped_reads filtered_contigs filtered_basepairs  \\\n",
       "0     17766              0     96.832853             3542           28188554   \n",
       "1    106963              0     99.480322              727           30282639   \n",
       "2     62331             13     99.414101             1279           30136999   \n",
       "3    170162              9     99.731491              517           30415305   \n",
       "4    234919              1     99.596146              350           30376419   \n",
       "..      ...            ...           ...              ...                ...   \n",
       "175    7727              0     91.098929             2172           12922094   \n",
       "176  351027              0     92.228441               21            2950393   \n",
       "177   46478              1     95.503607             1526           13088051   \n",
       "178  187479              3     96.948570              247           13333597   \n",
       "179       0              0      0.000000                0                  0   \n",
       "\n",
       "    filtered_n50 filtered_Ns         type                       algorythm  \\\n",
       "0          21029           0      Genomic    Single k-mer De Bruijn graph   \n",
       "1         106963           0  Metagenomic  Multiple k-mer De Bruijn graph   \n",
       "2          63253           0  Metagenomic  Multiple k-mer De Bruijn graph   \n",
       "3         171359           0  Metagenomic  Multiple k-mer De Bruijn graph   \n",
       "4         234919           0  Metagenomic  Multiple k-mer De Bruijn graph   \n",
       "..           ...         ...          ...                             ...   \n",
       "175        10582           0      Genomic    Single k-mer De Bruijn graph   \n",
       "176       351027           0      Genomic  Multiple k-mer De Bruijn graph   \n",
       "177        69361           0      Genomic  Multiple k-mer De Bruijn graph   \n",
       "178       187479           0      Genomic  Multiple k-mer De Bruijn graph   \n",
       "179            0           0      Genomic  Multiple k-mer De Bruijn graph   \n",
       "\n",
       "    distribution  \n",
       "0           Even  \n",
       "1           Even  \n",
       "2           Even  \n",
       "3           Even  \n",
       "4           Even  \n",
       "..           ...  \n",
       "175          Log  \n",
       "176          Log  \n",
       "177          Log  \n",
       "178          Log  \n",
       "179          Log  \n",
       "\n",
       "[180 rows x 17 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "global_pipeline_metrics_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Global statistics per assembler type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "stats_per_assembler_type = pd.DataFrame()\n",
    "for assembly_type in global_pipeline_metrics_df.type.unique():\n",
    "    row={'Assembler Type': assembly_type}\n",
    "    lala=global_pipeline_metrics_df[global_pipeline_metrics_df.type == assembly_type].describe()\n",
    "    for column in lala.columns:\n",
    "        mean = lala.loc['mean',column]\n",
    "        std = lala.loc['std',column]\n",
    "        row[column] = \"{}(±{})\".format(round(mean, 2), round(std, 2))\n",
    "    stats_per_assembler_type=stats_per_assembler_type.append(row, ignore_index=True)\n",
    "stats_per_assembler_type = stats_per_assembler_type.set_index('Assembler Type')\n",
    "stats_per_assembler_type.to_csv(\"Tables/Results/Global metrics per assembler type.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ns</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>contigs</th>\n",
       "      <th>mapped_reads</th>\n",
       "      <th>max_contig</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>n50</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler Type</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Genomic</th>\n",
       "      <td>632.23(±2690.54)</td>\n",
       "      <td>45034003.1(±77233098.8)</td>\n",
       "      <td>584983.59(±1922998.87)</td>\n",
       "      <td>74.63(±36.34)</td>\n",
       "      <td>407391.44(±369087.17)</td>\n",
       "      <td>18.17(±94.33)</td>\n",
       "      <td>75438.73(±96063.55)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Metagenomic</th>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>25546894.07(±9976656.57)</td>\n",
       "      <td>8085.12(±13481.19)</td>\n",
       "      <td>94.46(±6.08)</td>\n",
       "      <td>706300.71(±215974.24)</td>\n",
       "      <td>4.06(±4.63)</td>\n",
       "      <td>116904.35(±64562.91)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              Ns                 basepairs  \\\n",
       "Assembler Type                                               \n",
       "Genomic         632.23(±2690.54)   45034003.1(±77233098.8)   \n",
       "Metagenomic            0.0(±0.0)  25546894.07(±9976656.57)   \n",
       "\n",
       "                               contigs   mapped_reads             max_contig  \\\n",
       "Assembler Type                                                                 \n",
       "Genomic         584983.59(±1922998.87)  74.63(±36.34)  407391.44(±369087.17)   \n",
       "Metagenomic         8085.12(±13481.19)   94.46(±6.08)  706300.71(±215974.24)   \n",
       "\n",
       "                misassemblies                   n50  \n",
       "Assembler Type                                       \n",
       "Genomic         18.17(±94.33)   75438.73(±96063.55)  \n",
       "Metagenomic       4.06(±4.63)  116904.35(±64562.91)  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "stats_per_assembler_type"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Global stats per assembly algorithm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "stats_per_assembler_algo = pd.DataFrame()\n",
    "for assembly_algo in global_pipeline_metrics_df.algorythm.unique():\n",
    "    row={'Assembler Algorithm': assembly_algo}\n",
    "    lala=global_pipeline_metrics_df[global_pipeline_metrics_df.algorythm == assembly_algo].describe()\n",
    "    for column in lala.columns:\n",
    "        mean = lala.loc['mean',column]\n",
    "        std = lala.loc['std',column]\n",
    "        row[column] = \"{}(±{})\".format(round(mean, 2), round(std, 2))\n",
    "    stats_per_assembler_algo=stats_per_assembler_algo.append(row, ignore_index=True)\n",
    "stats_per_assembler_algo = stats_per_assembler_algo.set_index('Assembler Algorithm')\n",
    "stats_per_assembler_algo.to_csv(\"Tables/Results/Global metrics per assembler algorithm.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ns</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>contigs</th>\n",
       "      <th>mapped_reads</th>\n",
       "      <th>max_contig</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>n50</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler Algorithm</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Single k-mer De Bruijn graph</th>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>91615770.47(±120234711.8)</td>\n",
       "      <td>1691066.5(±3063933.44)</td>\n",
       "      <td>68.91(±38.27)</td>\n",
       "      <td>108171.83(±74130.16)</td>\n",
       "      <td>0.33(±0.76)</td>\n",
       "      <td>10205.89(±8991.9)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Multiple k-mer De Bruijn graph</th>\n",
       "      <td>474.17(±2343.51)</td>\n",
       "      <td>23645006.74(±12596918.3)</td>\n",
       "      <td>20013.63(±86808.16)</td>\n",
       "      <td>85.97(±26.61)</td>\n",
       "      <td>631650.97(±309107.29)</td>\n",
       "      <td>15.57(±81.74)</td>\n",
       "      <td>112479.75(±85835.01)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              Ns                  basepairs  \\\n",
       "Assembler Algorithm                                                           \n",
       "Single k-mer De Bruijn graph           0.0(±0.0)  91615770.47(±120234711.8)   \n",
       "Multiple k-mer De Bruijn graph  474.17(±2343.51)   23645006.74(±12596918.3)   \n",
       "\n",
       "                                               contigs   mapped_reads  \\\n",
       "Assembler Algorithm                                                     \n",
       "Single k-mer De Bruijn graph    1691066.5(±3063933.44)  68.91(±38.27)   \n",
       "Multiple k-mer De Bruijn graph     20013.63(±86808.16)  85.97(±26.61)   \n",
       "\n",
       "                                           max_contig  misassemblies  \\\n",
       "Assembler Algorithm                                                    \n",
       "Single k-mer De Bruijn graph     108171.83(±74130.16)    0.33(±0.76)   \n",
       "Multiple k-mer De Bruijn graph  631650.97(±309107.29)  15.57(±81.74)   \n",
       "\n",
       "                                                 n50  \n",
       "Assembler Algorithm                                   \n",
       "Single k-mer De Bruijn graph       10205.89(±8991.9)  \n",
       "Multiple k-mer De Bruijn graph  112479.75(±85835.01)  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "stats_per_assembler_algo"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Global stats per assembler"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "stats_per_assembler = pd.DataFrame()\n",
    "for assembler in global_pipeline_metrics_df.assembler.unique():\n",
    "    row={'Assembler': assembler, \n",
    "         'Type':global_pipeline_metrics_df[global_pipeline_metrics_df.assembler == assembler].type.unique()[0],\n",
    "         'Algorithm': global_pipeline_metrics_df[global_pipeline_metrics_df.assembler == assembler].algorythm.unique()[0]}\n",
    "    lala=global_pipeline_metrics_df[global_pipeline_metrics_df.assembler == assembler].describe()\n",
    "    for column in lala.columns:\n",
    "        mean = lala.loc['mean',column]\n",
    "        std = lala.loc['std',column]\n",
    "        row[column] = \"{}(±{})\".format(round(mean, 2), round(std, 2))\n",
    "    stats_per_assembler=stats_per_assembler.append(row, ignore_index=True)\n",
    "stats_per_assembler = stats_per_assembler.set_index('Assembler')\n",
    "stats_per_assembler.to_csv(\"Tables/Results/Global metrics per assembler.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>contigs</th>\n",
       "      <th>mapped_reads</th>\n",
       "      <th>max_contig</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>n50</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>157403460.17(±143207783.36)</td>\n",
       "      <td>3352423.83(±3671823.35)</td>\n",
       "      <td>46.33(±43.54)</td>\n",
       "      <td>58728.17(±75986.47)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>6220.67(±8741.64)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>23073291.67(±8534830.43)</td>\n",
       "      <td>2629.61(±2551.95)</td>\n",
       "      <td>94.65(±6.43)</td>\n",
       "      <td>640091.33(±67932.83)</td>\n",
       "      <td>1.83(±2.09)</td>\n",
       "      <td>102416.83(±26459.85)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>25378796.44(±9555093.25)</td>\n",
       "      <td>6832.39(±7740.59)</td>\n",
       "      <td>93.22(±6.12)</td>\n",
       "      <td>515095.83(±119980.22)</td>\n",
       "      <td>6.22(±4.73)</td>\n",
       "      <td>68440.89(±30988.57)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>25840461.5(±9906281.89)</td>\n",
       "      <td>6528.5(±8141.67)</td>\n",
       "      <td>94.98(±6.1)</td>\n",
       "      <td>847862.5(±225298.59)</td>\n",
       "      <td>6.33(±6.39)</td>\n",
       "      <td>147830.83(±47068.49)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>27895026.67(±11873670.3)</td>\n",
       "      <td>16350.0(±22765.29)</td>\n",
       "      <td>94.97(±6.0)</td>\n",
       "      <td>822153.17(±215081.62)</td>\n",
       "      <td>1.83(±0.92)</td>\n",
       "      <td>148928.83(±93410.42)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>25828080.78(±9458287.51)</td>\n",
       "      <td>29709.17(±19273.53)</td>\n",
       "      <td>91.49(±6.34)</td>\n",
       "      <td>157615.5(±19050.12)</td>\n",
       "      <td>0.67(±0.97)</td>\n",
       "      <td>14191.11(±7511.05)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>18472455.67(±12305308.04)</td>\n",
       "      <td>3671.5(±4029.8)</td>\n",
       "      <td>90.6(±8.05)</td>\n",
       "      <td>390219.83(±260046.23)</td>\n",
       "      <td>3.33(±4.39)</td>\n",
       "      <td>101495.67(±121656.06)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>28138604.33(±10712564.41)</td>\n",
       "      <td>19436.5(±17118.19)</td>\n",
       "      <td>94.21(±6.4)</td>\n",
       "      <td>697457.83(±215190.75)</td>\n",
       "      <td>98.0(±218.61)</td>\n",
       "      <td>107840.17(±84233.07)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>21030768.0(±9764955.84)</td>\n",
       "      <td>531.33(±325.41)</td>\n",
       "      <td>94.07(±6.74)</td>\n",
       "      <td>857592.0(±230735.33)</td>\n",
       "      <td>1.33(±1.41)</td>\n",
       "      <td>212762.33(±21896.83)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>3793.39(±5732.9)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>19330649.67(±21577560.73)</td>\n",
       "      <td>104129.22(±231388.14)</td>\n",
       "      <td>31.06(±44.61)</td>\n",
       "      <td>282735.28(±410950.64)</td>\n",
       "      <td>5.67(±8.27)</td>\n",
       "      <td>10122.44(±14699.0)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm                Ns  \\\n",
       "Assembler                                                             \n",
       "BCALM2               Single k-mer De Bruijn graph         0.0(±0.0)   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph         0.0(±0.0)   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph         0.0(±0.0)   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph         0.0(±0.0)   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph         0.0(±0.0)   \n",
       "MINIA                Single k-mer De Bruijn graph         0.0(±0.0)   \n",
       "SKESA              Multiple k-mer De Bruijn graph         0.0(±0.0)   \n",
       "SPAdes             Multiple k-mer De Bruijn graph         0.0(±0.0)   \n",
       "Unicycler          Multiple k-mer De Bruijn graph         0.0(±0.0)   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph  3793.39(±5732.9)   \n",
       "\n",
       "                          Type                    basepairs  \\\n",
       "Assembler                                                     \n",
       "BCALM2                 Genomic  157403460.17(±143207783.36)   \n",
       "GATBMiniaPipeline  Metagenomic     23073291.67(±8534830.43)   \n",
       "IDBA-UD            Metagenomic     25378796.44(±9555093.25)   \n",
       "MEGAHIT            Metagenomic      25840461.5(±9906281.89)   \n",
       "metaSPAdes         Metagenomic     27895026.67(±11873670.3)   \n",
       "MINIA                  Genomic     25828080.78(±9458287.51)   \n",
       "SKESA                  Genomic    18472455.67(±12305308.04)   \n",
       "SPAdes                 Genomic    28138604.33(±10712564.41)   \n",
       "Unicycler              Genomic      21030768.0(±9764955.84)   \n",
       "VelvetOptimizer        Genomic    19330649.67(±21577560.73)   \n",
       "\n",
       "                                   contigs   mapped_reads  \\\n",
       "Assembler                                                   \n",
       "BCALM2             3352423.83(±3671823.35)  46.33(±43.54)   \n",
       "GATBMiniaPipeline        2629.61(±2551.95)   94.65(±6.43)   \n",
       "IDBA-UD                  6832.39(±7740.59)   93.22(±6.12)   \n",
       "MEGAHIT                   6528.5(±8141.67)    94.98(±6.1)   \n",
       "metaSPAdes              16350.0(±22765.29)    94.97(±6.0)   \n",
       "MINIA                  29709.17(±19273.53)   91.49(±6.34)   \n",
       "SKESA                      3671.5(±4029.8)    90.6(±8.05)   \n",
       "SPAdes                  19436.5(±17118.19)    94.21(±6.4)   \n",
       "Unicycler                  531.33(±325.41)   94.07(±6.74)   \n",
       "VelvetOptimizer      104129.22(±231388.14)  31.06(±44.61)   \n",
       "\n",
       "                              max_contig  misassemblies                    n50  \n",
       "Assembler                                                                       \n",
       "BCALM2               58728.17(±75986.47)      0.0(±0.0)      6220.67(±8741.64)  \n",
       "GATBMiniaPipeline   640091.33(±67932.83)    1.83(±2.09)   102416.83(±26459.85)  \n",
       "IDBA-UD            515095.83(±119980.22)    6.22(±4.73)    68440.89(±30988.57)  \n",
       "MEGAHIT             847862.5(±225298.59)    6.33(±6.39)   147830.83(±47068.49)  \n",
       "metaSPAdes         822153.17(±215081.62)    1.83(±0.92)   148928.83(±93410.42)  \n",
       "MINIA                157615.5(±19050.12)    0.67(±0.97)     14191.11(±7511.05)  \n",
       "SKESA              390219.83(±260046.23)    3.33(±4.39)  101495.67(±121656.06)  \n",
       "SPAdes             697457.83(±215190.75)  98.0(±218.61)   107840.17(±84233.07)  \n",
       "Unicycler           857592.0(±230735.33)    1.33(±1.41)   212762.33(±21896.83)  \n",
       "VelvetOptimizer    282735.28(±410950.64)    5.67(±8.27)     10122.44(±14699.0)  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "stats_per_assembler"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Global stats per distribution"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Even\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>contigs</th>\n",
       "      <th>mapped_reads</th>\n",
       "      <th>max_contig</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>n50</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>81485532.0(±66888830.04)</td>\n",
       "      <td>1286040.67(±1671523.11)</td>\n",
       "      <td>57.29(±39.07)</td>\n",
       "      <td>61507.0(±75761.93)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>6186.0(±8690.2)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>31254751.44(±1167859.11)</td>\n",
       "      <td>2439.78(±2220.06)</td>\n",
       "      <td>93.87(±8.05)</td>\n",
       "      <td>677707.0(±68066.32)</td>\n",
       "      <td>2.67(±2.65)</td>\n",
       "      <td>110537.33(±9239.15)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>33523214.0(±4692045.42)</td>\n",
       "      <td>7943.22(±9186.33)</td>\n",
       "      <td>90.89(±6.89)</td>\n",
       "      <td>413513.0(±82638.0)</td>\n",
       "      <td>9.78(±4.35)</td>\n",
       "      <td>49109.78(±21114.03)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>34099350.67(±5300334.81)</td>\n",
       "      <td>7463.33(±9863.23)</td>\n",
       "      <td>94.35(±7.6)</td>\n",
       "      <td>868834.0(±259534.83)</td>\n",
       "      <td>12.0(±3.77)</td>\n",
       "      <td>143276.33(±41229.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>36624891.67(±9252202.76)</td>\n",
       "      <td>21002.33(±30476.03)</td>\n",
       "      <td>94.28(±7.47)</td>\n",
       "      <td>847661.67(±280794.32)</td>\n",
       "      <td>2.0(±0.87)</td>\n",
       "      <td>158678.0(±88651.34)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>34158216.89(±5270643.61)</td>\n",
       "      <td>33769.89(±25663.37)</td>\n",
       "      <td>90.39(±7.57)</td>\n",
       "      <td>147189.67(±21933.5)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>14482.0(±5545.5)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>30285330.33(±112007.2)</td>\n",
       "      <td>1219.0(±192.79)</td>\n",
       "      <td>92.53(±8.46)</td>\n",
       "      <td>540293.0(±80495.71)</td>\n",
       "      <td>2.67(±2.0)</td>\n",
       "      <td>85120.33(±16388.81)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>35828349.33(±8031371.09)</td>\n",
       "      <td>13949.33(±19394.94)</td>\n",
       "      <td>94.27(±7.53)</td>\n",
       "      <td>813593.33(±51167.5)</td>\n",
       "      <td>4.33(±4.27)</td>\n",
       "      <td>158629.67(±80482.08)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>30462753.0(±96627.2)</td>\n",
       "      <td>833.67(±134.55)</td>\n",
       "      <td>93.41(±8.33)</td>\n",
       "      <td>1011262.0(±219958.36)</td>\n",
       "      <td>1.67(±1.32)</td>\n",
       "      <td>215168.67(±26065.96)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>7586.78(±6120.74)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>38661299.33(±12190759.12)</td>\n",
       "      <td>208258.44(±298959.84)</td>\n",
       "      <td>62.11(±45.36)</td>\n",
       "      <td>565470.56(±423092.93)</td>\n",
       "      <td>11.33(±8.54)</td>\n",
       "      <td>20244.89(±15119.03)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm                 Ns  \\\n",
       "Assembler                                                              \n",
       "BCALM2               Single k-mer De Bruijn graph          0.0(±0.0)   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph          0.0(±0.0)   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph          0.0(±0.0)   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph          0.0(±0.0)   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph          0.0(±0.0)   \n",
       "MINIA                Single k-mer De Bruijn graph          0.0(±0.0)   \n",
       "SKESA              Multiple k-mer De Bruijn graph          0.0(±0.0)   \n",
       "SPAdes             Multiple k-mer De Bruijn graph          0.0(±0.0)   \n",
       "Unicycler          Multiple k-mer De Bruijn graph          0.0(±0.0)   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph  7586.78(±6120.74)   \n",
       "\n",
       "                          Type                  basepairs  \\\n",
       "Assembler                                                   \n",
       "BCALM2                 Genomic   81485532.0(±66888830.04)   \n",
       "GATBMiniaPipeline  Metagenomic   31254751.44(±1167859.11)   \n",
       "IDBA-UD            Metagenomic    33523214.0(±4692045.42)   \n",
       "MEGAHIT            Metagenomic   34099350.67(±5300334.81)   \n",
       "metaSPAdes         Metagenomic   36624891.67(±9252202.76)   \n",
       "MINIA                  Genomic   34158216.89(±5270643.61)   \n",
       "SKESA                  Genomic     30285330.33(±112007.2)   \n",
       "SPAdes                 Genomic   35828349.33(±8031371.09)   \n",
       "Unicycler              Genomic       30462753.0(±96627.2)   \n",
       "VelvetOptimizer        Genomic  38661299.33(±12190759.12)   \n",
       "\n",
       "                                   contigs   mapped_reads  \\\n",
       "Assembler                                                   \n",
       "BCALM2             1286040.67(±1671523.11)  57.29(±39.07)   \n",
       "GATBMiniaPipeline        2439.78(±2220.06)   93.87(±8.05)   \n",
       "IDBA-UD                  7943.22(±9186.33)   90.89(±6.89)   \n",
       "MEGAHIT                  7463.33(±9863.23)    94.35(±7.6)   \n",
       "metaSPAdes             21002.33(±30476.03)   94.28(±7.47)   \n",
       "MINIA                  33769.89(±25663.37)   90.39(±7.57)   \n",
       "SKESA                      1219.0(±192.79)   92.53(±8.46)   \n",
       "SPAdes                 13949.33(±19394.94)   94.27(±7.53)   \n",
       "Unicycler                  833.67(±134.55)   93.41(±8.33)   \n",
       "VelvetOptimizer      208258.44(±298959.84)  62.11(±45.36)   \n",
       "\n",
       "                              max_contig misassemblies                   n50  \n",
       "Assembler                                                                     \n",
       "BCALM2                61507.0(±75761.93)     0.0(±0.0)       6186.0(±8690.2)  \n",
       "GATBMiniaPipeline    677707.0(±68066.32)   2.67(±2.65)   110537.33(±9239.15)  \n",
       "IDBA-UD               413513.0(±82638.0)   9.78(±4.35)   49109.78(±21114.03)  \n",
       "MEGAHIT             868834.0(±259534.83)   12.0(±3.77)  143276.33(±41229.51)  \n",
       "metaSPAdes         847661.67(±280794.32)    2.0(±0.87)   158678.0(±88651.34)  \n",
       "MINIA                147189.67(±21933.5)     0.0(±0.0)      14482.0(±5545.5)  \n",
       "SKESA                540293.0(±80495.71)    2.67(±2.0)   85120.33(±16388.81)  \n",
       "SPAdes               813593.33(±51167.5)   4.33(±4.27)  158629.67(±80482.08)  \n",
       "Unicycler          1011262.0(±219958.36)   1.67(±1.32)  215168.67(±26065.96)  \n",
       "VelvetOptimizer    565470.56(±423092.93)  11.33(±8.54)   20244.89(±15119.03)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>contigs</th>\n",
       "      <th>mapped_reads</th>\n",
       "      <th>max_contig</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>n50</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>233321388.33(±161673853.51)</td>\n",
       "      <td>5418807.0(±4030943.57)</td>\n",
       "      <td>35.37(±47.25)</td>\n",
       "      <td>55949.33(±80699.36)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>6255.33(±9320.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>14891831.89(±1679744.93)</td>\n",
       "      <td>2819.44(±2971.41)</td>\n",
       "      <td>95.43(±4.65)</td>\n",
       "      <td>602475.67(±44609.0)</td>\n",
       "      <td>1.0(±0.87)</td>\n",
       "      <td>94296.33(±35412.28)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>17234378.89(±4769784.15)</td>\n",
       "      <td>5721.56(±6337.04)</td>\n",
       "      <td>95.56(±4.46)</td>\n",
       "      <td>616678.67(±23301.5)</td>\n",
       "      <td>2.67(±0.5)</td>\n",
       "      <td>87772.0(±27459.24)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>17581572.33(±5193388.33)</td>\n",
       "      <td>5593.67(±6450.57)</td>\n",
       "      <td>95.61(±4.51)</td>\n",
       "      <td>826891.0(±198786.22)</td>\n",
       "      <td>0.67(±0.5)</td>\n",
       "      <td>152385.33(±54417.57)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>19165161.67(±6520285.33)</td>\n",
       "      <td>11697.67(±11127.06)</td>\n",
       "      <td>95.66(±4.44)</td>\n",
       "      <td>796644.67(±134138.83)</td>\n",
       "      <td>1.67(±1.0)</td>\n",
       "      <td>139179.67(±102317.1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>17497944.67(±2488170.91)</td>\n",
       "      <td>25648.44(±9677.9)</td>\n",
       "      <td>92.59(±5.04)</td>\n",
       "      <td>168041.33(±6747.5)</td>\n",
       "      <td>1.33(±1.0)</td>\n",
       "      <td>13900.22(±9430.85)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>6659581.0(±2789663.83)</td>\n",
       "      <td>6124.0(±4575.82)</td>\n",
       "      <td>88.66(±7.6)</td>\n",
       "      <td>240146.67(±294188.94)</td>\n",
       "      <td>4.0(±6.0)</td>\n",
       "      <td>117871.0(±174867.04)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>20448859.33(±6805338.32)</td>\n",
       "      <td>24923.67(±13371.06)</td>\n",
       "      <td>94.15(±5.5)</td>\n",
       "      <td>581322.33(±255807.76)</td>\n",
       "      <td>191.67(±286.0)</td>\n",
       "      <td>57050.67(±52876.44)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>11598783.0(±1566338.53)</td>\n",
       "      <td>229.0(±35.38)</td>\n",
       "      <td>94.74(±5.12)</td>\n",
       "      <td>703922.0(±107787.75)</td>\n",
       "      <td>1.0(±1.5)</td>\n",
       "      <td>210356.0(±18066.91)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm         Ns         Type  \\\n",
       "Assembler                                                                   \n",
       "BCALM2               Single k-mer De Bruijn graph  0.0(±0.0)      Genomic   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph  0.0(±0.0)  Metagenomic   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  0.0(±0.0)  Metagenomic   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph  0.0(±0.0)  Metagenomic   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph  0.0(±0.0)  Metagenomic   \n",
       "MINIA                Single k-mer De Bruijn graph  0.0(±0.0)      Genomic   \n",
       "SKESA              Multiple k-mer De Bruijn graph  0.0(±0.0)      Genomic   \n",
       "SPAdes             Multiple k-mer De Bruijn graph  0.0(±0.0)      Genomic   \n",
       "Unicycler          Multiple k-mer De Bruijn graph  0.0(±0.0)      Genomic   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph  0.0(±0.0)      Genomic   \n",
       "\n",
       "                                     basepairs                 contigs  \\\n",
       "Assembler                                                                \n",
       "BCALM2             233321388.33(±161673853.51)  5418807.0(±4030943.57)   \n",
       "GATBMiniaPipeline     14891831.89(±1679744.93)       2819.44(±2971.41)   \n",
       "IDBA-UD               17234378.89(±4769784.15)       5721.56(±6337.04)   \n",
       "MEGAHIT               17581572.33(±5193388.33)       5593.67(±6450.57)   \n",
       "metaSPAdes            19165161.67(±6520285.33)     11697.67(±11127.06)   \n",
       "MINIA                 17497944.67(±2488170.91)       25648.44(±9677.9)   \n",
       "SKESA                   6659581.0(±2789663.83)        6124.0(±4575.82)   \n",
       "SPAdes                20448859.33(±6805338.32)     24923.67(±13371.06)   \n",
       "Unicycler              11598783.0(±1566338.53)           229.0(±35.38)   \n",
       "VelvetOptimizer                      0.0(±0.0)               0.0(±0.0)   \n",
       "\n",
       "                    mapped_reads             max_contig   misassemblies  \\\n",
       "Assembler                                                                 \n",
       "BCALM2             35.37(±47.25)    55949.33(±80699.36)       0.0(±0.0)   \n",
       "GATBMiniaPipeline   95.43(±4.65)    602475.67(±44609.0)      1.0(±0.87)   \n",
       "IDBA-UD             95.56(±4.46)    616678.67(±23301.5)      2.67(±0.5)   \n",
       "MEGAHIT             95.61(±4.51)   826891.0(±198786.22)      0.67(±0.5)   \n",
       "metaSPAdes          95.66(±4.44)  796644.67(±134138.83)      1.67(±1.0)   \n",
       "MINIA               92.59(±5.04)     168041.33(±6747.5)      1.33(±1.0)   \n",
       "SKESA                88.66(±7.6)  240146.67(±294188.94)       4.0(±6.0)   \n",
       "SPAdes               94.15(±5.5)  581322.33(±255807.76)  191.67(±286.0)   \n",
       "Unicycler           94.74(±5.12)   703922.0(±107787.75)       1.0(±1.5)   \n",
       "VelvetOptimizer        0.0(±0.0)              0.0(±0.0)       0.0(±0.0)   \n",
       "\n",
       "                                    n50  \n",
       "Assembler                                \n",
       "BCALM2                 6255.33(±9320.0)  \n",
       "GATBMiniaPipeline   94296.33(±35412.28)  \n",
       "IDBA-UD              87772.0(±27459.24)  \n",
       "MEGAHIT            152385.33(±54417.57)  \n",
       "metaSPAdes         139179.67(±102317.1)  \n",
       "MINIA                13900.22(±9430.85)  \n",
       "SKESA              117871.0(±174867.04)  \n",
       "SPAdes              57050.67(±52876.44)  \n",
       "Unicycler           210356.0(±18066.91)  \n",
       "VelvetOptimizer               0.0(±0.0)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "for distribution in reference_pipeline_metrics_df.distribution.unique():\n",
    "    stats_per_distribution = pd.DataFrame()\n",
    "    for assembler in global_pipeline_metrics_df.assembler.unique():\n",
    "        row={'Assembler': assembler, \n",
    "             'Type':global_pipeline_metrics_df[(global_pipeline_metrics_df.assembler == assembler) & (global_pipeline_metrics_df.distribution == distribution)].type.unique()[0],\n",
    "             'Algorithm': global_pipeline_metrics_df[global_pipeline_metrics_df.assembler == assembler].algorythm.unique()[0]}\n",
    "        lala=global_pipeline_metrics_df[(global_pipeline_metrics_df.assembler == assembler) & (global_pipeline_metrics_df.distribution == distribution)].describe()\n",
    "        for column in lala.columns:\n",
    "            mean = lala.loc['mean',column]\n",
    "            std = lala.loc['std',column]\n",
    "            row[column] = \"{}(±{})\".format(round(mean, 2), round(std, 2))\n",
    "        stats_per_distribution=stats_per_distribution.append(row, ignore_index=True)\n",
    "    stats_per_distribution = stats_per_distribution.set_index('Assembler')\n",
    "    stats_per_distribution.to_csv(\"Tables/Results/{} - Global metrics per assembler.csv\".format(distribution))\n",
    "    print(distribution)\n",
    "    display(stats_per_distribution)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Global metrics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pipeline_report_tables.json data from run1...\n",
      "Processing pipeline_report_tables.json data from run3...\n",
      "Processing pipeline_report_tables.json data from run2...\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>run</th>\n",
       "      <th>sample</th>\n",
       "      <th>assembler</th>\n",
       "      <th>reference</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>validity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>identity</th>\n",
       "      <th>...</th>\n",
       "      <th>L90</th>\n",
       "      <th>contigs</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>Ns</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>type</th>\n",
       "      <th>algorythm</th>\n",
       "      <th>distribution</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>run1</td>\n",
       "      <td>ENN</td>\n",
       "      <td>BCALM2</td>\n",
       "      <td>Bacillus_subtilis</td>\n",
       "      <td>0.032648</td>\n",
       "      <td>0.976835</td>\n",
       "      <td>1.000279</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000279</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>66</td>\n",
       "      <td>116</td>\n",
       "      <td>65138</td>\n",
       "      <td>64934</td>\n",
       "      <td>3951958</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>Even</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>run1</td>\n",
       "      <td>ENN</td>\n",
       "      <td>GATBMiniaPipeline</td>\n",
       "      <td>Bacillus_subtilis</td>\n",
       "      <td>0.097833</td>\n",
       "      <td>0.983819</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.999966</td>\n",
       "      <td>...</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>269695</td>\n",
       "      <td>263366</td>\n",
       "      <td>3980213</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>Even</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>run1</td>\n",
       "      <td>ENN</td>\n",
       "      <td>IDBA-UD</td>\n",
       "      <td>Bacillus_subtilis</td>\n",
       "      <td>0.077894</td>\n",
       "      <td>0.982940</td>\n",
       "      <td>1.000048</td>\n",
       "      <td>0.999999</td>\n",
       "      <td>1.000048</td>\n",
       "      <td>0.999558</td>\n",
       "      <td>...</td>\n",
       "      <td>21</td>\n",
       "      <td>38</td>\n",
       "      <td>211169</td>\n",
       "      <td>211169</td>\n",
       "      <td>3976658</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>Even</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>run1</td>\n",
       "      <td>ENN</td>\n",
       "      <td>MEGAHIT</td>\n",
       "      <td>Bacillus_subtilis</td>\n",
       "      <td>0.097848</td>\n",
       "      <td>0.985245</td>\n",
       "      <td>1.000233</td>\n",
       "      <td>0.999861</td>\n",
       "      <td>1.000372</td>\n",
       "      <td>0.998712</td>\n",
       "      <td>...</td>\n",
       "      <td>14</td>\n",
       "      <td>28</td>\n",
       "      <td>263587</td>\n",
       "      <td>263587</td>\n",
       "      <td>3985982</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>Even</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>run1</td>\n",
       "      <td>ENN</td>\n",
       "      <td>metaSPAdes</td>\n",
       "      <td>Bacillus_subtilis</td>\n",
       "      <td>0.255293</td>\n",
       "      <td>0.985404</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.998464</td>\n",
       "      <td>...</td>\n",
       "      <td>10</td>\n",
       "      <td>20</td>\n",
       "      <td>325068</td>\n",
       "      <td>325068</td>\n",
       "      <td>3986625</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>Even</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2155</th>\n",
       "      <td>run2</td>\n",
       "      <td>LHS</td>\n",
       "      <td>MINIA</td>\n",
       "      <td>Staphylococcus_aureus_plasmid3</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>Log</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2156</th>\n",
       "      <td>run2</td>\n",
       "      <td>LHS</td>\n",
       "      <td>SKESA</td>\n",
       "      <td>Staphylococcus_aureus_plasmid3</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>Log</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2157</th>\n",
       "      <td>run2</td>\n",
       "      <td>LHS</td>\n",
       "      <td>SPAdes</td>\n",
       "      <td>Staphylococcus_aureus_plasmid3</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>Log</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2158</th>\n",
       "      <td>run2</td>\n",
       "      <td>LHS</td>\n",
       "      <td>Unicycler</td>\n",
       "      <td>Staphylococcus_aureus_plasmid3</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>Log</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2159</th>\n",
       "      <td>run2</td>\n",
       "      <td>LHS</td>\n",
       "      <td>VelvetOptimizer</td>\n",
       "      <td>Staphylococcus_aureus_plasmid3</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>Log</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2160 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       run sample          assembler                       reference  \\\n",
       "0     run1    ENN             BCALM2               Bacillus_subtilis   \n",
       "1     run1    ENN  GATBMiniaPipeline               Bacillus_subtilis   \n",
       "2     run1    ENN            IDBA-UD               Bacillus_subtilis   \n",
       "3     run1    ENN            MEGAHIT               Bacillus_subtilis   \n",
       "4     run1    ENN         metaSPAdes               Bacillus_subtilis   \n",
       "...    ...    ...                ...                             ...   \n",
       "2155  run2    LHS              MINIA  Staphylococcus_aureus_plasmid3   \n",
       "2156  run2    LHS              SKESA  Staphylococcus_aureus_plasmid3   \n",
       "2157  run2    LHS             SPAdes  Staphylococcus_aureus_plasmid3   \n",
       "2158  run2    LHS          Unicycler  Staphylococcus_aureus_plasmid3   \n",
       "2159  run2    LHS    VelvetOptimizer  Staphylococcus_aureus_plasmid3   \n",
       "\n",
       "      contiguity  breadth_of_coverage  multiplicity  validity  parsimony  \\\n",
       "0       0.032648             0.976835      1.000279  1.000000   1.000279   \n",
       "1       0.097833             0.983819      1.000000  1.000000   1.000000   \n",
       "2       0.077894             0.982940      1.000048  0.999999   1.000048   \n",
       "3       0.097848             0.985245      1.000233  0.999861   1.000372   \n",
       "4       0.255293             0.985404      1.000000  1.000000   1.000000   \n",
       "...          ...                  ...           ...       ...        ...   \n",
       "2155    0.000000             0.000000      0.000000  0.000000   0.000000   \n",
       "2156    0.000000             0.000000      0.000000  0.000000   0.000000   \n",
       "2157    0.000000             0.000000      0.000000  0.000000   0.000000   \n",
       "2158    0.000000             0.000000      0.000000  0.000000   0.000000   \n",
       "2159    0.000000             0.000000      0.000000  0.000000   0.000000   \n",
       "\n",
       "      identity  ...  L90  contigs    NA50    NG50  basepairs  Ns  \\\n",
       "0     1.000000  ...   66      116   65138   64934    3951958   0   \n",
       "1     0.999966  ...   14       29  269695  263366    3980213   0   \n",
       "2     0.999558  ...   21       38  211169  211169    3976658   0   \n",
       "3     0.998712  ...   14       28  263587  263587    3985982   0   \n",
       "4     0.998464  ...   10       20  325068  325068    3986625   0   \n",
       "...        ...  ...  ...      ...     ...     ...        ...  ..   \n",
       "2155  0.000000  ...    0        0       0       0          0   0   \n",
       "2156  0.000000  ...    0        0       0       0          0   0   \n",
       "2157  0.000000  ...    0        0       0       0          0   0   \n",
       "2158  0.000000  ...    0        0       0       0          0   0   \n",
       "2159  0.000000  ...    0        0       0       0          0   0   \n",
       "\n",
       "      misassemblies         type                       algorythm distribution  \n",
       "0                 0      Genomic    Single k-mer De Bruijn graph         Even  \n",
       "1                 0  Metagenomic  Multiple k-mer De Bruijn graph         Even  \n",
       "2                 0  Metagenomic  Multiple k-mer De Bruijn graph         Even  \n",
       "3                 0  Metagenomic  Multiple k-mer De Bruijn graph         Even  \n",
       "4                 0  Metagenomic  Multiple k-mer De Bruijn graph         Even  \n",
       "...             ...          ...                             ...          ...  \n",
       "2155              0      Genomic    Single k-mer De Bruijn graph          Log  \n",
       "2156              0      Genomic  Multiple k-mer De Bruijn graph          Log  \n",
       "2157              0      Genomic  Multiple k-mer De Bruijn graph          Log  \n",
       "2158              0      Genomic  Multiple k-mer De Bruijn graph          Log  \n",
       "2159              0      Genomic  Multiple k-mer De Bruijn graph          Log  \n",
       "\n",
       "[2160 rows x 21 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "report_glob = glob.glob('../Results/*/report/pipeline_report_tables.json')\n",
    "reference_pipeline_metrics_df = pd.DataFrame(columns=METRICS_COLUMNS_REFERENCE)\n",
    "\n",
    "for pipeline_report_file in report_glob:\n",
    "    report_file_name = pipeline_report_file.split('/')[-1]\n",
    "    stats_run = pipeline_report_file.split('/')[-3]\n",
    "    print('Processing {0} data from {1}...'.format(report_file_name, stats_run))\n",
    "    \n",
    "    with open(pipeline_report_file) as _fh:\n",
    "        json_report = json.load(_fh)\n",
    "        for sample in json_report.keys():\n",
    "            for reference, data in json_report[sample]['ReferenceTables'].items():\n",
    "                for row in data:\n",
    "                    for item in row:\n",
    "                        reference_pipeline_metrics_df = reference_pipeline_metrics_df.append({'run': stats_run,\n",
    "                                                                                        'sample': sample,\n",
    "                                                                                        'assembler': item['assembler'],\n",
    "                                                                                        'reference': reference,\n",
    "                                                                                        'contiguity': item['contiguity'],\n",
    "                                                                                        'breadth_of_coverage': item['breadth_of_coverage'],\n",
    "                                                                                        'multiplicity': item['multiplicity'],\n",
    "                                                                                        'validity': item['validity'],\n",
    "                                                                                        'parsimony': item['parsimony'],\n",
    "                                                                                        'identity': item['identity'],\n",
    "                                                                                        'lowest_identity': item['lowest_identity'],\n",
    "                                                                                        'L90': item['L90'],\n",
    "                                                                                        'contigs': item['aligned_contigs'],\n",
    "                                                                                        'NA50': item['NA50'],\n",
    "                                                                                        'NG50': item['NG50'],\n",
    "                                                                                        'basepairs': item['aligned_basepairs'],\n",
    "                                                                                        'Ns': item['Ns'],\n",
    "                                                                                        'misassemblies': item['misassembled_contigs']},\n",
    "                                                                                       ignore_index=True)\n",
    "\n",
    "reference_pipeline_metrics_df['distribution'] = np.where(reference_pipeline_metrics_df['sample'].isin(log_distributed), 'Log', 'Even')\n",
    "reference_pipeline_metrics_df['type'] = np.where(reference_pipeline_metrics_df['assembler'].isin(genomic_assemblers), 'Genomic', 'Metagenomic')\n",
    "reference_pipeline_metrics_df['algorythm'] = np.where(reference_pipeline_metrics_df['assembler'].isin(single_kmer), 'Single k-mer De Bruijn graph', 'Multiple k-mer De Bruijn graph')\n",
    "reference_pipeline_metrics_df[['contigs','basepairs','L90','Ns','NA50','NG50','misassemblies']] = reference_pipeline_metrics_df[['contigs','basepairs','L90','Ns','NA50','NG50','misassemblies']].apply(pd.to_numeric)\n",
    "display(reference_pipeline_metrics_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "run                     object\n",
       "sample                  object\n",
       "assembler               object\n",
       "reference               object\n",
       "contiguity             float64\n",
       "breadth_of_coverage    float64\n",
       "multiplicity           float64\n",
       "validity               float64\n",
       "parsimony              float64\n",
       "identity               float64\n",
       "lowest_identity        float64\n",
       "L90                      int64\n",
       "contigs                  int64\n",
       "NA50                     int64\n",
       "NG50                     int64\n",
       "basepairs                int64\n",
       "Ns                       int64\n",
       "misassemblies            int64\n",
       "type                    object\n",
       "algorythm               object\n",
       "distribution            object\n",
       "dtype: object"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reference_pipeline_metrics_df.dtypes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reference statistics per assembler type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_stats_per_assembler_type = pd.DataFrame()\n",
    "for assembly_type in reference_pipeline_metrics_df.type.unique():\n",
    "    row={'Assembler Type': assembly_type}\n",
    "    lala=reference_pipeline_metrics_df[reference_pipeline_metrics_df.type == assembly_type].describe()\n",
    "    for column in lala.columns:\n",
    "        mean = lala.loc['mean',column]\n",
    "        std = lala.loc['std',column]\n",
    "        row[column] = \"{}(±{})\".format(round(mean, 2), round(std, 2))\n",
    "    ref_stats_per_assembler_type=ref_stats_per_assembler_type.append(row, ignore_index=True)\n",
    "ref_stats_per_assembler_type = ref_stats_per_assembler_type.set_index('Assembler Type')\n",
    "ref_stats_per_assembler_type.to_csv(\"Tables/Results/Reference metrics per assembler type.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>L90</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler Type</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Genomic</th>\n",
       "      <td>32.26(±116.1)</td>\n",
       "      <td>44530.55(±91863.12)</td>\n",
       "      <td>44301.7(±93231.26)</td>\n",
       "      <td>81.46(±800.9)</td>\n",
       "      <td>1392592.69(±2023387.45)</td>\n",
       "      <td>0.42(±0.47)</td>\n",
       "      <td>113.05(±296.65)</td>\n",
       "      <td>0.08(±0.2)</td>\n",
       "      <td>0.5(±0.5)</td>\n",
       "      <td>0.47(±0.49)</td>\n",
       "      <td>1.55(±27.22)</td>\n",
       "      <td>0.5(±0.5)</td>\n",
       "      <td>0.51(±0.51)</td>\n",
       "      <td>0.5(±0.5)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Metagenomic</th>\n",
       "      <td>24.21(±51.75)</td>\n",
       "      <td>82729.11(±123470.96)</td>\n",
       "      <td>81991.75(±122491.49)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1830607.91(±2235695.27)</td>\n",
       "      <td>0.55(±0.49)</td>\n",
       "      <td>45.2(±82.43)</td>\n",
       "      <td>0.14(±0.26)</td>\n",
       "      <td>0.66(±0.47)</td>\n",
       "      <td>0.61(±0.46)</td>\n",
       "      <td>0.39(±1.08)</td>\n",
       "      <td>0.67(±0.48)</td>\n",
       "      <td>0.69(±0.56)</td>\n",
       "      <td>0.66(±0.47)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                          L90                  NA50                  NG50  \\\n",
       "Assembler Type                                                              \n",
       "Genomic         32.26(±116.1)   44530.55(±91863.12)    44301.7(±93231.26)   \n",
       "Metagenomic     24.21(±51.75)  82729.11(±123470.96)  81991.75(±122491.49)   \n",
       "\n",
       "                           Ns                basepairs breadth_of_coverage  \\\n",
       "Assembler Type                                                               \n",
       "Genomic         81.46(±800.9)  1392592.69(±2023387.45)         0.42(±0.47)   \n",
       "Metagenomic         0.0(±0.0)  1830607.91(±2235695.27)         0.55(±0.49)   \n",
       "\n",
       "                        contigs   contiguity     identity lowest_identity  \\\n",
       "Assembler Type                                                              \n",
       "Genomic         113.05(±296.65)   0.08(±0.2)    0.5(±0.5)     0.47(±0.49)   \n",
       "Metagenomic        45.2(±82.43)  0.14(±0.26)  0.66(±0.47)     0.61(±0.46)   \n",
       "\n",
       "               misassemblies multiplicity    parsimony     validity  \n",
       "Assembler Type                                                       \n",
       "Genomic         1.55(±27.22)    0.5(±0.5)  0.51(±0.51)    0.5(±0.5)  \n",
       "Metagenomic      0.39(±1.08)  0.67(±0.48)  0.69(±0.56)  0.66(±0.47)  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ref_stats_per_assembler_type"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reference stats per assembly algorithm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_stats_per_assembler_algo = pd.DataFrame()\n",
    "for assembly_algo in reference_pipeline_metrics_df.algorythm.unique():\n",
    "    row={'Assembler Algorithm': assembly_algo}\n",
    "    lala=reference_pipeline_metrics_df[reference_pipeline_metrics_df.algorythm == assembly_algo].describe()\n",
    "    for column in lala.columns:\n",
    "        mean = lala.loc['mean',column]\n",
    "        std = lala.loc['std',column]\n",
    "        row[column] = \"{}(±{})\".format(round(mean, 2), round(std, 2))\n",
    "    ref_stats_per_assembler_algo=ref_stats_per_assembler_algo.append(row, ignore_index=True)\n",
    "ref_stats_per_assembler_algo = ref_stats_per_assembler_algo.set_index('Assembler Algorithm')\n",
    "ref_stats_per_assembler_algo.to_csv(\"Tables/Results/Reference metrics per assembler algorithm.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>L90</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler Algorithm</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Single k-mer De Bruijn graph</th>\n",
       "      <td>59.16(±179.99)</td>\n",
       "      <td>8841.57(±15783.42)</td>\n",
       "      <td>8407.69(±15608.21)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1281530.89(±1943438.06)</td>\n",
       "      <td>0.35(±0.44)</td>\n",
       "      <td>200.97(±381.52)</td>\n",
       "      <td>0.03(±0.13)</td>\n",
       "      <td>0.47(±0.5)</td>\n",
       "      <td>0.46(±0.5)</td>\n",
       "      <td>0.03(±0.25)</td>\n",
       "      <td>0.47(±0.5)</td>\n",
       "      <td>0.47(±0.5)</td>\n",
       "      <td>0.47(±0.5)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Multiple k-mer De Bruijn graph</th>\n",
       "      <td>21.51(±55.65)</td>\n",
       "      <td>72552.08(±116230.69)</td>\n",
       "      <td>72120.23(±116485.25)</td>\n",
       "      <td>61.1(±694.43)</td>\n",
       "      <td>1639365.75(±2158024.18)</td>\n",
       "      <td>0.5(±0.49)</td>\n",
       "      <td>57.15(±174.07)</td>\n",
       "      <td>0.12(±0.25)</td>\n",
       "      <td>0.59(±0.49)</td>\n",
       "      <td>0.54(±0.48)</td>\n",
       "      <td>1.34(±23.58)</td>\n",
       "      <td>0.59(±0.49)</td>\n",
       "      <td>0.6(±0.54)</td>\n",
       "      <td>0.59(±0.49)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           L90                  NA50  \\\n",
       "Assembler Algorithm                                                    \n",
       "Single k-mer De Bruijn graph    59.16(±179.99)    8841.57(±15783.42)   \n",
       "Multiple k-mer De Bruijn graph   21.51(±55.65)  72552.08(±116230.69)   \n",
       "\n",
       "                                                NG50             Ns  \\\n",
       "Assembler Algorithm                                                   \n",
       "Single k-mer De Bruijn graph      8407.69(±15608.21)      0.0(±0.0)   \n",
       "Multiple k-mer De Bruijn graph  72120.23(±116485.25)  61.1(±694.43)   \n",
       "\n",
       "                                              basepairs breadth_of_coverage  \\\n",
       "Assembler Algorithm                                                           \n",
       "Single k-mer De Bruijn graph    1281530.89(±1943438.06)         0.35(±0.44)   \n",
       "Multiple k-mer De Bruijn graph  1639365.75(±2158024.18)          0.5(±0.49)   \n",
       "\n",
       "                                        contigs   contiguity     identity  \\\n",
       "Assembler Algorithm                                                         \n",
       "Single k-mer De Bruijn graph    200.97(±381.52)  0.03(±0.13)   0.47(±0.5)   \n",
       "Multiple k-mer De Bruijn graph   57.15(±174.07)  0.12(±0.25)  0.59(±0.49)   \n",
       "\n",
       "                               lowest_identity misassemblies multiplicity  \\\n",
       "Assembler Algorithm                                                         \n",
       "Single k-mer De Bruijn graph        0.46(±0.5)   0.03(±0.25)   0.47(±0.5)   \n",
       "Multiple k-mer De Bruijn graph     0.54(±0.48)  1.34(±23.58)  0.59(±0.49)   \n",
       "\n",
       "                                 parsimony     validity  \n",
       "Assembler Algorithm                                      \n",
       "Single k-mer De Bruijn graph    0.47(±0.5)   0.47(±0.5)  \n",
       "Multiple k-mer De Bruijn graph  0.6(±0.54)  0.59(±0.49)  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ref_stats_per_assembler_algo"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reference stats per assembler"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_stats_per_assembler = pd.DataFrame()\n",
    "for assembler in reference_pipeline_metrics_df.assembler.unique():\n",
    "    row={'Assembler': assembler, \n",
    "         'Type':reference_pipeline_metrics_df[reference_pipeline_metrics_df.assembler == assembler].type.unique()[0],\n",
    "         'Algorithm': reference_pipeline_metrics_df[reference_pipeline_metrics_df.assembler == assembler].algorythm.unique()[0]}\n",
    "    lala=reference_pipeline_metrics_df[reference_pipeline_metrics_df.assembler == assembler].describe()\n",
    "    for column in lala.columns:\n",
    "        mean = lala.loc['mean',column]\n",
    "        std = lala.loc['std',column]\n",
    "        row[column] = \"{}(±{})\".format(round(mean, 2), round(std, 2))\n",
    "    ref_stats_per_assembler=ref_stats_per_assembler.append(row, ignore_index=True)\n",
    "ref_stats_per_assembler = ref_stats_per_assembler.set_index('Assembler')\n",
    "ref_stats_per_assembler.to_csv(\"Tables/Results/Reference metrics per assembler.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>44.26(±217.35)</td>\n",
       "      <td>5388.68(±13665.31)</td>\n",
       "      <td>5086.4(±13438.6)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>840447.21(±1648230.31)</td>\n",
       "      <td>0.23(±0.38)</td>\n",
       "      <td>188.96(±421.28)</td>\n",
       "      <td>0.02(±0.12)</td>\n",
       "      <td>0.33(±0.47)</td>\n",
       "      <td>0.33(±0.47)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.33(±0.47)</td>\n",
       "      <td>0.33(±0.47)</td>\n",
       "      <td>0.33(±0.47)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>26.58(±60.33)</td>\n",
       "      <td>75509.0(±112297.21)</td>\n",
       "      <td>70917.21(±103321.28)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1828005.45(±2235991.29)</td>\n",
       "      <td>0.56(±0.49)</td>\n",
       "      <td>47.59(±88.59)</td>\n",
       "      <td>0.14(±0.27)</td>\n",
       "      <td>0.62(±0.48)</td>\n",
       "      <td>0.61(±0.48)</td>\n",
       "      <td>0.17(±0.53)</td>\n",
       "      <td>0.63(±0.49)</td>\n",
       "      <td>0.63(±0.49)</td>\n",
       "      <td>0.62(±0.48)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>43.75(±74.53)</td>\n",
       "      <td>52040.9(±91871.34)</td>\n",
       "      <td>51122.97(±91883.14)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1821854.84(±2229476.67)</td>\n",
       "      <td>0.54(±0.48)</td>\n",
       "      <td>74.91(±120.52)</td>\n",
       "      <td>0.09(±0.2)</td>\n",
       "      <td>0.66(±0.47)</td>\n",
       "      <td>0.57(±0.46)</td>\n",
       "      <td>0.64(±1.62)</td>\n",
       "      <td>0.67(±0.47)</td>\n",
       "      <td>0.74(±0.75)</td>\n",
       "      <td>0.65(±0.47)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>12.94(±22.35)</td>\n",
       "      <td>105537.11(±143662.92)</td>\n",
       "      <td>109204.12(±146673.33)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1837433.01(±2247331.45)</td>\n",
       "      <td>0.56(±0.49)</td>\n",
       "      <td>31.08(±44.97)</td>\n",
       "      <td>0.17(±0.3)</td>\n",
       "      <td>0.68(±0.46)</td>\n",
       "      <td>0.61(±0.45)</td>\n",
       "      <td>0.57(±1.17)</td>\n",
       "      <td>0.68(±0.47)</td>\n",
       "      <td>0.69(±0.47)</td>\n",
       "      <td>0.68(±0.47)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>13.56(±20.66)</td>\n",
       "      <td>97829.43(±133239.5)</td>\n",
       "      <td>96722.68(±132486.45)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1835138.35(±2245485.34)</td>\n",
       "      <td>0.56(±0.49)</td>\n",
       "      <td>27.24(±38.21)</td>\n",
       "      <td>0.16(±0.27)</td>\n",
       "      <td>0.68(±0.47)</td>\n",
       "      <td>0.65(±0.46)</td>\n",
       "      <td>0.17(±0.5)</td>\n",
       "      <td>0.69(±0.47)</td>\n",
       "      <td>0.69(±0.47)</td>\n",
       "      <td>0.68(±0.47)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>74.05(±131.35)</td>\n",
       "      <td>12294.45(±16991.03)</td>\n",
       "      <td>11728.97(±16899.85)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1722614.56(±2112791.43)</td>\n",
       "      <td>0.48(±0.47)</td>\n",
       "      <td>212.97(±337.68)</td>\n",
       "      <td>0.04(±0.13)</td>\n",
       "      <td>0.61(±0.49)</td>\n",
       "      <td>0.59(±0.49)</td>\n",
       "      <td>0.07(±0.35)</td>\n",
       "      <td>0.61(±0.49)</td>\n",
       "      <td>0.61(±0.49)</td>\n",
       "      <td>0.61(±0.49)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>22.76(±44.4)</td>\n",
       "      <td>47300.06(±83310.7)</td>\n",
       "      <td>46891.18(±83381.95)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1436160.07(±1959266.05)</td>\n",
       "      <td>0.47(±0.47)</td>\n",
       "      <td>95.31(±280.36)</td>\n",
       "      <td>0.12(±0.26)</td>\n",
       "      <td>0.54(±0.5)</td>\n",
       "      <td>0.48(±0.48)</td>\n",
       "      <td>0.33(±1.49)</td>\n",
       "      <td>0.54(±0.5)</td>\n",
       "      <td>0.54(±0.5)</td>\n",
       "      <td>0.54(±0.5)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>25.32(±98.82)</td>\n",
       "      <td>83133.97(±116730.71)</td>\n",
       "      <td>83465.24(±117922.75)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1808988.64(±2231603.08)</td>\n",
       "      <td>0.52(±0.48)</td>\n",
       "      <td>72.15(±259.8)</td>\n",
       "      <td>0.12(±0.21)</td>\n",
       "      <td>0.7(±0.45)</td>\n",
       "      <td>0.64(±0.45)</td>\n",
       "      <td>8.28(±66.35)</td>\n",
       "      <td>0.72(±0.46)</td>\n",
       "      <td>0.73(±0.48)</td>\n",
       "      <td>0.7(±0.45)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>11.68(±21.53)</td>\n",
       "      <td>104906.65(±137292.36)</td>\n",
       "      <td>101808.62(±136709.67)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1742439.92(±2213254.72)</td>\n",
       "      <td>0.54(±0.48)</td>\n",
       "      <td>23.51(±35.57)</td>\n",
       "      <td>0.16(±0.26)</td>\n",
       "      <td>0.56(±0.5)</td>\n",
       "      <td>0.54(±0.49)</td>\n",
       "      <td>0.12(±0.53)</td>\n",
       "      <td>0.56(±0.5)</td>\n",
       "      <td>0.56(±0.5)</td>\n",
       "      <td>0.55(±0.5)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>15.51(±41.83)</td>\n",
       "      <td>14159.49(±47548.87)</td>\n",
       "      <td>16829.79(±64041.27)</td>\n",
       "      <td>488.78(±1914.05)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>804905.71(±1643235.15)</td>\n",
       "      <td>0.25(±0.42)</td>\n",
       "      <td>85.38(±254.08)</td>\n",
       "      <td>0.03(±0.13)</td>\n",
       "      <td>0.26(±0.44)</td>\n",
       "      <td>0.25(±0.42)</td>\n",
       "      <td>0.47(±1.67)</td>\n",
       "      <td>0.26(±0.44)</td>\n",
       "      <td>0.26(±0.44)</td>\n",
       "      <td>0.26(±0.44)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm             L90  \\\n",
       "Assembler                                                           \n",
       "BCALM2               Single k-mer De Bruijn graph  44.26(±217.35)   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph   26.58(±60.33)   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph   43.75(±74.53)   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph   12.94(±22.35)   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph   13.56(±20.66)   \n",
       "MINIA                Single k-mer De Bruijn graph  74.05(±131.35)   \n",
       "SKESA              Multiple k-mer De Bruijn graph    22.76(±44.4)   \n",
       "SPAdes             Multiple k-mer De Bruijn graph   25.32(±98.82)   \n",
       "Unicycler          Multiple k-mer De Bruijn graph   11.68(±21.53)   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph   15.51(±41.83)   \n",
       "\n",
       "                                    NA50                   NG50  \\\n",
       "Assembler                                                         \n",
       "BCALM2                5388.68(±13665.31)       5086.4(±13438.6)   \n",
       "GATBMiniaPipeline    75509.0(±112297.21)   70917.21(±103321.28)   \n",
       "IDBA-UD               52040.9(±91871.34)    51122.97(±91883.14)   \n",
       "MEGAHIT            105537.11(±143662.92)  109204.12(±146673.33)   \n",
       "metaSPAdes           97829.43(±133239.5)   96722.68(±132486.45)   \n",
       "MINIA                12294.45(±16991.03)    11728.97(±16899.85)   \n",
       "SKESA                 47300.06(±83310.7)    46891.18(±83381.95)   \n",
       "SPAdes              83133.97(±116730.71)   83465.24(±117922.75)   \n",
       "Unicycler          104906.65(±137292.36)  101808.62(±136709.67)   \n",
       "VelvetOptimizer      14159.49(±47548.87)    16829.79(±64041.27)   \n",
       "\n",
       "                                 Ns         Type                basepairs  \\\n",
       "Assembler                                                                   \n",
       "BCALM2                    0.0(±0.0)      Genomic   840447.21(±1648230.31)   \n",
       "GATBMiniaPipeline         0.0(±0.0)  Metagenomic  1828005.45(±2235991.29)   \n",
       "IDBA-UD                   0.0(±0.0)  Metagenomic  1821854.84(±2229476.67)   \n",
       "MEGAHIT                   0.0(±0.0)  Metagenomic  1837433.01(±2247331.45)   \n",
       "metaSPAdes                0.0(±0.0)  Metagenomic  1835138.35(±2245485.34)   \n",
       "MINIA                     0.0(±0.0)      Genomic  1722614.56(±2112791.43)   \n",
       "SKESA                     0.0(±0.0)      Genomic  1436160.07(±1959266.05)   \n",
       "SPAdes                    0.0(±0.0)      Genomic  1808988.64(±2231603.08)   \n",
       "Unicycler                 0.0(±0.0)      Genomic  1742439.92(±2213254.72)   \n",
       "VelvetOptimizer    488.78(±1914.05)      Genomic   804905.71(±1643235.15)   \n",
       "\n",
       "                  breadth_of_coverage          contigs   contiguity  \\\n",
       "Assembler                                                             \n",
       "BCALM2                    0.23(±0.38)  188.96(±421.28)  0.02(±0.12)   \n",
       "GATBMiniaPipeline         0.56(±0.49)    47.59(±88.59)  0.14(±0.27)   \n",
       "IDBA-UD                   0.54(±0.48)   74.91(±120.52)   0.09(±0.2)   \n",
       "MEGAHIT                   0.56(±0.49)    31.08(±44.97)   0.17(±0.3)   \n",
       "metaSPAdes                0.56(±0.49)    27.24(±38.21)  0.16(±0.27)   \n",
       "MINIA                     0.48(±0.47)  212.97(±337.68)  0.04(±0.13)   \n",
       "SKESA                     0.47(±0.47)   95.31(±280.36)  0.12(±0.26)   \n",
       "SPAdes                    0.52(±0.48)    72.15(±259.8)  0.12(±0.21)   \n",
       "Unicycler                 0.54(±0.48)    23.51(±35.57)  0.16(±0.26)   \n",
       "VelvetOptimizer           0.25(±0.42)   85.38(±254.08)  0.03(±0.13)   \n",
       "\n",
       "                      identity lowest_identity misassemblies multiplicity  \\\n",
       "Assembler                                                                   \n",
       "BCALM2             0.33(±0.47)     0.33(±0.47)     0.0(±0.0)  0.33(±0.47)   \n",
       "GATBMiniaPipeline  0.62(±0.48)     0.61(±0.48)   0.17(±0.53)  0.63(±0.49)   \n",
       "IDBA-UD            0.66(±0.47)     0.57(±0.46)   0.64(±1.62)  0.67(±0.47)   \n",
       "MEGAHIT            0.68(±0.46)     0.61(±0.45)   0.57(±1.17)  0.68(±0.47)   \n",
       "metaSPAdes         0.68(±0.47)     0.65(±0.46)    0.17(±0.5)  0.69(±0.47)   \n",
       "MINIA              0.61(±0.49)     0.59(±0.49)   0.07(±0.35)  0.61(±0.49)   \n",
       "SKESA               0.54(±0.5)     0.48(±0.48)   0.33(±1.49)   0.54(±0.5)   \n",
       "SPAdes              0.7(±0.45)     0.64(±0.45)  8.28(±66.35)  0.72(±0.46)   \n",
       "Unicycler           0.56(±0.5)     0.54(±0.49)   0.12(±0.53)   0.56(±0.5)   \n",
       "VelvetOptimizer    0.26(±0.44)     0.25(±0.42)   0.47(±1.67)  0.26(±0.44)   \n",
       "\n",
       "                     parsimony     validity  \n",
       "Assembler                                    \n",
       "BCALM2             0.33(±0.47)  0.33(±0.47)  \n",
       "GATBMiniaPipeline  0.63(±0.49)  0.62(±0.48)  \n",
       "IDBA-UD            0.74(±0.75)  0.65(±0.47)  \n",
       "MEGAHIT            0.69(±0.47)  0.68(±0.47)  \n",
       "metaSPAdes         0.69(±0.47)  0.68(±0.47)  \n",
       "MINIA              0.61(±0.49)  0.61(±0.49)  \n",
       "SKESA               0.54(±0.5)   0.54(±0.5)  \n",
       "SPAdes             0.73(±0.48)   0.7(±0.45)  \n",
       "Unicycler           0.56(±0.5)   0.55(±0.5)  \n",
       "VelvetOptimizer    0.26(±0.44)  0.26(±0.44)  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ref_stats_per_assembler"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reference stats per reference replicon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Bacillus_subtilis', 'Enterococcus_faecalis',\n",
       "       'Escherichia_coli_plasmid', 'Escherichia_coli',\n",
       "       'Lactobacillus_fermentum', 'Listeria_monocytogenes',\n",
       "       'Pseudomonas_aeruginosa', 'Salmonella_enterica',\n",
       "       'Staphylococcus_aureus', 'Staphylococcus_aureus_plasmid1',\n",
       "       'Staphylococcus_aureus_plasmid2', 'Staphylococcus_aureus_plasmid3'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reference_pipeline_metrics_df.reference.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Bacillus_subtilis\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>32.5(±50.86)</td>\n",
       "      <td>17016.5(±25096.26)</td>\n",
       "      <td>16507.83(±25340.74)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1913231.5(±1791735.3)</td>\n",
       "      <td>0.47(±0.44)</td>\n",
       "      <td>382.67(±527.71)</td>\n",
       "      <td>0.01(±0.01)</td>\n",
       "      <td>0.83(±0.38)</td>\n",
       "      <td>0.83(±0.38)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.83(±0.38)</td>\n",
       "      <td>0.83(±0.38)</td>\n",
       "      <td>0.83(±0.38)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>115.78(±163.82)</td>\n",
       "      <td>149567.5(±120326.96)</td>\n",
       "      <td>147301.83(±118434.41)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3960391.94(±36960.38)</td>\n",
       "      <td>0.98(±0.01)</td>\n",
       "      <td>165.39(±211.89)</td>\n",
       "      <td>0.08(±0.06)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.99(±0.01)</td>\n",
       "      <td>0.5(±0.79)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>92.22(±96.95)</td>\n",
       "      <td>111081.0(±92895.6)</td>\n",
       "      <td>110171.67(±92239.26)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3968675.5(±15308.63)</td>\n",
       "      <td>0.98(±0.0)</td>\n",
       "      <td>141.56(±137.78)</td>\n",
       "      <td>0.05(±0.03)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.98(±0.01)</td>\n",
       "      <td>0.83(±1.1)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>38.67(±50.29)</td>\n",
       "      <td>205803.83(±100307.84)</td>\n",
       "      <td>206013.67(±100176.43)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3983428.61(±6728.47)</td>\n",
       "      <td>0.98(±0.0)</td>\n",
       "      <td>65.33(±73.39)</td>\n",
       "      <td>0.11(±0.06)</td>\n",
       "      <td>1.0(±0.01)</td>\n",
       "      <td>0.87(±0.21)</td>\n",
       "      <td>0.5(±0.79)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>32.83(±21.97)</td>\n",
       "      <td>173649.83(±109893.2)</td>\n",
       "      <td>170909.5(±107527.12)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3977137.17(±5077.38)</td>\n",
       "      <td>0.98(±0.0)</td>\n",
       "      <td>60.0(±37.29)</td>\n",
       "      <td>0.13(±0.09)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.98(±0.01)</td>\n",
       "      <td>0.67(±0.77)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>193.33(±212.94)</td>\n",
       "      <td>40228.33(±21807.72)</td>\n",
       "      <td>39618.44(±21736.38)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>3928316.11(±49163.52)</td>\n",
       "      <td>0.97(±0.01)</td>\n",
       "      <td>278.61(±238.93)</td>\n",
       "      <td>0.02(±0.01)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.01)</td>\n",
       "      <td>0.67(±0.97)</td>\n",
       "      <td>1.0(±0.01)</td>\n",
       "      <td>1.0(±0.01)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>9.33(±9.92)</td>\n",
       "      <td>117416.17(±122527.71)</td>\n",
       "      <td>116361.33(±121157.86)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1984359.17(±2041889.94)</td>\n",
       "      <td>0.49(±0.5)</td>\n",
       "      <td>16.33(±17.19)</td>\n",
       "      <td>0.05(±0.05)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>27.17(±22.96)</td>\n",
       "      <td>214387.33(±100758.97)</td>\n",
       "      <td>222639.33(±107608.35)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>3974502.67(±6004.79)</td>\n",
       "      <td>0.98(±0.0)</td>\n",
       "      <td>49.33(±34.12)</td>\n",
       "      <td>0.13(±0.06)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.99(±0.01)</td>\n",
       "      <td>0.33(±0.77)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>28.83(±51.41)</td>\n",
       "      <td>180230.33(±139849.68)</td>\n",
       "      <td>162359.67(±153864.25)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2965228.33(±1541741.42)</td>\n",
       "      <td>0.73(±0.38)</td>\n",
       "      <td>43.5(±61.97)</td>\n",
       "      <td>0.11(±0.08)</td>\n",
       "      <td>0.83(±0.38)</td>\n",
       "      <td>0.83(±0.38)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.83(±0.38)</td>\n",
       "      <td>0.83(±0.38)</td>\n",
       "      <td>0.83(±0.38)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>20.5(±29.85)</td>\n",
       "      <td>22185.33(±32345.98)</td>\n",
       "      <td>22034.67(±32111.59)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1317971.0(±1917929.52)</td>\n",
       "      <td>0.33(±0.47)</td>\n",
       "      <td>37.17(±54.11)</td>\n",
       "      <td>0.02(±0.03)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm              L90  \\\n",
       "Assembler                                                            \n",
       "BCALM2               Single k-mer De Bruijn graph     32.5(±50.86)   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph  115.78(±163.82)   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph    92.22(±96.95)   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph    38.67(±50.29)   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph    32.83(±21.97)   \n",
       "MINIA                Single k-mer De Bruijn graph  193.33(±212.94)   \n",
       "SKESA              Multiple k-mer De Bruijn graph      9.33(±9.92)   \n",
       "SPAdes             Multiple k-mer De Bruijn graph    27.17(±22.96)   \n",
       "Unicycler          Multiple k-mer De Bruijn graph    28.83(±51.41)   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph     20.5(±29.85)   \n",
       "\n",
       "                                    NA50                   NG50         Ns  \\\n",
       "Assembler                                                                    \n",
       "BCALM2                17016.5(±25096.26)    16507.83(±25340.74)  0.0(±0.0)   \n",
       "GATBMiniaPipeline   149567.5(±120326.96)  147301.83(±118434.41)  0.0(±0.0)   \n",
       "IDBA-UD               111081.0(±92895.6)   110171.67(±92239.26)  0.0(±0.0)   \n",
       "MEGAHIT            205803.83(±100307.84)  206013.67(±100176.43)  0.0(±0.0)   \n",
       "metaSPAdes          173649.83(±109893.2)   170909.5(±107527.12)  0.0(±0.0)   \n",
       "MINIA                40228.33(±21807.72)    39618.44(±21736.38)  0.0(±0.0)   \n",
       "SKESA              117416.17(±122527.71)  116361.33(±121157.86)  0.0(±0.0)   \n",
       "SPAdes             214387.33(±100758.97)  222639.33(±107608.35)  0.0(±0.0)   \n",
       "Unicycler          180230.33(±139849.68)  162359.67(±153864.25)  0.0(±0.0)   \n",
       "VelvetOptimizer      22185.33(±32345.98)    22034.67(±32111.59)  0.0(±0.0)   \n",
       "\n",
       "                          Type                basepairs breadth_of_coverage  \\\n",
       "Assembler                                                                     \n",
       "BCALM2                 Genomic    1913231.5(±1791735.3)         0.47(±0.44)   \n",
       "GATBMiniaPipeline  Metagenomic    3960391.94(±36960.38)         0.98(±0.01)   \n",
       "IDBA-UD            Metagenomic     3968675.5(±15308.63)          0.98(±0.0)   \n",
       "MEGAHIT            Metagenomic     3983428.61(±6728.47)          0.98(±0.0)   \n",
       "metaSPAdes         Metagenomic     3977137.17(±5077.38)          0.98(±0.0)   \n",
       "MINIA                  Genomic    3928316.11(±49163.52)         0.97(±0.01)   \n",
       "SKESA                  Genomic  1984359.17(±2041889.94)          0.49(±0.5)   \n",
       "SPAdes                 Genomic     3974502.67(±6004.79)          0.98(±0.0)   \n",
       "Unicycler              Genomic  2965228.33(±1541741.42)         0.73(±0.38)   \n",
       "VelvetOptimizer        Genomic   1317971.0(±1917929.52)         0.33(±0.47)   \n",
       "\n",
       "                           contigs   contiguity     identity lowest_identity  \\\n",
       "Assembler                                                                      \n",
       "BCALM2             382.67(±527.71)  0.01(±0.01)  0.83(±0.38)     0.83(±0.38)   \n",
       "GATBMiniaPipeline  165.39(±211.89)  0.08(±0.06)    1.0(±0.0)     0.99(±0.01)   \n",
       "IDBA-UD            141.56(±137.78)  0.05(±0.03)    1.0(±0.0)     0.98(±0.01)   \n",
       "MEGAHIT              65.33(±73.39)  0.11(±0.06)   1.0(±0.01)     0.87(±0.21)   \n",
       "metaSPAdes            60.0(±37.29)  0.13(±0.09)    1.0(±0.0)     0.98(±0.01)   \n",
       "MINIA              278.61(±238.93)  0.02(±0.01)    1.0(±0.0)      1.0(±0.01)   \n",
       "SKESA                16.33(±17.19)  0.05(±0.05)   0.5(±0.51)      0.5(±0.51)   \n",
       "SPAdes               49.33(±34.12)  0.13(±0.06)    1.0(±0.0)     0.99(±0.01)   \n",
       "Unicycler             43.5(±61.97)  0.11(±0.08)  0.83(±0.38)     0.83(±0.38)   \n",
       "VelvetOptimizer      37.17(±54.11)  0.02(±0.03)  0.33(±0.49)     0.33(±0.49)   \n",
       "\n",
       "                  misassemblies multiplicity    parsimony     validity  \n",
       "Assembler                                                               \n",
       "BCALM2                0.0(±0.0)  0.83(±0.38)  0.83(±0.38)  0.83(±0.38)  \n",
       "GATBMiniaPipeline    0.5(±0.79)    1.0(±0.0)    1.0(±0.0)    1.0(±0.0)  \n",
       "IDBA-UD              0.83(±1.1)    1.0(±0.0)    1.0(±0.0)    1.0(±0.0)  \n",
       "MEGAHIT              0.5(±0.79)    1.0(±0.0)    1.0(±0.0)    1.0(±0.0)  \n",
       "metaSPAdes          0.67(±0.77)    1.0(±0.0)    1.0(±0.0)    1.0(±0.0)  \n",
       "MINIA               0.67(±0.97)   1.0(±0.01)   1.0(±0.01)    1.0(±0.0)  \n",
       "SKESA                 0.0(±0.0)   0.5(±0.51)   0.5(±0.51)   0.5(±0.51)  \n",
       "SPAdes              0.33(±0.77)    1.0(±0.0)    1.0(±0.0)    1.0(±0.0)  \n",
       "Unicycler             0.0(±0.0)  0.83(±0.38)  0.83(±0.38)  0.83(±0.38)  \n",
       "VelvetOptimizer       0.0(±0.0)  0.33(±0.49)  0.33(±0.49)  0.33(±0.49)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Enterococcus_faecalis\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>17.61(±40.53)</td>\n",
       "      <td>5311.67(±11581.76)</td>\n",
       "      <td>5064.17(±11652.13)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>650067.0(±1057244.09)</td>\n",
       "      <td>0.23(±0.37)</td>\n",
       "      <td>160.39(±299.47)</td>\n",
       "      <td>0.01(±0.02)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>9.17(±9.98)</td>\n",
       "      <td>83020.17(±87727.63)</td>\n",
       "      <td>84170.5(±89247.09)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1401643.72(±1442325.48)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>19.28(±20.33)</td>\n",
       "      <td>0.08(±0.09)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>32.72(±53.94)</td>\n",
       "      <td>41221.11(±53482.61)</td>\n",
       "      <td>41736.33(±54302.66)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1393139.22(±1433560.52)</td>\n",
       "      <td>0.49(±0.5)</td>\n",
       "      <td>56.06(±83.83)</td>\n",
       "      <td>0.05(±0.06)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.46(±0.48)</td>\n",
       "      <td>0.5(±0.79)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>7.0(±7.37)</td>\n",
       "      <td>117668.83(±121080.23)</td>\n",
       "      <td>118207.17(±121639.12)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1404943.83(±1445728.5)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>15.83(±16.34)</td>\n",
       "      <td>0.11(±0.12)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.48(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>7.5(±9.29)</td>\n",
       "      <td>103369.67(±111300.82)</td>\n",
       "      <td>103369.67(±111300.82)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1396282.67(±1436822.15)</td>\n",
       "      <td>0.49(±0.5)</td>\n",
       "      <td>15.17(±18.36)</td>\n",
       "      <td>0.08(±0.1)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.49(±0.5)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>59.28(±62.9)</td>\n",
       "      <td>14923.44(±13043.93)</td>\n",
       "      <td>13404.67(±14286.94)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1383418.67(±1413662.1)</td>\n",
       "      <td>0.49(±0.5)</td>\n",
       "      <td>93.5(±95.6)</td>\n",
       "      <td>0.02(±0.02)</td>\n",
       "      <td>0.83(±0.38)</td>\n",
       "      <td>0.83(±0.38)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.83(±0.38)</td>\n",
       "      <td>0.83(±0.38)</td>\n",
       "      <td>0.83(±0.38)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>15.83(±17.35)</td>\n",
       "      <td>49461.0(±51925.56)</td>\n",
       "      <td>47018.5(±51361.46)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1402020.17(±1437301.65)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>29.83(±29.27)</td>\n",
       "      <td>0.06(±0.06)</td>\n",
       "      <td>0.83(±0.38)</td>\n",
       "      <td>0.69(±0.39)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.83(±0.38)</td>\n",
       "      <td>0.84(±0.39)</td>\n",
       "      <td>0.83(±0.38)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>8.0(±9.49)</td>\n",
       "      <td>97227.67(±108468.98)</td>\n",
       "      <td>98739.33(±109480.51)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1412023.67(±1443008.6)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>21.67(±21.27)</td>\n",
       "      <td>0.1(±0.1)</td>\n",
       "      <td>0.79(±0.38)</td>\n",
       "      <td>0.71(±0.43)</td>\n",
       "      <td>1.67(±2.7)</td>\n",
       "      <td>0.87(±0.41)</td>\n",
       "      <td>0.96(±0.52)</td>\n",
       "      <td>0.78(±0.38)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>4.83(±5.12)</td>\n",
       "      <td>141763.17(±153616.5)</td>\n",
       "      <td>133592.17(±149461.16)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1396688.5(±1437221.46)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>11.5(±12.11)</td>\n",
       "      <td>0.13(±0.14)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>34.17(±49.73)</td>\n",
       "      <td>10149.33(±14769.45)</td>\n",
       "      <td>10133.5(±14746.43)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>911895.5(±1327002.92)</td>\n",
       "      <td>0.32(±0.47)</td>\n",
       "      <td>53.17(±77.38)</td>\n",
       "      <td>0.02(±0.02)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.48)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm            L90  \\\n",
       "Assembler                                                          \n",
       "BCALM2               Single k-mer De Bruijn graph  17.61(±40.53)   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph    9.17(±9.98)   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  32.72(±53.94)   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph     7.0(±7.37)   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph     7.5(±9.29)   \n",
       "MINIA                Single k-mer De Bruijn graph   59.28(±62.9)   \n",
       "SKESA              Multiple k-mer De Bruijn graph  15.83(±17.35)   \n",
       "SPAdes             Multiple k-mer De Bruijn graph     8.0(±9.49)   \n",
       "Unicycler          Multiple k-mer De Bruijn graph    4.83(±5.12)   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph  34.17(±49.73)   \n",
       "\n",
       "                                    NA50                   NG50         Ns  \\\n",
       "Assembler                                                                    \n",
       "BCALM2                5311.67(±11581.76)     5064.17(±11652.13)  0.0(±0.0)   \n",
       "GATBMiniaPipeline    83020.17(±87727.63)     84170.5(±89247.09)  0.0(±0.0)   \n",
       "IDBA-UD              41221.11(±53482.61)    41736.33(±54302.66)  0.0(±0.0)   \n",
       "MEGAHIT            117668.83(±121080.23)  118207.17(±121639.12)  0.0(±0.0)   \n",
       "metaSPAdes         103369.67(±111300.82)  103369.67(±111300.82)  0.0(±0.0)   \n",
       "MINIA                14923.44(±13043.93)    13404.67(±14286.94)  0.0(±0.0)   \n",
       "SKESA                 49461.0(±51925.56)     47018.5(±51361.46)  0.0(±0.0)   \n",
       "SPAdes              97227.67(±108468.98)   98739.33(±109480.51)  0.0(±0.0)   \n",
       "Unicycler           141763.17(±153616.5)  133592.17(±149461.16)  0.0(±0.0)   \n",
       "VelvetOptimizer      10149.33(±14769.45)     10133.5(±14746.43)  0.0(±0.0)   \n",
       "\n",
       "                          Type                basepairs breadth_of_coverage  \\\n",
       "Assembler                                                                     \n",
       "BCALM2                 Genomic    650067.0(±1057244.09)         0.23(±0.37)   \n",
       "GATBMiniaPipeline  Metagenomic  1401643.72(±1442325.48)         0.49(±0.51)   \n",
       "IDBA-UD            Metagenomic  1393139.22(±1433560.52)          0.49(±0.5)   \n",
       "MEGAHIT            Metagenomic   1404943.83(±1445728.5)         0.49(±0.51)   \n",
       "metaSPAdes         Metagenomic  1396282.67(±1436822.15)          0.49(±0.5)   \n",
       "MINIA                  Genomic   1383418.67(±1413662.1)          0.49(±0.5)   \n",
       "SKESA                  Genomic  1402020.17(±1437301.65)         0.49(±0.51)   \n",
       "SPAdes                 Genomic   1412023.67(±1443008.6)          0.5(±0.51)   \n",
       "Unicycler              Genomic   1396688.5(±1437221.46)         0.49(±0.51)   \n",
       "VelvetOptimizer        Genomic    911895.5(±1327002.92)         0.32(±0.47)   \n",
       "\n",
       "                           contigs   contiguity     identity lowest_identity  \\\n",
       "Assembler                                                                      \n",
       "BCALM2             160.39(±299.47)  0.01(±0.02)  0.33(±0.49)     0.33(±0.49)   \n",
       "GATBMiniaPipeline    19.28(±20.33)  0.08(±0.09)   0.5(±0.51)     0.49(±0.51)   \n",
       "IDBA-UD              56.06(±83.83)  0.05(±0.06)   0.5(±0.51)     0.46(±0.48)   \n",
       "MEGAHIT              15.83(±16.34)  0.11(±0.12)   0.5(±0.51)     0.48(±0.49)   \n",
       "metaSPAdes           15.17(±18.36)   0.08(±0.1)   0.5(±0.51)      0.49(±0.5)   \n",
       "MINIA                  93.5(±95.6)  0.02(±0.02)  0.83(±0.38)     0.83(±0.38)   \n",
       "SKESA                29.83(±29.27)  0.06(±0.06)  0.83(±0.38)     0.69(±0.39)   \n",
       "SPAdes               21.67(±21.27)    0.1(±0.1)  0.79(±0.38)     0.71(±0.43)   \n",
       "Unicycler             11.5(±12.11)  0.13(±0.14)   0.5(±0.51)      0.5(±0.51)   \n",
       "VelvetOptimizer      53.17(±77.38)  0.02(±0.02)  0.33(±0.49)     0.33(±0.48)   \n",
       "\n",
       "                  misassemblies multiplicity    parsimony     validity  \n",
       "Assembler                                                               \n",
       "BCALM2                0.0(±0.0)  0.33(±0.49)  0.33(±0.49)  0.33(±0.49)  \n",
       "GATBMiniaPipeline   0.33(±0.49)   0.5(±0.51)   0.5(±0.51)   0.5(±0.51)  \n",
       "IDBA-UD              0.5(±0.79)   0.5(±0.52)   0.5(±0.52)   0.5(±0.51)  \n",
       "MEGAHIT             0.33(±0.49)   0.5(±0.51)   0.5(±0.51)   0.5(±0.51)  \n",
       "metaSPAdes          0.17(±0.38)   0.5(±0.52)   0.5(±0.52)   0.5(±0.51)  \n",
       "MINIA                 0.0(±0.0)  0.83(±0.38)  0.83(±0.38)  0.83(±0.38)  \n",
       "SKESA               0.33(±0.49)  0.83(±0.38)  0.84(±0.39)  0.83(±0.38)  \n",
       "SPAdes               1.67(±2.7)  0.87(±0.41)  0.96(±0.52)  0.78(±0.38)  \n",
       "Unicycler             0.0(±0.0)   0.5(±0.51)   0.5(±0.51)   0.5(±0.51)  \n",
       "VelvetOptimizer       0.0(±0.0)  0.33(±0.49)  0.33(±0.49)  0.33(±0.49)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Escherichia_coli_plasmid\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.5(±1.15)</td>\n",
       "      <td>4473.0(±10291.9)</td>\n",
       "      <td>4473.0(±10291.9)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>17516.67(±40304.01)</td>\n",
       "      <td>0.16(±0.37)</td>\n",
       "      <td>0.83(±1.92)</td>\n",
       "      <td>0.06(±0.13)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>1.17(±1.25)</td>\n",
       "      <td>18771.83(±21362.47)</td>\n",
       "      <td>18771.83(±21362.47)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>54486.44(±56067.58)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>1.78(±1.86)</td>\n",
       "      <td>0.23(±0.24)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>1.67(±1.75)</td>\n",
       "      <td>14454.33(±14499.43)</td>\n",
       "      <td>14269.83(±14684.3)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>54250.67(±55445.89)</td>\n",
       "      <td>0.49(±0.5)</td>\n",
       "      <td>2.67(±2.63)</td>\n",
       "      <td>0.17(±0.18)</td>\n",
       "      <td>0.66(±0.48)</td>\n",
       "      <td>0.66(±0.48)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.67(±0.48)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>46160.5(±50608.08)</td>\n",
       "      <td>45910.0(±50844.85)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>55000.67(±56084.82)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.83(±0.71)</td>\n",
       "      <td>0.42(±0.46)</td>\n",
       "      <td>0.66(±0.48)</td>\n",
       "      <td>0.66(±0.48)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.67(±0.48)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>32132.83(±33398.64)</td>\n",
       "      <td>32132.83(±33398.64)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>54693.0(±56282.26)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>1.0(±1.03)</td>\n",
       "      <td>0.29(±0.3)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>2.0(±2.3)</td>\n",
       "      <td>11414.5(±12448.79)</td>\n",
       "      <td>11414.5(±12448.79)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>52479.67(±54002.8)</td>\n",
       "      <td>0.48(±0.49)</td>\n",
       "      <td>3.06(±3.3)</td>\n",
       "      <td>0.15(±0.16)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.33(±0.77)</td>\n",
       "      <td>41292.5(±50823.31)</td>\n",
       "      <td>41292.5(±50823.31)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>54529.83(±56114.44)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.83(±1.1)</td>\n",
       "      <td>0.41(±0.46)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>1.0(±1.19)</td>\n",
       "      <td>21183.33(±25418.92)</td>\n",
       "      <td>20931.67(±25630.87)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>54636.83(±55707.93)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>1.83(±1.72)</td>\n",
       "      <td>0.25(±0.26)</td>\n",
       "      <td>0.66(±0.48)</td>\n",
       "      <td>0.66(±0.48)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.83(±1.1)</td>\n",
       "      <td>25299.83(±28787.27)</td>\n",
       "      <td>24956.0(±28420.26)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>54046.83(±55616.63)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>1.33(±1.53)</td>\n",
       "      <td>0.25(±0.27)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>1.0(±1.46)</td>\n",
       "      <td>8984.33(±13074.3)</td>\n",
       "      <td>8984.33(±13074.3)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>35092.0(±51066.38)</td>\n",
       "      <td>0.32(±0.46)</td>\n",
       "      <td>1.5(±2.2)</td>\n",
       "      <td>0.12(±0.17)</td>\n",
       "      <td>0.33(±0.48)</td>\n",
       "      <td>0.33(±0.48)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm          L90  \\\n",
       "Assembler                                                        \n",
       "BCALM2               Single k-mer De Bruijn graph   0.5(±1.15)   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph  1.17(±1.25)   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  1.67(±1.75)   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph  0.17(±0.38)   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph   0.5(±0.51)   \n",
       "MINIA                Single k-mer De Bruijn graph    2.0(±2.3)   \n",
       "SKESA              Multiple k-mer De Bruijn graph  0.33(±0.77)   \n",
       "SPAdes             Multiple k-mer De Bruijn graph   1.0(±1.19)   \n",
       "Unicycler          Multiple k-mer De Bruijn graph   0.83(±1.1)   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph   1.0(±1.46)   \n",
       "\n",
       "                                  NA50                 NG50         Ns  \\\n",
       "Assembler                                                                \n",
       "BCALM2                4473.0(±10291.9)     4473.0(±10291.9)  0.0(±0.0)   \n",
       "GATBMiniaPipeline  18771.83(±21362.47)  18771.83(±21362.47)  0.0(±0.0)   \n",
       "IDBA-UD            14454.33(±14499.43)   14269.83(±14684.3)  0.0(±0.0)   \n",
       "MEGAHIT             46160.5(±50608.08)   45910.0(±50844.85)  0.0(±0.0)   \n",
       "metaSPAdes         32132.83(±33398.64)  32132.83(±33398.64)  0.0(±0.0)   \n",
       "MINIA               11414.5(±12448.79)   11414.5(±12448.79)  0.0(±0.0)   \n",
       "SKESA               41292.5(±50823.31)   41292.5(±50823.31)  0.0(±0.0)   \n",
       "SPAdes             21183.33(±25418.92)  20931.67(±25630.87)  0.0(±0.0)   \n",
       "Unicycler          25299.83(±28787.27)   24956.0(±28420.26)  0.0(±0.0)   \n",
       "VelvetOptimizer      8984.33(±13074.3)    8984.33(±13074.3)  0.0(±0.0)   \n",
       "\n",
       "                          Type            basepairs breadth_of_coverage  \\\n",
       "Assembler                                                                 \n",
       "BCALM2                 Genomic  17516.67(±40304.01)         0.16(±0.37)   \n",
       "GATBMiniaPipeline  Metagenomic  54486.44(±56067.58)          0.5(±0.51)   \n",
       "IDBA-UD            Metagenomic  54250.67(±55445.89)          0.49(±0.5)   \n",
       "MEGAHIT            Metagenomic  55000.67(±56084.82)          0.5(±0.51)   \n",
       "metaSPAdes         Metagenomic   54693.0(±56282.26)          0.5(±0.51)   \n",
       "MINIA                  Genomic   52479.67(±54002.8)         0.48(±0.49)   \n",
       "SKESA                  Genomic  54529.83(±56114.44)          0.5(±0.51)   \n",
       "SPAdes                 Genomic  54636.83(±55707.93)          0.5(±0.51)   \n",
       "Unicycler              Genomic  54046.83(±55616.63)         0.49(±0.51)   \n",
       "VelvetOptimizer        Genomic   35092.0(±51066.38)         0.32(±0.46)   \n",
       "\n",
       "                       contigs   contiguity     identity lowest_identity  \\\n",
       "Assembler                                                                  \n",
       "BCALM2             0.83(±1.92)  0.06(±0.13)  0.17(±0.38)     0.17(±0.38)   \n",
       "GATBMiniaPipeline  1.78(±1.86)  0.23(±0.24)   0.5(±0.51)      0.5(±0.51)   \n",
       "IDBA-UD            2.67(±2.63)  0.17(±0.18)  0.66(±0.48)     0.66(±0.48)   \n",
       "MEGAHIT            0.83(±0.71)  0.42(±0.46)  0.66(±0.48)     0.66(±0.48)   \n",
       "metaSPAdes          1.0(±1.03)   0.29(±0.3)   0.5(±0.51)      0.5(±0.51)   \n",
       "MINIA               3.06(±3.3)  0.15(±0.16)   0.5(±0.51)      0.5(±0.51)   \n",
       "SKESA               0.83(±1.1)  0.41(±0.46)   0.5(±0.51)      0.5(±0.51)   \n",
       "SPAdes             1.83(±1.72)  0.25(±0.26)  0.66(±0.48)     0.66(±0.48)   \n",
       "Unicycler          1.33(±1.53)  0.25(±0.27)   0.5(±0.51)      0.5(±0.51)   \n",
       "VelvetOptimizer      1.5(±2.2)  0.12(±0.17)  0.33(±0.48)     0.33(±0.48)   \n",
       "\n",
       "                  misassemblies multiplicity    parsimony     validity  \n",
       "Assembler                                                               \n",
       "BCALM2                0.0(±0.0)  0.17(±0.38)  0.17(±0.38)  0.17(±0.38)  \n",
       "GATBMiniaPipeline     0.0(±0.0)   0.5(±0.52)   0.5(±0.52)   0.5(±0.51)  \n",
       "IDBA-UD               0.0(±0.0)  0.67(±0.49)  0.67(±0.49)  0.67(±0.48)  \n",
       "MEGAHIT               0.0(±0.0)  0.67(±0.49)  0.67(±0.49)  0.67(±0.48)  \n",
       "metaSPAdes            0.0(±0.0)   0.5(±0.51)   0.5(±0.51)   0.5(±0.51)  \n",
       "MINIA                 0.0(±0.0)   0.5(±0.51)   0.5(±0.51)   0.5(±0.51)  \n",
       "SKESA                 0.0(±0.0)   0.5(±0.51)   0.5(±0.51)   0.5(±0.51)  \n",
       "SPAdes                0.0(±0.0)  0.67(±0.49)  0.67(±0.49)  0.67(±0.49)  \n",
       "Unicycler             0.0(±0.0)   0.5(±0.51)   0.5(±0.51)   0.5(±0.51)  \n",
       "VelvetOptimizer       0.0(±0.0)  0.33(±0.49)  0.33(±0.49)  0.33(±0.49)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Escherichia_coli\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1069.17(±1725.93)</td>\n",
       "      <td>741.67(±1224.89)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1098956.17(±1623404.39)</td>\n",
       "      <td>0.23(±0.34)</td>\n",
       "      <td>438.5(±648.59)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>50.67(±52.39)</td>\n",
       "      <td>29829.67(±30376.52)</td>\n",
       "      <td>28981.83(±29969.56)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2313708.11(±2379308.35)</td>\n",
       "      <td>0.49(±0.5)</td>\n",
       "      <td>94.56(±96.12)</td>\n",
       "      <td>0.01(±0.01)</td>\n",
       "      <td>0.66(±0.48)</td>\n",
       "      <td>0.57(±0.46)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.66(±0.48)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>123.44(±128.22)</td>\n",
       "      <td>15857.33(±11301.17)</td>\n",
       "      <td>10091.67(±10457.69)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2274671.22(±2334566.54)</td>\n",
       "      <td>0.48(±0.49)</td>\n",
       "      <td>202.22(±202.72)</td>\n",
       "      <td>0.01(±0.01)</td>\n",
       "      <td>0.95(±0.07)</td>\n",
       "      <td>0.44(±0.4)</td>\n",
       "      <td>2.83(±2.33)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.8(±1.76)</td>\n",
       "      <td>0.84(±0.31)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>24.5(±25.55)</td>\n",
       "      <td>42178.67(±42489.07)</td>\n",
       "      <td>47202.5(±50034.96)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2330990.0(±2389116.11)</td>\n",
       "      <td>0.49(±0.5)</td>\n",
       "      <td>70.83(±65.37)</td>\n",
       "      <td>0.03(±0.03)</td>\n",
       "      <td>0.99(±0.01)</td>\n",
       "      <td>0.78(±0.36)</td>\n",
       "      <td>1.83(±2.09)</td>\n",
       "      <td>1.0(±0.01)</td>\n",
       "      <td>1.02(±0.02)</td>\n",
       "      <td>0.99(±0.02)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>26.67(±33.3)</td>\n",
       "      <td>52767.0(±56876.84)</td>\n",
       "      <td>51810.83(±57876.45)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2331363.67(±2391879.28)</td>\n",
       "      <td>0.49(±0.5)</td>\n",
       "      <td>59.0(±61.71)</td>\n",
       "      <td>0.04(±0.05)</td>\n",
       "      <td>0.98(±0.03)</td>\n",
       "      <td>0.9(±0.17)</td>\n",
       "      <td>0.83(±1.1)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.02(±0.03)</td>\n",
       "      <td>0.98(±0.03)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>2224.06(±2288.63)</td>\n",
       "      <td>1579.56(±1625.89)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1881889.39(±1936468.6)</td>\n",
       "      <td>0.39(±0.41)</td>\n",
       "      <td>565.33(±581.77)</td>\n",
       "      <td>0.01(±0.01)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>62.5(±68.0)</td>\n",
       "      <td>22329.33(±24365.42)</td>\n",
       "      <td>22091.5(±24289.1)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2281684.5(±2347862.14)</td>\n",
       "      <td>0.48(±0.49)</td>\n",
       "      <td>113.17(±120.08)</td>\n",
       "      <td>0.01(±0.01)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.19(±0.37)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.52(±0.53)</td>\n",
       "      <td>0.48(±0.5)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>29.67(±32.65)</td>\n",
       "      <td>43076.33(±44276.14)</td>\n",
       "      <td>41704.67(±44299.08)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2319842.17(±2371139.22)</td>\n",
       "      <td>0.49(±0.5)</td>\n",
       "      <td>62.17(±52.95)</td>\n",
       "      <td>0.04(±0.04)</td>\n",
       "      <td>0.97(±0.05)</td>\n",
       "      <td>0.77(±0.3)</td>\n",
       "      <td>1.17(±1.1)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.03(±0.06)</td>\n",
       "      <td>0.98(±0.06)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>20.83(±22.92)</td>\n",
       "      <td>57316.5(±59993.84)</td>\n",
       "      <td>57285.67(±60194.67)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2312584.67(±2379674.43)</td>\n",
       "      <td>0.49(±0.5)</td>\n",
       "      <td>45.83(±48.4)</td>\n",
       "      <td>0.05(±0.05)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.43(±0.46)</td>\n",
       "      <td>0.67(±1.14)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1614.83(±2349.98)</td>\n",
       "      <td>1368.17(±1991.47)</td>\n",
       "      <td>1679.67(±2444.53)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1357997.17(±1976223.35)</td>\n",
       "      <td>0.28(±0.41)</td>\n",
       "      <td>375.67(±546.71)</td>\n",
       "      <td>0.0(±0.01)</td>\n",
       "      <td>0.33(±0.48)</td>\n",
       "      <td>0.28(±0.4)</td>\n",
       "      <td>2.17(±3.17)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.48)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm              L90  \\\n",
       "Assembler                                                            \n",
       "BCALM2               Single k-mer De Bruijn graph        0.0(±0.0)   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph    50.67(±52.39)   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  123.44(±128.22)   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph     24.5(±25.55)   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph     26.67(±33.3)   \n",
       "MINIA                Single k-mer De Bruijn graph        0.0(±0.0)   \n",
       "SKESA              Multiple k-mer De Bruijn graph      62.5(±68.0)   \n",
       "SPAdes             Multiple k-mer De Bruijn graph    29.67(±32.65)   \n",
       "Unicycler          Multiple k-mer De Bruijn graph    20.83(±22.92)   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph        0.0(±0.0)   \n",
       "\n",
       "                                  NA50                 NG50  \\\n",
       "Assembler                                                     \n",
       "BCALM2               1069.17(±1725.93)     741.67(±1224.89)   \n",
       "GATBMiniaPipeline  29829.67(±30376.52)  28981.83(±29969.56)   \n",
       "IDBA-UD            15857.33(±11301.17)  10091.67(±10457.69)   \n",
       "MEGAHIT            42178.67(±42489.07)   47202.5(±50034.96)   \n",
       "metaSPAdes          52767.0(±56876.84)  51810.83(±57876.45)   \n",
       "MINIA                2224.06(±2288.63)    1579.56(±1625.89)   \n",
       "SKESA              22329.33(±24365.42)    22091.5(±24289.1)   \n",
       "SPAdes             43076.33(±44276.14)  41704.67(±44299.08)   \n",
       "Unicycler           57316.5(±59993.84)  57285.67(±60194.67)   \n",
       "VelvetOptimizer      1614.83(±2349.98)    1368.17(±1991.47)   \n",
       "\n",
       "                                  Ns         Type                basepairs  \\\n",
       "Assembler                                                                    \n",
       "BCALM2                     0.0(±0.0)      Genomic  1098956.17(±1623404.39)   \n",
       "GATBMiniaPipeline          0.0(±0.0)  Metagenomic  2313708.11(±2379308.35)   \n",
       "IDBA-UD                    0.0(±0.0)  Metagenomic  2274671.22(±2334566.54)   \n",
       "MEGAHIT                    0.0(±0.0)  Metagenomic   2330990.0(±2389116.11)   \n",
       "metaSPAdes                 0.0(±0.0)  Metagenomic  2331363.67(±2391879.28)   \n",
       "MINIA                      0.0(±0.0)      Genomic   1881889.39(±1936468.6)   \n",
       "SKESA                      0.0(±0.0)      Genomic   2281684.5(±2347862.14)   \n",
       "SPAdes                     0.0(±0.0)      Genomic  2319842.17(±2371139.22)   \n",
       "Unicycler                  0.0(±0.0)      Genomic  2312584.67(±2379674.43)   \n",
       "VelvetOptimizer    1679.67(±2444.53)      Genomic  1357997.17(±1976223.35)   \n",
       "\n",
       "                  breadth_of_coverage          contigs   contiguity  \\\n",
       "Assembler                                                             \n",
       "BCALM2                    0.23(±0.34)   438.5(±648.59)    0.0(±0.0)   \n",
       "GATBMiniaPipeline          0.49(±0.5)    94.56(±96.12)  0.01(±0.01)   \n",
       "IDBA-UD                   0.48(±0.49)  202.22(±202.72)  0.01(±0.01)   \n",
       "MEGAHIT                    0.49(±0.5)    70.83(±65.37)  0.03(±0.03)   \n",
       "metaSPAdes                 0.49(±0.5)     59.0(±61.71)  0.04(±0.05)   \n",
       "MINIA                     0.39(±0.41)  565.33(±581.77)  0.01(±0.01)   \n",
       "SKESA                     0.48(±0.49)  113.17(±120.08)  0.01(±0.01)   \n",
       "SPAdes                     0.49(±0.5)    62.17(±52.95)  0.04(±0.04)   \n",
       "Unicycler                  0.49(±0.5)     45.83(±48.4)  0.05(±0.05)   \n",
       "VelvetOptimizer           0.28(±0.41)  375.67(±546.71)   0.0(±0.01)   \n",
       "\n",
       "                      identity lowest_identity misassemblies multiplicity  \\\n",
       "Assembler                                                                   \n",
       "BCALM2             0.33(±0.49)     0.33(±0.49)     0.0(±0.0)  0.33(±0.49)   \n",
       "GATBMiniaPipeline  0.66(±0.48)     0.57(±0.46)   0.17(±0.38)  0.67(±0.49)   \n",
       "IDBA-UD            0.95(±0.07)      0.44(±0.4)   2.83(±2.33)    1.0(±0.0)   \n",
       "MEGAHIT            0.99(±0.01)     0.78(±0.36)   1.83(±2.09)   1.0(±0.01)   \n",
       "metaSPAdes         0.98(±0.03)      0.9(±0.17)    0.83(±1.1)    1.0(±0.0)   \n",
       "MINIA               0.5(±0.51)     0.49(±0.51)     0.0(±0.0)   0.5(±0.52)   \n",
       "SKESA               0.5(±0.51)     0.19(±0.37)   0.33(±0.49)   0.5(±0.51)   \n",
       "SPAdes             0.97(±0.05)      0.77(±0.3)    1.17(±1.1)    1.0(±0.0)   \n",
       "Unicycler           0.5(±0.51)     0.43(±0.46)   0.67(±1.14)   0.5(±0.51)   \n",
       "VelvetOptimizer    0.33(±0.48)      0.28(±0.4)   2.17(±3.17)  0.33(±0.49)   \n",
       "\n",
       "                     parsimony     validity  \n",
       "Assembler                                    \n",
       "BCALM2             0.33(±0.49)  0.33(±0.49)  \n",
       "GATBMiniaPipeline  0.67(±0.49)  0.66(±0.48)  \n",
       "IDBA-UD             1.8(±1.76)  0.84(±0.31)  \n",
       "MEGAHIT            1.02(±0.02)  0.99(±0.02)  \n",
       "metaSPAdes         1.02(±0.03)  0.98(±0.03)  \n",
       "MINIA               0.5(±0.52)   0.5(±0.51)  \n",
       "SKESA              0.52(±0.53)   0.48(±0.5)  \n",
       "SPAdes             1.03(±0.06)  0.98(±0.06)  \n",
       "Unicycler           0.5(±0.52)   0.5(±0.51)  \n",
       "VelvetOptimizer    0.33(±0.49)  0.33(±0.48)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Lactobacillus_fermentum\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>34.67(±79.76)</td>\n",
       "      <td>2276.17(±4149.12)</td>\n",
       "      <td>1705.67(±3924.56)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>317679.39(±661045.74)</td>\n",
       "      <td>0.17(±0.35)</td>\n",
       "      <td>60.17(±93.09)</td>\n",
       "      <td>0.01(±0.01)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>27.0(±27.79)</td>\n",
       "      <td>20600.83(±20422.82)</td>\n",
       "      <td>19628.5(±20216.68)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>898814.67(±923546.39)</td>\n",
       "      <td>0.47(±0.48)</td>\n",
       "      <td>43.83(±44.47)</td>\n",
       "      <td>0.03(±0.03)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.66(±0.48)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>40.5(±48.98)</td>\n",
       "      <td>16478.5(±18171.06)</td>\n",
       "      <td>15759.67(±18171.16)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>893178.22(±917748.52)</td>\n",
       "      <td>0.47(±0.48)</td>\n",
       "      <td>57.83(±65.51)</td>\n",
       "      <td>0.03(±0.03)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.67(±0.48)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>21.5(±22.72)</td>\n",
       "      <td>22554.0(±22506.78)</td>\n",
       "      <td>21806.0(±22600.68)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>904999.06(±929803.14)</td>\n",
       "      <td>0.47(±0.49)</td>\n",
       "      <td>42.0(±42.65)</td>\n",
       "      <td>0.03(±0.03)</td>\n",
       "      <td>0.67(±0.48)</td>\n",
       "      <td>0.63(±0.46)</td>\n",
       "      <td>1.33(±1.41)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.67(±0.48)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>27.0(±28.35)</td>\n",
       "      <td>21254.67(±21139.13)</td>\n",
       "      <td>19114.17(±19802.59)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>896610.17(±919828.94)</td>\n",
       "      <td>0.47(±0.48)</td>\n",
       "      <td>43.5(±43.41)</td>\n",
       "      <td>0.03(±0.03)</td>\n",
       "      <td>0.83(±0.38)</td>\n",
       "      <td>0.82(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.83(±0.38)</td>\n",
       "      <td>0.84(±0.38)</td>\n",
       "      <td>0.83(±0.38)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>112.39(±117.25)</td>\n",
       "      <td>5502.33(±5219.43)</td>\n",
       "      <td>4980.17(±5203.16)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>874004.0(±898435.87)</td>\n",
       "      <td>0.46(±0.47)</td>\n",
       "      <td>126.22(±130.09)</td>\n",
       "      <td>0.01(±0.02)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.66(±0.48)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>32.17(±33.38)</td>\n",
       "      <td>18663.17(±19272.38)</td>\n",
       "      <td>17273.83(±17998.34)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>884407.0(±910066.32)</td>\n",
       "      <td>0.46(±0.48)</td>\n",
       "      <td>44.83(±46.29)</td>\n",
       "      <td>0.03(±0.03)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>25.67(±26.63)</td>\n",
       "      <td>22543.17(±21768.41)</td>\n",
       "      <td>21440.33(±22255.64)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>896103.67(±919245.35)</td>\n",
       "      <td>0.47(±0.48)</td>\n",
       "      <td>44.33(±44.59)</td>\n",
       "      <td>0.03(±0.03)</td>\n",
       "      <td>0.98(±0.03)</td>\n",
       "      <td>0.98(±0.02)</td>\n",
       "      <td>0.83(±1.92)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.01(±0.01)</td>\n",
       "      <td>0.99(±0.01)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>26.33(±27.24)</td>\n",
       "      <td>21890.17(±22610.14)</td>\n",
       "      <td>21005.33(±21709.7)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>887187.0(±912909.95)</td>\n",
       "      <td>0.47(±0.48)</td>\n",
       "      <td>39.83(±41.02)</td>\n",
       "      <td>0.03(±0.03)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>67.5(±98.26)</td>\n",
       "      <td>3881.33(±5648.78)</td>\n",
       "      <td>3681.33(±5357.33)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>583673.5(±849369.75)</td>\n",
       "      <td>0.31(±0.45)</td>\n",
       "      <td>77.17(±112.33)</td>\n",
       "      <td>0.01(±0.02)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm              L90  \\\n",
       "Assembler                                                            \n",
       "BCALM2               Single k-mer De Bruijn graph    34.67(±79.76)   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph     27.0(±27.79)   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph     40.5(±48.98)   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph     21.5(±22.72)   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph     27.0(±28.35)   \n",
       "MINIA                Single k-mer De Bruijn graph  112.39(±117.25)   \n",
       "SKESA              Multiple k-mer De Bruijn graph    32.17(±33.38)   \n",
       "SPAdes             Multiple k-mer De Bruijn graph    25.67(±26.63)   \n",
       "Unicycler          Multiple k-mer De Bruijn graph    26.33(±27.24)   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph     67.5(±98.26)   \n",
       "\n",
       "                                  NA50                 NG50         Ns  \\\n",
       "Assembler                                                                \n",
       "BCALM2               2276.17(±4149.12)    1705.67(±3924.56)  0.0(±0.0)   \n",
       "GATBMiniaPipeline  20600.83(±20422.82)   19628.5(±20216.68)  0.0(±0.0)   \n",
       "IDBA-UD             16478.5(±18171.06)  15759.67(±18171.16)  0.0(±0.0)   \n",
       "MEGAHIT             22554.0(±22506.78)   21806.0(±22600.68)  0.0(±0.0)   \n",
       "metaSPAdes         21254.67(±21139.13)  19114.17(±19802.59)  0.0(±0.0)   \n",
       "MINIA                5502.33(±5219.43)    4980.17(±5203.16)  0.0(±0.0)   \n",
       "SKESA              18663.17(±19272.38)  17273.83(±17998.34)  0.0(±0.0)   \n",
       "SPAdes             22543.17(±21768.41)  21440.33(±22255.64)  0.0(±0.0)   \n",
       "Unicycler          21890.17(±22610.14)   21005.33(±21709.7)  0.0(±0.0)   \n",
       "VelvetOptimizer      3881.33(±5648.78)    3681.33(±5357.33)  0.0(±0.0)   \n",
       "\n",
       "                          Type              basepairs breadth_of_coverage  \\\n",
       "Assembler                                                                   \n",
       "BCALM2                 Genomic  317679.39(±661045.74)         0.17(±0.35)   \n",
       "GATBMiniaPipeline  Metagenomic  898814.67(±923546.39)         0.47(±0.48)   \n",
       "IDBA-UD            Metagenomic  893178.22(±917748.52)         0.47(±0.48)   \n",
       "MEGAHIT            Metagenomic  904999.06(±929803.14)         0.47(±0.49)   \n",
       "metaSPAdes         Metagenomic  896610.17(±919828.94)         0.47(±0.48)   \n",
       "MINIA                  Genomic   874004.0(±898435.87)         0.46(±0.47)   \n",
       "SKESA                  Genomic   884407.0(±910066.32)         0.46(±0.48)   \n",
       "SPAdes                 Genomic  896103.67(±919245.35)         0.47(±0.48)   \n",
       "Unicycler              Genomic   887187.0(±912909.95)         0.47(±0.48)   \n",
       "VelvetOptimizer        Genomic   583673.5(±849369.75)         0.31(±0.45)   \n",
       "\n",
       "                           contigs   contiguity     identity lowest_identity  \\\n",
       "Assembler                                                                      \n",
       "BCALM2               60.17(±93.09)  0.01(±0.01)   0.5(±0.51)      0.5(±0.51)   \n",
       "GATBMiniaPipeline    43.83(±44.47)  0.03(±0.03)  0.67(±0.49)     0.66(±0.48)   \n",
       "IDBA-UD              57.83(±65.51)  0.03(±0.03)  0.67(±0.49)     0.67(±0.48)   \n",
       "MEGAHIT               42.0(±42.65)  0.03(±0.03)  0.67(±0.48)     0.63(±0.46)   \n",
       "metaSPAdes            43.5(±43.41)  0.03(±0.03)  0.83(±0.38)     0.82(±0.38)   \n",
       "MINIA              126.22(±130.09)  0.01(±0.02)  0.67(±0.49)     0.66(±0.48)   \n",
       "SKESA                44.83(±46.29)  0.03(±0.03)   0.5(±0.51)      0.5(±0.51)   \n",
       "SPAdes               44.33(±44.59)  0.03(±0.03)  0.98(±0.03)     0.98(±0.02)   \n",
       "Unicycler            39.83(±41.02)  0.03(±0.03)   0.5(±0.51)      0.5(±0.51)   \n",
       "VelvetOptimizer     77.17(±112.33)  0.01(±0.02)  0.33(±0.49)     0.33(±0.49)   \n",
       "\n",
       "                  misassemblies multiplicity    parsimony     validity  \n",
       "Assembler                                                               \n",
       "BCALM2                0.0(±0.0)   0.5(±0.51)   0.5(±0.51)   0.5(±0.51)  \n",
       "GATBMiniaPipeline     0.0(±0.0)  0.67(±0.49)  0.67(±0.49)  0.67(±0.49)  \n",
       "IDBA-UD               0.0(±0.0)  0.67(±0.49)  0.67(±0.49)  0.67(±0.49)  \n",
       "MEGAHIT             1.33(±1.41)  0.67(±0.49)  0.67(±0.49)  0.67(±0.48)  \n",
       "metaSPAdes          0.17(±0.38)  0.83(±0.38)  0.84(±0.38)  0.83(±0.38)  \n",
       "MINIA                 0.0(±0.0)  0.67(±0.49)  0.67(±0.49)  0.67(±0.49)  \n",
       "SKESA                 0.0(±0.0)   0.5(±0.51)   0.5(±0.51)   0.5(±0.51)  \n",
       "SPAdes              0.83(±1.92)    1.0(±0.0)  1.01(±0.01)  0.99(±0.01)  \n",
       "Unicycler             0.0(±0.0)   0.5(±0.51)   0.5(±0.51)   0.5(±0.51)  \n",
       "VelvetOptimizer       0.0(±0.0)  0.33(±0.49)  0.33(±0.49)  0.33(±0.49)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Listeria_monocytogenes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>20.33(±30.44)</td>\n",
       "      <td>18785.0(±27325.63)</td>\n",
       "      <td>17807.67(±26468.25)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1199536.17(±1342900.9)</td>\n",
       "      <td>0.4(±0.45)</td>\n",
       "      <td>180.94(±322.0)</td>\n",
       "      <td>0.02(±0.03)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>10.83(±0.92)</td>\n",
       "      <td>351932.67(±58746.49)</td>\n",
       "      <td>302389.5(±76869.69)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2944182.17(±16689.08)</td>\n",
       "      <td>0.98(±0.01)</td>\n",
       "      <td>21.06(±2.9)</td>\n",
       "      <td>0.2(±0.01)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.99(±0.01)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>27.33(±37.16)</td>\n",
       "      <td>269599.5(±138307.86)</td>\n",
       "      <td>269315.5(±138835.29)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2948346.94(±13320.78)</td>\n",
       "      <td>0.99(±0.0)</td>\n",
       "      <td>47.44(±57.22)</td>\n",
       "      <td>0.16(±0.06)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.99(±0.01)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>7.67(±0.97)</td>\n",
       "      <td>449821.33(±93866.14)</td>\n",
       "      <td>450683.67(±92115.79)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2948853.28(±17879.56)</td>\n",
       "      <td>0.99(±0.01)</td>\n",
       "      <td>19.17(±3.22)</td>\n",
       "      <td>0.25(±0.1)</td>\n",
       "      <td>0.99(±0.03)</td>\n",
       "      <td>0.82(±0.31)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>1.01(±0.02)</td>\n",
       "      <td>1.01(±0.02)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>9.0(±2.85)</td>\n",
       "      <td>372265.0(±124774.92)</td>\n",
       "      <td>372265.0(±124774.92)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2953982.17(±14214.38)</td>\n",
       "      <td>0.99(±0.0)</td>\n",
       "      <td>18.33(±6.25)</td>\n",
       "      <td>0.22(±0.07)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.99(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>130.11(±157.67)</td>\n",
       "      <td>32613.72(±22861.98)</td>\n",
       "      <td>31639.83(±22518.13)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2830602.61(±131725.36)</td>\n",
       "      <td>0.95(±0.04)</td>\n",
       "      <td>338.89(±319.16)</td>\n",
       "      <td>0.03(±0.02)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.83(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.01)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>47.17(±85.22)</td>\n",
       "      <td>194836.17(±146725.9)</td>\n",
       "      <td>194654.83(±146970.92)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2800686.83(±250347.09)</td>\n",
       "      <td>0.94(±0.08)</td>\n",
       "      <td>236.83(±369.15)</td>\n",
       "      <td>0.13(±0.09)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>146.83(±318.11)</td>\n",
       "      <td>217159.83(±181529.27)</td>\n",
       "      <td>217226.5(±181446.17)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2663180.83(±410722.25)</td>\n",
       "      <td>0.89(±0.14)</td>\n",
       "      <td>534.5(±766.84)</td>\n",
       "      <td>0.13(±0.1)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.91(±0.19)</td>\n",
       "      <td>94.33(±217.05)</td>\n",
       "      <td>1.07(±0.15)</td>\n",
       "      <td>1.07(±0.15)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>13.17(±11.48)</td>\n",
       "      <td>329234.17(±116931.4)</td>\n",
       "      <td>329234.17(±116931.4)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2944906.5(±13966.62)</td>\n",
       "      <td>0.98(±0.0)</td>\n",
       "      <td>25.33(±20.18)</td>\n",
       "      <td>0.2(±0.03)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.99(±0.01)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>22.5(±32.75)</td>\n",
       "      <td>16602.83(±24219.74)</td>\n",
       "      <td>15930.5(±23270.5)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>972774.33(±1415594.61)</td>\n",
       "      <td>0.33(±0.47)</td>\n",
       "      <td>39.5(±57.5)</td>\n",
       "      <td>0.02(±0.03)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm              L90  \\\n",
       "Assembler                                                            \n",
       "BCALM2               Single k-mer De Bruijn graph    20.33(±30.44)   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph     10.83(±0.92)   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph    27.33(±37.16)   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph      7.67(±0.97)   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph       9.0(±2.85)   \n",
       "MINIA                Single k-mer De Bruijn graph  130.11(±157.67)   \n",
       "SKESA              Multiple k-mer De Bruijn graph    47.17(±85.22)   \n",
       "SPAdes             Multiple k-mer De Bruijn graph  146.83(±318.11)   \n",
       "Unicycler          Multiple k-mer De Bruijn graph    13.17(±11.48)   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph     22.5(±32.75)   \n",
       "\n",
       "                                    NA50                   NG50         Ns  \\\n",
       "Assembler                                                                    \n",
       "BCALM2                18785.0(±27325.63)    17807.67(±26468.25)  0.0(±0.0)   \n",
       "GATBMiniaPipeline   351932.67(±58746.49)    302389.5(±76869.69)  0.0(±0.0)   \n",
       "IDBA-UD             269599.5(±138307.86)   269315.5(±138835.29)  0.0(±0.0)   \n",
       "MEGAHIT             449821.33(±93866.14)   450683.67(±92115.79)  0.0(±0.0)   \n",
       "metaSPAdes          372265.0(±124774.92)   372265.0(±124774.92)  0.0(±0.0)   \n",
       "MINIA                32613.72(±22861.98)    31639.83(±22518.13)  0.0(±0.0)   \n",
       "SKESA               194836.17(±146725.9)  194654.83(±146970.92)  0.0(±0.0)   \n",
       "SPAdes             217159.83(±181529.27)   217226.5(±181446.17)  0.0(±0.0)   \n",
       "Unicycler           329234.17(±116931.4)   329234.17(±116931.4)  0.0(±0.0)   \n",
       "VelvetOptimizer      16602.83(±24219.74)      15930.5(±23270.5)  0.0(±0.0)   \n",
       "\n",
       "                          Type               basepairs breadth_of_coverage  \\\n",
       "Assembler                                                                    \n",
       "BCALM2                 Genomic  1199536.17(±1342900.9)          0.4(±0.45)   \n",
       "GATBMiniaPipeline  Metagenomic   2944182.17(±16689.08)         0.98(±0.01)   \n",
       "IDBA-UD            Metagenomic   2948346.94(±13320.78)          0.99(±0.0)   \n",
       "MEGAHIT            Metagenomic   2948853.28(±17879.56)         0.99(±0.01)   \n",
       "metaSPAdes         Metagenomic   2953982.17(±14214.38)          0.99(±0.0)   \n",
       "MINIA                  Genomic  2830602.61(±131725.36)         0.95(±0.04)   \n",
       "SKESA                  Genomic  2800686.83(±250347.09)         0.94(±0.08)   \n",
       "SPAdes                 Genomic  2663180.83(±410722.25)         0.89(±0.14)   \n",
       "Unicycler              Genomic    2944906.5(±13966.62)          0.98(±0.0)   \n",
       "VelvetOptimizer        Genomic  972774.33(±1415594.61)         0.33(±0.47)   \n",
       "\n",
       "                           contigs   contiguity     identity lowest_identity  \\\n",
       "Assembler                                                                      \n",
       "BCALM2              180.94(±322.0)  0.02(±0.03)   0.5(±0.51)      0.5(±0.51)   \n",
       "GATBMiniaPipeline      21.06(±2.9)   0.2(±0.01)    1.0(±0.0)     0.99(±0.01)   \n",
       "IDBA-UD              47.44(±57.22)  0.16(±0.06)    1.0(±0.0)     0.99(±0.01)   \n",
       "MEGAHIT               19.17(±3.22)   0.25(±0.1)  0.99(±0.03)     0.82(±0.31)   \n",
       "metaSPAdes            18.33(±6.25)  0.22(±0.07)    1.0(±0.0)      0.99(±0.0)   \n",
       "MINIA              338.89(±319.16)  0.03(±0.02)    1.0(±0.0)     0.83(±0.38)   \n",
       "SKESA              236.83(±369.15)  0.13(±0.09)    1.0(±0.0)       1.0(±0.0)   \n",
       "SPAdes              534.5(±766.84)   0.13(±0.1)    1.0(±0.0)     0.91(±0.19)   \n",
       "Unicycler            25.33(±20.18)   0.2(±0.03)    1.0(±0.0)     0.99(±0.01)   \n",
       "VelvetOptimizer        39.5(±57.5)  0.02(±0.03)  0.33(±0.49)     0.33(±0.49)   \n",
       "\n",
       "                    misassemblies multiplicity    parsimony     validity  \n",
       "Assembler                                                                 \n",
       "BCALM2                  0.0(±0.0)   0.5(±0.52)   0.5(±0.52)   0.5(±0.51)  \n",
       "GATBMiniaPipeline       0.0(±0.0)    1.0(±0.0)    1.0(±0.0)    1.0(±0.0)  \n",
       "IDBA-UD                 0.0(±0.0)    1.0(±0.0)    1.0(±0.0)    1.0(±0.0)  \n",
       "MEGAHIT               0.33(±0.49)  1.01(±0.02)  1.01(±0.02)    1.0(±0.0)  \n",
       "metaSPAdes              0.0(±0.0)    1.0(±0.0)    1.0(±0.0)    1.0(±0.0)  \n",
       "MINIA                 0.17(±0.38)    1.0(±0.0)   1.0(±0.01)    1.0(±0.0)  \n",
       "SKESA                   0.0(±0.0)    1.0(±0.0)    1.0(±0.0)    1.0(±0.0)  \n",
       "SPAdes             94.33(±217.05)  1.07(±0.15)  1.07(±0.15)    1.0(±0.0)  \n",
       "Unicycler               0.0(±0.0)    1.0(±0.0)    1.0(±0.0)    1.0(±0.0)  \n",
       "VelvetOptimizer         0.0(±0.0)  0.33(±0.49)  0.33(±0.49)  0.33(±0.49)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pseudomonas_aeruginosa\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>406.83(±655.91)</td>\n",
       "      <td>8346.0(±10702.57)</td>\n",
       "      <td>8015.67(±10361.87)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>3187532.83(±3283009.66)</td>\n",
       "      <td>0.47(±0.48)</td>\n",
       "      <td>469.17(±657.93)</td>\n",
       "      <td>0.01(±0.01)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>47.0(±13.6)</td>\n",
       "      <td>173809.83(±50290.87)</td>\n",
       "      <td>172485.0(±51558.02)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>6694545.06(±29032.39)</td>\n",
       "      <td>0.99(±0.0)</td>\n",
       "      <td>112.67(±18.97)</td>\n",
       "      <td>0.06(±0.01)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.99(±0.03)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>65.06(±14.89)</td>\n",
       "      <td>118208.5(±31817.86)</td>\n",
       "      <td>115382.94(±31087.86)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>6700001.22(±4448.6)</td>\n",
       "      <td>0.99(±0.0)</td>\n",
       "      <td>146.56(±24.52)</td>\n",
       "      <td>0.05(±0.01)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.95(±0.02)</td>\n",
       "      <td>0.72(±0.89)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>28.33(±2.28)</td>\n",
       "      <td>249058.0(±21799.24)</td>\n",
       "      <td>250514.0(±24503.39)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>6733272.83(±3881.8)</td>\n",
       "      <td>0.99(±0.0)</td>\n",
       "      <td>79.67(±2.77)</td>\n",
       "      <td>0.07(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.97(±0.04)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>29.0(±7.65)</td>\n",
       "      <td>233006.33(±47928.11)</td>\n",
       "      <td>227289.33(±42851.17)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>6727853.5(±3660.1)</td>\n",
       "      <td>0.99(±0.0)</td>\n",
       "      <td>66.0(±16.22)</td>\n",
       "      <td>0.11(±0.03)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.98(±0.01)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1.0(±0.01)</td>\n",
       "      <td>1.0(±0.01)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>330.83(±39.97)</td>\n",
       "      <td>22430.78(±1966.48)</td>\n",
       "      <td>21759.0(±2084.92)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>6510191.22(±14217.43)</td>\n",
       "      <td>0.96(±0.0)</td>\n",
       "      <td>511.11(±46.55)</td>\n",
       "      <td>0.02(±0.01)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.99(±0.01)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>25.5(±27.51)</td>\n",
       "      <td>70226.5(±73300.89)</td>\n",
       "      <td>71537.67(±77206.75)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>4195442.67(±2764565.23)</td>\n",
       "      <td>0.62(±0.41)</td>\n",
       "      <td>560.67(±720.95)</td>\n",
       "      <td>0.03(±0.03)</td>\n",
       "      <td>0.83(±0.38)</td>\n",
       "      <td>0.77(±0.36)</td>\n",
       "      <td>3.0(±4.37)</td>\n",
       "      <td>0.83(±0.38)</td>\n",
       "      <td>0.83(±0.38)</td>\n",
       "      <td>0.83(±0.38)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>39.17(±13.59)</td>\n",
       "      <td>201858.33(±59886.53)</td>\n",
       "      <td>198958.83(±57401.18)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>6710119.33(±14969.94)</td>\n",
       "      <td>0.99(±0.0)</td>\n",
       "      <td>86.17(±25.37)</td>\n",
       "      <td>0.09(±0.04)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.98(±0.02)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>26.67(±3.74)</td>\n",
       "      <td>262392.33(±55848.4)</td>\n",
       "      <td>262392.33(±55848.4)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>6683616.17(±62106.21)</td>\n",
       "      <td>0.98(±0.01)</td>\n",
       "      <td>69.0(±9.26)</td>\n",
       "      <td>0.11(±0.03)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.97(±0.05)</td>\n",
       "      <td>0.5(±1.15)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>5.17(±7.52)</td>\n",
       "      <td>93246.0(±135976.2)</td>\n",
       "      <td>127325.67(±187046.92)</td>\n",
       "      <td>3093.67(±5137.7)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2235285.83(±3252822.59)</td>\n",
       "      <td>0.33(±0.48)</td>\n",
       "      <td>24.67(±36.29)</td>\n",
       "      <td>0.03(±0.05)</td>\n",
       "      <td>0.33(±0.48)</td>\n",
       "      <td>0.3(±0.45)</td>\n",
       "      <td>2.33(±3.6)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.48)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm              L90  \\\n",
       "Assembler                                                            \n",
       "BCALM2               Single k-mer De Bruijn graph  406.83(±655.91)   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph      47.0(±13.6)   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph    65.06(±14.89)   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph     28.33(±2.28)   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph      29.0(±7.65)   \n",
       "MINIA                Single k-mer De Bruijn graph   330.83(±39.97)   \n",
       "SKESA              Multiple k-mer De Bruijn graph     25.5(±27.51)   \n",
       "SPAdes             Multiple k-mer De Bruijn graph    39.17(±13.59)   \n",
       "Unicycler          Multiple k-mer De Bruijn graph     26.67(±3.74)   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph      5.17(±7.52)   \n",
       "\n",
       "                                   NA50                   NG50  \\\n",
       "Assembler                                                        \n",
       "BCALM2                8346.0(±10702.57)     8015.67(±10361.87)   \n",
       "GATBMiniaPipeline  173809.83(±50290.87)    172485.0(±51558.02)   \n",
       "IDBA-UD             118208.5(±31817.86)   115382.94(±31087.86)   \n",
       "MEGAHIT             249058.0(±21799.24)    250514.0(±24503.39)   \n",
       "metaSPAdes         233006.33(±47928.11)   227289.33(±42851.17)   \n",
       "MINIA                22430.78(±1966.48)      21759.0(±2084.92)   \n",
       "SKESA                70226.5(±73300.89)    71537.67(±77206.75)   \n",
       "SPAdes             201858.33(±59886.53)   198958.83(±57401.18)   \n",
       "Unicycler           262392.33(±55848.4)    262392.33(±55848.4)   \n",
       "VelvetOptimizer      93246.0(±135976.2)  127325.67(±187046.92)   \n",
       "\n",
       "                                 Ns         Type                basepairs  \\\n",
       "Assembler                                                                   \n",
       "BCALM2                    0.0(±0.0)      Genomic  3187532.83(±3283009.66)   \n",
       "GATBMiniaPipeline         0.0(±0.0)  Metagenomic    6694545.06(±29032.39)   \n",
       "IDBA-UD                   0.0(±0.0)  Metagenomic      6700001.22(±4448.6)   \n",
       "MEGAHIT                   0.0(±0.0)  Metagenomic      6733272.83(±3881.8)   \n",
       "metaSPAdes                0.0(±0.0)  Metagenomic       6727853.5(±3660.1)   \n",
       "MINIA                     0.0(±0.0)      Genomic    6510191.22(±14217.43)   \n",
       "SKESA                     0.0(±0.0)      Genomic  4195442.67(±2764565.23)   \n",
       "SPAdes                    0.0(±0.0)      Genomic    6710119.33(±14969.94)   \n",
       "Unicycler                 0.0(±0.0)      Genomic    6683616.17(±62106.21)   \n",
       "VelvetOptimizer    3093.67(±5137.7)      Genomic  2235285.83(±3252822.59)   \n",
       "\n",
       "                  breadth_of_coverage          contigs   contiguity  \\\n",
       "Assembler                                                             \n",
       "BCALM2                    0.47(±0.48)  469.17(±657.93)  0.01(±0.01)   \n",
       "GATBMiniaPipeline          0.99(±0.0)   112.67(±18.97)  0.06(±0.01)   \n",
       "IDBA-UD                    0.99(±0.0)   146.56(±24.52)  0.05(±0.01)   \n",
       "MEGAHIT                    0.99(±0.0)     79.67(±2.77)   0.07(±0.0)   \n",
       "metaSPAdes                 0.99(±0.0)     66.0(±16.22)  0.11(±0.03)   \n",
       "MINIA                      0.96(±0.0)   511.11(±46.55)  0.02(±0.01)   \n",
       "SKESA                     0.62(±0.41)  560.67(±720.95)  0.03(±0.03)   \n",
       "SPAdes                     0.99(±0.0)    86.17(±25.37)  0.09(±0.04)   \n",
       "Unicycler                 0.98(±0.01)      69.0(±9.26)  0.11(±0.03)   \n",
       "VelvetOptimizer           0.33(±0.48)    24.67(±36.29)  0.03(±0.05)   \n",
       "\n",
       "                      identity lowest_identity misassemblies multiplicity  \\\n",
       "Assembler                                                                   \n",
       "BCALM2              0.5(±0.51)      0.5(±0.51)     0.0(±0.0)   0.5(±0.52)   \n",
       "GATBMiniaPipeline    1.0(±0.0)     0.99(±0.03)     0.0(±0.0)    1.0(±0.0)   \n",
       "IDBA-UD              1.0(±0.0)     0.95(±0.02)   0.72(±0.89)    1.0(±0.0)   \n",
       "MEGAHIT              1.0(±0.0)     0.97(±0.04)   0.17(±0.38)    1.0(±0.0)   \n",
       "metaSPAdes           1.0(±0.0)     0.98(±0.01)     0.0(±0.0)   1.0(±0.01)   \n",
       "MINIA                1.0(±0.0)     0.99(±0.01)     0.0(±0.0)    1.0(±0.0)   \n",
       "SKESA              0.83(±0.38)     0.77(±0.36)    3.0(±4.37)  0.83(±0.38)   \n",
       "SPAdes               1.0(±0.0)     0.98(±0.02)   0.17(±0.38)    1.0(±0.0)   \n",
       "Unicycler            1.0(±0.0)     0.97(±0.05)    0.5(±1.15)    1.0(±0.0)   \n",
       "VelvetOptimizer    0.33(±0.48)      0.3(±0.45)    2.33(±3.6)  0.33(±0.49)   \n",
       "\n",
       "                     parsimony     validity  \n",
       "Assembler                                    \n",
       "BCALM2              0.5(±0.52)   0.5(±0.51)  \n",
       "GATBMiniaPipeline    1.0(±0.0)    1.0(±0.0)  \n",
       "IDBA-UD              1.0(±0.0)    1.0(±0.0)  \n",
       "MEGAHIT              1.0(±0.0)    1.0(±0.0)  \n",
       "metaSPAdes          1.0(±0.01)    1.0(±0.0)  \n",
       "MINIA                1.0(±0.0)    1.0(±0.0)  \n",
       "SKESA              0.83(±0.38)  0.83(±0.38)  \n",
       "SPAdes               1.0(±0.0)    1.0(±0.0)  \n",
       "Unicycler            1.0(±0.0)    1.0(±0.0)  \n",
       "VelvetOptimizer    0.33(±0.49)  0.33(±0.48)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salmonella_enterica\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1158.83(±1916.42)</td>\n",
       "      <td>804.33(±1351.7)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1112954.17(±1644329.61)</td>\n",
       "      <td>0.23(±0.35)</td>\n",
       "      <td>437.5(±651.59)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>40.67(±42.14)</td>\n",
       "      <td>31951.17(±32333.83)</td>\n",
       "      <td>31071.33(±32396.18)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2326649.83(±2392247.2)</td>\n",
       "      <td>0.49(±0.5)</td>\n",
       "      <td>84.0(±85.14)</td>\n",
       "      <td>0.03(±0.03)</td>\n",
       "      <td>0.83(±0.38)</td>\n",
       "      <td>0.76(±0.38)</td>\n",
       "      <td>0.67(±1.14)</td>\n",
       "      <td>0.83(±0.38)</td>\n",
       "      <td>0.84(±0.38)</td>\n",
       "      <td>0.83(±0.38)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>109.06(±114.37)</td>\n",
       "      <td>12896.33(±12227.84)</td>\n",
       "      <td>12073.33(±12634.8)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2295770.28(±2357256.56)</td>\n",
       "      <td>0.48(±0.5)</td>\n",
       "      <td>189.61(±194.38)</td>\n",
       "      <td>0.01(±0.01)</td>\n",
       "      <td>0.97(±0.04)</td>\n",
       "      <td>0.6(±0.35)</td>\n",
       "      <td>2.83(±3.43)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.03(±0.03)</td>\n",
       "      <td>0.98(±0.03)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>20.33(±21.43)</td>\n",
       "      <td>55259.0(±56426.16)</td>\n",
       "      <td>58692.83(±62326.73)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2342956.17(±2400081.6)</td>\n",
       "      <td>0.49(±0.5)</td>\n",
       "      <td>56.67(±50.93)</td>\n",
       "      <td>0.05(±0.05)</td>\n",
       "      <td>0.99(±0.01)</td>\n",
       "      <td>0.86(±0.18)</td>\n",
       "      <td>1.0(±1.46)</td>\n",
       "      <td>1.0(±0.01)</td>\n",
       "      <td>1.02(±0.01)</td>\n",
       "      <td>0.98(±0.01)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>21.33(±28.22)</td>\n",
       "      <td>72896.5(±83591.61)</td>\n",
       "      <td>71169.5(±82964.84)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2340048.0(±2395382.3)</td>\n",
       "      <td>0.49(±0.5)</td>\n",
       "      <td>47.5(±49.7)</td>\n",
       "      <td>0.04(±0.04)</td>\n",
       "      <td>0.98(±0.03)</td>\n",
       "      <td>0.82(±0.34)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.02(±0.06)</td>\n",
       "      <td>0.98(±0.06)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>3137.44(±1877.71)</td>\n",
       "      <td>1742.83(±1794.62)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1909524.06(±1963550.33)</td>\n",
       "      <td>0.4(±0.41)</td>\n",
       "      <td>545.67(±560.51)</td>\n",
       "      <td>0.01(±0.01)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.96(±0.07)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>59.0(±63.87)</td>\n",
       "      <td>23731.83(±25211.65)</td>\n",
       "      <td>23076.5(±24362.39)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2300371.83(±2367089.27)</td>\n",
       "      <td>0.48(±0.5)</td>\n",
       "      <td>104.5(±111.25)</td>\n",
       "      <td>0.02(±0.02)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.31(±0.39)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>17.83(±20.12)</td>\n",
       "      <td>68132.5(±72290.47)</td>\n",
       "      <td>67903.0(±73423.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2338166.67(±2386369.52)</td>\n",
       "      <td>0.49(±0.5)</td>\n",
       "      <td>51.0(±41.68)</td>\n",
       "      <td>0.05(±0.05)</td>\n",
       "      <td>0.97(±0.04)</td>\n",
       "      <td>0.72(±0.35)</td>\n",
       "      <td>0.67(±0.77)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.04(±0.06)</td>\n",
       "      <td>0.97(±0.05)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>13.67(±14.86)</td>\n",
       "      <td>96287.17(±101334.93)</td>\n",
       "      <td>96287.17(±101334.93)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2324824.0(±2392235.1)</td>\n",
       "      <td>0.49(±0.5)</td>\n",
       "      <td>33.0(±35.73)</td>\n",
       "      <td>0.05(±0.05)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.4(±0.46)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.51(±0.52)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1825.17(±2656.55)</td>\n",
       "      <td>1487.83(±2165.34)</td>\n",
       "      <td>1092.0(±1616.73)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1375574.83(±2001758.82)</td>\n",
       "      <td>0.29(±0.42)</td>\n",
       "      <td>360.0(±523.93)</td>\n",
       "      <td>0.0(±0.01)</td>\n",
       "      <td>0.33(±0.48)</td>\n",
       "      <td>0.26(±0.39)</td>\n",
       "      <td>1.17(±1.72)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.48)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm              L90  \\\n",
       "Assembler                                                            \n",
       "BCALM2               Single k-mer De Bruijn graph        0.0(±0.0)   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph    40.67(±42.14)   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  109.06(±114.37)   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph    20.33(±21.43)   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph    21.33(±28.22)   \n",
       "MINIA                Single k-mer De Bruijn graph        0.0(±0.0)   \n",
       "SKESA              Multiple k-mer De Bruijn graph     59.0(±63.87)   \n",
       "SPAdes             Multiple k-mer De Bruijn graph    17.83(±20.12)   \n",
       "Unicycler          Multiple k-mer De Bruijn graph    13.67(±14.86)   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph        0.0(±0.0)   \n",
       "\n",
       "                                   NA50                  NG50  \\\n",
       "Assembler                                                       \n",
       "BCALM2                1158.83(±1916.42)       804.33(±1351.7)   \n",
       "GATBMiniaPipeline   31951.17(±32333.83)   31071.33(±32396.18)   \n",
       "IDBA-UD             12896.33(±12227.84)    12073.33(±12634.8)   \n",
       "MEGAHIT              55259.0(±56426.16)   58692.83(±62326.73)   \n",
       "metaSPAdes           72896.5(±83591.61)    71169.5(±82964.84)   \n",
       "MINIA                 3137.44(±1877.71)     1742.83(±1794.62)   \n",
       "SKESA               23731.83(±25211.65)    23076.5(±24362.39)   \n",
       "SPAdes               68132.5(±72290.47)     67903.0(±73423.0)   \n",
       "Unicycler          96287.17(±101334.93)  96287.17(±101334.93)   \n",
       "VelvetOptimizer       1825.17(±2656.55)     1487.83(±2165.34)   \n",
       "\n",
       "                                 Ns         Type                basepairs  \\\n",
       "Assembler                                                                   \n",
       "BCALM2                    0.0(±0.0)      Genomic  1112954.17(±1644329.61)   \n",
       "GATBMiniaPipeline         0.0(±0.0)  Metagenomic   2326649.83(±2392247.2)   \n",
       "IDBA-UD                   0.0(±0.0)  Metagenomic  2295770.28(±2357256.56)   \n",
       "MEGAHIT                   0.0(±0.0)  Metagenomic   2342956.17(±2400081.6)   \n",
       "metaSPAdes                0.0(±0.0)  Metagenomic    2340048.0(±2395382.3)   \n",
       "MINIA                     0.0(±0.0)      Genomic  1909524.06(±1963550.33)   \n",
       "SKESA                     0.0(±0.0)      Genomic  2300371.83(±2367089.27)   \n",
       "SPAdes                    0.0(±0.0)      Genomic  2338166.67(±2386369.52)   \n",
       "Unicycler                 0.0(±0.0)      Genomic    2324824.0(±2392235.1)   \n",
       "VelvetOptimizer    1092.0(±1616.73)      Genomic  1375574.83(±2001758.82)   \n",
       "\n",
       "                  breadth_of_coverage          contigs   contiguity  \\\n",
       "Assembler                                                             \n",
       "BCALM2                    0.23(±0.35)   437.5(±651.59)    0.0(±0.0)   \n",
       "GATBMiniaPipeline          0.49(±0.5)     84.0(±85.14)  0.03(±0.03)   \n",
       "IDBA-UD                    0.48(±0.5)  189.61(±194.38)  0.01(±0.01)   \n",
       "MEGAHIT                    0.49(±0.5)    56.67(±50.93)  0.05(±0.05)   \n",
       "metaSPAdes                 0.49(±0.5)      47.5(±49.7)  0.04(±0.04)   \n",
       "MINIA                      0.4(±0.41)  545.67(±560.51)  0.01(±0.01)   \n",
       "SKESA                      0.48(±0.5)   104.5(±111.25)  0.02(±0.02)   \n",
       "SPAdes                     0.49(±0.5)     51.0(±41.68)  0.05(±0.05)   \n",
       "Unicycler                  0.49(±0.5)     33.0(±35.73)  0.05(±0.05)   \n",
       "VelvetOptimizer           0.29(±0.42)   360.0(±523.93)   0.0(±0.01)   \n",
       "\n",
       "                      identity lowest_identity misassemblies multiplicity  \\\n",
       "Assembler                                                                   \n",
       "BCALM2             0.33(±0.49)     0.33(±0.49)     0.0(±0.0)  0.33(±0.49)   \n",
       "GATBMiniaPipeline  0.83(±0.38)     0.76(±0.38)   0.67(±1.14)  0.83(±0.38)   \n",
       "IDBA-UD            0.97(±0.04)      0.6(±0.35)   2.83(±3.43)    1.0(±0.0)   \n",
       "MEGAHIT            0.99(±0.01)     0.86(±0.18)    1.0(±1.46)   1.0(±0.01)   \n",
       "metaSPAdes         0.98(±0.03)     0.82(±0.34)   0.17(±0.38)    1.0(±0.0)   \n",
       "MINIA                1.0(±0.0)     0.96(±0.07)     0.0(±0.0)    1.0(±0.0)   \n",
       "SKESA               0.5(±0.51)     0.31(±0.39)   0.33(±0.49)   0.5(±0.51)   \n",
       "SPAdes             0.97(±0.04)     0.72(±0.35)   0.67(±0.77)    1.0(±0.0)   \n",
       "Unicycler           0.5(±0.51)      0.4(±0.46)   0.17(±0.38)   0.5(±0.51)   \n",
       "VelvetOptimizer    0.33(±0.48)     0.26(±0.39)   1.17(±1.72)  0.33(±0.49)   \n",
       "\n",
       "                     parsimony     validity  \n",
       "Assembler                                    \n",
       "BCALM2             0.33(±0.49)  0.33(±0.49)  \n",
       "GATBMiniaPipeline  0.84(±0.38)  0.83(±0.38)  \n",
       "IDBA-UD            1.03(±0.03)  0.98(±0.03)  \n",
       "MEGAHIT            1.02(±0.01)  0.98(±0.01)  \n",
       "metaSPAdes         1.02(±0.06)  0.98(±0.06)  \n",
       "MINIA                1.0(±0.0)    1.0(±0.0)  \n",
       "SKESA               0.5(±0.52)   0.5(±0.51)  \n",
       "SPAdes             1.04(±0.06)  0.97(±0.05)  \n",
       "Unicycler          0.51(±0.52)   0.5(±0.51)  \n",
       "VelvetOptimizer    0.33(±0.49)  0.33(±0.48)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Staphylococcus_aureus\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>18.67(±42.95)</td>\n",
       "      <td>5171.83(±11274.96)</td>\n",
       "      <td>4860.83(±11184.27)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>586836.67(±989066.85)</td>\n",
       "      <td>0.22(±0.36)</td>\n",
       "      <td>137.17(±244.1)</td>\n",
       "      <td>0.01(±0.01)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>16.67(±17.46)</td>\n",
       "      <td>42504.0(±45154.94)</td>\n",
       "      <td>42085.83(±46400.33)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1337609.5(±1376390.95)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>27.67(±28.94)</td>\n",
       "      <td>0.05(±0.05)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.33(±0.77)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>32.83(±41.35)</td>\n",
       "      <td>22172.17(±24585.8)</td>\n",
       "      <td>22152.67(±24582.17)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1331546.17(±1370153.06)</td>\n",
       "      <td>0.49(±0.5)</td>\n",
       "      <td>54.17(±65.18)</td>\n",
       "      <td>0.03(±0.03)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.42(±0.45)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>7.17(±7.96)</td>\n",
       "      <td>73781.5(±85727.41)</td>\n",
       "      <td>107260.0(±134187.61)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1340715.33(±1379584.73)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>21.83(±22.66)</td>\n",
       "      <td>0.06(±0.07)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.42(±0.45)</td>\n",
       "      <td>1.33(±1.64)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>8.67(±11.19)</td>\n",
       "      <td>108771.83(±147827.93)</td>\n",
       "      <td>108771.83(±147827.93)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1339675.67(±1378517.87)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>15.33(±18.58)</td>\n",
       "      <td>0.1(±0.14)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>60.67(±63.16)</td>\n",
       "      <td>13787.83(±14346.77)</td>\n",
       "      <td>13552.67(±14105.71)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1299678.06(±1337358.65)</td>\n",
       "      <td>0.48(±0.49)</td>\n",
       "      <td>92.94(±96.25)</td>\n",
       "      <td>0.02(±0.02)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>21.33(±21.99)</td>\n",
       "      <td>26558.67(±27328.67)</td>\n",
       "      <td>26302.17(±27069.53)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1326801.67(±1365271.18)</td>\n",
       "      <td>0.49(±0.5)</td>\n",
       "      <td>35.5(±36.66)</td>\n",
       "      <td>0.03(±0.03)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>8.5(±11.11)</td>\n",
       "      <td>109600.33(±144920.98)</td>\n",
       "      <td>109600.33(±144920.98)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1336849.0(±1375608.95)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>14.33(±18.42)</td>\n",
       "      <td>0.09(±0.09)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>5.0(±5.82)</td>\n",
       "      <td>140460.5(±155972.74)</td>\n",
       "      <td>130585.33(±152026.96)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1336196.33(±1374936.83)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>12.0(±12.87)</td>\n",
       "      <td>0.16(±0.19)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>35.33(±51.43)</td>\n",
       "      <td>10368.67(±15105.21)</td>\n",
       "      <td>9955.5(±14487.97)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>867548.33(±1262468.35)</td>\n",
       "      <td>0.32(±0.46)</td>\n",
       "      <td>55.5(±80.78)</td>\n",
       "      <td>0.01(±0.02)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm            L90  \\\n",
       "Assembler                                                          \n",
       "BCALM2               Single k-mer De Bruijn graph  18.67(±42.95)   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph  16.67(±17.46)   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  32.83(±41.35)   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph    7.17(±7.96)   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph   8.67(±11.19)   \n",
       "MINIA                Single k-mer De Bruijn graph  60.67(±63.16)   \n",
       "SKESA              Multiple k-mer De Bruijn graph  21.33(±21.99)   \n",
       "SPAdes             Multiple k-mer De Bruijn graph    8.5(±11.11)   \n",
       "Unicycler          Multiple k-mer De Bruijn graph     5.0(±5.82)   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph  35.33(±51.43)   \n",
       "\n",
       "                                    NA50                   NG50         Ns  \\\n",
       "Assembler                                                                    \n",
       "BCALM2                5171.83(±11274.96)     4860.83(±11184.27)  0.0(±0.0)   \n",
       "GATBMiniaPipeline     42504.0(±45154.94)    42085.83(±46400.33)  0.0(±0.0)   \n",
       "IDBA-UD               22172.17(±24585.8)    22152.67(±24582.17)  0.0(±0.0)   \n",
       "MEGAHIT               73781.5(±85727.41)   107260.0(±134187.61)  0.0(±0.0)   \n",
       "metaSPAdes         108771.83(±147827.93)  108771.83(±147827.93)  0.0(±0.0)   \n",
       "MINIA                13787.83(±14346.77)    13552.67(±14105.71)  0.0(±0.0)   \n",
       "SKESA                26558.67(±27328.67)    26302.17(±27069.53)  0.0(±0.0)   \n",
       "SPAdes             109600.33(±144920.98)  109600.33(±144920.98)  0.0(±0.0)   \n",
       "Unicycler           140460.5(±155972.74)  130585.33(±152026.96)  0.0(±0.0)   \n",
       "VelvetOptimizer      10368.67(±15105.21)      9955.5(±14487.97)  0.0(±0.0)   \n",
       "\n",
       "                          Type                basepairs breadth_of_coverage  \\\n",
       "Assembler                                                                     \n",
       "BCALM2                 Genomic    586836.67(±989066.85)         0.22(±0.36)   \n",
       "GATBMiniaPipeline  Metagenomic   1337609.5(±1376390.95)         0.49(±0.51)   \n",
       "IDBA-UD            Metagenomic  1331546.17(±1370153.06)          0.49(±0.5)   \n",
       "MEGAHIT            Metagenomic  1340715.33(±1379584.73)         0.49(±0.51)   \n",
       "metaSPAdes         Metagenomic  1339675.67(±1378517.87)         0.49(±0.51)   \n",
       "MINIA                  Genomic  1299678.06(±1337358.65)         0.48(±0.49)   \n",
       "SKESA                  Genomic  1326801.67(±1365271.18)          0.49(±0.5)   \n",
       "SPAdes                 Genomic   1336849.0(±1375608.95)         0.49(±0.51)   \n",
       "Unicycler              Genomic  1336196.33(±1374936.83)         0.49(±0.51)   \n",
       "VelvetOptimizer        Genomic   867548.33(±1262468.35)         0.32(±0.46)   \n",
       "\n",
       "                          contigs   contiguity     identity lowest_identity  \\\n",
       "Assembler                                                                     \n",
       "BCALM2             137.17(±244.1)  0.01(±0.01)  0.33(±0.49)     0.33(±0.49)   \n",
       "GATBMiniaPipeline   27.67(±28.94)  0.05(±0.05)   0.5(±0.51)      0.5(±0.51)   \n",
       "IDBA-UD             54.17(±65.18)  0.03(±0.03)   0.5(±0.51)     0.42(±0.45)   \n",
       "MEGAHIT             21.83(±22.66)  0.06(±0.07)   0.5(±0.51)     0.42(±0.45)   \n",
       "metaSPAdes          15.33(±18.58)   0.1(±0.14)   0.5(±0.51)     0.49(±0.51)   \n",
       "MINIA               92.94(±96.25)  0.02(±0.02)   0.5(±0.51)      0.5(±0.51)   \n",
       "SKESA                35.5(±36.66)  0.03(±0.03)   0.5(±0.51)      0.5(±0.51)   \n",
       "SPAdes              14.33(±18.42)  0.09(±0.09)   0.5(±0.51)      0.5(±0.51)   \n",
       "Unicycler            12.0(±12.87)  0.16(±0.19)   0.5(±0.51)      0.5(±0.51)   \n",
       "VelvetOptimizer      55.5(±80.78)  0.01(±0.02)  0.33(±0.49)     0.33(±0.49)   \n",
       "\n",
       "                  misassemblies multiplicity    parsimony     validity  \n",
       "Assembler                                                               \n",
       "BCALM2                0.0(±0.0)  0.33(±0.49)  0.33(±0.49)  0.33(±0.49)  \n",
       "GATBMiniaPipeline   0.33(±0.77)   0.5(±0.51)   0.5(±0.51)   0.5(±0.51)  \n",
       "IDBA-UD               0.0(±0.0)   0.5(±0.52)   0.5(±0.52)   0.5(±0.51)  \n",
       "MEGAHIT             1.33(±1.64)   0.5(±0.52)   0.5(±0.52)   0.5(±0.51)  \n",
       "metaSPAdes            0.0(±0.0)   0.5(±0.51)   0.5(±0.51)   0.5(±0.51)  \n",
       "MINIA                 0.0(±0.0)   0.5(±0.51)   0.5(±0.51)   0.5(±0.51)  \n",
       "SKESA                 0.0(±0.0)   0.5(±0.51)   0.5(±0.51)   0.5(±0.51)  \n",
       "SPAdes              0.17(±0.38)   0.5(±0.51)   0.5(±0.51)   0.5(±0.51)  \n",
       "Unicycler           0.17(±0.38)   0.5(±0.51)   0.5(±0.51)   0.5(±0.51)  \n",
       "VelvetOptimizer       0.0(±0.0)  0.33(±0.49)  0.33(±0.49)  0.33(±0.49)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Staphylococcus_aureus_plasmid1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1056.0(±2429.75)</td>\n",
       "      <td>1056.0(±2429.75)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1056.0(±2429.75)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>3196.11(±3289.42)</td>\n",
       "      <td>3196.11(±3289.42)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3165.83(±3257.62)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1854.83(±2737.82)</td>\n",
       "      <td>1854.83(±2737.82)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1849.83(±2732.15)</td>\n",
       "      <td>0.29(±0.43)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.29(±0.43)</td>\n",
       "      <td>0.33(±0.48)</td>\n",
       "      <td>0.33(±0.48)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.48)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>3244.5(±3342.27)</td>\n",
       "      <td>3244.5(±3342.27)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3168.17(±3260.02)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.51(±0.53)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>2818.17(±2987.9)</td>\n",
       "      <td>2818.17(±2987.9)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3146.0(±3237.52)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.67(±0.77)</td>\n",
       "      <td>0.44(±0.47)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1271.0(±2380.67)</td>\n",
       "      <td>1056.0(±2429.75)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1271.0(±2380.67)</td>\n",
       "      <td>0.2(±0.38)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.2(±0.38)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>2287.5(±2787.9)</td>\n",
       "      <td>2287.5(±2787.9)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2819.33(±2928.59)</td>\n",
       "      <td>0.44(±0.46)</td>\n",
       "      <td>0.83(±1.1)</td>\n",
       "      <td>0.36(±0.44)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>2112.0(±3073.41)</td>\n",
       "      <td>2112.0(±3073.41)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2112.0(±3073.41)</td>\n",
       "      <td>0.33(±0.48)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.48)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>3137.5(±3228.81)</td>\n",
       "      <td>3137.5(±3228.81)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>3132.5(±3223.53)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1056.0(±2429.75)</td>\n",
       "      <td>1056.0(±2429.75)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1056.0(±2429.75)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm          L90  \\\n",
       "Assembler                                                        \n",
       "BCALM2               Single k-mer De Bruijn graph    0.0(±0.0)   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph    0.0(±0.0)   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph    0.0(±0.0)   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph    0.0(±0.0)   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph  0.17(±0.38)   \n",
       "MINIA                Single k-mer De Bruijn graph    0.0(±0.0)   \n",
       "SKESA              Multiple k-mer De Bruijn graph    0.0(±0.0)   \n",
       "SPAdes             Multiple k-mer De Bruijn graph    0.0(±0.0)   \n",
       "Unicycler          Multiple k-mer De Bruijn graph    0.0(±0.0)   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph    0.0(±0.0)   \n",
       "\n",
       "                                NA50               NG50         Ns  \\\n",
       "Assembler                                                            \n",
       "BCALM2              1056.0(±2429.75)   1056.0(±2429.75)  0.0(±0.0)   \n",
       "GATBMiniaPipeline  3196.11(±3289.42)  3196.11(±3289.42)  0.0(±0.0)   \n",
       "IDBA-UD            1854.83(±2737.82)  1854.83(±2737.82)  0.0(±0.0)   \n",
       "MEGAHIT             3244.5(±3342.27)   3244.5(±3342.27)  0.0(±0.0)   \n",
       "metaSPAdes          2818.17(±2987.9)   2818.17(±2987.9)  0.0(±0.0)   \n",
       "MINIA               1271.0(±2380.67)   1056.0(±2429.75)  0.0(±0.0)   \n",
       "SKESA                2287.5(±2787.9)    2287.5(±2787.9)  0.0(±0.0)   \n",
       "SPAdes              2112.0(±3073.41)   2112.0(±3073.41)  0.0(±0.0)   \n",
       "Unicycler           3137.5(±3228.81)   3137.5(±3228.81)  0.0(±0.0)   \n",
       "VelvetOptimizer     1056.0(±2429.75)   1056.0(±2429.75)  0.0(±0.0)   \n",
       "\n",
       "                          Type          basepairs breadth_of_coverage  \\\n",
       "Assembler                                                               \n",
       "BCALM2                 Genomic   1056.0(±2429.75)         0.17(±0.38)   \n",
       "GATBMiniaPipeline  Metagenomic  3165.83(±3257.62)          0.5(±0.51)   \n",
       "IDBA-UD            Metagenomic  1849.83(±2732.15)         0.29(±0.43)   \n",
       "MEGAHIT            Metagenomic  3168.17(±3260.02)          0.5(±0.51)   \n",
       "metaSPAdes         Metagenomic   3146.0(±3237.52)          0.5(±0.51)   \n",
       "MINIA                  Genomic   1271.0(±2380.67)          0.2(±0.38)   \n",
       "SKESA                  Genomic  2819.33(±2928.59)         0.44(±0.46)   \n",
       "SPAdes                 Genomic   2112.0(±3073.41)         0.33(±0.48)   \n",
       "Unicycler              Genomic   3132.5(±3223.53)         0.49(±0.51)   \n",
       "VelvetOptimizer        Genomic   1056.0(±2429.75)         0.17(±0.38)   \n",
       "\n",
       "                       contigs   contiguity     identity lowest_identity  \\\n",
       "Assembler                                                                  \n",
       "BCALM2             0.17(±0.38)  0.17(±0.38)  0.17(±0.38)     0.17(±0.38)   \n",
       "GATBMiniaPipeline   0.5(±0.51)   0.5(±0.52)   0.5(±0.51)      0.5(±0.51)   \n",
       "IDBA-UD            0.33(±0.49)  0.29(±0.43)  0.33(±0.48)     0.33(±0.48)   \n",
       "MEGAHIT             0.5(±0.51)   0.5(±0.52)  0.49(±0.51)     0.49(±0.51)   \n",
       "metaSPAdes         0.67(±0.77)  0.44(±0.47)   0.5(±0.51)      0.5(±0.51)   \n",
       "MINIA              0.33(±0.49)   0.2(±0.38)  0.33(±0.49)     0.33(±0.49)   \n",
       "SKESA               0.83(±1.1)  0.36(±0.44)   0.5(±0.51)     0.49(±0.51)   \n",
       "SPAdes             0.33(±0.49)  0.33(±0.48)  0.33(±0.49)     0.33(±0.49)   \n",
       "Unicycler           0.5(±0.51)  0.49(±0.51)   0.5(±0.51)      0.5(±0.51)   \n",
       "VelvetOptimizer    0.17(±0.38)  0.17(±0.38)  0.17(±0.38)     0.17(±0.38)   \n",
       "\n",
       "                  misassemblies multiplicity    parsimony     validity  \n",
       "Assembler                                                               \n",
       "BCALM2                0.0(±0.0)  0.17(±0.38)  0.17(±0.38)  0.17(±0.38)  \n",
       "GATBMiniaPipeline     0.0(±0.0)   0.5(±0.52)   0.5(±0.52)   0.5(±0.51)  \n",
       "IDBA-UD               0.0(±0.0)  0.33(±0.49)  0.33(±0.49)  0.33(±0.48)  \n",
       "MEGAHIT               0.0(±0.0)   0.5(±0.52)  0.51(±0.53)  0.49(±0.51)  \n",
       "metaSPAdes            0.0(±0.0)   0.5(±0.52)   0.5(±0.52)   0.5(±0.51)  \n",
       "MINIA                 0.0(±0.0)  0.33(±0.49)  0.33(±0.49)  0.33(±0.49)  \n",
       "SKESA                 0.0(±0.0)   0.5(±0.51)   0.5(±0.52)   0.5(±0.51)  \n",
       "SPAdes                0.0(±0.0)  0.33(±0.49)  0.33(±0.49)  0.33(±0.49)  \n",
       "Unicycler             0.0(±0.0)   0.5(±0.51)   0.5(±0.52)   0.5(±0.51)  \n",
       "VelvetOptimizer       0.0(±0.0)  0.17(±0.38)  0.17(±0.38)  0.17(±0.38)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Staphylococcus_aureus_plasmid2\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>398.0(±915.76)</td>\n",
       "      <td>398.0(±915.76)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>369.33(±849.8)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.18(±0.41)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.18(±0.41)</td>\n",
       "      <td>0.18(±0.41)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>385.83(±887.76)</td>\n",
       "      <td>385.83(±887.76)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>369.33(±849.8)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.4)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.17(±0.4)</td>\n",
       "      <td>0.17(±0.4)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>392.83(±903.87)</td>\n",
       "      <td>392.83(±903.87)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>369.33(±849.8)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.18(±0.41)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.18(±0.41)</td>\n",
       "      <td>0.18(±0.41)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>378.5(±870.89)</td>\n",
       "      <td>378.5(±870.89)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>369.33(±849.8)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>334.17(±768.88)</td>\n",
       "      <td>334.17(±768.88)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>334.17(±768.88)</td>\n",
       "      <td>0.15(±0.35)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.15(±0.35)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>326.83(±752.01)</td>\n",
       "      <td>326.83(±752.01)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>326.83(±752.01)</td>\n",
       "      <td>0.15(±0.34)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.15(±0.34)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>369.33(±849.8)</td>\n",
       "      <td>369.33(±849.8)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>369.33(±849.8)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm        L90             NA50  \\\n",
       "Assembler                                                                       \n",
       "BCALM2               Single k-mer De Bruijn graph  0.0(±0.0)        0.0(±0.0)   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph  0.0(±0.0)   398.0(±915.76)   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  0.0(±0.0)  385.83(±887.76)   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph  0.0(±0.0)  392.83(±903.87)   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph  0.0(±0.0)   378.5(±870.89)   \n",
       "MINIA                Single k-mer De Bruijn graph  0.0(±0.0)        0.0(±0.0)   \n",
       "SKESA              Multiple k-mer De Bruijn graph  0.0(±0.0)  334.17(±768.88)   \n",
       "SPAdes             Multiple k-mer De Bruijn graph  0.0(±0.0)  326.83(±752.01)   \n",
       "Unicycler          Multiple k-mer De Bruijn graph  0.0(±0.0)   369.33(±849.8)   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph  0.0(±0.0)        0.0(±0.0)   \n",
       "\n",
       "                              NG50         Ns         Type        basepairs  \\\n",
       "Assembler                                                                     \n",
       "BCALM2                   0.0(±0.0)  0.0(±0.0)      Genomic        0.0(±0.0)   \n",
       "GATBMiniaPipeline   398.0(±915.76)  0.0(±0.0)  Metagenomic   369.33(±849.8)   \n",
       "IDBA-UD            385.83(±887.76)  0.0(±0.0)  Metagenomic   369.33(±849.8)   \n",
       "MEGAHIT            392.83(±903.87)  0.0(±0.0)  Metagenomic   369.33(±849.8)   \n",
       "metaSPAdes          378.5(±870.89)  0.0(±0.0)  Metagenomic   369.33(±849.8)   \n",
       "MINIA                    0.0(±0.0)  0.0(±0.0)      Genomic        0.0(±0.0)   \n",
       "SKESA              334.17(±768.88)  0.0(±0.0)      Genomic  334.17(±768.88)   \n",
       "SPAdes             326.83(±752.01)  0.0(±0.0)      Genomic  326.83(±752.01)   \n",
       "Unicycler           369.33(±849.8)  0.0(±0.0)      Genomic   369.33(±849.8)   \n",
       "VelvetOptimizer          0.0(±0.0)  0.0(±0.0)      Genomic        0.0(±0.0)   \n",
       "\n",
       "                  breadth_of_coverage      contigs   contiguity     identity  \\\n",
       "Assembler                                                                      \n",
       "BCALM2                      0.0(±0.0)    0.0(±0.0)    0.0(±0.0)    0.0(±0.0)   \n",
       "GATBMiniaPipeline         0.17(±0.38)  0.17(±0.38)  0.18(±0.41)  0.17(±0.38)   \n",
       "IDBA-UD                   0.17(±0.38)  0.17(±0.38)   0.17(±0.4)  0.17(±0.38)   \n",
       "MEGAHIT                   0.17(±0.38)  0.17(±0.38)  0.18(±0.41)  0.17(±0.38)   \n",
       "metaSPAdes                0.17(±0.38)  0.17(±0.38)  0.17(±0.39)  0.17(±0.38)   \n",
       "MINIA                       0.0(±0.0)    0.0(±0.0)    0.0(±0.0)    0.0(±0.0)   \n",
       "SKESA                     0.15(±0.35)  0.17(±0.38)  0.15(±0.35)  0.17(±0.38)   \n",
       "SPAdes                    0.15(±0.34)  0.17(±0.38)  0.15(±0.34)  0.17(±0.38)   \n",
       "Unicycler                 0.17(±0.38)  0.17(±0.38)  0.17(±0.38)  0.17(±0.38)   \n",
       "VelvetOptimizer             0.0(±0.0)    0.0(±0.0)    0.0(±0.0)    0.0(±0.0)   \n",
       "\n",
       "                  lowest_identity misassemblies multiplicity    parsimony  \\\n",
       "Assembler                                                                   \n",
       "BCALM2                  0.0(±0.0)     0.0(±0.0)    0.0(±0.0)    0.0(±0.0)   \n",
       "GATBMiniaPipeline     0.17(±0.38)     0.0(±0.0)  0.18(±0.41)  0.18(±0.41)   \n",
       "IDBA-UD               0.17(±0.38)     0.0(±0.0)   0.17(±0.4)   0.17(±0.4)   \n",
       "MEGAHIT               0.17(±0.38)     0.0(±0.0)  0.18(±0.41)  0.18(±0.41)   \n",
       "metaSPAdes            0.17(±0.38)     0.0(±0.0)  0.17(±0.39)  0.17(±0.39)   \n",
       "MINIA                   0.0(±0.0)     0.0(±0.0)    0.0(±0.0)    0.0(±0.0)   \n",
       "SKESA                 0.17(±0.38)     0.0(±0.0)  0.17(±0.38)  0.17(±0.38)   \n",
       "SPAdes                0.17(±0.38)     0.0(±0.0)  0.17(±0.38)  0.17(±0.38)   \n",
       "Unicycler             0.17(±0.38)     0.0(±0.0)  0.17(±0.38)  0.17(±0.38)   \n",
       "VelvetOptimizer         0.0(±0.0)     0.0(±0.0)    0.0(±0.0)    0.0(±0.0)   \n",
       "\n",
       "                      validity  \n",
       "Assembler                       \n",
       "BCALM2               0.0(±0.0)  \n",
       "GATBMiniaPipeline  0.17(±0.38)  \n",
       "IDBA-UD            0.17(±0.38)  \n",
       "MEGAHIT            0.17(±0.38)  \n",
       "metaSPAdes         0.17(±0.38)  \n",
       "MINIA                0.0(±0.0)  \n",
       "SKESA              0.17(±0.38)  \n",
       "SPAdes             0.17(±0.38)  \n",
       "Unicycler          0.17(±0.38)  \n",
       "VelvetOptimizer      0.0(±0.0)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Staphylococcus_aureus_plasmid3\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>526.22(±1210.78)</td>\n",
       "      <td>526.22(±1210.78)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>498.83(±1147.76)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.18(±0.4)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.18(±0.4)</td>\n",
       "      <td>0.18(±0.4)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>281.33(±647.32)</td>\n",
       "      <td>281.33(±647.32)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>459.5(±1057.26)</td>\n",
       "      <td>0.15(±0.35)</td>\n",
       "      <td>0.33(±0.77)</td>\n",
       "      <td>0.09(±0.22)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.17(±0.4)</td>\n",
       "      <td>0.17(±0.4)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>522.33(±1201.83)</td>\n",
       "      <td>522.33(±1201.83)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>498.83(±1147.76)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.4)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.17(±0.4)</td>\n",
       "      <td>0.17(±0.4)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>642.83(±1479.09)</td>\n",
       "      <td>642.83(±1479.09)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>498.83(±1147.76)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.21(±0.49)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.21(±0.49)</td>\n",
       "      <td>0.21(±0.49)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>463.67(±1066.85)</td>\n",
       "      <td>463.67(±1066.85)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>463.67(±1066.85)</td>\n",
       "      <td>0.15(±0.36)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.15(±0.36)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>498.83(±1147.76)</td>\n",
       "      <td>498.83(±1147.76)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>498.83(±1147.76)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm          L90  \\\n",
       "Assembler                                                        \n",
       "BCALM2               Single k-mer De Bruijn graph    0.0(±0.0)   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph    0.0(±0.0)   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  0.17(±0.38)   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph    0.0(±0.0)   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph    0.0(±0.0)   \n",
       "MINIA                Single k-mer De Bruijn graph    0.0(±0.0)   \n",
       "SKESA              Multiple k-mer De Bruijn graph    0.0(±0.0)   \n",
       "SPAdes             Multiple k-mer De Bruijn graph    0.0(±0.0)   \n",
       "Unicycler          Multiple k-mer De Bruijn graph    0.0(±0.0)   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph    0.0(±0.0)   \n",
       "\n",
       "                               NA50              NG50         Ns         Type  \\\n",
       "Assembler                                                                       \n",
       "BCALM2                    0.0(±0.0)         0.0(±0.0)  0.0(±0.0)      Genomic   \n",
       "GATBMiniaPipeline  526.22(±1210.78)  526.22(±1210.78)  0.0(±0.0)  Metagenomic   \n",
       "IDBA-UD             281.33(±647.32)   281.33(±647.32)  0.0(±0.0)  Metagenomic   \n",
       "MEGAHIT            522.33(±1201.83)  522.33(±1201.83)  0.0(±0.0)  Metagenomic   \n",
       "metaSPAdes         642.83(±1479.09)  642.83(±1479.09)  0.0(±0.0)  Metagenomic   \n",
       "MINIA                     0.0(±0.0)         0.0(±0.0)  0.0(±0.0)      Genomic   \n",
       "SKESA              463.67(±1066.85)  463.67(±1066.85)  0.0(±0.0)      Genomic   \n",
       "SPAdes                    0.0(±0.0)         0.0(±0.0)  0.0(±0.0)      Genomic   \n",
       "Unicycler          498.83(±1147.76)  498.83(±1147.76)  0.0(±0.0)      Genomic   \n",
       "VelvetOptimizer           0.0(±0.0)         0.0(±0.0)  0.0(±0.0)      Genomic   \n",
       "\n",
       "                          basepairs breadth_of_coverage      contigs  \\\n",
       "Assembler                                                              \n",
       "BCALM2                    0.0(±0.0)           0.0(±0.0)    0.0(±0.0)   \n",
       "GATBMiniaPipeline  498.83(±1147.76)         0.17(±0.38)  0.17(±0.38)   \n",
       "IDBA-UD             459.5(±1057.26)         0.15(±0.35)  0.33(±0.77)   \n",
       "MEGAHIT            498.83(±1147.76)         0.17(±0.38)  0.17(±0.38)   \n",
       "metaSPAdes         498.83(±1147.76)         0.17(±0.38)  0.17(±0.38)   \n",
       "MINIA                     0.0(±0.0)           0.0(±0.0)    0.0(±0.0)   \n",
       "SKESA              463.67(±1066.85)         0.15(±0.36)  0.17(±0.38)   \n",
       "SPAdes                    0.0(±0.0)           0.0(±0.0)    0.0(±0.0)   \n",
       "Unicycler          498.83(±1147.76)         0.17(±0.38)  0.17(±0.38)   \n",
       "VelvetOptimizer           0.0(±0.0)           0.0(±0.0)    0.0(±0.0)   \n",
       "\n",
       "                    contiguity     identity lowest_identity misassemblies  \\\n",
       "Assembler                                                                   \n",
       "BCALM2               0.0(±0.0)    0.0(±0.0)       0.0(±0.0)     0.0(±0.0)   \n",
       "GATBMiniaPipeline   0.18(±0.4)  0.17(±0.38)     0.17(±0.38)     0.0(±0.0)   \n",
       "IDBA-UD            0.09(±0.22)  0.17(±0.38)     0.17(±0.38)     0.0(±0.0)   \n",
       "MEGAHIT             0.17(±0.4)  0.17(±0.38)     0.17(±0.38)     0.0(±0.0)   \n",
       "metaSPAdes         0.21(±0.49)  0.17(±0.38)     0.17(±0.38)     0.0(±0.0)   \n",
       "MINIA                0.0(±0.0)    0.0(±0.0)       0.0(±0.0)     0.0(±0.0)   \n",
       "SKESA              0.15(±0.36)  0.17(±0.38)     0.17(±0.38)     0.0(±0.0)   \n",
       "SPAdes               0.0(±0.0)    0.0(±0.0)       0.0(±0.0)     0.0(±0.0)   \n",
       "Unicycler          0.17(±0.38)  0.17(±0.38)     0.17(±0.38)     0.0(±0.0)   \n",
       "VelvetOptimizer      0.0(±0.0)    0.0(±0.0)       0.0(±0.0)     0.0(±0.0)   \n",
       "\n",
       "                  multiplicity    parsimony     validity  \n",
       "Assembler                                                 \n",
       "BCALM2               0.0(±0.0)    0.0(±0.0)    0.0(±0.0)  \n",
       "GATBMiniaPipeline   0.18(±0.4)   0.18(±0.4)  0.17(±0.38)  \n",
       "IDBA-UD             0.17(±0.4)   0.17(±0.4)  0.17(±0.38)  \n",
       "MEGAHIT             0.17(±0.4)   0.17(±0.4)  0.17(±0.38)  \n",
       "metaSPAdes         0.21(±0.49)  0.21(±0.49)  0.17(±0.38)  \n",
       "MINIA                0.0(±0.0)    0.0(±0.0)    0.0(±0.0)  \n",
       "SKESA              0.17(±0.38)  0.17(±0.38)  0.17(±0.38)  \n",
       "SPAdes               0.0(±0.0)    0.0(±0.0)    0.0(±0.0)  \n",
       "Unicycler          0.17(±0.38)  0.17(±0.38)  0.17(±0.38)  \n",
       "VelvetOptimizer      0.0(±0.0)    0.0(±0.0)    0.0(±0.0)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "for reference in reference_pipeline_metrics_df.reference.unique():\n",
    "    ref_stats_per_reference_per_assembler = pd.DataFrame()\n",
    "    for assembler in reference_pipeline_metrics_df.assembler.unique():\n",
    "        row={'Assembler': assembler, \n",
    "             'Type':reference_pipeline_metrics_df[reference_pipeline_metrics_df.assembler == assembler].type.unique()[0],\n",
    "             'Algorithm': reference_pipeline_metrics_df[reference_pipeline_metrics_df.assembler == assembler].algorythm.unique()[0]}\n",
    "        lala=reference_pipeline_metrics_df[(reference_pipeline_metrics_df.assembler == assembler) & (reference_pipeline_metrics_df.reference == reference)].describe()\n",
    "        for column in lala.columns:\n",
    "            mean = lala.loc['mean',column]\n",
    "            std = lala.loc['std',column]\n",
    "            row[column] = \"{}(±{})\".format(round(mean, 2), round(std, 2))\n",
    "        ref_stats_per_reference_per_assembler=ref_stats_per_reference_per_assembler.append(row, ignore_index=True)\n",
    "    ref_stats_per_reference_per_assembler = ref_stats_per_reference_per_assembler.set_index('Assembler')\n",
    "    ref_stats_per_reference_per_assembler.to_csv(\"Tables/Results/{} - Reference metrics per assembler.csv\".format(reference))\n",
    "    print(reference)\n",
    "    display(ref_stats_per_reference_per_assembler)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reference stats per distribution"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Even\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>74.94(±299.94)</td>\n",
       "      <td>7322.22(±14723.55)</td>\n",
       "      <td>6909.42(±14616.47)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1290698.37(±1823393.22)</td>\n",
       "      <td>0.37(±0.42)</td>\n",
       "      <td>342.06(±543.67)</td>\n",
       "      <td>0.04(±0.17)</td>\n",
       "      <td>0.5(±0.5)</td>\n",
       "      <td>0.5(±0.5)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.5)</td>\n",
       "      <td>0.5(±0.5)</td>\n",
       "      <td>0.5(±0.5)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>29.53(±32.72)</td>\n",
       "      <td>109768.25(±118113.5)</td>\n",
       "      <td>100966.81(±103048.46)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2524991.72(±2137594.31)</td>\n",
       "      <td>0.87(±0.31)</td>\n",
       "      <td>57.58(±62.8)</td>\n",
       "      <td>0.25(±0.34)</td>\n",
       "      <td>0.89(±0.32)</td>\n",
       "      <td>0.86(±0.32)</td>\n",
       "      <td>0.25(±0.64)</td>\n",
       "      <td>0.89(±0.32)</td>\n",
       "      <td>0.89(±0.32)</td>\n",
       "      <td>0.89(±0.32)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>71.31(±84.58)</td>\n",
       "      <td>56068.1(±73615.0)</td>\n",
       "      <td>55809.71(±73485.54)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2508186.43(±2123775.36)</td>\n",
       "      <td>0.83(±0.34)</td>\n",
       "      <td>120.27(±138.59)</td>\n",
       "      <td>0.16(±0.27)</td>\n",
       "      <td>0.86(±0.35)</td>\n",
       "      <td>0.75(±0.38)</td>\n",
       "      <td>1.01(±2.14)</td>\n",
       "      <td>0.87(±0.35)</td>\n",
       "      <td>0.87(±0.35)</td>\n",
       "      <td>0.86(±0.35)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>17.47(±17.84)</td>\n",
       "      <td>144475.11(±140689.52)</td>\n",
       "      <td>151952.72(±144268.39)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2533615.8(±2145228.58)</td>\n",
       "      <td>0.88(±0.31)</td>\n",
       "      <td>44.06(±44.41)</td>\n",
       "      <td>0.3(±0.37)</td>\n",
       "      <td>0.89(±0.31)</td>\n",
       "      <td>0.79(±0.35)</td>\n",
       "      <td>1.08(±1.47)</td>\n",
       "      <td>0.89(±0.32)</td>\n",
       "      <td>0.9(±0.32)</td>\n",
       "      <td>0.89(±0.31)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>19.92(±22.85)</td>\n",
       "      <td>134941.08(±125388.71)</td>\n",
       "      <td>133991.47(±125026.18)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2529572.39(±2143529.81)</td>\n",
       "      <td>0.87(±0.31)</td>\n",
       "      <td>38.94(±42.47)</td>\n",
       "      <td>0.28(±0.34)</td>\n",
       "      <td>0.89(±0.32)</td>\n",
       "      <td>0.88(±0.31)</td>\n",
       "      <td>0.17(±0.56)</td>\n",
       "      <td>0.9(±0.32)</td>\n",
       "      <td>0.9(±0.32)</td>\n",
       "      <td>0.89(±0.32)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>82.95(±109.73)</td>\n",
       "      <td>18327.94(±18094.39)</td>\n",
       "      <td>17669.59(±17958.75)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2345473.31(±1977463.42)</td>\n",
       "      <td>0.72(±0.38)</td>\n",
       "      <td>306.45(±390.73)</td>\n",
       "      <td>0.08(±0.18)</td>\n",
       "      <td>0.81(±0.4)</td>\n",
       "      <td>0.79(±0.39)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.81(±0.4)</td>\n",
       "      <td>0.81(±0.4)</td>\n",
       "      <td>0.81(±0.4)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>38.81(±44.45)</td>\n",
       "      <td>84178.14(±88597.63)</td>\n",
       "      <td>83571.72(±88877.24)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2505379.86(±2122746.26)</td>\n",
       "      <td>0.85(±0.31)</td>\n",
       "      <td>69.67(±78.4)</td>\n",
       "      <td>0.23(±0.34)</td>\n",
       "      <td>0.89(±0.32)</td>\n",
       "      <td>0.77(±0.38)</td>\n",
       "      <td>0.33(±0.75)</td>\n",
       "      <td>0.89(±0.32)</td>\n",
       "      <td>0.89(±0.32)</td>\n",
       "      <td>0.88(±0.31)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>19.72(±20.76)</td>\n",
       "      <td>129438.44(±125261.5)</td>\n",
       "      <td>129237.0(±124835.36)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2524523.92(±2139592.08)</td>\n",
       "      <td>0.81(±0.37)</td>\n",
       "      <td>39.03(±39.76)</td>\n",
       "      <td>0.21(±0.26)</td>\n",
       "      <td>0.83(±0.37)</td>\n",
       "      <td>0.8(±0.38)</td>\n",
       "      <td>0.39(±1.01)</td>\n",
       "      <td>0.83(±0.38)</td>\n",
       "      <td>0.84(±0.38)</td>\n",
       "      <td>0.83(±0.37)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>15.64(±17.26)</td>\n",
       "      <td>161424.75(±140701.13)</td>\n",
       "      <td>158207.14(±139990.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2521523.86(±2137646.62)</td>\n",
       "      <td>0.87(±0.31)</td>\n",
       "      <td>32.81(±33.23)</td>\n",
       "      <td>0.29(±0.32)</td>\n",
       "      <td>0.89(±0.32)</td>\n",
       "      <td>0.86(±0.33)</td>\n",
       "      <td>0.17(±0.56)</td>\n",
       "      <td>0.89(±0.32)</td>\n",
       "      <td>0.89(±0.32)</td>\n",
       "      <td>0.89(±0.32)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>31.03(±55.05)</td>\n",
       "      <td>28318.97(±64328.77)</td>\n",
       "      <td>33659.58(±87573.52)</td>\n",
       "      <td>977.56(±2622.81)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1609811.42(±2029242.0)</td>\n",
       "      <td>0.5(±0.47)</td>\n",
       "      <td>170.75(±339.12)</td>\n",
       "      <td>0.07(±0.18)</td>\n",
       "      <td>0.53(±0.5)</td>\n",
       "      <td>0.5(±0.48)</td>\n",
       "      <td>0.94(±2.27)</td>\n",
       "      <td>0.53(±0.5)</td>\n",
       "      <td>0.53(±0.5)</td>\n",
       "      <td>0.53(±0.5)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm             L90  \\\n",
       "Assembler                                                           \n",
       "BCALM2               Single k-mer De Bruijn graph  74.94(±299.94)   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph   29.53(±32.72)   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph   71.31(±84.58)   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph   17.47(±17.84)   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph   19.92(±22.85)   \n",
       "MINIA                Single k-mer De Bruijn graph  82.95(±109.73)   \n",
       "SKESA              Multiple k-mer De Bruijn graph   38.81(±44.45)   \n",
       "SPAdes             Multiple k-mer De Bruijn graph   19.72(±20.76)   \n",
       "Unicycler          Multiple k-mer De Bruijn graph   15.64(±17.26)   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph   31.03(±55.05)   \n",
       "\n",
       "                                    NA50                   NG50  \\\n",
       "Assembler                                                         \n",
       "BCALM2                7322.22(±14723.55)     6909.42(±14616.47)   \n",
       "GATBMiniaPipeline   109768.25(±118113.5)  100966.81(±103048.46)   \n",
       "IDBA-UD                56068.1(±73615.0)    55809.71(±73485.54)   \n",
       "MEGAHIT            144475.11(±140689.52)  151952.72(±144268.39)   \n",
       "metaSPAdes         134941.08(±125388.71)  133991.47(±125026.18)   \n",
       "MINIA                18327.94(±18094.39)    17669.59(±17958.75)   \n",
       "SKESA                84178.14(±88597.63)    83571.72(±88877.24)   \n",
       "SPAdes              129438.44(±125261.5)   129237.0(±124835.36)   \n",
       "Unicycler          161424.75(±140701.13)   158207.14(±139990.0)   \n",
       "VelvetOptimizer      28318.97(±64328.77)    33659.58(±87573.52)   \n",
       "\n",
       "                                 Ns         Type                basepairs  \\\n",
       "Assembler                                                                   \n",
       "BCALM2                    0.0(±0.0)      Genomic  1290698.37(±1823393.22)   \n",
       "GATBMiniaPipeline         0.0(±0.0)  Metagenomic  2524991.72(±2137594.31)   \n",
       "IDBA-UD                   0.0(±0.0)  Metagenomic  2508186.43(±2123775.36)   \n",
       "MEGAHIT                   0.0(±0.0)  Metagenomic   2533615.8(±2145228.58)   \n",
       "metaSPAdes                0.0(±0.0)  Metagenomic  2529572.39(±2143529.81)   \n",
       "MINIA                     0.0(±0.0)      Genomic  2345473.31(±1977463.42)   \n",
       "SKESA                     0.0(±0.0)      Genomic  2505379.86(±2122746.26)   \n",
       "SPAdes                    0.0(±0.0)      Genomic  2524523.92(±2139592.08)   \n",
       "Unicycler                 0.0(±0.0)      Genomic  2521523.86(±2137646.62)   \n",
       "VelvetOptimizer    977.56(±2622.81)      Genomic   1609811.42(±2029242.0)   \n",
       "\n",
       "                  breadth_of_coverage          contigs   contiguity  \\\n",
       "Assembler                                                             \n",
       "BCALM2                    0.37(±0.42)  342.06(±543.67)  0.04(±0.17)   \n",
       "GATBMiniaPipeline         0.87(±0.31)     57.58(±62.8)  0.25(±0.34)   \n",
       "IDBA-UD                   0.83(±0.34)  120.27(±138.59)  0.16(±0.27)   \n",
       "MEGAHIT                   0.88(±0.31)    44.06(±44.41)   0.3(±0.37)   \n",
       "metaSPAdes                0.87(±0.31)    38.94(±42.47)  0.28(±0.34)   \n",
       "MINIA                     0.72(±0.38)  306.45(±390.73)  0.08(±0.18)   \n",
       "SKESA                     0.85(±0.31)     69.67(±78.4)  0.23(±0.34)   \n",
       "SPAdes                    0.81(±0.37)    39.03(±39.76)  0.21(±0.26)   \n",
       "Unicycler                 0.87(±0.31)    32.81(±33.23)  0.29(±0.32)   \n",
       "VelvetOptimizer            0.5(±0.47)  170.75(±339.12)  0.07(±0.18)   \n",
       "\n",
       "                      identity lowest_identity misassemblies multiplicity  \\\n",
       "Assembler                                                                   \n",
       "BCALM2               0.5(±0.5)       0.5(±0.5)     0.0(±0.0)    0.5(±0.5)   \n",
       "GATBMiniaPipeline  0.89(±0.32)     0.86(±0.32)   0.25(±0.64)  0.89(±0.32)   \n",
       "IDBA-UD            0.86(±0.35)     0.75(±0.38)   1.01(±2.14)  0.87(±0.35)   \n",
       "MEGAHIT            0.89(±0.31)     0.79(±0.35)   1.08(±1.47)  0.89(±0.32)   \n",
       "metaSPAdes         0.89(±0.32)     0.88(±0.31)   0.17(±0.56)   0.9(±0.32)   \n",
       "MINIA               0.81(±0.4)     0.79(±0.39)     0.0(±0.0)   0.81(±0.4)   \n",
       "SKESA              0.89(±0.32)     0.77(±0.38)   0.33(±0.75)  0.89(±0.32)   \n",
       "SPAdes             0.83(±0.37)      0.8(±0.38)   0.39(±1.01)  0.83(±0.38)   \n",
       "Unicycler          0.89(±0.32)     0.86(±0.33)   0.17(±0.56)  0.89(±0.32)   \n",
       "VelvetOptimizer     0.53(±0.5)      0.5(±0.48)   0.94(±2.27)   0.53(±0.5)   \n",
       "\n",
       "                     parsimony     validity  \n",
       "Assembler                                    \n",
       "BCALM2               0.5(±0.5)    0.5(±0.5)  \n",
       "GATBMiniaPipeline  0.89(±0.32)  0.89(±0.32)  \n",
       "IDBA-UD            0.87(±0.35)  0.86(±0.35)  \n",
       "MEGAHIT             0.9(±0.32)  0.89(±0.31)  \n",
       "metaSPAdes          0.9(±0.32)  0.89(±0.32)  \n",
       "MINIA               0.81(±0.4)   0.81(±0.4)  \n",
       "SKESA              0.89(±0.32)  0.88(±0.31)  \n",
       "SPAdes             0.84(±0.38)  0.83(±0.37)  \n",
       "Unicycler          0.89(±0.32)  0.89(±0.32)  \n",
       "VelvetOptimizer     0.53(±0.5)   0.53(±0.5)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>13.58(±55.31)</td>\n",
       "      <td>3455.14(±12284.0)</td>\n",
       "      <td>3263.39(±11938.57)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>390196.06(±1313282.67)</td>\n",
       "      <td>0.09(±0.27)</td>\n",
       "      <td>35.86(±117.16)</td>\n",
       "      <td>0.0(±0.01)</td>\n",
       "      <td>0.17(±0.37)</td>\n",
       "      <td>0.17(±0.37)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.17(±0.37)</td>\n",
       "      <td>0.17(±0.37)</td>\n",
       "      <td>0.17(±0.37)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>23.63(±78.89)</td>\n",
       "      <td>41249.75(±94968.71)</td>\n",
       "      <td>40867.61(±94913.24)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1131019.19(±2120393.96)</td>\n",
       "      <td>0.25(±0.43)</td>\n",
       "      <td>37.59(±107.82)</td>\n",
       "      <td>0.02(±0.06)</td>\n",
       "      <td>0.36(±0.48)</td>\n",
       "      <td>0.36(±0.48)</td>\n",
       "      <td>0.08(±0.36)</td>\n",
       "      <td>0.36(±0.48)</td>\n",
       "      <td>0.36(±0.48)</td>\n",
       "      <td>0.36(±0.48)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>16.19(±49.73)</td>\n",
       "      <td>48013.69(±107273.76)</td>\n",
       "      <td>46436.22(±107328.74)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1135523.26(±2127500.35)</td>\n",
       "      <td>0.25(±0.43)</td>\n",
       "      <td>29.56(±76.34)</td>\n",
       "      <td>0.02(±0.06)</td>\n",
       "      <td>0.46(±0.49)</td>\n",
       "      <td>0.4(±0.47)</td>\n",
       "      <td>0.28(±0.65)</td>\n",
       "      <td>0.47(±0.5)</td>\n",
       "      <td>0.61(±0.99)</td>\n",
       "      <td>0.44(±0.49)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>8.42(±25.38)</td>\n",
       "      <td>66599.11(±136443.17)</td>\n",
       "      <td>66455.53(±136838.51)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1141250.22(±2137239.42)</td>\n",
       "      <td>0.25(±0.43)</td>\n",
       "      <td>18.11(±41.85)</td>\n",
       "      <td>0.03(±0.08)</td>\n",
       "      <td>0.47(±0.49)</td>\n",
       "      <td>0.43(±0.47)</td>\n",
       "      <td>0.06(±0.23)</td>\n",
       "      <td>0.47(±0.5)</td>\n",
       "      <td>0.48(±0.51)</td>\n",
       "      <td>0.47(±0.5)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>7.19(±15.94)</td>\n",
       "      <td>60717.78(±131029.85)</td>\n",
       "      <td>59453.89(±129745.62)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1140704.31(±2136193.13)</td>\n",
       "      <td>0.25(±0.43)</td>\n",
       "      <td>15.53(±29.22)</td>\n",
       "      <td>0.03(±0.07)</td>\n",
       "      <td>0.47(±0.5)</td>\n",
       "      <td>0.43(±0.48)</td>\n",
       "      <td>0.17(±0.44)</td>\n",
       "      <td>0.47(±0.5)</td>\n",
       "      <td>0.48(±0.51)</td>\n",
       "      <td>0.47(±0.5)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>65.15(±149.9)</td>\n",
       "      <td>6260.97(±13386.39)</td>\n",
       "      <td>5788.35(±13420.91)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1099755.82(±2067843.02)</td>\n",
       "      <td>0.24(±0.41)</td>\n",
       "      <td>119.49(±242.52)</td>\n",
       "      <td>0.01(±0.01)</td>\n",
       "      <td>0.42(±0.5)</td>\n",
       "      <td>0.39(±0.49)</td>\n",
       "      <td>0.14(±0.48)</td>\n",
       "      <td>0.42(±0.5)</td>\n",
       "      <td>0.42(±0.5)</td>\n",
       "      <td>0.42(±0.49)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>6.72(±38.29)</td>\n",
       "      <td>10421.97(±57890.12)</td>\n",
       "      <td>10210.64(±57920.98)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>366940.28(±948388.34)</td>\n",
       "      <td>0.1(±0.26)</td>\n",
       "      <td>120.94(±387.9)</td>\n",
       "      <td>0.01(±0.03)</td>\n",
       "      <td>0.19(±0.4)</td>\n",
       "      <td>0.19(±0.39)</td>\n",
       "      <td>0.33(±1.98)</td>\n",
       "      <td>0.19(±0.4)</td>\n",
       "      <td>0.19(±0.4)</td>\n",
       "      <td>0.19(±0.4)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>30.92(±138.3)</td>\n",
       "      <td>36829.5(±85794.72)</td>\n",
       "      <td>37693.47(±90157.5)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1093453.36(±2096483.13)</td>\n",
       "      <td>0.23(±0.41)</td>\n",
       "      <td>105.28(±363.08)</td>\n",
       "      <td>0.02(±0.05)</td>\n",
       "      <td>0.56(±0.48)</td>\n",
       "      <td>0.49(±0.47)</td>\n",
       "      <td>16.17(±93.38)</td>\n",
       "      <td>0.6(±0.51)</td>\n",
       "      <td>0.63(±0.55)</td>\n",
       "      <td>0.57(±0.48)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>7.72(±24.54)</td>\n",
       "      <td>48388.56(±107839.89)</td>\n",
       "      <td>45410.11(±107403.42)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>963355.97(±2011952.12)</td>\n",
       "      <td>0.2(±0.39)</td>\n",
       "      <td>14.22(±35.55)</td>\n",
       "      <td>0.03(±0.06)</td>\n",
       "      <td>0.22(±0.42)</td>\n",
       "      <td>0.22(±0.41)</td>\n",
       "      <td>0.08(±0.5)</td>\n",
       "      <td>0.22(±0.42)</td>\n",
       "      <td>0.22(±0.42)</td>\n",
       "      <td>0.22(±0.42)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm            L90  \\\n",
       "Assembler                                                          \n",
       "BCALM2               Single k-mer De Bruijn graph  13.58(±55.31)   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph  23.63(±78.89)   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  16.19(±49.73)   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph   8.42(±25.38)   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph   7.19(±15.94)   \n",
       "MINIA                Single k-mer De Bruijn graph  65.15(±149.9)   \n",
       "SKESA              Multiple k-mer De Bruijn graph   6.72(±38.29)   \n",
       "SPAdes             Multiple k-mer De Bruijn graph  30.92(±138.3)   \n",
       "Unicycler          Multiple k-mer De Bruijn graph   7.72(±24.54)   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph      0.0(±0.0)   \n",
       "\n",
       "                                   NA50                  NG50         Ns  \\\n",
       "Assembler                                                                  \n",
       "BCALM2                3455.14(±12284.0)    3263.39(±11938.57)  0.0(±0.0)   \n",
       "GATBMiniaPipeline   41249.75(±94968.71)   40867.61(±94913.24)  0.0(±0.0)   \n",
       "IDBA-UD            48013.69(±107273.76)  46436.22(±107328.74)  0.0(±0.0)   \n",
       "MEGAHIT            66599.11(±136443.17)  66455.53(±136838.51)  0.0(±0.0)   \n",
       "metaSPAdes         60717.78(±131029.85)  59453.89(±129745.62)  0.0(±0.0)   \n",
       "MINIA                6260.97(±13386.39)    5788.35(±13420.91)  0.0(±0.0)   \n",
       "SKESA               10421.97(±57890.12)   10210.64(±57920.98)  0.0(±0.0)   \n",
       "SPAdes               36829.5(±85794.72)    37693.47(±90157.5)  0.0(±0.0)   \n",
       "Unicycler          48388.56(±107839.89)  45410.11(±107403.42)  0.0(±0.0)   \n",
       "VelvetOptimizer               0.0(±0.0)             0.0(±0.0)  0.0(±0.0)   \n",
       "\n",
       "                          Type                basepairs breadth_of_coverage  \\\n",
       "Assembler                                                                     \n",
       "BCALM2                 Genomic   390196.06(±1313282.67)         0.09(±0.27)   \n",
       "GATBMiniaPipeline  Metagenomic  1131019.19(±2120393.96)         0.25(±0.43)   \n",
       "IDBA-UD            Metagenomic  1135523.26(±2127500.35)         0.25(±0.43)   \n",
       "MEGAHIT            Metagenomic  1141250.22(±2137239.42)         0.25(±0.43)   \n",
       "metaSPAdes         Metagenomic  1140704.31(±2136193.13)         0.25(±0.43)   \n",
       "MINIA                  Genomic  1099755.82(±2067843.02)         0.24(±0.41)   \n",
       "SKESA                  Genomic    366940.28(±948388.34)          0.1(±0.26)   \n",
       "SPAdes                 Genomic  1093453.36(±2096483.13)         0.23(±0.41)   \n",
       "Unicycler              Genomic   963355.97(±2011952.12)          0.2(±0.39)   \n",
       "VelvetOptimizer        Genomic                0.0(±0.0)           0.0(±0.0)   \n",
       "\n",
       "                           contigs   contiguity     identity lowest_identity  \\\n",
       "Assembler                                                                      \n",
       "BCALM2              35.86(±117.16)   0.0(±0.01)  0.17(±0.37)     0.17(±0.37)   \n",
       "GATBMiniaPipeline   37.59(±107.82)  0.02(±0.06)  0.36(±0.48)     0.36(±0.48)   \n",
       "IDBA-UD              29.56(±76.34)  0.02(±0.06)  0.46(±0.49)      0.4(±0.47)   \n",
       "MEGAHIT              18.11(±41.85)  0.03(±0.08)  0.47(±0.49)     0.43(±0.47)   \n",
       "metaSPAdes           15.53(±29.22)  0.03(±0.07)   0.47(±0.5)     0.43(±0.48)   \n",
       "MINIA              119.49(±242.52)  0.01(±0.01)   0.42(±0.5)     0.39(±0.49)   \n",
       "SKESA               120.94(±387.9)  0.01(±0.03)   0.19(±0.4)     0.19(±0.39)   \n",
       "SPAdes             105.28(±363.08)  0.02(±0.05)  0.56(±0.48)     0.49(±0.47)   \n",
       "Unicycler            14.22(±35.55)  0.03(±0.06)  0.22(±0.42)     0.22(±0.41)   \n",
       "VelvetOptimizer          0.0(±0.0)    0.0(±0.0)    0.0(±0.0)       0.0(±0.0)   \n",
       "\n",
       "                   misassemblies multiplicity    parsimony     validity  \n",
       "Assembler                                                                \n",
       "BCALM2                 0.0(±0.0)  0.17(±0.37)  0.17(±0.37)  0.17(±0.37)  \n",
       "GATBMiniaPipeline    0.08(±0.36)  0.36(±0.48)  0.36(±0.48)  0.36(±0.48)  \n",
       "IDBA-UD              0.28(±0.65)   0.47(±0.5)  0.61(±0.99)  0.44(±0.49)  \n",
       "MEGAHIT              0.06(±0.23)   0.47(±0.5)  0.48(±0.51)   0.47(±0.5)  \n",
       "metaSPAdes           0.17(±0.44)   0.47(±0.5)  0.48(±0.51)   0.47(±0.5)  \n",
       "MINIA                0.14(±0.48)   0.42(±0.5)   0.42(±0.5)  0.42(±0.49)  \n",
       "SKESA                0.33(±1.98)   0.19(±0.4)   0.19(±0.4)   0.19(±0.4)  \n",
       "SPAdes             16.17(±93.38)   0.6(±0.51)  0.63(±0.55)  0.57(±0.48)  \n",
       "Unicycler             0.08(±0.5)  0.22(±0.42)  0.22(±0.42)  0.22(±0.42)  \n",
       "VelvetOptimizer        0.0(±0.0)    0.0(±0.0)    0.0(±0.0)    0.0(±0.0)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "for distribution in reference_pipeline_metrics_df.distribution.unique():\n",
    "    ref_stats_per_reference_per_assembler = pd.DataFrame()\n",
    "    for assembler in reference_pipeline_metrics_df.assembler.unique():\n",
    "        row={'Assembler': assembler, \n",
    "             'Type':reference_pipeline_metrics_df[reference_pipeline_metrics_df.assembler == assembler].type.unique()[0],\n",
    "             'Algorithm': reference_pipeline_metrics_df[reference_pipeline_metrics_df.assembler == assembler].algorythm.unique()[0]}\n",
    "        lala=reference_pipeline_metrics_df[(reference_pipeline_metrics_df.assembler == assembler) & (reference_pipeline_metrics_df.distribution == distribution)].describe()\n",
    "        for column in lala.columns:\n",
    "            mean = lala.loc['mean',column]\n",
    "            std = lala.loc['std',column]\n",
    "            row[column] = \"{}(±{})\".format(round(mean, 2), round(std, 2))\n",
    "        ref_stats_per_reference_per_assembler=ref_stats_per_reference_per_assembler.append(row, ignore_index=True)\n",
    "    ref_stats_per_reference_per_assembler = ref_stats_per_reference_per_assembler.set_index('Assembler')\n",
    "    ref_stats_per_reference_per_assembler.to_csv(\"Tables/Results/{} - Reference metrics per assembler.csv\".format(distribution))\n",
    "    print(distribution)\n",
    "    display(ref_stats_per_reference_per_assembler)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
