{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Results Tables"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The success of an assembly is evaluated by the computation of metrics in two defined ways: globally through statistics inherent to the complete set of sequences that were assembled, and relative to the replicons present in the sample. \n",
    "\n",
    "The computation of the global metrics is done through custom python code from the complete set of sequences assembled. \n",
    "\n",
    "The following metrics are computed for the complete and filtered set of assembled sequences, restricted to contigs of length above a specified minimum size: \n",
    "\n",
    "- **Contig sizes**\n",
    "    - **Contigs:** The total number of contigs in the assembly;\n",
    "    - **Basepairs:** The total number of bases in the assembly;\n",
    "    - **Maximum sequence length:** The length of the largest contig in the assembly.\n",
    "    - **Number of ‘N’s:** Number of uncalled bases (N's) \n",
    "- **Contiguity**\n",
    "    - **Nx (where 0  < x  ⩽ 100):** Length for which the collection of all assembled sequences of that length or longer in an assembly covers at least a given percentage of the total length of the assembly\n",
    "- **Misassembly**\n",
    "    - **Misassemblies** - Number of aligned contigs that contain a misassembly event\n",
    "\n",
    "After filtering, the sequences are mapped with the reference replicons and the metrics are computed through custom python code. The metrics are calculated, filtered for a minimum length, for each replicon in the file provided input references. \n",
    "\n",
    "- **Contig sizes**\n",
    "    - **Contigs:** The total number of contigs in the assembly;\n",
    "    - **Basepairs:** The total number of bases in the assembly;\n",
    "    - **Number of ‘N’s:** Number of uncalled bases (N's) \n",
    "- **COMPASS**\n",
    "    - **(Breadth of) Coverage:** Ratio of covered sequence on the reference by aligned contigs;\n",
    "    - **Multiplicity:** Ratio of the length of alignable assembled sequence to covered sequence on the reference;\n",
    "    - **Validity:** Ratio of the length of the alignable assembled sequence to total basepairs in the aligned contigs;\n",
    "    - **Parsimony:** Cost of the assembly (multiplicity over validity);\n",
    "- **Contiguity**\n",
    "    - **Contiguity:** longest single alignment between the assembly and the reference, relative to the reference length;\n",
    "    - **NAx (where 0  < x  ⩽ 100):** Length for which the collection of aligned assembled sequences of that length or longer in an assembly covers at least a given percentage of the total length of the reference replicon;\n",
    "    - **NGx (where 0  < x  ⩽ 100):** Length for which the collection of aligned contigs of that length or longer covers at least a given percentage of the sequence of the reference.\n",
    "    - **Lx (where 0  < x  ⩽ 100):** Minimal number of contigs that cover x % of the sequence of the reference;\n",
    "- **Identity**\n",
    "    - **Identity:** Ratio of identical basepairs in all aligned contigs to the reference;\n",
    "    - **Lowest identity:** Identity of the lowest scoring contig to the reference.\n",
    "    - **PLS**: Phred-like score per contig, per assembler.\n",
    "- **Misassembly**\n",
    "    - **Misassemblies** - Number of aligned contigs that contain a misassembly event\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "from plotly.offline import plot\n",
    "import glob\n",
    "import fnmatch\n",
    "import plotly.graph_objects as go\n",
    "from plotly.subplots import make_subplots\n",
    "import json\n",
    "import pandas as pd\n",
    "from itertools import groupby\n",
    "import csv\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Global variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "METRICS_COLUMNS_GLOBAL = ['run','sample','assembler', 'contigs', 'basepairs', 'max_contig', 'Ns', 'n50', \n",
    "                          'misassemblies', 'mapped_reads', 'filtered_contigs', 'filtered_basepairs',\n",
    "                          'filtered_n50', 'filtered_Ns', 'type', 'algorythm']\n",
    "METRICS_COLUMNS_REFERENCE = ['run','sample','assembler', 'reference', 'contiguity', 'breadth_of_coverage', 'multiplicity', 'validity', \n",
    "                             'parsimony', 'identity', 'lowest_identity', 'L90', 'contigs', 'NA50', 'NG50', 'basepairs', 'Ns',\n",
    "                             'misassemblies', 'type', 'algorythm']\n",
    "\n",
    "ASSEMBLER_PROCESS_LIST = [\"BCALM2\", \"GATBMINIAPIPELINE\", \"MINIA\", \"MEGAHIT\", \"METASPADES\", \"UNICYCLER\", \"SPADES\",\n",
    "                          \"SKESA\", \"VELVETOPTIMIZER\", \"IDBA\"]\n",
    "PROCESS_TO_NAME = {\"BCALM2\": \"BCALM2\", \n",
    "                   \"GATBMINIAPIPELINE\": \"GATBMiniaPipeline\",\n",
    "                   \"MINIA\": \"MINIA\",\n",
    "                   \"MEGAHIT\": \"MEGAHIT\", \n",
    "                   \"METASPADES\": \"metaSPAdes\", \n",
    "                   \"UNICYCLER\": \"Unicycler\", \n",
    "                   \"SPADES\": \"SPAdes\",\n",
    "                    \"SKESA\": \"SKESA\",\n",
    "                   \"VELVETOPTIMIZER\": \"VelvetOptimizer\",\n",
    "                   \"IDBA\": \"IDBA-UD\"}\n",
    "\n",
    "log_distributed = ['LHS', 'LNN', 'ERR2935805']\n",
    "even_distribution = ['EMS', 'LNN', 'ERR2984773']\n",
    "\n",
    "genomic_assemblers = ['BCALM2', 'MINIA', 'SKESA', 'SPAdes', 'Unicycler', 'VelvetOptimizer']\n",
    "metagenomic_assemblers = ['GATBMiniaPipeline', 'IDBA-UD', 'MEGAHIT', 'metaSPAdes']\n",
    "single_kmer = ['BCALM2', 'MINIA']\n",
    "multiple_kmer = ['SKESA', 'SPAdes', 'Unicycler', 'VelvetOptimizer', 'GATBMiniaPipeline', \n",
    "                 'IDBA-UD', 'MEGAHIT', 'metaSPAdes']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Global metrics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pipeline_report_tables.json data from run3...\n",
      "Processing pipeline_report_tables.json data from run2...\n"
     ]
    }
   ],
   "source": [
    "report_glob = glob.glob('../Results/*/report/pipeline_report_tables.json')\n",
    "global_pipeline_metrics_df = pd.DataFrame(columns=METRICS_COLUMNS_GLOBAL)\n",
    "\n",
    "for pipeline_report_file in report_glob:\n",
    "    report_file_name = pipeline_report_file.split('/')[-1]\n",
    "    stats_run = pipeline_report_file.split('/')[-3]\n",
    "    print('Processing {0} data from {1}...'.format(report_file_name, stats_run))\n",
    "    \n",
    "    with open(pipeline_report_file) as _fh:\n",
    "        json_report = json.load(_fh)\n",
    "        for sample in json_report.keys():\n",
    "            for line in json_report[sample]['GlobalTable']:\n",
    "                assembler = line['assembler']\n",
    "                global_pipeline_metrics_df = global_pipeline_metrics_df.append({'run': stats_run,\n",
    "                                                                                'sample': sample,\n",
    "                                                                                'assembler': line['assembler'],\n",
    "                                                                                'contigs': int(line['original']['contigs']),\n",
    "                                                                                'basepairs': int(line['original']['basepairs']),\n",
    "                                                                                'max_contig': int(line['original']['max_contig_size']),\n",
    "                                                                                'n50': int(line['original']['N50']),\n",
    "                                                                                'mapped_reads': line['original']['mapped_reads'],\n",
    "                                                                                'Ns': int(line['original']['Ns']),\n",
    "                                                                                'misassemblies': line['filtered']['misassembled_contigs'],\n",
    "                                                                                'filtered_contigs': line['filtered']['contigs'],\n",
    "                                                                                'filtered_basepairs': line['filtered']['basepairs'],\n",
    "                                                                                'filtered_n50': line['filtered']['N50'],\n",
    "                                                                                'filtered_Ns': line['filtered']['Ns']\n",
    "                                                                               },\n",
    "                                                                               ignore_index=True)\n",
    "global_pipeline_metrics_df['distribution'] = np.where(global_pipeline_metrics_df['sample'].isin(log_distributed), 'Log', 'Even')\n",
    "global_pipeline_metrics_df['type'] = np.where(global_pipeline_metrics_df['assembler'].isin(genomic_assemblers), 'Genomic', 'Metagenomic')\n",
    "global_pipeline_metrics_df['algorythm'] = np.where(global_pipeline_metrics_df['assembler'].isin(single_kmer), 'Single k-mer De Bruijn graph', 'Multiple k-mer De Bruijn graph')\n",
    "global_pipeline_metrics_df[['contigs','basepairs','max_contig', 'Ns','n50','misassemblies']] = global_pipeline_metrics_df[['contigs','basepairs','max_contig', 'Ns','n50','misassemblies']].apply(pd.to_numeric)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Preview data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "run                    object\n",
       "sample                 object\n",
       "assembler              object\n",
       "contigs                 int64\n",
       "basepairs               int64\n",
       "max_contig              int64\n",
       "Ns                      int64\n",
       "n50                     int64\n",
       "misassemblies           int64\n",
       "mapped_reads          float64\n",
       "filtered_contigs       object\n",
       "filtered_basepairs     object\n",
       "filtered_n50           object\n",
       "filtered_Ns            object\n",
       "type                   object\n",
       "algorythm              object\n",
       "distribution           object\n",
       "dtype: object"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "global_pipeline_metrics_df.dtypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>run</th>\n",
       "      <th>sample</th>\n",
       "      <th>assembler</th>\n",
       "      <th>contigs</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>max_contig</th>\n",
       "      <th>Ns</th>\n",
       "      <th>n50</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>mapped_reads</th>\n",
       "      <th>filtered_contigs</th>\n",
       "      <th>filtered_basepairs</th>\n",
       "      <th>filtered_n50</th>\n",
       "      <th>filtered_Ns</th>\n",
       "      <th>type</th>\n",
       "      <th>algorythm</th>\n",
       "      <th>distribution</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>run3</td>\n",
       "      <td>ENN</td>\n",
       "      <td>BCALM2</td>\n",
       "      <td>25258</td>\n",
       "      <td>31040696</td>\n",
       "      <td>161812</td>\n",
       "      <td>0</td>\n",
       "      <td>17766</td>\n",
       "      <td>0</td>\n",
       "      <td>96.832836</td>\n",
       "      <td>3542</td>\n",
       "      <td>28188554</td>\n",
       "      <td>21029</td>\n",
       "      <td>0</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>Even</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>run3</td>\n",
       "      <td>ENN</td>\n",
       "      <td>GATBMiniaPipeline</td>\n",
       "      <td>1008</td>\n",
       "      <td>30445418</td>\n",
       "      <td>632215</td>\n",
       "      <td>0</td>\n",
       "      <td>106963</td>\n",
       "      <td>0</td>\n",
       "      <td>99.480265</td>\n",
       "      <td>727</td>\n",
       "      <td>30282639</td>\n",
       "      <td>106963</td>\n",
       "      <td>0</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>Even</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>run3</td>\n",
       "      <td>ENN</td>\n",
       "      <td>IDBA-UD</td>\n",
       "      <td>1825</td>\n",
       "      <td>30390287</td>\n",
       "      <td>468605</td>\n",
       "      <td>0</td>\n",
       "      <td>61778</td>\n",
       "      <td>12</td>\n",
       "      <td>99.414283</td>\n",
       "      <td>1279</td>\n",
       "      <td>30136999</td>\n",
       "      <td>62331</td>\n",
       "      <td>0</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>Even</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>run3</td>\n",
       "      <td>ENN</td>\n",
       "      <td>MEGAHIT</td>\n",
       "      <td>810</td>\n",
       "      <td>30541296</td>\n",
       "      <td>632274</td>\n",
       "      <td>0</td>\n",
       "      <td>170162</td>\n",
       "      <td>9</td>\n",
       "      <td>99.731417</td>\n",
       "      <td>517</td>\n",
       "      <td>30415305</td>\n",
       "      <td>171359</td>\n",
       "      <td>0</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>Even</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>run3</td>\n",
       "      <td>ENN</td>\n",
       "      <td>metaSPAdes</td>\n",
       "      <td>633</td>\n",
       "      <td>30448137</td>\n",
       "      <td>1036942</td>\n",
       "      <td>0</td>\n",
       "      <td>234919</td>\n",
       "      <td>1</td>\n",
       "      <td>99.596146</td>\n",
       "      <td>350</td>\n",
       "      <td>30376419</td>\n",
       "      <td>234919</td>\n",
       "      <td>0</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>Even</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>run2</td>\n",
       "      <td>LHS</td>\n",
       "      <td>MINIA</td>\n",
       "      <td>24652</td>\n",
       "      <td>15334650</td>\n",
       "      <td>163543</td>\n",
       "      <td>0</td>\n",
       "      <td>7727</td>\n",
       "      <td>0</td>\n",
       "      <td>91.099517</td>\n",
       "      <td>2171</td>\n",
       "      <td>12920930</td>\n",
       "      <td>10582</td>\n",
       "      <td>0</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>Log</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116</th>\n",
       "      <td>run2</td>\n",
       "      <td>LHS</td>\n",
       "      <td>SKESA</td>\n",
       "      <td>36</td>\n",
       "      <td>2957060</td>\n",
       "      <td>632013</td>\n",
       "      <td>0</td>\n",
       "      <td>351027</td>\n",
       "      <td>0</td>\n",
       "      <td>92.228441</td>\n",
       "      <td>21</td>\n",
       "      <td>2950393</td>\n",
       "      <td>351027</td>\n",
       "      <td>0</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>Log</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117</th>\n",
       "      <td>run2</td>\n",
       "      <td>LHS</td>\n",
       "      <td>SPAdes</td>\n",
       "      <td>31225</td>\n",
       "      <td>16224766</td>\n",
       "      <td>263645</td>\n",
       "      <td>0</td>\n",
       "      <td>46478</td>\n",
       "      <td>1</td>\n",
       "      <td>95.503607</td>\n",
       "      <td>1526</td>\n",
       "      <td>13088051</td>\n",
       "      <td>69361</td>\n",
       "      <td>0</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>Log</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>run2</td>\n",
       "      <td>LHS</td>\n",
       "      <td>Unicycler</td>\n",
       "      <td>276</td>\n",
       "      <td>13345058</td>\n",
       "      <td>632062</td>\n",
       "      <td>0</td>\n",
       "      <td>187479</td>\n",
       "      <td>3</td>\n",
       "      <td>96.948570</td>\n",
       "      <td>247</td>\n",
       "      <td>13333597</td>\n",
       "      <td>187479</td>\n",
       "      <td>0</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>Log</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>119</th>\n",
       "      <td>run2</td>\n",
       "      <td>LHS</td>\n",
       "      <td>VelvetOptimizer</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>Log</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>120 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      run sample          assembler  contigs  basepairs  max_contig  Ns  \\\n",
       "0    run3    ENN             BCALM2    25258   31040696      161812   0   \n",
       "1    run3    ENN  GATBMiniaPipeline     1008   30445418      632215   0   \n",
       "2    run3    ENN            IDBA-UD     1825   30390287      468605   0   \n",
       "3    run3    ENN            MEGAHIT      810   30541296      632274   0   \n",
       "4    run3    ENN         metaSPAdes      633   30448137     1036942   0   \n",
       "..    ...    ...                ...      ...        ...         ...  ..   \n",
       "115  run2    LHS              MINIA    24652   15334650      163543   0   \n",
       "116  run2    LHS              SKESA       36    2957060      632013   0   \n",
       "117  run2    LHS             SPAdes    31225   16224766      263645   0   \n",
       "118  run2    LHS          Unicycler      276   13345058      632062   0   \n",
       "119  run2    LHS    VelvetOptimizer        0          0           0   0   \n",
       "\n",
       "        n50  misassemblies  mapped_reads filtered_contigs filtered_basepairs  \\\n",
       "0     17766              0     96.832836             3542           28188554   \n",
       "1    106963              0     99.480265              727           30282639   \n",
       "2     61778             12     99.414283             1279           30136999   \n",
       "3    170162              9     99.731417              517           30415305   \n",
       "4    234919              1     99.596146              350           30376419   \n",
       "..      ...            ...           ...              ...                ...   \n",
       "115    7727              0     91.099517             2171           12920930   \n",
       "116  351027              0     92.228441               21            2950393   \n",
       "117   46478              1     95.503607             1526           13088051   \n",
       "118  187479              3     96.948570              247           13333597   \n",
       "119       0              0      0.000000                0                  0   \n",
       "\n",
       "    filtered_n50 filtered_Ns         type                       algorythm  \\\n",
       "0          21029           0      Genomic    Single k-mer De Bruijn graph   \n",
       "1         106963           0  Metagenomic  Multiple k-mer De Bruijn graph   \n",
       "2          62331           0  Metagenomic  Multiple k-mer De Bruijn graph   \n",
       "3         171359           0  Metagenomic  Multiple k-mer De Bruijn graph   \n",
       "4         234919           0  Metagenomic  Multiple k-mer De Bruijn graph   \n",
       "..           ...         ...          ...                             ...   \n",
       "115        10582           0      Genomic    Single k-mer De Bruijn graph   \n",
       "116       351027           0      Genomic  Multiple k-mer De Bruijn graph   \n",
       "117        69361           0      Genomic  Multiple k-mer De Bruijn graph   \n",
       "118       187479           0      Genomic  Multiple k-mer De Bruijn graph   \n",
       "119            0           0      Genomic  Multiple k-mer De Bruijn graph   \n",
       "\n",
       "    distribution  \n",
       "0           Even  \n",
       "1           Even  \n",
       "2           Even  \n",
       "3           Even  \n",
       "4           Even  \n",
       "..           ...  \n",
       "115          Log  \n",
       "116          Log  \n",
       "117          Log  \n",
       "118          Log  \n",
       "119          Log  \n",
       "\n",
       "[120 rows x 17 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "global_pipeline_metrics_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Save data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "global_pipeline_metrics_df.to_csv(\"Tables/Results/Global metrics.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Global statistics per assembler type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "stats_per_assembler_type = pd.DataFrame()\n",
    "for assembly_type in global_pipeline_metrics_df.type.unique():\n",
    "    row={'Assembler Type': assembly_type}\n",
    "    lala=global_pipeline_metrics_df[global_pipeline_metrics_df.type == assembly_type].describe()\n",
    "    for column in lala.columns:\n",
    "        mean = lala.loc['mean',column]\n",
    "        std = lala.loc['std',column]\n",
    "        row[column] = \"{}(±{})\".format(round(mean, 2), round(std, 2))\n",
    "    stats_per_assembler_type=stats_per_assembler_type.append(row, ignore_index=True)\n",
    "stats_per_assembler_type = stats_per_assembler_type.set_index('Assembler Type')\n",
    "stats_per_assembler_type.to_csv(\"Tables/Results/Global metrics per assembler type.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ns</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>contigs</th>\n",
       "      <th>mapped_reads</th>\n",
       "      <th>max_contig</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>n50</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler Type</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Genomic</th>\n",
       "      <td>632.49(±2696.83)</td>\n",
       "      <td>45049830.26(±77416335.87)</td>\n",
       "      <td>584820.46(±1927506.27)</td>\n",
       "      <td>74.63(±36.42)</td>\n",
       "      <td>407391.93(±369952.0)</td>\n",
       "      <td>18.17(±94.55)</td>\n",
       "      <td>75438.93(±96288.64)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Metagenomic</th>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>25547301.35(±10011582.38)</td>\n",
       "      <td>8085.02(±13528.83)</td>\n",
       "      <td>94.46(±6.1)</td>\n",
       "      <td>706300.71(±216738.75)</td>\n",
       "      <td>4.06(±4.6)</td>\n",
       "      <td>116760.52(±64826.32)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              Ns                  basepairs  \\\n",
       "Assembler Type                                                \n",
       "Genomic         632.49(±2696.83)  45049830.26(±77416335.87)   \n",
       "Metagenomic            0.0(±0.0)  25547301.35(±10011582.38)   \n",
       "\n",
       "                               contigs   mapped_reads             max_contig  \\\n",
       "Assembler Type                                                                 \n",
       "Genomic         584820.46(±1927506.27)  74.63(±36.42)   407391.93(±369952.0)   \n",
       "Metagenomic         8085.02(±13528.83)    94.46(±6.1)  706300.71(±216738.75)   \n",
       "\n",
       "                misassemblies                   n50  \n",
       "Assembler Type                                       \n",
       "Genomic         18.17(±94.55)   75438.93(±96288.64)  \n",
       "Metagenomic        4.06(±4.6)  116760.52(±64826.32)  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "stats_per_assembler_type"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Global stats per assembly algorithm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "stats_per_assembler_algo = pd.DataFrame()\n",
    "for assembly_algo in global_pipeline_metrics_df.algorythm.unique():\n",
    "    row={'Assembler Algorithm': assembly_algo}\n",
    "    lala=global_pipeline_metrics_df[global_pipeline_metrics_df.algorythm == assembly_algo].describe()\n",
    "    for column in lala.columns:\n",
    "        mean = lala.loc['mean',column]\n",
    "        std = lala.loc['std',column]\n",
    "        row[column] = \"{}(±{})\".format(round(mean, 2), round(std, 2))\n",
    "    stats_per_assembler_algo=stats_per_assembler_algo.append(row, ignore_index=True)\n",
    "stats_per_assembler_algo = stats_per_assembler_algo.set_index('Assembler Algorithm')\n",
    "stats_per_assembler_algo.to_csv(\"Tables/Results/Global metrics per assembler algorithm.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ns</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>contigs</th>\n",
       "      <th>mapped_reads</th>\n",
       "      <th>max_contig</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>n50</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler Algorithm</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Single k-mer De Bruijn graph</th>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>91615808.0(±121102826.02)</td>\n",
       "      <td>1691066.88(±3086055.78)</td>\n",
       "      <td>68.91(±38.55)</td>\n",
       "      <td>108171.83(±74665.4)</td>\n",
       "      <td>0.33(±0.76)</td>\n",
       "      <td>10206.33(±9056.65)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Multiple k-mer De Bruijn graph</th>\n",
       "      <td>474.36(±2347.62)</td>\n",
       "      <td>23657071.38(±12649027.02)</td>\n",
       "      <td>19891.14(±86129.94)</td>\n",
       "      <td>85.97(±26.66)</td>\n",
       "      <td>631651.34(±309648.34)</td>\n",
       "      <td>15.57(±81.88)</td>\n",
       "      <td>112407.88(±85994.72)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              Ns                  basepairs  \\\n",
       "Assembler Algorithm                                                           \n",
       "Single k-mer De Bruijn graph           0.0(±0.0)  91615808.0(±121102826.02)   \n",
       "Multiple k-mer De Bruijn graph  474.36(±2347.62)  23657071.38(±12649027.02)   \n",
       "\n",
       "                                                contigs   mapped_reads  \\\n",
       "Assembler Algorithm                                                      \n",
       "Single k-mer De Bruijn graph    1691066.88(±3086055.78)  68.91(±38.55)   \n",
       "Multiple k-mer De Bruijn graph      19891.14(±86129.94)  85.97(±26.66)   \n",
       "\n",
       "                                           max_contig  misassemblies  \\\n",
       "Assembler Algorithm                                                    \n",
       "Single k-mer De Bruijn graph      108171.83(±74665.4)    0.33(±0.76)   \n",
       "Multiple k-mer De Bruijn graph  631651.34(±309648.34)  15.57(±81.88)   \n",
       "\n",
       "                                                 n50  \n",
       "Assembler Algorithm                                   \n",
       "Single k-mer De Bruijn graph      10206.33(±9056.65)  \n",
       "Multiple k-mer De Bruijn graph  112407.88(±85994.72)  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "stats_per_assembler_algo"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Global stats per assembler"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "stats_per_assembler = pd.DataFrame()\n",
    "for assembler in global_pipeline_metrics_df.assembler.unique():\n",
    "    row={'Assembler': assembler, \n",
    "         'Type':global_pipeline_metrics_df[global_pipeline_metrics_df.assembler == assembler].type.unique()[0],\n",
    "         'Algorithm': global_pipeline_metrics_df[global_pipeline_metrics_df.assembler == assembler].algorythm.unique()[0]}\n",
    "    lala=global_pipeline_metrics_df[global_pipeline_metrics_df.assembler == assembler].describe()\n",
    "    for column in lala.columns:\n",
    "        mean = lala.loc['mean',column]\n",
    "        std = lala.loc['std',column]\n",
    "        row[column] = \"{}(±{})\".format(round(mean, 2), round(std, 2))\n",
    "    stats_per_assembler=stats_per_assembler.append(row, ignore_index=True)\n",
    "stats_per_assembler = stats_per_assembler.set_index('Assembler')\n",
    "stats_per_assembler.to_csv(\"Tables/Results/Global metrics per assembler.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>contigs</th>\n",
       "      <th>mapped_reads</th>\n",
       "      <th>max_contig</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>n50</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>157403460.17(±145361404.71)</td>\n",
       "      <td>3352423.83(±3727041.83)</td>\n",
       "      <td>46.33(±44.2)</td>\n",
       "      <td>58728.17(±77129.19)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>6220.67(±8873.1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>23074937.42(±8661768.88)</td>\n",
       "      <td>2629.42(±2589.73)</td>\n",
       "      <td>94.65(±6.52)</td>\n",
       "      <td>640091.33(±68954.44)</td>\n",
       "      <td>1.92(±2.07)</td>\n",
       "      <td>102424.0(±26843.99)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>25378779.83(±9698785.87)</td>\n",
       "      <td>6832.17(±7856.47)</td>\n",
       "      <td>93.22(±6.21)</td>\n",
       "      <td>515095.83(±121784.53)</td>\n",
       "      <td>6.17(±4.73)</td>\n",
       "      <td>67858.42(±30786.39)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>25840461.5(±10055256.9)</td>\n",
       "      <td>6528.5(±8264.11)</td>\n",
       "      <td>94.98(±6.19)</td>\n",
       "      <td>847862.5(±228686.73)</td>\n",
       "      <td>6.33(±6.49)</td>\n",
       "      <td>147830.83(±47776.33)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>27895026.67(±12052231.75)</td>\n",
       "      <td>16350.0(±23107.65)</td>\n",
       "      <td>94.97(±6.09)</td>\n",
       "      <td>822153.17(±218316.11)</td>\n",
       "      <td>1.83(±0.94)</td>\n",
       "      <td>148928.83(±94815.17)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>25828155.83(±9600619.33)</td>\n",
       "      <td>29709.92(±19566.17)</td>\n",
       "      <td>91.49(±6.43)</td>\n",
       "      <td>157615.5(±19336.61)</td>\n",
       "      <td>0.67(±0.98)</td>\n",
       "      <td>14192.0(±7623.07)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>18472455.67(±12490360.65)</td>\n",
       "      <td>3671.5(±4090.4)</td>\n",
       "      <td>90.6(±8.18)</td>\n",
       "      <td>390219.83(±263956.92)</td>\n",
       "      <td>3.33(±4.46)</td>\n",
       "      <td>101495.67(±123485.57)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>28138604.33(±10873664.65)</td>\n",
       "      <td>19436.5(±17375.62)</td>\n",
       "      <td>94.21(±6.49)</td>\n",
       "      <td>697457.83(±218426.88)</td>\n",
       "      <td>98.0(±221.9)</td>\n",
       "      <td>107840.17(±85499.81)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>21030768.0(±9911805.52)</td>\n",
       "      <td>531.33(±330.3)</td>\n",
       "      <td>94.07(±6.84)</td>\n",
       "      <td>857592.0(±234205.23)</td>\n",
       "      <td>1.33(±1.44)</td>\n",
       "      <td>212762.33(±22226.12)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>3794.92(±5818.15)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>19425537.58(±22071793.34)</td>\n",
       "      <td>103149.67(±232595.04)</td>\n",
       "      <td>31.06(±45.27)</td>\n",
       "      <td>282738.25(±417128.51)</td>\n",
       "      <td>5.67(±8.39)</td>\n",
       "      <td>10122.75(±14919.83)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm                 Ns  \\\n",
       "Assembler                                                              \n",
       "BCALM2               Single k-mer De Bruijn graph          0.0(±0.0)   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph          0.0(±0.0)   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph          0.0(±0.0)   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph          0.0(±0.0)   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph          0.0(±0.0)   \n",
       "MINIA                Single k-mer De Bruijn graph          0.0(±0.0)   \n",
       "SKESA              Multiple k-mer De Bruijn graph          0.0(±0.0)   \n",
       "SPAdes             Multiple k-mer De Bruijn graph          0.0(±0.0)   \n",
       "Unicycler          Multiple k-mer De Bruijn graph          0.0(±0.0)   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph  3794.92(±5818.15)   \n",
       "\n",
       "                          Type                    basepairs  \\\n",
       "Assembler                                                     \n",
       "BCALM2                 Genomic  157403460.17(±145361404.71)   \n",
       "GATBMiniaPipeline  Metagenomic     23074937.42(±8661768.88)   \n",
       "IDBA-UD            Metagenomic     25378779.83(±9698785.87)   \n",
       "MEGAHIT            Metagenomic      25840461.5(±10055256.9)   \n",
       "metaSPAdes         Metagenomic    27895026.67(±12052231.75)   \n",
       "MINIA                  Genomic     25828155.83(±9600619.33)   \n",
       "SKESA                  Genomic    18472455.67(±12490360.65)   \n",
       "SPAdes                 Genomic    28138604.33(±10873664.65)   \n",
       "Unicycler              Genomic      21030768.0(±9911805.52)   \n",
       "VelvetOptimizer        Genomic    19425537.58(±22071793.34)   \n",
       "\n",
       "                                   contigs   mapped_reads  \\\n",
       "Assembler                                                   \n",
       "BCALM2             3352423.83(±3727041.83)   46.33(±44.2)   \n",
       "GATBMiniaPipeline        2629.42(±2589.73)   94.65(±6.52)   \n",
       "IDBA-UD                  6832.17(±7856.47)   93.22(±6.21)   \n",
       "MEGAHIT                   6528.5(±8264.11)   94.98(±6.19)   \n",
       "metaSPAdes              16350.0(±23107.65)   94.97(±6.09)   \n",
       "MINIA                  29709.92(±19566.17)   91.49(±6.43)   \n",
       "SKESA                      3671.5(±4090.4)    90.6(±8.18)   \n",
       "SPAdes                  19436.5(±17375.62)   94.21(±6.49)   \n",
       "Unicycler                   531.33(±330.3)   94.07(±6.84)   \n",
       "VelvetOptimizer      103149.67(±232595.04)  31.06(±45.27)   \n",
       "\n",
       "                              max_contig misassemblies                    n50  \n",
       "Assembler                                                                      \n",
       "BCALM2               58728.17(±77129.19)     0.0(±0.0)       6220.67(±8873.1)  \n",
       "GATBMiniaPipeline   640091.33(±68954.44)   1.92(±2.07)    102424.0(±26843.99)  \n",
       "IDBA-UD            515095.83(±121784.53)   6.17(±4.73)    67858.42(±30786.39)  \n",
       "MEGAHIT             847862.5(±228686.73)   6.33(±6.49)   147830.83(±47776.33)  \n",
       "metaSPAdes         822153.17(±218316.11)   1.83(±0.94)   148928.83(±94815.17)  \n",
       "MINIA                157615.5(±19336.61)   0.67(±0.98)      14192.0(±7623.07)  \n",
       "SKESA              390219.83(±263956.92)   3.33(±4.46)  101495.67(±123485.57)  \n",
       "SPAdes             697457.83(±218426.88)  98.0(±221.9)   107840.17(±85499.81)  \n",
       "Unicycler           857592.0(±234205.23)   1.33(±1.44)   212762.33(±22226.12)  \n",
       "VelvetOptimizer    282738.25(±417128.51)   5.67(±8.39)    10122.75(±14919.83)  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "stats_per_assembler"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Global stats per distribution"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Even\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>contigs</th>\n",
       "      <th>mapped_reads</th>\n",
       "      <th>max_contig</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>n50</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>81485532.0(±69082486.61)</td>\n",
       "      <td>1286040.67(±1726341.64)</td>\n",
       "      <td>57.29(±40.35)</td>\n",
       "      <td>61507.0(±78246.58)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>6186.0(±8975.2)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>31254576.33(±1205888.55)</td>\n",
       "      <td>2439.5(±2292.43)</td>\n",
       "      <td>93.87(±8.31)</td>\n",
       "      <td>677707.0(±70298.6)</td>\n",
       "      <td>2.67(±2.73)</td>\n",
       "      <td>110537.33(±9542.15)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>33523229.67(±4845847.19)</td>\n",
       "      <td>7943.0(±9486.87)</td>\n",
       "      <td>90.89(±7.11)</td>\n",
       "      <td>413513.0(±85348.16)</td>\n",
       "      <td>9.67(±4.41)</td>\n",
       "      <td>49048.33(±21762.56)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>34099350.67(±5474162.25)</td>\n",
       "      <td>7463.33(±10186.7)</td>\n",
       "      <td>94.35(±7.85)</td>\n",
       "      <td>868834.0(±268046.42)</td>\n",
       "      <td>12.0(±3.9)</td>\n",
       "      <td>143276.33(±42581.65)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>36624891.67(±9555633.92)</td>\n",
       "      <td>21002.33(±31475.51)</td>\n",
       "      <td>94.28(±7.71)</td>\n",
       "      <td>847661.67(±290003.13)</td>\n",
       "      <td>2.0(±0.89)</td>\n",
       "      <td>158678.0(±91558.71)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>34158353.67(±5443602.16)</td>\n",
       "      <td>33771.5(±26508.92)</td>\n",
       "      <td>90.39(±7.82)</td>\n",
       "      <td>147189.67(±22652.82)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>14485.0(±5722.72)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>30285330.33(±115680.54)</td>\n",
       "      <td>1219.0(±199.11)</td>\n",
       "      <td>92.53(±8.74)</td>\n",
       "      <td>540293.0(±83135.61)</td>\n",
       "      <td>2.67(±2.07)</td>\n",
       "      <td>85120.33(±16926.29)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>35828349.33(±8294764.4)</td>\n",
       "      <td>13949.33(±20031.01)</td>\n",
       "      <td>94.27(±7.78)</td>\n",
       "      <td>813593.33(±52845.57)</td>\n",
       "      <td>4.33(±4.41)</td>\n",
       "      <td>158629.67(±83121.53)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>30462753.0(±99796.14)</td>\n",
       "      <td>833.67(±138.96)</td>\n",
       "      <td>93.41(±8.6)</td>\n",
       "      <td>1011262.0(±227172.01)</td>\n",
       "      <td>1.67(±1.37)</td>\n",
       "      <td>215168.67(±26920.81)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>7589.83(±6317.33)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>38851075.17(±12888650.42)</td>\n",
       "      <td>206299.33(±305753.73)</td>\n",
       "      <td>62.13(±46.83)</td>\n",
       "      <td>565476.5(±436959.29)</td>\n",
       "      <td>11.33(±8.82)</td>\n",
       "      <td>20245.5(±15613.92)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm                 Ns  \\\n",
       "Assembler                                                              \n",
       "BCALM2               Single k-mer De Bruijn graph          0.0(±0.0)   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph          0.0(±0.0)   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph          0.0(±0.0)   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph          0.0(±0.0)   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph          0.0(±0.0)   \n",
       "MINIA                Single k-mer De Bruijn graph          0.0(±0.0)   \n",
       "SKESA              Multiple k-mer De Bruijn graph          0.0(±0.0)   \n",
       "SPAdes             Multiple k-mer De Bruijn graph          0.0(±0.0)   \n",
       "Unicycler          Multiple k-mer De Bruijn graph          0.0(±0.0)   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph  7589.83(±6317.33)   \n",
       "\n",
       "                          Type                  basepairs  \\\n",
       "Assembler                                                   \n",
       "BCALM2                 Genomic   81485532.0(±69082486.61)   \n",
       "GATBMiniaPipeline  Metagenomic   31254576.33(±1205888.55)   \n",
       "IDBA-UD            Metagenomic   33523229.67(±4845847.19)   \n",
       "MEGAHIT            Metagenomic   34099350.67(±5474162.25)   \n",
       "metaSPAdes         Metagenomic   36624891.67(±9555633.92)   \n",
       "MINIA                  Genomic   34158353.67(±5443602.16)   \n",
       "SKESA                  Genomic    30285330.33(±115680.54)   \n",
       "SPAdes                 Genomic    35828349.33(±8294764.4)   \n",
       "Unicycler              Genomic      30462753.0(±99796.14)   \n",
       "VelvetOptimizer        Genomic  38851075.17(±12888650.42)   \n",
       "\n",
       "                                   contigs   mapped_reads  \\\n",
       "Assembler                                                   \n",
       "BCALM2             1286040.67(±1726341.64)  57.29(±40.35)   \n",
       "GATBMiniaPipeline         2439.5(±2292.43)   93.87(±8.31)   \n",
       "IDBA-UD                   7943.0(±9486.87)   90.89(±7.11)   \n",
       "MEGAHIT                  7463.33(±10186.7)   94.35(±7.85)   \n",
       "metaSPAdes             21002.33(±31475.51)   94.28(±7.71)   \n",
       "MINIA                   33771.5(±26508.92)   90.39(±7.82)   \n",
       "SKESA                      1219.0(±199.11)   92.53(±8.74)   \n",
       "SPAdes                 13949.33(±20031.01)   94.27(±7.78)   \n",
       "Unicycler                  833.67(±138.96)    93.41(±8.6)   \n",
       "VelvetOptimizer      206299.33(±305753.73)  62.13(±46.83)   \n",
       "\n",
       "                              max_contig misassemblies                   n50  \n",
       "Assembler                                                                     \n",
       "BCALM2                61507.0(±78246.58)     0.0(±0.0)       6186.0(±8975.2)  \n",
       "GATBMiniaPipeline     677707.0(±70298.6)   2.67(±2.73)   110537.33(±9542.15)  \n",
       "IDBA-UD              413513.0(±85348.16)   9.67(±4.41)   49048.33(±21762.56)  \n",
       "MEGAHIT             868834.0(±268046.42)    12.0(±3.9)  143276.33(±42581.65)  \n",
       "metaSPAdes         847661.67(±290003.13)    2.0(±0.89)   158678.0(±91558.71)  \n",
       "MINIA               147189.67(±22652.82)     0.0(±0.0)     14485.0(±5722.72)  \n",
       "SKESA                540293.0(±83135.61)   2.67(±2.07)   85120.33(±16926.29)  \n",
       "SPAdes              813593.33(±52845.57)   4.33(±4.41)  158629.67(±83121.53)  \n",
       "Unicycler          1011262.0(±227172.01)   1.67(±1.37)  215168.67(±26920.81)  \n",
       "VelvetOptimizer     565476.5(±436959.29)  11.33(±8.82)    20245.5(±15613.92)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>contigs</th>\n",
       "      <th>mapped_reads</th>\n",
       "      <th>max_contig</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>n50</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>233321388.33(±166976037.91)</td>\n",
       "      <td>5418807.0(±4163140.62)</td>\n",
       "      <td>35.37(±48.8)</td>\n",
       "      <td>55949.33(±83345.94)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>6255.33(±9625.66)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>14895298.5(±1740108.73)</td>\n",
       "      <td>2819.33(±3068.06)</td>\n",
       "      <td>95.43(±4.81)</td>\n",
       "      <td>602475.67(±46071.98)</td>\n",
       "      <td>1.17(±0.75)</td>\n",
       "      <td>94310.67(±36555.23)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>17234330.0(±4926154.48)</td>\n",
       "      <td>5721.33(±6544.52)</td>\n",
       "      <td>95.56(±4.6)</td>\n",
       "      <td>616678.67(±24065.69)</td>\n",
       "      <td>2.67(±0.52)</td>\n",
       "      <td>86668.5(±27611.4)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>17581572.33(±5363708.41)</td>\n",
       "      <td>5593.67(±6662.12)</td>\n",
       "      <td>95.61(±4.66)</td>\n",
       "      <td>826891.0(±205305.52)</td>\n",
       "      <td>0.67(±0.52)</td>\n",
       "      <td>152385.33(±56202.22)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>19165161.67(±6734121.74)</td>\n",
       "      <td>11697.67(±11491.98)</td>\n",
       "      <td>95.66(±4.59)</td>\n",
       "      <td>796644.67(±138537.99)</td>\n",
       "      <td>1.67(±1.03)</td>\n",
       "      <td>139179.67(±105672.65)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>17497958.0(±2569841.72)</td>\n",
       "      <td>25648.33(±9996.14)</td>\n",
       "      <td>92.59(±5.2)</td>\n",
       "      <td>168041.33(±6968.79)</td>\n",
       "      <td>1.33(±1.03)</td>\n",
       "      <td>13899.0(±9741.1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>6659581.0(±2881152.41)</td>\n",
       "      <td>6124.0(±4725.89)</td>\n",
       "      <td>88.66(±7.85)</td>\n",
       "      <td>240146.67(±303837.03)</td>\n",
       "      <td>4.0(±6.2)</td>\n",
       "      <td>117871.0(±180601.9)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>20448859.33(±7028523.19)</td>\n",
       "      <td>24923.67(±13809.57)</td>\n",
       "      <td>94.15(±5.68)</td>\n",
       "      <td>581322.33(±264197.12)</td>\n",
       "      <td>191.67(±295.38)</td>\n",
       "      <td>57050.67(±54610.55)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>11598783.0(±1617707.48)</td>\n",
       "      <td>229.0(±36.54)</td>\n",
       "      <td>94.74(±5.29)</td>\n",
       "      <td>703922.0(±111322.71)</td>\n",
       "      <td>1.0(±1.55)</td>\n",
       "      <td>210356.0(±18659.42)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm         Ns         Type  \\\n",
       "Assembler                                                                   \n",
       "BCALM2               Single k-mer De Bruijn graph  0.0(±0.0)      Genomic   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph  0.0(±0.0)  Metagenomic   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  0.0(±0.0)  Metagenomic   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph  0.0(±0.0)  Metagenomic   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph  0.0(±0.0)  Metagenomic   \n",
       "MINIA                Single k-mer De Bruijn graph  0.0(±0.0)      Genomic   \n",
       "SKESA              Multiple k-mer De Bruijn graph  0.0(±0.0)      Genomic   \n",
       "SPAdes             Multiple k-mer De Bruijn graph  0.0(±0.0)      Genomic   \n",
       "Unicycler          Multiple k-mer De Bruijn graph  0.0(±0.0)      Genomic   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph  0.0(±0.0)      Genomic   \n",
       "\n",
       "                                     basepairs                 contigs  \\\n",
       "Assembler                                                                \n",
       "BCALM2             233321388.33(±166976037.91)  5418807.0(±4163140.62)   \n",
       "GATBMiniaPipeline      14895298.5(±1740108.73)       2819.33(±3068.06)   \n",
       "IDBA-UD                17234330.0(±4926154.48)       5721.33(±6544.52)   \n",
       "MEGAHIT               17581572.33(±5363708.41)       5593.67(±6662.12)   \n",
       "metaSPAdes            19165161.67(±6734121.74)     11697.67(±11491.98)   \n",
       "MINIA                  17497958.0(±2569841.72)      25648.33(±9996.14)   \n",
       "SKESA                   6659581.0(±2881152.41)        6124.0(±4725.89)   \n",
       "SPAdes                20448859.33(±7028523.19)     24923.67(±13809.57)   \n",
       "Unicycler              11598783.0(±1617707.48)           229.0(±36.54)   \n",
       "VelvetOptimizer                      0.0(±0.0)               0.0(±0.0)   \n",
       "\n",
       "                   mapped_reads             max_contig    misassemblies  \\\n",
       "Assembler                                                                 \n",
       "BCALM2             35.37(±48.8)    55949.33(±83345.94)        0.0(±0.0)   \n",
       "GATBMiniaPipeline  95.43(±4.81)   602475.67(±46071.98)      1.17(±0.75)   \n",
       "IDBA-UD             95.56(±4.6)   616678.67(±24065.69)      2.67(±0.52)   \n",
       "MEGAHIT            95.61(±4.66)   826891.0(±205305.52)      0.67(±0.52)   \n",
       "metaSPAdes         95.66(±4.59)  796644.67(±138537.99)      1.67(±1.03)   \n",
       "MINIA               92.59(±5.2)    168041.33(±6968.79)      1.33(±1.03)   \n",
       "SKESA              88.66(±7.85)  240146.67(±303837.03)        4.0(±6.2)   \n",
       "SPAdes             94.15(±5.68)  581322.33(±264197.12)  191.67(±295.38)   \n",
       "Unicycler          94.74(±5.29)   703922.0(±111322.71)       1.0(±1.55)   \n",
       "VelvetOptimizer       0.0(±0.0)              0.0(±0.0)        0.0(±0.0)   \n",
       "\n",
       "                                     n50  \n",
       "Assembler                                 \n",
       "BCALM2                 6255.33(±9625.66)  \n",
       "GATBMiniaPipeline    94310.67(±36555.23)  \n",
       "IDBA-UD                86668.5(±27611.4)  \n",
       "MEGAHIT             152385.33(±56202.22)  \n",
       "metaSPAdes         139179.67(±105672.65)  \n",
       "MINIA                   13899.0(±9741.1)  \n",
       "SKESA                117871.0(±180601.9)  \n",
       "SPAdes               57050.67(±54610.55)  \n",
       "Unicycler            210356.0(±18659.42)  \n",
       "VelvetOptimizer                0.0(±0.0)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "for distribution in global_pipeline_metrics_df.distribution.unique():\n",
    "    stats_per_distribution = pd.DataFrame()\n",
    "    for assembler in global_pipeline_metrics_df.assembler.unique():\n",
    "        row={'Assembler': assembler, \n",
    "             'Type':global_pipeline_metrics_df[(global_pipeline_metrics_df.assembler == assembler) & (global_pipeline_metrics_df.distribution == distribution)].type.unique()[0],\n",
    "             'Algorithm': global_pipeline_metrics_df[global_pipeline_metrics_df.assembler == assembler].algorythm.unique()[0]}\n",
    "        lala=global_pipeline_metrics_df[(global_pipeline_metrics_df.assembler == assembler) & (global_pipeline_metrics_df.distribution == distribution)].describe()\n",
    "        for column in lala.columns:\n",
    "            mean = lala.loc['mean',column]\n",
    "            std = lala.loc['std',column]\n",
    "            row[column] = \"{}(±{})\".format(round(mean, 2), round(std, 2))\n",
    "        stats_per_distribution=stats_per_distribution.append(row, ignore_index=True)\n",
    "    stats_per_distribution = stats_per_distribution.set_index('Assembler')\n",
    "    stats_per_distribution.to_csv(\"Tables/Results/{} - Global metrics per assembler.csv\".format(distribution))\n",
    "    print(distribution)\n",
    "    display(stats_per_distribution)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Global metrics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pipeline_report_tables.json data from run3...\n",
      "Processing pipeline_report_tables.json data from run2...\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>run</th>\n",
       "      <th>sample</th>\n",
       "      <th>assembler</th>\n",
       "      <th>reference</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>validity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>identity</th>\n",
       "      <th>...</th>\n",
       "      <th>L90</th>\n",
       "      <th>contigs</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>Ns</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>type</th>\n",
       "      <th>algorythm</th>\n",
       "      <th>distribution</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>run3</td>\n",
       "      <td>ENN</td>\n",
       "      <td>BCALM2</td>\n",
       "      <td>Bacillus_subtilis</td>\n",
       "      <td>0.032648</td>\n",
       "      <td>0.976835</td>\n",
       "      <td>1.000279</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000279</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>66</td>\n",
       "      <td>116</td>\n",
       "      <td>65138</td>\n",
       "      <td>64934</td>\n",
       "      <td>3951958</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>Even</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>run3</td>\n",
       "      <td>ENN</td>\n",
       "      <td>GATBMiniaPipeline</td>\n",
       "      <td>Bacillus_subtilis</td>\n",
       "      <td>0.097833</td>\n",
       "      <td>0.983819</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.999966</td>\n",
       "      <td>...</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>269695</td>\n",
       "      <td>263366</td>\n",
       "      <td>3980213</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>Even</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>run3</td>\n",
       "      <td>ENN</td>\n",
       "      <td>IDBA-UD</td>\n",
       "      <td>Bacillus_subtilis</td>\n",
       "      <td>0.077894</td>\n",
       "      <td>0.982940</td>\n",
       "      <td>1.000048</td>\n",
       "      <td>0.999999</td>\n",
       "      <td>1.000048</td>\n",
       "      <td>0.999558</td>\n",
       "      <td>...</td>\n",
       "      <td>21</td>\n",
       "      <td>38</td>\n",
       "      <td>211169</td>\n",
       "      <td>211169</td>\n",
       "      <td>3976658</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>Even</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>run3</td>\n",
       "      <td>ENN</td>\n",
       "      <td>MEGAHIT</td>\n",
       "      <td>Bacillus_subtilis</td>\n",
       "      <td>0.097848</td>\n",
       "      <td>0.985261</td>\n",
       "      <td>1.000217</td>\n",
       "      <td>0.999861</td>\n",
       "      <td>1.000356</td>\n",
       "      <td>0.998712</td>\n",
       "      <td>...</td>\n",
       "      <td>14</td>\n",
       "      <td>28</td>\n",
       "      <td>263587</td>\n",
       "      <td>263587</td>\n",
       "      <td>3986046</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>Even</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>run3</td>\n",
       "      <td>ENN</td>\n",
       "      <td>metaSPAdes</td>\n",
       "      <td>Bacillus_subtilis</td>\n",
       "      <td>0.255293</td>\n",
       "      <td>0.985404</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.998464</td>\n",
       "      <td>...</td>\n",
       "      <td>10</td>\n",
       "      <td>20</td>\n",
       "      <td>325068</td>\n",
       "      <td>325068</td>\n",
       "      <td>3986625</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>Even</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1435</th>\n",
       "      <td>run2</td>\n",
       "      <td>LHS</td>\n",
       "      <td>MINIA</td>\n",
       "      <td>Staphylococcus_aureus_plasmid3</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>Log</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1436</th>\n",
       "      <td>run2</td>\n",
       "      <td>LHS</td>\n",
       "      <td>SKESA</td>\n",
       "      <td>Staphylococcus_aureus_plasmid3</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>Log</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1437</th>\n",
       "      <td>run2</td>\n",
       "      <td>LHS</td>\n",
       "      <td>SPAdes</td>\n",
       "      <td>Staphylococcus_aureus_plasmid3</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>Log</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1438</th>\n",
       "      <td>run2</td>\n",
       "      <td>LHS</td>\n",
       "      <td>Unicycler</td>\n",
       "      <td>Staphylococcus_aureus_plasmid3</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>Log</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1439</th>\n",
       "      <td>run2</td>\n",
       "      <td>LHS</td>\n",
       "      <td>VelvetOptimizer</td>\n",
       "      <td>Staphylococcus_aureus_plasmid3</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>Log</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1440 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       run sample          assembler                       reference  \\\n",
       "0     run3    ENN             BCALM2               Bacillus_subtilis   \n",
       "1     run3    ENN  GATBMiniaPipeline               Bacillus_subtilis   \n",
       "2     run3    ENN            IDBA-UD               Bacillus_subtilis   \n",
       "3     run3    ENN            MEGAHIT               Bacillus_subtilis   \n",
       "4     run3    ENN         metaSPAdes               Bacillus_subtilis   \n",
       "...    ...    ...                ...                             ...   \n",
       "1435  run2    LHS              MINIA  Staphylococcus_aureus_plasmid3   \n",
       "1436  run2    LHS              SKESA  Staphylococcus_aureus_plasmid3   \n",
       "1437  run2    LHS             SPAdes  Staphylococcus_aureus_plasmid3   \n",
       "1438  run2    LHS          Unicycler  Staphylococcus_aureus_plasmid3   \n",
       "1439  run2    LHS    VelvetOptimizer  Staphylococcus_aureus_plasmid3   \n",
       "\n",
       "      contiguity  breadth_of_coverage  multiplicity  validity  parsimony  \\\n",
       "0       0.032648             0.976835      1.000279  1.000000   1.000279   \n",
       "1       0.097833             0.983819      1.000000  1.000000   1.000000   \n",
       "2       0.077894             0.982940      1.000048  0.999999   1.000048   \n",
       "3       0.097848             0.985261      1.000217  0.999861   1.000356   \n",
       "4       0.255293             0.985404      1.000000  1.000000   1.000000   \n",
       "...          ...                  ...           ...       ...        ...   \n",
       "1435    0.000000             0.000000      0.000000  0.000000   0.000000   \n",
       "1436    0.000000             0.000000      0.000000  0.000000   0.000000   \n",
       "1437    0.000000             0.000000      0.000000  0.000000   0.000000   \n",
       "1438    0.000000             0.000000      0.000000  0.000000   0.000000   \n",
       "1439    0.000000             0.000000      0.000000  0.000000   0.000000   \n",
       "\n",
       "      identity  ...  L90  contigs    NA50    NG50  basepairs  Ns  \\\n",
       "0     1.000000  ...   66      116   65138   64934    3951958   0   \n",
       "1     0.999966  ...   14       29  269695  263366    3980213   0   \n",
       "2     0.999558  ...   21       38  211169  211169    3976658   0   \n",
       "3     0.998712  ...   14       28  263587  263587    3986046   0   \n",
       "4     0.998464  ...   10       20  325068  325068    3986625   0   \n",
       "...        ...  ...  ...      ...     ...     ...        ...  ..   \n",
       "1435  0.000000  ...    0        0       0       0          0   0   \n",
       "1436  0.000000  ...    0        0       0       0          0   0   \n",
       "1437  0.000000  ...    0        0       0       0          0   0   \n",
       "1438  0.000000  ...    0        0       0       0          0   0   \n",
       "1439  0.000000  ...    0        0       0       0          0   0   \n",
       "\n",
       "      misassemblies         type                       algorythm distribution  \n",
       "0                 0      Genomic    Single k-mer De Bruijn graph         Even  \n",
       "1                 0  Metagenomic  Multiple k-mer De Bruijn graph         Even  \n",
       "2                 0  Metagenomic  Multiple k-mer De Bruijn graph         Even  \n",
       "3                 0  Metagenomic  Multiple k-mer De Bruijn graph         Even  \n",
       "4                 0  Metagenomic  Multiple k-mer De Bruijn graph         Even  \n",
       "...             ...          ...                             ...          ...  \n",
       "1435              0      Genomic    Single k-mer De Bruijn graph          Log  \n",
       "1436              0      Genomic  Multiple k-mer De Bruijn graph          Log  \n",
       "1437              0      Genomic  Multiple k-mer De Bruijn graph          Log  \n",
       "1438              0      Genomic  Multiple k-mer De Bruijn graph          Log  \n",
       "1439              0      Genomic  Multiple k-mer De Bruijn graph          Log  \n",
       "\n",
       "[1440 rows x 21 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "report_glob = glob.glob('../Results/*/report/pipeline_report_tables.json')\n",
    "reference_pipeline_metrics_df = pd.DataFrame(columns=METRICS_COLUMNS_REFERENCE)\n",
    "\n",
    "for pipeline_report_file in report_glob:\n",
    "    report_file_name = pipeline_report_file.split('/')[-1]\n",
    "    stats_run = pipeline_report_file.split('/')[-3]\n",
    "    print('Processing {0} data from {1}...'.format(report_file_name, stats_run))\n",
    "    \n",
    "    with open(pipeline_report_file) as _fh:\n",
    "        json_report = json.load(_fh)\n",
    "        for sample in json_report.keys():\n",
    "            for reference, data in json_report[sample]['ReferenceTables'].items():\n",
    "                for row in data:\n",
    "                    for item in row:\n",
    "                        reference_pipeline_metrics_df = reference_pipeline_metrics_df.append({'run': stats_run,\n",
    "                                                                                        'sample': sample,\n",
    "                                                                                        'assembler': item['assembler'],\n",
    "                                                                                        'reference': reference,\n",
    "                                                                                        'contiguity': item['contiguity'],\n",
    "                                                                                        'breadth_of_coverage': item['breadth_of_coverage'],\n",
    "                                                                                        'multiplicity': item['multiplicity'],\n",
    "                                                                                        'validity': item['validity'],\n",
    "                                                                                        'parsimony': item['parsimony'],\n",
    "                                                                                        'identity': item['identity'],\n",
    "                                                                                        'lowest_identity': item['lowest_identity'],\n",
    "                                                                                        'L90': item['L90'],\n",
    "                                                                                        'contigs': item['aligned_contigs'],\n",
    "                                                                                        'NA50': item['NA50'],\n",
    "                                                                                        'NG50': item['NG50'],\n",
    "                                                                                        'basepairs': item['aligned_basepairs'],\n",
    "                                                                                        'Ns': item['Ns'],\n",
    "                                                                                        'misassemblies': item['misassembled_contigs']},\n",
    "                                                                                       ignore_index=True)\n",
    "\n",
    "reference_pipeline_metrics_df['distribution'] = np.where(reference_pipeline_metrics_df['sample'].isin(log_distributed), 'Log', 'Even')\n",
    "reference_pipeline_metrics_df['type'] = np.where(reference_pipeline_metrics_df['assembler'].isin(genomic_assemblers), 'Genomic', 'Metagenomic')\n",
    "reference_pipeline_metrics_df['algorythm'] = np.where(reference_pipeline_metrics_df['assembler'].isin(single_kmer), 'Single k-mer De Bruijn graph', 'Multiple k-mer De Bruijn graph')\n",
    "reference_pipeline_metrics_df[['contigs','basepairs','L90','Ns','NA50','NG50','misassemblies', 'multiplicity','validity','parsimony','identity','lowest_identity']] = reference_pipeline_metrics_df[['contigs','basepairs','L90','Ns','NA50','NG50','misassemblies','multiplicity','validity','parsimony','identity','lowest_identity']].apply(pd.to_numeric)\n",
    "display(reference_pipeline_metrics_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "run                     object\n",
       "sample                  object\n",
       "assembler               object\n",
       "reference               object\n",
       "contiguity             float64\n",
       "breadth_of_coverage    float64\n",
       "multiplicity           float64\n",
       "validity               float64\n",
       "parsimony              float64\n",
       "identity               float64\n",
       "lowest_identity        float64\n",
       "L90                      int64\n",
       "contigs                  int64\n",
       "NA50                     int64\n",
       "NG50                     int64\n",
       "basepairs                int64\n",
       "Ns                       int64\n",
       "misassemblies            int64\n",
       "type                    object\n",
       "algorythm               object\n",
       "distribution            object\n",
       "dtype: object"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reference_pipeline_metrics_df.dtypes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Save data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "reference_pipeline_metrics_df.to_csv(\"Tables/Results/Reference metrics.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reference statistics per assembler type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_stats_per_assembler_type = pd.DataFrame()\n",
    "for assembly_type in reference_pipeline_metrics_df.type.unique():\n",
    "    row={'Assembler Type': assembly_type}\n",
    "    lala=reference_pipeline_metrics_df[reference_pipeline_metrics_df.type == assembly_type].describe()\n",
    "    for column in lala.columns:\n",
    "        mean = lala.loc['mean',column]\n",
    "        std = lala.loc['std',column]\n",
    "        row[column] = \"{}(±{})\".format(round(mean, 2), round(std, 2))\n",
    "    ref_stats_per_assembler_type=ref_stats_per_assembler_type.append(row, ignore_index=True)\n",
    "ref_stats_per_assembler_type = ref_stats_per_assembler_type.set_index('Assembler Type')\n",
    "ref_stats_per_assembler_type.to_csv(\"Tables/Results/Reference metrics per assembler type.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>L90</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler Type</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Genomic</th>\n",
       "      <td>32.27(±116.14)</td>\n",
       "      <td>44541.82(±91876.27)</td>\n",
       "      <td>44301.36(±93249.33)</td>\n",
       "      <td>81.46(±801.06)</td>\n",
       "      <td>1392591.77(±2023755.37)</td>\n",
       "      <td>0.42(±0.47)</td>\n",
       "      <td>113.05(±296.71)</td>\n",
       "      <td>0.08(±0.2)</td>\n",
       "      <td>0.5(±0.5)</td>\n",
       "      <td>0.47(±0.49)</td>\n",
       "      <td>1.55(±27.23)</td>\n",
       "      <td>0.5(±0.5)</td>\n",
       "      <td>0.51(±0.51)</td>\n",
       "      <td>0.5(±0.5)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Metagenomic</th>\n",
       "      <td>24.19(±51.73)</td>\n",
       "      <td>82725.76(±123505.97)</td>\n",
       "      <td>81989.47(±122524.9)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1830606.01(±2236330.94)</td>\n",
       "      <td>0.55(±0.49)</td>\n",
       "      <td>45.2(±82.46)</td>\n",
       "      <td>0.14(±0.26)</td>\n",
       "      <td>0.66(±0.47)</td>\n",
       "      <td>0.61(±0.46)</td>\n",
       "      <td>0.39(±1.08)</td>\n",
       "      <td>0.67(±0.48)</td>\n",
       "      <td>0.69(±0.56)</td>\n",
       "      <td>0.66(±0.47)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           L90                  NA50                 NG50  \\\n",
       "Assembler Type                                                              \n",
       "Genomic         32.27(±116.14)   44541.82(±91876.27)  44301.36(±93249.33)   \n",
       "Metagenomic      24.19(±51.73)  82725.76(±123505.97)  81989.47(±122524.9)   \n",
       "\n",
       "                            Ns                basepairs breadth_of_coverage  \\\n",
       "Assembler Type                                                                \n",
       "Genomic         81.46(±801.06)  1392591.77(±2023755.37)         0.42(±0.47)   \n",
       "Metagenomic          0.0(±0.0)  1830606.01(±2236330.94)         0.55(±0.49)   \n",
       "\n",
       "                        contigs   contiguity     identity lowest_identity  \\\n",
       "Assembler Type                                                              \n",
       "Genomic         113.05(±296.71)   0.08(±0.2)    0.5(±0.5)     0.47(±0.49)   \n",
       "Metagenomic        45.2(±82.46)  0.14(±0.26)  0.66(±0.47)     0.61(±0.46)   \n",
       "\n",
       "               misassemblies multiplicity    parsimony     validity  \n",
       "Assembler Type                                                       \n",
       "Genomic         1.55(±27.23)    0.5(±0.5)  0.51(±0.51)    0.5(±0.5)  \n",
       "Metagenomic      0.39(±1.08)  0.67(±0.48)  0.69(±0.56)  0.66(±0.47)  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ref_stats_per_assembler_type"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reference stats per assembly algorithm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_stats_per_assembler_algo = pd.DataFrame()\n",
    "for assembly_algo in reference_pipeline_metrics_df.algorythm.unique():\n",
    "    row={'Assembler Algorithm': assembly_algo}\n",
    "    lala=reference_pipeline_metrics_df[reference_pipeline_metrics_df.algorythm == assembly_algo].describe()\n",
    "    for column in lala.columns:\n",
    "        mean = lala.loc['mean',column]\n",
    "        std = lala.loc['std',column]\n",
    "        row[column] = \"{}(±{})\".format(round(mean, 2), round(std, 2))\n",
    "    ref_stats_per_assembler_algo=ref_stats_per_assembler_algo.append(row, ignore_index=True)\n",
    "ref_stats_per_assembler_algo = ref_stats_per_assembler_algo.set_index('Assembler Algorithm')\n",
    "ref_stats_per_assembler_algo.to_csv(\"Tables/Results/Reference metrics per assembler algorithm.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>L90</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler Algorithm</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Single k-mer De Bruijn graph</th>\n",
       "      <td>59.17(±180.12)</td>\n",
       "      <td>8875.39(±15789.06)</td>\n",
       "      <td>8406.65(±15616.07)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1281528.14(±1944494.79)</td>\n",
       "      <td>0.35(±0.44)</td>\n",
       "      <td>200.96(±381.74)</td>\n",
       "      <td>0.03(±0.13)</td>\n",
       "      <td>0.47(±0.5)</td>\n",
       "      <td>0.46(±0.5)</td>\n",
       "      <td>0.03(±0.25)</td>\n",
       "      <td>0.47(±0.5)</td>\n",
       "      <td>0.47(±0.5)</td>\n",
       "      <td>0.47(±0.5)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Multiple k-mer De Bruijn graph</th>\n",
       "      <td>21.51(±55.64)</td>\n",
       "      <td>72550.4(±116246.96)</td>\n",
       "      <td>72119.09(±116500.93)</td>\n",
       "      <td>61.1(±694.53)</td>\n",
       "      <td>1639364.8(±2158330.22)</td>\n",
       "      <td>0.5(±0.49)</td>\n",
       "      <td>57.14(±174.1)</td>\n",
       "      <td>0.12(±0.25)</td>\n",
       "      <td>0.59(±0.49)</td>\n",
       "      <td>0.54(±0.48)</td>\n",
       "      <td>1.34(±23.59)</td>\n",
       "      <td>0.59(±0.49)</td>\n",
       "      <td>0.6(±0.54)</td>\n",
       "      <td>0.59(±0.49)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           L90                 NA50  \\\n",
       "Assembler Algorithm                                                   \n",
       "Single k-mer De Bruijn graph    59.17(±180.12)   8875.39(±15789.06)   \n",
       "Multiple k-mer De Bruijn graph   21.51(±55.64)  72550.4(±116246.96)   \n",
       "\n",
       "                                                NG50             Ns  \\\n",
       "Assembler Algorithm                                                   \n",
       "Single k-mer De Bruijn graph      8406.65(±15616.07)      0.0(±0.0)   \n",
       "Multiple k-mer De Bruijn graph  72119.09(±116500.93)  61.1(±694.53)   \n",
       "\n",
       "                                              basepairs breadth_of_coverage  \\\n",
       "Assembler Algorithm                                                           \n",
       "Single k-mer De Bruijn graph    1281528.14(±1944494.79)         0.35(±0.44)   \n",
       "Multiple k-mer De Bruijn graph   1639364.8(±2158330.22)          0.5(±0.49)   \n",
       "\n",
       "                                        contigs   contiguity     identity  \\\n",
       "Assembler Algorithm                                                         \n",
       "Single k-mer De Bruijn graph    200.96(±381.74)  0.03(±0.13)   0.47(±0.5)   \n",
       "Multiple k-mer De Bruijn graph    57.14(±174.1)  0.12(±0.25)  0.59(±0.49)   \n",
       "\n",
       "                               lowest_identity misassemblies multiplicity  \\\n",
       "Assembler Algorithm                                                         \n",
       "Single k-mer De Bruijn graph        0.46(±0.5)   0.03(±0.25)   0.47(±0.5)   \n",
       "Multiple k-mer De Bruijn graph     0.54(±0.48)  1.34(±23.59)  0.59(±0.49)   \n",
       "\n",
       "                                 parsimony     validity  \n",
       "Assembler Algorithm                                      \n",
       "Single k-mer De Bruijn graph    0.47(±0.5)   0.47(±0.5)  \n",
       "Multiple k-mer De Bruijn graph  0.6(±0.54)  0.59(±0.49)  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ref_stats_per_assembler_algo"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reference stats per assembler"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_stats_per_assembler = pd.DataFrame()\n",
    "for assembler in reference_pipeline_metrics_df.assembler.unique():\n",
    "    row={'Assembler': assembler, \n",
    "         'Type':reference_pipeline_metrics_df[reference_pipeline_metrics_df.assembler == assembler].type.unique()[0],\n",
    "         'Algorithm': reference_pipeline_metrics_df[reference_pipeline_metrics_df.assembler == assembler].algorythm.unique()[0]}\n",
    "    lala=reference_pipeline_metrics_df[reference_pipeline_metrics_df.assembler == assembler].describe()\n",
    "    for column in lala.columns:\n",
    "        mean = lala.loc['mean',column]\n",
    "        std = lala.loc['std',column]\n",
    "        row[column] = \"{}(±{})\".format(round(mean, 2), round(std, 2))\n",
    "    ref_stats_per_assembler=ref_stats_per_assembler.append(row, ignore_index=True)\n",
    "ref_stats_per_assembler = ref_stats_per_assembler.set_index('Assembler')\n",
    "ref_stats_per_assembler.to_csv(\"Tables/Results/Reference metrics per assembler.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>44.26(±217.6)</td>\n",
       "      <td>5388.68(±13681.22)</td>\n",
       "      <td>5086.6(±13454.63)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>840447.15(±1650142.0)</td>\n",
       "      <td>0.23(±0.38)</td>\n",
       "      <td>188.96(±421.77)</td>\n",
       "      <td>0.02(±0.12)</td>\n",
       "      <td>0.33(±0.47)</td>\n",
       "      <td>0.33(±0.47)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.33(±0.47)</td>\n",
       "      <td>0.33(±0.47)</td>\n",
       "      <td>0.33(±0.47)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>26.53(±60.32)</td>\n",
       "      <td>75511.67(±112427.0)</td>\n",
       "      <td>70925.78(±103438.3)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1828000.69(±2238597.0)</td>\n",
       "      <td>0.56(±0.49)</td>\n",
       "      <td>47.59(±88.73)</td>\n",
       "      <td>0.14(±0.27)</td>\n",
       "      <td>0.62(±0.49)</td>\n",
       "      <td>0.61(±0.48)</td>\n",
       "      <td>0.17(±0.53)</td>\n",
       "      <td>0.63(±0.49)</td>\n",
       "      <td>0.63(±0.49)</td>\n",
       "      <td>0.62(±0.49)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>43.74(±74.57)</td>\n",
       "      <td>52024.85(±91970.26)</td>\n",
       "      <td>51105.3(±91977.8)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1821846.85(±2232054.57)</td>\n",
       "      <td>0.54(±0.48)</td>\n",
       "      <td>74.9(±120.66)</td>\n",
       "      <td>0.09(±0.2)</td>\n",
       "      <td>0.66(±0.47)</td>\n",
       "      <td>0.57(±0.46)</td>\n",
       "      <td>0.64(±1.62)</td>\n",
       "      <td>0.67(±0.47)</td>\n",
       "      <td>0.74(±0.75)</td>\n",
       "      <td>0.65(±0.47)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>12.94(±22.37)</td>\n",
       "      <td>105537.11(±143830.26)</td>\n",
       "      <td>109204.12(±146844.18)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1837438.15(±2249917.86)</td>\n",
       "      <td>0.56(±0.49)</td>\n",
       "      <td>31.08(±45.0)</td>\n",
       "      <td>0.17(±0.3)</td>\n",
       "      <td>0.68(±0.46)</td>\n",
       "      <td>0.61(±0.45)</td>\n",
       "      <td>0.57(±1.17)</td>\n",
       "      <td>0.68(±0.47)</td>\n",
       "      <td>0.69(±0.47)</td>\n",
       "      <td>0.68(±0.47)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>13.56(±20.69)</td>\n",
       "      <td>97829.43(±133394.7)</td>\n",
       "      <td>96722.68(±132640.77)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1835138.35(±2248100.93)</td>\n",
       "      <td>0.56(±0.49)</td>\n",
       "      <td>27.24(±38.26)</td>\n",
       "      <td>0.16(±0.27)</td>\n",
       "      <td>0.68(±0.47)</td>\n",
       "      <td>0.65(±0.46)</td>\n",
       "      <td>0.17(±0.5)</td>\n",
       "      <td>0.69(±0.47)</td>\n",
       "      <td>0.69(±0.48)</td>\n",
       "      <td>0.68(±0.47)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>74.08(±131.58)</td>\n",
       "      <td>12362.1(±16990.35)</td>\n",
       "      <td>11726.7(±16917.51)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1722609.12(±2115127.99)</td>\n",
       "      <td>0.48(±0.47)</td>\n",
       "      <td>212.97(±338.05)</td>\n",
       "      <td>0.04(±0.13)</td>\n",
       "      <td>0.61(±0.49)</td>\n",
       "      <td>0.59(±0.49)</td>\n",
       "      <td>0.07(±0.35)</td>\n",
       "      <td>0.61(±0.49)</td>\n",
       "      <td>0.61(±0.49)</td>\n",
       "      <td>0.61(±0.49)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>22.76(±44.46)</td>\n",
       "      <td>47300.06(±83407.74)</td>\n",
       "      <td>46891.18(±83479.08)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1436160.07(±1961548.25)</td>\n",
       "      <td>0.47(±0.47)</td>\n",
       "      <td>95.31(±280.69)</td>\n",
       "      <td>0.12(±0.26)</td>\n",
       "      <td>0.54(±0.5)</td>\n",
       "      <td>0.48(±0.48)</td>\n",
       "      <td>0.33(±1.5)</td>\n",
       "      <td>0.54(±0.5)</td>\n",
       "      <td>0.54(±0.5)</td>\n",
       "      <td>0.54(±0.5)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>25.32(±98.93)</td>\n",
       "      <td>83133.97(±116866.68)</td>\n",
       "      <td>83465.24(±118060.11)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1808988.64(±2234202.5)</td>\n",
       "      <td>0.52(±0.48)</td>\n",
       "      <td>72.15(±260.1)</td>\n",
       "      <td>0.12(±0.21)</td>\n",
       "      <td>0.7(±0.45)</td>\n",
       "      <td>0.64(±0.45)</td>\n",
       "      <td>8.28(±66.43)</td>\n",
       "      <td>0.72(±0.46)</td>\n",
       "      <td>0.73(±0.48)</td>\n",
       "      <td>0.7(±0.45)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>11.68(±21.56)</td>\n",
       "      <td>104906.65(±137452.28)</td>\n",
       "      <td>101808.62(±136868.91)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1742439.92(±2215832.77)</td>\n",
       "      <td>0.54(±0.49)</td>\n",
       "      <td>23.51(±35.61)</td>\n",
       "      <td>0.16(±0.26)</td>\n",
       "      <td>0.56(±0.5)</td>\n",
       "      <td>0.54(±0.49)</td>\n",
       "      <td>0.12(±0.53)</td>\n",
       "      <td>0.56(±0.5)</td>\n",
       "      <td>0.56(±0.5)</td>\n",
       "      <td>0.55(±0.5)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>15.51(±41.88)</td>\n",
       "      <td>14159.49(±47604.26)</td>\n",
       "      <td>16829.79(±64115.87)</td>\n",
       "      <td>488.78(±1916.28)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>804905.71(±1645149.23)</td>\n",
       "      <td>0.25(±0.42)</td>\n",
       "      <td>85.38(±254.37)</td>\n",
       "      <td>0.03(±0.13)</td>\n",
       "      <td>0.26(±0.44)</td>\n",
       "      <td>0.25(±0.42)</td>\n",
       "      <td>0.47(±1.67)</td>\n",
       "      <td>0.26(±0.44)</td>\n",
       "      <td>0.26(±0.44)</td>\n",
       "      <td>0.26(±0.44)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm             L90  \\\n",
       "Assembler                                                           \n",
       "BCALM2               Single k-mer De Bruijn graph   44.26(±217.6)   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph   26.53(±60.32)   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph   43.74(±74.57)   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph   12.94(±22.37)   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph   13.56(±20.69)   \n",
       "MINIA                Single k-mer De Bruijn graph  74.08(±131.58)   \n",
       "SKESA              Multiple k-mer De Bruijn graph   22.76(±44.46)   \n",
       "SPAdes             Multiple k-mer De Bruijn graph   25.32(±98.93)   \n",
       "Unicycler          Multiple k-mer De Bruijn graph   11.68(±21.56)   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph   15.51(±41.88)   \n",
       "\n",
       "                                    NA50                   NG50  \\\n",
       "Assembler                                                         \n",
       "BCALM2                5388.68(±13681.22)      5086.6(±13454.63)   \n",
       "GATBMiniaPipeline    75511.67(±112427.0)    70925.78(±103438.3)   \n",
       "IDBA-UD              52024.85(±91970.26)      51105.3(±91977.8)   \n",
       "MEGAHIT            105537.11(±143830.26)  109204.12(±146844.18)   \n",
       "metaSPAdes           97829.43(±133394.7)   96722.68(±132640.77)   \n",
       "MINIA                 12362.1(±16990.35)     11726.7(±16917.51)   \n",
       "SKESA                47300.06(±83407.74)    46891.18(±83479.08)   \n",
       "SPAdes              83133.97(±116866.68)   83465.24(±118060.11)   \n",
       "Unicycler          104906.65(±137452.28)  101808.62(±136868.91)   \n",
       "VelvetOptimizer      14159.49(±47604.26)    16829.79(±64115.87)   \n",
       "\n",
       "                                 Ns         Type                basepairs  \\\n",
       "Assembler                                                                   \n",
       "BCALM2                    0.0(±0.0)      Genomic    840447.15(±1650142.0)   \n",
       "GATBMiniaPipeline         0.0(±0.0)  Metagenomic   1828000.69(±2238597.0)   \n",
       "IDBA-UD                   0.0(±0.0)  Metagenomic  1821846.85(±2232054.57)   \n",
       "MEGAHIT                   0.0(±0.0)  Metagenomic  1837438.15(±2249917.86)   \n",
       "metaSPAdes                0.0(±0.0)  Metagenomic  1835138.35(±2248100.93)   \n",
       "MINIA                     0.0(±0.0)      Genomic  1722609.12(±2115127.99)   \n",
       "SKESA                     0.0(±0.0)      Genomic  1436160.07(±1961548.25)   \n",
       "SPAdes                    0.0(±0.0)      Genomic   1808988.64(±2234202.5)   \n",
       "Unicycler                 0.0(±0.0)      Genomic  1742439.92(±2215832.77)   \n",
       "VelvetOptimizer    488.78(±1916.28)      Genomic   804905.71(±1645149.23)   \n",
       "\n",
       "                  breadth_of_coverage          contigs   contiguity  \\\n",
       "Assembler                                                             \n",
       "BCALM2                    0.23(±0.38)  188.96(±421.77)  0.02(±0.12)   \n",
       "GATBMiniaPipeline         0.56(±0.49)    47.59(±88.73)  0.14(±0.27)   \n",
       "IDBA-UD                   0.54(±0.48)    74.9(±120.66)   0.09(±0.2)   \n",
       "MEGAHIT                   0.56(±0.49)     31.08(±45.0)   0.17(±0.3)   \n",
       "metaSPAdes                0.56(±0.49)    27.24(±38.26)  0.16(±0.27)   \n",
       "MINIA                     0.48(±0.47)  212.97(±338.05)  0.04(±0.13)   \n",
       "SKESA                     0.47(±0.47)   95.31(±280.69)  0.12(±0.26)   \n",
       "SPAdes                    0.52(±0.48)    72.15(±260.1)  0.12(±0.21)   \n",
       "Unicycler                 0.54(±0.49)    23.51(±35.61)  0.16(±0.26)   \n",
       "VelvetOptimizer           0.25(±0.42)   85.38(±254.37)  0.03(±0.13)   \n",
       "\n",
       "                      identity lowest_identity misassemblies multiplicity  \\\n",
       "Assembler                                                                   \n",
       "BCALM2             0.33(±0.47)     0.33(±0.47)     0.0(±0.0)  0.33(±0.47)   \n",
       "GATBMiniaPipeline  0.62(±0.49)     0.61(±0.48)   0.17(±0.53)  0.63(±0.49)   \n",
       "IDBA-UD            0.66(±0.47)     0.57(±0.46)   0.64(±1.62)  0.67(±0.47)   \n",
       "MEGAHIT            0.68(±0.46)     0.61(±0.45)   0.57(±1.17)  0.68(±0.47)   \n",
       "metaSPAdes         0.68(±0.47)     0.65(±0.46)    0.17(±0.5)  0.69(±0.47)   \n",
       "MINIA              0.61(±0.49)     0.59(±0.49)   0.07(±0.35)  0.61(±0.49)   \n",
       "SKESA               0.54(±0.5)     0.48(±0.48)    0.33(±1.5)   0.54(±0.5)   \n",
       "SPAdes              0.7(±0.45)     0.64(±0.45)  8.28(±66.43)  0.72(±0.46)   \n",
       "Unicycler           0.56(±0.5)     0.54(±0.49)   0.12(±0.53)   0.56(±0.5)   \n",
       "VelvetOptimizer    0.26(±0.44)     0.25(±0.42)   0.47(±1.67)  0.26(±0.44)   \n",
       "\n",
       "                     parsimony     validity  \n",
       "Assembler                                    \n",
       "BCALM2             0.33(±0.47)  0.33(±0.47)  \n",
       "GATBMiniaPipeline  0.63(±0.49)  0.62(±0.49)  \n",
       "IDBA-UD            0.74(±0.75)  0.65(±0.47)  \n",
       "MEGAHIT            0.69(±0.47)  0.68(±0.47)  \n",
       "metaSPAdes         0.69(±0.48)  0.68(±0.47)  \n",
       "MINIA              0.61(±0.49)  0.61(±0.49)  \n",
       "SKESA               0.54(±0.5)   0.54(±0.5)  \n",
       "SPAdes             0.73(±0.48)   0.7(±0.45)  \n",
       "Unicycler           0.56(±0.5)   0.55(±0.5)  \n",
       "VelvetOptimizer    0.26(±0.44)  0.26(±0.44)  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ref_stats_per_assembler"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reference stats per reference replicon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Bacillus_subtilis', 'Enterococcus_faecalis',\n",
       "       'Escherichia_coli_plasmid', 'Escherichia_coli',\n",
       "       'Lactobacillus_fermentum', 'Listeria_monocytogenes',\n",
       "       'Pseudomonas_aeruginosa', 'Salmonella_enterica',\n",
       "       'Staphylococcus_aureus', 'Staphylococcus_aureus_plasmid1',\n",
       "       'Staphylococcus_aureus_plasmid2', 'Staphylococcus_aureus_plasmid3'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reference_pipeline_metrics_df.reference.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Bacillus_subtilis\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>32.5(±51.63)</td>\n",
       "      <td>17016.5(±25473.66)</td>\n",
       "      <td>16507.83(±25721.82)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1913231.5(±1818680.2)</td>\n",
       "      <td>0.47(±0.45)</td>\n",
       "      <td>382.67(±535.65)</td>\n",
       "      <td>0.01(±0.01)</td>\n",
       "      <td>0.83(±0.39)</td>\n",
       "      <td>0.83(±0.39)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.83(±0.39)</td>\n",
       "      <td>0.83(±0.39)</td>\n",
       "      <td>0.83(±0.39)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>115.25(±166.23)</td>\n",
       "      <td>149601.08(±122100.99)</td>\n",
       "      <td>147406.25(±120104.85)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3960376.67(±37546.65)</td>\n",
       "      <td>0.98(±0.01)</td>\n",
       "      <td>165.42(±215.2)</td>\n",
       "      <td>0.08(±0.06)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.99(±0.01)</td>\n",
       "      <td>0.58(±0.79)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>92.17(±98.39)</td>\n",
       "      <td>111161.67(±94220.24)</td>\n",
       "      <td>110292.0(±93518.39)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3968675.5(±15538.85)</td>\n",
       "      <td>0.98(±0.0)</td>\n",
       "      <td>141.5(±139.84)</td>\n",
       "      <td>0.05(±0.03)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.98(±0.01)</td>\n",
       "      <td>0.83(±1.11)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>38.67(±51.04)</td>\n",
       "      <td>205803.83(±101816.32)</td>\n",
       "      <td>206013.67(±101682.93)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3983434.17(±6834.76)</td>\n",
       "      <td>0.98(±0.0)</td>\n",
       "      <td>65.33(±74.49)</td>\n",
       "      <td>0.11(±0.06)</td>\n",
       "      <td>1.0(±0.01)</td>\n",
       "      <td>0.87(±0.21)</td>\n",
       "      <td>0.5(±0.8)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>32.83(±22.3)</td>\n",
       "      <td>173649.83(±111545.82)</td>\n",
       "      <td>170909.5(±109144.16)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3977137.17(±5153.73)</td>\n",
       "      <td>0.98(±0.0)</td>\n",
       "      <td>60.0(±37.85)</td>\n",
       "      <td>0.13(±0.1)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.98(±0.01)</td>\n",
       "      <td>0.67(±0.78)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>193.33(±216.14)</td>\n",
       "      <td>40222.25(±22140.45)</td>\n",
       "      <td>39606.08(±22072.87)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>3928355.33(±49914.34)</td>\n",
       "      <td>0.97(±0.01)</td>\n",
       "      <td>278.67(±242.5)</td>\n",
       "      <td>0.02(±0.01)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.01)</td>\n",
       "      <td>0.67(±0.98)</td>\n",
       "      <td>1.0(±0.01)</td>\n",
       "      <td>1.0(±0.01)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>9.33(±10.07)</td>\n",
       "      <td>117416.17(±124370.34)</td>\n",
       "      <td>116361.33(±122979.89)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1984359.17(±2072596.78)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>16.33(±17.45)</td>\n",
       "      <td>0.05(±0.05)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>27.17(±23.3)</td>\n",
       "      <td>214387.33(±102274.23)</td>\n",
       "      <td>222639.33(±109226.61)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>3974502.67(±6095.09)</td>\n",
       "      <td>0.98(±0.0)</td>\n",
       "      <td>49.33(±34.63)</td>\n",
       "      <td>0.13(±0.06)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.99(±0.01)</td>\n",
       "      <td>0.33(±0.78)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>28.83(±52.18)</td>\n",
       "      <td>180230.33(±141952.8)</td>\n",
       "      <td>162359.67(±156178.13)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2965228.33(±1564926.8)</td>\n",
       "      <td>0.73(±0.39)</td>\n",
       "      <td>43.5(±62.9)</td>\n",
       "      <td>0.11(±0.08)</td>\n",
       "      <td>0.83(±0.39)</td>\n",
       "      <td>0.83(±0.39)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.83(±0.39)</td>\n",
       "      <td>0.83(±0.39)</td>\n",
       "      <td>0.83(±0.39)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>20.5(±30.29)</td>\n",
       "      <td>22185.33(±32832.41)</td>\n",
       "      <td>22034.67(±32594.5)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1317971.0(±1946772.19)</td>\n",
       "      <td>0.33(±0.48)</td>\n",
       "      <td>37.17(±54.92)</td>\n",
       "      <td>0.02(±0.03)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm              L90  \\\n",
       "Assembler                                                            \n",
       "BCALM2               Single k-mer De Bruijn graph     32.5(±51.63)   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph  115.25(±166.23)   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph    92.17(±98.39)   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph    38.67(±51.04)   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph     32.83(±22.3)   \n",
       "MINIA                Single k-mer De Bruijn graph  193.33(±216.14)   \n",
       "SKESA              Multiple k-mer De Bruijn graph     9.33(±10.07)   \n",
       "SPAdes             Multiple k-mer De Bruijn graph     27.17(±23.3)   \n",
       "Unicycler          Multiple k-mer De Bruijn graph    28.83(±52.18)   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph     20.5(±30.29)   \n",
       "\n",
       "                                    NA50                   NG50         Ns  \\\n",
       "Assembler                                                                    \n",
       "BCALM2                17016.5(±25473.66)    16507.83(±25721.82)  0.0(±0.0)   \n",
       "GATBMiniaPipeline  149601.08(±122100.99)  147406.25(±120104.85)  0.0(±0.0)   \n",
       "IDBA-UD             111161.67(±94220.24)    110292.0(±93518.39)  0.0(±0.0)   \n",
       "MEGAHIT            205803.83(±101816.32)  206013.67(±101682.93)  0.0(±0.0)   \n",
       "metaSPAdes         173649.83(±111545.82)   170909.5(±109144.16)  0.0(±0.0)   \n",
       "MINIA                40222.25(±22140.45)    39606.08(±22072.87)  0.0(±0.0)   \n",
       "SKESA              117416.17(±124370.34)  116361.33(±122979.89)  0.0(±0.0)   \n",
       "SPAdes             214387.33(±102274.23)  222639.33(±109226.61)  0.0(±0.0)   \n",
       "Unicycler           180230.33(±141952.8)  162359.67(±156178.13)  0.0(±0.0)   \n",
       "VelvetOptimizer      22185.33(±32832.41)     22034.67(±32594.5)  0.0(±0.0)   \n",
       "\n",
       "                          Type                basepairs breadth_of_coverage  \\\n",
       "Assembler                                                                     \n",
       "BCALM2                 Genomic    1913231.5(±1818680.2)         0.47(±0.45)   \n",
       "GATBMiniaPipeline  Metagenomic    3960376.67(±37546.65)         0.98(±0.01)   \n",
       "IDBA-UD            Metagenomic     3968675.5(±15538.85)          0.98(±0.0)   \n",
       "MEGAHIT            Metagenomic     3983434.17(±6834.76)          0.98(±0.0)   \n",
       "metaSPAdes         Metagenomic     3977137.17(±5153.73)          0.98(±0.0)   \n",
       "MINIA                  Genomic    3928355.33(±49914.34)         0.97(±0.01)   \n",
       "SKESA                  Genomic  1984359.17(±2072596.78)         0.49(±0.51)   \n",
       "SPAdes                 Genomic     3974502.67(±6095.09)          0.98(±0.0)   \n",
       "Unicycler              Genomic   2965228.33(±1564926.8)         0.73(±0.39)   \n",
       "VelvetOptimizer        Genomic   1317971.0(±1946772.19)         0.33(±0.48)   \n",
       "\n",
       "                           contigs   contiguity     identity lowest_identity  \\\n",
       "Assembler                                                                      \n",
       "BCALM2             382.67(±535.65)  0.01(±0.01)  0.83(±0.39)     0.83(±0.39)   \n",
       "GATBMiniaPipeline   165.42(±215.2)  0.08(±0.06)    1.0(±0.0)     0.99(±0.01)   \n",
       "IDBA-UD             141.5(±139.84)  0.05(±0.03)    1.0(±0.0)     0.98(±0.01)   \n",
       "MEGAHIT              65.33(±74.49)  0.11(±0.06)   1.0(±0.01)     0.87(±0.21)   \n",
       "metaSPAdes            60.0(±37.85)   0.13(±0.1)    1.0(±0.0)     0.98(±0.01)   \n",
       "MINIA               278.67(±242.5)  0.02(±0.01)    1.0(±0.0)      1.0(±0.01)   \n",
       "SKESA                16.33(±17.45)  0.05(±0.05)   0.5(±0.52)      0.5(±0.52)   \n",
       "SPAdes               49.33(±34.63)  0.13(±0.06)    1.0(±0.0)     0.99(±0.01)   \n",
       "Unicycler              43.5(±62.9)  0.11(±0.08)  0.83(±0.39)     0.83(±0.39)   \n",
       "VelvetOptimizer      37.17(±54.92)  0.02(±0.03)  0.33(±0.49)     0.33(±0.49)   \n",
       "\n",
       "                  misassemblies multiplicity    parsimony     validity  \n",
       "Assembler                                                               \n",
       "BCALM2                0.0(±0.0)  0.83(±0.39)  0.83(±0.39)  0.83(±0.39)  \n",
       "GATBMiniaPipeline   0.58(±0.79)    1.0(±0.0)    1.0(±0.0)    1.0(±0.0)  \n",
       "IDBA-UD             0.83(±1.11)    1.0(±0.0)    1.0(±0.0)    1.0(±0.0)  \n",
       "MEGAHIT               0.5(±0.8)    1.0(±0.0)    1.0(±0.0)    1.0(±0.0)  \n",
       "metaSPAdes          0.67(±0.78)    1.0(±0.0)    1.0(±0.0)    1.0(±0.0)  \n",
       "MINIA               0.67(±0.98)   1.0(±0.01)   1.0(±0.01)    1.0(±0.0)  \n",
       "SKESA                 0.0(±0.0)   0.5(±0.52)   0.5(±0.52)   0.5(±0.52)  \n",
       "SPAdes              0.33(±0.78)    1.0(±0.0)    1.0(±0.0)    1.0(±0.0)  \n",
       "Unicycler             0.0(±0.0)  0.83(±0.39)  0.83(±0.39)  0.83(±0.39)  \n",
       "VelvetOptimizer       0.0(±0.0)  0.33(±0.49)  0.33(±0.49)  0.33(±0.49)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Enterococcus_faecalis\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>17.5(±40.88)</td>\n",
       "      <td>5311.67(±11755.93)</td>\n",
       "      <td>5066.58(±11833.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>651058.92(±1075258.97)</td>\n",
       "      <td>0.23(±0.38)</td>\n",
       "      <td>160.42(±303.97)</td>\n",
       "      <td>0.01(±0.02)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>9.17(±10.13)</td>\n",
       "      <td>83020.17(±89046.92)</td>\n",
       "      <td>84170.5(±90589.23)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1400651.25(±1462979.09)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>19.25(±20.61)</td>\n",
       "      <td>0.08(±0.09)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.49(±0.52)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>32.67(±54.62)</td>\n",
       "      <td>41193.67(±54299.03)</td>\n",
       "      <td>41736.33(±55119.29)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1393959.67(±1455984.8)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>56.17(±85.33)</td>\n",
       "      <td>0.05(±0.06)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.46(±0.49)</td>\n",
       "      <td>0.5(±0.8)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>7.0(±7.48)</td>\n",
       "      <td>117668.83(±122901.09)</td>\n",
       "      <td>118207.17(±123468.38)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1407900.08(±1470537.2)</td>\n",
       "      <td>0.49(±0.52)</td>\n",
       "      <td>15.92(±16.69)</td>\n",
       "      <td>0.11(±0.12)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.48(±0.5)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>7.5(±9.43)</td>\n",
       "      <td>103369.67(±112974.61)</td>\n",
       "      <td>103369.67(±112974.61)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1396282.67(±1458429.71)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>15.17(±18.64)</td>\n",
       "      <td>0.08(±0.1)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>59.25(±63.72)</td>\n",
       "      <td>15743.5(±12692.73)</td>\n",
       "      <td>13404.83(±14499.77)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1383381.67(±1432831.1)</td>\n",
       "      <td>0.49(±0.5)</td>\n",
       "      <td>93.33(±96.61)</td>\n",
       "      <td>0.02(±0.02)</td>\n",
       "      <td>0.83(±0.39)</td>\n",
       "      <td>0.83(±0.39)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.83(±0.39)</td>\n",
       "      <td>0.83(±0.39)</td>\n",
       "      <td>0.83(±0.39)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>15.83(±17.61)</td>\n",
       "      <td>49461.0(±52706.44)</td>\n",
       "      <td>47018.5(±52133.86)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1402020.17(±1458916.42)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>29.83(±29.71)</td>\n",
       "      <td>0.06(±0.06)</td>\n",
       "      <td>0.83(±0.39)</td>\n",
       "      <td>0.69(±0.4)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.83(±0.39)</td>\n",
       "      <td>0.84(±0.39)</td>\n",
       "      <td>0.83(±0.39)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>8.0(±9.63)</td>\n",
       "      <td>97227.67(±110100.19)</td>\n",
       "      <td>98739.33(±111126.93)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1412023.67(±1464709.19)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>21.67(±21.59)</td>\n",
       "      <td>0.1(±0.1)</td>\n",
       "      <td>0.79(±0.38)</td>\n",
       "      <td>0.71(±0.44)</td>\n",
       "      <td>1.67(±2.74)</td>\n",
       "      <td>0.87(±0.41)</td>\n",
       "      <td>0.96(±0.53)</td>\n",
       "      <td>0.78(±0.38)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>4.83(±5.2)</td>\n",
       "      <td>141763.17(±155926.65)</td>\n",
       "      <td>133592.17(±151708.82)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1396688.5(±1458835.03)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>11.5(±12.3)</td>\n",
       "      <td>0.13(±0.14)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>34.17(±50.48)</td>\n",
       "      <td>10149.33(±14991.56)</td>\n",
       "      <td>10133.5(±14968.2)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>911895.5(±1346958.97)</td>\n",
       "      <td>0.32(±0.47)</td>\n",
       "      <td>53.17(±78.55)</td>\n",
       "      <td>0.02(±0.02)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm            L90  \\\n",
       "Assembler                                                          \n",
       "BCALM2               Single k-mer De Bruijn graph   17.5(±40.88)   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph   9.17(±10.13)   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  32.67(±54.62)   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph     7.0(±7.48)   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph     7.5(±9.43)   \n",
       "MINIA                Single k-mer De Bruijn graph  59.25(±63.72)   \n",
       "SKESA              Multiple k-mer De Bruijn graph  15.83(±17.61)   \n",
       "SPAdes             Multiple k-mer De Bruijn graph     8.0(±9.63)   \n",
       "Unicycler          Multiple k-mer De Bruijn graph     4.83(±5.2)   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph  34.17(±50.48)   \n",
       "\n",
       "                                    NA50                   NG50         Ns  \\\n",
       "Assembler                                                                    \n",
       "BCALM2                5311.67(±11755.93)      5066.58(±11833.0)  0.0(±0.0)   \n",
       "GATBMiniaPipeline    83020.17(±89046.92)     84170.5(±90589.23)  0.0(±0.0)   \n",
       "IDBA-UD              41193.67(±54299.03)    41736.33(±55119.29)  0.0(±0.0)   \n",
       "MEGAHIT            117668.83(±122901.09)  118207.17(±123468.38)  0.0(±0.0)   \n",
       "metaSPAdes         103369.67(±112974.61)  103369.67(±112974.61)  0.0(±0.0)   \n",
       "MINIA                 15743.5(±12692.73)    13404.83(±14499.77)  0.0(±0.0)   \n",
       "SKESA                 49461.0(±52706.44)     47018.5(±52133.86)  0.0(±0.0)   \n",
       "SPAdes              97227.67(±110100.19)   98739.33(±111126.93)  0.0(±0.0)   \n",
       "Unicycler          141763.17(±155926.65)  133592.17(±151708.82)  0.0(±0.0)   \n",
       "VelvetOptimizer      10149.33(±14991.56)      10133.5(±14968.2)  0.0(±0.0)   \n",
       "\n",
       "                          Type                basepairs breadth_of_coverage  \\\n",
       "Assembler                                                                     \n",
       "BCALM2                 Genomic   651058.92(±1075258.97)         0.23(±0.38)   \n",
       "GATBMiniaPipeline  Metagenomic  1400651.25(±1462979.09)         0.49(±0.51)   \n",
       "IDBA-UD            Metagenomic   1393959.67(±1455984.8)         0.49(±0.51)   \n",
       "MEGAHIT            Metagenomic   1407900.08(±1470537.2)         0.49(±0.52)   \n",
       "metaSPAdes         Metagenomic  1396282.67(±1458429.71)         0.49(±0.51)   \n",
       "MINIA                  Genomic   1383381.67(±1432831.1)          0.49(±0.5)   \n",
       "SKESA                  Genomic  1402020.17(±1458916.42)         0.49(±0.51)   \n",
       "SPAdes                 Genomic  1412023.67(±1464709.19)          0.5(±0.51)   \n",
       "Unicycler              Genomic   1396688.5(±1458835.03)         0.49(±0.51)   \n",
       "VelvetOptimizer        Genomic    911895.5(±1346958.97)         0.32(±0.47)   \n",
       "\n",
       "                           contigs   contiguity     identity lowest_identity  \\\n",
       "Assembler                                                                      \n",
       "BCALM2             160.42(±303.97)  0.01(±0.02)  0.33(±0.49)     0.33(±0.49)   \n",
       "GATBMiniaPipeline    19.25(±20.61)  0.08(±0.09)   0.5(±0.52)     0.49(±0.52)   \n",
       "IDBA-UD              56.17(±85.33)  0.05(±0.06)   0.5(±0.52)     0.46(±0.49)   \n",
       "MEGAHIT              15.92(±16.69)  0.11(±0.12)   0.5(±0.52)      0.48(±0.5)   \n",
       "metaSPAdes           15.17(±18.64)   0.08(±0.1)   0.5(±0.52)     0.49(±0.51)   \n",
       "MINIA                93.33(±96.61)  0.02(±0.02)  0.83(±0.39)     0.83(±0.39)   \n",
       "SKESA                29.83(±29.71)  0.06(±0.06)  0.83(±0.39)      0.69(±0.4)   \n",
       "SPAdes               21.67(±21.59)    0.1(±0.1)  0.79(±0.38)     0.71(±0.44)   \n",
       "Unicycler              11.5(±12.3)  0.13(±0.14)   0.5(±0.52)      0.5(±0.52)   \n",
       "VelvetOptimizer      53.17(±78.55)  0.02(±0.02)  0.33(±0.49)     0.33(±0.49)   \n",
       "\n",
       "                  misassemblies multiplicity    parsimony     validity  \n",
       "Assembler                                                               \n",
       "BCALM2                0.0(±0.0)  0.33(±0.49)  0.33(±0.49)  0.33(±0.49)  \n",
       "GATBMiniaPipeline   0.33(±0.49)   0.5(±0.52)   0.5(±0.52)   0.5(±0.52)  \n",
       "IDBA-UD               0.5(±0.8)   0.5(±0.52)   0.5(±0.52)   0.5(±0.52)  \n",
       "MEGAHIT             0.33(±0.49)   0.5(±0.52)   0.5(±0.52)   0.5(±0.52)  \n",
       "metaSPAdes          0.17(±0.39)   0.5(±0.52)   0.5(±0.52)   0.5(±0.52)  \n",
       "MINIA                 0.0(±0.0)  0.83(±0.39)  0.83(±0.39)  0.83(±0.39)  \n",
       "SKESA               0.33(±0.49)  0.83(±0.39)  0.84(±0.39)  0.83(±0.39)  \n",
       "SPAdes              1.67(±2.74)  0.87(±0.41)  0.96(±0.53)  0.78(±0.38)  \n",
       "Unicycler             0.0(±0.0)   0.5(±0.52)   0.5(±0.52)   0.5(±0.52)  \n",
       "VelvetOptimizer       0.0(±0.0)  0.33(±0.49)  0.33(±0.49)  0.33(±0.49)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Escherichia_coli_plasmid\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.5(±1.17)</td>\n",
       "      <td>4473.0(±10446.68)</td>\n",
       "      <td>4473.0(±10446.68)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>17516.67(±40910.12)</td>\n",
       "      <td>0.16(±0.37)</td>\n",
       "      <td>0.83(±1.95)</td>\n",
       "      <td>0.06(±0.13)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>1.17(±1.27)</td>\n",
       "      <td>18771.83(±21683.73)</td>\n",
       "      <td>18771.83(±21683.73)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>54458.0(±56881.01)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>1.75(±1.86)</td>\n",
       "      <td>0.23(±0.24)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>1.67(±1.78)</td>\n",
       "      <td>14454.33(±14717.48)</td>\n",
       "      <td>14269.83(±14905.13)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>54250.67(±56279.71)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>2.67(±2.67)</td>\n",
       "      <td>0.17(±0.18)</td>\n",
       "      <td>0.66(±0.49)</td>\n",
       "      <td>0.66(±0.49)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>46160.5(±51369.14)</td>\n",
       "      <td>45910.0(±51609.48)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>55086.33(±57016.76)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.92(±0.9)</td>\n",
       "      <td>0.42(±0.47)</td>\n",
       "      <td>0.66(±0.49)</td>\n",
       "      <td>0.66(±0.49)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>32132.83(±33900.9)</td>\n",
       "      <td>32132.83(±33900.9)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>54693.0(±57128.66)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>1.0(±1.04)</td>\n",
       "      <td>0.29(±0.31)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>2.0(±2.34)</td>\n",
       "      <td>11414.5(±12636.0)</td>\n",
       "      <td>11414.5(±12636.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>52511.42(±54848.51)</td>\n",
       "      <td>0.48(±0.5)</td>\n",
       "      <td>3.08(±3.37)</td>\n",
       "      <td>0.15(±0.16)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.33(±0.78)</td>\n",
       "      <td>41292.5(±51587.61)</td>\n",
       "      <td>41292.5(±51587.61)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>54529.83(±56958.31)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.83(±1.11)</td>\n",
       "      <td>0.41(±0.47)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>1.0(±1.21)</td>\n",
       "      <td>21183.33(±25801.18)</td>\n",
       "      <td>20931.67(±26016.32)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>54636.83(±56545.69)</td>\n",
       "      <td>0.5(±0.51)</td>\n",
       "      <td>1.83(±1.75)</td>\n",
       "      <td>0.25(±0.27)</td>\n",
       "      <td>0.66(±0.49)</td>\n",
       "      <td>0.66(±0.49)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.83(±1.11)</td>\n",
       "      <td>25299.83(±29220.18)</td>\n",
       "      <td>24956.0(±28847.65)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>54046.83(±56453.02)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>1.33(±1.56)</td>\n",
       "      <td>0.25(±0.27)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>1.0(±1.48)</td>\n",
       "      <td>8984.33(±13270.92)</td>\n",
       "      <td>8984.33(±13270.92)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>35092.0(±51834.34)</td>\n",
       "      <td>0.32(±0.47)</td>\n",
       "      <td>1.5(±2.24)</td>\n",
       "      <td>0.12(±0.17)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm          L90  \\\n",
       "Assembler                                                        \n",
       "BCALM2               Single k-mer De Bruijn graph   0.5(±1.17)   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph  1.17(±1.27)   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  1.67(±1.78)   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph  0.17(±0.39)   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph   0.5(±0.52)   \n",
       "MINIA                Single k-mer De Bruijn graph   2.0(±2.34)   \n",
       "SKESA              Multiple k-mer De Bruijn graph  0.33(±0.78)   \n",
       "SPAdes             Multiple k-mer De Bruijn graph   1.0(±1.21)   \n",
       "Unicycler          Multiple k-mer De Bruijn graph  0.83(±1.11)   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph   1.0(±1.48)   \n",
       "\n",
       "                                  NA50                 NG50         Ns  \\\n",
       "Assembler                                                                \n",
       "BCALM2               4473.0(±10446.68)    4473.0(±10446.68)  0.0(±0.0)   \n",
       "GATBMiniaPipeline  18771.83(±21683.73)  18771.83(±21683.73)  0.0(±0.0)   \n",
       "IDBA-UD            14454.33(±14717.48)  14269.83(±14905.13)  0.0(±0.0)   \n",
       "MEGAHIT             46160.5(±51369.14)   45910.0(±51609.48)  0.0(±0.0)   \n",
       "metaSPAdes          32132.83(±33900.9)   32132.83(±33900.9)  0.0(±0.0)   \n",
       "MINIA                11414.5(±12636.0)    11414.5(±12636.0)  0.0(±0.0)   \n",
       "SKESA               41292.5(±51587.61)   41292.5(±51587.61)  0.0(±0.0)   \n",
       "SPAdes             21183.33(±25801.18)  20931.67(±26016.32)  0.0(±0.0)   \n",
       "Unicycler          25299.83(±29220.18)   24956.0(±28847.65)  0.0(±0.0)   \n",
       "VelvetOptimizer     8984.33(±13270.92)   8984.33(±13270.92)  0.0(±0.0)   \n",
       "\n",
       "                          Type            basepairs breadth_of_coverage  \\\n",
       "Assembler                                                                 \n",
       "BCALM2                 Genomic  17516.67(±40910.12)         0.16(±0.37)   \n",
       "GATBMiniaPipeline  Metagenomic   54458.0(±56881.01)          0.5(±0.52)   \n",
       "IDBA-UD            Metagenomic  54250.67(±56279.71)         0.49(±0.51)   \n",
       "MEGAHIT            Metagenomic  55086.33(±57016.76)          0.5(±0.52)   \n",
       "metaSPAdes         Metagenomic   54693.0(±57128.66)          0.5(±0.52)   \n",
       "MINIA                  Genomic  52511.42(±54848.51)          0.48(±0.5)   \n",
       "SKESA                  Genomic  54529.83(±56958.31)          0.5(±0.52)   \n",
       "SPAdes                 Genomic  54636.83(±56545.69)          0.5(±0.51)   \n",
       "Unicycler              Genomic  54046.83(±56453.02)         0.49(±0.51)   \n",
       "VelvetOptimizer        Genomic   35092.0(±51834.34)         0.32(±0.47)   \n",
       "\n",
       "                       contigs   contiguity     identity lowest_identity  \\\n",
       "Assembler                                                                  \n",
       "BCALM2             0.83(±1.95)  0.06(±0.13)  0.17(±0.39)     0.17(±0.39)   \n",
       "GATBMiniaPipeline  1.75(±1.86)  0.23(±0.24)   0.5(±0.52)      0.5(±0.52)   \n",
       "IDBA-UD            2.67(±2.67)  0.17(±0.18)  0.66(±0.49)     0.66(±0.49)   \n",
       "MEGAHIT             0.92(±0.9)  0.42(±0.47)  0.66(±0.49)     0.66(±0.49)   \n",
       "metaSPAdes          1.0(±1.04)  0.29(±0.31)   0.5(±0.52)      0.5(±0.52)   \n",
       "MINIA              3.08(±3.37)  0.15(±0.16)   0.5(±0.52)      0.5(±0.52)   \n",
       "SKESA              0.83(±1.11)  0.41(±0.47)   0.5(±0.52)      0.5(±0.52)   \n",
       "SPAdes             1.83(±1.75)  0.25(±0.27)  0.66(±0.49)     0.66(±0.49)   \n",
       "Unicycler          1.33(±1.56)  0.25(±0.27)   0.5(±0.52)      0.5(±0.52)   \n",
       "VelvetOptimizer     1.5(±2.24)  0.12(±0.17)  0.33(±0.49)     0.33(±0.49)   \n",
       "\n",
       "                  misassemblies multiplicity    parsimony     validity  \n",
       "Assembler                                                               \n",
       "BCALM2                0.0(±0.0)  0.17(±0.39)  0.17(±0.39)  0.17(±0.39)  \n",
       "GATBMiniaPipeline     0.0(±0.0)   0.5(±0.52)   0.5(±0.52)   0.5(±0.52)  \n",
       "IDBA-UD               0.0(±0.0)  0.67(±0.49)  0.67(±0.49)  0.67(±0.49)  \n",
       "MEGAHIT               0.0(±0.0)  0.67(±0.49)  0.67(±0.49)  0.67(±0.49)  \n",
       "metaSPAdes            0.0(±0.0)   0.5(±0.52)   0.5(±0.52)   0.5(±0.52)  \n",
       "MINIA                 0.0(±0.0)   0.5(±0.52)   0.5(±0.52)   0.5(±0.52)  \n",
       "SKESA                 0.0(±0.0)   0.5(±0.52)   0.5(±0.52)   0.5(±0.52)  \n",
       "SPAdes                0.0(±0.0)  0.67(±0.49)  0.67(±0.49)  0.67(±0.49)  \n",
       "Unicycler             0.0(±0.0)   0.5(±0.52)   0.5(±0.52)   0.5(±0.52)  \n",
       "VelvetOptimizer       0.0(±0.0)  0.33(±0.49)  0.33(±0.49)  0.33(±0.49)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Escherichia_coli\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1069.17(±1751.88)</td>\n",
       "      <td>741.67(±1243.32)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1098954.5(±1647814.92)</td>\n",
       "      <td>0.23(±0.35)</td>\n",
       "      <td>438.5(±658.35)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>50.67(±53.18)</td>\n",
       "      <td>29829.67(±30833.33)</td>\n",
       "      <td>28981.83(±30420.26)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2313731.67(±2415114.11)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>94.58(±97.59)</td>\n",
       "      <td>0.01(±0.01)</td>\n",
       "      <td>0.66(±0.49)</td>\n",
       "      <td>0.57(±0.46)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.67(±0.5)</td>\n",
       "      <td>0.66(±0.49)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>123.42(±130.12)</td>\n",
       "      <td>15857.33(±11471.13)</td>\n",
       "      <td>10091.67(±10614.96)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2274553.25(±2369552.74)</td>\n",
       "      <td>0.48(±0.5)</td>\n",
       "      <td>202.25(±205.78)</td>\n",
       "      <td>0.01(±0.01)</td>\n",
       "      <td>0.95(±0.07)</td>\n",
       "      <td>0.44(±0.41)</td>\n",
       "      <td>2.83(±2.37)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.8(±1.79)</td>\n",
       "      <td>0.84(±0.31)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>24.5(±25.94)</td>\n",
       "      <td>42178.67(±43128.04)</td>\n",
       "      <td>47202.5(±50787.41)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2330881.5(±2424930.54)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>70.75(±66.26)</td>\n",
       "      <td>0.03(±0.03)</td>\n",
       "      <td>0.99(±0.01)</td>\n",
       "      <td>0.78(±0.37)</td>\n",
       "      <td>1.83(±2.12)</td>\n",
       "      <td>1.0(±0.01)</td>\n",
       "      <td>1.02(±0.02)</td>\n",
       "      <td>0.99(±0.02)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>26.67(±33.8)</td>\n",
       "      <td>52767.0(±57732.18)</td>\n",
       "      <td>51810.83(±58746.82)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2331363.67(±2427849.41)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>59.0(±62.64)</td>\n",
       "      <td>0.04(±0.05)</td>\n",
       "      <td>0.98(±0.03)</td>\n",
       "      <td>0.9(±0.18)</td>\n",
       "      <td>0.83(±1.11)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.02(±0.03)</td>\n",
       "      <td>0.98(±0.03)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>2224.5(±2323.51)</td>\n",
       "      <td>1579.33(±1650.13)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1881469.42(±1965150.53)</td>\n",
       "      <td>0.39(±0.41)</td>\n",
       "      <td>565.08(±590.26)</td>\n",
       "      <td>0.01(±0.01)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.49(±0.52)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>62.5(±69.02)</td>\n",
       "      <td>22329.33(±24731.84)</td>\n",
       "      <td>22091.5(±24654.37)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2281684.5(±2383170.32)</td>\n",
       "      <td>0.48(±0.5)</td>\n",
       "      <td>113.17(±121.89)</td>\n",
       "      <td>0.01(±0.01)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.19(±0.37)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.52(±0.54)</td>\n",
       "      <td>0.48(±0.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>29.67(±33.14)</td>\n",
       "      <td>43076.33(±44941.99)</td>\n",
       "      <td>41704.67(±44965.26)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2319842.17(±2406797.45)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>62.17(±53.74)</td>\n",
       "      <td>0.04(±0.04)</td>\n",
       "      <td>0.97(±0.05)</td>\n",
       "      <td>0.77(±0.31)</td>\n",
       "      <td>1.17(±1.11)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.03(±0.07)</td>\n",
       "      <td>0.98(±0.06)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>20.83(±23.26)</td>\n",
       "      <td>57316.5(±60896.05)</td>\n",
       "      <td>57285.67(±61099.9)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2312584.67(±2415461.01)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>45.83(±49.13)</td>\n",
       "      <td>0.05(±0.05)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.43(±0.47)</td>\n",
       "      <td>0.67(±1.15)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.53)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1614.83(±2385.32)</td>\n",
       "      <td>1368.17(±2021.42)</td>\n",
       "      <td>1679.67(±2481.3)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1357997.17(±2005942.66)</td>\n",
       "      <td>0.28(±0.42)</td>\n",
       "      <td>375.67(±554.93)</td>\n",
       "      <td>0.0(±0.01)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.28(±0.41)</td>\n",
       "      <td>2.17(±3.21)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm              L90  \\\n",
       "Assembler                                                            \n",
       "BCALM2               Single k-mer De Bruijn graph        0.0(±0.0)   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph    50.67(±53.18)   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  123.42(±130.12)   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph     24.5(±25.94)   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph     26.67(±33.8)   \n",
       "MINIA                Single k-mer De Bruijn graph        0.0(±0.0)   \n",
       "SKESA              Multiple k-mer De Bruijn graph     62.5(±69.02)   \n",
       "SPAdes             Multiple k-mer De Bruijn graph    29.67(±33.14)   \n",
       "Unicycler          Multiple k-mer De Bruijn graph    20.83(±23.26)   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph        0.0(±0.0)   \n",
       "\n",
       "                                  NA50                 NG50                Ns  \\\n",
       "Assembler                                                                       \n",
       "BCALM2               1069.17(±1751.88)     741.67(±1243.32)         0.0(±0.0)   \n",
       "GATBMiniaPipeline  29829.67(±30833.33)  28981.83(±30420.26)         0.0(±0.0)   \n",
       "IDBA-UD            15857.33(±11471.13)  10091.67(±10614.96)         0.0(±0.0)   \n",
       "MEGAHIT            42178.67(±43128.04)   47202.5(±50787.41)         0.0(±0.0)   \n",
       "metaSPAdes          52767.0(±57732.18)  51810.83(±58746.82)         0.0(±0.0)   \n",
       "MINIA                 2224.5(±2323.51)    1579.33(±1650.13)         0.0(±0.0)   \n",
       "SKESA              22329.33(±24731.84)   22091.5(±24654.37)         0.0(±0.0)   \n",
       "SPAdes             43076.33(±44941.99)  41704.67(±44965.26)         0.0(±0.0)   \n",
       "Unicycler           57316.5(±60896.05)   57285.67(±61099.9)         0.0(±0.0)   \n",
       "VelvetOptimizer      1614.83(±2385.32)    1368.17(±2021.42)  1679.67(±2481.3)   \n",
       "\n",
       "                          Type                basepairs breadth_of_coverage  \\\n",
       "Assembler                                                                     \n",
       "BCALM2                 Genomic   1098954.5(±1647814.92)         0.23(±0.35)   \n",
       "GATBMiniaPipeline  Metagenomic  2313731.67(±2415114.11)         0.49(±0.51)   \n",
       "IDBA-UD            Metagenomic  2274553.25(±2369552.74)          0.48(±0.5)   \n",
       "MEGAHIT            Metagenomic   2330881.5(±2424930.54)         0.49(±0.51)   \n",
       "metaSPAdes         Metagenomic  2331363.67(±2427849.41)         0.49(±0.51)   \n",
       "MINIA                  Genomic  1881469.42(±1965150.53)         0.39(±0.41)   \n",
       "SKESA                  Genomic   2281684.5(±2383170.32)          0.48(±0.5)   \n",
       "SPAdes                 Genomic  2319842.17(±2406797.45)         0.49(±0.51)   \n",
       "Unicycler              Genomic  2312584.67(±2415461.01)         0.49(±0.51)   \n",
       "VelvetOptimizer        Genomic  1357997.17(±2005942.66)         0.28(±0.42)   \n",
       "\n",
       "                           contigs   contiguity     identity lowest_identity  \\\n",
       "Assembler                                                                      \n",
       "BCALM2              438.5(±658.35)    0.0(±0.0)  0.33(±0.49)     0.33(±0.49)   \n",
       "GATBMiniaPipeline    94.58(±97.59)  0.01(±0.01)  0.66(±0.49)     0.57(±0.46)   \n",
       "IDBA-UD            202.25(±205.78)  0.01(±0.01)  0.95(±0.07)     0.44(±0.41)   \n",
       "MEGAHIT              70.75(±66.26)  0.03(±0.03)  0.99(±0.01)     0.78(±0.37)   \n",
       "metaSPAdes            59.0(±62.64)  0.04(±0.05)  0.98(±0.03)      0.9(±0.18)   \n",
       "MINIA              565.08(±590.26)  0.01(±0.01)   0.5(±0.52)     0.49(±0.52)   \n",
       "SKESA              113.17(±121.89)  0.01(±0.01)   0.5(±0.52)     0.19(±0.37)   \n",
       "SPAdes               62.17(±53.74)  0.04(±0.04)  0.97(±0.05)     0.77(±0.31)   \n",
       "Unicycler            45.83(±49.13)  0.05(±0.05)   0.5(±0.52)     0.43(±0.47)   \n",
       "VelvetOptimizer    375.67(±554.93)   0.0(±0.01)  0.33(±0.49)     0.28(±0.41)   \n",
       "\n",
       "                  misassemblies multiplicity    parsimony     validity  \n",
       "Assembler                                                               \n",
       "BCALM2                0.0(±0.0)  0.33(±0.49)  0.33(±0.49)  0.33(±0.49)  \n",
       "GATBMiniaPipeline   0.17(±0.39)  0.67(±0.49)   0.67(±0.5)  0.66(±0.49)  \n",
       "IDBA-UD             2.83(±2.37)    1.0(±0.0)   1.8(±1.79)  0.84(±0.31)  \n",
       "MEGAHIT             1.83(±2.12)   1.0(±0.01)  1.02(±0.02)  0.99(±0.02)  \n",
       "metaSPAdes          0.83(±1.11)    1.0(±0.0)  1.02(±0.03)  0.98(±0.03)  \n",
       "MINIA                 0.0(±0.0)   0.5(±0.52)   0.5(±0.52)   0.5(±0.52)  \n",
       "SKESA               0.33(±0.49)   0.5(±0.52)  0.52(±0.54)  0.48(±0.51)  \n",
       "SPAdes              1.17(±1.11)    1.0(±0.0)  1.03(±0.07)  0.98(±0.06)  \n",
       "Unicycler           0.67(±1.15)   0.5(±0.52)   0.5(±0.53)   0.5(±0.52)  \n",
       "VelvetOptimizer     2.17(±3.21)  0.33(±0.49)  0.33(±0.49)  0.33(±0.49)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Lactobacillus_fermentum\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>34.67(±80.96)</td>\n",
       "      <td>2276.17(±4211.52)</td>\n",
       "      <td>1705.67(±3983.58)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>317681.17(±670990.97)</td>\n",
       "      <td>0.17(±0.35)</td>\n",
       "      <td>60.17(±94.49)</td>\n",
       "      <td>0.01(±0.01)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>27.0(±28.21)</td>\n",
       "      <td>20600.83(±20729.94)</td>\n",
       "      <td>19628.5(±20520.71)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>898812.75(±937432.93)</td>\n",
       "      <td>0.47(±0.49)</td>\n",
       "      <td>43.83(±45.14)</td>\n",
       "      <td>0.03(±0.03)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.66(±0.49)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>40.5(±49.71)</td>\n",
       "      <td>16478.5(±18444.33)</td>\n",
       "      <td>15759.67(±18444.43)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>893165.42(±931536.72)</td>\n",
       "      <td>0.47(±0.49)</td>\n",
       "      <td>57.83(±66.49)</td>\n",
       "      <td>0.03(±0.03)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>21.5(±23.06)</td>\n",
       "      <td>22554.0(±22845.25)</td>\n",
       "      <td>21806.0(±22940.55)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>905074.75(±943865.02)</td>\n",
       "      <td>0.48(±0.5)</td>\n",
       "      <td>42.0(±43.29)</td>\n",
       "      <td>0.03(±0.03)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.63(±0.47)</td>\n",
       "      <td>1.33(±1.44)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>27.0(±28.78)</td>\n",
       "      <td>21254.67(±21457.03)</td>\n",
       "      <td>19114.17(±20100.39)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>896610.17(±933661.73)</td>\n",
       "      <td>0.47(±0.49)</td>\n",
       "      <td>43.5(±44.07)</td>\n",
       "      <td>0.03(±0.03)</td>\n",
       "      <td>0.83(±0.39)</td>\n",
       "      <td>0.82(±0.38)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.83(±0.39)</td>\n",
       "      <td>0.84(±0.39)</td>\n",
       "      <td>0.83(±0.39)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>112.42(±119.05)</td>\n",
       "      <td>5502.33(±5297.92)</td>\n",
       "      <td>4980.17(±5281.41)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>874011.83(±911955.09)</td>\n",
       "      <td>0.46(±0.48)</td>\n",
       "      <td>126.25(±132.08)</td>\n",
       "      <td>0.01(±0.02)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.66(±0.48)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "      <td>0.67(±0.49)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>32.17(±33.88)</td>\n",
       "      <td>18663.17(±19562.2)</td>\n",
       "      <td>17273.83(±18269.01)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>884407.0(±923752.3)</td>\n",
       "      <td>0.46(±0.48)</td>\n",
       "      <td>44.83(±46.98)</td>\n",
       "      <td>0.03(±0.03)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>25.67(±27.03)</td>\n",
       "      <td>22543.17(±22095.77)</td>\n",
       "      <td>21440.33(±22590.33)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>896103.67(±933069.36)</td>\n",
       "      <td>0.47(±0.49)</td>\n",
       "      <td>44.33(±45.26)</td>\n",
       "      <td>0.03(±0.03)</td>\n",
       "      <td>0.98(±0.03)</td>\n",
       "      <td>0.98(±0.03)</td>\n",
       "      <td>0.83(±1.95)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.01(±0.01)</td>\n",
       "      <td>0.99(±0.01)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>26.33(±27.65)</td>\n",
       "      <td>21890.17(±22950.16)</td>\n",
       "      <td>21005.33(±22036.18)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>887187.0(±926638.69)</td>\n",
       "      <td>0.47(±0.49)</td>\n",
       "      <td>39.83(±41.63)</td>\n",
       "      <td>0.03(±0.03)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>67.5(±99.74)</td>\n",
       "      <td>3881.33(±5733.73)</td>\n",
       "      <td>3681.33(±5437.89)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>583673.5(±862142.94)</td>\n",
       "      <td>0.31(±0.45)</td>\n",
       "      <td>77.17(±114.02)</td>\n",
       "      <td>0.01(±0.02)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm              L90  \\\n",
       "Assembler                                                            \n",
       "BCALM2               Single k-mer De Bruijn graph    34.67(±80.96)   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph     27.0(±28.21)   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph     40.5(±49.71)   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph     21.5(±23.06)   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph     27.0(±28.78)   \n",
       "MINIA                Single k-mer De Bruijn graph  112.42(±119.05)   \n",
       "SKESA              Multiple k-mer De Bruijn graph    32.17(±33.88)   \n",
       "SPAdes             Multiple k-mer De Bruijn graph    25.67(±27.03)   \n",
       "Unicycler          Multiple k-mer De Bruijn graph    26.33(±27.65)   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph     67.5(±99.74)   \n",
       "\n",
       "                                  NA50                 NG50         Ns  \\\n",
       "Assembler                                                                \n",
       "BCALM2               2276.17(±4211.52)    1705.67(±3983.58)  0.0(±0.0)   \n",
       "GATBMiniaPipeline  20600.83(±20729.94)   19628.5(±20520.71)  0.0(±0.0)   \n",
       "IDBA-UD             16478.5(±18444.33)  15759.67(±18444.43)  0.0(±0.0)   \n",
       "MEGAHIT             22554.0(±22845.25)   21806.0(±22940.55)  0.0(±0.0)   \n",
       "metaSPAdes         21254.67(±21457.03)  19114.17(±20100.39)  0.0(±0.0)   \n",
       "MINIA                5502.33(±5297.92)    4980.17(±5281.41)  0.0(±0.0)   \n",
       "SKESA               18663.17(±19562.2)  17273.83(±18269.01)  0.0(±0.0)   \n",
       "SPAdes             22543.17(±22095.77)  21440.33(±22590.33)  0.0(±0.0)   \n",
       "Unicycler          21890.17(±22950.16)  21005.33(±22036.18)  0.0(±0.0)   \n",
       "VelvetOptimizer      3881.33(±5733.73)    3681.33(±5437.89)  0.0(±0.0)   \n",
       "\n",
       "                          Type              basepairs breadth_of_coverage  \\\n",
       "Assembler                                                                   \n",
       "BCALM2                 Genomic  317681.17(±670990.97)         0.17(±0.35)   \n",
       "GATBMiniaPipeline  Metagenomic  898812.75(±937432.93)         0.47(±0.49)   \n",
       "IDBA-UD            Metagenomic  893165.42(±931536.72)         0.47(±0.49)   \n",
       "MEGAHIT            Metagenomic  905074.75(±943865.02)          0.48(±0.5)   \n",
       "metaSPAdes         Metagenomic  896610.17(±933661.73)         0.47(±0.49)   \n",
       "MINIA                  Genomic  874011.83(±911955.09)         0.46(±0.48)   \n",
       "SKESA                  Genomic    884407.0(±923752.3)         0.46(±0.48)   \n",
       "SPAdes                 Genomic  896103.67(±933069.36)         0.47(±0.49)   \n",
       "Unicycler              Genomic   887187.0(±926638.69)         0.47(±0.49)   \n",
       "VelvetOptimizer        Genomic   583673.5(±862142.94)         0.31(±0.45)   \n",
       "\n",
       "                           contigs   contiguity     identity lowest_identity  \\\n",
       "Assembler                                                                      \n",
       "BCALM2               60.17(±94.49)  0.01(±0.01)   0.5(±0.52)      0.5(±0.52)   \n",
       "GATBMiniaPipeline    43.83(±45.14)  0.03(±0.03)  0.67(±0.49)     0.66(±0.49)   \n",
       "IDBA-UD              57.83(±66.49)  0.03(±0.03)  0.67(±0.49)     0.67(±0.49)   \n",
       "MEGAHIT               42.0(±43.29)  0.03(±0.03)  0.67(±0.49)     0.63(±0.47)   \n",
       "metaSPAdes            43.5(±44.07)  0.03(±0.03)  0.83(±0.39)     0.82(±0.38)   \n",
       "MINIA              126.25(±132.08)  0.01(±0.02)  0.67(±0.49)     0.66(±0.48)   \n",
       "SKESA                44.83(±46.98)  0.03(±0.03)   0.5(±0.52)      0.5(±0.52)   \n",
       "SPAdes               44.33(±45.26)  0.03(±0.03)  0.98(±0.03)     0.98(±0.03)   \n",
       "Unicycler            39.83(±41.63)  0.03(±0.03)   0.5(±0.52)      0.5(±0.52)   \n",
       "VelvetOptimizer     77.17(±114.02)  0.01(±0.02)  0.33(±0.49)     0.33(±0.49)   \n",
       "\n",
       "                  misassemblies multiplicity    parsimony     validity  \n",
       "Assembler                                                               \n",
       "BCALM2                0.0(±0.0)   0.5(±0.52)   0.5(±0.52)   0.5(±0.52)  \n",
       "GATBMiniaPipeline     0.0(±0.0)  0.67(±0.49)  0.67(±0.49)  0.67(±0.49)  \n",
       "IDBA-UD               0.0(±0.0)  0.67(±0.49)  0.67(±0.49)  0.67(±0.49)  \n",
       "MEGAHIT             1.33(±1.44)  0.67(±0.49)  0.67(±0.49)  0.67(±0.49)  \n",
       "metaSPAdes          0.17(±0.39)  0.83(±0.39)  0.84(±0.39)  0.83(±0.39)  \n",
       "MINIA                 0.0(±0.0)  0.67(±0.49)  0.67(±0.49)  0.67(±0.49)  \n",
       "SKESA                 0.0(±0.0)   0.5(±0.52)   0.5(±0.52)   0.5(±0.52)  \n",
       "SPAdes              0.83(±1.95)    1.0(±0.0)  1.01(±0.01)  0.99(±0.01)  \n",
       "Unicycler             0.0(±0.0)   0.5(±0.52)   0.5(±0.52)   0.5(±0.52)  \n",
       "VelvetOptimizer       0.0(±0.0)  0.33(±0.49)  0.33(±0.49)  0.33(±0.49)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Listeria_monocytogenes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>20.42(±31.06)</td>\n",
       "      <td>18785.0(±27736.56)</td>\n",
       "      <td>17807.67(±26866.29)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1198544.25(±1361745.87)</td>\n",
       "      <td>0.4(±0.46)</td>\n",
       "      <td>180.92(±326.85)</td>\n",
       "      <td>0.02(±0.03)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>10.83(±0.94)</td>\n",
       "      <td>351932.67(±59629.94)</td>\n",
       "      <td>302389.5(±78025.69)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2945179.25(±16640.4)</td>\n",
       "      <td>0.98(±0.01)</td>\n",
       "      <td>21.08(±2.97)</td>\n",
       "      <td>0.2(±0.01)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.99(±0.01)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>27.33(±37.72)</td>\n",
       "      <td>269599.5(±140387.79)</td>\n",
       "      <td>269315.5(±140923.15)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2947515.5(±15104.41)</td>\n",
       "      <td>0.99(±0.01)</td>\n",
       "      <td>47.33(±57.82)</td>\n",
       "      <td>0.16(±0.06)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.99(±0.01)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>7.67(±0.98)</td>\n",
       "      <td>449821.33(±95277.74)</td>\n",
       "      <td>450683.67(±93501.07)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2945895.08(±19057.69)</td>\n",
       "      <td>0.98(±0.01)</td>\n",
       "      <td>19.08(±3.29)</td>\n",
       "      <td>0.25(±0.1)</td>\n",
       "      <td>0.99(±0.03)</td>\n",
       "      <td>0.82(±0.32)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>1.01(±0.02)</td>\n",
       "      <td>1.01(±0.02)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>9.0(±2.89)</td>\n",
       "      <td>372265.0(±126651.34)</td>\n",
       "      <td>372265.0(±126651.34)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2953982.17(±14428.14)</td>\n",
       "      <td>0.99(±0.0)</td>\n",
       "      <td>18.33(±6.34)</td>\n",
       "      <td>0.22(±0.07)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.99(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>130.42(±160.69)</td>\n",
       "      <td>32615.25(±23203.96)</td>\n",
       "      <td>31617.83(±22863.1)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2830630.5(±134819.59)</td>\n",
       "      <td>0.95(±0.05)</td>\n",
       "      <td>339.0(±323.81)</td>\n",
       "      <td>0.03(±0.02)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.83(±0.38)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.01)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>47.17(±86.5)</td>\n",
       "      <td>194836.17(±148932.43)</td>\n",
       "      <td>194654.83(±149181.14)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2800686.83(±254111.92)</td>\n",
       "      <td>0.94(±0.08)</td>\n",
       "      <td>236.83(±374.7)</td>\n",
       "      <td>0.13(±0.09)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>146.83(±322.89)</td>\n",
       "      <td>217159.83(±184259.18)</td>\n",
       "      <td>217226.5(±184174.84)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2663180.83(±416898.87)</td>\n",
       "      <td>0.89(±0.14)</td>\n",
       "      <td>534.5(±778.37)</td>\n",
       "      <td>0.13(±0.1)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.91(±0.19)</td>\n",
       "      <td>94.33(±220.32)</td>\n",
       "      <td>1.07(±0.15)</td>\n",
       "      <td>1.07(±0.15)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>13.17(±11.65)</td>\n",
       "      <td>329234.17(±118689.87)</td>\n",
       "      <td>329234.17(±118689.87)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2944906.5(±14176.66)</td>\n",
       "      <td>0.98(±0.0)</td>\n",
       "      <td>25.33(±20.48)</td>\n",
       "      <td>0.2(±0.03)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.99(±0.01)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>22.5(±33.25)</td>\n",
       "      <td>16602.83(±24583.97)</td>\n",
       "      <td>15930.5(±23620.45)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>972774.33(±1436882.94)</td>\n",
       "      <td>0.33(±0.48)</td>\n",
       "      <td>39.5(±58.36)</td>\n",
       "      <td>0.02(±0.03)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm              L90  \\\n",
       "Assembler                                                            \n",
       "BCALM2               Single k-mer De Bruijn graph    20.42(±31.06)   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph     10.83(±0.94)   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph    27.33(±37.72)   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph      7.67(±0.98)   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph       9.0(±2.89)   \n",
       "MINIA                Single k-mer De Bruijn graph  130.42(±160.69)   \n",
       "SKESA              Multiple k-mer De Bruijn graph     47.17(±86.5)   \n",
       "SPAdes             Multiple k-mer De Bruijn graph  146.83(±322.89)   \n",
       "Unicycler          Multiple k-mer De Bruijn graph    13.17(±11.65)   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph     22.5(±33.25)   \n",
       "\n",
       "                                    NA50                   NG50         Ns  \\\n",
       "Assembler                                                                    \n",
       "BCALM2                18785.0(±27736.56)    17807.67(±26866.29)  0.0(±0.0)   \n",
       "GATBMiniaPipeline   351932.67(±59629.94)    302389.5(±78025.69)  0.0(±0.0)   \n",
       "IDBA-UD             269599.5(±140387.79)   269315.5(±140923.15)  0.0(±0.0)   \n",
       "MEGAHIT             449821.33(±95277.74)   450683.67(±93501.07)  0.0(±0.0)   \n",
       "metaSPAdes          372265.0(±126651.34)   372265.0(±126651.34)  0.0(±0.0)   \n",
       "MINIA                32615.25(±23203.96)     31617.83(±22863.1)  0.0(±0.0)   \n",
       "SKESA              194836.17(±148932.43)  194654.83(±149181.14)  0.0(±0.0)   \n",
       "SPAdes             217159.83(±184259.18)   217226.5(±184174.84)  0.0(±0.0)   \n",
       "Unicycler          329234.17(±118689.87)  329234.17(±118689.87)  0.0(±0.0)   \n",
       "VelvetOptimizer      16602.83(±24583.97)     15930.5(±23620.45)  0.0(±0.0)   \n",
       "\n",
       "                          Type                basepairs breadth_of_coverage  \\\n",
       "Assembler                                                                     \n",
       "BCALM2                 Genomic  1198544.25(±1361745.87)          0.4(±0.46)   \n",
       "GATBMiniaPipeline  Metagenomic     2945179.25(±16640.4)         0.98(±0.01)   \n",
       "IDBA-UD            Metagenomic     2947515.5(±15104.41)         0.99(±0.01)   \n",
       "MEGAHIT            Metagenomic    2945895.08(±19057.69)         0.98(±0.01)   \n",
       "metaSPAdes         Metagenomic    2953982.17(±14428.14)          0.99(±0.0)   \n",
       "MINIA                  Genomic    2830630.5(±134819.59)         0.95(±0.05)   \n",
       "SKESA                  Genomic   2800686.83(±254111.92)         0.94(±0.08)   \n",
       "SPAdes                 Genomic   2663180.83(±416898.87)         0.89(±0.14)   \n",
       "Unicycler              Genomic     2944906.5(±14176.66)          0.98(±0.0)   \n",
       "VelvetOptimizer        Genomic   972774.33(±1436882.94)         0.33(±0.48)   \n",
       "\n",
       "                           contigs   contiguity     identity lowest_identity  \\\n",
       "Assembler                                                                      \n",
       "BCALM2             180.92(±326.85)  0.02(±0.03)   0.5(±0.52)      0.5(±0.52)   \n",
       "GATBMiniaPipeline     21.08(±2.97)   0.2(±0.01)    1.0(±0.0)     0.99(±0.01)   \n",
       "IDBA-UD              47.33(±57.82)  0.16(±0.06)    1.0(±0.0)     0.99(±0.01)   \n",
       "MEGAHIT               19.08(±3.29)   0.25(±0.1)  0.99(±0.03)     0.82(±0.32)   \n",
       "metaSPAdes            18.33(±6.34)  0.22(±0.07)    1.0(±0.0)      0.99(±0.0)   \n",
       "MINIA               339.0(±323.81)  0.03(±0.02)    1.0(±0.0)     0.83(±0.38)   \n",
       "SKESA               236.83(±374.7)  0.13(±0.09)    1.0(±0.0)       1.0(±0.0)   \n",
       "SPAdes              534.5(±778.37)   0.13(±0.1)    1.0(±0.0)     0.91(±0.19)   \n",
       "Unicycler            25.33(±20.48)   0.2(±0.03)    1.0(±0.0)     0.99(±0.01)   \n",
       "VelvetOptimizer       39.5(±58.36)  0.02(±0.03)  0.33(±0.49)     0.33(±0.49)   \n",
       "\n",
       "                    misassemblies multiplicity    parsimony     validity  \n",
       "Assembler                                                                 \n",
       "BCALM2                  0.0(±0.0)   0.5(±0.52)   0.5(±0.52)   0.5(±0.52)  \n",
       "GATBMiniaPipeline       0.0(±0.0)    1.0(±0.0)    1.0(±0.0)    1.0(±0.0)  \n",
       "IDBA-UD                 0.0(±0.0)    1.0(±0.0)    1.0(±0.0)    1.0(±0.0)  \n",
       "MEGAHIT               0.33(±0.49)  1.01(±0.02)  1.01(±0.02)    1.0(±0.0)  \n",
       "metaSPAdes              0.0(±0.0)    1.0(±0.0)    1.0(±0.0)    1.0(±0.0)  \n",
       "MINIA                 0.17(±0.39)    1.0(±0.0)   1.0(±0.01)    1.0(±0.0)  \n",
       "SKESA                   0.0(±0.0)    1.0(±0.0)    1.0(±0.0)    1.0(±0.0)  \n",
       "SPAdes             94.33(±220.32)  1.07(±0.15)  1.07(±0.15)    1.0(±0.0)  \n",
       "Unicycler               0.0(±0.0)    1.0(±0.0)    1.0(±0.0)    1.0(±0.0)  \n",
       "VelvetOptimizer         0.0(±0.0)  0.33(±0.49)  0.33(±0.49)  0.33(±0.49)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pseudomonas_aeruginosa\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>406.83(±665.77)</td>\n",
       "      <td>8346.0(±10863.52)</td>\n",
       "      <td>8015.67(±10517.7)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>3187529.5(±3332377.59)</td>\n",
       "      <td>0.47(±0.49)</td>\n",
       "      <td>469.17(±667.83)</td>\n",
       "      <td>0.01(±0.01)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>47.0(±13.8)</td>\n",
       "      <td>173809.83(±51047.17)</td>\n",
       "      <td>172485.0(±52333.37)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>6694504.42(±29451.89)</td>\n",
       "      <td>0.99(±0.0)</td>\n",
       "      <td>112.67(±19.26)</td>\n",
       "      <td>0.06(±0.01)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.99(±0.03)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>65.25(±15.14)</td>\n",
       "      <td>117962.67(±32563.68)</td>\n",
       "      <td>115056.17(±31878.32)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>6699905.75(±4414.15)</td>\n",
       "      <td>0.99(±0.0)</td>\n",
       "      <td>146.5(±24.85)</td>\n",
       "      <td>0.05(±0.01)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.95(±0.03)</td>\n",
       "      <td>0.67(±0.78)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>28.33(±2.31)</td>\n",
       "      <td>249058.0(±22127.07)</td>\n",
       "      <td>250514.0(±24871.89)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>6733274.08(±3931.88)</td>\n",
       "      <td>0.99(±0.0)</td>\n",
       "      <td>79.67(±2.81)</td>\n",
       "      <td>0.07(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.97(±0.04)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>29.0(±7.77)</td>\n",
       "      <td>233006.33(±48648.87)</td>\n",
       "      <td>227289.33(±43495.58)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>6727853.5(±3715.14)</td>\n",
       "      <td>0.99(±0.0)</td>\n",
       "      <td>66.0(±16.46)</td>\n",
       "      <td>0.11(±0.03)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.98(±0.01)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1.0(±0.01)</td>\n",
       "      <td>1.0(±0.01)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>330.83(±40.76)</td>\n",
       "      <td>22426.83(±2003.94)</td>\n",
       "      <td>21766.17(±2116.41)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>6510048.42(±14555.32)</td>\n",
       "      <td>0.96(±0.0)</td>\n",
       "      <td>511.0(±47.51)</td>\n",
       "      <td>0.02(±0.01)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.99(±0.01)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>25.5(±27.92)</td>\n",
       "      <td>70226.5(±74403.23)</td>\n",
       "      <td>71537.67(±78367.82)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>4195442.67(±2806139.97)</td>\n",
       "      <td>0.62(±0.41)</td>\n",
       "      <td>560.67(±731.79)</td>\n",
       "      <td>0.03(±0.03)</td>\n",
       "      <td>0.83(±0.39)</td>\n",
       "      <td>0.77(±0.36)</td>\n",
       "      <td>3.0(±4.43)</td>\n",
       "      <td>0.83(±0.39)</td>\n",
       "      <td>0.83(±0.39)</td>\n",
       "      <td>0.83(±0.39)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>39.17(±13.8)</td>\n",
       "      <td>201858.33(±60787.13)</td>\n",
       "      <td>198958.83(±58264.4)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>6710119.33(±15195.06)</td>\n",
       "      <td>0.99(±0.0)</td>\n",
       "      <td>86.17(±25.75)</td>\n",
       "      <td>0.09(±0.04)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.98(±0.02)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>26.67(±3.8)</td>\n",
       "      <td>262392.33(±56688.27)</td>\n",
       "      <td>262392.33(±56688.27)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>6683616.17(±63040.19)</td>\n",
       "      <td>0.98(±0.01)</td>\n",
       "      <td>69.0(±9.4)</td>\n",
       "      <td>0.11(±0.03)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.97(±0.05)</td>\n",
       "      <td>0.5(±1.17)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>5.17(±7.64)</td>\n",
       "      <td>93246.0(±138021.07)</td>\n",
       "      <td>127325.67(±189859.82)</td>\n",
       "      <td>3093.67(±5214.96)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2235285.83(±3301739.97)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>24.67(±36.84)</td>\n",
       "      <td>0.03(±0.05)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.3(±0.45)</td>\n",
       "      <td>2.33(±3.65)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm              L90  \\\n",
       "Assembler                                                            \n",
       "BCALM2               Single k-mer De Bruijn graph  406.83(±665.77)   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph      47.0(±13.8)   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph    65.25(±15.14)   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph     28.33(±2.31)   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph      29.0(±7.77)   \n",
       "MINIA                Single k-mer De Bruijn graph   330.83(±40.76)   \n",
       "SKESA              Multiple k-mer De Bruijn graph     25.5(±27.92)   \n",
       "SPAdes             Multiple k-mer De Bruijn graph     39.17(±13.8)   \n",
       "Unicycler          Multiple k-mer De Bruijn graph      26.67(±3.8)   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph      5.17(±7.64)   \n",
       "\n",
       "                                   NA50                   NG50  \\\n",
       "Assembler                                                        \n",
       "BCALM2                8346.0(±10863.52)      8015.67(±10517.7)   \n",
       "GATBMiniaPipeline  173809.83(±51047.17)    172485.0(±52333.37)   \n",
       "IDBA-UD            117962.67(±32563.68)   115056.17(±31878.32)   \n",
       "MEGAHIT             249058.0(±22127.07)    250514.0(±24871.89)   \n",
       "metaSPAdes         233006.33(±48648.87)   227289.33(±43495.58)   \n",
       "MINIA                22426.83(±2003.94)     21766.17(±2116.41)   \n",
       "SKESA                70226.5(±74403.23)    71537.67(±78367.82)   \n",
       "SPAdes             201858.33(±60787.13)    198958.83(±58264.4)   \n",
       "Unicycler          262392.33(±56688.27)   262392.33(±56688.27)   \n",
       "VelvetOptimizer     93246.0(±138021.07)  127325.67(±189859.82)   \n",
       "\n",
       "                                  Ns         Type                basepairs  \\\n",
       "Assembler                                                                    \n",
       "BCALM2                     0.0(±0.0)      Genomic   3187529.5(±3332377.59)   \n",
       "GATBMiniaPipeline          0.0(±0.0)  Metagenomic    6694504.42(±29451.89)   \n",
       "IDBA-UD                    0.0(±0.0)  Metagenomic     6699905.75(±4414.15)   \n",
       "MEGAHIT                    0.0(±0.0)  Metagenomic     6733274.08(±3931.88)   \n",
       "metaSPAdes                 0.0(±0.0)  Metagenomic      6727853.5(±3715.14)   \n",
       "MINIA                      0.0(±0.0)      Genomic    6510048.42(±14555.32)   \n",
       "SKESA                      0.0(±0.0)      Genomic  4195442.67(±2806139.97)   \n",
       "SPAdes                     0.0(±0.0)      Genomic    6710119.33(±15195.06)   \n",
       "Unicycler                  0.0(±0.0)      Genomic    6683616.17(±63040.19)   \n",
       "VelvetOptimizer    3093.67(±5214.96)      Genomic  2235285.83(±3301739.97)   \n",
       "\n",
       "                  breadth_of_coverage          contigs   contiguity  \\\n",
       "Assembler                                                             \n",
       "BCALM2                    0.47(±0.49)  469.17(±667.83)  0.01(±0.01)   \n",
       "GATBMiniaPipeline          0.99(±0.0)   112.67(±19.26)  0.06(±0.01)   \n",
       "IDBA-UD                    0.99(±0.0)    146.5(±24.85)  0.05(±0.01)   \n",
       "MEGAHIT                    0.99(±0.0)     79.67(±2.81)   0.07(±0.0)   \n",
       "metaSPAdes                 0.99(±0.0)     66.0(±16.46)  0.11(±0.03)   \n",
       "MINIA                      0.96(±0.0)    511.0(±47.51)  0.02(±0.01)   \n",
       "SKESA                     0.62(±0.41)  560.67(±731.79)  0.03(±0.03)   \n",
       "SPAdes                     0.99(±0.0)    86.17(±25.75)  0.09(±0.04)   \n",
       "Unicycler                 0.98(±0.01)       69.0(±9.4)  0.11(±0.03)   \n",
       "VelvetOptimizer           0.33(±0.49)    24.67(±36.84)  0.03(±0.05)   \n",
       "\n",
       "                      identity lowest_identity misassemblies multiplicity  \\\n",
       "Assembler                                                                   \n",
       "BCALM2              0.5(±0.52)      0.5(±0.52)     0.0(±0.0)   0.5(±0.52)   \n",
       "GATBMiniaPipeline    1.0(±0.0)     0.99(±0.03)     0.0(±0.0)    1.0(±0.0)   \n",
       "IDBA-UD              1.0(±0.0)     0.95(±0.03)   0.67(±0.78)    1.0(±0.0)   \n",
       "MEGAHIT              1.0(±0.0)     0.97(±0.04)   0.17(±0.39)    1.0(±0.0)   \n",
       "metaSPAdes           1.0(±0.0)     0.98(±0.01)     0.0(±0.0)   1.0(±0.01)   \n",
       "MINIA                1.0(±0.0)     0.99(±0.01)     0.0(±0.0)    1.0(±0.0)   \n",
       "SKESA              0.83(±0.39)     0.77(±0.36)    3.0(±4.43)  0.83(±0.39)   \n",
       "SPAdes               1.0(±0.0)     0.98(±0.02)   0.17(±0.39)    1.0(±0.0)   \n",
       "Unicycler            1.0(±0.0)     0.97(±0.05)    0.5(±1.17)    1.0(±0.0)   \n",
       "VelvetOptimizer    0.33(±0.49)      0.3(±0.45)   2.33(±3.65)  0.33(±0.49)   \n",
       "\n",
       "                     parsimony     validity  \n",
       "Assembler                                    \n",
       "BCALM2              0.5(±0.52)   0.5(±0.52)  \n",
       "GATBMiniaPipeline    1.0(±0.0)    1.0(±0.0)  \n",
       "IDBA-UD              1.0(±0.0)    1.0(±0.0)  \n",
       "MEGAHIT              1.0(±0.0)    1.0(±0.0)  \n",
       "metaSPAdes          1.0(±0.01)    1.0(±0.0)  \n",
       "MINIA                1.0(±0.0)    1.0(±0.0)  \n",
       "SKESA              0.83(±0.39)  0.83(±0.39)  \n",
       "SPAdes               1.0(±0.0)    1.0(±0.0)  \n",
       "Unicycler            1.0(±0.0)    1.0(±0.0)  \n",
       "VelvetOptimizer    0.33(±0.49)  0.33(±0.49)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salmonella_enterica\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1158.83(±1945.24)</td>\n",
       "      <td>804.33(±1372.03)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1112956.58(±1669062.03)</td>\n",
       "      <td>0.23(±0.35)</td>\n",
       "      <td>437.5(±661.39)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>40.67(±42.77)</td>\n",
       "      <td>31951.17(±32820.08)</td>\n",
       "      <td>31071.33(±32883.36)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2326651.67(±2428224.81)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>84.0(±86.42)</td>\n",
       "      <td>0.03(±0.03)</td>\n",
       "      <td>0.83(±0.39)</td>\n",
       "      <td>0.76(±0.39)</td>\n",
       "      <td>0.67(±1.15)</td>\n",
       "      <td>0.83(±0.39)</td>\n",
       "      <td>0.84(±0.39)</td>\n",
       "      <td>0.83(±0.39)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>108.92(±115.95)</td>\n",
       "      <td>12896.33(±12411.72)</td>\n",
       "      <td>12067.75(±12819.8)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2295911.58(±2392852.98)</td>\n",
       "      <td>0.48(±0.5)</td>\n",
       "      <td>189.58(±197.3)</td>\n",
       "      <td>0.01(±0.01)</td>\n",
       "      <td>0.97(±0.04)</td>\n",
       "      <td>0.6(±0.36)</td>\n",
       "      <td>2.83(±3.49)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.03(±0.03)</td>\n",
       "      <td>0.98(±0.03)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>20.33(±21.75)</td>\n",
       "      <td>55259.0(±57274.72)</td>\n",
       "      <td>58692.83(±63264.03)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2342960.08(±2436179.16)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>56.67(±51.7)</td>\n",
       "      <td>0.05(±0.05)</td>\n",
       "      <td>0.99(±0.01)</td>\n",
       "      <td>0.86(±0.19)</td>\n",
       "      <td>1.0(±1.48)</td>\n",
       "      <td>1.0(±0.01)</td>\n",
       "      <td>1.02(±0.01)</td>\n",
       "      <td>0.98(±0.02)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>21.33(±28.64)</td>\n",
       "      <td>72896.5(±84848.7)</td>\n",
       "      <td>71169.5(±84212.5)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2340048.0(±2431405.11)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>47.5(±50.45)</td>\n",
       "      <td>0.04(±0.05)</td>\n",
       "      <td>0.98(±0.03)</td>\n",
       "      <td>0.82(±0.35)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.02(±0.06)</td>\n",
       "      <td>0.98(±0.06)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>3137.17(±1905.63)</td>\n",
       "      <td>1742.83(±1821.61)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1909982.0(±1993558.04)</td>\n",
       "      <td>0.4(±0.42)</td>\n",
       "      <td>545.92(±569.2)</td>\n",
       "      <td>0.01(±0.01)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>0.96(±0.08)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>59.0(±64.84)</td>\n",
       "      <td>23731.83(±25590.8)</td>\n",
       "      <td>23076.5(±24728.76)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2300371.83(±2402686.59)</td>\n",
       "      <td>0.48(±0.5)</td>\n",
       "      <td>104.5(±112.93)</td>\n",
       "      <td>0.02(±0.02)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.31(±0.4)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.53)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>17.83(±20.42)</td>\n",
       "      <td>68132.5(±73377.61)</td>\n",
       "      <td>67903.0(±74527.17)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2338166.67(±2422256.79)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>51.0(±42.3)</td>\n",
       "      <td>0.05(±0.05)</td>\n",
       "      <td>0.97(±0.04)</td>\n",
       "      <td>0.72(±0.35)</td>\n",
       "      <td>0.67(±0.78)</td>\n",
       "      <td>1.0(±0.0)</td>\n",
       "      <td>1.04(±0.06)</td>\n",
       "      <td>0.97(±0.05)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>13.67(±15.08)</td>\n",
       "      <td>96287.17(±102858.85)</td>\n",
       "      <td>96287.17(±102858.85)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2324824.0(±2428210.58)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>33.0(±36.27)</td>\n",
       "      <td>0.05(±0.05)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.4(±0.47)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.51(±0.53)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1825.17(±2696.5)</td>\n",
       "      <td>1487.83(±2197.9)</td>\n",
       "      <td>1092.0(±1641.04)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1375574.83(±2031862.14)</td>\n",
       "      <td>0.29(±0.43)</td>\n",
       "      <td>360.0(±531.81)</td>\n",
       "      <td>0.0(±0.01)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.26(±0.4)</td>\n",
       "      <td>1.17(±1.75)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm              L90  \\\n",
       "Assembler                                                            \n",
       "BCALM2               Single k-mer De Bruijn graph        0.0(±0.0)   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph    40.67(±42.77)   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  108.92(±115.95)   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph    20.33(±21.75)   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph    21.33(±28.64)   \n",
       "MINIA                Single k-mer De Bruijn graph        0.0(±0.0)   \n",
       "SKESA              Multiple k-mer De Bruijn graph     59.0(±64.84)   \n",
       "SPAdes             Multiple k-mer De Bruijn graph    17.83(±20.42)   \n",
       "Unicycler          Multiple k-mer De Bruijn graph    13.67(±15.08)   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph        0.0(±0.0)   \n",
       "\n",
       "                                   NA50                  NG50  \\\n",
       "Assembler                                                       \n",
       "BCALM2                1158.83(±1945.24)      804.33(±1372.03)   \n",
       "GATBMiniaPipeline   31951.17(±32820.08)   31071.33(±32883.36)   \n",
       "IDBA-UD             12896.33(±12411.72)    12067.75(±12819.8)   \n",
       "MEGAHIT              55259.0(±57274.72)   58692.83(±63264.03)   \n",
       "metaSPAdes            72896.5(±84848.7)     71169.5(±84212.5)   \n",
       "MINIA                 3137.17(±1905.63)     1742.83(±1821.61)   \n",
       "SKESA                23731.83(±25590.8)    23076.5(±24728.76)   \n",
       "SPAdes               68132.5(±73377.61)    67903.0(±74527.17)   \n",
       "Unicycler          96287.17(±102858.85)  96287.17(±102858.85)   \n",
       "VelvetOptimizer        1825.17(±2696.5)      1487.83(±2197.9)   \n",
       "\n",
       "                                 Ns         Type                basepairs  \\\n",
       "Assembler                                                                   \n",
       "BCALM2                    0.0(±0.0)      Genomic  1112956.58(±1669062.03)   \n",
       "GATBMiniaPipeline         0.0(±0.0)  Metagenomic  2326651.67(±2428224.81)   \n",
       "IDBA-UD                   0.0(±0.0)  Metagenomic  2295911.58(±2392852.98)   \n",
       "MEGAHIT                   0.0(±0.0)  Metagenomic  2342960.08(±2436179.16)   \n",
       "metaSPAdes                0.0(±0.0)  Metagenomic   2340048.0(±2431405.11)   \n",
       "MINIA                     0.0(±0.0)      Genomic   1909982.0(±1993558.04)   \n",
       "SKESA                     0.0(±0.0)      Genomic  2300371.83(±2402686.59)   \n",
       "SPAdes                    0.0(±0.0)      Genomic  2338166.67(±2422256.79)   \n",
       "Unicycler                 0.0(±0.0)      Genomic   2324824.0(±2428210.58)   \n",
       "VelvetOptimizer    1092.0(±1641.04)      Genomic  1375574.83(±2031862.14)   \n",
       "\n",
       "                  breadth_of_coverage         contigs   contiguity  \\\n",
       "Assembler                                                            \n",
       "BCALM2                    0.23(±0.35)  437.5(±661.39)    0.0(±0.0)   \n",
       "GATBMiniaPipeline         0.49(±0.51)    84.0(±86.42)  0.03(±0.03)   \n",
       "IDBA-UD                    0.48(±0.5)  189.58(±197.3)  0.01(±0.01)   \n",
       "MEGAHIT                   0.49(±0.51)    56.67(±51.7)  0.05(±0.05)   \n",
       "metaSPAdes                0.49(±0.51)    47.5(±50.45)  0.04(±0.05)   \n",
       "MINIA                      0.4(±0.42)  545.92(±569.2)  0.01(±0.01)   \n",
       "SKESA                      0.48(±0.5)  104.5(±112.93)  0.02(±0.02)   \n",
       "SPAdes                    0.49(±0.51)     51.0(±42.3)  0.05(±0.05)   \n",
       "Unicycler                 0.49(±0.51)    33.0(±36.27)  0.05(±0.05)   \n",
       "VelvetOptimizer           0.29(±0.43)  360.0(±531.81)   0.0(±0.01)   \n",
       "\n",
       "                      identity lowest_identity misassemblies multiplicity  \\\n",
       "Assembler                                                                   \n",
       "BCALM2             0.33(±0.49)     0.33(±0.49)     0.0(±0.0)  0.33(±0.49)   \n",
       "GATBMiniaPipeline  0.83(±0.39)     0.76(±0.39)   0.67(±1.15)  0.83(±0.39)   \n",
       "IDBA-UD            0.97(±0.04)      0.6(±0.36)   2.83(±3.49)    1.0(±0.0)   \n",
       "MEGAHIT            0.99(±0.01)     0.86(±0.19)    1.0(±1.48)   1.0(±0.01)   \n",
       "metaSPAdes         0.98(±0.03)     0.82(±0.35)   0.17(±0.39)    1.0(±0.0)   \n",
       "MINIA                1.0(±0.0)     0.96(±0.08)     0.0(±0.0)    1.0(±0.0)   \n",
       "SKESA               0.5(±0.52)      0.31(±0.4)   0.33(±0.49)   0.5(±0.52)   \n",
       "SPAdes             0.97(±0.04)     0.72(±0.35)   0.67(±0.78)    1.0(±0.0)   \n",
       "Unicycler           0.5(±0.52)      0.4(±0.47)   0.17(±0.39)   0.5(±0.52)   \n",
       "VelvetOptimizer    0.33(±0.49)      0.26(±0.4)   1.17(±1.75)  0.33(±0.49)   \n",
       "\n",
       "                     parsimony     validity  \n",
       "Assembler                                    \n",
       "BCALM2             0.33(±0.49)  0.33(±0.49)  \n",
       "GATBMiniaPipeline  0.84(±0.39)  0.83(±0.39)  \n",
       "IDBA-UD            1.03(±0.03)  0.98(±0.03)  \n",
       "MEGAHIT            1.02(±0.01)  0.98(±0.02)  \n",
       "metaSPAdes         1.02(±0.06)  0.98(±0.06)  \n",
       "MINIA                1.0(±0.0)    1.0(±0.0)  \n",
       "SKESA               0.5(±0.53)   0.5(±0.52)  \n",
       "SPAdes             1.04(±0.06)  0.97(±0.05)  \n",
       "Unicycler          0.51(±0.53)   0.5(±0.52)  \n",
       "VelvetOptimizer    0.33(±0.49)  0.33(±0.49)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Staphylococcus_aureus\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>18.67(±43.6)</td>\n",
       "      <td>5171.83(±11444.51)</td>\n",
       "      <td>4860.83(±11352.46)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>586836.67(±1003940.87)</td>\n",
       "      <td>0.22(±0.37)</td>\n",
       "      <td>137.17(±247.77)</td>\n",
       "      <td>0.01(±0.01)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>16.67(±17.72)</td>\n",
       "      <td>42504.0(±45834.0)</td>\n",
       "      <td>42085.83(±47098.12)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1337608.58(±1397088.76)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>27.67(±29.37)</td>\n",
       "      <td>0.05(±0.05)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.33(±0.78)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>32.83(±41.97)</td>\n",
       "      <td>22172.17(±24955.53)</td>\n",
       "      <td>22152.67(±24951.85)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1331546.17(±1390758.02)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>54.17(±66.16)</td>\n",
       "      <td>0.03(±0.03)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.42(±0.46)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>7.17(±8.08)</td>\n",
       "      <td>73781.5(±87016.61)</td>\n",
       "      <td>107260.0(±136205.58)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1340715.33(±1400331.53)</td>\n",
       "      <td>0.49(±0.52)</td>\n",
       "      <td>21.83(±23.01)</td>\n",
       "      <td>0.06(±0.07)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.42(±0.46)</td>\n",
       "      <td>1.33(±1.67)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>8.67(±11.36)</td>\n",
       "      <td>108771.83(±150051.03)</td>\n",
       "      <td>108771.83(±150051.03)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1339675.67(±1399248.63)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>15.33(±18.86)</td>\n",
       "      <td>0.1(±0.14)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.49(±0.52)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>60.67(±64.11)</td>\n",
       "      <td>13787.83(±14562.52)</td>\n",
       "      <td>13552.67(±14317.83)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1299647.92(±1357438.92)</td>\n",
       "      <td>0.48(±0.5)</td>\n",
       "      <td>92.92(±97.68)</td>\n",
       "      <td>0.02(±0.02)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>21.33(±22.32)</td>\n",
       "      <td>26558.67(±27739.65)</td>\n",
       "      <td>26302.17(±27476.61)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1326801.67(±1385802.73)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>35.5(±37.22)</td>\n",
       "      <td>0.03(±0.03)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>8.5(±11.28)</td>\n",
       "      <td>109600.33(±147100.36)</td>\n",
       "      <td>109600.33(±147100.36)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1336849.0(±1396295.96)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>14.33(±18.7)</td>\n",
       "      <td>0.09(±0.09)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>5.0(±5.91)</td>\n",
       "      <td>140460.5(±158318.33)</td>\n",
       "      <td>130585.33(±154313.21)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1336196.33(±1395613.73)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>12.0(±13.06)</td>\n",
       "      <td>0.16(±0.19)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>35.33(±52.2)</td>\n",
       "      <td>10368.67(±15332.37)</td>\n",
       "      <td>9955.5(±14705.84)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>867548.33(±1281453.9)</td>\n",
       "      <td>0.32(±0.47)</td>\n",
       "      <td>55.5(±81.99)</td>\n",
       "      <td>0.01(±0.02)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm            L90  \\\n",
       "Assembler                                                          \n",
       "BCALM2               Single k-mer De Bruijn graph   18.67(±43.6)   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph  16.67(±17.72)   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  32.83(±41.97)   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph    7.17(±8.08)   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph   8.67(±11.36)   \n",
       "MINIA                Single k-mer De Bruijn graph  60.67(±64.11)   \n",
       "SKESA              Multiple k-mer De Bruijn graph  21.33(±22.32)   \n",
       "SPAdes             Multiple k-mer De Bruijn graph    8.5(±11.28)   \n",
       "Unicycler          Multiple k-mer De Bruijn graph     5.0(±5.91)   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph   35.33(±52.2)   \n",
       "\n",
       "                                    NA50                   NG50         Ns  \\\n",
       "Assembler                                                                    \n",
       "BCALM2                5171.83(±11444.51)     4860.83(±11352.46)  0.0(±0.0)   \n",
       "GATBMiniaPipeline      42504.0(±45834.0)    42085.83(±47098.12)  0.0(±0.0)   \n",
       "IDBA-UD              22172.17(±24955.53)    22152.67(±24951.85)  0.0(±0.0)   \n",
       "MEGAHIT               73781.5(±87016.61)   107260.0(±136205.58)  0.0(±0.0)   \n",
       "metaSPAdes         108771.83(±150051.03)  108771.83(±150051.03)  0.0(±0.0)   \n",
       "MINIA                13787.83(±14562.52)    13552.67(±14317.83)  0.0(±0.0)   \n",
       "SKESA                26558.67(±27739.65)    26302.17(±27476.61)  0.0(±0.0)   \n",
       "SPAdes             109600.33(±147100.36)  109600.33(±147100.36)  0.0(±0.0)   \n",
       "Unicycler           140460.5(±158318.33)  130585.33(±154313.21)  0.0(±0.0)   \n",
       "VelvetOptimizer      10368.67(±15332.37)      9955.5(±14705.84)  0.0(±0.0)   \n",
       "\n",
       "                          Type                basepairs breadth_of_coverage  \\\n",
       "Assembler                                                                     \n",
       "BCALM2                 Genomic   586836.67(±1003940.87)         0.22(±0.37)   \n",
       "GATBMiniaPipeline  Metagenomic  1337608.58(±1397088.76)         0.49(±0.51)   \n",
       "IDBA-UD            Metagenomic  1331546.17(±1390758.02)         0.49(±0.51)   \n",
       "MEGAHIT            Metagenomic  1340715.33(±1400331.53)         0.49(±0.52)   \n",
       "metaSPAdes         Metagenomic  1339675.67(±1399248.63)         0.49(±0.51)   \n",
       "MINIA                  Genomic  1299647.92(±1357438.92)          0.48(±0.5)   \n",
       "SKESA                  Genomic  1326801.67(±1385802.73)         0.49(±0.51)   \n",
       "SPAdes                 Genomic   1336849.0(±1396295.96)         0.49(±0.51)   \n",
       "Unicycler              Genomic  1336196.33(±1395613.73)         0.49(±0.51)   \n",
       "VelvetOptimizer        Genomic    867548.33(±1281453.9)         0.32(±0.47)   \n",
       "\n",
       "                           contigs   contiguity     identity lowest_identity  \\\n",
       "Assembler                                                                      \n",
       "BCALM2             137.17(±247.77)  0.01(±0.01)  0.33(±0.49)     0.33(±0.49)   \n",
       "GATBMiniaPipeline    27.67(±29.37)  0.05(±0.05)   0.5(±0.52)      0.5(±0.52)   \n",
       "IDBA-UD              54.17(±66.16)  0.03(±0.03)   0.5(±0.52)     0.42(±0.46)   \n",
       "MEGAHIT              21.83(±23.01)  0.06(±0.07)   0.5(±0.52)     0.42(±0.46)   \n",
       "metaSPAdes           15.33(±18.86)   0.1(±0.14)   0.5(±0.52)     0.49(±0.52)   \n",
       "MINIA                92.92(±97.68)  0.02(±0.02)   0.5(±0.52)      0.5(±0.52)   \n",
       "SKESA                 35.5(±37.22)  0.03(±0.03)   0.5(±0.52)      0.5(±0.52)   \n",
       "SPAdes                14.33(±18.7)  0.09(±0.09)   0.5(±0.52)      0.5(±0.52)   \n",
       "Unicycler             12.0(±13.06)  0.16(±0.19)   0.5(±0.52)      0.5(±0.52)   \n",
       "VelvetOptimizer       55.5(±81.99)  0.01(±0.02)  0.33(±0.49)     0.33(±0.49)   \n",
       "\n",
       "                  misassemblies multiplicity    parsimony     validity  \n",
       "Assembler                                                               \n",
       "BCALM2                0.0(±0.0)  0.33(±0.49)  0.33(±0.49)  0.33(±0.49)  \n",
       "GATBMiniaPipeline   0.33(±0.78)   0.5(±0.52)   0.5(±0.52)   0.5(±0.52)  \n",
       "IDBA-UD               0.0(±0.0)   0.5(±0.52)   0.5(±0.52)   0.5(±0.52)  \n",
       "MEGAHIT             1.33(±1.67)   0.5(±0.52)   0.5(±0.52)   0.5(±0.52)  \n",
       "metaSPAdes            0.0(±0.0)   0.5(±0.52)   0.5(±0.52)   0.5(±0.52)  \n",
       "MINIA                 0.0(±0.0)   0.5(±0.52)   0.5(±0.52)   0.5(±0.52)  \n",
       "SKESA                 0.0(±0.0)   0.5(±0.52)   0.5(±0.52)   0.5(±0.52)  \n",
       "SPAdes              0.17(±0.39)   0.5(±0.52)   0.5(±0.52)   0.5(±0.52)  \n",
       "Unicycler           0.17(±0.39)   0.5(±0.52)   0.5(±0.52)   0.5(±0.52)  \n",
       "VelvetOptimizer       0.0(±0.0)  0.33(±0.49)  0.33(±0.49)  0.33(±0.49)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Staphylococcus_aureus_plasmid1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1056.0(±2466.28)</td>\n",
       "      <td>1056.0(±2466.28)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1056.0(±2466.28)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>3195.92(±3338.67)</td>\n",
       "      <td>3195.92(±3338.67)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3165.83(±3306.61)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.53)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.53)</td>\n",
       "      <td>0.5(±0.53)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1854.83(±2778.99)</td>\n",
       "      <td>1854.83(±2778.99)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1849.83(±2773.24)</td>\n",
       "      <td>0.29(±0.44)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.29(±0.44)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>3244.5(±3392.54)</td>\n",
       "      <td>3244.5(±3392.54)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3168.17(±3309.04)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.51(±0.54)</td>\n",
       "      <td>0.49(±0.51)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>2818.17(±3032.84)</td>\n",
       "      <td>2818.17(±3032.84)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>3146.0(±3286.21)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.67(±0.78)</td>\n",
       "      <td>0.44(±0.48)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1271.0(±2416.47)</td>\n",
       "      <td>1056.0(±2466.28)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1271.0(±2416.47)</td>\n",
       "      <td>0.2(±0.38)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.2(±0.38)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>2287.5(±2829.83)</td>\n",
       "      <td>2287.5(±2829.83)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2819.33(±2972.63)</td>\n",
       "      <td>0.44(±0.47)</td>\n",
       "      <td>0.83(±1.11)</td>\n",
       "      <td>0.36(±0.45)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.49(±0.52)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>2112.0(±3119.63)</td>\n",
       "      <td>2112.0(±3119.63)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2112.0(±3119.63)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "      <td>0.33(±0.49)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>3137.5(±3277.37)</td>\n",
       "      <td>3137.5(±3277.37)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>3132.5(±3272.01)</td>\n",
       "      <td>0.49(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.49(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "      <td>0.5(±0.52)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>1056.0(±2466.28)</td>\n",
       "      <td>1056.0(±2466.28)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1056.0(±2466.28)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm          L90  \\\n",
       "Assembler                                                        \n",
       "BCALM2               Single k-mer De Bruijn graph    0.0(±0.0)   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph    0.0(±0.0)   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph    0.0(±0.0)   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph    0.0(±0.0)   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph  0.17(±0.39)   \n",
       "MINIA                Single k-mer De Bruijn graph    0.0(±0.0)   \n",
       "SKESA              Multiple k-mer De Bruijn graph    0.0(±0.0)   \n",
       "SPAdes             Multiple k-mer De Bruijn graph    0.0(±0.0)   \n",
       "Unicycler          Multiple k-mer De Bruijn graph    0.0(±0.0)   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph    0.0(±0.0)   \n",
       "\n",
       "                                NA50               NG50         Ns  \\\n",
       "Assembler                                                            \n",
       "BCALM2              1056.0(±2466.28)   1056.0(±2466.28)  0.0(±0.0)   \n",
       "GATBMiniaPipeline  3195.92(±3338.67)  3195.92(±3338.67)  0.0(±0.0)   \n",
       "IDBA-UD            1854.83(±2778.99)  1854.83(±2778.99)  0.0(±0.0)   \n",
       "MEGAHIT             3244.5(±3392.54)   3244.5(±3392.54)  0.0(±0.0)   \n",
       "metaSPAdes         2818.17(±3032.84)  2818.17(±3032.84)  0.0(±0.0)   \n",
       "MINIA               1271.0(±2416.47)   1056.0(±2466.28)  0.0(±0.0)   \n",
       "SKESA               2287.5(±2829.83)   2287.5(±2829.83)  0.0(±0.0)   \n",
       "SPAdes              2112.0(±3119.63)   2112.0(±3119.63)  0.0(±0.0)   \n",
       "Unicycler           3137.5(±3277.37)   3137.5(±3277.37)  0.0(±0.0)   \n",
       "VelvetOptimizer     1056.0(±2466.28)   1056.0(±2466.28)  0.0(±0.0)   \n",
       "\n",
       "                          Type          basepairs breadth_of_coverage  \\\n",
       "Assembler                                                               \n",
       "BCALM2                 Genomic   1056.0(±2466.28)         0.17(±0.39)   \n",
       "GATBMiniaPipeline  Metagenomic  3165.83(±3306.61)          0.5(±0.52)   \n",
       "IDBA-UD            Metagenomic  1849.83(±2773.24)         0.29(±0.44)   \n",
       "MEGAHIT            Metagenomic  3168.17(±3309.04)          0.5(±0.52)   \n",
       "metaSPAdes         Metagenomic   3146.0(±3286.21)          0.5(±0.52)   \n",
       "MINIA                  Genomic   1271.0(±2416.47)          0.2(±0.38)   \n",
       "SKESA                  Genomic  2819.33(±2972.63)         0.44(±0.47)   \n",
       "SPAdes                 Genomic   2112.0(±3119.63)         0.33(±0.49)   \n",
       "Unicycler              Genomic   3132.5(±3272.01)         0.49(±0.52)   \n",
       "VelvetOptimizer        Genomic   1056.0(±2466.28)         0.17(±0.39)   \n",
       "\n",
       "                       contigs   contiguity     identity lowest_identity  \\\n",
       "Assembler                                                                  \n",
       "BCALM2             0.17(±0.39)  0.17(±0.39)  0.17(±0.39)     0.17(±0.39)   \n",
       "GATBMiniaPipeline   0.5(±0.52)   0.5(±0.53)   0.5(±0.52)      0.5(±0.52)   \n",
       "IDBA-UD            0.33(±0.49)  0.29(±0.44)  0.33(±0.49)     0.33(±0.49)   \n",
       "MEGAHIT             0.5(±0.52)   0.5(±0.52)  0.49(±0.51)     0.49(±0.51)   \n",
       "metaSPAdes         0.67(±0.78)  0.44(±0.48)   0.5(±0.52)      0.5(±0.52)   \n",
       "MINIA              0.33(±0.49)   0.2(±0.38)  0.33(±0.49)     0.33(±0.49)   \n",
       "SKESA              0.83(±1.11)  0.36(±0.45)   0.5(±0.52)     0.49(±0.52)   \n",
       "SPAdes             0.33(±0.49)  0.33(±0.49)  0.33(±0.49)     0.33(±0.49)   \n",
       "Unicycler           0.5(±0.52)  0.49(±0.52)   0.5(±0.52)      0.5(±0.52)   \n",
       "VelvetOptimizer    0.17(±0.39)  0.17(±0.39)  0.17(±0.39)     0.17(±0.39)   \n",
       "\n",
       "                  misassemblies multiplicity    parsimony     validity  \n",
       "Assembler                                                               \n",
       "BCALM2                0.0(±0.0)  0.17(±0.39)  0.17(±0.39)  0.17(±0.39)  \n",
       "GATBMiniaPipeline     0.0(±0.0)   0.5(±0.53)   0.5(±0.53)   0.5(±0.52)  \n",
       "IDBA-UD               0.0(±0.0)  0.33(±0.49)  0.33(±0.49)  0.33(±0.49)  \n",
       "MEGAHIT               0.0(±0.0)   0.5(±0.52)  0.51(±0.54)  0.49(±0.51)  \n",
       "metaSPAdes            0.0(±0.0)   0.5(±0.52)   0.5(±0.52)   0.5(±0.52)  \n",
       "MINIA                 0.0(±0.0)  0.33(±0.49)  0.33(±0.49)  0.33(±0.49)  \n",
       "SKESA                 0.0(±0.0)   0.5(±0.52)   0.5(±0.52)   0.5(±0.52)  \n",
       "SPAdes                0.0(±0.0)  0.33(±0.49)  0.33(±0.49)  0.33(±0.49)  \n",
       "Unicycler             0.0(±0.0)   0.5(±0.52)   0.5(±0.52)   0.5(±0.52)  \n",
       "VelvetOptimizer       0.0(±0.0)  0.17(±0.39)  0.17(±0.39)  0.17(±0.39)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Staphylococcus_aureus_plasmid2\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>397.0(±927.2)</td>\n",
       "      <td>397.0(±927.2)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>369.33(±862.58)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.18(±0.42)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.18(±0.42)</td>\n",
       "      <td>0.18(±0.42)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>385.83(±901.11)</td>\n",
       "      <td>385.83(±901.11)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>369.33(±862.58)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.41)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.17(±0.41)</td>\n",
       "      <td>0.17(±0.41)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>392.83(±917.46)</td>\n",
       "      <td>392.83(±917.46)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>369.33(±862.58)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.18(±0.41)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.18(±0.41)</td>\n",
       "      <td>0.18(±0.41)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>378.5(±883.99)</td>\n",
       "      <td>378.5(±883.99)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>369.33(±862.58)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.4)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.17(±0.4)</td>\n",
       "      <td>0.17(±0.4)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>334.17(±780.45)</td>\n",
       "      <td>334.17(±780.45)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>334.17(±780.45)</td>\n",
       "      <td>0.15(±0.35)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.15(±0.35)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>326.83(±763.32)</td>\n",
       "      <td>326.83(±763.32)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>326.83(±763.32)</td>\n",
       "      <td>0.15(±0.34)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.15(±0.34)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>369.33(±862.58)</td>\n",
       "      <td>369.33(±862.58)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>369.33(±862.58)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm        L90             NA50  \\\n",
       "Assembler                                                                       \n",
       "BCALM2               Single k-mer De Bruijn graph  0.0(±0.0)        0.0(±0.0)   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph  0.0(±0.0)    397.0(±927.2)   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  0.0(±0.0)  385.83(±901.11)   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph  0.0(±0.0)  392.83(±917.46)   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph  0.0(±0.0)   378.5(±883.99)   \n",
       "MINIA                Single k-mer De Bruijn graph  0.0(±0.0)        0.0(±0.0)   \n",
       "SKESA              Multiple k-mer De Bruijn graph  0.0(±0.0)  334.17(±780.45)   \n",
       "SPAdes             Multiple k-mer De Bruijn graph  0.0(±0.0)  326.83(±763.32)   \n",
       "Unicycler          Multiple k-mer De Bruijn graph  0.0(±0.0)  369.33(±862.58)   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph  0.0(±0.0)        0.0(±0.0)   \n",
       "\n",
       "                              NG50         Ns         Type        basepairs  \\\n",
       "Assembler                                                                     \n",
       "BCALM2                   0.0(±0.0)  0.0(±0.0)      Genomic        0.0(±0.0)   \n",
       "GATBMiniaPipeline    397.0(±927.2)  0.0(±0.0)  Metagenomic  369.33(±862.58)   \n",
       "IDBA-UD            385.83(±901.11)  0.0(±0.0)  Metagenomic  369.33(±862.58)   \n",
       "MEGAHIT            392.83(±917.46)  0.0(±0.0)  Metagenomic  369.33(±862.58)   \n",
       "metaSPAdes          378.5(±883.99)  0.0(±0.0)  Metagenomic  369.33(±862.58)   \n",
       "MINIA                    0.0(±0.0)  0.0(±0.0)      Genomic        0.0(±0.0)   \n",
       "SKESA              334.17(±780.45)  0.0(±0.0)      Genomic  334.17(±780.45)   \n",
       "SPAdes             326.83(±763.32)  0.0(±0.0)      Genomic  326.83(±763.32)   \n",
       "Unicycler          369.33(±862.58)  0.0(±0.0)      Genomic  369.33(±862.58)   \n",
       "VelvetOptimizer          0.0(±0.0)  0.0(±0.0)      Genomic        0.0(±0.0)   \n",
       "\n",
       "                  breadth_of_coverage      contigs   contiguity     identity  \\\n",
       "Assembler                                                                      \n",
       "BCALM2                      0.0(±0.0)    0.0(±0.0)    0.0(±0.0)    0.0(±0.0)   \n",
       "GATBMiniaPipeline         0.17(±0.39)  0.17(±0.39)  0.18(±0.42)  0.17(±0.39)   \n",
       "IDBA-UD                   0.17(±0.39)  0.17(±0.39)  0.17(±0.41)  0.17(±0.39)   \n",
       "MEGAHIT                   0.17(±0.39)  0.17(±0.39)  0.18(±0.41)  0.17(±0.39)   \n",
       "metaSPAdes                0.17(±0.39)  0.17(±0.39)   0.17(±0.4)  0.17(±0.39)   \n",
       "MINIA                       0.0(±0.0)    0.0(±0.0)    0.0(±0.0)    0.0(±0.0)   \n",
       "SKESA                     0.15(±0.35)  0.17(±0.39)  0.15(±0.35)  0.17(±0.39)   \n",
       "SPAdes                    0.15(±0.34)  0.17(±0.39)  0.15(±0.34)  0.17(±0.39)   \n",
       "Unicycler                 0.17(±0.39)  0.17(±0.39)  0.17(±0.39)  0.17(±0.39)   \n",
       "VelvetOptimizer             0.0(±0.0)    0.0(±0.0)    0.0(±0.0)    0.0(±0.0)   \n",
       "\n",
       "                  lowest_identity misassemblies multiplicity    parsimony  \\\n",
       "Assembler                                                                   \n",
       "BCALM2                  0.0(±0.0)     0.0(±0.0)    0.0(±0.0)    0.0(±0.0)   \n",
       "GATBMiniaPipeline     0.17(±0.39)     0.0(±0.0)  0.18(±0.42)  0.18(±0.42)   \n",
       "IDBA-UD               0.17(±0.39)     0.0(±0.0)  0.17(±0.41)  0.17(±0.41)   \n",
       "MEGAHIT               0.17(±0.39)     0.0(±0.0)  0.18(±0.41)  0.18(±0.41)   \n",
       "metaSPAdes            0.17(±0.39)     0.0(±0.0)   0.17(±0.4)   0.17(±0.4)   \n",
       "MINIA                   0.0(±0.0)     0.0(±0.0)    0.0(±0.0)    0.0(±0.0)   \n",
       "SKESA                 0.17(±0.39)     0.0(±0.0)  0.17(±0.39)  0.17(±0.39)   \n",
       "SPAdes                0.17(±0.39)     0.0(±0.0)  0.17(±0.39)  0.17(±0.39)   \n",
       "Unicycler             0.17(±0.39)     0.0(±0.0)  0.17(±0.39)  0.17(±0.39)   \n",
       "VelvetOptimizer         0.0(±0.0)     0.0(±0.0)    0.0(±0.0)    0.0(±0.0)   \n",
       "\n",
       "                      validity  \n",
       "Assembler                       \n",
       "BCALM2               0.0(±0.0)  \n",
       "GATBMiniaPipeline  0.17(±0.39)  \n",
       "IDBA-UD            0.17(±0.39)  \n",
       "MEGAHIT            0.17(±0.39)  \n",
       "metaSPAdes         0.17(±0.39)  \n",
       "MINIA                0.0(±0.0)  \n",
       "SKESA              0.17(±0.39)  \n",
       "SPAdes             0.17(±0.39)  \n",
       "Unicycler          0.17(±0.39)  \n",
       "VelvetOptimizer      0.0(±0.0)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Staphylococcus_aureus_plasmid3\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>525.83(±1228.08)</td>\n",
       "      <td>525.83(±1228.08)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>498.83(±1165.02)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.18(±0.41)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.18(±0.41)</td>\n",
       "      <td>0.18(±0.41)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>281.33(±657.05)</td>\n",
       "      <td>281.33(±657.05)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>459.5(±1073.16)</td>\n",
       "      <td>0.15(±0.36)</td>\n",
       "      <td>0.33(±0.78)</td>\n",
       "      <td>0.09(±0.22)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.17(±0.4)</td>\n",
       "      <td>0.17(±0.4)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>522.33(±1219.91)</td>\n",
       "      <td>522.33(±1219.91)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>498.83(±1165.02)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.41)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.17(±0.41)</td>\n",
       "      <td>0.17(±0.41)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>642.83(±1501.34)</td>\n",
       "      <td>642.83(±1501.34)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>498.83(±1165.02)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.21(±0.5)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.21(±0.5)</td>\n",
       "      <td>0.21(±0.5)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>463.67(±1082.89)</td>\n",
       "      <td>463.67(±1082.89)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>463.67(±1082.89)</td>\n",
       "      <td>0.15(±0.36)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.15(±0.36)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>498.83(±1165.02)</td>\n",
       "      <td>498.83(±1165.02)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>498.83(±1165.02)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "      <td>0.17(±0.39)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm          L90  \\\n",
       "Assembler                                                        \n",
       "BCALM2               Single k-mer De Bruijn graph    0.0(±0.0)   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph    0.0(±0.0)   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph  0.17(±0.39)   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph    0.0(±0.0)   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph    0.0(±0.0)   \n",
       "MINIA                Single k-mer De Bruijn graph    0.0(±0.0)   \n",
       "SKESA              Multiple k-mer De Bruijn graph    0.0(±0.0)   \n",
       "SPAdes             Multiple k-mer De Bruijn graph    0.0(±0.0)   \n",
       "Unicycler          Multiple k-mer De Bruijn graph    0.0(±0.0)   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph    0.0(±0.0)   \n",
       "\n",
       "                               NA50              NG50         Ns         Type  \\\n",
       "Assembler                                                                       \n",
       "BCALM2                    0.0(±0.0)         0.0(±0.0)  0.0(±0.0)      Genomic   \n",
       "GATBMiniaPipeline  525.83(±1228.08)  525.83(±1228.08)  0.0(±0.0)  Metagenomic   \n",
       "IDBA-UD             281.33(±657.05)   281.33(±657.05)  0.0(±0.0)  Metagenomic   \n",
       "MEGAHIT            522.33(±1219.91)  522.33(±1219.91)  0.0(±0.0)  Metagenomic   \n",
       "metaSPAdes         642.83(±1501.34)  642.83(±1501.34)  0.0(±0.0)  Metagenomic   \n",
       "MINIA                     0.0(±0.0)         0.0(±0.0)  0.0(±0.0)      Genomic   \n",
       "SKESA              463.67(±1082.89)  463.67(±1082.89)  0.0(±0.0)      Genomic   \n",
       "SPAdes                    0.0(±0.0)         0.0(±0.0)  0.0(±0.0)      Genomic   \n",
       "Unicycler          498.83(±1165.02)  498.83(±1165.02)  0.0(±0.0)      Genomic   \n",
       "VelvetOptimizer           0.0(±0.0)         0.0(±0.0)  0.0(±0.0)      Genomic   \n",
       "\n",
       "                          basepairs breadth_of_coverage      contigs  \\\n",
       "Assembler                                                              \n",
       "BCALM2                    0.0(±0.0)           0.0(±0.0)    0.0(±0.0)   \n",
       "GATBMiniaPipeline  498.83(±1165.02)         0.17(±0.39)  0.17(±0.39)   \n",
       "IDBA-UD             459.5(±1073.16)         0.15(±0.36)  0.33(±0.78)   \n",
       "MEGAHIT            498.83(±1165.02)         0.17(±0.39)  0.17(±0.39)   \n",
       "metaSPAdes         498.83(±1165.02)         0.17(±0.39)  0.17(±0.39)   \n",
       "MINIA                     0.0(±0.0)           0.0(±0.0)    0.0(±0.0)   \n",
       "SKESA              463.67(±1082.89)         0.15(±0.36)  0.17(±0.39)   \n",
       "SPAdes                    0.0(±0.0)           0.0(±0.0)    0.0(±0.0)   \n",
       "Unicycler          498.83(±1165.02)         0.17(±0.39)  0.17(±0.39)   \n",
       "VelvetOptimizer           0.0(±0.0)           0.0(±0.0)    0.0(±0.0)   \n",
       "\n",
       "                    contiguity     identity lowest_identity misassemblies  \\\n",
       "Assembler                                                                   \n",
       "BCALM2               0.0(±0.0)    0.0(±0.0)       0.0(±0.0)     0.0(±0.0)   \n",
       "GATBMiniaPipeline  0.18(±0.41)  0.17(±0.39)     0.17(±0.39)     0.0(±0.0)   \n",
       "IDBA-UD            0.09(±0.22)  0.17(±0.39)     0.17(±0.39)     0.0(±0.0)   \n",
       "MEGAHIT            0.17(±0.41)  0.17(±0.39)     0.17(±0.39)     0.0(±0.0)   \n",
       "metaSPAdes          0.21(±0.5)  0.17(±0.39)     0.17(±0.39)     0.0(±0.0)   \n",
       "MINIA                0.0(±0.0)    0.0(±0.0)       0.0(±0.0)     0.0(±0.0)   \n",
       "SKESA              0.15(±0.36)  0.17(±0.39)     0.17(±0.39)     0.0(±0.0)   \n",
       "SPAdes               0.0(±0.0)    0.0(±0.0)       0.0(±0.0)     0.0(±0.0)   \n",
       "Unicycler          0.17(±0.39)  0.17(±0.39)     0.17(±0.39)     0.0(±0.0)   \n",
       "VelvetOptimizer      0.0(±0.0)    0.0(±0.0)       0.0(±0.0)     0.0(±0.0)   \n",
       "\n",
       "                  multiplicity    parsimony     validity  \n",
       "Assembler                                                 \n",
       "BCALM2               0.0(±0.0)    0.0(±0.0)    0.0(±0.0)  \n",
       "GATBMiniaPipeline  0.18(±0.41)  0.18(±0.41)  0.17(±0.39)  \n",
       "IDBA-UD             0.17(±0.4)   0.17(±0.4)  0.17(±0.39)  \n",
       "MEGAHIT            0.17(±0.41)  0.17(±0.41)  0.17(±0.39)  \n",
       "metaSPAdes          0.21(±0.5)   0.21(±0.5)  0.17(±0.39)  \n",
       "MINIA                0.0(±0.0)    0.0(±0.0)    0.0(±0.0)  \n",
       "SKESA              0.17(±0.39)  0.17(±0.39)  0.17(±0.39)  \n",
       "SPAdes               0.0(±0.0)    0.0(±0.0)    0.0(±0.0)  \n",
       "Unicycler          0.17(±0.39)  0.17(±0.39)  0.17(±0.39)  \n",
       "VelvetOptimizer      0.0(±0.0)    0.0(±0.0)    0.0(±0.0)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "for reference in reference_pipeline_metrics_df.reference.unique():\n",
    "    ref_stats_per_reference_per_assembler = pd.DataFrame()\n",
    "    for assembler in reference_pipeline_metrics_df.assembler.unique():\n",
    "        row={'Assembler': assembler, \n",
    "             'Type':reference_pipeline_metrics_df[reference_pipeline_metrics_df.assembler == assembler].type.unique()[0],\n",
    "             'Algorithm': reference_pipeline_metrics_df[reference_pipeline_metrics_df.assembler == assembler].algorythm.unique()[0]}\n",
    "        lala=reference_pipeline_metrics_df[(reference_pipeline_metrics_df.assembler == assembler) & (reference_pipeline_metrics_df.reference == reference)].describe()\n",
    "        for column in lala.columns:\n",
    "            mean = lala.loc['mean',column]\n",
    "            std = lala.loc['std',column]\n",
    "            row[column] = \"{}(±{})\".format(round(mean, 2), round(std, 2))\n",
    "        ref_stats_per_reference_per_assembler=ref_stats_per_reference_per_assembler.append(row, ignore_index=True)\n",
    "    ref_stats_per_reference_per_assembler = ref_stats_per_reference_per_assembler.set_index('Assembler')\n",
    "    ref_stats_per_reference_per_assembler.to_csv(\"Tables/Results/{} - Reference metrics per assembler.csv\".format(reference))\n",
    "    print(reference)\n",
    "    display(ref_stats_per_reference_per_assembler)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reference stats per distribution"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Even\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>74.93(±300.64)</td>\n",
       "      <td>7322.22(±14758.07)</td>\n",
       "      <td>6909.82(±14651.4)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1290698.65(±1827654.8)</td>\n",
       "      <td>0.37(±0.42)</td>\n",
       "      <td>342.06(±544.94)</td>\n",
       "      <td>0.04(±0.17)</td>\n",
       "      <td>0.5(±0.5)</td>\n",
       "      <td>0.5(±0.5)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.5(±0.5)</td>\n",
       "      <td>0.5(±0.5)</td>\n",
       "      <td>0.5(±0.5)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>29.53(±32.8)</td>\n",
       "      <td>109767.99(±118390.68)</td>\n",
       "      <td>100966.54(±103290.32)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2524984.11(±2142615.69)</td>\n",
       "      <td>0.87(±0.31)</td>\n",
       "      <td>57.58(±62.96)</td>\n",
       "      <td>0.25(±0.34)</td>\n",
       "      <td>0.89(±0.32)</td>\n",
       "      <td>0.86(±0.32)</td>\n",
       "      <td>0.25(±0.64)</td>\n",
       "      <td>0.89(±0.32)</td>\n",
       "      <td>0.89(±0.32)</td>\n",
       "      <td>0.89(±0.32)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>71.28(±84.72)</td>\n",
       "      <td>56036.0(±73769.04)</td>\n",
       "      <td>55774.38(±73629.02)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2508176.47(±2128731.97)</td>\n",
       "      <td>0.83(±0.34)</td>\n",
       "      <td>120.25(±138.91)</td>\n",
       "      <td>0.16(±0.27)</td>\n",
       "      <td>0.86(±0.35)</td>\n",
       "      <td>0.75(±0.39)</td>\n",
       "      <td>1.0(±2.14)</td>\n",
       "      <td>0.87(±0.35)</td>\n",
       "      <td>0.87(±0.35)</td>\n",
       "      <td>0.86(±0.35)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>17.47(±17.88)</td>\n",
       "      <td>144475.11(±141019.39)</td>\n",
       "      <td>151952.72(±144606.65)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2533624.76(±2150186.44)</td>\n",
       "      <td>0.88(±0.31)</td>\n",
       "      <td>44.06(±44.47)</td>\n",
       "      <td>0.3(±0.37)</td>\n",
       "      <td>0.89(±0.32)</td>\n",
       "      <td>0.79(±0.35)</td>\n",
       "      <td>1.08(±1.47)</td>\n",
       "      <td>0.89(±0.32)</td>\n",
       "      <td>0.9(±0.32)</td>\n",
       "      <td>0.89(±0.32)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>19.92(±22.9)</td>\n",
       "      <td>134941.08(±125682.7)</td>\n",
       "      <td>133991.47(±125319.33)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>2529572.39(±2148555.68)</td>\n",
       "      <td>0.87(±0.31)</td>\n",
       "      <td>38.94(±42.57)</td>\n",
       "      <td>0.28(±0.34)</td>\n",
       "      <td>0.89(±0.32)</td>\n",
       "      <td>0.88(±0.31)</td>\n",
       "      <td>0.17(±0.56)</td>\n",
       "      <td>0.9(±0.32)</td>\n",
       "      <td>0.9(±0.32)</td>\n",
       "      <td>0.89(±0.32)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>82.97(±110.03)</td>\n",
       "      <td>18331.39(±18137.17)</td>\n",
       "      <td>17666.15(±17998.82)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2345476.71(±1982106.28)</td>\n",
       "      <td>0.72(±0.38)</td>\n",
       "      <td>306.47(±391.64)</td>\n",
       "      <td>0.08(±0.18)</td>\n",
       "      <td>0.81(±0.4)</td>\n",
       "      <td>0.79(±0.39)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.81(±0.4)</td>\n",
       "      <td>0.81(±0.4)</td>\n",
       "      <td>0.81(±0.4)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>38.81(±44.56)</td>\n",
       "      <td>84178.14(±88805.36)</td>\n",
       "      <td>83571.72(±89085.63)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2505379.86(±2127723.39)</td>\n",
       "      <td>0.85(±0.31)</td>\n",
       "      <td>69.67(±78.58)</td>\n",
       "      <td>0.23(±0.34)</td>\n",
       "      <td>0.89(±0.32)</td>\n",
       "      <td>0.77(±0.38)</td>\n",
       "      <td>0.33(±0.75)</td>\n",
       "      <td>0.89(±0.32)</td>\n",
       "      <td>0.89(±0.32)</td>\n",
       "      <td>0.88(±0.32)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>19.72(±20.8)</td>\n",
       "      <td>129438.44(±125555.2)</td>\n",
       "      <td>129237.0(±125128.06)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2524523.92(±2144608.71)</td>\n",
       "      <td>0.81(±0.37)</td>\n",
       "      <td>39.03(±39.85)</td>\n",
       "      <td>0.21(±0.26)</td>\n",
       "      <td>0.83(±0.38)</td>\n",
       "      <td>0.8(±0.38)</td>\n",
       "      <td>0.39(±1.01)</td>\n",
       "      <td>0.83(±0.38)</td>\n",
       "      <td>0.84(±0.38)</td>\n",
       "      <td>0.83(±0.38)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>15.64(±17.3)</td>\n",
       "      <td>161424.75(±141031.02)</td>\n",
       "      <td>158207.14(±140318.23)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>2521523.86(±2142658.7)</td>\n",
       "      <td>0.87(±0.31)</td>\n",
       "      <td>32.81(±33.31)</td>\n",
       "      <td>0.29(±0.32)</td>\n",
       "      <td>0.89(±0.32)</td>\n",
       "      <td>0.86(±0.33)</td>\n",
       "      <td>0.17(±0.56)</td>\n",
       "      <td>0.89(±0.32)</td>\n",
       "      <td>0.89(±0.32)</td>\n",
       "      <td>0.89(±0.32)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>31.03(±55.18)</td>\n",
       "      <td>28318.97(±64479.6)</td>\n",
       "      <td>33659.58(±87778.85)</td>\n",
       "      <td>977.56(±2628.96)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1609811.42(±2033999.9)</td>\n",
       "      <td>0.5(±0.48)</td>\n",
       "      <td>170.75(±339.91)</td>\n",
       "      <td>0.07(±0.18)</td>\n",
       "      <td>0.53(±0.5)</td>\n",
       "      <td>0.5(±0.48)</td>\n",
       "      <td>0.94(±2.28)</td>\n",
       "      <td>0.53(±0.5)</td>\n",
       "      <td>0.53(±0.5)</td>\n",
       "      <td>0.53(±0.5)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm             L90  \\\n",
       "Assembler                                                           \n",
       "BCALM2               Single k-mer De Bruijn graph  74.93(±300.64)   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph    29.53(±32.8)   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph   71.28(±84.72)   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph   17.47(±17.88)   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph    19.92(±22.9)   \n",
       "MINIA                Single k-mer De Bruijn graph  82.97(±110.03)   \n",
       "SKESA              Multiple k-mer De Bruijn graph   38.81(±44.56)   \n",
       "SPAdes             Multiple k-mer De Bruijn graph    19.72(±20.8)   \n",
       "Unicycler          Multiple k-mer De Bruijn graph    15.64(±17.3)   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph   31.03(±55.18)   \n",
       "\n",
       "                                    NA50                   NG50  \\\n",
       "Assembler                                                         \n",
       "BCALM2                7322.22(±14758.07)      6909.82(±14651.4)   \n",
       "GATBMiniaPipeline  109767.99(±118390.68)  100966.54(±103290.32)   \n",
       "IDBA-UD               56036.0(±73769.04)    55774.38(±73629.02)   \n",
       "MEGAHIT            144475.11(±141019.39)  151952.72(±144606.65)   \n",
       "metaSPAdes          134941.08(±125682.7)  133991.47(±125319.33)   \n",
       "MINIA                18331.39(±18137.17)    17666.15(±17998.82)   \n",
       "SKESA                84178.14(±88805.36)    83571.72(±89085.63)   \n",
       "SPAdes              129438.44(±125555.2)   129237.0(±125128.06)   \n",
       "Unicycler          161424.75(±141031.02)  158207.14(±140318.23)   \n",
       "VelvetOptimizer       28318.97(±64479.6)    33659.58(±87778.85)   \n",
       "\n",
       "                                 Ns         Type                basepairs  \\\n",
       "Assembler                                                                   \n",
       "BCALM2                    0.0(±0.0)      Genomic   1290698.65(±1827654.8)   \n",
       "GATBMiniaPipeline         0.0(±0.0)  Metagenomic  2524984.11(±2142615.69)   \n",
       "IDBA-UD                   0.0(±0.0)  Metagenomic  2508176.47(±2128731.97)   \n",
       "MEGAHIT                   0.0(±0.0)  Metagenomic  2533624.76(±2150186.44)   \n",
       "metaSPAdes                0.0(±0.0)  Metagenomic  2529572.39(±2148555.68)   \n",
       "MINIA                     0.0(±0.0)      Genomic  2345476.71(±1982106.28)   \n",
       "SKESA                     0.0(±0.0)      Genomic  2505379.86(±2127723.39)   \n",
       "SPAdes                    0.0(±0.0)      Genomic  2524523.92(±2144608.71)   \n",
       "Unicycler                 0.0(±0.0)      Genomic   2521523.86(±2142658.7)   \n",
       "VelvetOptimizer    977.56(±2628.96)      Genomic   1609811.42(±2033999.9)   \n",
       "\n",
       "                  breadth_of_coverage          contigs   contiguity  \\\n",
       "Assembler                                                             \n",
       "BCALM2                    0.37(±0.42)  342.06(±544.94)  0.04(±0.17)   \n",
       "GATBMiniaPipeline         0.87(±0.31)    57.58(±62.96)  0.25(±0.34)   \n",
       "IDBA-UD                   0.83(±0.34)  120.25(±138.91)  0.16(±0.27)   \n",
       "MEGAHIT                   0.88(±0.31)    44.06(±44.47)   0.3(±0.37)   \n",
       "metaSPAdes                0.87(±0.31)    38.94(±42.57)  0.28(±0.34)   \n",
       "MINIA                     0.72(±0.38)  306.47(±391.64)  0.08(±0.18)   \n",
       "SKESA                     0.85(±0.31)    69.67(±78.58)  0.23(±0.34)   \n",
       "SPAdes                    0.81(±0.37)    39.03(±39.85)  0.21(±0.26)   \n",
       "Unicycler                 0.87(±0.31)    32.81(±33.31)  0.29(±0.32)   \n",
       "VelvetOptimizer            0.5(±0.48)  170.75(±339.91)  0.07(±0.18)   \n",
       "\n",
       "                      identity lowest_identity misassemblies multiplicity  \\\n",
       "Assembler                                                                   \n",
       "BCALM2               0.5(±0.5)       0.5(±0.5)     0.0(±0.0)    0.5(±0.5)   \n",
       "GATBMiniaPipeline  0.89(±0.32)     0.86(±0.32)   0.25(±0.64)  0.89(±0.32)   \n",
       "IDBA-UD            0.86(±0.35)     0.75(±0.39)    1.0(±2.14)  0.87(±0.35)   \n",
       "MEGAHIT            0.89(±0.32)     0.79(±0.35)   1.08(±1.47)  0.89(±0.32)   \n",
       "metaSPAdes         0.89(±0.32)     0.88(±0.31)   0.17(±0.56)   0.9(±0.32)   \n",
       "MINIA               0.81(±0.4)     0.79(±0.39)     0.0(±0.0)   0.81(±0.4)   \n",
       "SKESA              0.89(±0.32)     0.77(±0.38)   0.33(±0.75)  0.89(±0.32)   \n",
       "SPAdes             0.83(±0.38)      0.8(±0.38)   0.39(±1.01)  0.83(±0.38)   \n",
       "Unicycler          0.89(±0.32)     0.86(±0.33)   0.17(±0.56)  0.89(±0.32)   \n",
       "VelvetOptimizer     0.53(±0.5)      0.5(±0.48)   0.94(±2.28)   0.53(±0.5)   \n",
       "\n",
       "                     parsimony     validity  \n",
       "Assembler                                    \n",
       "BCALM2               0.5(±0.5)    0.5(±0.5)  \n",
       "GATBMiniaPipeline  0.89(±0.32)  0.89(±0.32)  \n",
       "IDBA-UD            0.87(±0.35)  0.86(±0.35)  \n",
       "MEGAHIT             0.9(±0.32)  0.89(±0.32)  \n",
       "metaSPAdes          0.9(±0.32)  0.89(±0.32)  \n",
       "MINIA               0.81(±0.4)   0.81(±0.4)  \n",
       "SKESA              0.89(±0.32)  0.88(±0.32)  \n",
       "SPAdes             0.84(±0.38)  0.83(±0.38)  \n",
       "Unicycler          0.89(±0.32)  0.89(±0.32)  \n",
       "VelvetOptimizer     0.53(±0.5)   0.53(±0.5)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Algorithm</th>\n",
       "      <th>L90</th>\n",
       "      <th>NA50</th>\n",
       "      <th>NG50</th>\n",
       "      <th>Ns</th>\n",
       "      <th>Type</th>\n",
       "      <th>basepairs</th>\n",
       "      <th>breadth_of_coverage</th>\n",
       "      <th>contigs</th>\n",
       "      <th>contiguity</th>\n",
       "      <th>identity</th>\n",
       "      <th>lowest_identity</th>\n",
       "      <th>misassemblies</th>\n",
       "      <th>multiplicity</th>\n",
       "      <th>parsimony</th>\n",
       "      <th>validity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Assembler</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCALM2</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>13.58(±55.44)</td>\n",
       "      <td>3455.14(±12312.8)</td>\n",
       "      <td>3263.39(±11966.56)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>390195.64(±1316359.92)</td>\n",
       "      <td>0.09(±0.27)</td>\n",
       "      <td>35.86(±117.43)</td>\n",
       "      <td>0.0(±0.01)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.37)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "      <td>0.17(±0.38)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GATBMiniaPipeline</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>23.54(±78.96)</td>\n",
       "      <td>41255.35(±95190.78)</td>\n",
       "      <td>40885.01(±95133.88)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1131017.26(±2125360.46)</td>\n",
       "      <td>0.25(±0.43)</td>\n",
       "      <td>37.6(±108.12)</td>\n",
       "      <td>0.02(±0.06)</td>\n",
       "      <td>0.36(±0.48)</td>\n",
       "      <td>0.36(±0.48)</td>\n",
       "      <td>0.1(±0.38)</td>\n",
       "      <td>0.36(±0.48)</td>\n",
       "      <td>0.36(±0.48)</td>\n",
       "      <td>0.36(±0.48)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDBA-UD</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>16.21(±49.86)</td>\n",
       "      <td>48013.69(±107525.28)</td>\n",
       "      <td>46436.22(±107580.39)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1135517.22(±2132472.67)</td>\n",
       "      <td>0.25(±0.43)</td>\n",
       "      <td>29.56(±76.51)</td>\n",
       "      <td>0.02(±0.06)</td>\n",
       "      <td>0.46(±0.49)</td>\n",
       "      <td>0.4(±0.47)</td>\n",
       "      <td>0.28(±0.65)</td>\n",
       "      <td>0.47(±0.5)</td>\n",
       "      <td>0.61(±0.99)</td>\n",
       "      <td>0.44(±0.49)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEGAHIT</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>8.42(±25.44)</td>\n",
       "      <td>66599.11(±136763.09)</td>\n",
       "      <td>66455.53(±137159.35)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1141251.53(±2142253.99)</td>\n",
       "      <td>0.25(±0.43)</td>\n",
       "      <td>18.11(±41.95)</td>\n",
       "      <td>0.03(±0.08)</td>\n",
       "      <td>0.47(±0.5)</td>\n",
       "      <td>0.43(±0.47)</td>\n",
       "      <td>0.06(±0.23)</td>\n",
       "      <td>0.47(±0.5)</td>\n",
       "      <td>0.48(±0.51)</td>\n",
       "      <td>0.47(±0.5)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaSPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>7.19(±15.98)</td>\n",
       "      <td>60717.78(±131337.07)</td>\n",
       "      <td>59453.89(±130049.83)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Metagenomic</td>\n",
       "      <td>1140704.31(±2141201.79)</td>\n",
       "      <td>0.25(±0.43)</td>\n",
       "      <td>15.53(±29.28)</td>\n",
       "      <td>0.03(±0.07)</td>\n",
       "      <td>0.47(±0.5)</td>\n",
       "      <td>0.43(±0.48)</td>\n",
       "      <td>0.17(±0.44)</td>\n",
       "      <td>0.47(±0.5)</td>\n",
       "      <td>0.48(±0.51)</td>\n",
       "      <td>0.47(±0.5)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MINIA</th>\n",
       "      <td>Single k-mer De Bruijn graph</td>\n",
       "      <td>65.18(±150.35)</td>\n",
       "      <td>6392.81(±13423.27)</td>\n",
       "      <td>5787.25(±13451.03)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1099741.54(±2072424.11)</td>\n",
       "      <td>0.24(±0.41)</td>\n",
       "      <td>119.46(±243.01)</td>\n",
       "      <td>0.01(±0.01)</td>\n",
       "      <td>0.42(±0.5)</td>\n",
       "      <td>0.39(±0.49)</td>\n",
       "      <td>0.14(±0.48)</td>\n",
       "      <td>0.42(±0.5)</td>\n",
       "      <td>0.42(±0.5)</td>\n",
       "      <td>0.42(±0.5)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SKESA</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>6.72(±38.38)</td>\n",
       "      <td>10421.97(±58025.85)</td>\n",
       "      <td>10210.64(±58056.78)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>366940.28(±950611.99)</td>\n",
       "      <td>0.1(±0.26)</td>\n",
       "      <td>120.94(±388.81)</td>\n",
       "      <td>0.01(±0.03)</td>\n",
       "      <td>0.19(±0.4)</td>\n",
       "      <td>0.19(±0.39)</td>\n",
       "      <td>0.33(±1.99)</td>\n",
       "      <td>0.19(±0.4)</td>\n",
       "      <td>0.19(±0.4)</td>\n",
       "      <td>0.19(±0.4)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAdes</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>30.92(±138.63)</td>\n",
       "      <td>36829.5(±85995.88)</td>\n",
       "      <td>37693.47(±90368.89)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>1093453.36(±2101398.69)</td>\n",
       "      <td>0.23(±0.41)</td>\n",
       "      <td>105.28(±363.93)</td>\n",
       "      <td>0.02(±0.05)</td>\n",
       "      <td>0.56(±0.48)</td>\n",
       "      <td>0.49(±0.47)</td>\n",
       "      <td>16.17(±93.6)</td>\n",
       "      <td>0.6(±0.52)</td>\n",
       "      <td>0.63(±0.55)</td>\n",
       "      <td>0.57(±0.48)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unicycler</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>7.72(±24.6)</td>\n",
       "      <td>48388.56(±108092.74)</td>\n",
       "      <td>45410.11(±107655.24)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>963355.97(±2016669.48)</td>\n",
       "      <td>0.2(±0.39)</td>\n",
       "      <td>14.22(±35.63)</td>\n",
       "      <td>0.03(±0.06)</td>\n",
       "      <td>0.22(±0.42)</td>\n",
       "      <td>0.22(±0.41)</td>\n",
       "      <td>0.08(±0.5)</td>\n",
       "      <td>0.22(±0.42)</td>\n",
       "      <td>0.22(±0.42)</td>\n",
       "      <td>0.22(±0.42)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VelvetOptimizer</th>\n",
       "      <td>Multiple k-mer De Bruijn graph</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>Genomic</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "      <td>0.0(±0.0)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Algorithm             L90  \\\n",
       "Assembler                                                           \n",
       "BCALM2               Single k-mer De Bruijn graph   13.58(±55.44)   \n",
       "GATBMiniaPipeline  Multiple k-mer De Bruijn graph   23.54(±78.96)   \n",
       "IDBA-UD            Multiple k-mer De Bruijn graph   16.21(±49.86)   \n",
       "MEGAHIT            Multiple k-mer De Bruijn graph    8.42(±25.44)   \n",
       "metaSPAdes         Multiple k-mer De Bruijn graph    7.19(±15.98)   \n",
       "MINIA                Single k-mer De Bruijn graph  65.18(±150.35)   \n",
       "SKESA              Multiple k-mer De Bruijn graph    6.72(±38.38)   \n",
       "SPAdes             Multiple k-mer De Bruijn graph  30.92(±138.63)   \n",
       "Unicycler          Multiple k-mer De Bruijn graph     7.72(±24.6)   \n",
       "VelvetOptimizer    Multiple k-mer De Bruijn graph       0.0(±0.0)   \n",
       "\n",
       "                                   NA50                  NG50         Ns  \\\n",
       "Assembler                                                                  \n",
       "BCALM2                3455.14(±12312.8)    3263.39(±11966.56)  0.0(±0.0)   \n",
       "GATBMiniaPipeline   41255.35(±95190.78)   40885.01(±95133.88)  0.0(±0.0)   \n",
       "IDBA-UD            48013.69(±107525.28)  46436.22(±107580.39)  0.0(±0.0)   \n",
       "MEGAHIT            66599.11(±136763.09)  66455.53(±137159.35)  0.0(±0.0)   \n",
       "metaSPAdes         60717.78(±131337.07)  59453.89(±130049.83)  0.0(±0.0)   \n",
       "MINIA                6392.81(±13423.27)    5787.25(±13451.03)  0.0(±0.0)   \n",
       "SKESA               10421.97(±58025.85)   10210.64(±58056.78)  0.0(±0.0)   \n",
       "SPAdes               36829.5(±85995.88)   37693.47(±90368.89)  0.0(±0.0)   \n",
       "Unicycler          48388.56(±108092.74)  45410.11(±107655.24)  0.0(±0.0)   \n",
       "VelvetOptimizer               0.0(±0.0)             0.0(±0.0)  0.0(±0.0)   \n",
       "\n",
       "                          Type                basepairs breadth_of_coverage  \\\n",
       "Assembler                                                                     \n",
       "BCALM2                 Genomic   390195.64(±1316359.92)         0.09(±0.27)   \n",
       "GATBMiniaPipeline  Metagenomic  1131017.26(±2125360.46)         0.25(±0.43)   \n",
       "IDBA-UD            Metagenomic  1135517.22(±2132472.67)         0.25(±0.43)   \n",
       "MEGAHIT            Metagenomic  1141251.53(±2142253.99)         0.25(±0.43)   \n",
       "metaSPAdes         Metagenomic  1140704.31(±2141201.79)         0.25(±0.43)   \n",
       "MINIA                  Genomic  1099741.54(±2072424.11)         0.24(±0.41)   \n",
       "SKESA                  Genomic    366940.28(±950611.99)          0.1(±0.26)   \n",
       "SPAdes                 Genomic  1093453.36(±2101398.69)         0.23(±0.41)   \n",
       "Unicycler              Genomic   963355.97(±2016669.48)          0.2(±0.39)   \n",
       "VelvetOptimizer        Genomic                0.0(±0.0)           0.0(±0.0)   \n",
       "\n",
       "                           contigs   contiguity     identity lowest_identity  \\\n",
       "Assembler                                                                      \n",
       "BCALM2              35.86(±117.43)   0.0(±0.01)  0.17(±0.38)     0.17(±0.37)   \n",
       "GATBMiniaPipeline    37.6(±108.12)  0.02(±0.06)  0.36(±0.48)     0.36(±0.48)   \n",
       "IDBA-UD              29.56(±76.51)  0.02(±0.06)  0.46(±0.49)      0.4(±0.47)   \n",
       "MEGAHIT              18.11(±41.95)  0.03(±0.08)   0.47(±0.5)     0.43(±0.47)   \n",
       "metaSPAdes           15.53(±29.28)  0.03(±0.07)   0.47(±0.5)     0.43(±0.48)   \n",
       "MINIA              119.46(±243.01)  0.01(±0.01)   0.42(±0.5)     0.39(±0.49)   \n",
       "SKESA              120.94(±388.81)  0.01(±0.03)   0.19(±0.4)     0.19(±0.39)   \n",
       "SPAdes             105.28(±363.93)  0.02(±0.05)  0.56(±0.48)     0.49(±0.47)   \n",
       "Unicycler            14.22(±35.63)  0.03(±0.06)  0.22(±0.42)     0.22(±0.41)   \n",
       "VelvetOptimizer          0.0(±0.0)    0.0(±0.0)    0.0(±0.0)       0.0(±0.0)   \n",
       "\n",
       "                  misassemblies multiplicity    parsimony     validity  \n",
       "Assembler                                                               \n",
       "BCALM2                0.0(±0.0)  0.17(±0.38)  0.17(±0.38)  0.17(±0.38)  \n",
       "GATBMiniaPipeline    0.1(±0.38)  0.36(±0.48)  0.36(±0.48)  0.36(±0.48)  \n",
       "IDBA-UD             0.28(±0.65)   0.47(±0.5)  0.61(±0.99)  0.44(±0.49)  \n",
       "MEGAHIT             0.06(±0.23)   0.47(±0.5)  0.48(±0.51)   0.47(±0.5)  \n",
       "metaSPAdes          0.17(±0.44)   0.47(±0.5)  0.48(±0.51)   0.47(±0.5)  \n",
       "MINIA               0.14(±0.48)   0.42(±0.5)   0.42(±0.5)   0.42(±0.5)  \n",
       "SKESA               0.33(±1.99)   0.19(±0.4)   0.19(±0.4)   0.19(±0.4)  \n",
       "SPAdes             16.17(±93.6)   0.6(±0.52)  0.63(±0.55)  0.57(±0.48)  \n",
       "Unicycler            0.08(±0.5)  0.22(±0.42)  0.22(±0.42)  0.22(±0.42)  \n",
       "VelvetOptimizer       0.0(±0.0)    0.0(±0.0)    0.0(±0.0)    0.0(±0.0)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "for distribution in reference_pipeline_metrics_df.distribution.unique():\n",
    "    ref_stats_per_reference_per_assembler = pd.DataFrame()\n",
    "    for assembler in reference_pipeline_metrics_df.assembler.unique():\n",
    "        row={'Assembler': assembler, \n",
    "             'Type':reference_pipeline_metrics_df[reference_pipeline_metrics_df.assembler == assembler].type.unique()[0],\n",
    "             'Algorithm': reference_pipeline_metrics_df[reference_pipeline_metrics_df.assembler == assembler].algorythm.unique()[0]}\n",
    "        lala=reference_pipeline_metrics_df[(reference_pipeline_metrics_df.assembler == assembler) & (reference_pipeline_metrics_df.distribution == distribution)].describe()\n",
    "        for column in lala.columns:\n",
    "            mean = lala.loc['mean',column]\n",
    "            std = lala.loc['std',column]\n",
    "            row[column] = \"{}(±{})\".format(round(mean, 2), round(std, 2))\n",
    "        ref_stats_per_reference_per_assembler=ref_stats_per_reference_per_assembler.append(row, ignore_index=True)\n",
    "    ref_stats_per_reference_per_assembler = ref_stats_per_reference_per_assembler.set_index('Assembler')\n",
    "    ref_stats_per_reference_per_assembler.to_csv(\"Tables/Results/{} - Reference metrics per assembler.csv\".format(distribution))\n",
    "    print(distribution)\n",
    "    display(ref_stats_per_reference_per_assembler)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
